---
title: Assessment of single cell sequencing rapid run
author: John Blischak
output: html_document
---

`r Sys.Date()`

## Data processing

```{r setup, include=FALSE}
library("knitr")
opts_chunk$set(tidy = FALSE, fig.path = "figure/", echo = FALSE,
               warning = FALSE)
library("ggplot2")
theme_set(theme_bw(base_size = 14))
```

Po-Yuan sequenced two single cells on a rapid run flow cell.
In addition to the cellular RNA, she added the [ERCC RNA spike-in mix][ercc] separately to each cell to assess the efficiency of the single cell sequencing.
She also added in PhiX sample in order to boost the concentration.
The samples were multiplexed and the mix was sequenced on both lanes.
However, the sample barcode was not read so the data has to be analyzed jointly.

I processed the sequencing data using a pipeline which utilized Snakemake, R, umitools, Subread, and samtools.
The genome used for mapping was a combination of human genome hg19, the ERCC RNA spike-in sequences, and the PhiX genome.
The genomic features for counting were Ensembl exons from both coding and non-coding genes as well as the ERCC sequences.

For comparison, I also processed traditional RNA-sequencing on a population of cells for individual NA19239 from [Pickrell et al. 2010][Pickrell2010] (Yale data only because it was available in fastq format. The Argonne data is in map format).

[Pickrell2010]: http://www.ncbi.nlm.nih.gov/geo/query/acc.cgi?acc=GSE19480

```{r}
counts <- read.table("pipeline/data/count_matrix.txt", header = TRUE,
                     sep = "\t", stringsAsFactors = FALSE)
colnames(counts) <- c("gene", "chr", "start", "end", "strand", "length",
                      "reads", "molecules", "population")
counts$feature <- ifelse(grepl("ERCC", counts$gene), "control", "gene")
# colSums(counts[, c("reads", "molecules", "population")])
```

Here is the rough break down of the number of sequences obtained:

*  ~250 million raw reads
*  ~150 million reads with unique molecular identifier
*  ~114 million reads mapped to genome or controls
*  ~`r round(sum(counts$reads) / 10^6)` million reads mapped to human exons or control sequences
*  `r sum(counts$molecules)` unique molecules, which corresponds to `r sum(counts$molecules[counts$feature == "gene"] > 0)` genes

```{r clean}
# Remove genes with zero counts, but keep all controls
counts <- counts[counts$reads + counts$molecules + counts$population > 0 |
                 counts$feature == "control", ]
```

For the analyses below, I use only the `r sum(counts$feature == "gene")` genes with at least one molecule identified in this experiment or at least one read sequenced in [Pickrell et al. 2010][Pickrell2010] (as well as all `r sum(counts$feature == "control")` ERCC spike-ins).

```{r reads-v-molecules}
ggplot(counts, aes(x = molecules, y = reads)) + geom_point() + facet_wrap(~ feature) + geom_smooth(method = "lm") + ylim(0, 750000) + labs(title = "Read and molecule counts are highly correlated")
cor(counts$molecules, counts$reads)
```

### Plotting fold-change to mean

```{r log-fold-change-to-mean}
counts$reads_fc <- log2(counts$reads / mean(counts$reads))
counts$molecules_fc <- log2(counts$molecules / mean(counts$molecules))
ggplot(counts, aes(x = molecules_fc, y = reads_fc)) + geom_point() + facet_wrap(~ feature) + geom_abline(intercept = 0, slope = 1, col = "red") + labs(title = "Reads and molecules diverge at lower expression levels", x = "Log2 fold change of molecule counts to mean", y = "Log2 fold change of read counts to mean")
```

### ERCC spike-in mix

Po-Yuan added [ERCC RNA spike-in mix][ercc] 1 to the RNA from each single cell before tagging with unique molecular identifiers.
I downloaded the concentration information from Life Technologies website and combined it with the sequencing data.

[ercc]: http://www.lifetechnologies.com/order/catalog/product/4456740

```{r}
ercc <- read.table("pipeline/data/ercc_info.txt", header = TRUE, sep = "\t",
                   stringsAsFactors = FALSE)
colnames(ercc) <- c("num", "id", "subgroup", "conc_mix1", "conc_mix2",
                    "expected_fc", "log2_mix1_mix2")
# head(ercc)
ercc_umi <- merge(ercc, counts, by.x = "id", by.y = "gene")
```

Of the `r nrow(ercc)` ERCC spike-ins, only `r sum(ercc_umi$molecules > 0)` had at least one molecule sequenced.
As expected, the ones we sequenced were at a much higher concentration (attomoles/&mu;L) in the spike-in mix compared to those that we missed.

```{r concentration-of-sequenced-spike-ins}
ercc_umi$presence <- ifelse(ercc_umi$molecules > 0, "Present", "Absent")
ggplot(ercc_umi, aes(x = presence, y = conc_mix1)) + geom_boxplot() + scale_y_log10() + labs(title = "High concentration spike-ins were sequenced", x = "Was at least one molecule sequenced?", y = "log10 concentration in mix 1")
```

Of the `r sum(ercc_umi$molecules > 0)` spike-ins with at least one molecule sequenced, the concentration level was highly correlated with both the molecule (`r cor(ercc_umi$conc_mix1, ercc_umi$molecules)`) and read (`r cor(ercc_umi$conc_mix1, ercc_umi$reads)`) counts.

```{r}
ggplot(ercc_umi[ercc_umi$molecules > 0, ], aes(x = conc_mix1, y = molecules)) + geom_point() + geom_smooth(method = "lm") + labs(title = "Number of molecules is highly correlated with concentration", x = "Concentration in mix 1", y = "Number of molecules") + geom_text(aes(label = id), ercc_umi[ercc_umi$conc_mix1 > 10^4, ], size = 3, hjust = 1.1)
ggplot(ercc_umi[ercc_umi$molecules > 0, ], aes(x = conc_mix1, y = reads)) + geom_point() + geom_smooth(method = "lm") + labs(title = "Number of reads is highly correlated with concentration", x = "Concentration in mix 1", y = "Number of reads") + geom_text(aes(label = id), ercc_umi[ercc_umi$conc_mix1 > 10^4, ], size = 3, hjust = 1.1)
```

```{r}
summary(counts$reads)
summary(counts$molecules)
summary(counts$population)
counts$gene[counts$molecules > 1024]
```

