---
title: Assessment of single cell sequencing rapid run
author: John Blischak
output: html_document
---

`r Sys.Date()`

## Data processing

```{r setup, include=FALSE}
library("knitr")
opts_chunk$set(tidy = FALSE, fig.path = "figure/", echo = FALSE,
               warning = FALSE)
library("ggplot2")
theme_set(theme_bw(base_size = 14))
```

Po-Yuan sequenced two single cells on a rapid run flow cell.
In addition to the cellular RNA, she added the [ERCC RNA spike-in mix][ercc] separately to each cell to assess the efficiency of the single cell sequencing.
She also added in PhiX sample in order to boost the concentration.
The samples were multiplexed and the mix was sequenced on both lanes.
However, the sample barcode was not read so the data has to be analyzed jointly.

I processed the sequencing data using a pipeline which utilized Snakemake, R, umitools, Subread, and samtools.
The genome used for mapping was a combination of human genome hg19, the ERCC RNA spike-in sequences, and the PhiX genome.
The genomic features for counting were Ensembl exons from both coding and non-coding genes as well as the ERCC sequences.

For comparison, I also processed traditional RNA-sequencing on a population of cells for individual NA19239 from [Pickrell et al. 2010][Pickrell2010] (Yale data only because it was available in fastq format. The Argonne data is in map format).

[Pickrell2010]: http://www.ncbi.nlm.nih.gov/geo/query/acc.cgi?acc=GSE19480

```{r}
counts <- read.table("pipeline/data/count_matrix.txt", header = TRUE,
                     sep = "\t", stringsAsFactors = FALSE)
colnames(counts) <- c("gene", "chr", "start", "end", "strand", "length",
                      "reads", "molecules", "population")
counts$feature <- ifelse(grepl("ERCC", counts$gene), "control", "gene")
# colSums(counts[, c("reads", "molecules", "population")])
```

Here is the rough break down of the number of sequences obtained:

*  ~250 million raw reads
*  ~150 million reads with unique molecular identifier
*  ~114 million reads mapped to genome or controls
*  ~`r round(sum(counts$reads) / 10^6)` million reads mapped to human exons or control sequences
*  `r sum(counts$molecules)` unique molecules, which corresponds to `r sum(counts$molecules[counts$feature == "gene"] > 0)` genes

## ERCC spike-in mix

Po-Yuan added [ERCC RNA spike-in mix][ercc] 1 to the RNA from each single cell before tagging with unique molecular identifiers.
I downloaded the concentration information from Life Technologies website and combined it with the sequencing data.

[ercc]: http://www.lifetechnologies.com/order/catalog/product/4456740

```{r}
ercc <- read.table("pipeline/data/ercc_info.txt", header = TRUE, sep = "\t",
                   stringsAsFactors = FALSE)
colnames(ercc) <- c("num", "id", "subgroup", "conc_mix1", "conc_mix2",
                    "expected_fc", "log2_mix1_mix2")
# head(ercc)
ercc_umi <- merge(ercc, counts, by.x = "id", by.y = "gene")
```

Of the `r nrow(ercc)` ERCC spike-ins, only `r sum(ercc_umi$molecules > 0)` had at least one molecule sequenced.
As expected, the ones we sequenced were at a much higher concentration (attomoles/&mu;L) in the spike-in mix compared to those that we missed.

```{r concentration-of-sequenced-spike-ins}
ercc_umi$presence <- ifelse(ercc_umi$molecules > 0, "Present", "Absent")
ggplot(ercc_umi, aes(x = presence, y = conc_mix1)) + geom_boxplot() + scale_y_log10() + labs(title = "High concentration spike-ins were sequenced", x = "Was at least one molecule sequenced?", y = "log10 concentration in mix 1")
```

Of the `r sum(ercc_umi$molecules > 0)` spike-ins with at least one molecule sequenced, the concentration level was highly correlated with both the molecule (`r cor(ercc_umi$conc_mix1, ercc_umi$molecules)`) and read (`r cor(ercc_umi$conc_mix1, ercc_umi$reads)`) counts.

```{r}
ggplot(ercc_umi[ercc_umi$molecules > 0, ], aes(x = conc_mix1, y = molecules)) + geom_point() + geom_smooth(method = "lm") + labs(title = "Number of molecules is highly correlated with concentration", x = "Concentration in mix 1", y = "Number of molecules") + geom_text(aes(label = id), ercc_umi[ercc_umi$conc_mix1 > 10^4, ], size = 3, hjust = 1.1)
ggplot(ercc_umi[ercc_umi$molecules > 0, ], aes(x = conc_mix1, y = reads)) + geom_point() + geom_smooth(method = "lm") + labs(title = "Number of reads is highly correlated with concentration", x = "Concentration in mix 1", y = "Number of reads") + geom_text(aes(label = id), ercc_umi[ercc_umi$conc_mix1 > 10^4, ], size = 3, hjust = 1.1)
```

## Comparing reads and molecules

```{r clean}
# Remove genes with zero molecule counts
counts_present <- counts[counts$molecules > 0, ]
```

For the analyses below, I use only the `r sum(counts_present$feature == "gene")` genes and `r sum(counts_present$feature == "control")` controls with at least one molecule identified in this experiment.

The read and molecule counts are highly correlated for both the ERCC control spike-ins (`r cor(counts_present$molecules[counts_present$feature == "control"], counts_present$reads[counts_present$feature == "control"])`) and genes (`r cor(counts_present$molecules[counts_present$feature == "gene"], counts_present$reads[counts_present$feature == "gene"])`).

```{r reads-v-molecules}
ggplot(counts_present, aes(x = molecules, y = reads)) + geom_point() +
  facet_wrap(~ feature) + geom_smooth(method = "lm") + ylim(0, 750000) +
  labs(title = "Read and molecule counts are highly correlated")
```

Note that this plot has been truncated on the y-axis to exclude a few highly expressed genes.
Also note that the number of molecules is greater than 1024, which is the number of possible unique molecular identifiers.
This is likely due to genes that are highly expressed and have multiple transcription start sites, which would make it possible to tag more than 1024 molecules.
Overall, this occured only for a small fraction of the features (`r sum(counts$molecules > 1024 & counts$feature == "gene")` genes and `r sum(counts$molecules > 1024 & counts$feature == "control")` controls).

In order to find where the measurements differed, I created a plot similar to Figure S1 of [Islam et al., 2014][Islam2014].
In it, they calculate the log fold change of each count to the mean count, e.g. the log fold change of the number of molecules for a particular gene compared to the mean number of molecules across all genes.

[Islam2014]: http://www.nature.com/nmeth/journal/v11/n2/full/nmeth.2772.html

```{r log-fold-change-to-mean}
counts_present$reads_fc <- log2(counts_present$reads / mean(counts_present$reads))
counts_present$molecules_fc <- log2(counts_present$molecules / mean(counts_present$molecules))
ggplot(counts_present, aes(x = molecules_fc, y = reads_fc)) + geom_point() + facet_wrap(~ feature) + geom_abline(intercept = 0, slope = 1, col = "red") + labs(title = "Reads and molecules diverge at lower expression levels", x = "Log2 fold change of molecule counts to mean", y = "Log2 fold change of read counts to mean")
```

Similar to their result, the read and molecule counts disagree for lower expressed genes.
Normal sequencing underestimates the expression compared to the true number of molecules.

I also performed a similar analysis, but this time comparing to the median counts.
Overall the trend looks similar.

```{r log-fold-change-to-median}
counts_present$reads_fc_med <- log2(counts_present$reads / median(counts_present$reads))
counts_present$molecules_fc_med <- log2(counts_present$molecules / median(counts_present$molecules))
ggplot(counts_present, aes(x = molecules_fc_med, y = reads_fc_med)) + geom_point() + facet_wrap(~ feature) + geom_abline(intercept = 0, slope = 1, col = "red") + labs(title = "Reads and molecules diverge at lower expression levels", x = "Log2 fold change of molecule counts to median", y = "Log2 fold change of read counts to median")
```

The main difference is that the points fall above the diagonal because the median of the reads is so much lower than the mean.

*  mean read count: `r round(mean(counts_present$reads), 2)`
*  median read count: `r median(counts_present$reads)`
*  mean molecule count: `r round(mean(counts_present$molecules), 2)`
*  median molecule count: `r median(counts_present$molecules)`

## Comparing single cell versus traditional sequencing

```{r}
counts_pop <- counts[counts$molecules > 0 | counts$population > 0 &
                     counts$feature == "gene", ]
counts_pop$pop_cpbp <- counts_pop$population / counts_pop$length
```

For this analysis, I used only the `r nrow(counts_pop)` genes with at least one molecule sequenced in this experiment or at least one read sequenced in [Pickrell et al. 2010][Pickrell2010].
The single cell-sequencing does not need to be scaled by gene length because only the 5' end of each RNA molecule is sequenced.
However, the Pickrell data needs to standardized by gene length since longer genes have more RNA that can be fragmented and sequenced.
Thus below the traditional data is counts per base pair of exonic sequence.

Both the number of molecules (`r cor(counts_pop$molecules, counts_pop$pop_cpbp)`) and reads (`r cor(counts_pop$reads, counts_pop$pop_cpbp)`) from the single cell sequencing are modestly correlated with the counts per base pair from the traditional sequencing.

```{r molecules-v-population}
ggplot(counts_pop, aes(x = molecules, y = pop_cpbp)) + geom_point() +
    geom_smooth(method = "lm") +
  labs(title = "Molecule counts and counts per base pair are modestly correlated",
       x = "Number of molecules from single cell seq",
       y = "Counts per base pair")
```

```{r reads-v-population}
ggplot(counts_pop, aes(x = reads, y = pop_cpbp)) + geom_point() +
    geom_smooth(method = "lm") +
  labs(title = "Read counts and counts per base pair are modestly correlated",
       x = "Number of reads from single cell seq", y = "Counts per base pair")
```

## Rarefaction curve

We sequenced these two single cells on both lanes of a rapid run, which resulted in ~250 million raw reads and `r sum(counts$molecules)` unique molecules mapped to coding and non-coding genes as well as the ERCC controls.

```{r}
library("vegan")
```

Based on a rarefaction curve, implemented using the R package vegan, it appears we are close to saturation.

```{r rarefaction-curve-all}
options(scipen = 5)
rarecurve(x = matrix(counts_present$molecules, nrow = 1), step = 10^4,
          xlab = "Number of molecules", ylab = "Number of genes")
```

The above plot requires a minumum of only one unique molecule for a gene to be considered present.
Also, it includes non-coding genes and the ERRC controls.

```{r}
library("biomaRt")
ensembl <- useMart("ensembl", dataset = "hsapiens_gene_ensembl")
ens_biotype <- getBM(attributes = c("ensembl_gene_id" ,"gene_biotype"),
                     filters = "ensembl_gene_id",
                     values = counts_present$gene[counts_present$feature == "gene"],
                     mart = ensembl)
counts_type <- merge(counts_present, ens_biotype, by.x = "gene",
                     by.y = "ensembl_gene_id")
```

However, the result is similar when using only protein-coding genes.

```{r rarefaction-curve-coding}
rarecurve(x = matrix(counts_type$molecules[counts_type$gene_biotype == "protein_coding"], nrow = 1),
          step = 10^4, xlab = "Number of molecules",
          ylab = "Number of protein-coding genes")
```


