---
title: "Mixed effect model for batch correction - crossed design"
author: "Joyce Hsiao"
date: 2015-08-17
---

**Last updated:** `r Sys.Date()`

**Code version:** `r system("git log -1 --format='%H'", intern = TRUE)`

```{r chunk-options, include=FALSE}
source("chunk-options.R")
```


## Setup
```{r}
source("functions.R")
library("limma")
library("edgeR")
library(ggplot2)
```

## Prepare single cell molecule data


Input annotation

```{r}
anno <- read.table("../data/annotation.txt", header = TRUE,
                   stringsAsFactors = FALSE)
```


Input molecule counts

```{r}
molecules <- read.table("../data/molecules.txt", header = TRUE,
           stringsAsFactors = FALSE)
```


Input list of quality single cells

```{r}
quality_single_cells <-
  scan("../data/quality-single-cells.txt", what = "character")
```


Keep only the single cells that pass the QC filters. This also removes the bulk samples

```{r}
molecules_single <- molecules[, colnames(molecules) %in% quality_single_cells]
anno_single <- anno[anno$sample_id %in% quality_single_cells, ]
stopifnot(ncol(molecules_single) == nrow(anno_single),
          colnames(molecules_single) == anno_single$sample_id)
```


Also remove batch 2 of individual 19098.

```{r}
molecules_single <- 
  molecules_single[, !(anno_single$individual == 19098 & anno_single$batch == 2)]
anno_single <- 
  anno_single[!(anno_single$individual == 19098 & 
                anno_single$batch == 2), ]
stopifnot(ncol(molecules_single) == nrow(anno_single))
```


Remove genes with zero read counts in the single cells
```{r}
expressed_single <- rowSums(molecules_single) > 0
molecules_single <- molecules_single[expressed_single, ]
```


Remove genes with greater than or equal to 1,024 molecules 
in at least one of the cells

```{r}
overexpressed_genes <- 
  rownames(molecules_single)[ 
  apply(molecules_single, 1, 
          function(x) any(x >= 1024))]
molecules_single <-
  molecules_single[!(rownames(molecules_single) %in% overexpressed_genes), ]
```


Correct for collision probability

```{r}
molecules_single_collision <- -1024 * log(1 - molecules_single/1024)
```


Standardize the molecule counts to account for differences in sequencing
depth. This is necessary because the sequencing depth affects the total
molecule counts

```{r}
molecules_single_cpm <- cpm(molecules_single_collision, log = TRUE)
```


Prepare ERCC data
```{r}
gene_rows_single <- grep("ERCC", 
                         rownames(molecules_single), invert = TRUE)
dim(molecules_single_cpm[gene_rows_single, ])
```


## Mixed model for batch-effect correction

### Crossed Model

For every single gene, we will fit a mixed model assuming
differences between batches are not individual-specific as follows

$$ y_{ijk} = \mu + \alpha_i + b_j + \epsilon_{ijk} $$,

where $y_{ijk}$ is the log2 counts-per-million (cpm) for any gene 
in individual $i$, batch $j$, and cell $k$, $\mu$ is the gene-specific
expression level across all cells, $\alpha_i$ is the expression level
specific to individual $i$, $b_j$ is batch $j$'s deviation of expression
level from gene-specific expression levels, and $\epsilon_{ijk}$ is 
the models' residual error. 

We assume that $b_j$ follows a normal distribution with $b_j \sim N(0, \sigma^2_b)$ for $j = 1, \dots, 9$, and $\epsilon_{ijk} \sim N(0, \sigma^2_\epsilon)$ for $i = 1, 2, 3; j = 1, \dots, 9; and k = 1, \dots, n_{ij}$, where $n_ij$ denotes the number of cells in individual $i$, batch $j$. 


First, we create a unique identifying ID for the 9 batches.

```{r}
batch_unique <- 
  with(anno_single, paste(individual, "_", batch, sep = "") )
table(batch_unique)
```

Load the Humanzee package

```{r}
if (!require(Humanzee, quietly = TRUE)) {
  library(devtools)
  install_github("jhsiao999/Humanzee")
  library(Humanzee)
}
```

Create design matrix and compute a consensu correlation coefficient using
limma's duplicateCorrelation function.

```{r}
block <- batch_unique
design <- model.matrix(~ 1 + individual, data = anno_single)
dup_corrs <- 
    duplicateCorrelation(molecules_single_cpm[gene_rows_single, ],
                         design = design, block = block)
```

Fit a mixed model with the 9 batches being the random effect.

```{r, eval = FALSE}
gls_fit <- 
    Humanzee::ruv_mixed_model(molecules_single_cpm[gene_rows_single, ], 
                    ndups = 1,
                    design = design, block = block, 
                    correlation = dup_corrs$cons)
save(gls_fit, file = "../data/limma-crossed.rda")
```


Compute expression levels after removing variation due to random effects.

```{r}
load("../data/limma-crossed.rda")
molecule_batch_removed <- t( design %*% t(gls_fit$coef) ) + gls_fit$resid

pca_single_mixed_correction <- run_pca(molecule_batch_removed)
plot_pca(pca_single_mixed_correction$PCs, 
         explained = pca_single_mixed_correction$explained,
         metadata = anno_single, color = "individual",
         shape = "batch", factors = c("individual", "batch"))
```


### Nested Model

For every single gene, we will fit a mixed model assuming
differences between batches as homogeneous within individuals as follows

$$ y_{ijk} = \mu + \alpha_i + b_{ij} + \epsilon_{ijk} $$,

where the only difference from the Cross model is the batch random effect $b_{ij}$, with $i = 1, 2, 3$ and $j = 1, 2, 3$. Hence $\hat{b}_{i.}$ estimates for individual $i$ the random variation of expression levels from the entire population as a function of the variation across the three batches. Note that as before, $b_{ij}$ follows a normal distribution $N(0, \sigma^2_{b})$. 

Note that limma does not accommodate fitting of nested random effect. We will use other algorithms to remove unwanted variation under the nested model framework.



## Session information

```{r info}
sessionInfo()
```
