<!DOCTYPE html>

<html xmlns="http://www.w3.org/1999/xhtml">

<head>

<meta charset="utf-8">
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="generator" content="pandoc" />

<meta name="author" content="Joyce Hsiao" />

<meta name="date" content="2015-10-26" />

<title>Biological variation analysis summary</title>

<script src="libs/jquery-1.11.0/jquery.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="libs/bootstrap-3.3.1/css/united.min.css" rel="stylesheet" />
<script src="libs/bootstrap-3.3.1/js/bootstrap.min.js"></script>
<script src="libs/bootstrap-3.3.1/shim/html5shiv.min.js"></script>
<script src="libs/bootstrap-3.3.1/shim/respond.min.js"></script>

<style type="text/css">

/* padding for bootstrap navbar */
body {
  padding-top: 50px;
  padding-bottom: 40px;
}


/* offset scroll position for anchor links (for fixed navbar)  */
.section h2 {
  padding-top: 55px;
  margin-top: -55px;
}
.section h3 {
  padding-top: 55px;
  margin-top: -55px;
}



/* don't use link color in navbar */
.dropdown-menu>li>a {
  color: black;
}

/* some padding for disqus */
#disqus_thread {
  margin-top: 45px;
}

</style>

<link rel="stylesheet" href="libs/font-awesome-4.1.0/css/font-awesome.min.css"/>

<style type="text/css">code{white-space: pre;}</style>
<link rel="stylesheet"
      href="libs/highlight/textmate.css"
      type="text/css" />
<script src="libs/highlight/highlight.js"></script>
<style type="text/css">
  pre:not([class]) {
    background-color: white;
  }
</style>
<script type="text/javascript">
if (window.hljs && document.readyState && document.readyState === "complete") {
   window.setTimeout(function() {
      hljs.initHighlighting();
   }, 0);
}
</script>



</head>

<body>

<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
code {
  color: inherit;
  background-color: rgba(0, 0, 0, 0.04);
}
img { 
  max-width:100%; 
  height: auto; 
}
</style>
<div class="container-fluid main-container">


<div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">singleCellSeq</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li><a href="index.html">Home</a></li>
        <li><a href="about.html">About</a></li>
        <li><a href="license.html">License</a></li>
        <li><a href="https://github.com/jdblischak/singleCellSeq">GitHub</a></li>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<div id="header">
<h1 class="title">Biological variation analysis summary</h1>
<h4 class="author"><em>Joyce Hsiao</em></h4>
<h4 class="date"><em>2015-10-26</em></h4>
</div>

<div id="TOC">
<ul>
<li><a href="#cell-to-cell-heterogeneity">cell-to-cell-heterogeneity</a></li>
<li><a href="#transcriptome-wide-cell-to-cell-heterogeneity-between-individuals">Transcriptome-wide cell-to-cell heterogeneity between individuals</a></li>
</ul>
</div>

<div id="cell-to-cell-heterogeneity" class="section level2">
<h2>cell-to-cell-heterogeneity</h2>
<p>We evaluate the variability in gene expression profiles for each individual samples. The principal component analysis suggest that after our normalization steps, the major source of variation is not driven by replicate labels but instead is highly correlated with mean molecule counts across replicates. Therefore, for transcriptome-wide comparison of cell-to-cell heterogeneity, we combine replicates of each individual samples and compute each gene’s coefficients of variation across the combined samples. <a href="http://jdblischak.github.io/singleCellSeq/analysis/cv-adjusted-transcriptome-wide.html">link1</a></p>
<p>To account for the confounding effect of mean molecule counts in the individual difference of coefficients of variation, we fit a data-wide PCA and filtered the first PC (29% of variation), which is highly correlated with mean molecule counts (r = .90).</p>
</div>
<div id="transcriptome-wide-cell-to-cell-heterogeneity-between-individuals" class="section level2">
<h2>Transcriptome-wide cell-to-cell heterogeneity between individuals</h2>
<p>To compare cell-to-cell heterogeneity of gene expression profiles between individuals, we used coefficients of variation of the normalized counts. We combine the replicates to compute per gene coefficient of variation for each individual samples (<a href="http://jdblischak.github.io/singleCellSeq/analysis/cv-adjusted-transcriptome-wide.html">link1</a>); principal component analysis of each individual samples consistenly suggest that the main source of sample variation is mean gene expression level, and is not a function of replicate labels. We observed that the CV profiles differ between individuals by the Friedman rank sum test (p &lt; 1e-5). To investigate individual differencees in cell-to-cell heterogeneity beyond the effect of mean gene expression levels, we remove the first PC from the normalized counts, which accounts for 29% of variation in the data and is highly correlated with mean gene expression levels (r = .9) (link2).</p>
<p>Between individuals, the CV profiles are highly correlated prior to PC1 removal, possibly due to the high correlation between PC1 and mean expression levels. The adjusted CV profiles are orthogonal of each other, suggesting the presence of individual-specific cell-to-cell heterogeneity profiles in single-cell sequencing data that is independent of mean expression levels. <a href="http://jdblischak.github.io/singleCellSeq/analysis/svd-filtered-high-low.html">link3</a></p>
<p>We furthered investigated genes with extremely high or low adjusted CV profiles (mean +/- 2 standard deviation). [Summarize functional categories…]</p>
<p>We further investigate potential sources of biological variation in iPSC cells that may be due to cel-cycle and pluripotency. The adjusted CV profiles of the cell-cycle genes significantly differed between individuals (p &lt; .0005), and so are the non-cell-cycle genes (p &lt; 1e-6). However, we observed that the adjusted CV profiles do not differ between the pluripotent genes. (<a href="http://jdblischak.github.io/singleCellSeq/analysis/svd-filtered.html">link2</a>)</p>
</div>


<!-- some extra javascript for older browsers -->
<script type="text/javascript" src="libs/polyfill.js"></script>

<script>

// manage active state of menu based on current page
$(document).ready(function () {

    // active menu
    href = window.location.pathname
    href = href.substr(href.lastIndexOf('/') + 1)
    $('a[href="' + href + '"]').parent().addClass('active');

    // manage active menu header
    if (href.startsWith('authoring_'))
      $('a[href="' + 'authoring' + '"]').parent().addClass('active');
    else if (href.endsWith('_format.html'))
      $('a[href="' + 'formats' + '"]').parent().addClass('active');
    else if (href.startsWith('developer_'))
      $('a[href="' + 'developer' + '"]').parent().addClass('active');

});

</script>

</div>

<script>

// add bootstrap table styles to pandoc tables
$(document).ready(function () {
  $('tr.header').parent('thead').parent('table').addClass('table table-condensed');
});

</script>


</body>
</html>
