<!DOCTYPE html>

<html xmlns="http://www.w3.org/1999/xhtml">

<head>

<meta charset="utf-8">
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="generator" content="pandoc" />

<meta name="author" content="John Blischak" />

<meta name="date" content="2015-02-19" />

<title>Tracking down bug in reads to molecules conversion - Attempt 02</title>

<script src="libs/jquery-1.11.0/jquery.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="libs/bootstrap-3.3.1/css/united.min.css" rel="stylesheet" />
<script src="libs/bootstrap-3.3.1/js/bootstrap.min.js"></script>
<script src="libs/bootstrap-3.3.1/shim/html5shiv.min.js"></script>
<script src="libs/bootstrap-3.3.1/shim/respond.min.js"></script>

<style type="text/css">

/* padding for bootstrap navbar */
body {
  padding-top: 50px;
  padding-bottom: 40px;
}


/* offset scroll position for anchor links (for fixed navbar)  */
.section h2 {
  padding-top: 55px;
  margin-top: -55px;
}
.section h3 {
  padding-top: 55px;
  margin-top: -55px;
}



/* don't use link color in navbar */
.dropdown-menu>li>a {
  color: black;
}

/* some padding for disqus */
#disqus_thread {
  margin-top: 45px;
}

</style>

<link rel="stylesheet" href="libs/font-awesome-4.1.0/css/font-awesome.min.css"/>

<style type="text/css">code{white-space: pre;}</style>
<link rel="stylesheet"
      href="libs/highlight/textmate.css"
      type="text/css" />
<script src="libs/highlight/highlight.js"></script>
<style type="text/css">
  pre:not([class]) {
    background-color: white;
  }
</style>
<script type="text/javascript">
if (window.hljs && document.readyState && document.readyState === "complete") {
   window.setTimeout(function() {
      hljs.initHighlighting();
   }, 0);
}
</script>



</head>

<body>

<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
code {
  color: inherit;
  background-color: rgba(0, 0, 0, 0.04);
}
img { 
  max-width:100%; 
  height: auto; 
}
</style>
<div class="container-fluid main-container">


<div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">singleCellSeq</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li><a href="index.html">Home</a></li>
        <li><a href="about.html">About</a></li>
        <li><a href="license.html">License</a></li>
        <li><a href="https://github.com/jdblischak/singleCellSeq">GitHub</a></li>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<div id="header">
<h1 class="title">Tracking down bug in reads to molecules conversion - Attempt 02</h1>
<h4 class="author"><em>John Blischak</em></h4>
<h4 class="date"><em>2015-02-19</em></h4>
</div>

<div id="TOC">
<ul>
<li><a href="#input">Input</a></li>
<li><a href="#identifying-problem-genes">Identifying problem genes</a></li>
<li><a href="#searching-for-where-the-reads-are-lost-during-the-processing-pipeline">Searching for where the reads are lost during the processing pipeline</a></li>
<li><a href="#using-the-correct-de-deduplicated-sequences">Using the correct de-deduplicated sequences</a></li>
<li><a href="#next-step">Next step</a></li>
<li><a href="#session-information">Session information</a></li>
</ul>
</div>

<p><strong>Last updated:</strong> 2016-02-19</p>
<p><strong>Code version:</strong> a82b08d63e9164bb57307a210fd5ac09189f61f2</p>
<p>After <a href="bug-conversion-01.html">identifying a source of problematic reads</a> and re-running featureCounts <a href="read2pos.html">using the read2pos flag</a>, the <a href="https://github.com/jdblischak/singleCellSeq/tree/master/tests">tests</a> still fail. This anaysis explores the remaining bugs lurking in the <a href="process-samples.html">sequence processsing pipeline</a>.</p>
<div id="input" class="section level2">
<h2>Input</h2>
<p>Input annotation.</p>
<pre class="r"><code>anno &lt;- read.table(&quot;../data/annotation.txt&quot;, header = TRUE,
                   stringsAsFactors = FALSE)</code></pre>
<p>Input read counts.</p>
<pre class="r"><code>reads &lt;- read.table(&quot;../data/reads.txt&quot;, header = TRUE,
                    stringsAsFactors = FALSE)
stopifnot(ncol(reads) == nrow(anno),
          colnames(reads) == anno$sample_id)</code></pre>
<p>Input molecule counts.</p>
<pre class="r"><code>molecules &lt;- read.table(&quot;../data/molecules.txt&quot;, header = TRUE,
                    stringsAsFactors = FALSE)
stopifnot(ncol(molecules) == nrow(anno),
          colnames(molecules) == anno$sample_id)</code></pre>
<p>Input list of quality single cells.</p>
<pre class="r"><code>quality_single_cells &lt;- scan(&quot;../data/quality-single-cells.txt&quot;,
                             what = &quot;character&quot;)</code></pre>
</div>
<div id="identifying-problem-genes" class="section level2">
<h2>Identifying problem genes</h2>
<p>How often is this a problem? How many genes does it affect? For affected genes, how many samples are affected?</p>
<pre class="r"><code>discordant_zeros &lt;- (reads == 0) != (molecules == 0)
all_genes &lt;- rowSums(discordant_zeros)
names(all_genes) &lt;- rownames(reads)
problem_genes &lt;- all_genes[all_genes &gt; 0]
length(problem_genes)</code></pre>
<pre><code>[1] 13</code></pre>
<pre class="r"><code>length(problem_genes) / length(all_genes)</code></pre>
<pre><code>[1] 0.0007751014</code></pre>
<pre class="r"><code>summary(problem_genes)</code></pre>
<pre><code>   Min. 1st Qu.  Median    Mean 3rd Qu.    Max. 
  1.000   1.000   1.000   1.077   1.000   2.000 </code></pre>
<p>This problem affects 13 out of the 16772 total genes (0.08%). For these problem genes, the median number of affected samples is 1 and the max number of affected samples is 2.</p>
<p>Next I identify a problem gene-sample combination so that I can track down what happened.</p>
<pre class="r"><code>x &lt;- names(problem_genes)[1]
plot(as.numeric(reads[x, ]), as.numeric(molecules[x, ]))</code></pre>
<p><img src="figure/bug-conversion-02.Rmd/problem-1-1.png" title="" alt="" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>problem_sample &lt;- colnames(reads)[discordant_zeros[x, ] == TRUE]
reads[x, problem_sample]</code></pre>
<pre><code>[1] 33</code></pre>
<pre class="r"><code>molecules[x, problem_sample]</code></pre>
<pre><code>[1] 0</code></pre>
<p>For gene ENSG00000143294, the sample NA19098.r3.E09 was assigned 33 reads but 0 molecules. What happened?</p>
<p>Note that NA19098.r3.E09 is not a high quality single cell, so this particular error has no effect on our downstream results.</p>
<pre class="r"><code>problem_sample %in% quality_single_cells</code></pre>
<pre><code>[1] FALSE</code></pre>
</div>
<div id="searching-for-where-the-reads-are-lost-during-the-processing-pipeline" class="section level2">
<h2>Searching for where the reads are lost during the processing pipeline</h2>
<p>Now I am going to search for the bug by inspecting the intermediate data files from the <a href="process-samples.html">sequence processsing pipeline</a>.</p>
<p>The following chunks are all Bash commands run from the data directory.</p>
<pre class="r"><code>opts_chunk$set(engine = &quot;bash&quot;)
opts_knit$set(root.dir = &quot;/mnt/gluster/home/jdblischak/ssd&quot;)</code></pre>
<p>First I confirm that this difference is observed at the <a href="process-samples.html#count-reads-per-gene">featureCounts step</a>:</p>
<pre class="bash"><code># reads per lane
grep ENSG00000143294 counts/19098.3.E09*trim.sickle.sorted.genecounts.txt | cut -f1,7</code></pre>
<pre><code>counts/19098.3.E09.TCCCATGG.L001.R1.C723YACXX.trim.sickle.sorted.genecounts.txt:ENSG00000143294 11
counts/19098.3.E09.TCCCATGG.L003.R1.C72JMACXX.trim.sickle.sorted.genecounts.txt:ENSG00000143294 13
counts/19098.3.E09.TCCCATGG.L007.R1.C6WYKACXX.trim.sickle.sorted.genecounts.txt:ENSG00000143294 9</code></pre>
<pre class="bash"><code># molecules per lane
grep ENSG00000143294 counts/19098.3.E09*trim.sickle.sorted.rmdup.genecounts.txt | cut -f1,7</code></pre>
<pre><code>counts/19098.3.E09.TCCCATGG.L001.R1.C723YACXX.trim.sickle.sorted.rmdup.genecounts.txt:ENSG00000143294   0
counts/19098.3.E09.TCCCATGG.L003.R1.C72JMACXX.trim.sickle.sorted.rmdup.genecounts.txt:ENSG00000143294   0
counts/19098.3.E09.TCCCATGG.L007.R1.C6WYKACXX.trim.sickle.sorted.rmdup.genecounts.txt:ENSG00000143294   0</code></pre>
<pre class="bash"><code># molecules per sample
grep ENSG00000143294 counts/19098.3.E09.trim.sickle.sorted.combined.rmdup.genecounts.txt | cut -f1,7</code></pre>
<pre><code>ENSG00000143294 0</code></pre>
<p>It is. The featureCounts results corroborate that 33 reads are mapped to this sample but zero molecules. Therefore the problem is upstream of my <a href="process-samples.html#gather-gene-counts">Python script to gather the gene counts</a>.</p>
<p>Next I inspect the featureCounts assigment results (flag <code>-R</code>) to corroborate the output file results. Because all the lanes failed similarly, Iâ€™ll focus on the lane with only 9 reads for simplicity.</p>
<pre class="bash"><code># reads per lane
grep ENSG00000143294 counts/19098.3.E09.TCCCATGG.L007.R1.C6WYKACXX.trim.sickle.sorted.bam.featureCounts</code></pre>
<pre><code>HWI-700819F:303:C6WYKACXX:7:1104:4034:6156:UMI_GTCTGGGG Assigned    ENSG00000143294 *
HWI-700819F:303:C6WYKACXX:7:1201:15253:37760:UMI_GTCTGGGG   Assigned    ENSG00000143294 *
HWI-700819F:303:C6WYKACXX:7:1205:20480:54449:UMI_GTCTGGGG   Assigned    ENSG00000143294 *
HWI-700819F:303:C6WYKACXX:7:1208:12235:61153:UMI_GTCTGGGG   Assigned    ENSG00000143294 *
HWI-700819F:303:C6WYKACXX:7:1213:13458:22728:UMI_GTCTGGGG   Assigned    ENSG00000143294 *
HWI-700819F:303:C6WYKACXX:7:2105:9174:45224:UMI_GTCTGGGG    Assigned    ENSG00000143294 *
HWI-700819F:303:C6WYKACXX:7:2109:16882:12730:UMI_GTCTGGGG   Assigned    ENSG00000143294 *
HWI-700819F:303:C6WYKACXX:7:2113:2934:40550:UMI_GTCTGGGG    Assigned    ENSG00000143294 *
HWI-700819F:303:C6WYKACXX:7:2215:7988:79679:UMI_GTCTGGGG    Assigned    ENSG00000143294 *</code></pre>
<p>All 9 reads have the same UMI sequence.</p>
<pre class="bash"><code># molecules per lane
grep ENSG00000143294 counts/19098.3.E09.TCCCATGG.L007.R1.C6WYKACXX.trim.sickle.sorted.rmdup.bam.featureCounts
# I have to add the following to change the exit status. When grep does not find
# a match, its exit status is 1. Since this is an error exit status, knitr
# assumes something has failed and stops execution.
exit 0</code></pre>
<p>But no molecules!</p>
<p>Now I want to know if featureCounts reports having seen any of these reads, but does not assign them for some reason (e.g.Â ambiguity, mapping to more than one feature).</p>
<pre class="bash"><code>problem_reads=`grep ENSG00000143294 counts/19098.3.E09.TCCCATGG.L007.R1.C6WYKACXX.trim.sickle.sorted.bam.featureCounts | cut -f1`
for read in $problem_reads
do
  echo &quot;read: $read&quot;
  grep $read counts/19098.3.E09.TCCCATGG.L007.R1.C6WYKACXX.trim.sickle.sorted.rmdup.bam.featureCounts
done
exit 0</code></pre>
<pre><code>read: HWI-700819F:303:C6WYKACXX:7:1104:4034:6156:UMI_GTCTGGGG
read: HWI-700819F:303:C6WYKACXX:7:1201:15253:37760:UMI_GTCTGGGG
read: HWI-700819F:303:C6WYKACXX:7:1205:20480:54449:UMI_GTCTGGGG
read: HWI-700819F:303:C6WYKACXX:7:1208:12235:61153:UMI_GTCTGGGG
read: HWI-700819F:303:C6WYKACXX:7:1213:13458:22728:UMI_GTCTGGGG
read: HWI-700819F:303:C6WYKACXX:7:2105:9174:45224:UMI_GTCTGGGG
read: HWI-700819F:303:C6WYKACXX:7:2109:16882:12730:UMI_GTCTGGGG
read: HWI-700819F:303:C6WYKACXX:7:2113:2934:40550:UMI_GTCTGGGG
read: HWI-700819F:303:C6WYKACXX:7:2215:7988:79679:UMI_GTCTGGGG</code></pre>
<p>featureCounts has no record of having seen any of these reads and then not assigning them. This suggests the problem is upstream. Are any of these reads present in the bam file passed to featureCounts? These are the output files from the <a href="http://jdblischak.github.io/singleCellSeq/analysis/process-samples.html#remove-duplicate-umis">step to remove duplicate reads</a> (i.e.Â convert reads to molecules).</p>
<pre class="bash"><code>problem_reads=`grep ENSG00000143294 counts/19098.3.E09.TCCCATGG.L007.R1.C6WYKACXX.trim.sickle.sorted.bam.featureCounts | cut -f1`
for read in $problem_reads
do
  echo &quot;read: $read&quot;
  samtools view bam-rmdup-umi/19098.3.E09.TCCCATGG.L007.R1.C6WYKACXX.trim.sickle.sorted.rmdup.bam | grep $read
done
exit 0</code></pre>
<pre><code>read: HWI-700819F:303:C6WYKACXX:7:1104:4034:6156:UMI_GTCTGGGG
read: HWI-700819F:303:C6WYKACXX:7:1201:15253:37760:UMI_GTCTGGGG
read: HWI-700819F:303:C6WYKACXX:7:1205:20480:54449:UMI_GTCTGGGG
read: HWI-700819F:303:C6WYKACXX:7:1208:12235:61153:UMI_GTCTGGGG
read: HWI-700819F:303:C6WYKACXX:7:1213:13458:22728:UMI_GTCTGGGG
read: HWI-700819F:303:C6WYKACXX:7:2105:9174:45224:UMI_GTCTGGGG
read: HWI-700819F:303:C6WYKACXX:7:2109:16882:12730:UMI_GTCTGGGG
read: HWI-700819F:303:C6WYKACXX:7:2113:2934:40550:UMI_GTCTGGGG
read: HWI-700819F:303:C6WYKACXX:7:2215:7988:79679:UMI_GTCTGGGG</code></pre>
<p>Nothing there either.</p>
</div>
<div id="using-the-correct-de-deduplicated-sequences" class="section level2">
<h2>Using the correct de-deduplicated sequences</h2>
<p>This was strange that none of these reads were maintained in the molecules file. I took a closer look at the data directory and realized there was still a <code>bam-dedup-umi</code> directory. I had used that when I was initially testing dedup_umi.py from UMI-tools. Checking the dates, I realized that these files were more recent!</p>
<pre class="bash"><code>ls -lhS bam-rmdup-umi/*bam | head</code></pre>
<pre><code>-rw-r--r-- 1 jdblischak jdblischak   22M Jan  6 12:20 bam-rmdup-umi/19101.3.D02.trim.sickle.sorted.combined.rmdup.bam
-rw-r--r-- 1 jdblischak jdblischak   21M Jan  6 12:17 bam-rmdup-umi/19098.1.F01.trim.sickle.sorted.combined.rmdup.bam
-rw-r--r-- 1 jdblischak jdblischak   20M Jan  6 12:20 bam-rmdup-umi/19101.3.C02.trim.sickle.sorted.combined.rmdup.bam
-rw-r--r-- 1 jdblischak jdblischak   19M Jan  6 12:20 bam-rmdup-umi/19101.3.H04.trim.sickle.sorted.combined.rmdup.bam
-rw-r--r-- 1 jdblischak jdblischak   19M Jan  6 12:20 bam-rmdup-umi/19101.3.G04.trim.sickle.sorted.combined.rmdup.bam
-rw-r--r-- 1 jdblischak jdblischak   19M Jan  6 12:20 bam-rmdup-umi/19101.3.D11.trim.sickle.sorted.combined.rmdup.bam
-rw-r--r-- 1 jdblischak jdblischak   18M Jan  6 12:19 bam-rmdup-umi/19101.2.E08.trim.sickle.sorted.combined.rmdup.bam
-rw-r--r-- 1 jdblischak jdblischak   18M Jan  6 12:20 bam-rmdup-umi/19101.3.B03.trim.sickle.sorted.combined.rmdup.bam
-rw-r--r-- 1 jdblischak jdblischak   18M Jan  6 12:17 bam-rmdup-umi/19098.2.G12.trim.sickle.sorted.combined.rmdup.bam
-rw-r--r-- 1 jdblischak jdblischak   17M Jan  6 12:12 bam-rmdup-umi/19101.3.D02.AGTCCACC.L004.R1.C6WYKACXX.trim.sickle.sorted.rmdup.bam</code></pre>
<pre class="bash"><code>ls -lhS bam-dedup-umi/*bam | head</code></pre>
<pre><code>-rw-r--r-- 1 jdblischak jdblischak   24M Jan 11 23:13 bam-dedup-umi/19101.3.D02.trim.sickle.sorted.combined.rmdup.bam
-rw-r--r-- 1 jdblischak jdblischak   22M Jan 11 23:08 bam-dedup-umi/19098.1.F01.trim.sickle.sorted.combined.rmdup.bam
-rw-r--r-- 1 jdblischak jdblischak   21M Jan 11 23:13 bam-dedup-umi/19101.3.C02.trim.sickle.sorted.combined.rmdup.bam
-rw-r--r-- 1 jdblischak jdblischak   21M Jan 11 23:14 bam-dedup-umi/19101.3.H04.trim.sickle.sorted.combined.rmdup.bam
-rw-r--r-- 1 jdblischak jdblischak   21M Jan 11 23:14 bam-dedup-umi/19101.3.G04.trim.sickle.sorted.combined.rmdup.bam
-rw-r--r-- 1 jdblischak jdblischak   20M Jan 11 23:12 bam-dedup-umi/19101.2.E08.trim.sickle.sorted.combined.rmdup.bam
-rw-r--r-- 1 jdblischak jdblischak   20M Jan 11 23:14 bam-dedup-umi/19101.3.D11.trim.sickle.sorted.combined.rmdup.bam
-rw-r--r-- 1 jdblischak jdblischak   19M Jan 11 23:13 bam-dedup-umi/19101.3.B03.trim.sickle.sorted.combined.rmdup.bam
-rw-r--r-- 1 jdblischak jdblischak   19M Jan 11 23:02 bam-dedup-umi/19101.3.D02.AGTCCACC.L004.R1.C6WYKACXX.trim.sickle.sorted.rmdup.bam
-rw-r--r-- 1 jdblischak jdblischak   19M Jan 11 23:09 bam-dedup-umi/19098.2.G12.trim.sickle.sorted.combined.rmdup.bam</code></pre>
<p>This is not good! The bam files in <code>bam-dedup-umi</code> are more recent than those in <code>bam-rmdup-umi</code>. It is the files that are in <code>bam-rmdup-umi</code> that get passed to featureCounts. Thus it appears I am passing results from <code>umitools rmdup</code> instead of <code>dedup_umi.py</code>. The creation of the files in <code>bam-dedup-umi</code> on January 11th is consistent with my commit history. I implemented UMI-tools <code>dedup_umi.py</code> in commit <a href="https://github.com/jdblischak/singleCellSeq/commit/2e024cdae1a8e9b3375aa037c3f7deb7c4b60057">2e024cd</a> on January 12th. I know that the results from these two different methods are not drastically different because I compared results from the two, but perhaps this is what is causing these slight discrepancies.</p>
<p>Looking in <code>bam-dedup-umi</code>, it does contain one of the sequences in the molecule file.</p>
<pre class="bash"><code>problem_reads=`grep ENSG00000143294 counts/19098.3.E09.TCCCATGG.L007.R1.C6WYKACXX.trim.sickle.sorted.bam.featureCounts | cut -f1`
for read in $problem_reads
do
  echo &quot;read: $read&quot;
  samtools view bam-dedup-umi/19098.3.E09.TCCCATGG.L007.R1.C6WYKACXX.trim.sickle.sorted.rmdup.bam | grep $read
done
exit 0</code></pre>
<pre><code>read: HWI-700819F:303:C6WYKACXX:7:1104:4034:6156:UMI_GTCTGGGG
read: HWI-700819F:303:C6WYKACXX:7:1201:15253:37760:UMI_GTCTGGGG
read: HWI-700819F:303:C6WYKACXX:7:1205:20480:54449:UMI_GTCTGGGG
read: HWI-700819F:303:C6WYKACXX:7:1208:12235:61153:UMI_GTCTGGGG
HWI-700819F:303:C6WYKACXX:7:1208:12235:61153:UMI_GTCTGGGG   0   chr1    156756791   57  6S86M   *   0   0   GCAGCTCTTCAGGGACGGAACCAGAGCCGGCTTTCCAGGACGATGCAGCCAATGCCCCCCTTGAATTCAAGATGGCAGCAGGTTCAAGTGGG    BFFFFFFIIIIIIIIIIFFFIFIFIIIFIIIFFFFFFFFFFFFFFBFBBFFFBBBB7BBBBBBB0B&lt;&lt;&lt;BBBBBBBBFBFFB0&lt;&lt;0&lt;&lt;00&lt;B    HI:i:1  NH:i:1  NM:i:2
read: HWI-700819F:303:C6WYKACXX:7:1213:13458:22728:UMI_GTCTGGGG
read: HWI-700819F:303:C6WYKACXX:7:2105:9174:45224:UMI_GTCTGGGG
read: HWI-700819F:303:C6WYKACXX:7:2109:16882:12730:UMI_GTCTGGGG
read: HWI-700819F:303:C6WYKACXX:7:2113:2934:40550:UMI_GTCTGGGG
read: HWI-700819F:303:C6WYKACXX:7:2215:7988:79679:UMI_GTCTGGGG</code></pre>
<p>There is only one sequence, consistent with collapsing the 9 reads with the same UMI to one molecule. Focusing in on just this one sequence:</p>
<pre class="bash"><code>samtools view bam-dedup-umi/19098.3.E09.TCCCATGG.L007.R1.C6WYKACXX.trim.sickle.sorted.rmdup.bam | grep HWI-700819F:303:C6WYKACXX:7:1208:12235:61153:UMI_GTCTGGGG</code></pre>
<pre><code>HWI-700819F:303:C6WYKACXX:7:1208:12235:61153:UMI_GTCTGGGG   0   chr1    156756791   57  6S86M   *   0   0   GCAGCTCTTCAGGGACGGAACCAGAGCCGGCTTTCCAGGACGATGCAGCCAATGCCCCCCTTGAATTCAAGATGGCAGCAGGTTCAAGTGGG    BFFFFFFIIIIIIIIIIFFFIFIFIIIFIIIFFFFFFFFFFFFFFBFBBFFFBBBB7BBBBBBB0B&lt;&lt;&lt;BBBBBBBBFBFFB0&lt;&lt;0&lt;&lt;00&lt;B    HI:i:1  NH:i:1  NM:i:2</code></pre>
<p>Searching this sequence with BLAT, it does map to <a href="http://www.genome.ucsc.edu/cgi-bin/hgc?hgsid=476159087_GlgSvATrzrn0XnTPl8co67sJzTrQ&amp;c=chr1&amp;o=156737345&amp;t=156770602&amp;g=ensGene&amp;i=ENST00000353233">ENSG00000143294</a>.</p>
</div>
<div id="next-step" class="section level2">
<h2>Next step</h2>
<p>My plan is to delete all the de-duplicated files and re-run the <a href="http://jdblischak.github.io/singleCellSeq/analysis/process-samples.html#remove-duplicate-umis">de-duplication</a> and <a href="process-samples.html#count-reads-per-gene">gene counting</a> steps. If this doesnâ€™t fix the problem, it will at least make it easier to investigate and diagnose the exact problem.</p>
</div>
<div id="session-information" class="section level2">
<h2>Session information</h2>
<pre class="r"><code>sessionInfo()</code></pre>
<pre><code>R version 3.2.0 (2015-04-16)
Platform: x86_64-unknown-linux-gnu (64-bit)

locale:
 [1] LC_CTYPE=en_US.UTF-8       LC_NUMERIC=C              
 [3] LC_TIME=en_US.UTF-8        LC_COLLATE=en_US.UTF-8    
 [5] LC_MONETARY=en_US.UTF-8    LC_MESSAGES=en_US.UTF-8   
 [7] LC_PAPER=en_US.UTF-8       LC_NAME=C                 
 [9] LC_ADDRESS=C               LC_TELEPHONE=C            
[11] LC_MEASUREMENT=en_US.UTF-8 LC_IDENTIFICATION=C       

attached base packages:
[1] stats     graphics  grDevices utils     datasets  methods   base     

other attached packages:
[1] knitr_1.10.5

loaded via a namespace (and not attached):
 [1] httr_0.6.1       magrittr_1.5     formatR_1.2      htmltools_0.2.6 
 [5] tools_3.2.0      RCurl_1.95-4.6   yaml_2.1.13      codetools_0.2-11
 [9] rmarkdown_0.6.1  stringi_0.4-1    digest_0.6.8     stringr_1.0.0   
[13] bitops_1.0-6     evaluate_0.7    </code></pre>
</div>


<!-- some extra javascript for older browsers -->
<script type="text/javascript" src="libs/polyfill.js"></script>

<script>

// manage active state of menu based on current page
$(document).ready(function () {

    // active menu
    href = window.location.pathname
    href = href.substr(href.lastIndexOf('/') + 1)
    $('a[href="' + href + '"]').parent().addClass('active');

    // manage active menu header
    if (href.startsWith('authoring_'))
      $('a[href="' + 'authoring' + '"]').parent().addClass('active');
    else if (href.endsWith('_format.html'))
      $('a[href="' + 'formats' + '"]').parent().addClass('active');
    else if (href.startsWith('developer_'))
      $('a[href="' + 'developer' + '"]').parent().addClass('active');

});

</script>

</div>

<script>

// add bootstrap table styles to pandoc tables
$(document).ready(function () {
  $('tr.header').parent('thead').parent('table').addClass('table table-condensed');
});

</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://cdn.mathjax.org/mathjax/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
