<!DOCTYPE html>

<html xmlns="http://www.w3.org/1999/xhtml">

<head>

<meta charset="utf-8">
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="generator" content="pandoc" />

<meta name="author" content="Po-Yuan Tung" />

<meta name="date" content="2015-07-06" />

<title>Cell-Cycle Analysis</title>

<script src="libs/jquery-1.11.0/jquery.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="libs/bootstrap-3.3.1/css/united.min.css" rel="stylesheet" />
<script src="libs/bootstrap-3.3.1/js/bootstrap.min.js"></script>
<script src="libs/bootstrap-3.3.1/shim/html5shiv.min.js"></script>
<script src="libs/bootstrap-3.3.1/shim/respond.min.js"></script>

<style type="text/css">

/* padding for bootstrap navbar */
body {
  padding-top: 50px;
  padding-bottom: 40px;
}


/* offset scroll position for anchor links (for fixed navbar)  */
.section h2 {
  padding-top: 55px;
  margin-top: -55px;
}
.section h3 {
  padding-top: 55px;
  margin-top: -55px;
}



/* don't use link color in navbar */
.dropdown-menu>li>a {
  color: black;
}

/* some padding for disqus */
#disqus_thread {
  margin-top: 45px;
}

</style>

<link rel="stylesheet" href="libs/font-awesome-4.1.0/css/font-awesome.min.css"/>

<style type="text/css">code{white-space: pre;}</style>
<link rel="stylesheet"
      href="libs/highlight/textmate.css"
      type="text/css" />
<script src="libs/highlight/highlight.js"></script>
<style type="text/css">
  pre:not([class]) {
    background-color: white;
  }
</style>
<script type="text/javascript">
if (window.hljs && document.readyState && document.readyState === "complete") {
   window.setTimeout(function() {
      hljs.initHighlighting();
   }, 0);
}
</script>



</head>

<body>

<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
code {
  color: inherit;
  background-color: rgba(0, 0, 0, 0.04);
}
img { 
  max-width:100%; 
  height: auto; 
}
</style>
<div class="container-fluid main-container">


<div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">singleCellSeq</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li><a href="index.html">Home</a></li>
        <li><a href="about.html">About</a></li>
        <li><a href="license.html">License</a></li>
        <li><a href="https://github.com/jdblischak/singleCellSeq">GitHub</a></li>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<div id="header">
<h1 class="title">Cell-Cycle Analysis</h1>
<h4 class="author"><em>Po-Yuan Tung</em></h4>
<h4 class="date"><em>2015-07-06</em></h4>
</div>

<div id="TOC">
<ul>
<li><a href="#input">Input</a></li>
<li><a href="#create-phase-scores-for-each-cell">Create phase scores for each cell</a></li>
<li><a href="#two-step-normalization-of-the-phase-specific-scores">Two step normalization of the phase-specific scores</a></li>
<li><a href="#assign-phase-for-each-cell-based-on-phase-specific-score">Assign phase for each cell based on phase-specific score</a></li>
<li><a href="#total-molecule-counts-and-reads-counts-of-each-cell">Total molecule counts and reads counts of each cell</a></li>
<li><a href="#use-the-original-assigned-phases">Use the original assigned phases</a></li>
<li><a href="#the-ratios-of-endogenous-gene-and-ercc">The ratios of endogenous gene and ERCC</a></li>
<li><a href="#satuation-or-not">Satuation or not</a></li>
<li><a href="#session-information">Session information</a></li>
</ul>
</div>

<p><strong>Last updated:</strong> 2015-09-28</p>
<p><strong>Code version:</strong> 3f442a23f405d3564d9ab3197ec337df22fe9383</p>
<p>Carry out cell cycle analysis of single cell iPSCs using the method by <a href="http://www.sciencedirect.com/science/article/pii/S0092867415005498">Macosko2015</a></p>
<p>Gene sets reflecting five phases of the HeLa cell cycle (G1/S, S, G2/M, M and M/G1) were taken from Whitfield et al. (Whitfield et al., 2002) (Table S2), and refined by examining the correlation between the expression pattern of each gene and the average expression pattern of all genes in the respective gene-set, and excluding genes with a low correlation (R&lt;0.3). This step removed genes that were identified as phase-specific in HeLa cells but did not correlate with that phase in our single-cell data. The remaining genes in each refined gene-set were highly correlated (not shown). We then averaged the normalized expression levels (log2(TPM+1)) of the genes in each gene-set to define the phase-specific scores of each cell. These scores were then subjected to two normalization steps. First, for each phase, the scores were centered and divided by their standard deviation. Second, the normalized scores of each cell were centered and normalized.</p>
<div id="input" class="section level2">
<h2>Input</h2>
<pre class="r"><code>library(&quot;dplyr&quot;)
library(&quot;ggplot2&quot;)
theme_set(theme_bw(base_size = 16))
library(&quot;edgeR&quot;)
library(&quot;gplots&quot;)</code></pre>
<p>Input annotation.</p>
<pre class="r"><code>anno &lt;- read.table(&quot;../data/annotation.txt&quot;, header = TRUE,
                   stringsAsFactors = FALSE)
head(anno)</code></pre>
<pre><code>  individual batch well     sample_id
1      19098     1  A01 NA19098.1.A01
2      19098     1  A02 NA19098.1.A02
3      19098     1  A03 NA19098.1.A03
4      19098     1  A04 NA19098.1.A04
5      19098     1  A05 NA19098.1.A05
6      19098     1  A06 NA19098.1.A06</code></pre>
<p>Summary counts from featureCounts. Created with <a href="https://github.com/jdblischak/singleCellSeq/blob/master/code/gather-summary-counts.py">gather-summary-counts.py</a>. These data were collected from the summary files of the full combined samples.</p>
<pre class="r"><code>summary_counts &lt;- read.table(&quot;../data/summary-counts.txt&quot;, header = TRUE,
                             stringsAsFactors = FALSE)</code></pre>
<p>Currently this file only contains data from sickle-trimmed reads, so the code below simply ensures this and then removes the column.</p>
<pre class="r"><code>summary_per_sample &lt;- summary_counts %&gt;%
  filter(sickle == &quot;quality-trimmed&quot;) %&gt;%
  select(-sickle) %&gt;%
  arrange(individual, batch, well, rmdup) %&gt;%
  as.data.frame

summary_per_sample_reads &lt;- summary_per_sample %&gt;% filter(rmdup == &quot;reads&quot;)
summary_per_sample_reads$sample_id &lt;- anno$sample_id</code></pre>
<p>Input read counts.</p>
<pre class="r"><code>reads &lt;- read.table(&quot;../data/reads.txt&quot;, header = TRUE,
                    stringsAsFactors = FALSE)</code></pre>
<p>Input molecule counts.</p>
<pre class="r"><code>molecules &lt;- read.table(&quot;../data/molecules.txt&quot;, header = TRUE,
                    stringsAsFactors = FALSE)</code></pre>
<p>Input list of quality single cells.</p>
<pre class="r"><code>quality_single_cells &lt;- scan(&quot;../data/quality-single-cells.txt&quot;,
                             what = &quot;character&quot;)</code></pre>
<p>Keep only the single cells that passed the <a href="qc-cell-ipsc.html">QC filters</a> and the bulk samples.</p>
<pre class="r"><code>molecules &lt;- molecules[, grepl(&quot;bulk&quot;, colnames(molecules)) |
                         colnames(molecules) %in% quality_single_cells]
anno &lt;- anno[anno$well == &quot;bulk&quot; | anno$sample_id %in% quality_single_cells, ]
stopifnot(ncol(molecules) == nrow(anno),
          colnames(molecules) == anno$sample_id)

reads &lt;- reads[, grepl(&quot;bulk&quot;, colnames(reads)) |
                         colnames(reads) %in% quality_single_cells]
stopifnot(ncol(reads) == nrow(anno),
          colnames(reads) == anno$sample_id)

summary_per_sample_reads_qc &lt;- summary_per_sample_reads[summary_per_sample_reads$sample_id %in% anno$sample_id,]
stopifnot(summary_per_sample_reads_qc$sample_id == anno$sample_id,
          colnames(reads) == summary_per_sample_reads_qc$sample_id)</code></pre>
<p>Remove genes with zero read counts in the single cells or bulk samples.</p>
<pre class="r"><code>expressed &lt;- rowSums(molecules[, anno$well == &quot;bulk&quot;]) &gt; 0 &amp;
             rowSums(molecules[, anno$well != &quot;bulk&quot;]) &gt; 0
molecules &lt;- molecules[expressed, ]
dim(molecules)</code></pre>
<pre><code>[1] 17153   587</code></pre>
<pre class="r"><code>expressed &lt;- rowSums(reads[, anno$well == &quot;bulk&quot;]) &gt; 0 &amp;
             rowSums(reads[, anno$well != &quot;bulk&quot;]) &gt; 0
reads &lt;- reads[expressed, ]
dim(reads)</code></pre>
<pre><code>[1] 17162   587</code></pre>
<p>Keep only single samples.</p>
<pre class="r"><code>molecules_single &lt;- molecules %&gt;% select(-contains(&quot;bulk&quot;))
reads_single &lt;- reads %&gt;% select(-contains(&quot;bulk&quot;))</code></pre>
<p>Remove genes with max molecule numer larger than 1024</p>
<pre class="r"><code>molecules_single &lt;- molecules_single[apply(molecules_single,1,max) &lt; 1024,]</code></pre>
<p>Input cell cycle gene Gene sets reflecting 5 cell cycle phases were taken from Table S2 of <a href="http://www.sciencedirect.com/science/article/pii/S0092867415005498">Macosko2015</a> Gene ID conversion was done by using the DAVID <a href="http://david.abcc.ncifcrf.gov" class="uri">http://david.abcc.ncifcrf.gov</a></p>
<pre class="r"><code>cell_cycle_genes &lt;- read.table(&quot;../data/cellcyclegenes.txt&quot;, header = TRUE, sep=&quot;\t&quot;)

## create 5 lists of 5 phases (de-level and then remove &quot;&quot;)
cell_cycle_genes_list &lt;- lapply(1:5,function(x){
  temp &lt;- as.character(cell_cycle_genes[,x])
  temp[temp!=&quot;&quot;]
})</code></pre>
</div>
<div id="create-phase-scores-for-each-cell" class="section level2">
<h2>Create phase scores for each cell</h2>
<pre class="r"><code>ans &lt;-
sapply(cell_cycle_genes_list,function(xx){
  #### create table of each phase
  reads_single_phase &lt;- reads_single[rownames(reads_single) %in% unlist(xx) ,]
  #### add average expression of all genes in the phase
  combined_matrix &lt;- rbind(reads_single_phase,average=apply(reads_single_phase,2,mean))
  #### use transpose to compute cor matrix
  cor_matrix &lt;- cor(t(combined_matrix))
  #### take the numbers
  cor_vector &lt;- cor_matrix[,dim(cor_matrix)[1]]
  #### restrict to correlation &gt;= 0.3 
  reads_single_phase_restricted &lt;- reads_single_phase[rownames(reads_single_phase) %in% names(cor_vector[cor_vector &gt;= 0.3]),]
  #### apply normalization to reads
  norm_factors_single &lt;- calcNormFactors(reads_single_phase_restricted, method = &quot;TMM&quot;)
  reads_single_cpm &lt;- cpm(reads_single_phase_restricted, log = TRUE,
                            lib.size = colSums(reads_single) * norm_factors_single)
  #### output the phase specific scores (mean of normalized expression levels in the phase)
  apply(reads_single_cpm,2,mean)

})

head(ans)</code></pre>
<pre><code>                  [,1]     [,2]     [,3]     [,4]     [,5]
NA19098.1.A01 6.155267 5.792305 4.765536 5.263269 7.369384
NA19098.1.A02 6.708742 5.997631 5.422097 5.699212 7.579759
NA19098.1.A04 7.006426 5.932785 5.136684 5.351807 7.282850
NA19098.1.A05 7.265333 6.942455 5.626943 5.667291 7.806992
NA19098.1.A06 6.519006 7.043136 5.871447 6.564793 7.839005
NA19098.1.A07 7.293232 6.441728 5.278740 5.859825 7.910691</code></pre>
</div>
<div id="two-step-normalization-of-the-phase-specific-scores" class="section level2">
<h2>Two step normalization of the phase-specific scores</h2>
<pre class="r"><code>#### normalization function
flexible_normalization &lt;- function(data_in,by_row=TRUE){
  if(by_row){
    row_mean &lt;- apply(data_in,1,mean)
    row_sd   &lt;- apply(data_in,1,sd)
    output &lt;- data_in
    for(i in 1:dim(data_in)[1]){
      output[i,] &lt;- (data_in[i,] - row_mean[i])/row_sd[i]
    }
  }
  #### if by column
  if(!by_row){
    col_mean &lt;- apply(data_in,2,mean)
    col_sd   &lt;- apply(data_in,2,sd)
    output &lt;- data_in
    for(i in 1:dim(data_in)[2]){
      output[,i] &lt;- (data_in[,i] - col_mean[i])/col_sd[i]
    }
  }
  output
}

#### apply the normalization function
## first normalized for each phase
ans_normed &lt;- flexible_normalization(ans,by_row=FALSE)
## then normalized of each cell
ans_normed_normed &lt;- flexible_normalization(ans_normed,by_row=TRUE)

head(ans_normed_normed)</code></pre>
<pre><code>                    [,1]       [,2]        [,3]       [,4]       [,5]
NA19098.1.A01 -0.3262126 -1.4409558  0.03914774  0.4623441  1.2656766
NA19098.1.A02  0.4918863 -1.7310702  0.29151567  0.1366215  0.8110468
NA19098.1.A04  1.5958666 -1.1564594  0.03792022 -0.2236181 -0.2537094
NA19098.1.A05  1.0606822  0.4971191 -0.67722194 -1.3840816  0.5035022
NA19098.1.A06 -1.6641313  0.7088817 -0.20733444  0.6497222  0.5128619
NA19098.1.A07  1.1268010 -0.8172340 -0.89233759 -0.4490102  1.0317808</code></pre>
<pre class="r"><code>summary(ans_normed_normed)</code></pre>
<pre><code>       V1                 V2                 V3          
 Min.   :-1.77787   Min.   :-1.76589   Min.   :-1.74602  
 1st Qu.:-0.90666   1st Qu.:-0.71638   1st Qu.:-0.67815  
 Median : 0.08765   Median :-0.01321   Median :-0.08626  
 Mean   : 0.04186   Mean   :-0.00684   Mean   :-0.03876  
 3rd Qu.: 1.00655   3rd Qu.: 0.66862   3rd Qu.: 0.63972  
 Max.   : 1.77266   Max.   : 1.75940   Max.   : 1.69325  
       V4                   V5           
 Min.   :-1.5758929   Min.   :-1.764853  
 1st Qu.:-0.6757738   1st Qu.:-0.706826  
 Median : 0.0486712   Median : 0.010415  
 Mean   :-0.0001804   Mean   : 0.003921  
 3rd Qu.: 0.6731066   3rd Qu.: 0.735029  
 Max.   : 1.6948942   Max.   : 1.765184  </code></pre>
<pre class="r"><code>heatmap.2(ans_normed_normed, trace=&quot;none&quot;, cexRow=1,cexCol=1,margins=c(8,8),xlab=&quot;cell-cycle phase&quot;, ylab= &quot;cells&quot;)</code></pre>
<p><img src="figure/cell-cycle.Rmd/normalization-1.png" title="" alt="" width="672" style="display: block; margin: auto;" /></p>
</div>
<div id="assign-phase-for-each-cell-based-on-phase-specific-score" class="section level2">
<h2>Assign phase for each cell based on phase-specific score</h2>
<pre class="r"><code>cell_phase &lt;- apply(ans_normed_normed,1,function(x) colnames(cell_cycle_genes)[which.max(x)])
assign_cell_phase &lt;- data.frame(cell_phase)

## plot
phase_order &lt;- c(&quot;G1.S&quot;,&quot;S&quot;,&quot;G2.M&quot;,&quot;M&quot;,&quot;M.G1&quot;)
cbPalette &lt;- c(&quot;#999999&quot;, &quot;#E69F00&quot;, &quot;#56B4E9&quot;, &quot;#009E73&quot;, &quot;#F0E442&quot;, &quot;#0072B2&quot;, &quot;#D55E00&quot;, &quot;#CC79A7&quot;)
ggplot(assign_cell_phase, aes(x=cell_phase)) + geom_histogram(aes(fill=factor(cell_phase))) + scale_x_discrete(limits=phase_order, labels=c(&quot;G1/S&quot;,&quot;S&quot;,&quot;G2/M&quot;,&quot;M&quot;,&quot;M/G1&quot;)) + scale_fill_manual(values=cbPalette, breaks=phase_order)</code></pre>
<p><img src="figure/cell-cycle.Rmd/assign-phase-1.png" title="" alt="" width="672" style="display: block; margin: auto;" /></p>
</div>
<div id="total-molecule-counts-and-reads-counts-of-each-cell" class="section level2">
<h2>Total molecule counts and reads counts of each cell</h2>
<pre class="r"><code>anno_single &lt;- anno %&gt;% filter(well != &quot;bulk&quot;)
anno_single$cell_phase &lt;- assign_cell_phase$cell_phase
anno_single$total_reads &lt;- apply(reads_single, 2, sum)
anno_single$total_molecules &lt;- apply(molecules_single, 2, sum)

b &lt;- ggplot(anno_single, aes(x = as.factor(cell_phase), y = total_molecules)) + geom_boxplot(aes(fill=factor(cell_phase))) + scale_x_discrete(limits=phase_order, labels=c(&quot;G1/S&quot;,&quot;S&quot;,&quot;G2/M&quot;,&quot;M&quot;,&quot;M/G1&quot;)) + scale_fill_manual(values=cbPalette, breaks=phase_order) + theme(axis.text.x = element_text(angle = 90, hjust = 0.9, vjust = 0.5)) + xlab(&quot;Cell cycle phase&quot;) + ylab(&quot;Total molecules&quot;)
b</code></pre>
<p><img src="figure/cell-cycle.Rmd/counts-1.png" title="" alt="" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>b + facet_grid(individual ~ batch) </code></pre>
<p><img src="figure/cell-cycle.Rmd/counts-2.png" title="" alt="" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>p &lt;- ggplot(anno_single, aes(x = total_reads, y = total_molecules, col = cell_phase)) + geom_point(size = 3, alpha = 0.5, fontface=3) + scale_colour_manual(values=cbPalette, breaks=phase_order) + theme(axis.text.x = element_text(angle = 90, hjust = 0.9, vjust = 0.5)) + xlab(&quot;Total reads&quot;) + ylab(&quot;Total molecules&quot;)
p</code></pre>
<p><img src="figure/cell-cycle.Rmd/counts-3.png" title="" alt="" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>p + facet_grid(individual ~ batch) </code></pre>
<p><img src="figure/cell-cycle.Rmd/counts-4.png" title="" alt="" width="672" style="display: block; margin: auto;" /></p>
</div>
<div id="use-the-original-assigned-phases" class="section level2">
<h2>Use the original assigned phases</h2>
<p>In the original paper that assigned genes to each phase in <a href="http://www.molbiolcell.org/content/13/6/1977.long">Whitfield2002</a> , they use G1/S, S, G2, G2/M, M/G1 instead of G1/S, S, G2/M, M, M/G1 in <a href="http://www.sciencedirect.com/science/article/pii/S0092867415005498">Macosko2015</a></p>
<pre class="r"><code>ggplot(assign_cell_phase, aes(x=cell_phase)) + geom_histogram(aes(fill=factor(cell_phase))) + scale_x_discrete(limits=phase_order, labels=c(&quot;G1/S&quot;,&quot;S&quot;,&quot;G2&quot;,&quot;G2/M&quot;,&quot;M/G1&quot;)) + scale_fill_manual(values=cbPalette, breaks=phase_order, labels=c(&quot;G1/S&quot;,&quot;S&quot;,&quot;G2&quot;,&quot;G2/M&quot;,&quot;M/G1&quot;))</code></pre>
<p><img src="figure/cell-cycle.Rmd/original-phase-1.png" title="" alt="" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>b &lt;- ggplot(anno_single, aes(x = as.factor(cell_phase), y = total_molecules)) + geom_boxplot(aes(fill=factor(cell_phase))) + scale_x_discrete(limits=phase_order, labels=c(&quot;G1/S&quot;,&quot;S&quot;,&quot;G2&quot;,&quot;G2/M&quot;,&quot;M/G1&quot;)) + scale_fill_manual(values=cbPalette, breaks=phase_order, labels=c(&quot;G1/S&quot;,&quot;S&quot;,&quot;G2&quot;,&quot;G2/M&quot;,&quot;M/G1&quot;)) + theme(axis.text.x = element_text(angle = 90, hjust = 0.9, vjust = 0.5)) + xlab(&quot;Cell cycle phase&quot;) + ylab(&quot;Total molecules&quot;)
b</code></pre>
<p><img src="figure/cell-cycle.Rmd/original-phase-2.png" title="" alt="" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>b + facet_grid(individual ~ batch) </code></pre>
<p><img src="figure/cell-cycle.Rmd/original-phase-3.png" title="" alt="" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>p &lt;- ggplot(anno_single, aes(x = total_reads, y = total_molecules, col = cell_phase)) + geom_point(size = 3, alpha = 0.5, fontface=3) + scale_colour_manual(values=cbPalette, breaks=phase_order, labels=c(&quot;G1/S&quot;,&quot;S&quot;,&quot;G2&quot;,&quot;G2/M&quot;,&quot;M/G1&quot;)) + theme(axis.text.x = element_text(angle = 90, hjust = 0.9, vjust = 0.5)) + xlab(&quot;Total reads&quot;) + ylab(&quot;Total molecules&quot;)
p</code></pre>
<p><img src="figure/cell-cycle.Rmd/original-phase-4.png" title="" alt="" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>p + facet_grid(individual ~ batch) </code></pre>
<p><img src="figure/cell-cycle.Rmd/original-phase-5.png" title="" alt="" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>p + facet_grid(individual ~ cell_phase)</code></pre>
<p><img src="figure/cell-cycle.Rmd/original-phase-6.png" title="" alt="" width="672" style="display: block; margin: auto;" /></p>
</div>
<div id="the-ratios-of-endogenous-gene-and-ercc" class="section level2">
<h2>The ratios of endogenous gene and ERCC</h2>
<ul>
<li>Hypothesis: cells with more mRNA molecule (presumbely G2 phase) will have higher endogenous gene ratio and lower ERCC ratio.</li>
<li>Definition: endogenous gene ratios (total gene molecule/ total molecule) ERCC ratio (total ERCC molecule/ total molecule) total reads (total number of reads from both ERCC and endogenous genes) total molecules (total number of molecules from both ERCC and endogenous genes)</li>
<li>Conclusion:</li>
</ul>
<ol style="list-style-type: decimal">
<li>ERCC ratios strongly corrlate with total molecules G2, G2/M have lower ERCC ratios than others</li>
</ol>
<pre class="r"><code># remove bulk
summary_per_sample_reads_single &lt;- summary_per_sample_reads_qc[summary_per_sample_reads_qc$well!=&quot;bulk&quot;,]

# create total mapped reads
summary_per_sample_reads_single$total_mapped &lt;- apply(summary_per_sample_reads_single[,5:8],1,sum)

# total ERCC molecules 
summary_per_sample_reads_single$total_ERCC_mol &lt;- apply(molecules_single[grep(&quot;ERCC&quot;, rownames(molecules_single)), ],2,sum)

# creat ERCC molecule ratios
summary_per_sample_reads_single$ERCC_mol_ratios &lt;- apply(molecules_single[grep(&quot;ERCC&quot;, rownames(molecules_single)), ],2,sum)/apply(molecules_single,2,sum)

# creat endogenous gene molecule
summary_per_sample_reads_single$total_gene_mol &lt;- apply(molecules_single[grep(&quot;ENSG&quot;, rownames(molecules_single)), ],2,sum)

# creat endogenous gene molecule ratios
summary_per_sample_reads_single$gene_mol_ratios &lt;- apply(molecules_single[grep(&quot;ENSG&quot;, rownames(molecules_single)), ],2,sum)/apply(molecules_single,2,sum)

# plot
ggplot(summary_per_sample_reads_single, aes(x = total_mapped, y = ERCC_mol_ratios, col = as.factor(individual), shape = as.factor(batch))) + geom_point(size = 3, alpha = 0.5)</code></pre>
<p><img src="figure/cell-cycle.Rmd/ratios-1.png" title="" alt="" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>ggplot(summary_per_sample_reads_single, aes(x = total_mapped, y = gene_mol_ratios, col = as.factor(individual), shape = as.factor(batch))) + geom_point(size = 3, alpha = 0.5)</code></pre>
<p><img src="figure/cell-cycle.Rmd/ratios-2.png" title="" alt="" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>anno_single$gene_mol_ratios &lt;- summary_per_sample_reads_single$gene_mol_ratios
anno_single$ERCC_mol_ratios &lt;- summary_per_sample_reads_single$ERCC_mol_ratios

ggplot(anno_single, aes(x = gene_mol_ratios, y = total_molecules, col = cell_phase)) + geom_point(size = 3, alpha = 0.5, fontface=3) + scale_colour_manual(values=cbPalette, breaks=phase_order, labels=c(&quot;G1/S&quot;,&quot;S&quot;,&quot;G2&quot;,&quot;G2/M&quot;,&quot;M/G1&quot;)) + theme(axis.text.x = element_text(angle = 90, hjust = 0.9, vjust = 0.5)) + xlab(&quot;total endogenou gene molecule/total molecule&quot;) + ylab(&quot;Total molecules&quot;)</code></pre>
<p><img src="figure/cell-cycle.Rmd/cell-cycle-ratio-1.png" title="" alt="" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>ggplot(anno_single, aes(x = ERCC_mol_ratios, y = total_molecules, col = cell_phase)) + geom_point(size = 3, alpha = 0.5, fontface=3) + scale_colour_manual(values=cbPalette, breaks=phase_order, labels=c(&quot;G1/S&quot;,&quot;S&quot;,&quot;G2&quot;,&quot;G2/M&quot;,&quot;M/G1&quot;)) + theme(axis.text.x = element_text(angle = 90, hjust = 0.9, vjust = 0.5)) + xlab(&quot;total ERCC molecule/total molecule&quot;) + ylab(&quot;Total molecules&quot;)</code></pre>
<p><img src="figure/cell-cycle.Rmd/cell-cycle-ratio-2.png" title="" alt="" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code># box plots
r &lt;- ggplot(anno_single, aes(x = as.factor(cell_phase), y = gene_mol_ratios)) + geom_boxplot(aes(fill=factor(cell_phase))) + scale_x_discrete(limits=phase_order, labels=c(&quot;G1/S&quot;,&quot;S&quot;,&quot;G2&quot;,&quot;G2/M&quot;,&quot;M/G1&quot;)) + scale_fill_manual(values=cbPalette, breaks=phase_order, labels=c(&quot;G1/S&quot;,&quot;S&quot;,&quot;G2&quot;,&quot;G2/M&quot;,&quot;M/G1&quot;)) + theme(axis.text.x = element_text(angle = 90, hjust = 0.9, vjust = 0.5)) + xlab(&quot;Cell cycle phase&quot;) + ylab(&quot;total endogenou gene molecule/total molecules&quot;)
r</code></pre>
<p><img src="figure/cell-cycle.Rmd/cell-cycle-ratio-3.png" title="" alt="" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>r + facet_grid(individual ~ batch) </code></pre>
<p><img src="figure/cell-cycle.Rmd/cell-cycle-ratio-4.png" title="" alt="" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>e &lt;- ggplot(anno_single, aes(x = as.factor(cell_phase), y= ERCC_mol_ratios)) + geom_boxplot(aes(fill=factor(cell_phase))) + scale_x_discrete(limits=phase_order, labels=c(&quot;G1/S&quot;,&quot;S&quot;,&quot;G2&quot;,&quot;G2/M&quot;,&quot;M/G1&quot;)) + scale_fill_manual(values=cbPalette, breaks=phase_order, labels=c(&quot;G1/S&quot;,&quot;S&quot;,&quot;G2&quot;,&quot;G2/M&quot;,&quot;M/G1&quot;)) + theme(axis.text.x = element_text(angle = 90, hjust = 0.9, vjust = 0.5)) + xlab(&quot;Cell cycle phase&quot;) + ylab(&quot;total ERCC molecule/total molecules&quot;)
e</code></pre>
<p><img src="figure/cell-cycle.Rmd/cell-cycle-ratio-5.png" title="" alt="" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>e + facet_grid(individual ~ batch) </code></pre>
<p><img src="figure/cell-cycle.Rmd/cell-cycle-ratio-6.png" title="" alt="" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code># density plots
d &lt;- ggplot(anno_single, aes(x = gene_mol_ratios, fill = cell_phase)) + geom_density(alpha = 0.5)
d + facet_grid(individual ~ batch)</code></pre>
<p><img src="figure/cell-cycle.Rmd/cell-cycle-ratio-7.png" title="" alt="" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>d + facet_grid(cell_phase ~ individual)</code></pre>
<p><img src="figure/cell-cycle.Rmd/cell-cycle-ratio-8.png" title="" alt="" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>s &lt;- ggplot(anno_single, aes(x = ERCC_mol_ratios, fill = cell_phase)) + geom_density(alpha = 0.5)
s + facet_grid(individual ~ batch)</code></pre>
<p><img src="figure/cell-cycle.Rmd/cell-cycle-ratio-9.png" title="" alt="" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>s + facet_grid(cell_phase ~ individual)</code></pre>
<p><img src="figure/cell-cycle.Rmd/cell-cycle-ratio-10.png" title="" alt="" width="672" style="display: block; margin: auto;" /></p>
</div>
<div id="satuation-or-not" class="section level2">
<h2>Satuation or not</h2>
<p>Check if total endogenous gene molecule or total ERCC molecule is reaching the satuation point!</p>
<pre class="r"><code># add total ERCC molecule
anno_single$total_ERCC_molecule &lt;- apply(molecules_single[grep(&quot;ERCC&quot;, rownames(molecules_single)), ],2,sum)

anno_single$total_gene_molecule &lt;- apply(molecules_single[grep(&quot;ENSG&quot;, rownames(molecules_single)), ],2,sum)

total_gene &lt;- ggplot(anno_single, aes(x = total_gene_molecule, y = total_reads, col = cell_phase)) + geom_point(size = 3, alpha = 0.5, fontface=3) + scale_colour_manual(values=cbPalette, breaks=phase_order, labels=c(&quot;G1/S&quot;,&quot;S&quot;,&quot;G2&quot;,&quot;G2/M&quot;,&quot;M/G1&quot;)) + theme(axis.text.x = element_text(angle = 90, hjust = 0.9, vjust = 0.5)) + xlab(&quot;total gene molecules&quot;) + ylab(&quot;Total reads&quot;)

total_gene</code></pre>
<p><img src="figure/cell-cycle.Rmd/test-test-1.png" title="" alt="" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>total_gene + facet_grid(individual ~ cell_phase) + geom_smooth()</code></pre>
<pre><code>geom_smooth: method=&quot;auto&quot; and size of largest group is &lt;1000, so using loess. Use &#39;method = x&#39; to change the smoothing method.
geom_smooth: method=&quot;auto&quot; and size of largest group is &lt;1000, so using loess. Use &#39;method = x&#39; to change the smoothing method.
geom_smooth: method=&quot;auto&quot; and size of largest group is &lt;1000, so using loess. Use &#39;method = x&#39; to change the smoothing method.
geom_smooth: method=&quot;auto&quot; and size of largest group is &lt;1000, so using loess. Use &#39;method = x&#39; to change the smoothing method.
geom_smooth: method=&quot;auto&quot; and size of largest group is &lt;1000, so using loess. Use &#39;method = x&#39; to change the smoothing method.
geom_smooth: method=&quot;auto&quot; and size of largest group is &lt;1000, so using loess. Use &#39;method = x&#39; to change the smoothing method.
geom_smooth: method=&quot;auto&quot; and size of largest group is &lt;1000, so using loess. Use &#39;method = x&#39; to change the smoothing method.
geom_smooth: method=&quot;auto&quot; and size of largest group is &lt;1000, so using loess. Use &#39;method = x&#39; to change the smoothing method.
geom_smooth: method=&quot;auto&quot; and size of largest group is &lt;1000, so using loess. Use &#39;method = x&#39; to change the smoothing method.
geom_smooth: method=&quot;auto&quot; and size of largest group is &lt;1000, so using loess. Use &#39;method = x&#39; to change the smoothing method.
geom_smooth: method=&quot;auto&quot; and size of largest group is &lt;1000, so using loess. Use &#39;method = x&#39; to change the smoothing method.
geom_smooth: method=&quot;auto&quot; and size of largest group is &lt;1000, so using loess. Use &#39;method = x&#39; to change the smoothing method.
geom_smooth: method=&quot;auto&quot; and size of largest group is &lt;1000, so using loess. Use &#39;method = x&#39; to change the smoothing method.
geom_smooth: method=&quot;auto&quot; and size of largest group is &lt;1000, so using loess. Use &#39;method = x&#39; to change the smoothing method.
geom_smooth: method=&quot;auto&quot; and size of largest group is &lt;1000, so using loess. Use &#39;method = x&#39; to change the smoothing method.</code></pre>
<p><img src="figure/cell-cycle.Rmd/test-test-2.png" title="" alt="" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>total_ERCC &lt;- ggplot(anno_single, aes(x = total_ERCC_molecule, y = total_reads, col = cell_phase)) + geom_point(size = 3, alpha = 0.5, fontface=3) + scale_colour_manual(values=cbPalette, breaks=phase_order, labels=c(&quot;G1/S&quot;,&quot;S&quot;,&quot;G2&quot;,&quot;G2/M&quot;,&quot;M/G1&quot;)) + theme(axis.text.x = element_text(angle = 90, hjust = 0.9, vjust = 0.5)) + xlab(&quot;total ERCC molecules&quot;) + ylab(&quot;Total reads&quot;) 

total_ERCC</code></pre>
<p><img src="figure/cell-cycle.Rmd/test-test-3.png" title="" alt="" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>total_ERCC + facet_grid(individual ~ cell_phase) + geom_smooth() + scale_x_continuous(limits=c(0, 1000))</code></pre>
<pre><code>geom_smooth: method=&quot;auto&quot; and size of largest group is &lt;1000, so using loess. Use &#39;method = x&#39; to change the smoothing method.</code></pre>
<pre><code>Warning: Removed 3 rows containing missing values (stat_smooth).</code></pre>
<pre><code>geom_smooth: method=&quot;auto&quot; and size of largest group is &lt;1000, so using loess. Use &#39;method = x&#39; to change the smoothing method.</code></pre>
<pre><code>Warning: Removed 1 rows containing missing values (stat_smooth).</code></pre>
<pre><code>geom_smooth: method=&quot;auto&quot; and size of largest group is &lt;1000, so using loess. Use &#39;method = x&#39; to change the smoothing method.</code></pre>
<pre><code>Warning: Removed 6 rows containing missing values (stat_smooth).</code></pre>
<pre><code>geom_smooth: method=&quot;auto&quot; and size of largest group is &lt;1000, so using loess. Use &#39;method = x&#39; to change the smoothing method.</code></pre>
<pre><code>Warning: Removed 3 rows containing missing values (stat_smooth).</code></pre>
<pre><code>geom_smooth: method=&quot;auto&quot; and size of largest group is &lt;1000, so using loess. Use &#39;method = x&#39; to change the smoothing method.</code></pre>
<pre><code>Warning: Removed 2 rows containing missing values (stat_smooth).</code></pre>
<pre><code>geom_smooth: method=&quot;auto&quot; and size of largest group is &lt;1000, so using loess. Use &#39;method = x&#39; to change the smoothing method.
geom_smooth: method=&quot;auto&quot; and size of largest group is &lt;1000, so using loess. Use &#39;method = x&#39; to change the smoothing method.
geom_smooth: method=&quot;auto&quot; and size of largest group is &lt;1000, so using loess. Use &#39;method = x&#39; to change the smoothing method.
geom_smooth: method=&quot;auto&quot; and size of largest group is &lt;1000, so using loess. Use &#39;method = x&#39; to change the smoothing method.
geom_smooth: method=&quot;auto&quot; and size of largest group is &lt;1000, so using loess. Use &#39;method = x&#39; to change the smoothing method.
geom_smooth: method=&quot;auto&quot; and size of largest group is &lt;1000, so using loess. Use &#39;method = x&#39; to change the smoothing method.
geom_smooth: method=&quot;auto&quot; and size of largest group is &lt;1000, so using loess. Use &#39;method = x&#39; to change the smoothing method.
geom_smooth: method=&quot;auto&quot; and size of largest group is &lt;1000, so using loess. Use &#39;method = x&#39; to change the smoothing method.
geom_smooth: method=&quot;auto&quot; and size of largest group is &lt;1000, so using loess. Use &#39;method = x&#39; to change the smoothing method.
geom_smooth: method=&quot;auto&quot; and size of largest group is &lt;1000, so using loess. Use &#39;method = x&#39; to change the smoothing method.</code></pre>
<pre><code>Warning: Removed 3 rows containing missing values (geom_point).</code></pre>
<pre><code>Warning: Removed 1 rows containing missing values (geom_point).</code></pre>
<pre><code>Warning: Removed 6 rows containing missing values (geom_point).</code></pre>
<pre><code>Warning: Removed 3 rows containing missing values (geom_point).</code></pre>
<pre><code>Warning: Removed 2 rows containing missing values (geom_point).</code></pre>
<p><img src="figure/cell-cycle.Rmd/test-test-4.png" title="" alt="" width="672" style="display: block; margin: auto;" /></p>
</div>
<div id="session-information" class="section level2">
<h2>Session information</h2>
<pre class="r"><code>sessionInfo()</code></pre>
<pre><code>R version 3.2.0 (2015-04-16)
Platform: x86_64-unknown-linux-gnu (64-bit)

locale:
 [1] LC_CTYPE=en_US.UTF-8       LC_NUMERIC=C              
 [3] LC_TIME=en_US.UTF-8        LC_COLLATE=en_US.UTF-8    
 [5] LC_MONETARY=en_US.UTF-8    LC_MESSAGES=en_US.UTF-8   
 [7] LC_PAPER=en_US.UTF-8       LC_NAME=C                 
 [9] LC_ADDRESS=C               LC_TELEPHONE=C            
[11] LC_MEASUREMENT=en_US.UTF-8 LC_IDENTIFICATION=C       

attached base packages:
[1] stats     graphics  grDevices utils     datasets  methods   base     

other attached packages:
[1] gplots_2.17.0 edgeR_3.10.2  limma_3.24.9  ggplot2_1.0.1 dplyr_0.4.2  
[6] knitr_1.10.5 

loaded via a namespace (and not attached):
 [1] Rcpp_0.12.0        magrittr_1.5       MASS_7.3-40       
 [4] munsell_0.4.2      colorspace_1.2-6   R6_2.1.1          
 [7] stringr_1.0.0      httr_0.6.1         plyr_1.8.3        
[10] caTools_1.17.1     tools_3.2.0        parallel_3.2.0    
[13] grid_3.2.0         gtable_0.1.2       KernSmooth_2.23-14
[16] DBI_0.3.1          gtools_3.5.0       htmltools_0.2.6   
[19] lazyeval_0.1.10    yaml_2.1.13        assertthat_0.1    
[22] digest_0.6.8       reshape2_1.4.1     formatR_1.2       
[25] bitops_1.0-6       RCurl_1.95-4.6     evaluate_0.7      
[28] rmarkdown_0.6.1    labeling_0.3       gdata_2.16.1      
[31] stringi_0.4-1      scales_0.2.4       proto_0.3-10      </code></pre>
</div>


<!-- some extra javascript for older browsers -->
<script type="text/javascript" src="libs/polyfill.js"></script>

<script>

// manage active state of menu based on current page
$(document).ready(function () {

    // active menu
    href = window.location.pathname
    href = href.substr(href.lastIndexOf('/') + 1)
    $('a[href="' + href + '"]').parent().addClass('active');

    // manage active menu header
    if (href.startsWith('authoring_'))
      $('a[href="' + 'authoring' + '"]').parent().addClass('active');
    else if (href.endsWith('_format.html'))
      $('a[href="' + 'formats' + '"]').parent().addClass('active');
    else if (href.startsWith('developer_'))
      $('a[href="' + 'developer' + '"]').parent().addClass('active');

});

</script>

</div>

<script>

// add bootstrap table styles to pandoc tables
$(document).ready(function () {
  $('tr.header').parent('thead').parent('table').addClass('table table-condensed');
});

</script>


</body>
</html>
