---
title: "Cell-Cycle Analysis"
author: "Po-Yuan Tung"
date: 2015-07-06
---

**Last updated:** `r Sys.Date()`

**Code version:** `r system("git log -1 --format='%H'", intern = TRUE)`

```{r chunk-options, include=FALSE}
source("chunk-options.R")
```

Carry out cell cycle analysis of single cell iPSCs using the method by [Macosko2015]

[Macosko2015]:http://www.sciencedirect.com/science/article/pii/S0092867415005498

Gene sets reflecting five phases of the HeLa cell cycle (G1/S, S, G2/M, M and M/G1) were taken from Whitfield et al. (Whitfield et al., 2002) (Table S2), and refined by examining the correlation between the expression pattern of each gene and the average expression pattern of all genes in the respective gene-set, and excluding genes with a low correlation (R<0.3). This step removed genes that were identified as phase-specific in HeLa cells but did not correlate with that phase in our single-cell data. The remaining genes in each refined gene-set were highly correlated (not shown). We then averaged the normalized expression levels (log2(TPM+1)) of the genes in each gene-set to define the phase-specific scores of each cell. These scores were then subjected to two normalization steps. First, for each phase, the scores were centered and divided by their standard deviation. Second, the normalized scores of each cell were centered and normalized.

## Input

```{r packages, message=FALSE}
library("dplyr")
library("ggplot2")
theme_set(theme_bw(base_size = 16))
library("edgeR")
library("gplots")
```

Input annotation.

```{r input-annotation}
anno <- read.table("../data/annotation.txt", header = TRUE,
                   stringsAsFactors = FALSE)
head(anno)
```

Summary counts from featureCounts.
Created with [gather-summary-counts.py](https://github.com/jdblischak/singleCellSeq/blob/master/code/gather-summary-counts.py).
These data were collected from the summary files of the full combined samples.

```{r input-summary-counts}
summary_counts <- read.table("../data/summary-counts.txt", header = TRUE,
                             stringsAsFactors = FALSE)
```

Currently this file only contains data from sickle-trimmed reads, so the code below simply ensures this and then removes the column.

```{r clean-summary-counts}
summary_per_sample <- summary_counts %>%
  filter(sickle == "quality-trimmed") %>%
  select(-sickle) %>%
  arrange(individual, batch, well, rmdup) %>%
  as.data.frame

summary_per_sample_reads <- summary_per_sample %>% filter(rmdup == "reads")
summary_per_sample_reads$sample_id <- anno$sample_id
```

Input read counts.

```{r input-read-counts}
reads <- read.table("../data/reads.txt", header = TRUE,
                    stringsAsFactors = FALSE)
```

Input molecule counts.

```{r input-molecule-counts}
molecules <- read.table("../data/molecules.txt", header = TRUE,
                    stringsAsFactors = FALSE)
```

Input list of quality single cells.

```{r input-quality-single-cells}
quality_single_cells <- scan("../data/quality-single-cells.txt",
                             what = "character")
```

Keep only the single cells that passed the [QC filters](qc-cell-ipsc.html) and the bulk samples.

```{r qc-filter}
molecules <- molecules[, grepl("bulk", colnames(molecules)) |
                         colnames(molecules) %in% quality_single_cells]
anno <- anno[anno$well == "bulk" | anno$sample_id %in% quality_single_cells, ]
stopifnot(ncol(molecules) == nrow(anno),
          colnames(molecules) == anno$sample_id)

reads <- reads[, grepl("bulk", colnames(reads)) |
                         colnames(reads) %in% quality_single_cells]
stopifnot(ncol(reads) == nrow(anno),
          colnames(reads) == anno$sample_id)

summary_per_sample_reads_qc <- summary_per_sample_reads[summary_per_sample_reads$sample_id %in% anno$sample_id,]
stopifnot(summary_per_sample_reads_qc$sample_id == anno$sample_id,
          colnames(reads) == summary_per_sample_reads_qc$sample_id)
```

Remove genes with zero read counts in the single cells or bulk samples.

```{r remove-non-expressed-genes}
expressed <- rowSums(molecules[, anno$well == "bulk"]) > 0 &
             rowSums(molecules[, anno$well != "bulk"]) > 0
molecules <- molecules[expressed, ]
dim(molecules)

expressed <- rowSums(reads[, anno$well == "bulk"]) > 0 &
             rowSums(reads[, anno$well != "bulk"]) > 0
reads <- reads[expressed, ]
dim(reads)
```

Keep only single samples.

```{r keep-single}
molecules_single <- molecules %>% select(-contains("bulk"))
reads_single <- reads %>% select(-contains("bulk"))
```

Remove genes with max molecule numer larger than 1024
```{r remove-1024}
molecules_single <- molecules_single[apply(molecules_single,1,max) < 1024,]
```

Input cell cycle gene
Gene sets reflecting 5 cell cycle phases were taken from Table S2 of [Macosko2015]
Gene ID conversion was done by using the DAVID http://david.abcc.ncifcrf.gov
```{r input-cell-cycle-gene}
cell_cycle_genes <- read.table("../data/cellcyclegenes.txt", header = TRUE, sep="\t")

## create 5 lists of 5 phases (de-level and then remove "")
cell_cycle_genes_list <- lapply(1:5,function(x){
  temp <- as.character(cell_cycle_genes[,x])
  temp[temp!=""]
})
```

## Create phase scores for each cell
```{r phase-score}
ans <-
sapply(cell_cycle_genes_list,function(xx){
  #### create table of each phase
  reads_single_phase <- reads_single[rownames(reads_single) %in% unlist(xx) ,]
  #### add average expression of all genes in the phase
  combined_matrix <- rbind(reads_single_phase,average=apply(reads_single_phase,2,mean))
  #### use transpose to compute cor matrix
  cor_matrix <- cor(t(combined_matrix))
  #### take the numbers
  cor_vector <- cor_matrix[,dim(cor_matrix)[1]]
  #### restrict to correlation >= 0.3 
  reads_single_phase_restricted <- reads_single_phase[rownames(reads_single_phase) %in% names(cor_vector[cor_vector >= 0.3]),]
  #### apply normalization to reads
  norm_factors_single <- calcNormFactors(reads_single_phase_restricted, method = "TMM")
  reads_single_cpm <- cpm(reads_single_phase_restricted, log = TRUE,
                            lib.size = colSums(reads_single) * norm_factors_single)
  #### output the phase specific scores (mean of normalized expression levels in the phase)
  apply(reads_single_cpm,2,mean)

})

head(ans)
```

## Two step normalization of the phase-specific scores
```{r normalization}
#### normalization function
flexible_normalization <- function(data_in,by_row=TRUE){
  if(by_row){
    row_mean <- apply(data_in,1,mean)
    row_sd   <- apply(data_in,1,sd)
    output <- data_in
    for(i in 1:dim(data_in)[1]){
      output[i,] <- (data_in[i,] - row_mean[i])/row_sd[i]
    }
  }
  #### if by column
  if(!by_row){
    col_mean <- apply(data_in,2,mean)
    col_sd   <- apply(data_in,2,sd)
    output <- data_in
    for(i in 1:dim(data_in)[2]){
      output[,i] <- (data_in[,i] - col_mean[i])/col_sd[i]
    }
  }
  output
}

#### apply the normalization function
## first normalized for each phase
ans_normed <- flexible_normalization(ans,by_row=FALSE)
## then normalized of each cell
ans_normed_normed <- flexible_normalization(ans_normed,by_row=TRUE)

head(ans_normed_normed)
summary(ans_normed_normed)
heatmap.2(ans_normed_normed, trace="none", cexRow=1,cexCol=1,margins=c(8,8),xlab="cell-cycle phase", ylab= "cells")
```

## Assign phase for each cell based on phase-specific score
```{r assign-phase}
cell_phase <- apply(ans_normed_normed,1,function(x) colnames(cell_cycle_genes)[which.max(x)])
assign_cell_phase <- data.frame(cell_phase)

## plot
phase_order <- c("G1.S","S","G2.M","M","M.G1")
cbPalette <- c("#999999", "#E69F00", "#56B4E9", "#009E73", "#F0E442", "#0072B2", "#D55E00", "#CC79A7")
ggplot(assign_cell_phase, aes(x=cell_phase)) + geom_histogram(aes(fill=factor(cell_phase))) + scale_x_discrete(limits=phase_order, labels=c("G1/S","S","G2/M","M","M/G1")) + scale_fill_manual(values=cbPalette, breaks=phase_order)
```

## Total molecule counts and reads counts of each cell
```{r counts}
anno_single <- anno %>% filter(well != "bulk")
anno_single$cell_phase <- assign_cell_phase$cell_phase
anno_single$total_reads <- apply(reads_single, 2, sum)
anno_single$total_molecules <- apply(molecules_single, 2, sum)

b <- ggplot(anno_single, aes(x = as.factor(cell_phase), y = total_molecules)) + geom_boxplot(aes(fill=factor(cell_phase))) + scale_x_discrete(limits=phase_order, labels=c("G1/S","S","G2/M","M","M/G1")) + scale_fill_manual(values=cbPalette, breaks=phase_order) + theme(axis.text.x = element_text(angle = 90, hjust = 0.9, vjust = 0.5)) + xlab("Cell cycle phase") + ylab("Total molecules")
b
b + facet_grid(individual ~ batch) 

p <- ggplot(anno_single, aes(x = total_reads, y = total_molecules, col = cell_phase)) + geom_point(size = 3, alpha = 0.5, fontface=3) + scale_colour_manual(values=cbPalette, breaks=phase_order) + theme(axis.text.x = element_text(angle = 90, hjust = 0.9, vjust = 0.5)) + xlab("Total reads") + ylab("Total molecules")
p
p + facet_grid(individual ~ batch) 
```

## Use the original assigned phases
In the original paper that assigned genes to each phase in [Whitfield2002] , they use G1/S, S, G2, G2/M, M/G1 instead of G1/S, S, G2/M, M, M/G1 in [Macosko2015]

[Whitfield2002]:http://www.molbiolcell.org/content/13/6/1977.long

```{r original-phase}
ggplot(assign_cell_phase, aes(x=cell_phase)) + geom_histogram(aes(fill=factor(cell_phase))) + scale_x_discrete(limits=phase_order, labels=c("G1/S","S","G2","G2/M","M/G1")) + scale_fill_manual(values=cbPalette, breaks=phase_order, labels=c("G1/S","S","G2","G2/M","M/G1"))

b <- ggplot(anno_single, aes(x = as.factor(cell_phase), y = total_molecules)) + geom_boxplot(aes(fill=factor(cell_phase))) + scale_x_discrete(limits=phase_order, labels=c("G1/S","S","G2","G2/M","M/G1")) + scale_fill_manual(values=cbPalette, breaks=phase_order, labels=c("G1/S","S","G2","G2/M","M/G1")) + theme(axis.text.x = element_text(angle = 90, hjust = 0.9, vjust = 0.5)) + xlab("Cell cycle phase") + ylab("Total molecules")
b
b + facet_grid(individual ~ batch) 

p <- ggplot(anno_single, aes(x = total_reads, y = total_molecules, col = cell_phase)) + geom_point(size = 3, alpha = 0.5, fontface=3) + scale_colour_manual(values=cbPalette, breaks=phase_order, labels=c("G1/S","S","G2","G2/M","M/G1")) + theme(axis.text.x = element_text(angle = 90, hjust = 0.9, vjust = 0.5)) + xlab("Total reads") + ylab("Total molecules")
p
p + facet_grid(individual ~ batch) 
p + facet_grid(individual ~ cell_phase)
```

## The ratios of endogenous gene and ERCC 
Hypothesis: cells with more mRNA molecule (presumbely G2 phase) will have higher endogenous gene ratio and lower ERCC ratio. 
Definition: 
endogenous gene ratios (total gene reads/ total mapped reads)
ERCC ratio             (total ERCC reads/ total mapped reads)
total reads            (total number of reads from both ERCC and endogenous genes)
total molecules        (total number of molecules from both ERCC and endogenous genes)
Conclusion: 
1. endogenous gene ratios do not corelate with total mapped reads, ERCC ratios, 
   endogenous gene ratios are pretty similar across individual and across cell cycle
2. ERCC ratios strongly corrlate with total molecules
   G2, G2/M have lower ERCC ratios than others

```{r ratios}
# remove bulk
summary_per_sample_reads_single <- summary_per_sample_reads_qc[summary_per_sample_reads_qc$well!="bulk",]

# create total mapped reads
summary_per_sample_reads_single$total_mapped <- apply(summary_per_sample_reads_single[,5:8],1,sum)

# total ERCC reads 
summary_per_sample_reads_single$total_ERCC <- apply(reads_single[grep("ERCC", rownames(reads_single)), ],2,sum)

# creat ERCC ratios
summary_per_sample_reads_single$ERCC_ratios <- apply(reads_single[grep("ERCC", rownames(reads_single)), ],2,sum)/apply(summary_per_sample_reads_single[,5:8],1,sum)

# creat endogenous gene ratios
summary_per_sample_reads_single$gene_ratios <- apply(reads_single[grep("ENSG", rownames(reads_single)), ],2,sum)/apply(summary_per_sample_reads_single[,5:8],1,sum)

# plot
ggplot(summary_per_sample_reads_single, aes(x = total_mapped, y = ERCC_ratios, col = as.factor(individual), shape = as.factor(batch))) + geom_point(size = 3, alpha = 0.5)

ggplot(summary_per_sample_reads_single, aes(x = total_mapped, y = gene_ratios, col = as.factor(individual), shape = as.factor(batch))) + geom_point(size = 3, alpha = 0.5)

ggplot(summary_per_sample_reads_single, aes(x = ERCC_ratios, y = gene_ratios, col = as.factor(individual), shape = as.factor(batch))) + geom_point(size = 3, alpha = 0.5)
```

```{r cell-cycle-ratio}
anno_single$gene_ratios <- summary_per_sample_reads_single$gene_ratios
anno_single$ERCC_ratios <- summary_per_sample_reads_single$ERCC_ratios

ggplot(anno_single, aes(x = gene_ratios, y = total_molecules, col = cell_phase)) + geom_point(size = 3, alpha = 0.5, fontface=3) + scale_colour_manual(values=cbPalette, breaks=phase_order, labels=c("G1/S","S","G2","G2/M","M/G1")) + theme(axis.text.x = element_text(angle = 90, hjust = 0.9, vjust = 0.5)) + xlab("total endogenou gene reads/total mapped") + ylab("Total molecules")

ggplot(anno_single, aes(x = gene_ratios, y = total_reads, col = cell_phase)) + geom_point(size = 3, alpha = 0.5, fontface=3) + scale_colour_manual(values=cbPalette, breaks=phase_order, labels=c("G1/S","S","G2","G2/M","M/G1")) + theme(axis.text.x = element_text(angle = 90, hjust = 0.9, vjust = 0.5)) + xlab("total endogenou gene reads/total mapped") + ylab("Total reads")

ggplot(anno_single, aes(x = ERCC_ratios, y = total_molecules, col = cell_phase)) + geom_point(size = 3, alpha = 0.5, fontface=3) + scale_colour_manual(values=cbPalette, breaks=phase_order, labels=c("G1/S","S","G2","G2/M","M/G1")) + theme(axis.text.x = element_text(angle = 90, hjust = 0.9, vjust = 0.5)) + xlab("total ERCC reads/total mapped") + ylab("Total molecules")

ggplot(anno_single, aes(x = ERCC_ratios, y = total_reads, col = cell_phase)) + geom_point(size = 3, alpha = 0.5, fontface=3) + scale_colour_manual(values=cbPalette, breaks=phase_order, labels=c("G1/S","S","G2","G2/M","M/G1")) + theme(axis.text.x = element_text(angle = 90, hjust = 0.9, vjust = 0.5)) + xlab("total ERCC reads/total mapped") + ylab("Total reads")

# box plots
r <- ggplot(anno_single, aes(x = as.factor(cell_phase), y = gene_ratios)) + geom_boxplot(aes(fill=factor(cell_phase))) + scale_x_discrete(limits=phase_order, labels=c("G1/S","S","G2","G2/M","M/G1")) + scale_fill_manual(values=cbPalette, breaks=phase_order, labels=c("G1/S","S","G2","G2/M","M/G1")) + theme(axis.text.x = element_text(angle = 90, hjust = 0.9, vjust = 0.5)) + xlab("Cell cycle phase") + ylab("total endogenou gene reads/total mapped")
r
r + facet_grid(individual ~ batch) 

e <- ggplot(anno_single, aes(x = as.factor(cell_phase), y= ERCC_ratios)) + geom_boxplot(aes(fill=factor(cell_phase))) + scale_x_discrete(limits=phase_order, labels=c("G1/S","S","G2","G2/M","M/G1")) + scale_fill_manual(values=cbPalette, breaks=phase_order, labels=c("G1/S","S","G2","G2/M","M/G1")) + theme(axis.text.x = element_text(angle = 90, hjust = 0.9, vjust = 0.5)) + xlab("Cell cycle phase") + ylab("total ERCC reads/total mapped")
e
e + facet_grid(individual ~ batch) 

# density plots
d <- ggplot(anno_single, aes(x = gene_ratios, fill = cell_phase)) + geom_density(alpha = 0.5)
d + facet_grid(individual ~ batch)
d + facet_grid(cell_phase ~ individual)

s <- ggplot(anno_single, aes(x = ERCC_ratios, fill = cell_phase)) + geom_density(alpha = 0.5)
s + facet_grid(individual ~ batch)
s + facet_grid(cell_phase ~ individual)
```
## Session information

```{r info}
sessionInfo()
```
