---
title: "Explore distribituions"
author: "Joyce Hsiao"
date: 2016-02-01  
---

**Last updated:** `r Sys.Date()`

**Code version:** `r system("git log -1 --format='%H'", intern = TRUE)`

```{r chunk-options, include=FALSE}
source("chunk-options.R")

library(knitr)
opts_chunk$set(message = FALSE, warning = FALSE, eval = TRUE, 
               echo = TRUE)
```



## Objective

Consider three scenarios for comparing distributions between different individuals 

1. Significant individual difference in noise meausre, with one or more individual
    distributions tightly regulated (one mode).

2. No significant individual difference in noise measure, with all individual distributions
   tightly regulated.

3. No significant individual differences in noise measure, with none of the individual
   distributions show a pattern consistent with tight regulation.



## Set up

```{r, message=FALSE, warning=FALSE}
library("data.table")
library("dplyr")
library("limma")
library("edgeR")
library("ggplot2")
library("grid")
theme_set(theme_bw(base_size = 12))
source("functions.R")
library("Humanzee")
library("cowplot")
library("MASS")
library("matrixStats")
```


## Prepare data

Input annotation of only QC-filtered single cells, with NA19098.r2 removed.

```{r import-annotation}
anno_filter <- read.table("../data/annotation-filter.txt", 
                      header = TRUE,
                      stringsAsFactors = FALSE)
dim(anno_filter)
head(anno_filter, 2)
```

Import molecule counts after filtering and before any correction.

```{r import-molecule-counts-after-filtering}
molecules_filter <- read.table("../data/molecules-filter.txt",
                               header = TRUE, stringsAsFactors = FALSE)
stopifnot(NROW(anno_filter) == NCOL(molecules_filter))
```


Import final processed molecule counts of endogeneous genes.

```{r import-ensg-final-transformed-log2counts}
molecules_final <- read.table("../data/molecules-final.txt", 
                             header = TRUE, stringsAsFactors = FALSE)
dim(molecules_final)
stopifnot(NROW(anno_filter) == NCOL(molecules_final))
```


Import gene symbols.

```{r}
gene_symbols <- read.table(file = "../data/gene-info.txt", sep = "\t",
                           header = TRUE, stringsAsFactors = FALSE, quote = "")
```

Import pluripotent gene list

```{r}
cell_cycle_genes <- read.table("../data/cellcyclegenes.txt",
                               header = TRUE, sep = "\t",
                               stringsAsFactors = FALSE)

pluripotency_genes <- read.table("../data/pluripotency-genes.txt",
                               header = TRUE, sep = "\t",
                               stringsAsFactors = FALSE)$To
```


Import distance meausre

```{r import-median-of-absolute-deviation, eval = TRUE}
load("rda/cv-adjusted-summary-pois/adj-cv.rda")
```


Import permuted p-values

```{r import-permuted-pvals, eval = TRUE}
load("rda/cv-adjusted-summary-pois/permuted-pval.rda")
```


## Helper functions

*plot_density

Per gene plot of overlaied density curves computed from individual cell lines.


```{r overlapping-density-function}
plot_density <- function(molecules_ENSG, annotation,
                         individuals, batches = NULL,
                         which_gene, labels, 
                         xlims = NULL, ylims = NULL, gene_symbols) {
  if_present <- which(rownames(molecules_ENSG) == which_gene)
  if(length(if_present) == 0) {
    stop("Gene not present in the data")
  }
  
  library(scales)
  library(broman)
  crayon <- brocolors("crayon")
  
  if (is.null(batches)) {
    individuals <- unique(annotation$individual)
    colors <- c("Sunset Orange", "Tropical Rain Forest", "Denim")
    dens <- lapply(1:3, function(per_individual) {
      which_individual <- annotation$individual == individuals[per_individual]
      density(unlist( molecules_ENSG[ rownames(molecules_ENSG) == which_gene, 
                                      which_individual] ) )
    })

    if (is.null(xlims)) xlims <- range(sapply(dens, function(obj) obj$x))
    if (is.null(ylims)) ylims <- range(sapply(dens, function(obj) obj$y))
    
    plot(dens[[1]], 
         xlab = "log2 gene expression", main = "",
         ylab = "Density", axes = F, lwd = 0, xlim = xlims, ylim = ylims)
    for (i in 1:length(individuals)) {
      polygon(dens[[i]], 
              col = alpha(crayon[colors[i]], .4), 
              border = "grey40")
    }
    axis(1); axis(2)
    mtext(text = labels, side = 3)
    title(main = with(gene_symbols, 
                      external_gene_name[which(ensembl_gene_id == which_gene)]) )
  }
  
  if (!is.null(batches)) {
    
    colors <- c("Sunset Orange", "Tropical Rain Forest", "Denim")
    individuals <- unique(annotation$individual)
    #    colors <- c("Sunset Orange", "Tropical Rain Forest", "Denim")
    dens <- lapply(1:length(individuals), function(per_individual) {
      which_individual <- annotation$individual == individuals[per_individual]
      annotation_sub <- annotation[which_individual, ]
      molecules_sub <- molecules_ENSG[ , which_individual]
      replicates <- unique(annotation_sub$replicate)
      dens_batch <- lapply(1:length(replicates), function(per_replicate) {
        which_replicate <- annotation_sub$replicate == replicates[per_replicate]
        density(unlist( molecules_sub[ rownames(molecules_ENSG) == which_gene, 
                                       which_replicate] ) )
      })
    })
    
    if (is.null(xlims)) {
      xlims <- range( c( sapply(dens, function(obj_individual) {
        c( sapply(obj_individual, function(obj) {
          range(obj$x)
        }) )
      }) ) )
    }
    if (is.null(ylims)) {
      ylims <- range( c( sapply(dens, function(obj_individual) {
        c( sapply(obj_individual, function(obj) {
          range(obj$y)
        }) )
      }) ) )
    }
    
#    par(mfrow = c(1,3))
    for (i in 1:length(dens)) {
      plot(dens[[i]][[1]], 
           xlab = "log2 gene expression", main = "",
           ylab = "Density", axes = F, lwd = 0, xlim = xlims, ylim = ylims)
      for (j in 1:length(dens[[i]])) {
        polygon(dens[[i]][[j]], 
                col = alpha(crayon[colors[i]], .4), 
                border = "grey40")
      }
    }
    axis(1); axis(2)
    mtext(text = labels, side = 3)
    title(main = with(gene_symbols, 
                      external_gene_name[which(ensembl_gene_id == which_gene)]) )
  }
}
```



## Pluripotency gene distributions

`r length(pluripotency_genes[which(pluripotency_genes %in% rownames(molecules_final))]
)` pluripotency genes are included in the data set after filtering. 



```{r}
pluri_pvals <- data.frame(pvals = permuted_pval,
                          ENSG = rownames(molecules_final))
pluri_pvals <- pluri_pvals[which(rownames(molecules_final) %in% 
                                   pluripotency_genes), ]
pluri_symbols <- gene_symbols[which(gene_symbols$ensembl_gene_id %in% pluri_pvals$ENSG) , 
                              c(1,3)]
pluri_results <- merge(pluri_pvals, pluri_symbols,
                       by.x = c("ENSG"), by.y = "ensembl_gene_id") 
pluri_results <- pluri_results[order(pluri_results$mad_pval), ]
pluri_results
```



### Batch-corrected log2 molecule count

#### Collapsing across batches

```{r pluripotent-density, fig.width=8, fig.height=2.5, eval=T}
par(mfrow = c(1,4))
for (i in 1:nrow(pluri_results)) {
  plot_density(molecules_ENSG = molecules_final,
               annotation = anno_filter,
               which_gene = pluri_results$ENSG[i], 
    #                   labels = round(genes_plot$dist[i], 6),
               labels = "",
               xlims = c(1,14), ylims = NULL,
               gene_symbols = gene_symbols)
}
```



#### One batch at a time

```{r pluripotent-density-replicates, fig.width=8, fig.height=3, eval=T}
for (i in 1:nrow(pluri_results)) {
    par(mfrow = c(1,3))
    for (which_individual in unique(anno_filter$individual)) {
        plot_density(molecules_ENSG = molecules_final[ , anno_filter$individual == which_individual],
                     annotation = anno_filter[anno_filter$individual == which_individual, ],
                     individuals = anno_filter$individual,
                     batches = anno_filter$sample_id,
                     which_gene = pluri_results$ENSG[i],
                     labels = which_individual,
                     xlims = c(2,14),
                     gene_symbols = gene_symbols)
    }
}
```


## Session information

```{r info}
sessionInfo()
```
