<!DOCTYPE html>

<html xmlns="http://www.w3.org/1999/xhtml">

<head>

<meta charset="utf-8">
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="generator" content="pandoc" />


<meta name="date" content="2015-06-11" />

<title>Compare read and molecule counts</title>

<script src="libs/jquery-1.11.0/jquery.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="libs/bootstrap-3.3.1/css/united.min.css" rel="stylesheet" />
<script src="libs/bootstrap-3.3.1/js/bootstrap.min.js"></script>
<script src="libs/bootstrap-3.3.1/shim/html5shiv.min.js"></script>
<script src="libs/bootstrap-3.3.1/shim/respond.min.js"></script>

<style type="text/css">

/* padding for bootstrap navbar */
body {
  padding-top: 50px;
  padding-bottom: 40px;
}


/* offset scroll position for anchor links (for fixed navbar)  */
.section h2 {
  padding-top: 55px;
  margin-top: -55px;
}
.section h3 {
  padding-top: 55px;
  margin-top: -55px;
}



/* don't use link color in navbar */
.dropdown-menu>li>a {
  color: black;
}

/* some padding for disqus */
#disqus_thread {
  margin-top: 45px;
}

</style>

<link rel="stylesheet" href="libs/font-awesome-4.1.0/css/font-awesome.min.css"/>

<style type="text/css">code{white-space: pre;}</style>
<link rel="stylesheet"
      href="libs/highlight/textmate.css"
      type="text/css" />
<script src="libs/highlight/highlight.js"></script>
<style type="text/css">
  pre:not([class]) {
    background-color: white;
  }
</style>
<script type="text/javascript">
if (window.hljs && document.readyState && document.readyState === "complete") {
   window.setTimeout(function() {
      hljs.initHighlighting();
   }, 0);
}
</script>



</head>

<body>

<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
code {
  color: inherit;
  background-color: rgba(0, 0, 0, 0.04);
}
img { 
  max-width:100%; 
  height: auto; 
}
</style>
<div class="container-fluid main-container">


<div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">singleCellSeq</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li><a href="index.html">Home</a></li>
        <li><a href="about.html">About</a></li>
        <li><a href="license.html">License</a></li>
        <li><a href="https://github.com/jdblischak/singleCellSeq">GitHub</a></li>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<div id="header">
<h1 class="title">Compare read and molecule counts</h1>
<h4 class="date"><em>2015-06-11</em></h4>
</div>

<div id="TOC">
<ul>
<li><a href="#input">Input</a></li>
<li><a href="#filter">Filter</a></li>
<li><a href="#sequencing-depth">Sequencing depth</a></li>
<li><a href="#calculate-counts-per-million-cpm">Calculate counts per million (cpm)</a></li>
<li><a href="#compare-reads-and-molecules">Compare reads and molecules</a></li>
<li><a href="#effect-of-sequencing-depth-on-molecule-count">Effect of sequencing depth on molecule count</a><ul>
<li><a href="#pairwise-distance-between-cells">Pairwise distance between cells</a></li>
<li><a href="#reads-to-molecule-conversion-rate">Reads to molecule conversion rate</a></li>
<li><a href="#pca">PCA</a></li>
</ul></li>
<li><a href="#compare-reads-and-standardized-molecules">Compare reads and standardized molecules</a></li>
<li><a href="#session-information">Session information</a></li>
</ul>
</div>

<p><strong>Last updated:</strong> 2015-09-28</p>
<p><strong>Code version:</strong> 3f442a23f405d3564d9ab3197ec337df22fe9383</p>
<p>First, compared counts via three methods:</p>
<ul>
<li><strong>reads_cpm</strong> - standard counts per million</li>
<li><strong>molecules</strong> - counts of molecules identified using UMIs</li>
<li><strong>molecules_per_lane</strong> - counts of molecules identified using UMIs per each sequencing lane and then summed per sample</li>
</ul>
<p>Then investigated the relationship between sequencing depth and total molecule count per sample. Found that sequencing depth affects the total molecule count, which in turn affects PC1. Will use TMM-normalize molecule counts per million mapped (cpm) for downstream analyses.</p>
<p>Therefore reran the original comparisons between reads and molecules, but this time using TMM-normalized counts per million for the molecules similar to the reads. The correlation of the mean expression improved.</p>
<div id="input" class="section level2">
<h2>Input</h2>
<pre class="r"><code>library(&quot;dplyr&quot;)
library(&quot;ggplot2&quot;)
theme_set(theme_bw(base_size = 16))
library(&quot;edgeR&quot;)
source(&quot;functions.R&quot;)</code></pre>
<p>Input annotation.</p>
<pre class="r"><code>anno &lt;- read.table(&quot;../data/annotation.txt&quot;, header = TRUE,
                   stringsAsFactors = FALSE)
head(anno)</code></pre>
<pre><code>  individual batch well     sample_id
1      19098     1  A01 NA19098.1.A01
2      19098     1  A02 NA19098.1.A02
3      19098     1  A03 NA19098.1.A03
4      19098     1  A04 NA19098.1.A04
5      19098     1  A05 NA19098.1.A05
6      19098     1  A06 NA19098.1.A06</code></pre>
<p>Input read counts.</p>
<pre class="r"><code>reads &lt;- read.table(&quot;../data/reads.txt&quot;, header = TRUE,
                    stringsAsFactors = FALSE)</code></pre>
<p>Input molecule counts.</p>
<pre class="r"><code>molecules &lt;- read.table(&quot;../data/molecules.txt&quot;, header = TRUE,
                    stringsAsFactors = FALSE)</code></pre>
<p>Input molecule counts summed across lanes.</p>
<pre class="r"><code>molecules_per_lane &lt;- read.table(&quot;../data/molecules-per-lane.txt&quot;, header = TRUE,
                    stringsAsFactors = FALSE)</code></pre>
<p>Input list of quality single cells.</p>
<pre class="r"><code>quality_single_cells &lt;- scan(&quot;../data/quality-single-cells.txt&quot;,
                             what = &quot;character&quot;)</code></pre>
</div>
<div id="filter" class="section level2">
<h2>Filter</h2>
<p>Keep only the single cells that passed the <a href="qc-cell-ipsc.html">QC filters</a> and the bulk samples.</p>
<pre class="r"><code>reads &lt;- reads[, grepl(&quot;bulk&quot;, colnames(reads)) |
                 colnames(reads) %in% quality_single_cells]
molecules &lt;- molecules[, grepl(&quot;bulk&quot;, colnames(molecules)) |
                         colnames(molecules) %in% quality_single_cells]
molecules_per_lane &lt;- molecules_per_lane[, grepl(&quot;bulk&quot;, colnames(molecules_per_lane)) |
                                           colnames(molecules_per_lane) %in% quality_single_cells]
anno &lt;- anno[anno$well == &quot;bulk&quot; | anno$sample_id %in% quality_single_cells, ]
stopifnot(dim(reads) == dim(molecules),
          nrow(anno) == ncol(molecules_per_lane))</code></pre>
<p>Remove genes with zero read or molecule counts in the single cell or bulk samples.</p>
<pre class="r"><code>expressed &lt;- rowSums(reads[anno$well == &quot;bulk&quot;]) &gt; 0 &amp;
             rowSums(reads[anno$well != &quot;bulk&quot;]) &gt; 0 &amp;
             rowSums(molecules[anno$well == &quot;bulk&quot;]) &gt; 0 &amp;
             rowSums(molecules[anno$well != &quot;bulk&quot;]) &gt; 0
reads &lt;- reads[expressed, ]
molecules &lt;- molecules[expressed, ]
molecules_per_lane &lt;- molecules_per_lane[expressed, ]</code></pre>
</div>
<div id="sequencing-depth" class="section level2">
<h2>Sequencing depth</h2>
<p>Calculate the number of reads per molecule of each gene in each cell.</p>
<pre class="r"><code>reads_per_molecule &lt;- as.matrix(reads/molecules)
hist(reads_per_molecule, breaks=100)</code></pre>
<p><img src="figure/compare-reads-v-molecules.Rmd/seq-depth-1.png" title="" alt="" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>plot(density(reads_per_molecule, na.rm = TRUE))</code></pre>
<p><img src="figure/compare-reads-v-molecules.Rmd/seq-depth-2.png" title="" alt="" width="672" style="display: block; margin: auto;" /></p>
</div>
<div id="calculate-counts-per-million-cpm" class="section level2">
<h2>Calculate counts per million (cpm)</h2>
<p>Calculate cpm for the reads data using TMM-normalization.</p>
<pre class="r"><code>norm_factors_reads &lt;- calcNormFactors(reads, method = &quot;TMM&quot;)
reads_cpm &lt;- cpm(reads, lib.size = colSums(reads) * norm_factors_reads)</code></pre>
<p>And for the molecules.</p>
<pre class="r"><code>norm_factors_mol &lt;- calcNormFactors(molecules, method = &quot;TMM&quot;)
molecules_cpm &lt;- cpm(molecules, lib.size = colSums(molecules) * norm_factors_mol)</code></pre>
<p>And for the molecules summed per lane.</p>
<pre class="r"><code>norm_factors_mol_per_lane &lt;- calcNormFactors(molecules_per_lane, method = &quot;TMM&quot;)
molecules_per_lane_cpm &lt;- cpm(molecules_per_lane,
                              lib.size = colSums(molecules_per_lane) *
                                         norm_factors_mol_per_lane)</code></pre>
</div>
<div id="compare-reads-and-molecules" class="section level2">
<h2>Compare reads and molecules</h2>
<p>Compare the means of each gene obtained via the different methods.</p>
<pre class="r"><code>mean_data &lt;- data.frame(reads_cpm = rowMeans(reads_cpm),
                        molecules = rowMeans(molecules),
                        molecules_per_lane = rowMeans(molecules_per_lane))
cor(mean_data)</code></pre>
<pre><code>                   reads_cpm molecules molecules_per_lane
reads_cpm          1.0000000 0.8909973          0.8769356
molecules          0.8909973 1.0000000          0.9944794
molecules_per_lane 0.8769356 0.9944794          1.0000000</code></pre>
<p>All three are highly correlated.</p>
<pre class="r"><code>mean_data$type &lt;- ifelse(grepl(&quot;ERCC&quot;, rownames(mean_data)), &quot;ERCC&quot;, &quot;gene&quot;)
ggplot(mean_data, aes(x = reads_cpm, y = molecules)) +
  geom_point() +
  geom_smooth(method = &quot;lm&quot;) +
  facet_wrap(~ type)</code></pre>
<p><img src="figure/compare-reads-v-molecules.Rmd/reads-v-molecules-1.png" title="" alt="" width="672" style="display: block; margin: auto;" /></p>
<p>There are only a few genes with molecule counts greater than the number of UMIs.</p>
<pre class="r"><code>rownames(molecules)[rowMeans(molecules) &gt; 1024]</code></pre>
<pre><code>[1] &quot;ENSG00000198712&quot; &quot;ENSG00000198938&quot; &quot;ENSG00000198886&quot;</code></pre>
<p>They are highly expressed mitochondrial genes.</p>
<pre class="r"><code>ggplot(mean_data, aes(x = reads_cpm, y = molecules)) +
  geom_point() +
  geom_smooth(method = &quot;lm&quot;) +
  facet_wrap(~ type) +
  ylim(0, 1100)</code></pre>
<p><img src="figure/compare-reads-v-molecules.Rmd/reads-v-molecules-range-restricted-1.png" title="" alt="" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>ggplot(mean_data, aes(x = reads_cpm, y = molecules_per_lane)) +
  geom_point() +
  geom_smooth(method = &quot;lm&quot;) +
  facet_wrap(~ type)</code></pre>
<p><img src="figure/compare-reads-v-molecules.Rmd/reads-v-molecules-per-lane-1.png" title="" alt="" width="672" style="display: block; margin: auto;" /></p>
<p>The molecule counts and the molecule counts summed per sequencing lane are highly correlated. This indicates that most of the bias is introduced in the library preparation step and not during sequencing.</p>
<pre class="r"><code>ggplot(mean_data, aes(x = molecules_per_lane, y = molecules)) +
  geom_point() +
  geom_smooth(method = &quot;lm&quot;) +
  facet_wrap(~ type)</code></pre>
<p><img src="figure/compare-reads-v-molecules.Rmd/molecules-v-molecules-per-lane-1.png" title="" alt="" width="672" style="display: block; margin: auto;" /></p>
</div>
<div id="effect-of-sequencing-depth-on-molecule-count" class="section level2">
<h2>Effect of sequencing depth on molecule count</h2>
<p>How dependent are the molecule counts on the total molecule count for a given sample? Should we standardize by the total molecule count per sample? <a href="http://www.nature.com/nmeth/journal/v11/n2/full/nmeth.2772.html">Islam et al. 2014</a> argue that this is not necessary, “scales for molecule-counting scatterplots (Fig. 2d,e) are absolute and would not change appreciably if the number of reads were increased.” Let’s check this assumption.</p>
<p>Does the total number of molecules per sample vary with the total number of reads? If it is not necessary to standardize the molecule counts, the molecule counts should be consistent across varying read depths.</p>
<pre class="r"><code>total_counts_data &lt;- data.frame(total_reads = colSums(reads) / 10^6,
                                total_molecules = colSums(molecules) / 10^3,
                                anno)
str(total_counts_data)</code></pre>
<pre><code>&#39;data.frame&#39;:   587 obs. of  6 variables:
 $ total_reads    : num  1.89 1.99 1.27 1.99 1.75 ...
 $ total_molecules: num  93.1 95.1 74.7 104.7 98.2 ...
 $ individual     : int  19098 19098 19098 19098 19098 19098 19098 19098 19098 19098 ...
 $ batch          : int  1 1 1 1 1 1 1 1 1 1 ...
 $ well           : chr  &quot;A01&quot; &quot;A02&quot; &quot;A04&quot; &quot;A05&quot; ...
 $ sample_id      : chr  &quot;NA19098.1.A01&quot; &quot;NA19098.1.A02&quot; &quot;NA19098.1.A04&quot; &quot;NA19098.1.A05&quot; ...</code></pre>
<pre class="r"><code>total_counts_single &lt;- ggplot(total_counts_data[total_counts_data$well != &quot;bulk&quot;, ],
                              aes(x = total_reads, y = total_molecules)) +
  geom_point(aes(col = as.factor(individual), shape = as.factor(batch))) +
  geom_smooth(method = &quot;lm&quot;) +
  labs(x = &quot;Total number of reads (x10^6)&quot;,
       y = &quot;Total number of molecules (x10^3)&quot;,
       title = &quot;Effect of read depth on single cells&quot;)
total_counts_single</code></pre>
<p><img src="figure/compare-reads-v-molecules.Rmd/total-mol-v-reads-single-1.png" title="" alt="" width="768" style="display: block; margin: auto;" /></p>
<pre class="r"><code>total_counts_bulk &lt;- total_counts_single %+%
  total_counts_data[total_counts_data$well == &quot;bulk&quot;, ] +
  labs(x = &quot;Total number of reads (x10^6)&quot;,
       y = &quot;Total number of molecules (x10^3)&quot;,
       title = &quot;Effect of read depth on bulk samples&quot;)
total_counts_bulk</code></pre>
<p><img src="figure/compare-reads-v-molecules.Rmd/total-mol-v-reads-bulk-1.png" title="" alt="" width="672" style="display: block; margin: auto;" /></p>
<p>So this is clearly not the case. Perhaps in the ideal case where all the cells are sequenced to saturation, then any increasing sequencing would not make a difference in the molecule counts.</p>
<p>Also, there is a difference in total molecule count between the three individuals.</p>
<pre class="r"><code>total_molecules_ind &lt;- ggplot(total_counts_data[total_counts_data$well != &quot;bulk&quot;, ],
                              aes(x = as.factor(individual), y = total_molecules)) +
  geom_boxplot(aes(fill = as.factor(batch))) +
  scale_fill_brewer(type = &quot;qual&quot;, palette = &quot;Dark2&quot;, name =&quot;Batch&quot;) +
  labs(x = &quot;Individual&quot;,
       y = &quot;Molecules (x10^3)&quot;,
       title = &quot;Total molecule counts vary across individuals&quot;)
total_molecules_ind</code></pre>
<p><img src="figure/compare-reads-v-molecules.Rmd/total-molecules-by-ind-1.png" title="" alt="" width="672" style="display: block; margin: auto;" /></p>
<p>But not for the total number of reads, as expected from the plot above of the effect of read depth where all three individuals span the x-axis of the total number of reads.</p>
<pre class="r"><code>total_reads_ind &lt;- total_molecules_ind %+% aes(y = total_reads) +
  labs(y = &quot;Reads (x10^6)&quot;,
       title = &quot;Total read counts vary across individuals&quot;)
total_reads_ind</code></pre>
<p><img src="figure/compare-reads-v-molecules.Rmd/total-reads-by-ind-1.png" title="" alt="" width="672" style="display: block; margin: auto;" /></p>
<p>There is a clear difference between individuals. Is there a difference between the full 9 batches?</p>
<pre class="r"><code>total_counts_single_batch &lt;- ggplot(total_counts_data[total_counts_data$well != &quot;bulk&quot;, ],
                              aes(x = total_reads, y = total_molecules,
                                  col = paste(individual, batch, sep = &quot;.&quot;))) +
  geom_point() +
  scale_color_brewer(palette = &quot;Set1&quot;, name = &quot;9 batches&quot;) +
  labs(x = &quot;Total number of reads (x10^6)&quot;,
       y = &quot;Total number of molecules (x10^3)&quot;,
       title = &quot;Effect of read depth on single cells&quot;)
total_counts_single_batch</code></pre>
<p><img src="figure/compare-reads-v-molecules.Rmd/total-mol-v-reads-single-9-batches-1.png" title="" alt="" width="768" style="display: block; margin: auto;" /></p>
<p>It is diffcult to see when all 9 are plotted at once. Here are the batches split by individual.</p>
<pre class="r"><code>total_counts_single_ind &lt;- ggplot(total_counts_data[total_counts_data$well != &quot;bulk&quot;, ],
                              aes(x = total_reads, y = total_molecules)) +
  geom_point(aes(color = as.factor(batch))) +
  facet_wrap(~individual, nrow = 3) +
  scale_color_brewer(palette = &quot;Dark2&quot;, name = &quot;batch&quot;) +
  labs(x = &quot;Total number of reads (x10^6)&quot;,
       y = &quot;Total number of molecules (x10^3)&quot;,
       title = &quot;Effect of read depth on single cells&quot;)
total_counts_single_ind</code></pre>
<p><img src="figure/compare-reads-v-molecules.Rmd/total-mol-v-reads-single-per-ind-1.png" title="" alt="" width="768" style="display: block; margin: auto;" /></p>
<div id="pairwise-distance-between-cells" class="section level3">
<h3>Pairwise distance between cells</h3>
<p>Pairwise distance in (total reads, total molecules) between batches or individuals.</p>
<ul>
<li>Compute pairwise Euclidean distance between cells.</li>
</ul>
<pre class="r"><code>total_counts_single &lt;- total_counts_data[total_counts_data$well != &quot;bulk&quot;, ]
total_counts_single_dist &lt;- as.matrix( dist(total_counts_single[ , c(&quot;total_reads&quot;, &quot;total_molecules&quot;)]) )
rownames(total_counts_single_dist) &lt;- with(total_counts_single,
                                            paste(individual, batch, sep = &quot;_&quot;))
colnames(total_counts_single_dist) &lt;- rownames(total_counts_single_dist)
total_counts_single[1:2, 1:2]</code></pre>
<pre><code>              total_reads total_molecules
NA19098.1.A01    1.892562          93.117
NA19098.1.A02    1.988496          95.125</code></pre>
<pre class="r"><code>ind_index &lt;- (total_counts_single$individual)
ind_batch_index &lt;- with(total_counts_single,paste(individual, batch, sep = &quot;_&quot;))

same_ind_index &lt;- outer(ind_index,ind_index,function(x,y) x==y)
same_batch_index &lt;- outer(ind_batch_index,ind_batch_index,function(x,y) x==y)

dim_temp &lt;- dim(total_counts_single_dist)
dist_index_matrix &lt;- matrix(&quot;diff_ind&quot;,nrow=dim_temp[1],ncol=dim_temp[2])

dist_index_matrix[same_ind_index &amp; !same_batch_index] &lt;- &quot;same_ind_diff_batch&quot;
dist_index_matrix[same_batch_index] &lt;- &quot;same_batch&quot;

ans &lt;- lapply(unique(c(dist_index_matrix)),function(x){
  temp &lt;- c(total_counts_single_dist[(dist_index_matrix==x)&amp;(upper.tri(dist_index_matrix,diag=FALSE))])
  data.frame(dist=temp,type=rep(x,length(temp)))
})
ans1 &lt;- do.call(rbind,ans)

boxplot(dist~type,data=ans1)</code></pre>
<p><img src="figure/compare-reads-v-molecules.Rmd/three-groups-1.png" title="" alt="" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>plot(density(ans1$dist[ans1$type==&quot;same_batch&quot;]))
lines(density(ans1$dist[ans1$type==&quot;same_ind_diff_batch&quot;]),col=2)
lines(density(ans1$dist[ans1$type==&quot;diff_ind&quot;]),col=3)</code></pre>
<p><img src="figure/compare-reads-v-molecules.Rmd/three-groups-2.png" title="" alt="" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>ggplot(ans1, aes(x= factor(type), y = dist, col = factor(type)), height = 600, width = 2000) +
geom_boxplot(outlier.shape = NA, alpha = .01, width = .2, position = position_dodge(width = .9)) +
ylim(0, 100) +
labs(title = &quot;cell-cell distance (total molecule and total reads)&quot;) +
theme(axis.text.x = element_text(hjust=1, angle = 45))</code></pre>
<pre><code>Warning: Removed 1285 rows containing non-finite values (stat_boxplot).</code></pre>
<pre><code>Warning: Removed 292 rows containing missing values (geom_point).</code></pre>
<pre><code>Warning: Removed 613 rows containing missing values (geom_point).</code></pre>
<pre><code>Warning: Removed 1195 rows containing missing values (geom_point).</code></pre>
<p><img src="figure/compare-reads-v-molecules.Rmd/three-groups-3.png" title="" alt="" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>summary(lm(dist~type,data=ans1))</code></pre>
<pre><code>
Call:
lm(formula = dist ~ type, data = ans1)

Residuals:
    Min      1Q  Median      3Q     Max 
-29.126 -16.646  -4.327  12.006 138.499 

Coefficients:
                        Estimate Std. Error t value Pr(&gt;|t|)    
(Intercept)              25.0283     0.1512 165.572  &lt; 2e-16 ***
typesame_ind_diff_batch   0.8673     0.1883   4.606  4.1e-06 ***
typediff_ind              4.1509     0.1644  25.254  &lt; 2e-16 ***
---
Signif. codes:  0 &#39;***&#39; 0.001 &#39;**&#39; 0.01 &#39;*&#39; 0.05 &#39;.&#39; 0.1 &#39; &#39; 1

Residual standard error: 21.43 on 166750 degrees of freedom
Multiple R-squared:  0.006382,  Adjusted R-squared:  0.00637 
F-statistic: 535.5 on 2 and 166750 DF,  p-value: &lt; 2.2e-16</code></pre>
</div>
<div id="reads-to-molecule-conversion-rate" class="section level3">
<h3>Reads to molecule conversion rate</h3>
<pre class="r"><code>## calculate conversion rate
total_counts_single$conversion_rate &lt;- total_counts_single$total_reads/ total_counts_single$total_molecules
plot(total_counts_single$conversion_rate,col=total_counts_single$batch)</code></pre>
<p><img src="figure/compare-reads-v-molecules.Rmd/conversion-rate-1.png" title="" alt="" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>## create 9 batches
total_counts_single$batch_num &lt;- paste(total_counts_single$individual,total_counts_single$batch,sep=&quot;_&quot;)

## calculate individual mean for standardization
avg_conversion_rate &lt;- do.call(rbind,
lapply(unique(total_counts_single$individual),function(x){
  c(individual=x,avg_conv_rate=mean(total_counts_single$conversion_rate[total_counts_single$individual==x]))
}))

total_counts_single &lt;-merge(total_counts_single,avg_conversion_rate,by=&quot;individual&quot;)

## create std_conv_rate within individual to remove individual effect
total_counts_single$std_conv_rate &lt;- total_counts_single$conversion_rate - total_counts_single$avg_conv_rate
plot(total_counts_single$std_conv_rate,col=total_counts_single$batch)</code></pre>
<p><img src="figure/compare-reads-v-molecules.Rmd/conversion-rate-2.png" title="" alt="" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>summary(lm(std_conv_rate~batch_num,data=total_counts_single))</code></pre>
<pre><code>
Call:
lm(formula = std_conv_rate ~ batch_num, data = total_counts_single)

Residuals:
      Min        1Q    Median        3Q       Max 
-0.017995 -0.001046  0.000454  0.001550  0.007489 

Coefficients:
                   Estimate Std. Error t value Pr(&gt;|t|)    
(Intercept)      -2.439e-04  3.055e-04  -0.798 0.425037    
batch_num19098_2 -3.580e-03  7.888e-04  -4.539 6.91e-06 ***
batch_num19098_3  1.614e-03  4.822e-04   3.347 0.000871 ***
batch_num19101_1  3.796e-05  4.431e-04   0.086 0.931757    
batch_num19101_2  1.144e-03  4.583e-04   2.496 0.012834 *  
batch_num19101_3 -6.282e-04  4.959e-04  -1.267 0.205748    
batch_num19239_1 -1.306e-03  4.416e-04  -2.956 0.003242 ** 
batch_num19239_2  5.405e-04  4.601e-04   1.175 0.240640    
batch_num19239_3  1.522e-03  4.402e-04   3.458 0.000584 ***
---
Signif. codes:  0 &#39;***&#39; 0.001 &#39;**&#39; 0.01 &#39;*&#39; 0.05 &#39;.&#39; 0.1 &#39; &#39; 1

Residual standard error: 0.002817 on 569 degrees of freedom
Multiple R-squared:  0.1427,    Adjusted R-squared:  0.1307 
F-statistic: 11.84 on 8 and 569 DF,  p-value: 1.14e-15</code></pre>
<pre class="r"><code>## violin plots
ggplot(total_counts_single, aes(x= factor(batch_num), y = conversion_rate, fill = factor(batch_num)), height = 600, width = 2000) +
geom_violin(alpha = .5) + 
geom_boxplot(alpha = .01, width = .2, position = position_dodge(width = .9)) +
labs(x = &quot;batch&quot;, y = &quot;conversion rate&quot;, title = &quot;individual batch effect of conversion rate&quot;) +
theme(axis.text.x = element_text(hjust=1, angle = 45))</code></pre>
<p><img src="figure/compare-reads-v-molecules.Rmd/conversion-rate-3.png" title="" alt="" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>ggplot(total_counts_single, aes(x= factor(batch_num), y = std_conv_rate, fill = factor(batch_num)), height = 600, width = 2000) +
geom_violin(alpha = .5) + 
geom_boxplot(alpha = .01, width = .2, position = position_dodge(width = .9)) +
labs(x = &quot;batch&quot;, y = &quot;std conversion rate&quot;, title = &quot;batch effect of conversion rate&quot;) +
theme(axis.text.x = element_text(hjust=1, angle = 45))</code></pre>
<p><img src="figure/compare-reads-v-molecules.Rmd/conversion-rate-4.png" title="" alt="" width="672" style="display: block; margin: auto;" /></p>
</div>
<div id="pca" class="section level3">
<h3>PCA</h3>
<p>What effect does this difference in total molecule count have in PCA?</p>
<pre class="r"><code>pca_single &lt;- run_pca(molecules[, anno$well != &quot;bulk&quot;])</code></pre>
<pre class="r"><code>plot_pca(pca_single$PCs, explained = pca_single$explained,
         metadata = anno[anno$well != &quot;bulk&quot;, ], color = &quot;individual&quot;,
         shape = &quot;batch&quot;, factors = c(&quot;individual&quot;, &quot;batch&quot;)) +
  labs(title = &quot;PCA single cells uncorrected for depth&quot;)</code></pre>
<p><img src="figure/compare-reads-v-molecules.Rmd/pca-single-uncorrected-1.png" title="" alt="" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>pc1_v_total_mol_uncorrected &lt;- ggplot(cbind(total_counts_data[total_counts_data$well != &quot;bulk&quot;, ], pca_single$PCs),
       aes(x = total_molecules, y = PC1, col = as.factor(individual),
           shape = as.factor(batch))) +
  geom_point(alpha = 0.5) +
  labs(x = &quot;Total number of molecules (x10^3)&quot;)
pc1_v_total_mol_uncorrected</code></pre>
<p><img src="figure/compare-reads-v-molecules.Rmd/pc1-v-total-molecules-uncorrected-1.png" title="" alt="" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>pc2_v_total_mol_uncorrected &lt;- pc1_v_total_mol_uncorrected %+% aes(y = PC2)
pc2_v_total_mol_uncorrected</code></pre>
<p><img src="figure/compare-reads-v-molecules.Rmd/pc2-v-total-molecules-uncorrected-1.png" title="" alt="" width="672" style="display: block; margin: auto;" /></p>
<p>The total molecule depth per sample is highly correlated with PC1. However, it did not affect PC2, which captures the individual effect.</p>
<p>What happens to the PCA results when depth is properly accounted for using TMM-normalized counts per million?</p>
<pre class="r"><code>norm_factors_mol_single &lt;- calcNormFactors(molecules[, anno$well != &quot;bulk&quot;],
                                           method = &quot;TMM&quot;)
molecules_cpm_single &lt;- cpm(molecules[, anno$well != &quot;bulk&quot;],
                            lib.size = colSums(molecules[, anno$well != &quot;bulk&quot;]) *
                                       norm_factors_mol_single)</code></pre>
<pre class="r"><code>pca_single_cpm &lt;- run_pca(molecules_cpm_single)</code></pre>
<pre class="r"><code>plot_pca(pca_single_cpm$PCs, explained = pca_single_cpm$explained,
         metadata = anno[anno$well != &quot;bulk&quot;, ], color = &quot;individual&quot;,
         shape = &quot;batch&quot;, factors = c(&quot;individual&quot;, &quot;batch&quot;)) +
  labs(title = &quot;PCA single cells *corrected* for depth&quot;)</code></pre>
<p><img src="figure/compare-reads-v-molecules.Rmd/pca-single-corrected-1.png" title="" alt="" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>pc1_v_total_mol_corrected &lt;- pc1_v_total_mol_uncorrected %+%
  cbind(total_counts_data[total_counts_data$well != &quot;bulk&quot;, ], pca_single_cpm$PCs) +
  labs(title = &quot;PCA single cells *corrected* for depth&quot;)
pc1_v_total_mol_corrected</code></pre>
<p><img src="figure/compare-reads-v-molecules.Rmd/pc1-v-total-molecules-corrected-1.png" title="" alt="" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>pc2_v_total_mol_corrected &lt;- pc1_v_total_mol_corrected %+% aes(y = PC2)
pc2_v_total_mol_corrected</code></pre>
<p><img src="figure/compare-reads-v-molecules.Rmd/pc2-v-total-molecules-corrected-1.png" title="" alt="" width="672" style="display: block; margin: auto;" /></p>
<p>PC1 is no longer associated with sequencing depth!</p>
</div>
</div>
<div id="compare-reads-and-standardized-molecules" class="section level2">
<h2>Compare reads and standardized molecules</h2>
<p>This time standardize the molecule counts for the sequencing depth.</p>
<p>Compare the means of each gene obtained via the different methods.</p>
<pre class="r"><code>mean_data_std &lt;- data.frame(reads_cpm = rowMeans(reads_cpm),
                        molecules_cpm = rowMeans(molecules_cpm),
                        molecules_per_lane_cpm = rowMeans(molecules_per_lane_cpm))
cor(mean_data_std)</code></pre>
<pre><code>                       reads_cpm molecules_cpm molecules_per_lane_cpm
reads_cpm              1.0000000     0.9704740              0.9703383
molecules_cpm          0.9704740     1.0000000              0.9986011
molecules_per_lane_cpm 0.9703383     0.9986011              1.0000000</code></pre>
<p>All three are even more highly correlated now that the molecules are standardized.</p>
<pre class="r"><code>mean_data_std$type &lt;- ifelse(grepl(&quot;ERCC&quot;, rownames(mean_data_std)), &quot;ERCC&quot;, &quot;gene&quot;)
ggplot(mean_data_std, aes(x = reads_cpm, y = molecules_cpm)) +
  geom_point() +
  geom_smooth(method = &quot;lm&quot;) +
  facet_wrap(~ type)</code></pre>
<p><img src="figure/compare-reads-v-molecules.Rmd/reads-v-molecules-std-1.png" title="" alt="" width="672" style="display: block; margin: auto;" /></p>
<p>Examining the lower range where most genes are:</p>
<pre class="r"><code>ggplot(mean_data_std, aes(x = reads_cpm, y = molecules_cpm)) +
  geom_point() +
  geom_smooth(method = &quot;lm&quot;) +
  facet_wrap(~ type) +
  ylim(0, 10000)</code></pre>
<p><img src="figure/compare-reads-v-molecules.Rmd/reads-v-molecules-range-restricted-std-1.png" title="" alt="" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>ggplot(mean_data_std, aes(x = reads_cpm, y = molecules_per_lane_cpm)) +
  geom_point() +
  geom_smooth(method = &quot;lm&quot;) +
  facet_wrap(~ type)</code></pre>
<p><img src="figure/compare-reads-v-molecules.Rmd/reads-v-molecules-per-lane-std-1.png" title="" alt="" width="672" style="display: block; margin: auto;" /></p>
<p>And as above, the molecule counts and the molecule counts summed per sequencing lane are highly correlated, which indicates that most of the bias is introduced in the library preparation step and not during sequencing.</p>
<pre class="r"><code>ggplot(mean_data_std, aes(x = molecules_per_lane_cpm, y = molecules_cpm)) +
  geom_point() +
  geom_smooth(method = &quot;lm&quot;) +
  facet_wrap(~ type)</code></pre>
<p><img src="figure/compare-reads-v-molecules.Rmd/molecules-v-molecules-per-lane-std-1.png" title="" alt="" width="672" style="display: block; margin: auto;" /></p>
</div>
<div id="session-information" class="section level2">
<h2>Session information</h2>
<pre class="r"><code>sessionInfo()</code></pre>
<pre><code>R version 3.2.0 (2015-04-16)
Platform: x86_64-unknown-linux-gnu (64-bit)

locale:
 [1] LC_CTYPE=en_US.UTF-8       LC_NUMERIC=C              
 [3] LC_TIME=en_US.UTF-8        LC_COLLATE=en_US.UTF-8    
 [5] LC_MONETARY=en_US.UTF-8    LC_MESSAGES=en_US.UTF-8   
 [7] LC_PAPER=en_US.UTF-8       LC_NAME=C                 
 [9] LC_ADDRESS=C               LC_TELEPHONE=C            
[11] LC_MEASUREMENT=en_US.UTF-8 LC_IDENTIFICATION=C       

attached base packages:
[1] stats     graphics  grDevices utils     datasets  methods   base     

other attached packages:
[1] testit_0.4    edgeR_3.10.2  limma_3.24.9  ggplot2_1.0.1 dplyr_0.4.2  
[6] knitr_1.10.5 

loaded via a namespace (and not attached):
 [1] Rcpp_0.12.0        magrittr_1.5       MASS_7.3-40       
 [4] munsell_0.4.2      colorspace_1.2-6   R6_2.1.1          
 [7] stringr_1.0.0      httr_0.6.1         plyr_1.8.3        
[10] tools_3.2.0        parallel_3.2.0     grid_3.2.0        
[13] gtable_0.1.2       DBI_0.3.1          htmltools_0.2.6   
[16] yaml_2.1.13        assertthat_0.1     digest_0.6.8      
[19] RColorBrewer_1.1-2 reshape2_1.4.1     formatR_1.2       
[22] bitops_1.0-6       RCurl_1.95-4.6     evaluate_0.7      
[25] rmarkdown_0.6.1    labeling_0.3       stringi_0.4-1     
[28] scales_0.2.4       proto_0.3-10      </code></pre>
</div>


<!-- some extra javascript for older browsers -->
<script type="text/javascript" src="libs/polyfill.js"></script>

<script>

// manage active state of menu based on current page
$(document).ready(function () {

    // active menu
    href = window.location.pathname
    href = href.substr(href.lastIndexOf('/') + 1)
    $('a[href="' + href + '"]').parent().addClass('active');

    // manage active menu header
    if (href.startsWith('authoring_'))
      $('a[href="' + 'authoring' + '"]').parent().addClass('active');
    else if (href.endsWith('_format.html'))
      $('a[href="' + 'formats' + '"]').parent().addClass('active');
    else if (href.startsWith('developer_'))
      $('a[href="' + 'developer' + '"]').parent().addClass('active');

});

</script>

</div>

<script>

// add bootstrap table styles to pandoc tables
$(document).ready(function () {
  $('tr.header').parent('thead').parent('table').addClass('table table-condensed');
});

</script>


</body>
</html>
