<!DOCTYPE html>

<html xmlns="http://www.w3.org/1999/xhtml">

<head>

<meta charset="utf-8">
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="generator" content="pandoc" />

<meta name="author" content="Joyce Hsiao &amp; PoYuan Tung" />

<meta name="date" content="2015-10-23" />

<title>Batch effects of conversion efficiency</title>

<script src="libs/jquery-1.11.0/jquery.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="libs/bootstrap-3.3.1/css/united.min.css" rel="stylesheet" />
<script src="libs/bootstrap-3.3.1/js/bootstrap.min.js"></script>
<script src="libs/bootstrap-3.3.1/shim/html5shiv.min.js"></script>
<script src="libs/bootstrap-3.3.1/shim/respond.min.js"></script>

<style type="text/css">

/* padding for bootstrap navbar */
body {
  padding-top: 50px;
  padding-bottom: 40px;
}


/* offset scroll position for anchor links (for fixed navbar)  */
.section h2 {
  padding-top: 55px;
  margin-top: -55px;
}
.section h3 {
  padding-top: 55px;
  margin-top: -55px;
}



/* don't use link color in navbar */
.dropdown-menu>li>a {
  color: black;
}

/* some padding for disqus */
#disqus_thread {
  margin-top: 45px;
}

</style>

<link rel="stylesheet" href="libs/font-awesome-4.1.0/css/font-awesome.min.css"/>

<style type="text/css">code{white-space: pre;}</style>
<link rel="stylesheet"
      href="libs/highlight/textmate.css"
      type="text/css" />
<script src="libs/highlight/highlight.js"></script>
<style type="text/css">
  pre:not([class]) {
    background-color: white;
  }
</style>
<script type="text/javascript">
if (window.hljs && document.readyState && document.readyState === "complete") {
   window.setTimeout(function() {
      hljs.initHighlighting();
   }, 0);
}
</script>



</head>

<body>

<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
code {
  color: inherit;
  background-color: rgba(0, 0, 0, 0.04);
}
img { 
  max-width:100%; 
  height: auto; 
}
</style>
<div class="container-fluid main-container">


<div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">singleCellSeq</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li><a href="index.html">Home</a></li>
        <li><a href="about.html">About</a></li>
        <li><a href="license.html">License</a></li>
        <li><a href="https://github.com/jdblischak/singleCellSeq">GitHub</a></li>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<div id="header">
<h1 class="title">Batch effects of conversion efficiency</h1>
<h4 class="author"><em>Joyce Hsiao &amp; PoYuan Tung</em></h4>
<h4 class="date"><em>2015-10-23</em></h4>
</div>

<div id="TOC">
<ul>
<li><a href="#objective">Objective</a></li>
<li><a href="#input">Input</a></li>
<li><a href="#reads-to-molecule-conversion-efficiency">Reads to molecule conversion efficiency</a></li>
<li><a href="#compare-variances-of-conversion-rates-between-batches">Compare variances of conversion rates between batches</a><ul>
<li><a href="#ensg">ENSG</a></li>
<li><a href="#ercc">ERCC</a></li>
</ul></li>
<li><a href="#summary-plots">Summary plots</a></li>
<li><a href="#session-information">Session information</a></li>
</ul>
</div>

<p><strong>Last updated:</strong> 2016-06-11</p>
<p><strong>Code version:</strong> 0242e47aaea33879b60143338a1f1f3232683dc6</p>
<div id="objective" class="section level2">
<h2>Objective</h2>
<p>Compute conversion efficiency of each cell using the filerted set of cells. Compare conversion efficiency between batches: is there an individual effect or batch effect of coversion efficiency?</p>
<p>The definition of conversion efficiency: total molecule counts/ total read counts</p>
<p>We will use analysis of variance to test whether conversion rates are significantly different between individuals. Results below are described at an alpha level of .05.</p>
</div>
<div id="input" class="section level2">
<h2>Input</h2>
<pre class="r"><code>library(&quot;dplyr&quot;)
library(&quot;ggplot2&quot;)
library(&quot;cowplot&quot;)
theme_set(theme_bw(base_size = 12))
theme_update(panel.grid.minor.x = element_blank(),
             panel.grid.minor.y = element_blank(),
             panel.grid.major.x = element_blank(),
             panel.grid.major.y = element_blank())
library(&quot;edgeR&quot;)
source(&quot;functions.R&quot;)</code></pre>
<p>Input annotation.</p>
<pre class="r"><code>anno &lt;- read.table(&quot;../data/annotation.txt&quot;, header = TRUE,
                   stringsAsFactors = FALSE)
head(anno)</code></pre>
<pre><code>  individual replicate well      batch      sample_id
1    NA19098        r1  A01 NA19098.r1 NA19098.r1.A01
2    NA19098        r1  A02 NA19098.r1 NA19098.r1.A02
3    NA19098        r1  A03 NA19098.r1 NA19098.r1.A03
4    NA19098        r1  A04 NA19098.r1 NA19098.r1.A04
5    NA19098        r1  A05 NA19098.r1 NA19098.r1.A05
6    NA19098        r1  A06 NA19098.r1 NA19098.r1.A06</code></pre>
<p>Input filtered read counts.</p>
<pre class="r"><code>reads &lt;- read.table(&quot;../data/reads-filter.txt&quot;, header = TRUE,
                    stringsAsFactors = FALSE)</code></pre>
<p>Input filtered molecule counts.</p>
<pre class="r"><code>molecules &lt;- read.table(&quot;../data/molecules-filter.txt&quot;, header = TRUE,
                    stringsAsFactors = FALSE)</code></pre>
<p>Input list of quality single cells.</p>
<pre class="r"><code>quality_single_cells &lt;- scan(&quot;../data/quality-single-cells.txt&quot;,
                             what = &quot;character&quot;)</code></pre>
<p>Keep only the single cells that passed the <a href="qc-filter-ipsc.html">QC filters</a>.</p>
<pre class="r"><code>reads &lt;- reads[, colnames(reads) %in% quality_single_cells]
molecules &lt;- molecules[, colnames(molecules) %in% quality_single_cells]
anno &lt;- anno[anno$sample_id %in% quality_single_cells, ]
stopifnot(dim(reads) == dim(molecules),
          nrow(anno) == ncol(reads))</code></pre>
<p>Seperate ERCC and endogenous genes</p>
<pre class="r"><code>reads_ENSG &lt;- reads[grep(&quot;ERCC&quot;, rownames(reads), invert = TRUE), ]
reads_ERCC &lt;- reads[grep(&quot;ERCC&quot;, rownames(reads), invert = FALSE), ]

molecules_ENSG &lt;- molecules[grep(&quot;ERCC&quot;, rownames(molecules), invert = TRUE), ]
molecules_ERCC &lt;- molecules[grep(&quot;ERCC&quot;, rownames(molecules), invert = FALSE), ]</code></pre>
</div>
<div id="reads-to-molecule-conversion-efficiency" class="section level2">
<h2>Reads to molecule conversion efficiency</h2>
<pre class="r"><code>total_counts_ERCC &lt;- data.frame(total_reads = colSums(reads_ERCC),
                                total_molecules = colSums(molecules_ERCC))
total_counts_ERCC$conversion &lt;- with(total_counts_ERCC,
                                     total_molecules/total_reads)
summary(total_counts_ERCC)</code></pre>
<pre><code>  total_reads    total_molecules   conversion     
 Min.   :10906   Min.   : 666    Min.   :0.01403  
 1st Qu.:31564   1st Qu.: 914    1st Qu.:0.02321  
 Median :38934   Median :1028    Median :0.02740  
 Mean   :39700   Mean   :1102    Mean   :0.02965  
 3rd Qu.:46506   3rd Qu.:1276    3rd Qu.:0.03427  
 Max.   :88786   Max.   :2035    Max.   :0.07813  </code></pre>
<pre class="r"><code>total_counts_ENSG &lt;- data.frame(total_reads = colSums(reads_ENSG),
                                total_molecules = colSums(molecules_ENSG))
total_counts_ENSG$conversion &lt;- with(total_counts_ENSG,
                                     total_molecules/total_reads)
summary(total_counts_ENSG)</code></pre>
<pre><code>  total_reads      total_molecules    conversion     
 Min.   : 625855   Min.   : 30408   Min.   :0.01712  
 1st Qu.:1959019   1st Qu.: 55561   1st Qu.:0.02471  
 Median :2396897   Median : 65636   Median :0.02882  
 Mean   :2334326   Mean   : 66634   Mean   :0.02944  
 3rd Qu.:2727310   3rd Qu.: 78173   3rd Qu.:0.03318  
 Max.   :4143812   Max.   :113832   Max.   :0.07225  </code></pre>
<pre class="r"><code>## create a color palette with one color per individual and different shades for repplicates
great_color &lt;- c(&quot;#CC3300&quot;, &quot;#FF9966&quot;, &quot;#006633&quot;, &quot;#009900&quot;, &quot;#99FF99&quot;, &quot;#3366FF&quot;, &quot;#6699FF&quot;, &quot;#66CCFF&quot;)

depth_plot_ensg &lt;- ggplot(data.frame(total_counts_ENSG,
                  batch = anno$batch), 
       aes(x = total_reads/10^6, y = total_molecules/10^3, col = factor(batch))) +
  geom_point() +
  scale_color_manual(values = great_color) +
  labs(x = &quot;Total number of reads (x10^6)&quot;,
       y = &quot;Total number of molecules (x10^3)&quot;,
       title = &quot;Effect of read depth on single cells (genes)&quot;)

depth_plot_ensg</code></pre>
<p><img src="figure/compare-reads-v-molecules-per-batch-test.Rmd/plot-convertion-1.png" title="" alt="" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>depth_plot_ercc &lt;- ggplot(data.frame(total_counts_ERCC,
                  batch = anno$batch), 
       aes(x = total_reads/10^6, y = total_molecules/10^3, col = factor(batch))) +
  geom_point() +
  scale_color_manual(values = great_color) +
  labs(x = &quot;Total number of reads (x10^6)&quot;,
       y = &quot;Total number of molecules (x10^3)&quot;,
       title = &quot;Effect of read depth on single cells (ERCC spike-in)&quot;)

depth_plot_ercc</code></pre>
<p><img src="figure/compare-reads-v-molecules-per-batch-test.Rmd/plot-convertion-2.png" title="" alt="" width="672" style="display: block; margin: auto;" /></p>
</div>
<div id="compare-variances-of-conversion-rates-between-batches" class="section level2">
<h2>Compare variances of conversion rates between batches</h2>
<div id="ensg" class="section level3">
<h3>ENSG</h3>
<pre class="r"><code>convertion_ensg &lt;- ggplot(data.frame(total_counts_ENSG,
                  batch = anno$batch),
       aes(x = factor(batch), y = conversion,
           fill = factor(batch)), height = 600, width = 2000) +
geom_violin(alpha = .5) + 
geom_boxplot(alpha = .01, width = .2, position = position_dodge(width = .9)) +
  scale_fill_manual(values = great_color) +
labs(x = &quot;Batch&quot;, y = &quot;Conversion efficiency&quot;, title = &quot;ENSG batch effect of conversion efficiency&quot;) +
theme(axis.text.x = element_text(hjust=1, angle = 45))

convertion_ensg</code></pre>
<p><img src="figure/compare-reads-v-molecules-per-batch-test.Rmd/unnamed-chunk-1-1.png" title="" alt="" width="672" style="display: block; margin: auto;" /></p>
<p>*Between individual</p>
<p>For individual <span class="math">\(i\)</span>, sample <span class="math">\(j\)</span>, we have <span class="math">\(n_{ij}\)</span> cells. Here we compare the per cell conversion rate from molecule count to read count between the three individuals:</p>
<p><span class="math">\[ Y_{ijk} = \mu_i + b_{ij} + \epsilon_{ijk} \]</span></p>
<p>where <span class="math">\(i = 1, 2, 3\)</span>, <span class="math">\(j = 1, 2, 3\)</span>, and <span class="math">\(k = 1, 2, \dots, n_{ij}\)</span>.</p>
<pre class="r"><code>df_ENSG &lt;- data.frame(total_counts_ENSG,
                      individual = factor(anno$individual),
                      replicate = factor(anno$replicate),
                      batch = factor(anno$batch))

library(lme4)</code></pre>
<pre><code>Loading required package: Matrix</code></pre>
<pre class="r"><code>lmer_ENSG_full &lt;- lmer(conversion ~ individual + (1| individual:replicate), 
                        data = df_ENSG)
lmer_ENSG_null &lt;- lmer(conversion ~ 1 + (1| individual:replicate), 
                        data = df_ENSG)
anova(lmer_ENSG_null, lmer_ENSG_full)</code></pre>
<pre><code>refitting model(s) with ML (instead of REML)</code></pre>
<pre><code>Data: df_ENSG
Models:
lmer_ENSG_null: conversion ~ 1 + (1 | individual:replicate)
lmer_ENSG_full: conversion ~ individual + (1 | individual:replicate)
               Df     AIC     BIC logLik deviance  Chisq Chi Df Pr(&gt;Chisq)
lmer_ENSG_null  3 -4506.8 -4493.8 2256.4  -4512.8                         
lmer_ENSG_full  5 -4523.8 -4502.2 2266.9  -4533.8 21.077      2  2.649e-05
                  
lmer_ENSG_null    
lmer_ENSG_full ***
---
Signif. codes:  0 &#39;***&#39; 0.001 &#39;**&#39; 0.01 &#39;*&#39; 0.05 &#39;.&#39; 0.1 &#39; &#39; 1</code></pre>
<p>*For each individual, between batches.</p>
<pre class="r"><code>for (i in 1:length(unique(anno$individual))) {
  print(unique(anno$individual)[i])
  select_individual &lt;- with(anno, individual == unique(individual)[i])
  print( summary(aov(conversion ~ batch, 
                     data = df_ENSG[select_individual, ]) ) )
}</code></pre>
<pre><code>[1] &quot;NA19098&quot;
             Df   Sum Sq   Mean Sq F value Pr(&gt;F)
batch         1 0.000035 3.474e-05    0.88   0.35
Residuals   140 0.005528 3.949e-05               
[1] &quot;NA19101&quot;
             Df   Sum Sq   Mean Sq F value  Pr(&gt;F)   
batch         2 0.000125 6.250e-05   5.722 0.00384 **
Residuals   198 0.002163 1.092e-05                   
---
Signif. codes:  0 &#39;***&#39; 0.001 &#39;**&#39; 0.01 &#39;*&#39; 0.05 &#39;.&#39; 0.1 &#39; &#39; 1
[1] &quot;NA19239&quot;
             Df   Sum Sq  Mean Sq F value   Pr(&gt;F)    
batch         2 0.000750 3.75e-04   32.11 5.99e-13 ***
Residuals   218 0.002546 1.17e-05                     
---
Signif. codes:  0 &#39;***&#39; 0.001 &#39;**&#39; 0.01 &#39;*&#39; 0.05 &#39;.&#39; 0.1 &#39; &#39; 1</code></pre>
</div>
<div id="ercc" class="section level3">
<h3>ERCC</h3>
<pre class="r"><code>convertion_ercc &lt;- ggplot(data.frame(total_counts_ERCC,
                  batch = anno$batch),
       aes(x = factor(batch), y = conversion,
           fill = factor(batch)), height = 600, width = 2000) +
geom_violin(alpha = .5) + 
geom_boxplot(alpha = .01, width = .2, position = position_dodge(width = .9)) +
  scale_fill_manual(values = great_color) +
labs(x = &quot;Batch&quot;, y = &quot;Conversion efficiency&quot;, title = &quot;ERCC batch effect of conversion efficiency&quot;) +
theme(axis.text.x = element_text(hjust=1, angle = 45))

convertion_ercc</code></pre>
<p><img src="figure/compare-reads-v-molecules-per-batch-test.Rmd/unnamed-chunk-4-1.png" title="" alt="" width="672" style="display: block; margin: auto;" /></p>
<ul>
<li>Between individuals</li>
</ul>
<pre class="r"><code>df_ERCC &lt;- data.frame(total_counts_ERCC,
                      individual = factor(anno$individual),
                      replicate = factor(anno$replicate),
                      batch = factor(anno$batch))

library(lme4)
lmer_ERCC_full &lt;- lmer(conversion ~ individual + (1| individual:replicate), 
                        data = df_ERCC)
lmer_ERCC_null &lt;- lmer(conversion ~ 1 + (1| individual:replicate), 
                        data = df_ERCC)
anova(lmer_ERCC_null, lmer_ERCC_full)</code></pre>
<pre><code>refitting model(s) with ML (instead of REML)</code></pre>
<pre><code>Data: df_ERCC
Models:
lmer_ERCC_null: conversion ~ 1 + (1 | individual:replicate)
lmer_ERCC_full: conversion ~ individual + (1 | individual:replicate)
               Df     AIC     BIC logLik deviance  Chisq Chi Df Pr(&gt;Chisq)
lmer_ERCC_null  3 -3811.1 -3798.1 1908.5  -3817.1                         
lmer_ERCC_full  5 -3827.1 -3805.4 1918.5  -3837.1 19.994      2  4.553e-05
                  
lmer_ERCC_null    
lmer_ERCC_full ***
---
Signif. codes:  0 &#39;***&#39; 0.001 &#39;**&#39; 0.01 &#39;*&#39; 0.05 &#39;.&#39; 0.1 &#39; &#39; 1</code></pre>
<p>*For each individual, between batches.</p>
<pre class="r"><code>for (i in 1:length(unique(anno$individual))) {
  print(unique(anno$individual)[i])
  select_individual &lt;- with(anno, individual == unique(individual)[i])
  print( summary(aov(conversion ~ batch, 
                     data = df_ERCC[select_individual, ]) ) )
}</code></pre>
<pre><code>[1] &quot;NA19098&quot;
             Df   Sum Sq   Mean Sq F value Pr(&gt;F)
batch         1 0.000181 1.813e-04   1.823  0.179
Residuals   140 0.013922 9.944e-05               
[1] &quot;NA19101&quot;
             Df   Sum Sq   Mean Sq F value Pr(&gt;F)
batch         2 0.000008 4.190e-06   0.125  0.883
Residuals   198 0.006637 3.352e-05               
[1] &quot;NA19239&quot;
             Df   Sum Sq  Mean Sq F value   Pr(&gt;F)    
batch         2 0.001112 5.56e-04   8.019 0.000436 ***
Residuals   218 0.015114 6.93e-05                     
---
Signif. codes:  0 &#39;***&#39; 0.001 &#39;**&#39; 0.01 &#39;*&#39; 0.05 &#39;.&#39; 0.1 &#39; &#39; 1</code></pre>
</div>
</div>
<div id="summary-plots" class="section level2">
<h2>Summary plots</h2>
<pre class="r"><code>library(gridExtra)
grid.arrange(
  ggplot(data.frame(total_counts_ERCC,
                    batch = anno$batch),
         aes(x = factor(batch), y = conversion,
             fill = factor(batch)), height = 600, width = 2000) +
  geom_violin(alpha = .5) + 
  geom_boxplot(alpha = .01, width = .2, position = position_dodge(width = .9)) +
    scale_fill_manual(values = great_color) +
  labs(x = &quot;Batch&quot;, y = &quot;Conversion rate&quot;, title = &quot;Batch effect of conversion rate (ERCC spike-in)&quot;) +
  theme(axis.text.x = element_text(hjust=1, angle = 45))
  ,
  ggplot(data.frame(total_counts_ENSG,
                    batch = anno$batch),
         aes(x = factor(batch), y = conversion,
             fill = factor(batch)), height = 600, width = 2000) +
  geom_violin(alpha = .5) + 
  geom_boxplot(alpha = .01, width = .2, position = position_dodge(width = .9)) +
    scale_fill_manual(values = great_color) +
  labs(x = &quot;Batch&quot;, y = &quot;Conversion rate&quot;, title = &quot;Batch effect of conversion rate (genes)&quot;) +
  theme(axis.text.x = element_text(hjust=1, angle = 45)),
  ncol = 2
)</code></pre>
<p><img src="figure/compare-reads-v-molecules-per-batch-test.Rmd/unnamed-chunk-7-1.png" title="" alt="" width="1152" style="display: block; margin: auto;" /></p>
<pre class="r"><code>plot_grid(depth_plot_ensg + theme(legend.position = &quot;none&quot;), 
          convertion_ensg + theme(legend.position = &quot;none&quot;),
          depth_plot_ercc + theme(legend.position = &quot;none&quot;),
          convertion_ercc + theme(legend.position = &quot;none&quot;),
          labels = LETTERS[1:4])</code></pre>
<p><img src="figure/compare-reads-v-molecules-per-batch-test.Rmd/plot-reads-molecule-1.png" title="" alt="" width="1152" style="display: block; margin: auto;" /></p>
<pre class="r"><code>plot_grid(convertion_ensg + theme(legend.position = &quot;none&quot;),
          convertion_ercc + theme(legend.position = &quot;none&quot;),
          labels = LETTERS[5:6])</code></pre>
<p><img src="figure/compare-reads-v-molecules-per-batch-test.Rmd/plot-reads-molecule-paper-1.png" title="" alt="" width="1152" style="display: block; margin: auto;" /></p>
</div>
<div id="session-information" class="section level2">
<h2>Session information</h2>
<pre class="r"><code>sessionInfo()</code></pre>
<pre><code>R version 3.2.0 (2015-04-16)
Platform: x86_64-unknown-linux-gnu (64-bit)

locale:
 [1] LC_CTYPE=en_US.UTF-8       LC_NUMERIC=C              
 [3] LC_TIME=en_US.UTF-8        LC_COLLATE=en_US.UTF-8    
 [5] LC_MONETARY=en_US.UTF-8    LC_MESSAGES=en_US.UTF-8   
 [7] LC_PAPER=en_US.UTF-8       LC_NAME=C                 
 [9] LC_ADDRESS=C               LC_TELEPHONE=C            
[11] LC_MEASUREMENT=en_US.UTF-8 LC_IDENTIFICATION=C       

attached base packages:
[1] stats     graphics  grDevices utils     datasets  methods   base     

other attached packages:
[1] gridExtra_2.0.0 lme4_1.1-10     Matrix_1.2-1    edgeR_3.10.2   
[5] limma_3.24.9    cowplot_0.3.1   ggplot2_1.0.1   dplyr_0.4.2    
[9] knitr_1.10.5   

loaded via a namespace (and not attached):
 [1] Rcpp_0.12.4      formatR_1.2      nloptr_1.0.4     plyr_1.8.3      
 [5] bitops_1.0-6     tools_3.2.0      digest_0.6.8     evaluate_0.7    
 [9] gtable_0.1.2     nlme_3.1-120     lattice_0.20-31  DBI_0.3.1       
[13] yaml_2.1.13      parallel_3.2.0   proto_0.3-10     httr_0.6.1      
[17] stringr_1.0.0    grid_3.2.0       R6_2.1.1         rmarkdown_0.6.1 
[21] minqa_1.2.4      reshape2_1.4.1   magrittr_1.5     scales_0.4.0    
[25] htmltools_0.2.6  MASS_7.3-40      splines_3.2.0    assertthat_0.1  
[29] colorspace_1.2-6 labeling_0.3     stringi_1.0-1    RCurl_1.95-4.6  
[33] munsell_0.4.3   </code></pre>
</div>


<!-- some extra javascript for older browsers -->
<script type="text/javascript" src="libs/polyfill.js"></script>

<script>

// manage active state of menu based on current page
$(document).ready(function () {

    // active menu
    href = window.location.pathname
    href = href.substr(href.lastIndexOf('/') + 1)
    $('a[href="' + href + '"]').parent().addClass('active');

    // manage active menu header
    if (href.startsWith('authoring_'))
      $('a[href="' + 'authoring' + '"]').parent().addClass('active');
    else if (href.endsWith('_format.html'))
      $('a[href="' + 'formats' + '"]').parent().addClass('active');
    else if (href.startsWith('developer_'))
      $('a[href="' + 'developer' + '"]').parent().addClass('active');

});

</script>

</div>

<script>

// add bootstrap table styles to pandoc tables
$(document).ready(function () {
  $('tr.header').parent('thead').parent('table').addClass('table table-condensed');
});

</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://cdn.mathjax.org/mathjax/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
