---
title: "Compare read and molecule counts in single cells per batch"
author: "PoYuan Tung"
date: 2015-08-24
---

**Last updated:** `r Sys.Date()`

**Code version:** `r system("git log -1 --format='%H'", intern = TRUE)`

```{r chunk-options, include=FALSE}
source("chunk-options.R")
opts_chunk$set(cache = FALSE)
```

Comparing the conversion of reads to molecules for each of the 9 batches.
Used three different metrics:

*  Raw counts
*  Log2 counts (pseudocount of 1)
*  Log2 TMM-normalized counts per million (pseudocount of 0.25)

## Input

```{r packages, message=FALSE, cache=FALSE}
library("dplyr")
library("ggplot2")
theme_set(theme_bw(base_size = 16))
library("edgeR")
source("functions.R")
library("tidyr")
```

Input annotation.

```{r input-annotation}
anno <- read.table("../data/annotation.txt", header = TRUE,
                   stringsAsFactors = FALSE)
head(anno)
```

Input read counts.

```{r input-read-counts}
reads <- read.table("../data/reads.txt", header = TRUE,
                    stringsAsFactors = FALSE)
```

Input molecule counts.

```{r input-molecule-counts}
molecules <- read.table("../data/molecules.txt", header = TRUE,
                    stringsAsFactors = FALSE)
```

Input list of quality single cells.

```{r input-quality-single-cells}
quality_single_cells <- scan("../data/quality-single-cells.txt",
                             what = "character")
```

Keep only the single cells that passed the [QC filters](qc-cell-ipsc.html).

```{r filter-cells}
reads <- reads[, colnames(reads) %in% quality_single_cells]
molecules <- molecules[, colnames(molecules) %in% quality_single_cells]
anno <- anno[anno$sample_id %in% quality_single_cells, ]
stopifnot(dim(reads) == dim(molecules),
          nrow(anno) == ncol(reads))
```

## Distribution 
Look at the distribution of fold change to mean.  

```{r distribution}
## calculate mean
reads_mean     <- apply(reads, 1, mean)
molecules_mean <- apply(molecules, 1, mean)
distribution <- data.frame(reads_mean, molecules_mean)

## calculate fold change to mean
distribution$fold_change_read     <- log2(reads_mean/mean(reads_mean))
distribution$fold_change_molecule <- log2(molecules_mean/mean(molecules_mean))

## select ERCC
distribution$ERCC <- grepl("ERCC", rownames(distribution))

## color palette
cbPalette <- c("#999999", "#0000FF", "#990033", "#F0E442", "#0072B2", "#D55E00", "#CC79A7", "#009E73")

ggplot(distribution, aes(x = fold_change_molecule, y = fold_change_read, col = ERCC)) + geom_point(size = 3, alpha = 0.5) + scale_colour_manual(values=cbPalette) + stat_function(fun= function(x) {x}, col= "#56B4E9") + labs(x = "log2 fold change to mean (molecule)", y =  "log2 fold change to mean (reads)")
```

## Filter

Only keep the following genes:

*  ERCC genes with at least one molecule observed in at least one single cell

```{r keep-ercc}
ercc_keep <- rownames(molecules)[grepl("ERCC", rownames(molecules)) &
                                 rowSums(molecules) > 1]
```

*  The top expressed endogenous genes in terms of mean molecule counts per million

```{r top-expressed-genes}
num_genes <- 12000
mean_cpm <- molecules %>%
  filter(!grepl("ERCC", rownames(molecules))) %>%
  cpm %>%
  rowMeans
gene_keep <- rownames(molecules)[!grepl("ERCC", rownames(molecules))][order(mean_cpm, decreasing = TRUE)][1:num_genes]
```

Filter the genes:

```{r filter-genes}
reads <- reads[rownames(reads) %in% c(gene_keep, ercc_keep), ]
molecules <- molecules[rownames(molecules) %in% c(gene_keep, ercc_keep), ]
```

## Transformation

In addition to comparing the raw counts of reads and molecules, we compare the log2 counts and the log counts per million.

For the log counts, I add a pseudocount of 1.

```{r log-counts}
reads_log <- log2(reads + 1)
molecules_log <- log2(molecules + 1)
```

Calculate cpm for the reads data using TMM-normalization.

```{r calc-cpm-reads}
norm_factors_reads <- calcNormFactors(reads, method = "TMM")
reads_cpm <- cpm(reads, lib.size = colSums(reads) * norm_factors_reads,
                 log = TRUE)
```

And for the molecules.

```{r calc-cpm-molecules}
norm_factors_mol <- calcNormFactors(molecules, method = "TMM")
molecules_cpm <- cpm(molecules, lib.size = colSums(molecules) * norm_factors_mol,
                     log = TRUE)
```

## conversion in each single cell
```{r conversion-in-cell}
reads1 <- reads[,1]
molecules1 <- molecules[,1]
lm.D <- lm(molecules1 ~ reads1)
lm.DD <- lm(molecules1 ~ reads1 - 1) # omitting intercept

reads_cpm1 <- reads_cpm[,1]
molecules_cpm1 <- molecules_cpm[,1]
lm.E <- lm(molecules_cpm1 ~ reads_cpm1)

regression_table <- as.data.frame(do.call(rbind,lapply(names(reads),function(x){
    c(x,lm(molecules[,x]~reads[,x])$coef)
})))
names(regression_table) <- c("sample_id","Intercept","slope")
regression_table$Intercept <- as.numeric(as.character(regression_table$Intercept))
regression_table$slope <- as.numeric(as.character(regression_table$slope))


anno_regression <- merge(anno,regression_table,by="sample_id")

ggplot(anno_regression,aes(x=Intercept,y=slope,col=as.factor(individual),shape=as.factor(batch)))+geom_point()
ggplot(anno_regression,aes(x=Intercept,y=slope,shape=as.factor(individual),col=as.factor(batch)))+geom_point()
```

```{r I-want-to-learn}
data_use <- data.frame(y=c(molecules_cpm),sample_id=rep(colnames(molecules_cpm),each=dim(molecules_cpm)[1]))
data_use <- merge(data_use,anno,by="sample_id")
head(data_use)
data_use$individual <- "19098"
data_use[grep("19101",data_use$sample_id),"individual"] <- "19101"
data_use[grep("19239",data_use$sample_id),"individual"] <- "19239"

data_use$batch <- as.factor(substr(data_use$sample_id,3,9))
head(data_use)

library(nlme)
iii <- sample(1:dim(data_use)[1],10000)
fit2 <- lme(y~individual,random=~1|batch,data=data_use[iii,])


molecule_batch_removed <- (summary(fit2))$fitted [,2]

plot(data_use$y[iii],molecule_batch_removed)

pca_single_mixed_correction <- run_pca(molecule_batch_removed)
plot_pca(pca_single_mixed_correction$PCs, 
         explained = pca_single_mixed_correction$explained,
         metadata = anno_single, color = "individual",
         shape = "batch", factors = c("individual", "batch"))

```

## Session information

```{r info}
sessionInfo()
```
