<!DOCTYPE html>

<html xmlns="http://www.w3.org/1999/xhtml">

<head>

<meta charset="utf-8">
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="generator" content="pandoc" />

<meta name="author" content="PoYuan Tung" />

<meta name="date" content="2015-08-24" />

<title>Compare read and molecule counts in single cells per batch</title>

<script src="libs/jquery-1.11.0/jquery.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="libs/bootstrap-3.3.1/css/united.min.css" rel="stylesheet" />
<script src="libs/bootstrap-3.3.1/js/bootstrap.min.js"></script>
<script src="libs/bootstrap-3.3.1/shim/html5shiv.min.js"></script>
<script src="libs/bootstrap-3.3.1/shim/respond.min.js"></script>

<style type="text/css">

/* padding for bootstrap navbar */
body {
  padding-top: 50px;
  padding-bottom: 40px;
}


/* offset scroll position for anchor links (for fixed navbar)  */
.section h2 {
  padding-top: 55px;
  margin-top: -55px;
}
.section h3 {
  padding-top: 55px;
  margin-top: -55px;
}



/* don't use link color in navbar */
.dropdown-menu>li>a {
  color: black;
}

/* some padding for disqus */
#disqus_thread {
  margin-top: 45px;
}

</style>

<link rel="stylesheet" href="libs/font-awesome-4.1.0/css/font-awesome.min.css"/>

<style type="text/css">code{white-space: pre;}</style>
<link rel="stylesheet"
      href="libs/highlight/textmate.css"
      type="text/css" />
<script src="libs/highlight/highlight.js"></script>
<style type="text/css">
  pre:not([class]) {
    background-color: white;
  }
</style>
<script type="text/javascript">
if (window.hljs && document.readyState && document.readyState === "complete") {
   window.setTimeout(function() {
      hljs.initHighlighting();
   }, 0);
}
</script>



</head>

<body>

<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
code {
  color: inherit;
  background-color: rgba(0, 0, 0, 0.04);
}
img { 
  max-width:100%; 
  height: auto; 
}
</style>
<div class="container-fluid main-container">


<div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">singleCellSeq</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li><a href="index.html">Home</a></li>
        <li><a href="about.html">About</a></li>
        <li><a href="license.html">License</a></li>
        <li><a href="https://github.com/jdblischak/singleCellSeq">GitHub</a></li>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<div id="header">
<h1 class="title">Compare read and molecule counts in single cells per batch</h1>
<h4 class="author"><em>PoYuan Tung</em></h4>
<h4 class="date"><em>2015-08-24</em></h4>
</div>

<div id="TOC">
<ul>
<li><a href="#input">Input</a></li>
<li><a href="#distribution">Distribution</a></li>
<li><a href="#filter">Filter</a></li>
<li><a href="#transformation">Transformation</a></li>
<li><a href="#conversion-in-each-single-cell">conversion in each single cell</a><ul>
<li><a href="#counts">Counts</a></li>
<li><a href="#log2-counts">Log2 counts</a></li>
<li><a href="#counts-per-million">counts per million</a></li>
<li><a href="#tmm-normalized-counts-per-million">TMM-normalized counts per million</a></li>
</ul></li>
<li><a href="#session-information">Session information</a></li>
</ul>
</div>

<p><strong>Last updated:</strong> 2015-09-04</p>
<p><strong>Code version:</strong> 2ceea9387c87573d9a83285ea3beeb7e13e4c13a</p>
<p>Comparing the conversion of reads to molecules for each cell. Used three different metrics:</p>
<ul>
<li>Raw counts</li>
<li>Log2 counts (pseudocount of 1)</li>
<li>cpm counts (no log transformation)</li>
<li>Log2 TMM-normalized counts per million (pseudocount of 0.25)</li>
</ul>
<div id="input" class="section level2">
<h2>Input</h2>
<pre class="r"><code>library(&quot;dplyr&quot;)
library(&quot;ggplot2&quot;)
theme_set(theme_bw(base_size = 16))
library(&quot;edgeR&quot;)
source(&quot;functions.R&quot;)
library(&quot;tidyr&quot;)</code></pre>
<p>Input annotation.</p>
<pre class="r"><code>anno &lt;- read.table(&quot;../data/annotation.txt&quot;, header = TRUE,
                   stringsAsFactors = FALSE)
head(anno)</code></pre>
<pre><code>  individual batch well     sample_id
1      19098     1  A01 NA19098.1.A01
2      19098     1  A02 NA19098.1.A02
3      19098     1  A03 NA19098.1.A03
4      19098     1  A04 NA19098.1.A04
5      19098     1  A05 NA19098.1.A05
6      19098     1  A06 NA19098.1.A06</code></pre>
<p>Input read counts.</p>
<pre class="r"><code>reads &lt;- read.table(&quot;../data/reads.txt&quot;, header = TRUE,
                    stringsAsFactors = FALSE)</code></pre>
<p>Input molecule counts.</p>
<pre class="r"><code>molecules &lt;- read.table(&quot;../data/molecules.txt&quot;, header = TRUE,
                    stringsAsFactors = FALSE)</code></pre>
<p>Input list of quality single cells.</p>
<pre class="r"><code>quality_single_cells &lt;- scan(&quot;../data/quality-single-cells.txt&quot;,
                             what = &quot;character&quot;)</code></pre>
<p>Keep only the single cells that passed the <a href="qc-cell-ipsc.html">QC filters</a>.</p>
<pre class="r"><code>reads &lt;- reads[, colnames(reads) %in% quality_single_cells]
molecules &lt;- molecules[, colnames(molecules) %in% quality_single_cells]
anno &lt;- anno[anno$sample_id %in% quality_single_cells, ]
stopifnot(dim(reads) == dim(molecules),
          nrow(anno) == ncol(reads))</code></pre>
</div>
<div id="distribution" class="section level2">
<h2>Distribution</h2>
<p>Look at the distribution of fold change to mean.</p>
<pre class="r"><code>## calculate mean
reads_mean     &lt;- apply(reads, 1, mean)
molecules_mean &lt;- apply(molecules, 1, mean)
distribution &lt;- data.frame(reads_mean, molecules_mean)

## calculate fold change to mean
distribution$fold_change_read     &lt;- log2(reads_mean/mean(reads_mean))
distribution$fold_change_molecule &lt;- log2(molecules_mean/mean(molecules_mean))

## select ERCC
distribution$ERCC &lt;- grepl(&quot;ERCC&quot;, rownames(distribution))

## color palette
cbPalette &lt;- c(&quot;#999999&quot;, &quot;#0000FF&quot;, &quot;#990033&quot;, &quot;#F0E442&quot;, &quot;#0072B2&quot;, &quot;#D55E00&quot;, &quot;#CC79A7&quot;, &quot;#009E73&quot;)

ggplot(distribution, aes(x = fold_change_molecule, y = fold_change_read, col = ERCC)) + geom_point(size = 3, alpha = 0.5) + scale_colour_manual(values=cbPalette) + stat_function(fun= function(x) {x}, col= &quot;#56B4E9&quot;) + labs(x = &quot;log2 fold change to mean (molecule)&quot;, y =  &quot;log2 fold change to mean (reads)&quot;)</code></pre>
<p><img src="figure/compare-reads-v-molecules-in-cells.Rmd/distribution-1.png" title="" alt="" width="672" style="display: block; margin: auto;" /></p>
</div>
<div id="filter" class="section level2">
<h2>Filter</h2>
<p>Only keep the following genes:</p>
<ul>
<li>ERCC genes with at least one molecule observed in at least one single cell</li>
</ul>
<pre class="r"><code>ercc_keep &lt;- rownames(molecules)[grepl(&quot;ERCC&quot;, rownames(molecules)) &amp;
                                 rowSums(molecules) &gt; 1]</code></pre>
<ul>
<li>The top expressed endogenous genes in terms of mean molecule counts per million</li>
</ul>
<pre class="r"><code>num_genes &lt;- 12000
mean_cpm &lt;- molecules %&gt;%
  filter(!grepl(&quot;ERCC&quot;, rownames(molecules))) %&gt;%
  cpm %&gt;%
  rowMeans
gene_keep &lt;- rownames(molecules)[!grepl(&quot;ERCC&quot;, rownames(molecules))][order(mean_cpm, decreasing = TRUE)][1:num_genes]</code></pre>
<p>Filter the genes:</p>
<pre class="r"><code>reads &lt;- reads[rownames(reads) %in% c(gene_keep, ercc_keep), ]
molecules &lt;- molecules[rownames(molecules) %in% c(gene_keep, ercc_keep), ]</code></pre>
</div>
<div id="transformation" class="section level2">
<h2>Transformation</h2>
<p>In addition to comparing the raw counts of reads and molecules, we compare the log2 counts and the log counts per million.</p>
<p>For the log counts, I add a pseudocount of 1.</p>
<pre class="r"><code>reads_log &lt;- log2(reads + 1)
molecules_log &lt;- log2(molecules + 1)</code></pre>
<p>standardized by cmp and log-transform</p>
<pre class="r"><code>reads_cpm &lt;- cpm(reads, log = FALSE)
molecules_cpm &lt;- cpm(molecules, log = FALSE)</code></pre>
<p>Calculate cpm for the reads data using TMM-normalization.</p>
<pre class="r"><code>norm_factors_reads &lt;- calcNormFactors(reads, method = &quot;TMM&quot;)
reads_tmm &lt;- cpm(reads, lib.size = colSums(reads) * norm_factors_reads,
                 log = TRUE)</code></pre>
<p>And for the molecules.</p>
<pre class="r"><code>norm_factors_mol &lt;- calcNormFactors(molecules, method = &quot;TMM&quot;)
molecules_tmm &lt;- cpm(molecules, lib.size = colSums(molecules) * norm_factors_mol,
                     log = TRUE)</code></pre>
</div>
<div id="conversion-in-each-single-cell" class="section level2">
<h2>conversion in each single cell</h2>
<div id="counts" class="section level3">
<h3>Counts</h3>
<p>Compare the counts. (1) all genes. (2) only ERCC. (3) without ERCC, only endogenous genes</p>
<pre class="r"><code>## linear regression per cell and make a table with intercept, slope, and r-squared
regression_table &lt;- as.data.frame(do.call(rbind,lapply(names(reads),function(x){
  fit.temp &lt;- lm(molecules[,x]~reads[,x])
    c(x,fit.temp$coefficients,summary(fit.temp)$adj.r.squared)
})))
names(regression_table) &lt;- c(&quot;sample_id&quot;,&quot;Intercept&quot;,&quot;slope&quot;,&quot;r2&quot;)
regression_table$Intercept &lt;- as.numeric(as.character(regression_table$Intercept))
regression_table$slope &lt;- as.numeric(as.character(regression_table$slope))
regression_table$r2 &lt;- as.numeric(as.character(regression_table$r2))
plot(regression_table$r2)</code></pre>
<p><img src="figure/compare-reads-v-molecules-in-cells.Rmd/conversion-in-cell-1.png" title="" alt="" width="864" style="display: block; margin: auto;" /></p>
<pre class="r"><code>anno_regression &lt;- merge(anno,regression_table,by=&quot;sample_id&quot;)

ggplot(anno_regression,aes(x=Intercept,y=slope,col=as.factor(individual),shape=as.factor(batch))) + geom_point() + labs(x = &quot;intercept&quot;, y = &quot;slope&quot;, title = &quot;read-molecule conversion (counts)&quot;)</code></pre>
<p><img src="figure/compare-reads-v-molecules-in-cells.Rmd/conversion-in-cell-2.png" title="" alt="" width="864" style="display: block; margin: auto;" /></p>
<pre class="r"><code>ggplot(anno_regression,aes(x=Intercept,y=slope,col=as.factor(individual),shape=as.factor(batch))) + geom_point() + labs(x = &quot;intercept&quot;, y = &quot;slope&quot;, title = &quot;read-molecule conversion (counts)&quot;) + facet_grid(individual ~ batch)</code></pre>
<p><img src="figure/compare-reads-v-molecules-in-cells.Rmd/conversion-in-cell-3.png" title="" alt="" width="864" style="display: block; margin: auto;" /></p>
<pre class="r"><code>## plot all the lines
anno_regression$reads_mean &lt;- apply(reads, 2, mean)
anno_regression$molecules_mean &lt;- apply(molecules, 2, mean)

ggplot(anno_regression, aes(x= reads_mean, y= molecules_mean)) + geom_point() + geom_abline(aes(intercept=Intercept, slope=slope, col=as.factor(individual), alpha= 0.5), data=anno_regression) </code></pre>
<p><img src="figure/compare-reads-v-molecules-in-cells.Rmd/conversion-in-cell-4.png" title="" alt="" width="864" style="display: block; margin: auto;" /></p>
<pre class="r"><code>ggplot(anno_regression, aes(x= reads_mean, y= molecules_mean)) + geom_point() + geom_abline(aes(intercept=Intercept, slope=slope, col=as.factor(individual), alpha= 0.5), data=anno_regression) + facet_grid(individual ~ batch)</code></pre>
<p><img src="figure/compare-reads-v-molecules-in-cells.Rmd/conversion-in-cell-5.png" title="" alt="" width="864" style="display: block; margin: auto;" /></p>
<p>Only ERCC genes</p>
<pre class="r"><code>## grep ERCC
reads_ERCC &lt;- reads[grep(&quot;ERCC&quot;, rownames(reads)), ]
molecules_ERCC &lt;- molecules[grep(&quot;ERCC&quot;, rownames(molecules)), ]

## linear regression per cell and make a table with intercept, slope, and r-squared
regression_table_ERCC &lt;- as.data.frame(do.call(rbind,lapply(names(reads_ERCC),function(x){
  fit.temp &lt;- lm(molecules_ERCC[,x]~reads_ERCC[,x])
    c(x,fit.temp$coefficients,summary(fit.temp)$adj.r.squared)
})))
names(regression_table_ERCC) &lt;- c(&quot;sample_id&quot;,&quot;Intercept&quot;,&quot;slope&quot;,&quot;r2&quot;)
regression_table_ERCC$Intercept &lt;- as.numeric(as.character(regression_table_ERCC$Intercept))
regression_table_ERCC$slope &lt;- as.numeric(as.character(regression_table_ERCC$slope))
regression_table_ERCC$r2 &lt;- as.numeric(as.character(regression_table_ERCC$r2))
plot(regression_table_ERCC$r2)</code></pre>
<p><img src="figure/compare-reads-v-molecules-in-cells.Rmd/conversion-in-cell-ERCC-1.png" title="" alt="" width="864" style="display: block; margin: auto;" /></p>
<pre class="r"><code>anno_regression_ERCC &lt;- merge(anno,regression_table_ERCC,by=&quot;sample_id&quot;)
ggplot(anno_regression_ERCC,aes(x=Intercept,y=slope,col=as.factor(individual),shape=as.factor(batch))) + geom_point() + labs(x = &quot;intercept&quot;, y = &quot;slope&quot;, title = &quot;read-molecule conversion ERCC genes (counts)&quot;) + facet_grid(individual ~ batch)</code></pre>
<p><img src="figure/compare-reads-v-molecules-in-cells.Rmd/conversion-in-cell-ERCC-2.png" title="" alt="" width="864" style="display: block; margin: auto;" /></p>
<pre class="r"><code>## plot all the lines
anno_regression_ERCC$reads_mean &lt;- apply(reads_ERCC, 2, mean)
anno_regression_ERCC$molecules_mean &lt;- apply(molecules_ERCC, 2, mean)

ggplot(anno_regression_ERCC, aes(x= reads_mean, y= molecules_mean)) + geom_point() + geom_abline(aes(intercept=Intercept, slope=slope, col=as.factor(individual), alpha= 0.5), data=anno_regression_ERCC) </code></pre>
<p><img src="figure/compare-reads-v-molecules-in-cells.Rmd/conversion-in-cell-ERCC-3.png" title="" alt="" width="864" style="display: block; margin: auto;" /></p>
<pre class="r"><code>ggplot(anno_regression_ERCC, aes(x= reads_mean, y= molecules_mean)) + geom_point() + geom_abline(aes(intercept=Intercept, slope=slope, col=as.factor(individual), alpha= 0.5), data=anno_regression) + facet_grid(individual ~ batch)</code></pre>
<p><img src="figure/compare-reads-v-molecules-in-cells.Rmd/conversion-in-cell-ERCC-4.png" title="" alt="" width="864" style="display: block; margin: auto;" /></p>
<p>without ERCC genes, only endogenous genes</p>
<pre class="r"><code>## grep ENSG
reads_ENSG &lt;- reads[grep(&quot;ENSG&quot;, rownames(reads)), ]
molecules_ENSG &lt;- molecules[grep(&quot;ENSG&quot;, rownames(molecules)), ]

## linear regression per cell and make a table with intercept, slope, and r-squared
regression_table_ENSG &lt;- as.data.frame(do.call(rbind,lapply(names(reads_ENSG),function(x){
  fit.temp &lt;- lm(molecules_ENSG[,x]~reads_ENSG[,x])
    c(x,fit.temp$coefficients,summary(fit.temp)$adj.r.squared)
})))
names(regression_table_ENSG) &lt;- c(&quot;sample_id&quot;,&quot;Intercept&quot;,&quot;slope&quot;,&quot;r2&quot;)
regression_table_ENSG$Intercept &lt;- as.numeric(as.character(regression_table_ENSG$Intercept))
regression_table_ENSG$slope &lt;- as.numeric(as.character(regression_table_ENSG$slope))
regression_table_ENSG$r2 &lt;- as.numeric(as.character(regression_table_ENSG$r2))
plot(regression_table_ENSG$r2)</code></pre>
<p><img src="figure/compare-reads-v-molecules-in-cells.Rmd/conversion-in-cell-gene-1.png" title="" alt="" width="864" style="display: block; margin: auto;" /></p>
<pre class="r"><code>anno_regression_ENSG &lt;- merge(anno,regression_table_ENSG,by=&quot;sample_id&quot;)
ggplot(anno_regression_ENSG,aes(x=Intercept,y=slope,col=as.factor(individual),shape=as.factor(batch))) + geom_point() + labs(x = &quot;intercept&quot;, y = &quot;slope&quot;, title = &quot;read-molecule conversion endogenous genes (counts)&quot;)</code></pre>
<p><img src="figure/compare-reads-v-molecules-in-cells.Rmd/conversion-in-cell-gene-2.png" title="" alt="" width="864" style="display: block; margin: auto;" /></p>
<pre class="r"><code>ggplot(anno_regression_ENSG,aes(x=Intercept,y=slope,col=as.factor(individual),shape=as.factor(batch))) + geom_point() + labs(x = &quot;intercept&quot;, y = &quot;slope&quot;, title = &quot;read-molecule conversion endogenous genes (counts)&quot;) + facet_grid(individual ~ batch)</code></pre>
<p><img src="figure/compare-reads-v-molecules-in-cells.Rmd/conversion-in-cell-gene-3.png" title="" alt="" width="864" style="display: block; margin: auto;" /></p>
<pre class="r"><code>## plot all the lines
anno_regression_ENSG$reads_mean &lt;- apply(reads_ENSG, 2, mean)
anno_regression_ENSG$molecules_mean &lt;- apply(molecules_ENSG, 2, mean)

ggplot(anno_regression_ENSG, aes(x= reads_mean, y= molecules_mean)) + geom_point() + geom_abline(aes(intercept=Intercept, slope=slope, col=as.factor(individual), alpha= 0.5), data=anno_regression_ENSG) </code></pre>
<p><img src="figure/compare-reads-v-molecules-in-cells.Rmd/conversion-in-cell-gene-4.png" title="" alt="" width="864" style="display: block; margin: auto;" /></p>
<pre class="r"><code>ggplot(anno_regression_ENSG, aes(x= reads_mean, y= molecules_mean)) + geom_point() + geom_abline(aes(intercept=Intercept, slope=slope, col=as.factor(individual), alpha= 0.5), data=anno_regression_ENSG) + facet_grid(individual ~ batch)</code></pre>
<p><img src="figure/compare-reads-v-molecules-in-cells.Rmd/conversion-in-cell-gene-5.png" title="" alt="" width="864" style="display: block; margin: auto;" /></p>
<pre class="r"><code>## 19098 have more total endogenous gene molecules and more ERCC molecule
anno_regression$total_molecule_ERCC &lt;- apply(molecules_ERCC, 2, sum)
anno_regression$total_molecule_ENSG &lt;- apply(molecules_ENSG, 2, sum)
anno_regression$total_molecule &lt;- apply(molecules, 2, sum) 
ggplot(anno_regression, aes(x= total_molecule_ERCC, y= total_molecule,col=as.factor(individual),shape=as.factor(batch))) + geom_point()</code></pre>
<p><img src="figure/compare-reads-v-molecules-in-cells.Rmd/total-molecule-1.png" title="" alt="" width="864" style="display: block; margin: auto;" /></p>
<pre class="r"><code>ggplot(anno_regression, aes(x = as.factor(individual), y = total_molecule_ENSG)) + geom_boxplot(aes(fill = as.factor(batch)))</code></pre>
<p><img src="figure/compare-reads-v-molecules-in-cells.Rmd/total-molecule-2.png" title="" alt="" width="864" style="display: block; margin: auto;" /></p>
<pre class="r"><code>ggplot(anno_regression, aes(x = as.factor(individual), y = total_molecule_ERCC)) + geom_boxplot(aes(fill = as.factor(batch)))</code></pre>
<p><img src="figure/compare-reads-v-molecules-in-cells.Rmd/total-molecule-3.png" title="" alt="" width="864" style="display: block; margin: auto;" /></p>
</div>
<div id="log2-counts" class="section level3">
<h3>Log2 counts</h3>
<p>Compare the log counts. The r squared values of lm are relativiely low (around 0.85).</p>
<pre class="r"><code>regression_table_log &lt;- as.data.frame(do.call(rbind,lapply(names(reads_log),function(x){
  fit.temp &lt;- lm(molecules_log[,x]~reads_log[,x])
    c(x,fit.temp$coefficients,summary(fit.temp)$adj.r.squared)
})))
names(regression_table_log) &lt;- c(&quot;sample_id&quot;,&quot;Intercept&quot;,&quot;slope&quot;,&quot;r2&quot;)
regression_table_log$Intercept &lt;- as.numeric(as.character(regression_table_log$Intercept))
regression_table_log$slope &lt;- as.numeric(as.character(regression_table_log$slope))
regression_table_log$r2 &lt;- as.numeric(as.character(regression_table_log$r2))
plot(regression_table_log$r2)</code></pre>
<p><img src="figure/compare-reads-v-molecules-in-cells.Rmd/conversion-in-cell-log-1.png" title="" alt="" width="864" style="display: block; margin: auto;" /></p>
<pre class="r"><code>anno_regression_log &lt;- merge(anno,regression_table_log,by=&quot;sample_id&quot;)
ggplot(anno_regression_log,aes(x=Intercept,y=slope,col=as.factor(individual),shape=as.factor(batch))) + geom_point() + labs(x = &quot;intercept&quot;, y = &quot;slope&quot;, title = &quot;read-molecule conversion (log2)&quot;)</code></pre>
<p><img src="figure/compare-reads-v-molecules-in-cells.Rmd/conversion-in-cell-log-2.png" title="" alt="" width="864" style="display: block; margin: auto;" /></p>
</div>
<div id="counts-per-million" class="section level3">
<h3>counts per million</h3>
<p>Compare the counts per million.</p>
<pre class="r"><code>regression_table_cpm &lt;- as.data.frame(do.call(rbind,lapply(colnames(reads_cpm),function(x){
  fit.temp &lt;- lm(molecules_cpm[,x]~reads_cpm[,x])
    c(x,fit.temp$coefficients,summary(fit.temp)$adj.r.squared)
})))
names(regression_table_cpm) &lt;- c(&quot;sample_id&quot;,&quot;Intercept&quot;,&quot;slope&quot;,&quot;r2&quot;)
regression_table_cpm$Intercept &lt;- as.numeric(as.character(regression_table_cpm$Intercept))
regression_table_cpm$slope &lt;- as.numeric(as.character(regression_table_cpm$slope))
regression_table_cpm$r2 &lt;- as.numeric(as.character(regression_table_cpm$r2))
plot(regression_table_cpm$r2)</code></pre>
<p><img src="figure/compare-reads-v-molecules-in-cells.Rmd/conversion-in-cell-cpm-1.png" title="" alt="" width="864" style="display: block; margin: auto;" /></p>
<pre class="r"><code>anno_regression_cpm &lt;- merge(anno,regression_table_cpm,by=&quot;sample_id&quot;)

ggplot(anno_regression_cpm,aes(x=Intercept,y=slope,col=as.factor(individual),shape=as.factor(batch))) + geom_point() + labs(x = &quot;intercept&quot;, y = &quot;slope&quot;, title = &quot;read-molecule conversion (cpm no log)&quot;)</code></pre>
<p><img src="figure/compare-reads-v-molecules-in-cells.Rmd/conversion-in-cell-cpm-2.png" title="" alt="" width="864" style="display: block; margin: auto;" /></p>
<pre class="r"><code>ggplot(anno_regression_cpm,aes(x=Intercept,y=slope,col=as.factor(individual),shape=as.factor(batch))) + geom_point() + labs(x = &quot;intercept&quot;, y = &quot;slope&quot;, title = &quot;read-molecule conversion (cpm no log)&quot;) + facet_grid(individual ~ batch) </code></pre>
<p><img src="figure/compare-reads-v-molecules-in-cells.Rmd/conversion-in-cell-cpm-3.png" title="" alt="" width="864" style="display: block; margin: auto;" /></p>
<pre class="r"><code>## plot all the lines
anno_regression_cpm$reads_mean &lt;- apply(reads_cpm, 2, mean)
anno_regression_cpm$molecules_mean &lt;- apply(molecules_cpm, 2, mean)

ggplot(anno_regression_cpm, aes(x= reads_mean, y= molecules_mean)) + geom_point() + geom_abline(aes(intercept=Intercept, slope=slope, col=as.factor(individual), alpha = 0.01), data=anno_regression_cpm) + facet_grid(individual ~ batch) </code></pre>
<p><img src="figure/compare-reads-v-molecules-in-cells.Rmd/conversion-in-cell-cpm-4.png" title="" alt="" width="864" style="display: block; margin: auto;" /></p>
<pre class="r"><code>## look at just the slope using total raw molecule counts (total cpm molecule counts are all the same across cells)
anno_regression_cpm$total_molecule &lt;- apply(molecules, 2, sum)
ggplot(anno_regression_cpm, aes(x= total_molecule, y= slope,col=as.factor(individual),shape=as.factor(batch))) + geom_point()</code></pre>
<p><img src="figure/compare-reads-v-molecules-in-cells.Rmd/conversion-in-cell-cpm-5.png" title="" alt="" width="864" style="display: block; margin: auto;" /></p>
<p>Only ERCC genes</p>
<pre class="r"><code>## grep ERCC
reads_cpm_ERCC &lt;- as.data.frame(reads_cpm[grep(&quot;ERCC&quot;, rownames(reads_cpm)), ])
molecules_cpm_ERCC &lt;- as.data.frame(molecules_cpm[grep(&quot;ERCC&quot;, rownames(molecules_cpm)), ])

## linear regression per cell and make a table with intercept, slope, and r-squared
regression_table_cpm_ERCC &lt;- as.data.frame(do.call(rbind,lapply(names(reads_cpm_ERCC),function(x){
  fit.temp &lt;- lm(molecules_cpm_ERCC[,x]~reads_cpm_ERCC[,x])
    c(x,fit.temp$coefficients,summary(fit.temp)$adj.r.squared)
})))
names(regression_table_cpm_ERCC) &lt;- c(&quot;sample_id&quot;,&quot;Intercept&quot;,&quot;slope&quot;,&quot;r2&quot;)
regression_table_cpm_ERCC$Intercept &lt;- as.numeric(as.character(regression_table_cpm_ERCC$Intercept))
regression_table_cpm_ERCC$slope &lt;- as.numeric(as.character(regression_table_cpm_ERCC$slope))
regression_table_cpm_ERCC$r2 &lt;- as.numeric(as.character(regression_table_cpm_ERCC$r2))
plot(regression_table_cpm_ERCC$r2)</code></pre>
<p><img src="figure/compare-reads-v-molecules-in-cells.Rmd/conversion-in-cell-cpm-ERCC-1.png" title="" alt="" width="864" style="display: block; margin: auto;" /></p>
<pre class="r"><code>anno_regression_cpm_ERCC &lt;- merge(anno,regression_table_cpm_ERCC,by=&quot;sample_id&quot;)
ggplot(anno_regression_cpm_ERCC,aes(x=Intercept,y=slope,col=as.factor(individual),shape=as.factor(batch))) + geom_point() + labs(x = &quot;intercept&quot;, y = &quot;slope&quot;, title = &quot;read-molecule conversion ERCC genes (cpm no log)&quot;) </code></pre>
<p><img src="figure/compare-reads-v-molecules-in-cells.Rmd/conversion-in-cell-cpm-ERCC-2.png" title="" alt="" width="864" style="display: block; margin: auto;" /></p>
<pre class="r"><code>## plot all the lines
anno_regression_cpm_ERCC$reads_mean &lt;- apply(reads_cpm_ERCC, 2, mean)
anno_regression_cpm_ERCC$molecules_mean &lt;- apply(molecules_cpm_ERCC, 2, mean)

ggplot(anno_regression_cpm_ERCC, aes(x= reads_mean, y= molecules_mean)) + geom_point() + geom_abline(aes(intercept=Intercept, slope=slope, col=as.factor(individual), alpha= 0.5), data=anno_regression_cpm_ERCC) </code></pre>
<p><img src="figure/compare-reads-v-molecules-in-cells.Rmd/conversion-in-cell-cpm-ERCC-3.png" title="" alt="" width="864" style="display: block; margin: auto;" /></p>
<pre class="r"><code>ggplot(anno_regression_ERCC, aes(x= reads_mean, y= molecules_mean)) + geom_point() + geom_abline(aes(intercept=Intercept, slope=slope, col=as.factor(individual), alpha= 0.5), data=anno_regression) + facet_grid(individual ~ batch)</code></pre>
<p><img src="figure/compare-reads-v-molecules-in-cells.Rmd/conversion-in-cell-cpm-ERCC-4.png" title="" alt="" width="864" style="display: block; margin: auto;" /></p>
</div>
<div id="tmm-normalized-counts-per-million" class="section level3">
<h3>TMM-normalized counts per million</h3>
<p>Compare TMM-normalized log2-transformed cpm .</p>
<pre class="r"><code>regression_table_tmm &lt;- as.data.frame(do.call(rbind,lapply(colnames(reads_tmm),function(x){
  fit.temp &lt;- lm(molecules_tmm[,x]~reads_tmm[,x])
    c(x,fit.temp$coefficients,summary(fit.temp)$adj.r.squared)
})))
names(regression_table_tmm) &lt;- c(&quot;sample_id&quot;,&quot;Intercept&quot;,&quot;slope&quot;,&quot;r2&quot;)
regression_table_tmm$Intercept &lt;- as.numeric(as.character(regression_table_tmm$Intercept))
regression_table_tmm$slope &lt;- as.numeric(as.character(regression_table_tmm$slope))
regression_table_tmm$r2 &lt;- as.numeric(as.character(regression_table_tmm$r2))
plot(regression_table_tmm$r2)</code></pre>
<p><img src="figure/compare-reads-v-molecules-in-cells.Rmd/conversion-in-cell-tmm-1.png" title="" alt="" width="864" style="display: block; margin: auto;" /></p>
<pre class="r"><code>anno_regression_tmm &lt;- merge(anno,regression_table_tmm,by=&quot;sample_id&quot;)

ggplot(anno_regression_tmm,aes(x=Intercept,y=slope,col=as.factor(individual),shape=as.factor(batch))) + geom_point() + labs(x = &quot;intercept&quot;, y = &quot;slope&quot;, title = &quot;read-molecule conversion (tmm cpm log)&quot;)</code></pre>
<p><img src="figure/compare-reads-v-molecules-in-cells.Rmd/conversion-in-cell-tmm-2.png" title="" alt="" width="864" style="display: block; margin: auto;" /></p>
<pre class="r"><code>ggplot(anno_regression_tmm,aes(x=Intercept,y=slope,col=as.factor(individual),shape=as.factor(batch))) + geom_point() + labs(x = &quot;intercept&quot;, y = &quot;slope&quot;, title = &quot;read-molecule conversion (tmm cpm log)&quot;) + facet_grid(individual ~ batch) </code></pre>
<p><img src="figure/compare-reads-v-molecules-in-cells.Rmd/conversion-in-cell-tmm-3.png" title="" alt="" width="864" style="display: block; margin: auto;" /></p>
<pre class="r"><code>## plot all the lines
anno_regression_tmm$reads_mean &lt;- apply(reads_tmm, 2, mean)
anno_regression_tmm$molecules_mean &lt;- apply(molecules_tmm, 2, mean)

ggplot(anno_regression_tmm, aes(x= reads_mean, y= molecules_mean)) + geom_point() + geom_abline(aes(intercept=Intercept, slope=slope, col=as.factor(individual), alpha = 0.01), data=anno_regression_tmm) </code></pre>
<p><img src="figure/compare-reads-v-molecules-in-cells.Rmd/conversion-in-cell-tmm-4.png" title="" alt="" width="864" style="display: block; margin: auto;" /></p>
<pre class="r"><code>ggplot(anno_regression_tmm, aes(x= reads_mean, y= molecules_mean)) + geom_point() + geom_abline(aes(intercept=Intercept, slope=slope, col=as.factor(individual), alpha = 0.01), data=anno_regression_tmm) + facet_grid(individual ~ batch) </code></pre>
<p><img src="figure/compare-reads-v-molecules-in-cells.Rmd/conversion-in-cell-tmm-5.png" title="" alt="" width="864" style="display: block; margin: auto;" /></p>
</div>
</div>
<div id="session-information" class="section level2">
<h2>Session information</h2>
<pre class="r"><code>sessionInfo()</code></pre>
<pre><code>R version 3.2.0 (2015-04-16)
Platform: x86_64-unknown-linux-gnu (64-bit)

locale:
 [1] LC_CTYPE=en_US.UTF-8       LC_NUMERIC=C              
 [3] LC_TIME=en_US.UTF-8        LC_COLLATE=en_US.UTF-8    
 [5] LC_MONETARY=en_US.UTF-8    LC_MESSAGES=en_US.UTF-8   
 [7] LC_PAPER=en_US.UTF-8       LC_NAME=C                 
 [9] LC_ADDRESS=C               LC_TELEPHONE=C            
[11] LC_MEASUREMENT=en_US.UTF-8 LC_IDENTIFICATION=C       

attached base packages:
[1] stats     graphics  grDevices utils     datasets  methods   base     

other attached packages:
[1] tidyr_0.2.0   edgeR_3.10.2  limma_3.24.9  ggplot2_1.0.1 dplyr_0.4.2  
[6] knitr_1.10.5 

loaded via a namespace (and not attached):
 [1] Rcpp_0.12.0      magrittr_1.5     MASS_7.3-40      munsell_0.4.2   
 [5] colorspace_1.2-6 R6_2.1.1         stringr_1.0.0    plyr_1.8.3      
 [9] tools_3.2.0      parallel_3.2.0   grid_3.2.0       gtable_0.1.2    
[13] DBI_0.3.1        htmltools_0.2.6  lazyeval_0.1.10  yaml_2.1.13     
[17] assertthat_0.1   digest_0.6.8     reshape2_1.4.1   formatR_1.2     
[21] evaluate_0.7     rmarkdown_0.6.1  labeling_0.3     stringi_0.4-1   
[25] scales_0.2.4     proto_0.3-10    </code></pre>
</div>


<!-- some extra javascript for older browsers -->
<script type="text/javascript" src="libs/polyfill.js"></script>

<script>

// manage active state of menu based on current page
$(document).ready(function () {

    // active menu
    href = window.location.pathname
    href = href.substr(href.lastIndexOf('/') + 1)
    $('a[href="' + href + '"]').parent().addClass('active');

    // manage active menu header
    if (href.startsWith('authoring_'))
      $('a[href="' + 'authoring' + '"]').parent().addClass('active');
    else if (href.endsWith('_format.html'))
      $('a[href="' + 'formats' + '"]').parent().addClass('active');
    else if (href.startsWith('developer_'))
      $('a[href="' + 'developer' + '"]').parent().addClass('active');

});

</script>

</div>

<script>

// add bootstrap table styles to pandoc tables
$(document).ready(function () {
  $('tr.header').parent('thead').parent('table').addClass('table table-condensed');
});

</script>


</body>
</html>
