<!DOCTYPE html>

<html xmlns="http://www.w3.org/1999/xhtml">

<head>

<meta charset="utf-8">
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="generator" content="pandoc" />

<meta name="author" content="PoYuan Tung &amp; Joyce Hsiao" />

<meta name="date" content="2015-09-22" />

<title>Compare read and molecule counts in single cells per batch</title>

<script src="libs/jquery-1.11.0/jquery.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="libs/bootstrap-3.3.1/css/united.min.css" rel="stylesheet" />
<script src="libs/bootstrap-3.3.1/js/bootstrap.min.js"></script>
<script src="libs/bootstrap-3.3.1/shim/html5shiv.min.js"></script>
<script src="libs/bootstrap-3.3.1/shim/respond.min.js"></script>

<style type="text/css">

/* padding for bootstrap navbar */
body {
  padding-top: 50px;
  padding-bottom: 40px;
}


/* offset scroll position for anchor links (for fixed navbar)  */
.section h2 {
  padding-top: 55px;
  margin-top: -55px;
}
.section h3 {
  padding-top: 55px;
  margin-top: -55px;
}



/* don't use link color in navbar */
.dropdown-menu>li>a {
  color: black;
}

/* some padding for disqus */
#disqus_thread {
  margin-top: 45px;
}

</style>

<link rel="stylesheet" href="libs/font-awesome-4.1.0/css/font-awesome.min.css"/>

<style type="text/css">code{white-space: pre;}</style>
<link rel="stylesheet"
      href="libs/highlight/textmate.css"
      type="text/css" />
<script src="libs/highlight/highlight.js"></script>
<style type="text/css">
  pre:not([class]) {
    background-color: white;
  }
</style>
<script type="text/javascript">
if (window.hljs && document.readyState && document.readyState === "complete") {
   window.setTimeout(function() {
      hljs.initHighlighting();
   }, 0);
}
</script>



</head>

<body>

<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
code {
  color: inherit;
  background-color: rgba(0, 0, 0, 0.04);
}
img { 
  max-width:100%; 
  height: auto; 
}
</style>
<div class="container-fluid main-container">


<div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">singleCellSeq</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li><a href="index.html">Home</a></li>
        <li><a href="about.html">About</a></li>
        <li><a href="license.html">License</a></li>
        <li><a href="https://github.com/jdblischak/singleCellSeq">GitHub</a></li>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<div id="header">
<h1 class="title">Compare read and molecule counts in single cells per batch</h1>
<h4 class="author"><em>PoYuan Tung &amp; Joyce Hsiao</em></h4>
<h4 class="date"><em>2015-09-22</em></h4>
</div>

<div id="TOC">
<ul>
<li><a href="#input">Input</a></li>
<li><a href="#distribution-of-fold-change-to-mean">Distribution of fold change to mean</a></li>
<li><a href="#transformation">Transformation</a></li>
<li><a href="#conversion-in-each-single-cell">conversion in each single cell</a><ul>
<li><a href="#counts">Counts</a></li>
<li><a href="#log2-counts">Log2 counts</a></li>
<li><a href="#counts-per-million">counts per million</a></li>
<li><a href="#tmm-normalized-counts-per-million">TMM-normalized counts per million</a></li>
</ul></li>
<li><a href="#session-information">Session information</a></li>
</ul>
</div>

<p><strong>Last updated:</strong> 2015-09-24</p>
<p><strong>Code version:</strong> 9d87919cfee4ceb697409c1ff079d1ed897d8e73</p>
<p>Comparing the conversion of reads to molecules for each cell. Used three different metrics:</p>
<ul>
<li>Raw counts</li>
<li>Log2 counts (pseudocount of 1)</li>
<li>cpm counts (no log transformation)</li>
<li>Log2 TMM-normalized counts per million (pseudocount of 0.25)</li>
</ul>
<div id="input" class="section level2">
<h2>Input</h2>
<pre class="r"><code>library(&quot;dplyr&quot;)
library(&quot;ggplot2&quot;)
theme_set(theme_bw(base_size = 16))
library(&quot;edgeR&quot;)
source(&quot;functions.R&quot;)
library(&quot;tidyr&quot;)</code></pre>
<p>Input annotation.</p>
<pre class="r"><code>anno &lt;- read.table(&quot;../data/annotation.txt&quot;, header = TRUE,
                   stringsAsFactors = FALSE)
head(anno)</code></pre>
<pre><code>  individual batch well     sample_id
1      19098     1  A01 NA19098.1.A01
2      19098     1  A02 NA19098.1.A02
3      19098     1  A03 NA19098.1.A03
4      19098     1  A04 NA19098.1.A04
5      19098     1  A05 NA19098.1.A05
6      19098     1  A06 NA19098.1.A06</code></pre>
<p>Input read counts.</p>
<pre class="r"><code>reads &lt;- read.table(&quot;../data/reads.txt&quot;, header = TRUE,
                    stringsAsFactors = FALSE)</code></pre>
<p>Input molecule counts.</p>
<pre class="r"><code>molecules &lt;- read.table(&quot;../data/molecules.txt&quot;, header = TRUE,
                    stringsAsFactors = FALSE)</code></pre>
<p>Input list of quality single cells.</p>
<pre class="r"><code>quality_single_cells &lt;- scan(&quot;../data/quality-single-cells.txt&quot;,
                             what = &quot;character&quot;)</code></pre>
<p>Keep only the single cells that passed the <a href="qc-cell-ipsc.html">QC filters</a>.</p>
<pre class="r"><code>reads &lt;- reads[, colnames(reads) %in% quality_single_cells]
molecules &lt;- molecules[, colnames(molecules) %in% quality_single_cells]
anno &lt;- anno[anno$sample_id %in% quality_single_cells, ]
stopifnot(dim(reads) == dim(molecules),
          nrow(anno) == ncol(reads))</code></pre>
</div>
<div id="distribution-of-fold-change-to-mean" class="section level2">
<h2>Distribution of fold change to mean</h2>
<p>Look at the distribution of fold change to mean. As being reported by others, the lowly expressed genes show divergent read and molecule counts</p>
<pre class="r"><code>## calculate mean
reads_mean     &lt;- apply(reads, 1, mean)
molecules_mean &lt;- apply(molecules, 1, mean)
distribution &lt;- data.frame(reads_mean, molecules_mean)

## calculate fold change to mean
distribution$fold_change_read     &lt;- log2(reads_mean/mean(reads_mean))
distribution$fold_change_molecule &lt;- log2(molecules_mean/mean(molecules_mean))

## select ERCC
distribution$ERCC &lt;- grepl(&quot;ERCC&quot;, rownames(distribution))

## color palette
cbPalette &lt;- c(&quot;#999999&quot;, &quot;#0000FF&quot;, &quot;#990033&quot;, &quot;#F0E442&quot;, &quot;#0072B2&quot;, &quot;#D55E00&quot;, &quot;#CC79A7&quot;, &quot;#009E73&quot;)

ggplot(distribution, aes(x = fold_change_molecule, y = fold_change_read, col = ERCC)) + geom_point(size = 3, alpha = 0.5) + scale_colour_manual(values=cbPalette) + stat_function(fun= function(x) {x}, col= &quot;#56B4E9&quot;) + labs(x = &quot;log2 fold change to mean (molecule)&quot;, y =  &quot;log2 fold change to mean (reads)&quot;)</code></pre>
<p><img src="figure/compare-reads-v-molecules-in-cells.Rmd/distribution-1.png" title="" alt="" width="672" style="display: block; margin: auto;" /></p>
</div>
<div id="transformation" class="section level2">
<h2>Transformation</h2>
<p>In addition to comparing the raw counts of reads and molecules, we compare the log2 counts and the log counts per million.</p>
<p>For the log counts, I add a pseudocount of 1.</p>
<pre class="r"><code>reads_log &lt;- log2(reads + 1)
molecules_log &lt;- log2(molecules + 1)</code></pre>
<p>standardized by cmp and log-transform</p>
<pre class="r"><code>reads_cpm &lt;- cpm(reads, log = FALSE)
molecules_cpm &lt;- cpm(molecules, log = FALSE)</code></pre>
<p>Calculate cpm for the reads data using TMM-normalization.</p>
<pre class="r"><code>norm_factors_reads &lt;- calcNormFactors(reads, method = &quot;TMM&quot;)
reads_tmm &lt;- cpm(reads, lib.size = colSums(reads) * norm_factors_reads,
                 log = TRUE)</code></pre>
<p>And for the molecules.</p>
<pre class="r"><code>norm_factors_mol &lt;- calcNormFactors(molecules, method = &quot;TMM&quot;)
molecules_tmm &lt;- cpm(molecules, lib.size = colSums(molecules) * norm_factors_mol,
                     log = TRUE)</code></pre>
</div>
<div id="conversion-in-each-single-cell" class="section level2">
<h2>conversion in each single cell</h2>
<div id="counts" class="section level3">
<h3>Counts</h3>
<p>Compare the counts. (1) ERCC genes. (2) without ERCC, only endogenous genes</p>
<div id="ercc-genes" class="section level4">
<h4>ERCC genes</h4>
<pre class="r"><code>## grep ERCC
reads_ERCC &lt;- reads[grep(&quot;ERCC&quot;, rownames(reads)), ]
molecules_ERCC &lt;- molecules[grep(&quot;ERCC&quot;, rownames(molecules)), ]

## linear regression per cell and make a table with intercept, slope, and r-squared
regression_table_ERCC &lt;- as.data.frame(do.call(rbind,lapply(names(reads_ERCC),function(x){
  fit.temp &lt;- lm(molecules_ERCC[,x]~reads_ERCC[,x])
    c(x,fit.temp$coefficients,summary(fit.temp)$adj.r.squared)
})))
names(regression_table_ERCC) &lt;- c(&quot;sample_id&quot;,&quot;Intercept&quot;,&quot;slope&quot;,&quot;r2&quot;)
regression_table_ERCC$Intercept &lt;- as.numeric(as.character(regression_table_ERCC$Intercept))
regression_table_ERCC$slope &lt;- as.numeric(as.character(regression_table_ERCC$slope))
regression_table_ERCC$r2 &lt;- as.numeric(as.character(regression_table_ERCC$r2))
plot(regression_table_ERCC$r2)</code></pre>
<p><img src="figure/compare-reads-v-molecules-in-cells.Rmd/conversion-in-cell-ERCC-1.png" title="" alt="" width="864" style="display: block; margin: auto;" /></p>
<pre class="r"><code>anno_regression_ERCC &lt;- merge(anno,regression_table_ERCC,by=&quot;sample_id&quot;)
ggplot(anno_regression_ERCC,aes(x=Intercept,y=slope,col=as.factor(individual),shape=as.factor(batch))) + geom_point() + labs(x = &quot;intercept&quot;, y = &quot;slope&quot;, title = &quot;read-molecule conversion ERCC genes (counts)&quot;) + facet_grid(individual ~ batch)</code></pre>
<p><img src="figure/compare-reads-v-molecules-in-cells.Rmd/conversion-in-cell-ERCC-2.png" title="" alt="" width="864" style="display: block; margin: auto;" /></p>
<pre class="r"><code>## plot all the lines
anno_regression_ERCC$reads_mean &lt;- apply(reads_ERCC, 2, mean)
anno_regression_ERCC$molecules_mean &lt;- apply(molecules_ERCC, 2, mean)

ggplot(anno_regression_ERCC, aes(x= reads_mean, y= molecules_mean)) + geom_point() + geom_abline(aes(intercept=Intercept, slope=slope, col=as.factor(individual), alpha= 0.5), data=anno_regression_ERCC) + facet_grid(individual ~ batch)</code></pre>
<p><img src="figure/compare-reads-v-molecules-in-cells.Rmd/conversion-in-cell-ERCC-3.png" title="" alt="" width="864" style="display: block; margin: auto;" /></p>
<p>Pairwise distance in (intercept, slope) between batches or individuals.</p>
<ul>
<li>Compute pairwise Euclidean distance between cells.</li>
</ul>
<pre class="r"><code>anno_regression_ERCC_dist &lt;- as.matrix( dist(anno_regression_ERCC[ , c(&quot;Intercept&quot;, &quot;slope&quot;)]) )
rownames(anno_regression_ERCC_dist) &lt;- with(anno_regression_ERCC,
                                            paste(individual, batch, sep = &quot;_&quot;))
colnames(anno_regression_ERCC_dist) &lt;- rownames(anno_regression_ERCC_dist)
anno_regression_ERCC_dist[1:2, 1:2]</code></pre>
<pre><code>          19098_1   19098_1
19098_1 0.0000000 0.6214327
19098_1 0.6214327 0.0000000</code></pre>
<ul>
<li>Cell-to-cell Euclidean distance within each individiaual, batch</li>
</ul>
<pre class="r"><code>## All cells
dist_mean &lt;- data.frame(dist = rowSums(anno_regression_ERCC_dist)/(ncol(anno_regression_ERCC_dist) - 1))
ggplot(dist_mean, aes(y = dist, x = 1)) + geom_violin(alpha = .5) + 
  geom_boxplot(alpha = .01, width = .2, position = position_dodge(width = .9)) +
  labs(title = &quot;Mean cell-to-cell distance in (intercept, slope)&quot;, x = &quot;&quot;) +
  theme(axis.text.x = element_text(hjust=1, angle = 45))</code></pre>
<p><img src="figure/compare-reads-v-molecules-in-cells.Rmd/unnamed-chunk-2-1.png" title="" alt="" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>## Within batch 
batch_unique &lt;- with(anno_regression_ERCC, paste(individual, batch, sep = &quot;_&quot;))
dist_vec &lt;- lapply(1:length(unique(batch_unique)), function(per_batch) {
  dist_foo &lt;- anno_regression_ERCC_dist[batch_unique == unique(batch_unique)[per_batch],
                             batch_unique == unique(batch_unique)[per_batch] ]
  data.frame(dist = c(rowSums(dist_foo)/(ncol(dist_foo) - 1)),
             batch_unique = rep(unique(batch_unique)[per_batch], ncol(dist_foo) )  )
})
dist_vec &lt;- do.call(rbind, dist_vec)
str(dist_vec)</code></pre>
<pre><code>&#39;data.frame&#39;:   578 obs. of  2 variables:
 $ dist        : num  0.442 0.606 0.608 0.436 0.441 ...
 $ batch_unique: Factor w/ 9 levels &quot;19098_1&quot;,&quot;19098_2&quot;,..: 1 1 1 1 1 1 1 1 1 1 ...</code></pre>
<pre class="r"><code>ggplot(dist_vec, aes(x= factor(batch_unique), y = dist, fill = factor(batch_unique)),
       height = 600, width = 2000) +
  geom_violin(alpha = .5) + geom_boxplot(alpha = .01, width = .2, position = position_dodge(width = .9)) +
  labs(title = &quot;Mean cell-to-cell distance within batch \n
                in (intercept, slope)&quot;) +
  theme(axis.text.x = element_text(hjust=1, angle = 45))</code></pre>
<p><img src="figure/compare-reads-v-molecules-in-cells.Rmd/unnamed-chunk-2-2.png" title="" alt="" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>## Within individual
individual &lt;- anno_regression_ERCC$individual
dist_vec &lt;- lapply(1:length(unique(individual)), function(per_individual) {
  dist_foo &lt;- anno_regression_ERCC_dist[individual == unique(individual)[per_individual],
                             individual == unique(individual)[per_individual] ]
  data.frame(dist = c(rowSums(dist_foo)/(ncol(dist_foo) - 1)),
             individual = rep(unique(individual)[per_individual], ncol(dist_foo) )  ) 
})
dist_vec &lt;- do.call(rbind, dist_vec)
str(dist_vec)</code></pre>
<pre><code>&#39;data.frame&#39;:   578 obs. of  2 variables:
 $ dist      : num  0.514 0.729 0.732 0.515 0.535 ...
 $ individual: int  19098 19098 19098 19098 19098 19098 19098 19098 19098 19098 ...</code></pre>
<pre class="r"><code>ggplot(dist_vec, aes(x= factor(individual), y = dist, fill = factor(individual)),
       height = 600, width = 2000) +
  geom_violin(alpha = .5) + geom_boxplot(alpha = .01, width = .2, position = position_dodge(width = .9)) +
  labs(title = &quot;Mean cell-to-cell distance within individual \n
                in (intercept, slope)&quot;) +
  theme(axis.text.x = element_text(hjust=1, angle = 45))</code></pre>
<p><img src="figure/compare-reads-v-molecules-in-cells.Rmd/unnamed-chunk-2-3.png" title="" alt="" width="672" style="display: block; margin: auto;" /></p>
<ul>
<li>compare the distance of two cells (1) within batch, (2) between batch of same individual, and (3) between individual</li>
</ul>
<pre class="r"><code>ind_index &lt;- (anno_regression_ERCC$individual)
ind_batch_index &lt;- with(anno_regression_ERCC,paste(individual, batch, sep = &quot;_&quot;))

same_ind_index &lt;- outer(ind_index,ind_index,function(x,y) x==y)
same_batch_index &lt;- outer(ind_batch_index,ind_batch_index,function(x,y) x==y)

dim_temp &lt;- dim(anno_regression_ERCC_dist)
dist_index_matrix &lt;- matrix(&quot;diff_ind&quot;,nrow=dim_temp[1],ncol=dim_temp[2])

dist_index_matrix[same_ind_index &amp; !same_batch_index] &lt;- &quot;same_ind_diff_batch&quot;
dist_index_matrix[same_batch_index] &lt;- &quot;same_batch&quot;

ans_ERCC &lt;- lapply(unique(c(dist_index_matrix)),function(x){
  temp &lt;- c(anno_regression_ERCC_dist[(dist_index_matrix==x)&amp;(upper.tri(dist_index_matrix,diag=FALSE))])
  data.frame(dist=temp,type=rep(x,length(temp)))
})
ans1_ERCC &lt;- do.call(rbind,ans_ERCC)

boxplot(dist~type,data=ans1_ERCC)</code></pre>
<p><img src="figure/compare-reads-v-molecules-in-cells.Rmd/three-groups-ercc-1.png" title="" alt="" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>plot(density(ans1_ERCC$dist[ans1_ERCC$type==&quot;same_batch&quot;]))
lines(density(ans1_ERCC$dist[ans1_ERCC$type==&quot;same_ind_diff_batch&quot;]),col=2)
lines(density(ans1_ERCC$dist[ans1_ERCC$type==&quot;diff_ind&quot;]),col=3)</code></pre>
<p><img src="figure/compare-reads-v-molecules-in-cells.Rmd/three-groups-ercc-2.png" title="" alt="" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>ggplot(ans1_ERCC, aes(x= factor(type), y = dist, col = factor(type)), height = 600, width = 2000) +
geom_boxplot(outlier.shape = NA, alpha = .01, width = .2, position = position_dodge(width = .9)) +
ylim(0,1.5) +
labs(title = &quot;cell-cell distance (slope and intercept of lm) ERCC&quot;) +
theme(axis.text.x = element_text(hjust=1, angle = 45))</code></pre>
<pre><code>Warning: Removed 5336 rows containing non-finite values (stat_boxplot).</code></pre>
<pre><code>Warning: Removed 254 rows containing missing values (geom_point).</code></pre>
<pre><code>Warning: Removed 612 rows containing missing values (geom_point).</code></pre>
<pre><code>Warning: Removed 1150 rows containing missing values (geom_point).</code></pre>
<p><img src="figure/compare-reads-v-molecules-in-cells.Rmd/three-groups-ercc-3.png" title="" alt="" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>summary(lm(dist~type,data=ans1_ERCC))</code></pre>
<pre><code>
Call:
lm(formula = dist ~ type, data = ans1_ERCC)

Residuals:
    Min      1Q  Median      3Q     Max 
-0.5171 -0.3175 -0.1125  0.1798  5.2540 

Coefficients:
                        Estimate Std. Error t value Pr(&gt;|t|)    
(Intercept)             0.458805   0.003349 136.990  &lt; 2e-16 ***
typesame_ind_diff_batch 0.023554   0.004172   5.646 1.65e-08 ***
typediff_ind            0.058664   0.003642  16.109  &lt; 2e-16 ***
---
Signif. codes:  0 &#39;***&#39; 0.001 &#39;**&#39; 0.01 &#39;*&#39; 0.05 &#39;.&#39; 0.1 &#39; &#39; 1

Residual standard error: 0.4748 on 166750 degrees of freedom
Multiple R-squared:  0.002066,  Adjusted R-squared:  0.002055 
F-statistic: 172.7 on 2 and 166750 DF,  p-value: &lt; 2.2e-16</code></pre>
</div>
<div id="endogenous-genes" class="section level4">
<h4>Endogenous genes</h4>
<pre class="r"><code>## grep ENSG
reads_ENSG &lt;- reads[grep(&quot;ENSG&quot;, rownames(reads)), ]
molecules_ENSG &lt;- molecules[grep(&quot;ENSG&quot;, rownames(molecules)), ]

## linear regression per cell and make a table with intercept, slope, and r-squared
regression_table_ENSG &lt;- as.data.frame(do.call(rbind,lapply(names(reads_ENSG),function(x){
  fit.temp &lt;- lm(molecules_ENSG[,x]~reads_ENSG[,x])
    c(x,fit.temp$coefficients,summary(fit.temp)$adj.r.squared)
})))
names(regression_table_ENSG) &lt;- c(&quot;sample_id&quot;,&quot;Intercept&quot;,&quot;slope&quot;,&quot;r2&quot;)
regression_table_ENSG$Intercept &lt;- as.numeric(as.character(regression_table_ENSG$Intercept))
regression_table_ENSG$slope &lt;- as.numeric(as.character(regression_table_ENSG$slope))
regression_table_ENSG$r2 &lt;- as.numeric(as.character(regression_table_ENSG$r2))
plot(regression_table_ENSG$r2)</code></pre>
<p><img src="figure/compare-reads-v-molecules-in-cells.Rmd/conversion-in-cell-gene-1.png" title="" alt="" width="864" style="display: block; margin: auto;" /></p>
<pre class="r"><code>anno_regression_ENSG &lt;- merge(anno,regression_table_ENSG,by=&quot;sample_id&quot;)

ggplot(anno_regression_ENSG,aes(x=Intercept,y=slope,col=as.factor(individual),shape=as.factor(batch))) + geom_point() + labs(x = &quot;intercept&quot;, y = &quot;slope&quot;, title = &quot;lm of read-molecule conversion endogenous genes in each cell (counts)&quot;)</code></pre>
<p><img src="figure/compare-reads-v-molecules-in-cells.Rmd/conversion-in-cell-gene-2.png" title="" alt="" width="864" style="display: block; margin: auto;" /></p>
<pre class="r"><code>ggplot(anno_regression_ENSG,aes(x=Intercept,y=slope,col=as.factor(individual),shape=as.factor(batch))) + geom_point() + labs(x = &quot;intercept&quot;, y = &quot;slope&quot;, title = &quot;lm of read-molecule conversion endogenous genes in each cell (counts)&quot;) + facet_grid(individual ~ batch)</code></pre>
<p><img src="figure/compare-reads-v-molecules-in-cells.Rmd/conversion-in-cell-gene-3.png" title="" alt="" width="864" style="display: block; margin: auto;" /></p>
<pre class="r"><code>## plot all the lines
anno_regression_ENSG$reads_mean &lt;- apply(reads_ENSG, 2, mean)
anno_regression_ENSG$molecules_mean &lt;- apply(molecules_ENSG, 2, mean)

ggplot(anno_regression_ENSG, aes(x= reads_mean, y= molecules_mean)) + geom_point() + geom_abline(aes(intercept=Intercept, slope=slope, col=as.factor(individual), alpha= 0.5), data=anno_regression_ENSG)</code></pre>
<p><img src="figure/compare-reads-v-molecules-in-cells.Rmd/conversion-in-cell-gene-4.png" title="" alt="" width="864" style="display: block; margin: auto;" /></p>
<pre class="r"><code>ggplot(anno_regression_ENSG, aes(x= reads_mean, y= molecules_mean)) + geom_point() + geom_abline(aes(intercept=Intercept, slope=slope, col=as.factor(individual), alpha= 0.5), data=anno_regression_ENSG) + facet_grid(individual ~ batch)</code></pre>
<p><img src="figure/compare-reads-v-molecules-in-cells.Rmd/conversion-in-cell-gene-5.png" title="" alt="" width="864" style="display: block; margin: auto;" /></p>
<p>There is a clear difference between individuals. How about difference between batches? Use 9 different colors to visualize <a href="compare-reads-v-molecules.html#effect-of-sequencing-depth-on-molecule-count">the 9 batches</a></p>
<pre class="r"><code>ggplot(anno_regression_ENSG, aes(x = Intercept, y = slope, col = paste(individual, batch, sep = &quot;.&quot;))) +
  geom_point() +
  scale_color_brewer(palette = &quot;Set1&quot;, name = &quot;9 batches&quot;) +
  labs(title = &quot;lm of read-molecule conversion endogenous genes in each cell (counts)&quot;)</code></pre>
<p><img src="figure/compare-reads-v-molecules-in-cells.Rmd/mol-v-reads-single-9-batches-1.png" title="" alt="" width="768" style="display: block; margin: auto;" /></p>
<p>Here are the batches split by individual.</p>
<pre class="r"><code>ggplot(anno_regression_ENSG, aes(x = Intercept, y = slope)) +
geom_point(aes(color = as.factor(batch))) +
facet_wrap(~individual, nrow = 3) +
scale_color_brewer(palette = &quot;Dark2&quot;, name = &quot;batch&quot;) +
labs(title = &quot;lm of read-molecule conversion endogenous genes in each cell (counts)&quot;)</code></pre>
<p><img src="figure/compare-reads-v-molecules-in-cells.Rmd/mol-v-reads-single-per-ind-1.png" title="" alt="" width="768" style="display: block; margin: auto;" /></p>
<p>Pairwise distance in (intercept, slope) between batches or individuals.</p>
<ul>
<li>Compute pairwise Euclidean distance between cells.</li>
</ul>
<pre class="r"><code>anno_regression_ENSG_dist &lt;- as.matrix( dist(anno_regression_ENSG[ , c(&quot;Intercept&quot;, &quot;slope&quot;)]) )
rownames(anno_regression_ENSG_dist) &lt;- with(anno_regression_ENSG,
                                            paste(individual, batch, sep = &quot;_&quot;))
colnames(anno_regression_ENSG_dist) &lt;- rownames(anno_regression_ENSG_dist)
anno_regression_ENSG_dist[1:2, 1:2]</code></pre>
<pre><code>          19098_1   19098_1
19098_1 0.0000000 0.4931873
19098_1 0.4931873 0.0000000</code></pre>
<ul>
<li>Cell-to-cell Euclidean distance within each individiaual, batch</li>
</ul>
<pre class="r"><code>## All cells
dist_mean &lt;- data.frame(dist = rowSums(anno_regression_ENSG_dist)/(ncol(anno_regression_ENSG_dist) - 1))
ggplot(dist_mean, aes(y = dist, x = 1)) + geom_violin(alpha = .5) + 
  geom_boxplot(alpha = .01, width = .2, position = position_dodge(width = .9)) +
  labs(title = &quot;Mean cell-to-cell distance in (intercept, slope)&quot;, x = &quot;&quot;) +
  theme(axis.text.x = element_text(hjust=1, angle = 45))</code></pre>
<p><img src="figure/compare-reads-v-molecules-in-cells.Rmd/unnamed-chunk-4-1.png" title="" alt="" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>## Within batch 
batch_unique &lt;- with(anno_regression_ENSG, paste(individual, batch, sep = &quot;_&quot;))
dist_vec &lt;- lapply(1:length(unique(batch_unique)), function(per_batch) {
  dist_foo &lt;- anno_regression_ENSG_dist[batch_unique == unique(batch_unique)[per_batch],
                             batch_unique == unique(batch_unique)[per_batch] ]
  data.frame(dist = c(rowSums(dist_foo)/(ncol(dist_foo) - 1)),
             batch_unique = rep(unique(batch_unique)[per_batch], ncol(dist_foo) )  )
})
dist_vec &lt;- do.call(rbind, dist_vec)
str(dist_vec)</code></pre>
<pre><code>&#39;data.frame&#39;:   578 obs. of  2 variables:
 $ dist        : num  0.404 0.344 0.363 0.332 0.629 ...
 $ batch_unique: Factor w/ 9 levels &quot;19098_1&quot;,&quot;19098_2&quot;,..: 1 1 1 1 1 1 1 1 1 1 ...</code></pre>
<pre class="r"><code>ggplot(dist_vec, aes(x= factor(batch_unique), y = dist, fill = factor(batch_unique)),
       height = 600, width = 2000) +
  geom_violin(alpha = .5) + geom_boxplot(alpha = .01, width = .2, position = position_dodge(width = .9)) +
  labs(title = &quot;Mean cell-to-cell distance within batch \n
                in (intercept, slope)&quot;) +
  theme(axis.text.x = element_text(hjust=1, angle = 45))</code></pre>
<p><img src="figure/compare-reads-v-molecules-in-cells.Rmd/unnamed-chunk-4-2.png" title="" alt="" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>## Within individual
individual &lt;- anno_regression_ENSG$individual
dist_vec &lt;- lapply(1:length(unique(individual)), function(per_individual) {
  dist_foo &lt;- anno_regression_ENSG_dist[individual == unique(individual)[per_individual],
                             individual == unique(individual)[per_individual] ]
  data.frame(dist = c(rowSums(dist_foo)/(ncol(dist_foo) - 1)),
             individual = rep(unique(individual)[per_individual], ncol(dist_foo) )  )
})
dist_vec &lt;- do.call(rbind, dist_vec)
str(dist_vec)</code></pre>
<pre><code>&#39;data.frame&#39;:   578 obs. of  2 variables:
 $ dist      : num  0.457 0.429 0.446 0.404 0.699 ...
 $ individual: int  19098 19098 19098 19098 19098 19098 19098 19098 19098 19098 ...</code></pre>
<pre class="r"><code>ggplot(dist_vec, aes(x= factor(individual), y = dist, fill = factor(individual)),
       height = 600, width = 2000) +
  geom_violin(alpha = .5) + geom_boxplot(alpha = .01, width = .2, position = position_dodge(width = .9)) +
  labs(title = &quot;Mean cell-to-cell distance within individual \n
                in (intercept, slope)&quot;) +
  theme(axis.text.x = element_text(hjust=1, angle = 45))</code></pre>
<p><img src="figure/compare-reads-v-molecules-in-cells.Rmd/unnamed-chunk-4-3.png" title="" alt="" width="672" style="display: block; margin: auto;" /></p>
<ul>
<li>compare the distance of two cells (1) within batch, (2) between batch of same individual, and (3) between individual</li>
</ul>
<pre class="r"><code>ind_index &lt;- (anno_regression_ENSG$individual)
ind_batch_index &lt;- with(anno_regression_ENSG,paste(individual, batch, sep = &quot;_&quot;))

same_ind_index &lt;- outer(ind_index,ind_index,function(x,y) x==y)
same_batch_index &lt;- outer(ind_batch_index,ind_batch_index,function(x,y) x==y)

dim_temp &lt;- dim(anno_regression_ENSG_dist)
dist_index_matrix &lt;- matrix(&quot;diff_ind&quot;,nrow=dim_temp[1],ncol=dim_temp[2])

dist_index_matrix[same_ind_index &amp; !same_batch_index] &lt;- &quot;same_ind_diff_batch&quot;
dist_index_matrix[same_batch_index] &lt;- &quot;same_batch&quot;

ans &lt;- lapply(unique(c(dist_index_matrix)),function(x){
  temp &lt;- c(anno_regression_ENSG_dist[(dist_index_matrix==x)&amp;(upper.tri(dist_index_matrix,diag=FALSE))])
  data.frame(dist=temp,type=rep(x,length(temp)))
})
ans1 &lt;- do.call(rbind,ans)

boxplot(dist~type,data=ans1)</code></pre>
<p><img src="figure/compare-reads-v-molecules-in-cells.Rmd/three-groups-1.png" title="" alt="" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>plot(density(ans1$dist[ans1$type==&quot;same_batch&quot;]))
lines(density(ans1$dist[ans1$type==&quot;same_ind_diff_batch&quot;]),col=2)
lines(density(ans1$dist[ans1$type==&quot;diff_ind&quot;]),col=3)</code></pre>
<p><img src="figure/compare-reads-v-molecules-in-cells.Rmd/three-groups-2.png" title="" alt="" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>ggplot(ans1, aes(x= factor(type), y = dist, col = factor(type)), height = 600, width = 2000) +
geom_boxplot(outlier.shape = NA, alpha = .01, width = .2, position = position_dodge(width = .9)) +
ylim(0,1.5) +
labs(title = &quot;cell-cell distance (slope and intercept of lm)&quot;) +
theme(axis.text.x = element_text(hjust=1, angle = 45))</code></pre>
<pre><code>Warning: Removed 6712 rows containing non-finite values (stat_boxplot).</code></pre>
<pre><code>Warning: Removed 300 rows containing missing values (geom_point).</code></pre>
<pre><code>Warning: Removed 413 rows containing missing values (geom_point).</code></pre>
<pre><code>Warning: Removed 660 rows containing missing values (geom_point).</code></pre>
<p><img src="figure/compare-reads-v-molecules-in-cells.Rmd/three-groups-3.png" title="" alt="" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>summary(lm(dist~type,data=ans1))</code></pre>
<pre><code>
Call:
lm(formula = dist ~ type, data = ans1)

Residuals:
    Min      1Q  Median      3Q     Max 
-0.5457 -0.3339 -0.1172  0.1769  3.5505 

Coefficients:
                        Estimate Std. Error t value Pr(&gt;|t|)    
(Intercept)             0.468258   0.003516  133.18   &lt;2e-16 ***
typesame_ind_diff_batch 0.036000   0.004380    8.22   &lt;2e-16 ***
typediff_ind            0.077562   0.003823   20.29   &lt;2e-16 ***
---
Signif. codes:  0 &#39;***&#39; 0.001 &#39;**&#39; 0.01 &#39;*&#39; 0.05 &#39;.&#39; 0.1 &#39; &#39; 1

Residual standard error: 0.4984 on 166750 degrees of freedom
Multiple R-squared:  0.003061,  Adjusted R-squared:  0.003049 
F-statistic:   256 on 2 and 166750 DF,  p-value: &lt; 2.2e-16</code></pre>
<p>19098 have more total endogenous gene molecules and more ERCC molecule</p>
<pre class="r"><code>anno_regression_ENSG$total_molecule_ERCC &lt;- apply(molecules_ERCC, 2, sum)
anno_regression_ENSG$total_molecule_ENSG &lt;- apply(molecules_ENSG, 2, sum)
anno_regression_ENSG$total_molecule &lt;- apply(molecules, 2, sum) 
ggplot(anno_regression_ENSG, aes(x= total_molecule_ERCC, y= total_molecule,col=as.factor(individual),shape=as.factor(batch))) + geom_point()</code></pre>
<p><img src="figure/compare-reads-v-molecules-in-cells.Rmd/total-molecule-1.png" title="" alt="" width="864" style="display: block; margin: auto;" /></p>
<pre class="r"><code>ggplot(anno_regression_ENSG, aes(x = as.factor(individual), y = total_molecule_ENSG)) + geom_boxplot(aes(fill = as.factor(batch)))</code></pre>
<p><img src="figure/compare-reads-v-molecules-in-cells.Rmd/total-molecule-2.png" title="" alt="" width="864" style="display: block; margin: auto;" /></p>
<pre class="r"><code>ggplot(anno_regression_ENSG, aes(x = as.factor(individual), y = total_molecule_ERCC)) + geom_boxplot(aes(fill = as.factor(batch)))</code></pre>
<p><img src="figure/compare-reads-v-molecules-in-cells.Rmd/total-molecule-3.png" title="" alt="" width="864" style="display: block; margin: auto;" /></p>
</div>
</div>
<div id="log2-counts" class="section level3">
<h3>Log2 counts</h3>
<p>Compare the log counts. The r squared values of lm are relativiely low (around 0.85).</p>
<pre class="r"><code>regression_table_log &lt;- as.data.frame(do.call(rbind,lapply(names(reads_log),function(x){
  fit.temp &lt;- lm(molecules_log[,x]~reads_log[,x])
    c(x,fit.temp$coefficients,summary(fit.temp)$adj.r.squared)
})))
names(regression_table_log) &lt;- c(&quot;sample_id&quot;,&quot;Intercept&quot;,&quot;slope&quot;,&quot;r2&quot;)
regression_table_log$Intercept &lt;- as.numeric(as.character(regression_table_log$Intercept))
regression_table_log$slope &lt;- as.numeric(as.character(regression_table_log$slope))
regression_table_log$r2 &lt;- as.numeric(as.character(regression_table_log$r2))
plot(regression_table_log$r2)</code></pre>
<p><img src="figure/compare-reads-v-molecules-in-cells.Rmd/conversion-in-cell-log-1.png" title="" alt="" width="864" style="display: block; margin: auto;" /></p>
<pre class="r"><code>anno_regression_log &lt;- merge(anno,regression_table_log,by=&quot;sample_id&quot;)
ggplot(anno_regression_log,aes(x=Intercept,y=slope,col=as.factor(individual),shape=as.factor(batch))) + geom_point() + labs(x = &quot;intercept&quot;, y = &quot;slope&quot;, title = &quot;read-molecule conversion (log2)&quot;)</code></pre>
<p><img src="figure/compare-reads-v-molecules-in-cells.Rmd/conversion-in-cell-log-2.png" title="" alt="" width="864" style="display: block; margin: auto;" /></p>
</div>
<div id="counts-per-million" class="section level3">
<h3>counts per million</h3>
<p>Compare the counts per million.</p>
<pre class="r"><code>regression_table_cpm &lt;- as.data.frame(do.call(rbind,lapply(colnames(reads_cpm),function(x){
  fit.temp &lt;- lm(molecules_cpm[,x]~reads_cpm[,x])
    c(x,fit.temp$coefficients,summary(fit.temp)$adj.r.squared)
})))
names(regression_table_cpm) &lt;- c(&quot;sample_id&quot;,&quot;Intercept&quot;,&quot;slope&quot;,&quot;r2&quot;)
regression_table_cpm$Intercept &lt;- as.numeric(as.character(regression_table_cpm$Intercept))
regression_table_cpm$slope &lt;- as.numeric(as.character(regression_table_cpm$slope))
regression_table_cpm$r2 &lt;- as.numeric(as.character(regression_table_cpm$r2))
plot(regression_table_cpm$r2)</code></pre>
<p><img src="figure/compare-reads-v-molecules-in-cells.Rmd/conversion-in-cell-cpm-1.png" title="" alt="" width="864" style="display: block; margin: auto;" /></p>
<pre class="r"><code>anno_regression_cpm &lt;- merge(anno,regression_table_cpm,by=&quot;sample_id&quot;)

ggplot(anno_regression_cpm,aes(x=Intercept,y=slope,col=as.factor(individual),shape=as.factor(batch))) + geom_point() + labs(x = &quot;intercept&quot;, y = &quot;slope&quot;, title = &quot;read-molecule conversion (cpm no log)&quot;)</code></pre>
<p><img src="figure/compare-reads-v-molecules-in-cells.Rmd/conversion-in-cell-cpm-2.png" title="" alt="" width="864" style="display: block; margin: auto;" /></p>
<pre class="r"><code>ggplot(anno_regression_cpm,aes(x=Intercept,y=slope,col=as.factor(individual),shape=as.factor(batch))) + geom_point() + labs(x = &quot;intercept&quot;, y = &quot;slope&quot;, title = &quot;read-molecule conversion (cpm no log)&quot;) + facet_grid(individual ~ batch) </code></pre>
<p><img src="figure/compare-reads-v-molecules-in-cells.Rmd/conversion-in-cell-cpm-3.png" title="" alt="" width="864" style="display: block; margin: auto;" /></p>
<pre class="r"><code>## plot all the lines
anno_regression_cpm$reads_mean &lt;- apply(reads_cpm, 2, mean)
anno_regression_cpm$molecules_mean &lt;- apply(molecules_cpm, 2, mean)

ggplot(anno_regression_cpm, aes(x= reads_mean, y= molecules_mean)) + geom_point() + geom_abline(aes(intercept=Intercept, slope=slope, col=as.factor(individual), alpha = 0.01), data=anno_regression_cpm) + facet_grid(individual ~ batch) </code></pre>
<p><img src="figure/compare-reads-v-molecules-in-cells.Rmd/conversion-in-cell-cpm-4.png" title="" alt="" width="864" style="display: block; margin: auto;" /></p>
<pre class="r"><code>## look at just the slope using total raw molecule counts (total cpm molecule counts are all the same across cells)
anno_regression_cpm$total_molecule &lt;- apply(molecules, 2, sum)
ggplot(anno_regression_cpm, aes(x= total_molecule, y= slope,col=as.factor(individual),shape=as.factor(batch))) + geom_point()</code></pre>
<p><img src="figure/compare-reads-v-molecules-in-cells.Rmd/conversion-in-cell-cpm-5.png" title="" alt="" width="864" style="display: block; margin: auto;" /></p>
<p>Only ERCC genes</p>
<pre class="r"><code>## grep ERCC
reads_cpm_ERCC &lt;- as.data.frame(reads_cpm[grep(&quot;ERCC&quot;, rownames(reads_cpm)), ])
molecules_cpm_ERCC &lt;- as.data.frame(molecules_cpm[grep(&quot;ERCC&quot;, rownames(molecules_cpm)), ])

## linear regression per cell and make a table with intercept, slope, and r-squared
regression_table_cpm_ERCC &lt;- as.data.frame(do.call(rbind,lapply(names(reads_cpm_ERCC),function(x){
  fit.temp &lt;- lm(molecules_cpm_ERCC[,x]~reads_cpm_ERCC[,x])
    c(x,fit.temp$coefficients,summary(fit.temp)$adj.r.squared)
})))
names(regression_table_cpm_ERCC) &lt;- c(&quot;sample_id&quot;,&quot;Intercept&quot;,&quot;slope&quot;,&quot;r2&quot;)
regression_table_cpm_ERCC$Intercept &lt;- as.numeric(as.character(regression_table_cpm_ERCC$Intercept))
regression_table_cpm_ERCC$slope &lt;- as.numeric(as.character(regression_table_cpm_ERCC$slope))
regression_table_cpm_ERCC$r2 &lt;- as.numeric(as.character(regression_table_cpm_ERCC$r2))
plot(regression_table_cpm_ERCC$r2)</code></pre>
<p><img src="figure/compare-reads-v-molecules-in-cells.Rmd/conversion-in-cell-cpm-ERCC-1.png" title="" alt="" width="864" style="display: block; margin: auto;" /></p>
<pre class="r"><code>anno_regression_cpm_ERCC &lt;- merge(anno,regression_table_cpm_ERCC,by=&quot;sample_id&quot;)

ggplot(anno_regression_cpm_ERCC,aes(x=Intercept,y=slope,col=as.factor(individual),shape=as.factor(batch))) + geom_point() + labs(x = &quot;intercept&quot;, y = &quot;slope&quot;, title = &quot;read-molecule conversion ERCC genes (cpm no log)&quot;) </code></pre>
<p><img src="figure/compare-reads-v-molecules-in-cells.Rmd/conversion-in-cell-cpm-ERCC-2.png" title="" alt="" width="864" style="display: block; margin: auto;" /></p>
<pre class="r"><code>## plot all the lines
anno_regression_cpm_ERCC$reads_mean &lt;- apply(reads_cpm_ERCC, 2, mean)
anno_regression_cpm_ERCC$molecules_mean &lt;- apply(molecules_cpm_ERCC, 2, mean)

ggplot(anno_regression_cpm_ERCC, aes(x= reads_mean, y= molecules_mean)) + geom_point() + geom_abline(aes(intercept=Intercept, slope=slope, col=as.factor(individual), alpha= 0.5), data=anno_regression_cpm_ERCC) + facet_grid(individual ~ batch)</code></pre>
<p><img src="figure/compare-reads-v-molecules-in-cells.Rmd/conversion-in-cell-cpm-ERCC-3.png" title="" alt="" width="864" style="display: block; margin: auto;" /></p>
</div>
<div id="tmm-normalized-counts-per-million" class="section level3">
<h3>TMM-normalized counts per million</h3>
<p>Compare TMM-normalized log2-transformed cpm .</p>
<pre class="r"><code>regression_table_tmm &lt;- as.data.frame(do.call(rbind,lapply(colnames(reads_tmm),function(x){
  fit.temp &lt;- lm(molecules_tmm[,x]~reads_tmm[,x])
    c(x,fit.temp$coefficients,summary(fit.temp)$adj.r.squared)
})))
names(regression_table_tmm) &lt;- c(&quot;sample_id&quot;,&quot;Intercept&quot;,&quot;slope&quot;,&quot;r2&quot;)
regression_table_tmm$Intercept &lt;- as.numeric(as.character(regression_table_tmm$Intercept))
regression_table_tmm$slope &lt;- as.numeric(as.character(regression_table_tmm$slope))
regression_table_tmm$r2 &lt;- as.numeric(as.character(regression_table_tmm$r2))
plot(regression_table_tmm$r2)</code></pre>
<p><img src="figure/compare-reads-v-molecules-in-cells.Rmd/conversion-in-cell-tmm-1.png" title="" alt="" width="864" style="display: block; margin: auto;" /></p>
<pre class="r"><code>anno_regression_tmm &lt;- merge(anno,regression_table_tmm,by=&quot;sample_id&quot;)

ggplot(anno_regression_tmm,aes(x=Intercept,y=slope,col=as.factor(individual),shape=as.factor(batch))) + geom_point() + labs(x = &quot;intercept&quot;, y = &quot;slope&quot;, title = &quot;read-molecule conversion (tmm cpm log)&quot;)</code></pre>
<p><img src="figure/compare-reads-v-molecules-in-cells.Rmd/conversion-in-cell-tmm-2.png" title="" alt="" width="864" style="display: block; margin: auto;" /></p>
<pre class="r"><code>ggplot(anno_regression_tmm,aes(x=Intercept,y=slope,col=as.factor(individual),shape=as.factor(batch))) + geom_point() + labs(x = &quot;intercept&quot;, y = &quot;slope&quot;, title = &quot;read-molecule conversion (tmm cpm log)&quot;) + facet_grid(individual ~ batch) </code></pre>
<p><img src="figure/compare-reads-v-molecules-in-cells.Rmd/conversion-in-cell-tmm-3.png" title="" alt="" width="864" style="display: block; margin: auto;" /></p>
<pre class="r"><code>## plot all the lines
anno_regression_tmm$reads_mean &lt;- apply(reads_tmm, 2, mean)
anno_regression_tmm$molecules_mean &lt;- apply(molecules_tmm, 2, mean)

ggplot(anno_regression_tmm, aes(x= reads_mean, y= molecules_mean)) + geom_point() + geom_abline(aes(intercept=Intercept, slope=slope, col=as.factor(individual), alpha = 0.01), data=anno_regression_tmm) + facet_grid(individual ~ batch) </code></pre>
<p><img src="figure/compare-reads-v-molecules-in-cells.Rmd/conversion-in-cell-tmm-4.png" title="" alt="" width="864" style="display: block; margin: auto;" /></p>
</div>
</div>
<div id="session-information" class="section level2">
<h2>Session information</h2>
<pre class="r"><code>sessionInfo()</code></pre>
<pre><code>R version 3.2.0 (2015-04-16)
Platform: x86_64-unknown-linux-gnu (64-bit)

locale:
 [1] LC_CTYPE=en_US.UTF-8       LC_NUMERIC=C              
 [3] LC_TIME=en_US.UTF-8        LC_COLLATE=en_US.UTF-8    
 [5] LC_MONETARY=en_US.UTF-8    LC_MESSAGES=en_US.UTF-8   
 [7] LC_PAPER=en_US.UTF-8       LC_NAME=C                 
 [9] LC_ADDRESS=C               LC_TELEPHONE=C            
[11] LC_MEASUREMENT=en_US.UTF-8 LC_IDENTIFICATION=C       

attached base packages:
[1] stats     graphics  grDevices utils     datasets  methods   base     

other attached packages:
[1] tidyr_0.2.0   edgeR_3.10.2  limma_3.24.9  ggplot2_1.0.1 dplyr_0.4.2  
[6] knitr_1.10.5 

loaded via a namespace (and not attached):
 [1] Rcpp_0.12.0        magrittr_1.5       MASS_7.3-40       
 [4] munsell_0.4.2      colorspace_1.2-6   R6_2.1.1          
 [7] stringr_1.0.0      plyr_1.8.3         tools_3.2.0       
[10] parallel_3.2.0     grid_3.2.0         gtable_0.1.2      
[13] DBI_0.3.1          htmltools_0.2.6    yaml_2.1.13       
[16] assertthat_0.1     digest_0.6.8       RColorBrewer_1.1-2
[19] reshape2_1.4.1     formatR_1.2        evaluate_0.7      
[22] rmarkdown_0.6.1    labeling_0.3       stringi_0.4-1     
[25] scales_0.2.4       proto_0.3-10      </code></pre>
</div>


<!-- some extra javascript for older browsers -->
<script type="text/javascript" src="libs/polyfill.js"></script>

<script>

// manage active state of menu based on current page
$(document).ready(function () {

    // active menu
    href = window.location.pathname
    href = href.substr(href.lastIndexOf('/') + 1)
    $('a[href="' + href + '"]').parent().addClass('active');

    // manage active menu header
    if (href.startsWith('authoring_'))
      $('a[href="' + 'authoring' + '"]').parent().addClass('active');
    else if (href.endsWith('_format.html'))
      $('a[href="' + 'formats' + '"]').parent().addClass('active');
    else if (href.startsWith('developer_'))
      $('a[href="' + 'developer' + '"]').parent().addClass('active');

});

</script>

</div>

<script>

// add bootstrap table styles to pandoc tables
$(document).ready(function () {
  $('tr.header').parent('thead').parent('table').addClass('table table-condensed');
});

</script>


</body>
</html>
