<!DOCTYPE html>

<html xmlns="http://www.w3.org/1999/xhtml">

<head>

<meta charset="utf-8">
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="generator" content="pandoc" />


<meta name="date" content="2015-07-02" />

<title>Subsample: Correlation of single and bulk cells</title>

<script src="libs/jquery-1.11.0/jquery.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="libs/bootstrap-3.3.1/css/united.min.css" rel="stylesheet" />
<script src="libs/bootstrap-3.3.1/js/bootstrap.min.js"></script>
<script src="libs/bootstrap-3.3.1/shim/html5shiv.min.js"></script>
<script src="libs/bootstrap-3.3.1/shim/respond.min.js"></script>

<style type="text/css">

/* padding for bootstrap navbar */
body {
  padding-top: 50px;
  padding-bottom: 40px;
}


/* offset scroll position for anchor links (for fixed navbar)  */
.section h2 {
  padding-top: 55px;
  margin-top: -55px;
}
.section h3 {
  padding-top: 55px;
  margin-top: -55px;
}



/* don't use link color in navbar */
.dropdown-menu>li>a {
  color: black;
}

/* some padding for disqus */
#disqus_thread {
  margin-top: 45px;
}

</style>

<link rel="stylesheet" href="libs/font-awesome-4.1.0/css/font-awesome.min.css"/>

<style type="text/css">code{white-space: pre;}</style>
<link rel="stylesheet"
      href="libs/highlight/textmate.css"
      type="text/css" />
<script src="libs/highlight/highlight.js"></script>
<style type="text/css">
  pre:not([class]) {
    background-color: white;
  }
</style>
<script type="text/javascript">
if (window.hljs && document.readyState && document.readyState === "complete") {
   window.setTimeout(function() {
      hljs.initHighlighting();
   }, 0);
}
</script>



</head>

<body>

<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
code {
  color: inherit;
  background-color: rgba(0, 0, 0, 0.04);
}
img { 
  max-width:100%; 
  height: auto; 
}
</style>
<div class="container-fluid main-container">


<div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">singleCellSeq</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li><a href="index.html">Home</a></li>
        <li><a href="about.html">About</a></li>
        <li><a href="license.html">License</a></li>
        <li><a href="https://github.com/jdblischak/singleCellSeq">GitHub</a></li>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<div id="header">
<h1 class="title">Subsample: Correlation of single and bulk cells</h1>
<h4 class="date"><em>2015-07-02</em></h4>
</div>

<div id="TOC">
<ul>
<li><a href="#batch-process-each-subsampled-data-set">Batch process each subsampled data set</a></li>
<li><a href="#results">Results</a></li>
<li><a href="#pollen-et-al.-2014">Pollen et al. 2014</a></li>
<li><a href="#results-by-expression-quantiles">Results by expression quantiles</a></li>
<li><a href="#session-information">Session information</a></li>
</ul>
</div>

<p><strong>Last updated:</strong> 2015-07-08</p>
<p><strong>Code version:</strong> e78dbba2bdcc461056a93867890f364a1425b46b</p>
<p>The mean counts per million in a subsampled set of single cells (both sequencing depth and number of cells is varied) is compared to the mean counts per million of the bulk cells (mean across the three batches). The mean counts are log<sub>2</sub> transformed with an added pseudocount of 1.</p>
<pre class="r"><code>library(&quot;dplyr&quot;)
library(&quot;ggplot2&quot;)
theme_set(theme_bw(base_size = 14))</code></pre>
<div id="batch-process-each-subsampled-data-set" class="section level2">
<h2>Batch process each subsampled data set</h2>
<p>Run 10 iterations for each individual for each sequencing depth for each subsample of cells. The analysis is performed by <a href="https://github.com/jdblischak/singleCellSeq/blob/master/code/correlate-single-to-bulk.R">correlate-single-to-bulk.R</a>.</p>
<pre class="bash"><code>cd $ssd/subsampled
mkdir -p correlation correlation-quantiles
mkdir -p ~/log/correlate-single-to-bulk.R
for IND in 19098 19101 19239
do
  for NUM in 200000 400000 600000 800000 1000000 1200000 1400000 1600000 1800000 2000000 2400000 2800000 3200000 3600000 4000000
  do
    for CELLS in 5 10 15 20 25 50 75 100 125 150
    do
      for SEED in {1..10}
      do
        # Correlation across all genes
        CMD=&quot;correlate-single-to-bulk.R $CELLS $SEED molecule-counts-$NUM.txt $ssc/data/reads.txt --individual=$IND --good_cells=/mnt/lustre/home/jdblischak/singleCellSeq/data/quality-single-cells.txt&quot;
        DEST=&quot;correlation/$IND-$CELLS-$SEED-$NUM.txt&quot;
        echo &quot;$CMD &gt; $DEST&quot; | qsub -l h_vmem=2g -cwd -V -N cor-$IND-$CELLS-$SEED-$NUM -j y -o ~/log/correlate-single-to-bulk.R -l &#39;hostname=!bigmem01&#39;
        sleep .01s
        # Correlation for genes separated by the specified quantiles
        CMD=&quot;correlate-single-to-bulk.R $CELLS $SEED molecule-counts-$NUM.txt $ssc/data/reads.txt --individual=$IND --good_cells=/mnt/lustre/home/jdblischak/singleCellSeq/data/quality-single-cells.txt -q .25 -q .5 -q .75&quot;
        DEST=&quot;correlation-quantiles/$IND-$CELLS-$SEED-$NUM.txt&quot;
        echo &quot;$CMD &gt; $DEST&quot; | qsub -l h_vmem=2g -cwd -V -N cor-$IND-$CELLS-$SEED-$NUM-quantiles -j y -o ~/log/correlate-single-to-bulk.R -l &#39;hostname=!bigmem01&#39;
        sleep .01s
      done
    done
  done
done</code></pre>
<p>Convert to one file using Python. Run from <code>$ssd/subsampled</code>.</p>
<pre class="python"><code>import os
import glob

def gather(files, outfile):
    out = open(outfile, &quot;w&quot;)
    out.write(&quot;ind\tdepth\tnum_cells\tseed\tquantiles\tr\tnum_genes\n&quot;)
    for fname in files:
        fname_parts = os.path.basename(fname).rstrip(&quot;.txt&quot;).split(&quot;-&quot;)
        ind = fname_parts[0]
        depth = fname_parts[3]
        f = open(fname, &quot;r&quot;)
        for line in f:
            out.write(ind + &quot;\t&quot; + depth + &quot;\t&quot; + line)
        f.close()
    out.close()

files = glob.glob(&quot;correlation/*txt&quot;)
gather(files, &quot;correlation.txt&quot;)

files = glob.glob(&quot;correlation-quantiles/*txt&quot;)
gather(files, &quot;correlation-quantiles.txt&quot;)</code></pre>
</div>
<div id="results" class="section level2">
<h2>Results</h2>
<pre class="r"><code>r_data &lt;- read.table(&quot;/mnt/gluster/data/internal_supp/singleCellSeq/subsampled/correlation.txt&quot;,
                     header = TRUE, sep = &quot;\t&quot;, stringsAsFactors = FALSE)</code></pre>
<p>Calculate the mean and standard error of the mean (sem) for each of the 10 iterations.</p>
<pre class="r"><code>r_data_plot &lt;- r_data %&gt;%
  group_by(ind, depth, num_cells) %&gt;%
  summarize(mean = mean(r), sem = sd(r) / length(r))</code></pre>
<pre class="r"><code>p &lt;- ggplot(r_data_plot, aes(x = num_cells, y = mean, color = as.factor(depth))) +
  geom_line() +
  geom_errorbar(aes(ymin = mean - sem, ymax = mean + sem), width = 1) +
  facet_wrap(~ind) +
  labs(x = &quot;Number of subsampled cells&quot;,
       y = &quot;Pearson&#39;s r (mean +/- sem)&quot;,
       color = &quot;Depth&quot;,
       title = &quot;Subsample: Correlation of single and bulk cells&quot;)
p</code></pre>
<p><img src="figure/correlate-single-to-bulk.Rmd/subsample-correlate-single-cell-to-bulk-1.png" title="" alt="" width="960" style="display: block; margin: auto;" /></p>
<p>Focus on the lower end of the distribution of the number of subsampled cells.</p>
<pre class="r"><code>p + xlim(0, 30)</code></pre>
<p><img src="figure/correlate-single-to-bulk.Rmd/subsample-correlate-single-cell-to-bulk-xlim-1.png" title="" alt="" width="960" style="display: block; margin: auto;" /></p>
</div>
<div id="pollen-et-al.-2014" class="section level2">
<h2>Pollen et al. 2014</h2>
<p><a href="http://www.ncbi.nlm.nih.gov/pmc/articles/PMC4191988/">Pollen et al. 2014</a> observed that the correlation between the average expression in a group of single cells and the expression in a bulk sample of 100 cells quickly saturates at about 10 single cells (Figure 2ab). Each point is the mean of 10 subsamples. The high coverage samples were sequenced at a depth of ~8.9 × 10<sup>6</sup> reads per cell, and the low coverage at ~2.7 × 10<sup>5</sup> reads per cell. The data comes from 46 K562 cells.</p>
<p><img src="figure/correlate-single-to-bulk.Rmd/pollen-2014-figure-2ab.png" alt="Pollen et al. 2014 - Figure 2ab" /></p>
</div>
<div id="results-by-expression-quantiles" class="section level2">
<h2>Results by expression quantiles</h2>
<p>In the analysis above, all genes were used. By default, the expression level in the bulk cells is used to remove the bottom 25% of genes from the analysis. This basically removes genes that are clearly unexpressed. Starting from this list of “expressed” genes, I split them into their expression quartiles. Thus we can see how the correlation is affected by the expression level.</p>
<pre class="r"><code>r_q_data &lt;- read.table(&quot;/mnt/gluster/data/internal_supp/singleCellSeq/subsampled/correlation-quantiles.txt&quot;,
                     header = TRUE, sep = &quot;\t&quot;, stringsAsFactors = FALSE)</code></pre>
<p>Calculate the mean and standard error of the mean (sem) for each of the 10 iterations.</p>
<pre class="r"><code>r_q_data_plot &lt;- r_q_data %&gt;%
  group_by(ind, depth, num_cells, quantiles) %&gt;%
  summarize(mean = mean(r), sem = sd(r) / length(r), num_genes = num_genes[1])</code></pre>
<p>Some single cells did not have enough reads for the higher subsamples.</p>
<pre class="r"><code>r_q_data_plot &lt;- na.omit(r_q_data_plot)</code></pre>
<p>The full results.</p>
<pre class="r"><code>p &lt;- ggplot(r_q_data_plot, aes(x = num_cells, y = mean, color = as.factor(depth))) +
  geom_line() +
  geom_errorbar(aes(ymin = mean - sem, ymax = mean + sem), width = 1) +
  facet_grid(quantiles~ind) +
  labs(x = &quot;Number of subsampled cells&quot;,
       y = &quot;Pearson&#39;s r (mean +/- sem)&quot;,
       color = &quot;Depth&quot;,
       title = &quot;Subsample: Correlation of single and bulk cells&quot;)
p</code></pre>
<p><img src="figure/correlate-single-to-bulk.Rmd/subsample-correlate-single-cell-to-bulk-quantiles-1.png" title="" alt="" width="960" style="display: block; margin: auto;" /></p>
<p>Focus on one cell line, two sequencing depths, and the lower end of the number cells.</p>
<pre class="r"><code>p %+% r_q_data_plot[r_q_data_plot$ind == 19098 &amp;
                    r_q_data_plot$depth %in% c(2e5, 4e6), ] +
  xlim(0, 75)</code></pre>
<p><img src="figure/correlate-single-to-bulk.Rmd/subsample-correlate-single-cell-to-bulk-xlim-quantiles-1.png" title="" alt="" width="960" style="display: block; margin: auto;" /></p>
</div>
<div id="session-information" class="section level2">
<h2>Session information</h2>
<pre class="r"><code>sessionInfo()</code></pre>
<pre><code>R version 3.2.0 (2015-04-16)
Platform: x86_64-unknown-linux-gnu (64-bit)

locale:
 [1] LC_CTYPE=en_US.UTF-8       LC_NUMERIC=C              
 [3] LC_TIME=en_US.UTF-8        LC_COLLATE=en_US.UTF-8    
 [5] LC_MONETARY=en_US.UTF-8    LC_MESSAGES=en_US.UTF-8   
 [7] LC_PAPER=en_US.UTF-8       LC_NAME=C                 
 [9] LC_ADDRESS=C               LC_TELEPHONE=C            
[11] LC_MEASUREMENT=en_US.UTF-8 LC_IDENTIFICATION=C       

attached base packages:
[1] stats     graphics  grDevices utils     datasets  methods   base     

other attached packages:
[1] ggplot2_1.0.1 dplyr_0.4.1   knitr_1.10.5 

loaded via a namespace (and not attached):
 [1] Rcpp_0.11.6      magrittr_1.5     MASS_7.3-40      munsell_0.4.2   
 [5] colorspace_1.2-6 stringr_1.0.0    plyr_1.8.2       tools_3.2.0     
 [9] parallel_3.2.0   grid_3.2.0       gtable_0.1.2     DBI_0.3.1       
[13] htmltools_0.2.6  yaml_2.1.13      lazyeval_0.1.10  assertthat_0.1  
[17] digest_0.6.8     reshape2_1.4.1   formatR_1.2      evaluate_0.7    
[21] rmarkdown_0.6.1  labeling_0.3     stringi_0.4-1    scales_0.2.4    
[25] proto_0.3-10    </code></pre>
</div>


<!-- some extra javascript for older browsers -->
<script type="text/javascript" src="libs/polyfill.js"></script>

<script>

// manage active state of menu based on current page
$(document).ready(function () {

    // active menu
    href = window.location.pathname
    href = href.substr(href.lastIndexOf('/') + 1)
    $('a[href="' + href + '"]').parent().addClass('active');

    // manage active menu header
    if (href.startsWith('authoring_'))
      $('a[href="' + 'authoring' + '"]').parent().addClass('active');
    else if (href.endsWith('_format.html'))
      $('a[href="' + 'formats' + '"]').parent().addClass('active');
    else if (href.startsWith('developer_'))
      $('a[href="' + 'developer' + '"]').parent().addClass('active');

});

</script>

</div>

<script>

// add bootstrap table styles to pandoc tables
$(document).ready(function () {
  $('tr.header').parent('thead').parent('table').addClass('table table-condensed');
});

</script>


</body>
</html>
