<!DOCTYPE html>

<html xmlns="http://www.w3.org/1999/xhtml">

<head>

<meta charset="utf-8">
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="generator" content="pandoc" />

<meta name="author" content="Joyce Hsiao" />

<meta name="date" content="2015-10-15" />

<title>Normalize coefficients of variation</title>

<script src="libs/jquery-1.11.0/jquery.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="libs/bootstrap-3.3.1/css/united.min.css" rel="stylesheet" />
<script src="libs/bootstrap-3.3.1/js/bootstrap.min.js"></script>
<script src="libs/bootstrap-3.3.1/shim/html5shiv.min.js"></script>
<script src="libs/bootstrap-3.3.1/shim/respond.min.js"></script>

<style type="text/css">

/* padding for bootstrap navbar */
body {
  padding-top: 50px;
  padding-bottom: 40px;
}


/* offset scroll position for anchor links (for fixed navbar)  */
.section h2 {
  padding-top: 55px;
  margin-top: -55px;
}
.section h3 {
  padding-top: 55px;
  margin-top: -55px;
}



/* don't use link color in navbar */
.dropdown-menu>li>a {
  color: black;
}

/* some padding for disqus */
#disqus_thread {
  margin-top: 45px;
}

</style>

<link rel="stylesheet" href="libs/font-awesome-4.1.0/css/font-awesome.min.css"/>

<style type="text/css">code{white-space: pre;}</style>
<link rel="stylesheet"
      href="libs/highlight/textmate.css"
      type="text/css" />
<script src="libs/highlight/highlight.js"></script>
<style type="text/css">
  pre:not([class]) {
    background-color: white;
  }
</style>
<script type="text/javascript">
if (window.hljs && document.readyState && document.readyState === "complete") {
   window.setTimeout(function() {
      hljs.initHighlighting();
   }, 0);
}
</script>



</head>

<body>

<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
code {
  color: inherit;
  background-color: rgba(0, 0, 0, 0.04);
}
img { 
  max-width:100%; 
  height: auto; 
}
</style>
<div class="container-fluid main-container">


<div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">singleCellSeq</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li><a href="index.html">Home</a></li>
        <li><a href="about.html">About</a></li>
        <li><a href="license.html">License</a></li>
        <li><a href="https://github.com/jdblischak/singleCellSeq">GitHub</a></li>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<div id="header">
<h1 class="title">Normalize coefficients of variation</h1>
<h4 class="author"><em>Joyce Hsiao</em></h4>
<h4 class="date"><em>2015-10-15</em></h4>
</div>

<div id="TOC">
<ul>
<li><a href="#objective">Objective</a></li>
<li><a href="#model">Model</a></li>
<li><a href="#set-up">Set up</a></li>
<li><a href="#prepare-data">Prepare data</a></li>
<li><a href="#compute-coefficient-of-variation">Compute coefficient of variation</a></li>
<li><a href="#distance-to-the-median">Distance-to-the-median</a></li>
<li><a href="#normalize-coefficient-of-variation">Normalize coefficient of variation</a></li>
<li><a href="#session-information">Session information</a></li>
</ul>
</div>

<p><strong>Last updated:</strong> 2015-12-11</p>
<p><strong>Code version:</strong> 884b335605607bd2cf5c9a1ada1e345466388866</p>
<div id="objective" class="section level2">
<h2>Objective</h2>
<p>We would like to compare coefficients of variation of gene expression across individuals, a metric that is widely employed for quantifying heterogeneity of gene expression in single cell sequencing data. Setting the discussion of heterogeneity aside, the major challenge associated with the analysis of coefficient of variation resides in the nature of count data having high coefficient of variation at the low count level. We verified the mean-CV relationship in the scatter plots of per gene coefficient of variation across cells versus mean molecule cout across cells, for each individual samples and across all samples.</p>
<p><a href="http://dx.doi.org/10.1016/j.stem.2015.09.011">Kolodziejczyk et al. 2015</a> surveyed the transcriptome profiles of mESCs cultured in three different conditions and propose DM (distance-to-the-median), a corrected version of CV that is independent of the mean, as a metric of heterogenity comparison. They followed the analytical strategy in <a href="http://dx.doi.org/10.1038/nature04785">Newman et al. 2006</a>.</p>
<p>In this document, we adopt the same strategy as in Kolodziejczyk et al. when removing the dependency of squared coefficient of variation on mean molecule count per gene. These “normalized” squared coefficients of variation will be used in all downstream analyses of cell-to-cell heterogeneity individual differences.</p>
<p>Our results indicate that after this normalization step, the coefficient of variations no longer has a polynomial relationship with mean gene molecule count.</p>
</div>
<div id="model" class="section level2">
<h2>Model</h2>
<p><br /><span class="math display"><em>l</em><em>o</em><em>g</em>10(<em>C</em><em>V</em><sub><em>g</em><em>k</em></sub><sup>2</sup>) = <em>l</em><em>o</em><em>g</em>10(<em>C</em><em>V</em><sub><em>g</em></sub><sup>2</sup>) + <em>ϵ</em><sub><em>g</em><em>k</em></sub></span><br /></p>
<p>where <span class="math inline"><em>ϵ</em><sub><em>g</em><em>k</em></sub></span> is independent distributed as a normal random variables for each gene <span class="math inline"><em>g</em></span> and sample <span class="math inline"><em>k</em></span>, and <span class="math inline"><em>C</em><em>V</em><sub><em>g</em></sub><sup>2</sup></span> is modeled as a smooth function of <span class="math inline"><em>μ</em><sub><em>g</em></sub></span>, the mean molecule count for gene <span class="math inline"><em>g</em></span>.</p>
<p>Equivalently,</p>
<p><br /><span class="math display">$$log10\left(\frac{\sigma^2_{gk}}{\mu^2_{gk}} \right) = log10 \left(\frac{\sigma^2_{g}}{\mu^2_{g}} \right) + \epsilon_{gk}$$</span><br />.</p>
</div>
<div id="set-up" class="section level2">
<h2>Set up</h2>
<pre class="r"><code>library(&quot;data.table&quot;)
library(&quot;dplyr&quot;)
library(&quot;limma&quot;)
library(&quot;edgeR&quot;)
library(&quot;ggplot2&quot;)
library(&quot;grid&quot;)
library(&quot;zoo&quot;)
theme_set(theme_bw(base_size = 12))
source(&quot;functions.R&quot;)</code></pre>
</div>
<div id="prepare-data" class="section level2">
<h2>Prepare data</h2>
<p>Input annotation of only QC-filtered single cells</p>
<pre class="r"><code>anno_qc &lt;- read.table(&quot;../data/annotation-filter.txt&quot;, header = TRUE,
                   stringsAsFactors = FALSE)
head(anno_qc)</code></pre>
<pre><code>  individual replicate well      batch      sample_id
1    NA19098        r1  A01 NA19098.r1 NA19098.r1.A01
2    NA19098        r1  A02 NA19098.r1 NA19098.r1.A02
3    NA19098        r1  A04 NA19098.r1 NA19098.r1.A04
4    NA19098        r1  A05 NA19098.r1 NA19098.r1.A05
5    NA19098        r1  A06 NA19098.r1 NA19098.r1.A06
6    NA19098        r1  A07 NA19098.r1 NA19098.r1.A07</code></pre>
<p>Input endogeneous gene molecule counts that are QC-filtered, CPM-normalized, ERCC-normalized, and also processed to remove unwanted variation from batch effet. ERCC genes are removed from this file.</p>
<pre class="r"><code>molecules_ENSG &lt;- read.table(&quot;../data/molecules-final.txt&quot;, header = TRUE, stringsAsFactors = FALSE)</code></pre>
<p>Input ERCC gene moleclue counts that are QC-filtered and CPM-normalized.</p>
<pre class="r"><code>molecules_ERCC &lt;- read.table(&quot;../data/molecules-cpm-ercc.txt&quot;, header = TRUE, stringsAsFactors = FALSE)</code></pre>
<p>Combine endogeneous and ERCC genes.</p>
<pre class="r"><code>molecules_all_genes &lt;- rbind(molecules_ENSG, molecules_ERCC)</code></pre>
<p>Input endogeneous and ERCC gene moleclule counts before log2 CPM transformation. This file is used to compute percent zero-count cells per sample.</p>
<pre class="r"><code>molecules_filter &lt;- read.table(&quot;../data/molecules-filter.txt&quot;, header = TRUE, stringsAsFactors = FALSE)

all.equal(rownames(molecules_all_genes), rownames(molecules_filter) )</code></pre>
<pre><code>[1] TRUE</code></pre>
<pre class="r"><code>tail(rownames(molecules_all_genes))</code></pre>
<pre><code>[1] &quot;ERCC-00131&quot; &quot;ERCC-00136&quot; &quot;ERCC-00145&quot; &quot;ERCC-00162&quot; &quot;ERCC-00165&quot;
[6] &quot;ERCC-00171&quot;</code></pre>
<pre class="r"><code>tail(rownames(molecules_filter))</code></pre>
<pre><code>[1] &quot;ERCC-00131&quot; &quot;ERCC-00136&quot; &quot;ERCC-00145&quot; &quot;ERCC-00162&quot; &quot;ERCC-00165&quot;
[6] &quot;ERCC-00171&quot;</code></pre>
</div>
<div id="compute-coefficient-of-variation" class="section level2">
<h2>Compute coefficient of variation</h2>
<p>Compute per batch coefficient of variation based on transformed molecule counts (on count scale).</p>
<p>Include only genes with positive coefficient of variation. Some genes in this data may have zero coefficient of variation, because we include gene with more than 0 count across all cells.</p>
<pre class="r"><code># Compute CV and mean of normalized molecule counts (take 2^(log2-normalized count))

molecules_cv_batch &lt;- 
  lapply(1:length(unique(anno_qc$batch)), function(per_batch) {
      molecules_per_batch &lt;- 2^molecules_all_genes[ , unique(anno_qc$batch) == unique(anno_qc$batch)[per_batch] ]
      mean_per_gene &lt;- apply(molecules_per_batch, 1, mean, na.rm = TRUE)
      sd_per_gene &lt;- apply(molecules_per_batch, 1, sd, na.rm = TRUE)
      cv_per_gene &lt;- data.frame(mean = mean_per_gene,
                                sd = sd_per_gene,
                                cv = sd_per_gene/mean_per_gene)
      rownames(cv_per_gene) &lt;- rownames(molecules_all_genes)
  
      # cv_per_gene &lt;- cv_per_gene[rowSums(is.na(cv_per_gene)) == 0, ]
      cv_per_gene$batch &lt;- unique(anno_qc$batch)[per_batch]
      
      # Add sparsity percent
      molecules_count &lt;- molecules_filter[ , unique(anno_qc$batch) == unique(anno_qc$batch)[per_batch]]
      cv_per_gene$sparse &lt;- rowMeans(as.matrix(molecules_count) == 0)
        
      return(cv_per_gene)
      }) 
names(molecules_cv_batch) &lt;- unique(anno_qc$batch)

sapply(molecules_cv_batch, dim)</code></pre>
<pre><code>     NA19098.r1 NA19098.r3 NA19101.r1 NA19101.r2 NA19101.r3 NA19239.r1
[1,]      10598      10598      10598      10598      10598      10598
[2,]          5          5          5          5          5          5
     NA19239.r2 NA19239.r3
[1,]      10598      10598
[2,]          5          5</code></pre>
</div>
<div id="distance-to-the-median" class="section level2">
<h2>Distance-to-the-median</h2>
<p>*This method was designed for comparison of variation profile across genes, while we are intersted in comparison of heterogeneity profiles on a per-gene basis.</p>
<p>The computation of DM for gene <em>i</em> in <a href="http://dx.doi.org/10.1016/j.stem.2015.09.011">Kolodziejczyk et al. 2015</a> involves two steps:</p>
<ol style="list-style-type: decimal">
<li>Correct for mean dependency:
<ul>
<li>log10 (CV^2 / rolling median log10 of squared CV)</li>
</ul></li>
<li>Correct for dependency on gene length
<ul>
<li>Corrected CV - gene length (union of all exons); equivalently, log10 of CV on count scale divided by gene length</li>
</ul></li>
</ol>
<p>In studies that count reads instead of molecules, gene length is a possible confounder in expression levels. However in our study, UMI is used to count the number of RNA molecules in each cell. Hence, we may not need to adjust coefficient of variation for correlation with gene lenght.</p>
</div>
<div id="normalize-coefficient-of-variation" class="section level2">
<h2>Normalize coefficient of variation</h2>
<p>Merge summary data.frames.</p>
<pre class="r"><code>df_plot &lt;- do.call(rbind, molecules_cv_batch)</code></pre>
<p>Compute rolling medians across all samples.</p>
<blockquote>
<p>We aggregate normalized molecule counts across samples to compute sample-wide rolling medians. Specifically, we ordered data-wide coefficients of variations according to their corresponding values of molecule counts. Then, we take rolling medians of the ordered coefficients of variations, using the same parameters in <a href="http://dx.doi.org/10.1016/j.stem.2015.09.011">Kolodziejczyk et al. 2015</a>: the number of genes in each window is 50, and the number of genes that overlap between windows is 25. Finally, we substract data-wide rolling medians of CVs from each sample’s CVs.</p>
</blockquote>
<pre class="r"><code># Compute a data-wide coefficient of variation on CPM normalized counts.
data_cv &lt;- apply(2^molecules_all_genes, 1, sd)/apply(2^molecules_all_genes, 1, mean)

# Order of genes by mean expression levels
order_gene &lt;- order(apply(2^molecules_all_genes, 1, mean))

# Rolling medians of log10 squared CV by mean expression levels
roll_medians &lt;- rollapply(log10(data_cv^2)[order_gene], width = 50, by = 25,
                         FUN = median, fill = list(&quot;extend&quot;, &quot;extend&quot;, &quot;NA&quot;) )
ii_na &lt;- which( is.na(roll_medians) )
roll_medians[ii_na] &lt;- median( log10(data_cv^2)[order_gene][ii_na] )

names(roll_medians) &lt;- rownames(molecules_all_genes)[order_gene]

# re-order rolling medians
reorder_gene &lt;- match(rownames(molecules_all_genes), names(roll_medians) )
head(reorder_gene)</code></pre>
<pre><code>[1]  277 3669 4688 6230 3391 2286</code></pre>
<pre class="r"><code>roll_medians &lt;- roll_medians[ reorder_gene ]

stopifnot( all.equal(names(roll_medians), rownames(molecules_all_genes) ) )</code></pre>
<p>Sanity check for the computation of rolling median.</p>
<p>Compared to data before normalization and transformation (<a href="http://jdblischak.github.io/singleCellSeq/analysis/noisy-gene.html">link</a>), ERCC gene mean molecule counts are higher than endogeneous molecule counts after normalization and transformation.</p>
<pre class="r"><code>ggplot(data.frame(cv2 = log10(data_cv^2), 
                  roll_medians = roll_medians,
                  mean = log10(apply(2^molecules_all_genes, 1, mean) ),
                  is_ERCC = (1:length(data_cv) %in% grep(&quot;ERCC&quot;, names(data_cv)) )  ) ) +
  geom_point( aes(x = mean, y = cv2, shape = factor(is_ERCC) ), col = &quot;red&quot; ) + 
  geom_point(aes(x = mean, y = roll_medians), col = &quot;blue&quot;, alpha = .7) +
  labs(x = &quot;log10 data-wide per gene molecule count&quot;,
       y = &quot;log10 squared coefficient of variation&quot;)</code></pre>
<p><img src="figure/cv-adjusted.Rmd/unnamed-chunk-10-1.png" title="" alt="" width="672" style="display: block; margin: auto;" /></p>
<p>Compute adjusted coefficient of variation.</p>
<pre class="r"><code># adjusted coefficient of variation on log10 scale
log10cv2_adj &lt;- 
  lapply(1:length(molecules_cv_batch), function(per_batch) {
    foo &lt;- log10(molecules_cv_batch[[per_batch]]$cv^2) - roll_medians
    return(foo)
})
df_plot$log10cv2_adj &lt;- do.call(c, log10cv2_adj)
df_plot$is_ERCC &lt;- ( 1:dim(df_plot)[1] %in% grep(&quot;ERCC&quot;, rownames(df_plot)) )</code></pre>
<p>Adjusted squared coefficient of variation versus log10 mean count (CPM corrected).</p>
<p>ERCC remain the outliers after substracting out rolling medians.</p>
<pre class="r"><code>ggplot( df_plot, aes(x = log10(mean), y = log10cv2_adj) ) +
  geom_point( aes(col = as.factor(batch), shape = factor(is_ERCC)), cex = .9 ) + 
  facet_wrap( ~ batch) +
  labs(x = &quot;log10(Mean CPM)&quot;, y = &quot;log10(Adjusted Squared coefficient of variation&quot;) </code></pre>
<p><img src="figure/cv-adjusted.Rmd/unnamed-chunk-12-1.png" title="" alt="" width="672" style="display: block; margin: auto;" /></p>
<p>All genes: coefficient of variation after adjustment.</p>
<pre class="r"><code>ggplot(df_plot, aes(x= factor(batch), y = log10cv2_adj, fill = factor(batch) ) ) +
  geom_violin(alpha = .5) + 
  geom_boxplot(alpha = .01, width = .2, position = position_dodge(width = .9)) +
  labs(xlab = &quot;log10 adjusted Squared coefficient of variation&quot;) +
  ggtitle( &quot;log10 CV^2 after adjustment&quot; )</code></pre>
<p><img src="figure/cv-adjusted.Rmd/unnamed-chunk-13-1.png" title="" alt="" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>  theme(axis.text.x = element_text(hjust=1, angle = 45))</code></pre>
<pre><code>List of 1
 $ axis.text.x:List of 8
  ..$ family    : NULL
  ..$ face      : NULL
  ..$ colour    : NULL
  ..$ size      : NULL
  ..$ hjust     : num 1
  ..$ vjust     : NULL
  ..$ angle     : num 45
  ..$ lineheight: NULL
  ..- attr(*, &quot;class&quot;)= chr [1:2] &quot;element_text&quot; &quot;element&quot;
 - attr(*, &quot;class&quot;)= chr [1:2] &quot;theme&quot; &quot;gg&quot;
 - attr(*, &quot;complete&quot;)= logi FALSE</code></pre>
<p>All genes: coefficient of variation before adjustment.</p>
<pre class="r"><code>ggplot(df_plot, aes(x= factor(batch), y = log10(cv^2), fill = factor(batch) ) ) +
  geom_violin(alpha = .5) + 
  geom_boxplot(alpha = .01, width = .2, position = position_dodge(width = .9)) +
  labs(xlab = &quot;log10 unadjusted Squared coefficient of variation&quot;) +
  ggtitle( &quot;log10 CV^2 before adjustment&quot; )</code></pre>
<p><img src="figure/cv-adjusted.Rmd/unnamed-chunk-14-1.png" title="" alt="" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>  theme(axis.text.x = element_text(hjust=1, angle = 45))</code></pre>
<pre><code>List of 1
 $ axis.text.x:List of 8
  ..$ family    : NULL
  ..$ face      : NULL
  ..$ colour    : NULL
  ..$ size      : NULL
  ..$ hjust     : num 1
  ..$ vjust     : NULL
  ..$ angle     : num 45
  ..$ lineheight: NULL
  ..- attr(*, &quot;class&quot;)= chr [1:2] &quot;element_text&quot; &quot;element&quot;
 - attr(*, &quot;class&quot;)= chr [1:2] &quot;theme&quot; &quot;gg&quot;
 - attr(*, &quot;complete&quot;)= logi FALSE</code></pre>
<p>Endogeneous genes: coefficient of variation after adjustment.</p>
<pre class="r"><code>ggplot(df_plot[which(!df_plot$is_ERCC), ], aes(x= factor(batch), y = log10cv2_adj, fill = factor(batch) ) ) +
  geom_violin(alpha = .5) + 
  geom_boxplot(alpha = .01, width = .2, position = position_dodge(width = .9)) +
  labs(xlab = &quot;log10 adjusted Squared coefficient of variation&quot;) +
  ggtitle( &quot;log10 CV^2 after adjustment w/o ERCC&quot; )</code></pre>
<p><img src="figure/cv-adjusted.Rmd/unnamed-chunk-15-1.png" title="" alt="" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>  theme(axis.text.x = element_text(hjust=1, angle = 45))</code></pre>
<pre><code>List of 1
 $ axis.text.x:List of 8
  ..$ family    : NULL
  ..$ face      : NULL
  ..$ colour    : NULL
  ..$ size      : NULL
  ..$ hjust     : num 1
  ..$ vjust     : NULL
  ..$ angle     : num 45
  ..$ lineheight: NULL
  ..- attr(*, &quot;class&quot;)= chr [1:2] &quot;element_text&quot; &quot;element&quot;
 - attr(*, &quot;class&quot;)= chr [1:2] &quot;theme&quot; &quot;gg&quot;
 - attr(*, &quot;complete&quot;)= logi FALSE</code></pre>
<p>Endogeneous genes: coefficient of variation before adjustment.</p>
<pre class="r"><code>ggplot(df_plot[which(!df_plot$is_ERCC), ], aes(x= factor(batch), y = log10(cv^2), fill = factor(batch) ) ) +
  geom_violin(alpha = .5) + 
  geom_boxplot(alpha = .01, width = .2, position = position_dodge(width = .9)) +
  labs(xlab = &quot;log10 unadjusted Squared coefficient of variation&quot;) +
  ggtitle( &quot;log10 CV^2 before adjustment w/o ERCC&quot; )</code></pre>
<p><img src="figure/cv-adjusted.Rmd/unnamed-chunk-16-1.png" title="" alt="" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>  theme(axis.text.x = element_text(hjust=1, angle = 45))</code></pre>
<pre><code>List of 1
 $ axis.text.x:List of 8
  ..$ family    : NULL
  ..$ face      : NULL
  ..$ colour    : NULL
  ..$ size      : NULL
  ..$ hjust     : num 1
  ..$ vjust     : NULL
  ..$ angle     : num 45
  ..$ lineheight: NULL
  ..- attr(*, &quot;class&quot;)= chr [1:2] &quot;element_text&quot; &quot;element&quot;
 - attr(*, &quot;class&quot;)= chr [1:2] &quot;theme&quot; &quot;gg&quot;
 - attr(*, &quot;complete&quot;)= logi FALSE</code></pre>
</div>
<div id="session-information" class="section level2">
<h2>Session information</h2>
<pre class="r"><code>sessionInfo()</code></pre>
<pre><code>R version 3.2.1 (2015-06-18)
Platform: x86_64-apple-darwin13.4.0 (64-bit)
Running under: OS X 10.10.5 (Yosemite)

locale:
[1] en_US.UTF-8/en_US.UTF-8/en_US.UTF-8/C/en_US.UTF-8/en_US.UTF-8

attached base packages:
[1] grid      stats     graphics  grDevices utils     datasets  methods  
[8] base     

other attached packages:
[1] zoo_1.7-12       ggplot2_1.0.1    edgeR_3.10.5     limma_3.24.15   
[5] dplyr_0.4.3      data.table_1.9.6 knitr_1.11      

loaded via a namespace (and not attached):
 [1] Rcpp_0.12.2      magrittr_1.5     MASS_7.3-45      munsell_0.4.2   
 [5] lattice_0.20-33  colorspace_1.2-6 R6_2.1.1         stringr_1.0.0   
 [9] plyr_1.8.3       tools_3.2.1      parallel_3.2.1   gtable_0.1.2    
[13] DBI_0.3.1        htmltools_0.2.6  yaml_2.1.13      assertthat_0.1  
[17] digest_0.6.8     reshape2_1.4.1   formatR_1.2.1    evaluate_0.8    
[21] rmarkdown_0.8.1  labeling_0.3     stringi_1.0-1    scales_0.3.0    
[25] chron_2.3-47     proto_0.3-10    </code></pre>
</div>


<!-- some extra javascript for older browsers -->
<script type="text/javascript" src="libs/polyfill.js"></script>

<script>

// manage active state of menu based on current page
$(document).ready(function () {

    // active menu
    href = window.location.pathname
    href = href.substr(href.lastIndexOf('/') + 1)
    $('a[href="' + href + '"]').parent().addClass('active');

    // manage active menu header
    if (href.startsWith('authoring_'))
      $('a[href="' + 'authoring' + '"]').parent().addClass('active');
    else if (href.endsWith('_format.html'))
      $('a[href="' + 'formats' + '"]').parent().addClass('active');
    else if (href.startsWith('developer_'))
      $('a[href="' + 'developer' + '"]').parent().addClass('active');

});

</script>

</div>

<script>

// add bootstrap table styles to pandoc tables
$(document).ready(function () {
  $('tr.header').parent('thead').parent('table').addClass('table table-condensed');
});

</script>


</body>
</html>
