---
title: "Cell-to-cell variation analysis: non-detected cells"
author: "Joyce Hsiao"
date: 2016-03-01  
---

**Last updated:** `r Sys.Date()`

**Code version:** `r system("git log -1 --format='%H'", intern = TRUE)`

```{r chunk-options, include=FALSE}
source("chunk-options.R")

library(knitr)
opts_chunk$set(message = FALSE, warning = FALSE, eval = TRUE, 
               echo = TRUE)
```


## Background and some observations

1. __Non-detection rate__: We found that the median of non-detected cells rate is 26% across the three individuals and ranges between 21 to 30% between the three individuals. 

2. __Mean, Var, CV__: Expectedly, gene abundance and variance was associated with percent of non-detected cells; genes with as low as 10 percent of non-detected cells were restricted in the dynamic range of abudance levels. Gene coefficient of variation was not highly correlated with percent of non-detected genes, and after correcting for mean abudance levels, we observed a zero-correlation between adjusted CV and percent of non-detected cells.





## Set up

```{r, message=FALSE, warning=FALSE}
library("data.table")
library("dplyr")
library("limma")
library("edgeR")
library("ggplot2")
library("grid")
theme_set(theme_bw(base_size = 12))
source("functions.R")
library("Humanzee")
library("cowplot")
library("MASS")
library("matrixStats")
source("../code/cv-functions.r")
source("../code/plotting-functions.r")
library("mygene")
library("aod")
```



## Prepare data

We import molecule counts before standardizing and transformation and also log2-transformed counts after batch-correction. Biological variation analysis of the individuals is performed on the batch-corrected and log2-transformed counts. 


```{r import-data}
# Import filtered annotations
anno_filter <- read.table("../data/annotation-filter.txt", 
                      header = TRUE,
                      stringsAsFactors = FALSE)

# Import filtered molecule counts
molecules_filter <- read.table("../data/molecules-filter.txt",
                               header = TRUE, stringsAsFactors = FALSE)
stopifnot(NROW(anno_filter) == NCOL(molecules_filter))

# Import final processed molecule counts of endogeneous genes
molecules_final <- read.table("../data/molecules-final.txt", 
                             header = TRUE, stringsAsFactors = FALSE)
stopifnot(NROW(anno_filter) == NCOL(molecules_final))

# Import gene symbols
gene_symbols <- read.table(file = "../data/gene-info.txt", sep = "\t",
                           header = TRUE, stringsAsFactors = FALSE, quote = "")

# Import cell-cycle gene list
cell_cycle_genes <- read.table("../data/cellcyclegenes.txt",
                               header = TRUE, sep = "\t",
                               stringsAsFactors = FALSE)

# Import pluripotency gene list

pluripotency_genes <- read.table("../data/pluripotency-genes.txt",
                               header = TRUE, sep = "\t",
                               stringsAsFactors = FALSE)$To
```


Load CV results of all cells from [previous analysis][link]

[link]: http://jdblischak.github.io/singleCellSeq/analysis/cv-adjusted-summary-pois.html

```{r}
load("../data/cv-all-cells.rda")
load("../data/cv-expressed-cells.rda")
```


Compute a matrix of 0's and 1's indicating non-detected and detected cells, respectively.

```{r}
molecules_expressed <- molecules_filter
molecules_expressed[which(molecules_filter > 0 , arr.ind = TRUE)] <- 1
molecules_expressed <- as.matrix((molecules_expressed))
```


Take the gene subset included in the final data.

```{r}
genes_included <- intersect(rownames(molecules_final),
                               rownames(expressed_cv$NA19098) )

molecules_filter_subset <- molecules_filter[
  which(rownames(molecules_filter) %in% genes_included), ]

molecules_final_subset <- molecules_final[
  which(rownames(molecules_final) %in% genes_included), ]

molecules_expressed_subset <- molecules_expressed[
  which(rownames(molecules_expressed) %in% genes_included), ]

dim(molecules_expressed_subset)
all.equal(rownames(expressed_cv$NA19098), 
          rownames(molecules_expressed_subset) )
```




## Drop-out rates

Compute drop-out rates.

```{r compute-dropout}
drop_out <- lapply(unique(anno_filter$individual), function(ind) {
  temp_df <- molecules_filter_subset[,anno_filter$individual == ind]
  zero_count <- rowMeans(temp_df == 0)
  return(zero_count)
})  
names(drop_out) <- unique(anno_filter$individual)
drop_out$all <- rowMeans(as.matrix(molecules_filter_subset) == 0)


summary(drop_out$NA19098)
summary(drop_out$NA19101)
summary(drop_out$NA19239)
summary(drop_out$all)
```



Histogram of drop-out percents.

```{r}
par(mfrow = c(2,2))
for (i in 1:4) {
h1 <- hist(drop_out[[i]], breaks = seq(0,1, by = .1), 
           plot = FALSE)
h1$density <- h1$counts/sum(h1$counts)*100
plot(h1, ylab = "Percent",
         xlab = "Proportion of non-detected cells",
         main = names(drop_out)[i], 
     freq=FALSE)
}

sapply(drop_out, function(x) mean(x>=.5))
```




Explore the role of drop-outs

NA19098, expressed cells

```{r, fig.height=6, fig.width=6}
par(mfrow = c(2,2))
plot(y = log10(expressed_cv$NA19098$expr_mean),
     x = drop_out$NA19098,
     xlab = "Percent of drop-outs",
     ylab = "log10 mean expression",
     pch = 16, cex =.3)
lines(lowess( expressed_cv$NA19098$expr_mean ~ drop_out$NA19098),
      col = "red")
plot(y = log10(expressed_cv$NA19098$expr_var),
     x = drop_out$NA19098,
     xlab = "Percent of drop-outs",
     ylab = "log10 variance of expression",
     pch = 16, cex =.3)
lines(lowess( log10(expressed_cv$NA19098$expr_var) ~ drop_out$NA19098),
      col = "red")
plot(y = expressed_cv$NA19098$expr_cv,
     x = drop_out$NA19098,
     xlab = "Percent of drop-outs",
     ylab = "CV",
     pch = 16, cex =.3)
lines(lowess( expressed_cv$NA19098$expr_cv ~ drop_out$NA19098),
      col = "red")
plot(y = expressed_dm$NA19098,
     x = drop_out$NA19098,
     xlab = "Percent of drop-outs",
     ylab = "Corrected CV",
     pch = 16, cex =.3)
lines(lowess( expressed_dm$NA19098 ~ drop_out$NA19098),
      col = "red")
title(main = "Expressed cells, NA19098", outer = TRUE, line = -1)
```


All individuals, expressed cells

```{r, fig.height=6, fig.width=6}
par(mfrow = c(2,2))
# abundance
plot(y = log10(expressed_cv$all$expr_mean),
     x = drop_out$all,
     xlab = "Percent of drop-outs",
     ylab = "log10 mean expression",
     pch = 16, cex =.3)
lines(lowess( log10(expressed_cv$all$expr_mean) ~ drop_out$all),
      col = "red")
# variance
plot(y = log10(expressed_cv$all$expr_var),
     x = drop_out$all,
     xlab = "Percent of drop-outs",
     ylab = "log10 variance of expression",
     pch = 16, cex =.3)
lines(lowess( log10(expressed_cv$all$expr_var) ~ drop_out$all),
      col = "red")
# CV
plot(y = expressed_cv$all$expr_cv,
     x = drop_out$all,
     xlab = "Percent of drop-outs",
     ylab = "CV",
     pch = 16, cex =.3)
lines(lowess( expressed_cv$all$expr_cv ~ drop_out$all),
      col = "red")
title(main = "All 3 individuals", outer = TRUE, line = -1)
```



CV and mean plus drop-out rates. (TBD)

```{r}
par(mfrow = c(2,2))
plot(y = expressed_cv$NA19098$expr_cv,
     x = log10(expressed_cv$NA19098$expr_mean),
     cex = .9, col = alpha("grey40", .8), lwd = .6,
     xlab = "log10 gene mean abundance",
     ylab = "CV",
     main = "Expressed cells, NA19098",
     xlim = c(1,5), ylim = c(0,2.5))
points(y = expressed_cv$NA19098$expr_cv,
       x = log10(expressed_cv$NA19098$expr_mean), 
       col = rev(RColorBrewer::brewer.pal(10, "RdYlBu"))[
                cut(drop_out$NA19098, breaks = seq(0, 1, by = .1),
                     include.lowest = TRUE)], 
       cex = .6, pch = 16)


plot(y = expressed_cv$all$expr_cv,
     x = log10(expressed_cv$all$expr_mean),
     cex = .9, col = alpha("grey40", .8), lwd = .6,
     xlab = "log10 gene mean abundance",
     ylab = "CV",
     main = "Expressed cells, All 3 individuals",
     xlim = c(1,5), ylim = c(0,2.5))
points(y = expressed_cv$all$expr_cv,
       x = log10(expressed_cv$all$expr_mean), 
       col = rev(RColorBrewer::brewer.pal(10, "RdYlBu"))[
                cut(drop_out$all, breaks = seq(0, 1, by = .1),
                     include.lowest = TRUE)], 
       cex = .6, pch = 16)


plot(y = expressed_dm$NA19098,
     x = log10(expressed_cv$NA19098$expr_mean),
     cex = .9, col = alpha("grey40", .8), lwd = .6,
     xlab = "log10 gene mean abundance",
     ylab = "Adjusted CV",
     main = "Expressed cells, NA19098",
     xlim = c(1,5))
points(y = expressed_dm$NA19098,
       x = log10(expressed_cv$NA19098$expr_mean), 
       col = rev(RColorBrewer::brewer.pal(10, "RdYlBu"))[
                cut(drop_out$NA19098, breaks = seq(0, 1, by = .1),
                     include.lowest = TRUE)], 
       cex = .6, pch = 16)

plot(1:10, pch = 15, cex = 2, axes = F, xlab = "", ylab = "",
     col = rev(RColorBrewer::brewer.pal(10, "RdYlBu")), xlim = c(0, 13))
text(labels = levels(cut(drop_out$NA19098, breaks = seq(0, 1, by = .1),
                     include.lowest = TRUE)),
     y = 1:10, x = 1:10 + 2, cex = .7)
title(main = "drop-out rate")
```



Variance and mean plus drop-out rates. (TBD)

```{r}
par(mfrow = c(2,2))
plot(y = log10(expressed_cv$NA19098$expr_var^(1/4)),
     x = log10(expressed_cv$NA19098$expr_mean),
     cex = .9, col = alpha("grey40", .8), lwd = .6,
     xlab = "log10 gene mean abundance",
     ylab = "log10 variance^(1/4)",
     main = "Expressed cells, NA19098",
     xlim = c(1,5) )
points(y = log10(expressed_cv$NA19098$expr_var^(1/4)),
       x = log10(expressed_cv$NA19098$expr_mean), 
       col = rev(RColorBrewer::brewer.pal(10, "RdYlBu"))[
                cut(drop_out$NA19098, breaks = seq(0, 1, by = .1),
                     include.lowest = TRUE)], 
       cex = .6, pch = 16)

plot(y = log10(expressed_cv$all$expr_var^(1/4)),
     x = log10(expressed_cv$all$expr_mean),
     cex = .9, col = alpha("grey40", .8), lwd = .6,
     xlab = "log10 gene mean abundance",
     ylab = "log10 variance^(1/4)",
     main = "Expressed cells, All 3 individuals",
     xlim = c(1,5))
points(y = log10(expressed_cv$all$expr_var^(1/4)),
       x = log10(expressed_cv$all$expr_mean), 
       col = rev(RColorBrewer::brewer.pal(10, "RdYlBu"))[
                cut(drop_out$all, breaks = seq(0, 1, by = .1),
                     include.lowest = TRUE)], 
       cex = .6, pch = 16)

plot(1:10, pch = 15, cex = 2, axes = F, xlab = "", ylab = "",
     col = rev(RColorBrewer::brewer.pal(10, "RdYlBu")), xlim = c(0, 13))
text(labels = levels(cut(drop_out$NA19098, breaks = seq(0, 1, by = .1),
                     include.lowest = TRUE)),
     y = 1:10, x = 1:10 + 2, cex = .7)
title(main = "drop-out rate")
```


## Compare non-detection rates between individuals

Fisher's exact test for between individual comparisons. For some genes, we observed no undetected cells, a scenario that calls for Fisher's exact test instead of Chi-square test or logistic regression. 


```{r}
file_name <- "../data/sig-test-zeros.rda"
if (file.exists(file_name)) {
  load(file_name)
} else {
  ind <- factor(anno_filter$individual,
              levels = unique(anno_filter$individual),
              labels = unique(anno_filter$individual) )
  fit_zero <- do.call(rbind,
      lapply(1:NROW(molecules_expressed_subset), function(ii) {
#      lapply(1:100, function(ii) {
          tab <- table(molecules_expressed_subset[ii, ], ind)
          if (dim(tab)[1] == 1) {
             if (as.numeric(rownames(tab)) == 1) {
               tab <- rbind(rep(0,3), tab)
               rownames(tab) <- c(0,1)
             } else {
               tab <- rbind(rep(1,3), tab)
               rownames(tab) <- c(0,1)
             }
          }
          pval <- lapply(rev(1:dim(tab)[2]),
                         function(i) {
                              fisher.test(tab[,-i])$p.value                 
                         })
          names(pval) <- sapply(rev(1:dim(tab)[2]), 
                               function(i) {
                                 paste(colnames(tab[,-i])[1],
                                      colnames(tab[,-1])[2],
                                      sep = "-") })
          df <- data.frame(
                do.call(cbind, pval),
                    row.names = rownames(molecules_expressed_subset)[ii] ) 
          colnames(df) <- names(pval)
          return(df)
  }) )
  fit_zero_qval <- do.call(cbind,
    lapply(1:3, function(i) {
    stats::p.adjust(fit_zero[,i], method = "fdr")
  }))
  colnames(fit_zero_qval) <- colnames(fit_zero)
  
  save(fit_zero, fit_zero_qval, file = file_name)
}
```


```{r}
par(mfrow = c(2,2))
for (i in 1:3) {
  hist(fit_zero[,i],
       main = colnames(fit_zero)[i],
       xlab = "p-value")
}
title(main = "p-value histogram", outer = TRUE, line = -1)

# genes with almost 1 p-value
table(rowSums(fit_zero > .999))

par(mfrow = c(2,2))
for (i in 1:3) {
  plot(ecdf(fit_zero[,i]),
       main = colnames(fit_zero)[i],
       xlab = "p-value",
       axes = F)
  axis(1); axis(2)
}
title(main = "p-value empirical distribution function", outer = TRUE, line = -1)


par(mfrow = c(2,2))
for (i in 1:3) {
  hist(fit_zero_qval[,i],
       main = colnames(fit_zero_qval)[i],
       xlab = "q-value")
}
title(main = "q-value histogram", outer = TRUE, line = -1)


par(mfrow = c(2,2))
for (i in 1:3) {
  plot(ecdf(fit_zero_qval[,i]),
       main = colnames(fit_zero_qval)[i],
       xlab = "q-value",
       axes = F)
  axis(1); axis(2)
}
title(main = "q-value empirical distribution function", outer = TRUE, line = -1)

table(rowSums(fit_zero_qval < .01))


library(VennDiagram)
library(gridExtra)
genes <- rownames(molecules_expressed_subset)
overlap_list <- lapply(1:3, function(i)
      genes[ which( fit_zero_qval[ ,i] < .01 ) ] )
names(overlap_list) <- colnames(fit_zero_qval)

grid.arrange(gTree(children = venn.diagram(overlap_list,filename = NULL,
                          category.names = names(overlap_list),
                          name = "q-val < .01")) )

# compute proportion of zeros in each individual
zero_prop <- do.call(cbind,
  lapply(1:3, function(i)
  rowMeans( molecules_expressed_subset[, 
          anno_filter$individual == unique(anno_filter$individual)[i]]) ) )
colnames(zero_prop) <- unique(anno_filter$individual)

genes <- rownames(molecules_expressed_subset)
overlap_list <- lapply(1:3, function(i)
      genes[ which( zero_prop[ ,i] < .5 ) ] )
names(overlap_list) <- colnames(fit_zero_qval)

grid.arrange(gTree(children = venn.diagram(overlap_list,filename = NULL,
                          category.names = names(overlap_list),
                          name = "Percent undetected cells < .5")) )


genes <- rownames(molecules_expressed_subset)
overlap_list <- lapply(1:3, function(i)
      genes[ which( zero_prop[ ,i] > .9 ) ] )
names(overlap_list) <- colnames(fit_zero_qval)

grid.arrange(gTree(children = venn.diagram(overlap_list,filename = NULL,
                          category.names = names(overlap_list),
                          name = "Percent undeteced cells > .9")) )
```




Density plot of a gene with sig. difference in proportion of non-detected cells.

```{r}
gene_elect <- 
  rownames(molecules_expressed_subset)[which(rowSums(fit_zero_qval) == 3)]
par(mfrow = c(2,2))
for (i in 1:4) {
plot_density_overlay(
  molecules = molecules_final_subset*molecules_expressed_subset,
  annotation = anno_filter,
  individual_label = NULL,
  which_gene = gene_elect[i], 
  labels = "",
  xlims = c(1,14), ylims = NULL,
  gene_symbols = gene_symbols)
}
```


## non-detected and mean expression

Compute mean expression

```{r}

```



```{r, eval = FALSE, include = FALSE}
#you have to define the newMRexperiment object
#and set libSize
#when you define it
#then in fitZig use the `useCSSoffset=FALSE`
#and make sure the normalized counts aren't logged
#obj@expSummary$expSummary` - this is where the libsize and the normfactors are hidden

# Use zero-inflated log-normal mixture implemented in metagenomeSeq.
library(metagenomeSeq)
MRobj <- newMRexperiment(
  molecules_final,
  phenoData = AnnotatedDataFrame(
    data.frame(anno_filter,
        row.names = colnames(molecules_final),
        stringsAsFactors = FALSE) ),
  featureData = AnnotatedDataFrame(
    data.frame(gene_id = rownames(molecules_final),
        row.names = rownames(molecules_final),
        stringsAsFactors = FALSE) ) ) 
                                   
design <- model.matrix(~ factor(individual,
                                levels = unique(individual)), 
                       data = pData(MRobj))

control <- zigControl(maxit=50,verbose=FALSE)


fit_zig <- fitZig(obj = MRobj,
                  mod = design,
                  zeroMod = design,
                  useCSSoffset = FALSE,
                  useMixedModel = FALSE,
                  control = control)
```



## Proportion of detected genes

```{r}
detection_gene_prop <- lapply(molecules_filter, function(x) {
                                 mean(x < 1 ) 
                              })
names(detection_gene_prop) <- colnames(molecules_filter)
detection_gene_prop <- do.call(c, detection_gene_prop)
summary(detection_gene_prop)

ggplot(data.frame(anno_filter,
                  detection_gene_prop),
       aes(x = as.factor(individual), 
           y = detection_gene_prop, 
                        fill = as.factor(batch))) + 
    geom_violin(alpha = 0.5) + 
    geom_boxplot(alpha = 0.4, 
                 width = 0.1, 
                 position = position_dodge(width = 0.9)) + 
    labs( title = "Proportion of detected genes")
```



## Session information

```{r info}
sessionInfo()
```
