<!DOCTYPE html>

<html xmlns="http://www.w3.org/1999/xhtml">

<head>

<meta charset="utf-8">
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="generator" content="pandoc" />

<meta name="author" content="Joyce Hsiao" />

<meta name="date" content="2015-10-16" />

<title>Compare Coefficient of variation between individuals</title>

<script src="libs/jquery-1.11.0/jquery.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="libs/bootstrap-3.3.1/css/united.min.css" rel="stylesheet" />
<script src="libs/bootstrap-3.3.1/js/bootstrap.min.js"></script>
<script src="libs/bootstrap-3.3.1/shim/html5shiv.min.js"></script>
<script src="libs/bootstrap-3.3.1/shim/respond.min.js"></script>

<style type="text/css">

/* padding for bootstrap navbar */
body {
  padding-top: 50px;
  padding-bottom: 40px;
}


/* offset scroll position for anchor links (for fixed navbar)  */
.section h2 {
  padding-top: 55px;
  margin-top: -55px;
}
.section h3 {
  padding-top: 55px;
  margin-top: -55px;
}



/* don't use link color in navbar */
.dropdown-menu>li>a {
  color: black;
}

/* some padding for disqus */
#disqus_thread {
  margin-top: 45px;
}

</style>

<link rel="stylesheet" href="libs/font-awesome-4.1.0/css/font-awesome.min.css"/>

<style type="text/css">code{white-space: pre;}</style>
<link rel="stylesheet"
      href="libs/highlight/textmate.css"
      type="text/css" />
<script src="libs/highlight/highlight.js"></script>
<style type="text/css">
  pre:not([class]) {
    background-color: white;
  }
</style>
<script type="text/javascript">
if (window.hljs && document.readyState && document.readyState === "complete") {
   window.setTimeout(function() {
      hljs.initHighlighting();
   }, 0);
}
</script>



</head>

<body>

<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
code {
  color: inherit;
  background-color: rgba(0, 0, 0, 0.04);
}
img { 
  max-width:100%; 
  height: auto; 
}
</style>
<div class="container-fluid main-container">


<div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">singleCellSeq</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li><a href="index.html">Home</a></li>
        <li><a href="about.html">About</a></li>
        <li><a href="license.html">License</a></li>
        <li><a href="https://github.com/jdblischak/singleCellSeq">GitHub</a></li>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<div id="header">
<h1 class="title">Compare Coefficient of variation between individuals</h1>
<h4 class="author"><em>Joyce Hsiao</em></h4>
<h4 class="date"><em>2015-10-16</em></h4>
</div>

<div id="TOC">
<ul>
<li><a href="#objective">Objective</a></li>
<li><a href="#set-up">Set up</a></li>
<li><a href="#prepare-data">Prepare data</a></li>
<li><a href="#normalize-coefficient-of-variation">Normalize coefficient of variation</a></li>
<li><a href="#limma">limma</a></li>
<li><a href="#overlapy-with-pluripotency-genes">Overlapy with pluripotency genes</a></li>
<li><a href="#overlap-with-cell-cycle-genes">Overlap with cell-cycle genes</a><ul>
<li><a href="#gene-annotations">Gene annotations</a></li>
</ul></li>
<li><a href="#session-information">Session information</a></li>
</ul>
</div>

<p><strong>Last updated:</strong> 2015-12-11</p>
<p><strong>Code version:</strong> 884b335605607bd2cf5c9a1ada1e345466388866</p>
<div id="objective" class="section level2">
<h2>Objective</h2>
<p>Previously, we used the entire set of data including all individuals and all batches to perform normalization of coefficient of variations (<a href="http://jdblischak.github.io/singleCellSeq/analysis/cv-adjusted.html">link1</a>) and performed</p>
<p>In this document, we wil perform per gene differential analysis of coefficients of variations using only coefficients of variation from ENSG genes and also from QC-approved batches (no NA19098.r2).</p>
</div>
<div id="set-up" class="section level2">
<h2>Set up</h2>
<pre class="r"><code>library(&quot;data.table&quot;)
library(&quot;dplyr&quot;)
library(&quot;limma&quot;)
library(&quot;edgeR&quot;)
library(&quot;ggplot2&quot;)
library(&quot;grid&quot;)
library(&quot;zoo&quot;)
theme_set(theme_bw(base_size = 12))
source(&quot;functions.R&quot;)</code></pre>
</div>
<div id="prepare-data" class="section level2">
<h2>Prepare data</h2>
<p>Input annotation of only QC-filtered single cells</p>
<pre class="r"><code>anno_qc_filter &lt;- read.table(&quot;../data/annotation-filter.txt&quot;, header = TRUE,
                   stringsAsFactors = FALSE)
head(anno_qc_filter)</code></pre>
<pre><code>  individual replicate well      batch      sample_id
1    NA19098        r1  A01 NA19098.r1 NA19098.r1.A01
2    NA19098        r1  A02 NA19098.r1 NA19098.r1.A02
3    NA19098        r1  A04 NA19098.r1 NA19098.r1.A04
4    NA19098        r1  A05 NA19098.r1 NA19098.r1.A05
5    NA19098        r1  A06 NA19098.r1 NA19098.r1.A06
6    NA19098        r1  A07 NA19098.r1 NA19098.r1.A07</code></pre>
<p>Import endogeneous gene molecule counts that are QC-filtered, CPM-normalized, ERCC-normalized, and also processed to remove unwanted variation from batch effet. ERCC genes are removed from this file.</p>
<pre class="r"><code>molecules_ENSG &lt;- read.table(&quot;../data/molecules-final.txt&quot;, header = TRUE, stringsAsFactors = FALSE)</code></pre>
<p>Input moleclule counts before log2 CPM transformation. This file is used to compute percent zero-count cells per sample.</p>
<pre class="r"><code>molecules_sparse &lt;- read.table(&quot;../data/molecules-filter.txt&quot;, header = TRUE, stringsAsFactors = FALSE)

molecules_sparse &lt;- molecules_sparse[grep(&quot;ENSG&quot;, rownames(molecules_sparse)), ]
stopifnot( all.equal(rownames(molecules_ENSG), rownames(molecules_sparse)) )</code></pre>
</div>
<div id="normalize-coefficient-of-variation" class="section level2">
<h2>Normalize coefficient of variation</h2>
<p>Compute coefficient of variation.</p>
<pre class="r"><code># Compute CV and mean of normalized molecule counts (take 2^(log2-normalized count))

molecules_cv_batch_ENSG &lt;- 
  lapply(1:length(unique(anno_qc_filter$batch)), function(per_batch) {
      molecules_per_batch &lt;- 2^molecules_ENSG[ , unique(anno_qc_filter$batch) == unique(anno_qc_filter$batch)[per_batch] ]
      mean_per_gene &lt;- apply(molecules_per_batch, 1, mean, na.rm = TRUE)
      sd_per_gene &lt;- apply(molecules_per_batch, 1, sd, na.rm = TRUE)
      cv_per_gene &lt;- data.frame(mean = mean_per_gene,
                                sd = sd_per_gene,
                                cv = sd_per_gene/mean_per_gene)
      rownames(cv_per_gene) &lt;- rownames(molecules_ENSG)
  
      # cv_per_gene &lt;- cv_per_gene[rowSums(is.na(cv_per_gene)) == 0, ]
      cv_per_gene$batch &lt;- unique(anno_qc_filter$batch)[per_batch]
      
      # Add sparsity percent
      molecules_sparse_per_batch &lt;- molecules_sparse[ , unique(anno_qc_filter$batch) == unique(anno_qc_filter$batch)[per_batch]]
      cv_per_gene$sparse &lt;- rowMeans(as.matrix(molecules_sparse_per_batch) == 0)
        
      return(cv_per_gene)
      }) 
names(molecules_cv_batch_ENSG) &lt;- unique(anno_qc_filter$batch)

sapply(molecules_cv_batch_ENSG, dim)</code></pre>
<pre><code>     NA19098.r1 NA19098.r3 NA19101.r1 NA19101.r2 NA19101.r3 NA19239.r1
[1,]      10564      10564      10564      10564      10564      10564
[2,]          5          5          5          5          5          5
     NA19239.r2 NA19239.r3
[1,]      10564      10564
[2,]          5          5</code></pre>
<p>Merge summary data.frames.</p>
<pre class="r"><code>df_ENSG &lt;- do.call(rbind, molecules_cv_batch_ENSG)</code></pre>
<p>Compute rolling medians across all samples.</p>
<pre class="r"><code># Compute a data-wide coefficient of variation on CPM normalized counts.
data_cv_ENSG &lt;- apply(2^molecules_ENSG, 1, sd)/apply(2^molecules_ENSG, 1, mean)

# Order of genes by mean expression levels
order_gene &lt;- order(apply(2^molecules_ENSG, 1, mean))

# Rolling medians of log10 squared CV by mean expression levels
roll_medians &lt;- rollapply(log10(data_cv_ENSG^2)[order_gene], width = 50, by = 25,
                         FUN = median, fill = list(&quot;extend&quot;, &quot;extend&quot;, &quot;NA&quot;) )
ii_na &lt;- which( is.na(roll_medians) )
roll_medians[ii_na] &lt;- median( log10(data_cv_ENSG^2)[order_gene][ii_na] )

names(roll_medians) &lt;- rownames(molecules_ENSG)[order_gene]


# re-order rolling medians
reorder_gene &lt;- match(rownames(molecules_ENSG), names(roll_medians) )
roll_medians &lt;- roll_medians[ reorder_gene ]

stopifnot( all.equal(names(roll_medians), rownames(molecules_ENSG) ) )</code></pre>
<p>Compute adjusted coefficient of variation.</p>
<pre class="r"><code># adjusted coefficient of variation on log10 scale
log10cv2_adj_ENSG &lt;- 
  lapply(1:length(molecules_cv_batch_ENSG), function(per_batch) {
    foo &lt;- log10(molecules_cv_batch_ENSG[[per_batch]]$cv^2) - roll_medians
    return(foo)
})
df_ENSG$log10cv2_adj_ENSG &lt;- do.call(c, log10cv2_adj_ENSG)</code></pre>
</div>
<div id="limma" class="section level2">
<h2>limma</h2>
<p>We use the <em>lmFit</em> function in the limma package to fit linear models comparing coefficients of variations between individuals for all genes. <em>lmFit</em> provides a fast algorithm for least-square estimation, when design matrix is the same for all genes.</p>
<pre class="r"><code>library(limma)

df_limma &lt;- matrix(df_ENSG$log10cv2_adj_ENSG, 
                      nrow = nrow(molecules_ENSG), ncol = 8, byrow = FALSE)

design &lt;- data.frame(individual = factor(rep(unique(anno_qc_filter$individual), each = 3) ),
                     rep = factor(rep(c(1:3), times = 3)) )
design &lt;- design[ with(design, !(individual == &quot;NA19098&quot; &amp; rep == &quot;2&quot;)), ]

colnames(df_limma) &lt;- with(design, paste0(individual, rep))

fit_limma &lt;- lmFit(df_limma, design = model.matrix( ~ individual, data = design))                      
fit_limma &lt;- eBayes(fit_limma)</code></pre>
<p>False discover control adjutment.</p>
<pre class="r"><code>F.p.adj &lt;- p.adjust(fit_limma$F.p.value, method = &quot;fdr&quot;)</code></pre>
<p>Cutoffs</p>
<pre class="r"><code>df_cuts &lt;- data.frame(cuts = c(.001, .01, .05, .1, .15, .2))
df_cuts$sig_count &lt;- sapply(1:6, function(per_cut) {
                    sum(F.p.adj &lt; df_cuts$cuts[per_cut] )
                    })
df_cuts</code></pre>
<pre><code>   cuts sig_count
1 0.001        91
2 0.010       192
3 0.050       385
4 0.100       560
5 0.150       780
6 0.200      1004</code></pre>
</div>
<div id="overlapy-with-pluripotency-genes" class="section level2">
<h2>Overlapy with pluripotency genes</h2>
<p>Input pluripotency genes. A list of 27 pluripotency genes used to demonstrate iPSC heterogeneity in Narshinh2011 Gene ID conversion was done by using the DAVID <a href="http://david.abcc.ncifcrf.gov" class="uri">http://david.abcc.ncifcrf.gov</a></p>
<pre class="r"><code>pluripotency_genes &lt;- read.table(&quot;../data/pluripotency-genes.txt&quot;, header = TRUE, sep=&quot;\t&quot;)</code></pre>
<pre class="r"><code>pluripotent_count &lt;- lapply(1:nrow(df_cuts), function(per_cut) {
    is_pluripotent &lt;- rownames(molecules_ENSG)[which(F.p.adj&lt; df_cuts$cuts[per_cut])] %in% pluripotency_genes$To
    sum(is_pluripotent)
  })
pluripotent_count &lt;- do.call(rbind, pluripotent_count)

rownames(pluripotent_count) &lt;- df_cuts$cuts
kable(pluripotent_count)</code></pre>
<table>
<tbody>
<tr class="odd">
<td align="left">0.001</td>
<td align="right">1</td>
</tr>
<tr class="even">
<td align="left">0.01</td>
<td align="right">2</td>
</tr>
<tr class="odd">
<td align="left">0.05</td>
<td align="right">4</td>
</tr>
<tr class="even">
<td align="left">0.1</td>
<td align="right">4</td>
</tr>
<tr class="odd">
<td align="left">0.15</td>
<td align="right">4</td>
</tr>
<tr class="even">
<td align="left">0.2</td>
<td align="right">5</td>
</tr>
</tbody>
</table>
</div>
<div id="overlap-with-cell-cycle-genes" class="section level2">
<h2>Overlap with cell-cycle genes</h2>
<p>Input cell cycle gene Gene sets reflecting 5 cell cycle phases were taken from Table S2 of Macosko2015 Gene ID conversion was done by using the DAVID <a href="http://david.abcc.ncifcrf.gov" class="uri">http://david.abcc.ncifcrf.gov</a></p>
<pre class="r"><code>cell_cycle_genes &lt;- read.table(&quot;../data/cellcyclegenes.txt&quot;, header = TRUE, sep=&quot;\t&quot;)</code></pre>
<pre class="r"><code>cell_cycle_count &lt;- lapply(1:nrow(df_cuts), function(per_cut) {
  per_cycle &lt;- sapply(1:ncol(cell_cycle_genes), function(per_cycle) {
        is_cell_cycle &lt;- rownames(molecules_ENSG)[which(F.p.adj&lt; df_cuts$cuts[per_cut])] %in% cell_cycle_genes[[per_cycle]]
        sum(is_cell_cycle)
  })
})
cell_cycle_count &lt;- do.call(rbind, cell_cycle_count)

colnames(cell_cycle_count) &lt;- colnames(cell_cycle_genes)
rownames(cell_cycle_count) &lt;- df_cuts$cuts
cell_cycle_count &lt;- cbind(df_cuts, cell_cycle_count)

kable(cell_cycle_count)</code></pre>
<table>
<thead>
<tr class="header">
<th align="left"></th>
<th align="right">cuts</th>
<th align="right">sig_count</th>
<th align="right">G1.S</th>
<th align="right">S</th>
<th align="right">G2.M</th>
<th align="right">M</th>
<th align="right">M.G1</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td align="left">0.001</td>
<td align="right">0.001</td>
<td align="right">91</td>
<td align="right">0</td>
<td align="right">2</td>
<td align="right">3</td>
<td align="right">2</td>
<td align="right">2</td>
</tr>
<tr class="even">
<td align="left">0.01</td>
<td align="right">0.010</td>
<td align="right">192</td>
<td align="right">1</td>
<td align="right">2</td>
<td align="right">3</td>
<td align="right">3</td>
<td align="right">2</td>
</tr>
<tr class="odd">
<td align="left">0.05</td>
<td align="right">0.050</td>
<td align="right">385</td>
<td align="right">1</td>
<td align="right">3</td>
<td align="right">3</td>
<td align="right">5</td>
<td align="right">2</td>
</tr>
<tr class="even">
<td align="left">0.1</td>
<td align="right">0.100</td>
<td align="right">560</td>
<td align="right">1</td>
<td align="right">5</td>
<td align="right">6</td>
<td align="right">8</td>
<td align="right">4</td>
</tr>
<tr class="odd">
<td align="left">0.15</td>
<td align="right">0.150</td>
<td align="right">780</td>
<td align="right">1</td>
<td align="right">6</td>
<td align="right">7</td>
<td align="right">10</td>
<td align="right">5</td>
</tr>
<tr class="even">
<td align="left">0.2</td>
<td align="right">0.200</td>
<td align="right">1004</td>
<td align="right">2</td>
<td align="right">6</td>
<td align="right">9</td>
<td align="right">13</td>
<td align="right">6</td>
</tr>
</tbody>
</table>
<div id="gene-annotations" class="section level3">
<h3>Gene annotations</h3>
<p>For genes that we found to have FDR &lt; .001 (18 genes) in our analysis comparing coefficient of variations between individuals, we look up the gene annotations as follows</p>
<pre class="r"><code>library(knitr)
library(biomaRt)

ensembl &lt;- useMart(host = &quot;grch37.ensembl.org&quot;,
                   biomart = &quot;ENSEMBL_MART_ENSEMBL&quot;,
                   dataset = &quot;hsapiens_gene_ensembl&quot;)

diff_cv_genes &lt;- rownames(molecules_ENSG)[which(F.p.adj&lt; .001)]

diff_cv_genes_info &lt;- getBM(attributes = c(&quot;ensembl_gene_id&quot;, &quot;chromosome_name&quot;,
                                           &quot;external_gene_name&quot;, &quot;transcript_count&quot;,
                                           &quot;description&quot;),
                            filters = &quot;ensembl_gene_id&quot;,
                            values = diff_cv_genes[grep(&quot;ENSG&quot;, diff_cv_genes)],
                            mart = ensembl)
kable(diff_cv_genes_info)</code></pre>
<table>
<thead>
<tr class="header">
<th align="left">ensembl_gene_id</th>
<th align="left">chromosome_name</th>
<th align="left">external_gene_name</th>
<th align="right">transcript_count</th>
<th align="left">description</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td align="left">ENSG00000003436</td>
<td align="left">2</td>
<td align="left">TFPI</td>
<td align="right">13</td>
<td align="left">tissue factor pathway inhibitor (lipoprotein-associated coagulation inhibitor) [Source:HGNC Symbol;Acc:11760]</td>
</tr>
<tr class="even">
<td align="left">ENSG00000010932</td>
<td align="left">1</td>
<td align="left">FMO1</td>
<td align="right">8</td>
<td align="left">flavin containing monooxygenase 1 [Source:HGNC Symbol;Acc:3769]</td>
</tr>
<tr class="odd">
<td align="left">ENSG00000018625</td>
<td align="left">1</td>
<td align="left">ATP1A2</td>
<td align="right">8</td>
<td align="left">ATPase, Na+/K+ transporting, alpha 2 polypeptide [Source:HGNC Symbol;Acc:800]</td>
</tr>
<tr class="even">
<td align="left">ENSG00000022556</td>
<td align="left">19</td>
<td align="left">NLRP2</td>
<td align="right">20</td>
<td align="left">NLR family, pyrin domain containing 2 [Source:HGNC Symbol;Acc:22948]</td>
</tr>
<tr class="odd">
<td align="left">ENSG00000026025</td>
<td align="left">10</td>
<td align="left">VIM</td>
<td align="right">10</td>
<td align="left">vimentin [Source:HGNC Symbol;Acc:12692]</td>
</tr>
<tr class="even">
<td align="left">ENSG00000053438</td>
<td align="left">20</td>
<td align="left">NNAT</td>
<td align="right">2</td>
<td align="left">neuronatin [Source:HGNC Symbol;Acc:7860]</td>
</tr>
<tr class="odd">
<td align="left">ENSG00000067066</td>
<td align="left">2</td>
<td align="left">SP100</td>
<td align="right">20</td>
<td align="left">SP100 nuclear antigen [Source:HGNC Symbol;Acc:11206]</td>
</tr>
<tr class="even">
<td align="left">ENSG00000077152</td>
<td align="left">1</td>
<td align="left">UBE2T</td>
<td align="right">3</td>
<td align="left">ubiquitin-conjugating enzyme E2T (putative) [Source:HGNC Symbol;Acc:25009]</td>
</tr>
<tr class="odd">
<td align="left">ENSG00000088305</td>
<td align="left">20</td>
<td align="left">DNMT3B</td>
<td align="right">8</td>
<td align="left">DNA (cytosine-5-)-methyltransferase 3 beta [Source:HGNC Symbol;Acc:2979]</td>
</tr>
<tr class="even">
<td align="left">ENSG00000089116</td>
<td align="left">12</td>
<td align="left">LHX5</td>
<td align="right">2</td>
<td align="left">LIM homeobox 5 [Source:HGNC Symbol;Acc:14216]</td>
</tr>
<tr class="odd">
<td align="left">ENSG00000092421</td>
<td align="left">5</td>
<td align="left">SEMA6A</td>
<td align="right">16</td>
<td align="left">sema domain, transmembrane domain (TM), and cytoplasmic domain, (semaphorin) 6A [Source:HGNC Symbol;Acc:10738]</td>
</tr>
<tr class="even">
<td align="left">ENSG00000092964</td>
<td align="left">8</td>
<td align="left">DPYSL2</td>
<td align="right">8</td>
<td align="left">dihydropyrimidinase-like 2 [Source:HGNC Symbol;Acc:3014]</td>
</tr>
<tr class="odd">
<td align="left">ENSG00000106153</td>
<td align="left">7</td>
<td align="left">CHCHD2</td>
<td align="right">2</td>
<td align="left">coiled-coil-helix-coiled-coil-helix domain containing 2 [Source:HGNC Symbol;Acc:21645]</td>
</tr>
<tr class="even">
<td align="left">ENSG00000108518</td>
<td align="left">17</td>
<td align="left">PFN1</td>
<td align="right">3</td>
<td align="left">profilin 1 [Source:HGNC Symbol;Acc:8881]</td>
</tr>
<tr class="odd">
<td align="left">ENSG00000109971</td>
<td align="left">11</td>
<td align="left">HSPA8</td>
<td align="right">24</td>
<td align="left">heat shock 70kDa protein 8 [Source:HGNC Symbol;Acc:5241]</td>
</tr>
<tr class="even">
<td align="left">ENSG00000110492</td>
<td align="left">11</td>
<td align="left">MDK</td>
<td align="right">12</td>
<td align="left">midkine (neurite growth-promoting factor 2) [Source:HGNC Symbol;Acc:6972]</td>
</tr>
<tr class="odd">
<td align="left">ENSG00000112511</td>
<td align="left">6</td>
<td align="left">PHF1</td>
<td align="right">11</td>
<td align="left">PHD finger protein 1 [Source:HGNC Symbol;Acc:8919]</td>
</tr>
<tr class="even">
<td align="left">ENSG00000112655</td>
<td align="left">6</td>
<td align="left">PTK7</td>
<td align="right">20</td>
<td align="left">protein tyrosine kinase 7 [Source:HGNC Symbol;Acc:9618]</td>
</tr>
<tr class="odd">
<td align="left">ENSG00000117318</td>
<td align="left">1</td>
<td align="left">ID3</td>
<td align="right">3</td>
<td align="left">inhibitor of DNA binding 3, dominant negative helix-loop-helix protein [Source:HGNC Symbol;Acc:5362]</td>
</tr>
<tr class="even">
<td align="left">ENSG00000117724</td>
<td align="left">1</td>
<td align="left">CENPF</td>
<td align="right">5</td>
<td align="left">centromere protein F, 350/400kDa [Source:HGNC Symbol;Acc:1857]</td>
</tr>
<tr class="odd">
<td align="left">ENSG00000118523</td>
<td align="left">6</td>
<td align="left">CTGF</td>
<td align="right">1</td>
<td align="left">connective tissue growth factor [Source:HGNC Symbol;Acc:2500]</td>
</tr>
<tr class="even">
<td align="left">ENSG00000120738</td>
<td align="left">5</td>
<td align="left">EGR1</td>
<td align="right">1</td>
<td align="left">early growth response 1 [Source:HGNC Symbol;Acc:3238]</td>
</tr>
<tr class="odd">
<td align="left">ENSG00000124766</td>
<td align="left">6</td>
<td align="left">SOX4</td>
<td align="right">2</td>
<td align="left">SRY (sex determining region Y)-box 4 [Source:HGNC Symbol;Acc:11200]</td>
</tr>
<tr class="even">
<td align="left">ENSG00000125144</td>
<td align="left">16</td>
<td align="left">MT1G</td>
<td align="right">4</td>
<td align="left">metallothionein 1G [Source:HGNC Symbol;Acc:7399]</td>
</tr>
<tr class="odd">
<td align="left">ENSG00000125148</td>
<td align="left">16</td>
<td align="left">MT2A</td>
<td align="right">5</td>
<td align="left">metallothionein 2A [Source:HGNC Symbol;Acc:7406]</td>
</tr>
<tr class="even">
<td align="left">ENSG00000131747</td>
<td align="left">17</td>
<td align="left">TOP2A</td>
<td align="right">5</td>
<td align="left">topoisomerase (DNA) II alpha 170kDa [Source:HGNC Symbol;Acc:11989]</td>
</tr>
<tr class="odd">
<td align="left">ENSG00000133316</td>
<td align="left">11</td>
<td align="left">WDR74</td>
<td align="right">17</td>
<td align="left">WD repeat domain 74 [Source:HGNC Symbol;Acc:25529]</td>
</tr>
<tr class="even">
<td align="left">ENSG00000135549</td>
<td align="left">6</td>
<td align="left">PKIB</td>
<td align="right">9</td>
<td align="left">protein kinase (cAMP-dependent, catalytic) inhibitor beta [Source:HGNC Symbol;Acc:9018]</td>
</tr>
<tr class="odd">
<td align="left">ENSG00000138035</td>
<td align="left">2</td>
<td align="left">PNPT1</td>
<td align="right">6</td>
<td align="left">polyribonucleotide nucleotidyltransferase 1 [Source:HGNC Symbol;Acc:23166]</td>
</tr>
<tr class="even">
<td align="left">ENSG00000139318</td>
<td align="left">12</td>
<td align="left">DUSP6</td>
<td align="right">5</td>
<td align="left">dual specificity phosphatase 6 [Source:HGNC Symbol;Acc:3072]</td>
</tr>
<tr class="odd">
<td align="left">ENSG00000142544</td>
<td align="left">19</td>
<td align="left">CTU1</td>
<td align="right">1</td>
<td align="left">cytosolic thiouridylase subunit 1 [Source:HGNC Symbol;Acc:29590]</td>
</tr>
<tr class="even">
<td align="left">ENSG00000142871</td>
<td align="left">1</td>
<td align="left">CYR61</td>
<td align="right">2</td>
<td align="left">cysteine-rich, angiogenic inducer, 61 [Source:HGNC Symbol;Acc:2654]</td>
</tr>
<tr class="odd">
<td align="left">ENSG00000143248</td>
<td align="left">1</td>
<td align="left">RGS5</td>
<td align="right">11</td>
<td align="left">regulator of G-protein signaling 5 [Source:HGNC Symbol;Acc:10001]</td>
</tr>
<tr class="even">
<td align="left">ENSG00000143375</td>
<td align="left">1</td>
<td align="left">CGN</td>
<td align="right">8</td>
<td align="left">cingulin [Source:HGNC Symbol;Acc:17429]</td>
</tr>
<tr class="odd">
<td align="left">ENSG00000143537</td>
<td align="left">1</td>
<td align="left">ADAM15</td>
<td align="right">33</td>
<td align="left">ADAM metallopeptidase domain 15 [Source:HGNC Symbol;Acc:193]</td>
</tr>
<tr class="even">
<td align="left">ENSG00000149591</td>
<td align="left">11</td>
<td align="left">TAGLN</td>
<td align="right">8</td>
<td align="left">transgelin [Source:HGNC Symbol;Acc:11553]</td>
</tr>
<tr class="odd">
<td align="left">ENSG00000152661</td>
<td align="left">6</td>
<td align="left">GJA1</td>
<td align="right">1</td>
<td align="left">gap junction protein, alpha 1, 43kDa [Source:HGNC Symbol;Acc:4274]</td>
</tr>
<tr class="even">
<td align="left">ENSG00000154556</td>
<td align="left">4</td>
<td align="left">SORBS2</td>
<td align="right">65</td>
<td align="left">sorbin and SH3 domain containing 2 [Source:HGNC Symbol;Acc:24098]</td>
</tr>
<tr class="odd">
<td align="left">ENSG00000158106</td>
<td align="left">8</td>
<td align="left">RHPN1</td>
<td align="right">3</td>
<td align="left">rhophilin, Rho GTPase binding protein 1 [Source:HGNC Symbol;Acc:19973]</td>
</tr>
<tr class="even">
<td align="left">ENSG00000158717</td>
<td align="left">16</td>
<td align="left">RNF166</td>
<td align="right">11</td>
<td align="left">ring finger protein 166 [Source:HGNC Symbol;Acc:28856]</td>
</tr>
<tr class="odd">
<td align="left">ENSG00000159239</td>
<td align="left">2</td>
<td align="left">C2orf81</td>
<td align="right">5</td>
<td align="left">chromosome 2 open reading frame 81 [Source:HGNC Symbol;Acc:34350]</td>
</tr>
<tr class="even">
<td align="left">ENSG00000163191</td>
<td align="left">1</td>
<td align="left">S100A11</td>
<td align="right">2</td>
<td align="left">S100 calcium binding protein A11 [Source:HGNC Symbol;Acc:10488]</td>
</tr>
<tr class="odd">
<td align="left">ENSG00000163702</td>
<td align="left">3</td>
<td align="left">IL17RC</td>
<td align="right">30</td>
<td align="left">interleukin 17 receptor C [Source:HGNC Symbol;Acc:18358]</td>
</tr>
<tr class="even">
<td align="left">ENSG00000164434</td>
<td align="left">6</td>
<td align="left">FABP7</td>
<td align="right">2</td>
<td align="left">fatty acid binding protein 7, brain [Source:HGNC Symbol;Acc:3562]</td>
</tr>
<tr class="odd">
<td align="left">ENSG00000164897</td>
<td align="left">7</td>
<td align="left">TMUB1</td>
<td align="right">7</td>
<td align="left">transmembrane and ubiquitin-like domain containing 1 [Source:HGNC Symbol;Acc:21709]</td>
</tr>
<tr class="even">
<td align="left">ENSG00000166426</td>
<td align="left">15</td>
<td align="left">CRABP1</td>
<td align="right">3</td>
<td align="left">cellular retinoic acid binding protein 1 [Source:HGNC Symbol;Acc:2338]</td>
</tr>
<tr class="odd">
<td align="left">ENSG00000166450</td>
<td align="left">15</td>
<td align="left">PRTG</td>
<td align="right">5</td>
<td align="left">protogenin [Source:HGNC Symbol;Acc:26373]</td>
</tr>
<tr class="even">
<td align="left">ENSG00000167996</td>
<td align="left">11</td>
<td align="left">FTH1</td>
<td align="right">11</td>
<td align="left">ferritin, heavy polypeptide 1 [Source:HGNC Symbol;Acc:3976]</td>
</tr>
<tr class="odd">
<td align="left">ENSG00000168614</td>
<td align="left">1</td>
<td align="left">NBPF9</td>
<td align="right">14</td>
<td align="left">neuroblastoma breakpoint family, member 9 [Source:HGNC Symbol;Acc:31991]</td>
</tr>
<tr class="even">
<td align="left">ENSG00000168758</td>
<td align="left">2</td>
<td align="left">SEMA4C</td>
<td align="right">6</td>
<td align="left">sema domain, immunoglobulin domain (Ig), transmembrane domain (TM) and short cytoplasmic domain, (semaphorin) 4C [Source:HGNC Symbol;Acc:10731]</td>
</tr>
<tr class="odd">
<td align="left">ENSG00000169218</td>
<td align="left">1</td>
<td align="left">RSPO1</td>
<td align="right">6</td>
<td align="left">R-spondin 1 [Source:HGNC Symbol;Acc:21679]</td>
</tr>
<tr class="even">
<td align="left">ENSG00000169230</td>
<td align="left">5</td>
<td align="left">PRELID1</td>
<td align="right">8</td>
<td align="left">PRELI domain containing 1 [Source:HGNC Symbol;Acc:30255]</td>
</tr>
<tr class="odd">
<td align="left">ENSG00000169715</td>
<td align="left">16</td>
<td align="left">MT1E</td>
<td align="right">3</td>
<td align="left">metallothionein 1E [Source:HGNC Symbol;Acc:7397]</td>
</tr>
<tr class="even">
<td align="left">ENSG00000170561</td>
<td align="left">5</td>
<td align="left">IRX2</td>
<td align="right">3</td>
<td align="left">iroquois homeobox 2 [Source:HGNC Symbol;Acc:14359]</td>
</tr>
<tr class="odd">
<td align="left">ENSG00000172717</td>
<td align="left">14</td>
<td align="left">FAM71D</td>
<td align="right">10</td>
<td align="left">family with sequence similarity 71, member D [Source:HGNC Symbol;Acc:20101]</td>
</tr>
<tr class="even">
<td align="left">ENSG00000174720</td>
<td align="left">4</td>
<td align="left">LARP7</td>
<td align="right">15</td>
<td align="left">La ribonucleoprotein domain family, member 7 [Source:HGNC Symbol;Acc:24912]</td>
</tr>
<tr class="odd">
<td align="left">ENSG00000175063</td>
<td align="left">20</td>
<td align="left">UBE2C</td>
<td align="right">7</td>
<td align="left">ubiquitin-conjugating enzyme E2C [Source:HGNC Symbol;Acc:15937]</td>
</tr>
<tr class="even">
<td align="left">ENSG00000175137</td>
<td align="left">1</td>
<td align="left">SH3BP5L</td>
<td align="right">5</td>
<td align="left">SH3-binding domain protein 5-like [Source:HGNC Symbol;Acc:29360]</td>
</tr>
<tr class="odd">
<td align="left">ENSG00000176386</td>
<td align="left">9</td>
<td align="left">CDC26</td>
<td align="right">2</td>
<td align="left">cell division cycle 26 [Source:HGNC Symbol;Acc:17839]</td>
</tr>
<tr class="even">
<td align="left">ENSG00000177954</td>
<td align="left">1</td>
<td align="left">RPS27</td>
<td align="right">4</td>
<td align="left">ribosomal protein S27 [Source:HGNC Symbol;Acc:10416]</td>
</tr>
<tr class="odd">
<td align="left">ENSG00000178531</td>
<td align="left">19</td>
<td align="left">CTXN1</td>
<td align="right">1</td>
<td align="left">cortexin 1 [Source:HGNC Symbol;Acc:31108]</td>
</tr>
<tr class="even">
<td align="left">ENSG00000182481</td>
<td align="left">17</td>
<td align="left">KPNA2</td>
<td align="right">7</td>
<td align="left">karyopherin alpha 2 (RAG cohort 1, importin alpha 1) [Source:HGNC Symbol;Acc:6395]</td>
</tr>
<tr class="odd">
<td align="left">ENSG00000184674</td>
<td align="left">22</td>
<td align="left">GSTT1</td>
<td align="right">13</td>
<td align="left">glutathione S-transferase theta 1 [Source:HGNC Symbol;Acc:4641]</td>
</tr>
<tr class="even">
<td align="left">ENSG00000185088</td>
<td align="left">15</td>
<td align="left">RPS27L</td>
<td align="right">7</td>
<td align="left">ribosomal protein S27-like [Source:HGNC Symbol;Acc:18476]</td>
</tr>
<tr class="odd">
<td align="left">ENSG00000185885</td>
<td align="left">11</td>
<td align="left">IFITM1</td>
<td align="right">4</td>
<td align="left">interferon induced transmembrane protein 1 [Source:HGNC Symbol;Acc:5412]</td>
</tr>
<tr class="even">
<td align="left">ENSG00000186010</td>
<td align="left">19</td>
<td align="left">NDUFA13</td>
<td align="right">9</td>
<td align="left">NADH dehydrogenase (ubiquinone) 1 alpha subcomplex, 13 [Source:HGNC Symbol;Acc:17194]</td>
</tr>
<tr class="odd">
<td align="left">ENSG00000187193</td>
<td align="left">16</td>
<td align="left">MT1X</td>
<td align="right">4</td>
<td align="left">metallothionein 1X [Source:HGNC Symbol;Acc:7405]</td>
</tr>
<tr class="even">
<td align="left">ENSG00000188822</td>
<td align="left">1</td>
<td align="left">CNR2</td>
<td align="right">2</td>
<td align="left">cannabinoid receptor 2 (macrophage) [Source:HGNC Symbol;Acc:2160]</td>
</tr>
<tr class="odd">
<td align="left">ENSG00000196700</td>
<td align="left">20</td>
<td align="left">ZNF512B</td>
<td align="right">3</td>
<td align="left">zinc finger protein 512B [Source:HGNC Symbol;Acc:29212]</td>
</tr>
<tr class="even">
<td align="left">ENSG00000196862</td>
<td align="left">2</td>
<td align="left">RGPD4</td>
<td align="right">2</td>
<td align="left">RANBP2-like and GRIP domain containing 4 [Source:HGNC Symbol;Acc:32417]</td>
</tr>
<tr class="odd">
<td align="left">ENSG00000198417</td>
<td align="left">16</td>
<td align="left">MT1F</td>
<td align="right">5</td>
<td align="left">metallothionein 1F [Source:HGNC Symbol;Acc:7398]</td>
</tr>
<tr class="even">
<td align="left">ENSG00000198695</td>
<td align="left">MT</td>
<td align="left">MT-ND6</td>
<td align="right">1</td>
<td align="left">mitochondrially encoded NADH dehydrogenase 6 [Source:HGNC Symbol;Acc:7462]</td>
</tr>
<tr class="odd">
<td align="left">ENSG00000198825</td>
<td align="left">10</td>
<td align="left">INPP5F</td>
<td align="right">5</td>
<td align="left">inositol polyphosphate-5-phosphatase F [Source:HGNC Symbol;Acc:17054]</td>
</tr>
<tr class="even">
<td align="left">ENSG00000198840</td>
<td align="left">MT</td>
<td align="left">MT-ND3</td>
<td align="right">1</td>
<td align="left">mitochondrially encoded NADH dehydrogenase 3 [Source:HGNC Symbol;Acc:7458]</td>
</tr>
<tr class="odd">
<td align="left">ENSG00000198865</td>
<td align="left">5</td>
<td align="left">CCDC152</td>
<td align="right">2</td>
<td align="left">coiled-coil domain containing 152 [Source:HGNC Symbol;Acc:34438]</td>
</tr>
<tr class="even">
<td align="left">ENSG00000205358</td>
<td align="left">16</td>
<td align="left">MT1H</td>
<td align="right">2</td>
<td align="left">metallothionein 1H [Source:HGNC Symbol;Acc:7400]</td>
</tr>
<tr class="odd">
<td align="left">ENSG00000205542</td>
<td align="left">X</td>
<td align="left">TMSB4X</td>
<td align="right">4</td>
<td align="left">thymosin beta 4, X-linked [Source:HGNC Symbol;Acc:11881]</td>
</tr>
<tr class="even">
<td align="left">ENSG00000212907</td>
<td align="left">MT</td>
<td align="left">MT-ND4L</td>
<td align="right">1</td>
<td align="left">mitochondrially encoded NADH dehydrogenase 4L [Source:HGNC Symbol;Acc:7460]</td>
</tr>
<tr class="odd">
<td align="left">ENSG00000222036</td>
<td align="left">14</td>
<td align="left">POTEG</td>
<td align="right">3</td>
<td align="left">POTE ankyrin domain family, member G [Source:HGNC Symbol;Acc:33896]</td>
</tr>
<tr class="even">
<td align="left">ENSG00000226784</td>
<td align="left">X</td>
<td align="left">PGAM4</td>
<td align="right">1</td>
<td align="left">phosphoglycerate mutase family member 4 [Source:HGNC Symbol;Acc:21731]</td>
</tr>
<tr class="odd">
<td align="left">ENSG00000236334</td>
<td align="left">1</td>
<td align="left">PPIAL4G</td>
<td align="right">1</td>
<td align="left">peptidylprolyl isomerase A (cyclophilin A)-like 4G [Source:HGNC Symbol;Acc:33996]</td>
</tr>
<tr class="even">
<td align="left">ENSG00000240038</td>
<td align="left">1</td>
<td align="left">AMY2B</td>
<td align="right">7</td>
<td align="left">amylase, alpha 2B (pancreatic) [Source:HGNC Symbol;Acc:478]</td>
</tr>
<tr class="odd">
<td align="left">ENSG00000241343</td>
<td align="left">X</td>
<td align="left">RPL36A</td>
<td align="right">9</td>
<td align="left">ribosomal protein L36a [Source:HGNC Symbol;Acc:10359]</td>
</tr>
<tr class="even">
<td align="left">ENSG00000249222</td>
<td align="left">22</td>
<td align="left">ATP5L2</td>
<td align="right">1</td>
<td align="left">ATP synthase, H+ transporting, mitochondrial Fo complex, subunit G2 [Source:HGNC Symbol;Acc:13213]</td>
</tr>
<tr class="odd">
<td align="left">ENSG00000250254</td>
<td align="left">4</td>
<td align="left">PTTG2</td>
<td align="right">1</td>
<td align="left">pituitary tumor-transforming 2 [Source:HGNC Symbol;Acc:9691]</td>
</tr>
<tr class="even">
<td align="left">ENSG00000255823</td>
<td align="left">11</td>
<td align="left">MTRNR2L8</td>
<td align="right">1</td>
<td align="left">MT-RNR2-like 8 [Source:HGNC Symbol;Acc:37165]</td>
</tr>
<tr class="odd">
<td align="left">ENSG00000256618</td>
<td align="left">17</td>
<td align="left">MTRNR2L1</td>
<td align="right">1</td>
<td align="left">MT-RNR2-like 1 [Source:HGNC Symbol;Acc:37155]</td>
</tr>
<tr class="even">
<td align="left">ENSG00000261221</td>
<td align="left">19</td>
<td align="left">ZNF865</td>
<td align="right">1</td>
<td align="left">zinc finger protein 865 [Source:HGNC Symbol;Acc:38705]</td>
</tr>
<tr class="odd">
<td align="left">ENSG00000261740</td>
<td align="left">16</td>
<td align="left">RP11-345J4.5</td>
<td align="right">3</td>
<td align="left">BolA-like protein 2 [Source:UniProtKB/TrEMBL;Acc:H3BV85]</td>
</tr>
<tr class="even">
<td align="left">ENSG00000269028</td>
<td align="left">3</td>
<td align="left">MTRNR2L12</td>
<td align="right">1</td>
<td align="left">MT-RNR2-like 12 (pseudogene) [Source:HGNC Symbol;Acc:37169]</td>
</tr>
<tr class="odd">
<td align="left">ENSG00000271043</td>
<td align="left">5</td>
<td align="left">MTRNR2L2</td>
<td align="right">1</td>
<td align="left">MT-RNR2-like 2 [Source:HGNC Symbol;Acc:37156]</td>
</tr>
</tbody>
</table>
</div>
</div>
<div id="session-information" class="section level2">
<h2>Session information</h2>
<pre class="r"><code>sessionInfo()</code></pre>
<pre><code>R version 3.2.1 (2015-06-18)
Platform: x86_64-apple-darwin13.4.0 (64-bit)
Running under: OS X 10.10.5 (Yosemite)

locale:
[1] en_US.UTF-8/en_US.UTF-8/en_US.UTF-8/C/en_US.UTF-8/en_US.UTF-8

attached base packages:
[1] grid      stats     graphics  grDevices utils     datasets  methods  
[8] base     

other attached packages:
[1] biomaRt_2.24.1   zoo_1.7-12       ggplot2_1.0.1    edgeR_3.10.5    
[5] limma_3.24.15    dplyr_0.4.3      data.table_1.9.6 knitr_1.11      

loaded via a namespace (and not attached):
 [1] Rcpp_0.12.2          GenomeInfoDb_1.4.3   formatR_1.2.1       
 [4] plyr_1.8.3           highr_0.5.1          bitops_1.0-6        
 [7] tools_3.2.1          digest_0.6.8         evaluate_0.8        
[10] RSQLite_1.0.0        gtable_0.1.2         lattice_0.20-33     
[13] DBI_0.3.1            yaml_2.1.13          parallel_3.2.1      
[16] proto_0.3-10         stringr_1.0.0        IRanges_2.2.9       
[19] S4Vectors_0.6.6      stats4_3.2.1         Biobase_2.28.0      
[22] R6_2.1.1             AnnotationDbi_1.30.1 XML_3.98-1.3        
[25] rmarkdown_0.8.1      reshape2_1.4.1       magrittr_1.5        
[28] scales_0.3.0         htmltools_0.2.6      MASS_7.3-45         
[31] BiocGenerics_0.14.0  assertthat_0.1       colorspace_1.2-6    
[34] stringi_1.0-1        RCurl_1.95-4.7       munsell_0.4.2       
[37] chron_2.3-47        </code></pre>
</div>


<!-- some extra javascript for older browsers -->
<script type="text/javascript" src="libs/polyfill.js"></script>

<script>

// manage active state of menu based on current page
$(document).ready(function () {

    // active menu
    href = window.location.pathname
    href = href.substr(href.lastIndexOf('/') + 1)
    $('a[href="' + href + '"]').parent().addClass('active');

    // manage active menu header
    if (href.startsWith('authoring_'))
      $('a[href="' + 'authoring' + '"]').parent().addClass('active');
    else if (href.endsWith('_format.html'))
      $('a[href="' + 'formats' + '"]').parent().addClass('active');
    else if (href.startsWith('developer_'))
      $('a[href="' + 'developer' + '"]').parent().addClass('active');

});

</script>

</div>

<script>

// add bootstrap table styles to pandoc tables
$(document).ready(function () {
  $('tr.header').parent('thead').parent('table').addClass('table table-condensed');
});

</script>


</body>
</html>
