<!DOCTYPE html>

<html xmlns="http://www.w3.org/1999/xhtml">

<head>

<meta charset="utf-8">
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="generator" content="pandoc" />

<meta name="author" content="Joyce Hsiao" />

<meta name="date" content="2015-11-12" />

<title>Per gene statistical comparison of CVs</title>

<script src="libs/jquery-1.11.0/jquery.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="libs/bootstrap-3.3.1/css/united.min.css" rel="stylesheet" />
<script src="libs/bootstrap-3.3.1/js/bootstrap.min.js"></script>
<script src="libs/bootstrap-3.3.1/shim/html5shiv.min.js"></script>
<script src="libs/bootstrap-3.3.1/shim/respond.min.js"></script>

<style type="text/css">

/* padding for bootstrap navbar */
body {
  padding-top: 50px;
  padding-bottom: 40px;
}


/* offset scroll position for anchor links (for fixed navbar)  */
.section h2 {
  padding-top: 55px;
  margin-top: -55px;
}
.section h3 {
  padding-top: 55px;
  margin-top: -55px;
}



/* don't use link color in navbar */
.dropdown-menu>li>a {
  color: black;
}

/* some padding for disqus */
#disqus_thread {
  margin-top: 45px;
}

</style>

<link rel="stylesheet" href="libs/font-awesome-4.1.0/css/font-awesome.min.css"/>

<style type="text/css">code{white-space: pre;}</style>
<link rel="stylesheet"
      href="libs/highlight/textmate.css"
      type="text/css" />
<script src="libs/highlight/highlight.js"></script>
<style type="text/css">
  pre:not([class]) {
    background-color: white;
  }
</style>
<script type="text/javascript">
if (window.hljs && document.readyState && document.readyState === "complete") {
   window.setTimeout(function() {
      hljs.initHighlighting();
   }, 0);
}
</script>



</head>

<body>

<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
code {
  color: inherit;
  background-color: rgba(0, 0, 0, 0.04);
}
img { 
  max-width:100%; 
  height: auto; 
}
</style>
<div class="container-fluid main-container">


<div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">singleCellSeq</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li><a href="index.html">Home</a></li>
        <li><a href="about.html">About</a></li>
        <li><a href="license.html">License</a></li>
        <li><a href="https://github.com/jdblischak/singleCellSeq">GitHub</a></li>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<div id="header">
<h1 class="title">Per gene statistical comparison of CVs</h1>
<h4 class="author"><em>Joyce Hsiao</em></h4>
<h4 class="date"><em>2015-11-12</em></h4>
</div>

<div id="TOC">
<ul>
<li><a href="#objective">Objective</a></li>
<li><a href="#set-up">Set up</a></li>
<li><a href="#prepare-data">Prepare data</a></li>
<li><a href="#import-gene-symbols">Import gene symbols</a></li>
<li><a href="#compute-normalized-cv">Compute normalized CV</a></li>
<li><a href="#compute-summary-measure-of-deviation">Compute summary measure of deviation</a></li>
<li><a href="#bootstrapped-cis">Bootstrapped CIs</a><ul>
<li><a href="#sample-size-is-the-orignal-sample-size">Sample size is the orignal sample size</a></li>
</ul></li>
<li><a href="#session-information">Session information</a></li>
</ul>
</div>

<p><strong>Last updated:</strong> 2015-12-11</p>
<p><strong>Code version:</strong> 7a25b2aef5c8329d4b5394cce595bbfc2b5b0c3e</p>
<div id="objective" class="section level2">
<h2>Objective</h2>
<p>Quantify statistical significance of individual differences between adjusted CVs. Here we focus on bootstrapped CIs of individual CVs.</p>
<p>For cells within each individual cell line, we constructed 1000 samples with replacment and then computed 95% confidence interval. This method resulted in narrow confidence interval.</p>
<p>Since sample size is a major factor in the width of confidence interval, instead of the entire set of cells for each individual, we sample a subset of the cells with replacement; the number of the sample is set to be 90, close to the number of wells on each C1 plate.</p>
</div>
<div id="set-up" class="section level2">
<h2>Set up</h2>
<pre class="r"><code>library(&quot;data.table&quot;)
library(&quot;dplyr&quot;)
library(&quot;limma&quot;)
library(&quot;edgeR&quot;)
library(&quot;ggplot2&quot;)
library(&quot;grid&quot;)
library(&quot;Humanzee&quot;)
theme_set(theme_bw(base_size = 12))
source(&quot;functions.R&quot;)</code></pre>
</div>
<div id="prepare-data" class="section level2">
<h2>Prepare data</h2>
<p>Input quality single cells</p>
<pre class="r"><code>quality_single_cells &lt;- read.table(&quot;../data/quality-single-cells.txt&quot;, 
                                  header = FALSE,
                                  stringsAsFactors = FALSE)
str(quality_single_cells)</code></pre>
<pre><code>&#39;data.frame&#39;:   568 obs. of  1 variable:
 $ V1: chr  &quot;NA19098.r1.A01&quot; &quot;NA19098.r1.A02&quot; &quot;NA19098.r1.A04&quot; &quot;NA19098.r1.A05&quot; ...</code></pre>
<p>Input annotation of only QC-filtered single cells. Remove NA19098.r2</p>
<pre class="r"><code>anno_filter &lt;- read.table(&quot;../data/annotation-filter.txt&quot;, 
                    header = TRUE,
                    stringsAsFactors = FALSE)
dim(anno_filter)</code></pre>
<pre><code>[1] 568   5</code></pre>
<p>Import endogeneous gene molecule counts that are QC-filtered, CPM-normalized, ERCC-normalized, and also processed to remove unwanted variation from batch effet. ERCC genes are removed from this file.</p>
<pre class="r"><code>molecules_ENSG &lt;- read.table(&quot;../data/molecules-final.txt&quot;, 
                             header = TRUE, stringsAsFactors = FALSE)
stopifnot(NCOL(molecules_ENSG) == NROW(anno_filter))</code></pre>
</div>
<div id="import-gene-symbols" class="section level2">
<h2>Import gene symbols</h2>
<pre class="r"><code>gene_info &lt;- read.table(&quot;../data/gene-info.txt&quot;, sep = &quot;\t&quot;,
                        header = TRUE, stringsAsFactors = FALSE)
str(gene_info)</code></pre>
<pre><code>&#39;data.frame&#39;:   6152 obs. of  5 variables:
 $ ensembl_gene_id   : chr  &quot;ENSG00000000003&quot; &quot;ENSG00000000005&quot; &quot;ENSG00000000419&quot; &quot;ENSG00000000457&quot; ...
 $ chromosome_name   : chr  &quot;X&quot; &quot;X&quot; &quot;20&quot; &quot;1&quot; ...
 $ external_gene_name: chr  &quot;TSPAN6&quot; &quot;TNMD&quot; &quot;DPM1&quot; &quot;SCYL3&quot; ...
 $ transcript_count  : int  3 2 7 5 10 4 12 2 13 8 ...
 $ description       : chr  &quot;tetraspanin 6 [Source:HGNC Symbol;Acc:11858]&quot; &quot;tenomodulin [Source:HGNC Symbol;Acc:17757]&quot; &quot;dolichyl-phosphate mannosyltransferase polypeptide 1, catalytic subunit [Source:HGNC Symbol;Acc:3005]&quot; &quot;SCY1-like 3 (S. cerevisiae) [Source:HGNC Symbol;Acc:19285]&quot; ...</code></pre>
</div>
<div id="compute-normalized-cv" class="section level2">
<h2>Compute normalized CV</h2>
<p>We compute squared CV across cells for each individual and then for each individual CV profile, account for mean dependency by computing distance with respect to the data-wide coefficient variation on the log10 scale.</p>
<pre class="r"><code>ENSG_cv &lt;- Humanzee::compute_cv(log2counts = molecules_ENSG,
                      grouping_vector = anno_filter$individual)

ENSG_cv_adj &lt;- Humanzee::normalize_cv(group_cv = ENSG_cv, 
                            log2counts = molecules_ENSG, 
                            anno = anno_filter)</code></pre>
</div>
<div id="compute-summary-measure-of-deviation" class="section level2">
<h2>Compute summary measure of deviation</h2>
<ul>
<li>Standardize the each CV vectors</li>
</ul>
<p>Individual CV vectors are standarized for individual CV mean and coefficients of variation across genes.</p>
<pre class="r"><code>df_cv &lt;- data.frame(NA19098 = ENSG_cv_adj[[1]]$log10cv2_adj,
                    NA19101 = ENSG_cv_adj[[2]]$log10cv2_adj,
                    NA19239 = ENSG_cv_adj[[3]]$log10cv2_adj)

library(matrixStats)
df_norm &lt;- sweep(df_cv, MARGIN = 2, STATS = colMeans(as.matrix(df_cv)), FUN = &quot;-&quot;)
df_norm &lt;- sweep(df_norm, MARGIN = 2, STATS = sqrt(colVars(as.matrix(df_cv))), FUN = &quot;/&quot;)
colnames(df_norm) &lt;- names(ENSG_cv_adj)</code></pre>
<ul>
<li>SSM and SAM</li>
</ul>
<p>Compute metrics for quantifying similarity between the three individual coefficients of variation.</p>
<pre class="r"><code>library(matrixStats)

df_norm &lt;- as.data.frame(df_norm)
df_norm$squared_dev &lt;- rowSums( ( df_norm - rowMedians(as.matrix(df_norm)) )^2 )
df_norm$abs_dev &lt;- rowSums(abs( df_norm - rowMedians(as.matrix(df_norm)) ))</code></pre>
</div>
<div id="bootstrapped-cis" class="section level2">
<h2>Bootstrapped CIs</h2>
<div id="sample-size-is-the-orignal-sample-size" class="section level3">
<h3>Sample size is the orignal sample size</h3>
<pre class="r"><code>load(&quot;rda/cv-adjusted-statistical-test-bootstrap/bootstrap-ci.rda&quot;)
rownames(abs_dev_pnt) &lt;- rownames(molecules_ENSG)
rownames(squared_dev_pnt) &lt;- rownames(molecules_ENSG)

head(cbind(squared_dev_pnt, df_norm$squared_dev))</code></pre>
<pre><code>                      25%     97.5%           
ENSG00000237683 0.4301777  5.343718 0.02747056
ENSG00000188976 0.4824864  4.999148 0.14337580
ENSG00000187608 3.3836449 10.265239 2.51312909
ENSG00000188157 0.2737914  3.458007 0.49346477
ENSG00000078808 0.2217499  2.585129 0.75898223
ENSG00000176022 0.4122886  7.099454 0.10725585</code></pre>
<pre class="r"><code>head(cbind(abs_dev_pnt, df_norm$abs_dev))</code></pre>
<pre><code>                      25%    97.5%          
ENSG00000237683 0.8048241 2.826562 0.6859387
ENSG00000188976 0.8860306 2.749277 1.1479143
ENSG00000187608 2.4104778 4.375608 4.5782764
ENSG00000188157 0.6458733 2.250246 3.6291050
ENSG00000078808 0.5955066 1.991099 3.2945706
ENSG00000176022 0.8035054 3.271384 0.9898609</code></pre>
<p>Genes with SSM outside of the 95% boostrapped CI are consistented to be statistical significant.</p>
<pre class="r"><code>sig_gene_CI_squared_dev &lt;- df_norm$squared_dev &gt; squared_dev_pnt[, 2] | df_norm$squared_dev &lt; squared_dev_pnt[, 1]

sig_gene_CI_abs_dev &lt;- df_norm$abs_dev &gt; abs_dev_pnt[, 2] | df_norm$abs_dev &lt; abs_dev_pnt[, 1]</code></pre>
<pre class="r"><code>library(broman)
crayon &lt;- brocolors(&quot;crayon&quot;)
ggplot(data.frame(index = c(mean(sig_gene_CI_squared_dev == 0), 
                            mean(sig_gene_CI_squared_dev == 1),
                            mean(sig_gene_CI_abs_dev == 0 ),
                            mean(sig_gene_CI_abs_dev == 1) ),
                  type = factor( rep(c(0,1), times = 2),
                                 labels = c(&quot;Non-sig.&quot;, &quot;Sig.&quot;) ),
                  metric = rep( c(&quot;SSM&quot;, &quot;SAM&quot;), each = 2)  ),
      aes(x = metric, y = 100*index, fill = type) ) +
  geom_bar(stat = &quot;identity&quot;, colour = &quot;white&quot;) +
  scale_fill_manual(values = c(&quot;orange&quot;, &quot;grey80&quot;)) +
  ylab(&quot;Percent&quot;) + xlab(&quot;Similarity metric&quot;) </code></pre>
<p><img src="figure/cv-adjusted-statistical-test-bootstrap.Rmd/unnamed-chunk-5-1.png" title="" alt="" width="672" style="display: block; margin: auto;" /></p>
<p>Correspondance of SSM and SAM results.</p>
<pre class="r"><code>library(gplots)
venn(list(SSM = rownames(molecules_ENSG)[which(sig_gene_CI_squared_dev)],
          SAM = rownames(molecules_ENSG)[which(sig_gene_CI_abs_dev)] ))</code></pre>
<p><img src="figure/cv-adjusted-statistical-test-bootstrap.Rmd/unnamed-chunk-6-1.png" title="" alt="" width="672" style="display: block; margin: auto;" /></p>
</div>
</div>
<div id="session-information" class="section level2">
<h2>Session information</h2>
<pre class="r"><code>sessionInfo()</code></pre>
<pre><code>R version 3.2.1 (2015-06-18)
Platform: x86_64-apple-darwin13.4.0 (64-bit)
Running under: OS X 10.10.5 (Yosemite)

locale:
[1] en_US.UTF-8/en_US.UTF-8/en_US.UTF-8/C/en_US.UTF-8/en_US.UTF-8

attached base packages:
[1] grid      stats     graphics  grDevices utils     datasets  methods  
[8] base     

other attached packages:
 [1] gplots_2.17.0      broman_0.59-5      matrixStats_0.15.0
 [4] zoo_1.7-12         Humanzee_0.1.0     ggplot2_1.0.1     
 [7] edgeR_3.10.5       limma_3.24.15      dplyr_0.4.3       
[10] data.table_1.9.6   knitr_1.11        

loaded via a namespace (and not attached):
 [1] Rcpp_0.12.2        formatR_1.2.1      plyr_1.8.3        
 [4] bitops_1.0-6       tools_3.2.1        digest_0.6.8      
 [7] evaluate_0.8       gtable_0.1.2       lattice_0.20-33   
[10] DBI_0.3.1          yaml_2.1.13        parallel_3.2.1    
[13] proto_0.3-10       stringr_1.0.0      gtools_3.5.0      
[16] caTools_1.17.1     R6_2.1.1           rmarkdown_0.8.1   
[19] gdata_2.17.0       reshape2_1.4.1     magrittr_1.5      
[22] scales_0.3.0       htmltools_0.2.6    MASS_7.3-45       
[25] assertthat_0.1     colorspace_1.2-6   labeling_0.3      
[28] KernSmooth_2.23-15 stringi_1.0-1      munsell_0.4.2     
[31] chron_2.3-47      </code></pre>
</div>


<!-- some extra javascript for older browsers -->
<script type="text/javascript" src="libs/polyfill.js"></script>

<script>

// manage active state of menu based on current page
$(document).ready(function () {

    // active menu
    href = window.location.pathname
    href = href.substr(href.lastIndexOf('/') + 1)
    $('a[href="' + href + '"]').parent().addClass('active');

    // manage active menu header
    if (href.startsWith('authoring_'))
      $('a[href="' + 'authoring' + '"]').parent().addClass('active');
    else if (href.endsWith('_format.html'))
      $('a[href="' + 'formats' + '"]').parent().addClass('active');
    else if (href.startsWith('developer_'))
      $('a[href="' + 'developer' + '"]').parent().addClass('active');

});

</script>

</div>

<script>

// add bootstrap table styles to pandoc tables
$(document).ready(function () {
  $('tr.header').parent('thead').parent('table').addClass('table table-condensed');
});

</script>


</body>
</html>
