---
title: "Compare transcriptome-wide profile of CVs"
author: "Joyce Hsiao"
date: 2015-10-25
---

**Last updated:** `r Sys.Date()`

**Code version:** `r system("git log -1 --format='%H'", intern = TRUE)`

```{r chunk-options, include=FALSE}
source("chunk-options.R")

library(knitr)
opts_chunk$set(message = FALSE, warning = FALSE, eval = TRUE, 
               echo = TRUE)
```


## Objective

Compare transcriptome profile of cell-to-cell heterogeneity within individuals betwen batches and between individuals. 

Here we compute CV for each of the batches and then follow the algorithm described in [Kolodziejczk et al. (2015)](link1) to remove mean dependency of the batch-specific CVs. Then, in order to compare CVs between individuals, we need to account for the multiple CVs for each gene for each individual cell line. The statistical test of choice would be a nonparametric equivalent of two-way ANOVA, which turns out to be difficult to compute. 





[link1]: http://dx.doi.org/10.1016/j.stem.2015.09.011

## Set up

```{r, message=FALSE, warning=FALSE}
library("data.table")
library("dplyr")
library("limma")
library("edgeR")
library("ggplot2")
library("grid")
theme_set(theme_bw(base_size = 12))
source("functions.R")
```


## Prepare data

Input annotation of only QC-filtered single cells. Remove NA19098.r2

```{r}
anno_qc <- read.table("../data/annotation-filter.txt", header = TRUE,
                   stringsAsFactors = FALSE)
is_include <- anno_qc$batch != "NA19098.r2"
anno_qc_filter <- anno_qc[which(is_include), ]
```


Import endogeneous gene molecule counts that are QC-filtered, CPM-normalized, ERCC-normalized, and also processed to remove unwanted variation from batch effet. ERCC genes are removed from this file.

```{r}
molecules_ENSG <- read.table("../data/molecules-final.txt", header = TRUE, stringsAsFactors = FALSE)
molecules_ENSG <- molecules_ENSG[ , is_include]
```

Input moleclule counts before log2 CPM transformation. This file is used to compute percent zero-count cells per sample.

```{r}
molecules_sparse <- read.table("../data/molecules-filter.txt", header = TRUE, stringsAsFactors = FALSE)

molecules_sparse <- molecules_sparse[grep("ENSG", rownames(molecules_sparse)), ]
stopifnot( all.equal(rownames(molecules_ENSG), rownames(molecules_sparse)) )
```


## Compute normalized CV 

```{r}
source("../code/cv-functions.r")

# compute_cv() takes log2 counts as input
ENSG_cv <- compute_cv(counts = molecules_ENSG,
                           anno = anno_qc_filter)

ENSG_cv_adj <- normalize_cv(batch_cv = ENSG_cv, 
                                 counts = molecules_ENSG, 
                                 anno = anno_qc_filter)
```


## Compare transcriptome profile within individuals


### Within each individual

The CV normalization method employed here scales each sample individual with respect to 
data-wide coefficients of variation. Therefore, we would expect that the transcriptome-wide
profile relationship between samples (with or between individuals) to be consistent before 
versus after normalization.


* Before normalization

```{r}
print(ENSG_cv[[1]]$individual[1])
print(friedman.test( cbind(ENSG_cv[[1]]$cv, ENSG_cv[[2]]$cv) ) )

print(ENSG_cv[[3]]$individual[3])
print(friedman.test( cbind(ENSG_cv[[3]]$cv, ENSG_cv[[4]]$cv, ENSG_cv[[5]]$cv) ) )

print(ENSG_cv[[6]]$individual[6])
print(friedman.test( cbind(ENSG_cv[[6]]$cv, ENSG_cv[[7]]$cv, ENSG_cv[[8]]$cv) ) )
```

* After normalization

```{r}
print(ENSG_cv_adj[[1]]$individual[1])
print(friedman.test( cbind(ENSG_cv_adj[[1]]$log10cv2_adj, ENSG_cv_adj[[2]]$log10cv2_adj) ) )

print(ENSG_cv[[3]]$individual[3])
print(friedman.test( cbind(ENSG_cv_adj[[3]]$log10cv2_adj, 
                           ENSG_cv_adj[[4]]$log10cv2_adj, ENSG_cv_adj[[5]]$log10cv2_adj) ) )

print(ENSG_cv[[6]]$individual[6])
print(friedman.test( cbind(ENSG_cv_adj[[6]]$log10cv2_adj, 
                           ENSG_cv_adj[[7]]$log10cv2_adj, ENSG_cv_adj[[8]]$log10cv2_adj) ) )
```

*Plot CV before/after normalization

The normalization method we used is a non-linear transformation. The method aims to remove heterogeneity of variances across mean gene expression levels. 

```{r}
print(ENSG_cv[[6]]$individual[6])
plot(x = rank(log10((ENSG_cv_adj[[6]]$cv)^2)),
     y = rank(ENSG_cv_adj[[6]]$log10cv2_adj),
     xlab = "Before normalization", ylab = "After normalization",
     cex = .4, pch = 16, main = "Ranks of CV")
abline(0, 1, col = "red")

print(ENSG_cv[[6]]$individual[6])
plot(x = (log10((ENSG_cv_adj[[6]]$cv)^2)),
     y = (ENSG_cv_adj[[6]]$log10cv2_adj),
     xlab = "Before normalization", ylab = "After normalization",
     cex = .4, pch = 16, main = "log10 squared-CV")
abline(0, 1, col = "red")
```


## Compare transcriptome profile between individual, adjusted CV


* PCA

Perform PCA for each individual. Consider replicates. Then compare PCA gene loadings between individuals.

Use *prcomp* to perform PCA:
* output$x: loadings
* output$rotation: loadings
* output$center: variable means
* output$scale: variable standard deviation

```{r}
library(matrixStats)
pca_individuals <- lapply(1:3, function(ii_individual) {
  if (ii_individual == 1) {
  df_ind <- cbind(ENSG_cv_adj$NA19098.r1$log10cv2_adj,
                  ENSG_cv_adj$NA19098.r3$log10cv2_adj ) 
  df_ind <- (df_ind - rowMeans(df_ind))/sqrt(rowVars(df_ind))

  }
  if (ii_individual == 2) {
  df_ind <- cbind(ENSG_cv_adj$NA19101.r1$log10cv2_adj,
                  ENSG_cv_adj$NA19101.r2$log10cv2_adj,
                  ENSG_cv_adj$NA19101.r3$log10cv2_adj )
  df_ind <- (df_ind - rowMeans(df_ind))/sqrt(rowVars(df_ind))

  }
  if (ii_individual == 3) {
  df_ind <- cbind(ENSG_cv_adj$NA19239.r1$log10cv2_adj,
                  ENSG_cv_adj$NA19239.r2$log10cv2_adj,
                  ENSG_cv_adj$NA19239.r3$log10cv2_adj )
  df_ind <- (df_ind - rowMeans(df_ind))/sqrt(rowVars(df_ind))
  }
  pca_results <- prcomp( t(df_ind), retx = TRUE, center = TRUE, scale = TRUE )
  prop_var <- (pca_results$sdev^2) / sum(pca_results$sdev^2)
  
  list(loadings = pca_results$x,
       prop_var = prop_var)
})


sapply(pca_individuals, function(xx) round(xx$prop_var,2) )

pca_all <- cbind(pca_individuals[[1]]$loadings[ , 1],
                 pca_individuals[[2]]$loadings[ , 1],
                 pca_individuals[[3]]$loadings[ , 1] )

friedman_results <- friedman.test(pca_all)
friedman_results
```




* Subsampling (NOT RUN).

```{r, eval = FALSE}
source("../code/cv-functions.r")

sizes <- c(10, 20, 30, 40, 60, 70, 80, 90, 100)
subsamples_compare <- lapply(sizes, function(per_size) {
  per_size_test <- sapply(1:100, function(per_sample) {
      subsample_obj <- make_subsample(counts = molecules_ENSG,
                                      anno = anno_qc_filter,
                                      subsample_size = per_size,
                                      number_subsample = 1) 
      
      subsample_cv <- compute_cv(counts = subsample_obj$counts_subsample,
                                 anno = subsample_obj$anno_subsample)
      
      subsample_cv_adj <- normalize_cv(batch_cv = subsample_cv, 
                                       counts = subsample_obj$counts_subsample, 
                                       anno = subsample_obj$anno_subsample)
      df_compare <- do.call(cbind, lapply(subsample_cv_adj, "[[", 7) )
      res <- friedman.test(df_compare)
      res$p.value
  })
  per_size_test
})
names(subsamples_compare) <- sizes
```



### Between individual, unadjusted CV

* PCA for each individual


```{r}
pca_individuals <- lapply(1:3, function(ii_individual) {
  if (ii_individual == 1) {
  df_ind <- log10(cbind(ENSG_cv_adj$NA19098.r1$cv,
                        ENSG_cv_adj$NA19098.r3$cv )^2 ) 
  }
  if (ii_individual == 2) {
  df_ind <- log10(cbind(ENSG_cv_adj$NA19101.r1$cv,
                        ENSG_cv_adj$NA19101.r2$cv,
                        ENSG_cv_adj$NA19101.r3$cv )^2)
  }
  if (ii_individual == 3) {
  df_ind <- log10(cbind(ENSG_cv_adj$NA19239.r1$cv,
                        ENSG_cv_adj$NA19239.r2$cv,
                        ENSG_cv_adj$NA19239.r3$cv )^2 )
  }
  pca_results <- prcomp( df_ind, retx = TRUE, center = TRUE, scale = TRUE)
  prop_var <- (pca_results$sdev^2) / sum(pca_results$sdev^2)
  
  list(scores = pca_results,
       loadings = pca_results$x,
       prop_var = prop_var)
})

sapply(pca_individuals, function(xx) xx$prop_var)

pca_all <- cbind(pca_individuals[[1]]$loadings[ , 1],
                 pca_individuals[[2]]$loadings[ , 1],
                 pca_individuals[[3]]$loadings[ , 1] )
                 
friedman_results <- friedman.test(pca_all)
friedman_results
```


* Compute CV across replicates

First we make PCA of each individual samples to evaluate whether replicate is a main source of variation. If replicate is not a main source of variation, we will combine the replicates to perform between indvidual comparisons.

```{r}
temp <- molecules_ENSG[, anno_qc_filter$individual == "NA19098"]
anno_temp <- anno_qc_filter[anno_qc_filter$individual == "NA19098",]
pca_temp <- prcomp( t(temp - rowMeans(temp)) )
(pca_temp$sdev^2)[1:3]/sum(pca_temp$sdev^2)
xy <- cbind(pca_temp$x[,1], pca_temp$x[,2] )

ggplot(data.frame(pc1 = pca_temp$x[,1],
                  pc2 = pca_temp$x[,2],
                  replicate = anno_temp$replicate),
       aes(x = pc1, y = pc2, col = replicate)) +
  geom_point() +
  ggtitle("NA19098")
par(mfrow = c(1,2))
plot(x = rowMeans(temp), y = pca_temp$rotation[ ,1],
     xlab = "mean molecule counts", ylab = "first PC loadings")
plot(x = rowMeans(temp), y = pca_temp$rotation[ ,2],
     xlab = "mean molecule counts", ylab = "second PC loadings")
title(main = "NA19098", outer = TRUE, line = -1)
cor(rowMeans(temp), pca_temp$rotation[ ,1], method = "spearman")
```


```{r}
temp <- molecules_ENSG[, anno_qc_filter$individual == "NA19101"]
anno_temp <- anno_qc_filter[anno_qc_filter$individual == "NA19101",]
pca_temp <- prcomp( t(temp - rowMeans(temp)))
(pca_temp$sdev^2)[1:3]/sum(pca_temp$sdev^2)
xy <- cbind(pca_temp$x[,1], pca_temp$x[,2] )

ggplot(data.frame(pc1 = pca_temp$x[,1],
                  pc2 = pca_temp$x[,2],
                  replicate = anno_temp$replicate),
       aes(x = pc1, y = pc2, col = replicate)) +
  geom_point() +
  ggtitle("NA19101")

par(mfrow = c(1,2))
plot(x = rowMeans(temp), y = pca_temp$rotation[ ,1],
     xlab = "mean molecule counts", ylab = "first PC loadings")
plot(x = rowMeans(temp), y = pca_temp$rotation[ ,2],
     xlab = "mean molecule counts", ylab = "second PC loadings")
title(main = "NA19101", outer = TRUE)
cor(rowMeans(temp), pca_temp$rotation[ ,1], method = "spearman")
```



```{r}
temp <- molecules_ENSG[, anno_qc_filter$individual == "NA19239"]
anno_temp <- anno_qc_filter[anno_qc_filter$individual == "NA19239",]
pca_temp <- prcomp( t(temp - rowMeans(temp)), retx = TRUE, center = TRUE, scale = TRUE )
(pca_temp$sdev^2)[1:3]/sum(pca_temp$sdev^2)

xy <- cbind(pca_temp$x[,1], pca_temp$x[,2] )

ggplot(data.frame(pc1 = pca_temp$x[,1],
                  pc2 = pca_temp$x[,2],
                  replicate = anno_temp$replicate),
       aes(x = pc1, y = pc2, col = replicate)) +
  geom_point() +
  ggtitle("NA19239")

par(mfrow = c(1,2))
plot(x = rowMeans(temp), y = pca_temp$rotation[ ,1],
     xlab = "mean molecule counts", ylab = "first PC loadings")
plot(x = rowMeans(temp), y = pca_temp$rotation[ ,2],
     xlab = "mean molecule counts", ylab = "second PC loadings")
title(main = "NA19239", outer = TRUE)
```


Compute CV across replicates.

```{r}
cv_individual <- lapply(1:3, function(ii_individual) {
    individuals <- unique(anno_qc_filter$individual)
    counts <- molecules_ENSG[ , anno_qc_filter$individual == individuals [ii_individual]]
    means <- apply(counts, 1, mean)
    sds <- apply(counts, 1, sd)
    cv <- sds/means
    return(cv)
})
names(cv_individual) <- unique(anno_qc_filter$individual)
cv_individual <- do.call(cbind, cv_individual)
friedman.test(cv_individual)
```




## Session information

```{r info}
sessionInfo()
```
