<!DOCTYPE html>

<html xmlns="http://www.w3.org/1999/xhtml">

<head>

<meta charset="utf-8">
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="generator" content="pandoc" />

<meta name="author" content="Joyce Hsiao" />

<meta name="date" content="2016-07-08" />

<title>Cell-to-cell variation analysis: noisy genes among all cells</title>

<script src="libs/jquery-1.11.0/jquery.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="libs/bootstrap-3.3.1/css/united.min.css" rel="stylesheet" />
<script src="libs/bootstrap-3.3.1/js/bootstrap.min.js"></script>
<script src="libs/bootstrap-3.3.1/shim/html5shiv.min.js"></script>
<script src="libs/bootstrap-3.3.1/shim/respond.min.js"></script>

<style type="text/css">

/* padding for bootstrap navbar */
body {
  padding-top: 50px;
  padding-bottom: 40px;
}


/* offset scroll position for anchor links (for fixed navbar)  */
.section h2 {
  padding-top: 55px;
  margin-top: -55px;
}
.section h3 {
  padding-top: 55px;
  margin-top: -55px;
}



/* don't use link color in navbar */
.dropdown-menu>li>a {
  color: black;
}

/* some padding for disqus */
#disqus_thread {
  margin-top: 45px;
}

</style>

<link rel="stylesheet" href="libs/font-awesome-4.1.0/css/font-awesome.min.css"/>

<style type="text/css">code{white-space: pre;}</style>
<link rel="stylesheet"
      href="libs/highlight/textmate.css"
      type="text/css" />
<script src="libs/highlight/highlight.js"></script>
<style type="text/css">
  pre:not([class]) {
    background-color: white;
  }
</style>
<script type="text/javascript">
if (window.hljs && document.readyState && document.readyState === "complete") {
   window.setTimeout(function() {
      hljs.initHighlighting();
   }, 0);
}
</script>



</head>

<body>

<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
code {
  color: inherit;
  background-color: rgba(0, 0, 0, 0.04);
}
img { 
  max-width:100%; 
  height: auto; 
}
</style>
<div class="container-fluid main-container">


<div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">singleCellSeq</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li><a href="index.html">Home</a></li>
        <li><a href="about.html">About</a></li>
        <li><a href="license.html">License</a></li>
        <li><a href="https://github.com/jdblischak/singleCellSeq">GitHub</a></li>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<div id="header">
<h1 class="title">Cell-to-cell variation analysis: noisy genes among all cells</h1>
<h4 class="author"><em>Joyce Hsiao</em></h4>
<h4 class="date"><em>2016-07-08</em></h4>
</div>

<div id="TOC">
<ul>
<li><a href="#background-and-some-observations">Background and some observations</a></li>
<li><a href="#set-up">Set up</a></li>
<li><a href="#prepare-data">Prepare data</a></li>
<li><a href="#compute-cv-and-adjusted-cv-values">Compute CV and adjusted CV values</a><ul>
<li><a href="#sanity-check-plots.">Sanity-check plots.</a></li>
<li><a href="#check-pluripotency-and-cell-cycle">Check pluripotency and cell-cycle</a></li>
</ul></li>
<li><a href="#top-ranked-dm-genes">Top ranked DM genes</a><ul>
<li><a href="#venn-diagram-1000-genes">Venn diagram, 1000 genes</a></li>
<li><a href="#venn-diagram-100-genes">Venn diagram, 100 genes</a></li>
<li><a href="#go-analysis---individual-top-genes">GO analysis - individual top genes</a></li>
<li><a href="#go-analysis---shared-noisy-genes">GO analysis - shared noisy genes</a></li>
</ul></li>
<li><a href="#differential-dms">Differential DMs</a><ul>
<li><a href="#top-100-ranked-genes-by-mad.">Top 100 ranked genes by MAD.</a></li>
</ul></li>
<li><a href="#session-information">Session information</a></li>
</ul>
</div>

<p><strong>Last updated:</strong> 2016-07-08</p>
<p><strong>Code version:</strong> f3271c626c586079652907b55fa357b6f569fd26</p>
<div id="background-and-some-observations" class="section level2">
<h2>Background and some observations</h2>
<p>Preliminary investigation into expression noise at the transcriptional level. We looked at expression noise across all the cells, including the cells that were not detected (i.e., count = 0) in the experiment. The investigatino begins with coefficeint of variation - a popular measure for quantifying variation in the data.</p>
<ol style="list-style-type: decimal">
<li><p>Computation of CV: We were interested in the variation in our batch-corrected data. But since these data are log2-transformed counts. We take 2 to the power of log-transformed counts and then compute the CV based on these corrected counts.</p></li>
<li><p>Corrected CV: Because in sequencing data, CVs are confounded with abundance levels, we performed a calculation that transformed each CV into a measure of deviation from the median of the CVs from the genes of similar abudance levels.</p></li>
</ol>
</div>
<div id="set-up" class="section level2">
<h2>Set up</h2>
<pre class="r"><code>library(&quot;data.table&quot;)
library(&quot;dplyr&quot;)
library(&quot;limma&quot;)
library(&quot;edgeR&quot;)
library(&quot;ggplot2&quot;)
library(&quot;grid&quot;)
theme_set(theme_bw(base_size = 12))
source(&quot;functions.R&quot;)
library(&quot;Humanzee&quot;)
library(&quot;cowplot&quot;)
library(&quot;MASS&quot;)
library(&quot;matrixStats&quot;)
library(&quot;mygene&quot;)</code></pre>
</div>
<div id="prepare-data" class="section level2">
<h2>Prepare data</h2>
<p>Input annotation of only QC-filtered single cells, with NA19098.r2 removed.</p>
<pre class="r"><code>anno_filter &lt;- read.table(&quot;../data/annotation-filter.txt&quot;, 
                      header = TRUE,
                      stringsAsFactors = FALSE)
dim(anno_filter)</code></pre>
<pre><code>[1] 564   5</code></pre>
<pre class="r"><code>head(anno_filter, 2)</code></pre>
<pre><code>  individual replicate well      batch      sample_id
1    NA19098        r1  A01 NA19098.r1 NA19098.r1.A01
2    NA19098        r1  A02 NA19098.r1 NA19098.r1.A02</code></pre>
<p>Import molecule counts after filtering and before any correction.</p>
<pre class="r"><code>molecules_filter &lt;- read.table(&quot;../data/molecules-filter.txt&quot;,
                               header = TRUE, stringsAsFactors = FALSE)
stopifnot(NROW(anno_filter) == NCOL(molecules_filter))</code></pre>
<p>Import final processed molecule counts of endogeneous genes.</p>
<pre class="r"><code>molecules_final &lt;- read.table(&quot;../data/molecules-final.txt&quot;, 
                             header = TRUE, stringsAsFactors = FALSE)
dim(molecules_final)</code></pre>
<pre><code>[1] 13058   564</code></pre>
<pre class="r"><code>stopifnot(NROW(anno_filter) == NCOL(molecules_final))</code></pre>
<p>Import gene symbols.</p>
<pre class="r"><code>gene_symbols &lt;- read.table(file = &quot;../data/gene-info.txt&quot;, sep = &quot;\t&quot;,
                           header = TRUE, stringsAsFactors = FALSE, quote = &quot;&quot;)</code></pre>
<p>Import cell cycle and pluripotency genes.</p>
<pre class="r"><code>cell_cycle_genes &lt;- read.table(&quot;../data/cellcyclegenes.txt&quot;,
                               header = TRUE, sep = &quot;\t&quot;,
                               stringsAsFactors = FALSE)

pluripotency_genes &lt;- read.table(&quot;../data/pluripotency-genes.txt&quot;,
                               header = TRUE, sep = &quot;\t&quot;,
                               stringsAsFactors = FALSE)$To</code></pre>
</div>
<div id="compute-cv-and-adjusted-cv-values" class="section level2">
<h2>Compute CV and adjusted CV values</h2>
<ol style="list-style-type: decimal">
<li>Compute Squared Coefficients of Variation across cells for each individual;</li>
<li>Adjust Squared CVs for confounding effect with the mean:
<ul>
<li>Compute rolling medians of gene expression levels,</li>
<li>Compute Squared CVs corresponding to rolling medians of gene expression levels, deviation of adjusted CVs.</li>
</ul></li>
</ol>
<div id="sanity-check-plots." class="section level3">
<h3>Sanity-check plots.</h3>
<p>Supplemenetal for the Methods section.</p>
<pre class="r"><code>theme_set(theme_bw(base_size = 8))
plot_grid(
  ggplot(data.frame(log10cv_1 = log10(ENSG_cv_adj$NA19098$cv^2),
                    log10cv_2 = log10(ENSG_cv_adj$NA19101$cv^2)),
       aes(x = log10cv_1, y = log10cv_2)) +
    geom_point(cex = .4) +
    xlab(&quot;NA19098 log10 squared-CV values&quot;) +
    ylab(&quot;NA19101 log10 squared-CV values&quot;) +
    ggtitle(&quot;Relationship between individual DM values&quot;) +
    theme(legend.position = &quot;none&quot;),
  ggplot(data.frame(dm1 = ENSG_cv_adj$NA19098$log10cv2_adj,
                    dm2 = ENSG_cv_adj$NA19101$log10cv2_adj),
       aes(x = dm1, y = dm2)) +
    geom_point(cex = .4) +
    xlab(&quot;NA19098 DM values&quot;) +
    ylab(&quot;NA19101 DM values&quot;) +
    ggtitle(&quot;Relationship between individual DM values&quot;) +
    theme(legend.position = &quot;none&quot;),
  ggplot(data.frame(dm = ENSG_cv_adj$NA19098$log10cv2_adj,
                    log10_mean = log10(ENSG_cv_adj$NA19098$mean)),
         aes(x = log10_mean, y = dm)) +
      geom_point(cex = .4) +
      xlab(&quot;log10 average molecule count&quot;) +
      ylab(&quot;DM values&quot;) +
      ggtitle(&quot;NA19098&quot;) +
      theme(legend.position = &quot;none&quot;),
    ggplot(data.frame(dm = ENSG_cv_adj$NA19101$log10cv2_adj,
                    log10_mean = log10(ENSG_cv_adj$NA19101$mean)),
         aes(x = log10_mean, y = dm)) +
      geom_point(cex = .4) +
      xlab(&quot;log10 average molecule count&quot;) +
      ylab(&quot;DM values&quot;) +
      ggtitle(&quot;NA19101&quot;) +
      theme(legend.position = &quot;none&quot;),
  ggplot(data.frame(dm = ENSG_cv_adj$NA19239$log10cv2_adj,
                    log10_mean = log10(ENSG_cv_adj$NA19239$mean)),
         aes(x = log10_mean, y = dm)) +
      geom_point(cex = .4) +
      xlab(&quot;log10 average molecule count&quot;) +
      ylab(&quot;DM values&quot;) +
      ggtitle(&quot;NA19239&quot;) +
      theme(legend.position = &quot;none&quot;),
  ncol = 2,
  labels = LETTERS[1:5] )</code></pre>
<p><img src="figure/cv-adjusted-summary-pois.Rmd/sanity-plots-dm-1.png" title="" alt="" width="768" style="display: block; margin: auto;" /></p>
</div>
<div id="check-pluripotency-and-cell-cycle" class="section level3">
<h3>Check pluripotency and cell-cycle</h3>
<p>Mark cell-cycle genes.</p>
<pre class="r"><code>genes &lt;- rownames(ENSG_cv[[1]])
ii_cellcycle_genes &lt;- lapply(1:3, function(per_individual) {
  genes %in% unlist(cell_cycle_genes)
})
names(ii_cellcycle_genes) &lt;- names(ENSG_cv)[1:3]
ii_cellcycle_genes &lt;- do.call(c, ii_cellcycle_genes)

ggplot(data.frame(do.call(rbind, ENSG_cv_adj[1:3]),
                  dm = c(ENSG_cv_adj$NA19098$log10cv2_adj, 
                         ENSG_cv_adj$NA19101$log10cv2_adj, 
                         ENSG_cv_adj$NA19239$log10cv2_adj) ),
       aes(x = log10(mean), y = dm )) +
  geom_point(aes(col = group), cex = 1.2) + facet_wrap( ~ group) +
  ggtitle(&quot;Cell-cycle genes&quot;) + 
  geom_point(
      data = subset(data.frame(do.call(rbind, ENSG_cv_adj[1:3]),
                              dm = c(ENSG_cv_adj$NA19098$log10cv2_adj, 
                                     ENSG_cv_adj$NA19101$log10cv2_adj, 
                                     ENSG_cv_adj$NA19239$log10cv2_adj) ), 
                 ii_cellcycle_genes), 
       colour = &quot;grey20&quot;, cex = 1.2) +
  labs(x = &quot;log10 average gene expression level&quot;,
       y = &quot;DM values&quot;)</code></pre>
<p><img src="figure/cv-adjusted-summary-pois.Rmd/unnamed-chunk-2-1.png" title="" alt="" width="672" style="display: block; margin: auto;" /></p>
<p>Mark pluripotent genes</p>
<pre class="r"><code>ii_pluripotent_genes &lt;- lapply(1:3, function(per_individual) {
  genes %in% unlist(pluripotency_genes)
})
names(ii_pluripotent_genes) &lt;- names(ENSG_cv)[1:3]
ii_pluripotent_genes &lt;- do.call(c, ii_pluripotent_genes)

ggplot(data.frame(do.call(rbind, ENSG_cv_adj[1:3]),
                  dm = c(ENSG_cv_adj$NA19098$log10cv2_adj, 
                         ENSG_cv_adj$NA19101$log10cv2_adj, 
                         ENSG_cv_adj$NA19239$log10cv2_adj) ),
       aes(x = log10(mean), y = dm )) +
  geom_point(aes(col = group), cex = 1.2) + facet_wrap( ~ group) +
  ggtitle(&quot;Pluripotent genes&quot;) + 
  geom_point(
      data = subset(data.frame(do.call(rbind, ENSG_cv_adj[1:3]),
                              dm = c(ENSG_cv_adj$NA19098$log10cv2_adj, 
                                     ENSG_cv_adj$NA19101$log10cv2_adj, 
                                     ENSG_cv_adj$NA19239$log10cv2_adj) ), 
                 ii_pluripotent_genes), 
       colour = &quot;grey20&quot;, cex = 1.2) +
  labs(x = &quot;log10 average gene expression level&quot;,
       y = &quot;DM values&quot;)</code></pre>
<p><img src="figure/cv-adjusted-summary-pois.Rmd/unnamed-chunk-3-1.png" title="" alt="" width="672" style="display: block; margin: auto;" /></p>
</div>
</div>
<div id="top-ranked-dm-genes" class="section level2">
<h2>Top ranked DM genes</h2>
<div id="venn-diagram-1000-genes" class="section level3">
<h3>Venn diagram, 1000 genes</h3>
<p>Top 1000 genes based on DM.</p>
<pre class="r"><code>genes &lt;- rownames(ENSG_cv[[1]])
library(gplots)
venn_cv_rank &lt;- gplots::venn(
  list(NA19098 = genes[ which( rank(ENSG_cv_adj$NA19098$log10cv2_adj) 
                               &gt; length(genes) - 1000 ) ],
       NA19101 = genes[ which( rank(ENSG_cv_adj$NA19101$log10cv2_adj) 
                               &gt; length(genes) - 1000 ) ],
       NA19239 = genes[ which( rank(ENSG_cv_adj$NA19239$log10cv2_adj) 
                               &gt; length(genes) - 1000 ) ] ))</code></pre>
<p><img src="figure/cv-adjusted-summary-pois.Rmd/venn-top-1000-dm-1.png" title="" alt="" width="672" style="display: block; margin: auto;" /></p>
<p>Bottom 1000 genes based on DM.</p>
<pre class="r"><code>genes &lt;- rownames(ENSG_cv[[1]])
library(gplots)
gplots::venn(
  list(NA19098 = genes[ which( rank(ENSG_cv_adj$NA19098$log10cv2_adj) 
                               &lt;= 1000 ) ],
       NA19101 = genes[ which( rank(ENSG_cv_adj$NA19101$log10cv2_adj) 
                               &lt;= 1000 ) ],
       NA19239 = genes[ which( rank(ENSG_cv_adj$NA19239$log10cv2_adj) 
                               &lt;= 1000 ) ] ))</code></pre>
<p><img src="figure/cv-adjusted-summary-pois.Rmd/venn-bottom-1000-dm-1.png" title="" alt="" width="672" style="display: block; margin: auto;" /></p>
<p>Top 1000 genes based on Means.</p>
<pre class="r"><code>genes &lt;- rownames(ENSG_cv[[1]])
library(gplots)
gplots::venn( 
  list(NA19098 = genes[ which(rank(ENSG_cv[[1]]$mean) &gt; length(genes) - 1000 ) ],
       NA19101 = genes[ which(rank(ENSG_cv[[2]]$mean) &gt; length(genes) - 1000 ) ],
       NA19239 = genes[ which(rank(ENSG_cv[[3]]$mean) &gt; length(genes) - 1000 ) ] ) )</code></pre>
<p><img src="figure/cv-adjusted-summary-pois.Rmd/venn-top-1000-means-1.png" title="" alt="" width="672" style="display: block; margin: auto;" /></p>
<p>Mark top ranked genes based on individual DM values.</p>
<pre class="r"><code>df_plot &lt;- data.frame(
  cvs = c(ENSG_cv_adj[[1]]$log10cv2_adj, ENSG_cv_adj[[2]]$log10cv2_adj,
          ENSG_cv_adj[[3]]$log10cv2_adj),
  means = c(ENSG_cv[[1]]$mean, ENSG_cv[[2]]$mean, ENSG_cv[[3]]$mean),
  individual = as.factor(rep(names(ENSG_cv)[1:3], each = NROW(ENSG_cv[[1]])) ) ) 

cowplot::plot_grid(
  ggplot( df_plot,
        aes(x = log10(means), y = cvs ) ) +
        geom_point( aes(col = as.factor(individual)), cex = 1.2 ) + 
        facet_wrap( ~ individual) +
        labs(x = &quot;log10 average gene expression level&quot;, 
             y = &quot;DM values&quot;) +
        geom_point(
          data = df_plot[ rep( rank(ENSG_cv_adj$NA19098$log10cv2_adj) 
                               &gt; length(genes) - 1000, 3), ],
          colour = &quot;grey20&quot;, cex = 1.2 ) +
        ggtitle(&quot;Top 1,000 genes in NA19098 based on DM values&quot;) +
        theme(legend.position = &quot;none&quot;),
  ggplot( df_plot,
        aes(x = log10(means), y = cvs ) ) +
        geom_point( aes(col = as.factor(individual)), cex = 1.2 ) + 
        facet_wrap( ~ individual) +
        labs(x = &quot;log10 average gene expression level&quot;, 
             y = &quot;DM values&quot;) +
        geom_point(
          data = df_plot[ rep( rank(ENSG_cv_adj$NA19101$log10cv2_adj) 
                               &gt; length(genes) - 1000, 3), ],
          colour = &quot;grey20&quot;, cex = 1.2 ) +
        ggtitle(&quot;Top 1,000 genes in NA19101 based on DM values&quot;) +
        theme(legend.position = &quot;none&quot;),
  ggplot( df_plot,
        aes(x = log10(means), y = cvs ) ) +
        geom_point( aes(col = as.factor(individual)), cex = 1.2 ) + 
        facet_wrap( ~ individual) +
        labs(x = &quot;log10 average gene expression level&quot;, 
             y = &quot;DM values&quot;) +
        geom_point(
          data = df_plot[ rep( rank(ENSG_cv_adj$NA19239$log10cv2_adj) 
                               &gt; length(genes) - 1000, 3), ],
          colour = &quot;grey20&quot;, cex = 1.2 ) +
        ggtitle(&quot;Top 1,000 genes in NA19239 based on DM values&quot;) +
        theme(legend.position = &quot;none&quot;),
  labels = LETTERS[1:4] )  </code></pre>
<p><img src="figure/cv-adjusted-summary-pois.Rmd/unnamed-chunk-4-1.png" title="" alt="" width="768" style="display: block; margin: auto;" /></p>
</div>
<div id="venn-diagram-100-genes" class="section level3">
<h3>Venn diagram, 100 genes</h3>
<p>Top 100 genes based on DM.</p>
<pre class="r"><code>genes &lt;- rownames(ENSG_cv[[1]])
library(gplots)
venn_cv_rank &lt;- gplots::venn(
  list(NA19098 = genes[ which( rank(ENSG_cv_adj$NA19098$log10cv2_adj) 
                               &gt; length(genes) - 100 ) ],
       NA19101 = genes[ which( rank(ENSG_cv_adj$NA19101$log10cv2_adj) 
                               &gt; length(genes) - 100 ) ],
       NA19239 = genes[ which( rank(ENSG_cv_adj$NA19239$log10cv2_adj) 
                               &gt; length(genes) - 100 ) ] ))</code></pre>
<p><img src="figure/cv-adjusted-summary-pois.Rmd/venn-top-100-dm-1.png" title="" alt="" width="672" style="display: block; margin: auto;" /></p>
<p>Bottom 100 genes based on DM.</p>
<pre class="r"><code>genes &lt;- rownames(ENSG_cv[[1]])
library(gplots)
gplots::venn(
  list(NA19098 = genes[ which( rank(ENSG_cv_adj$NA19098$log10cv2_adj) 
                               &lt;= 100 ) ],
       NA19101 = genes[ which( rank(ENSG_cv_adj$NA19101$log10cv2_adj) 
                               &lt;= 100 ) ],
       NA19239 = genes[ which( rank(ENSG_cv_adj$NA19239$log10cv2_adj) 
                               &lt;= 100 ) ] ))</code></pre>
<p><img src="figure/cv-adjusted-summary-pois.Rmd/venn-bottom-100-dm-1.png" title="" alt="" width="672" style="display: block; margin: auto;" /></p>
<p>Top 100 genes based on Means.</p>
<pre class="r"><code>genes &lt;- rownames(ENSG_cv[[1]])
library(gplots)
gplots::venn( 
  list(NA19098 = genes[ which(rank(ENSG_cv[[1]]$mean) &gt; length(genes) - 100 ) ],
       NA19101 = genes[ which(rank(ENSG_cv[[2]]$mean) &gt; length(genes) - 100 ) ],
       NA19239 = genes[ which(rank(ENSG_cv[[3]]$mean) &gt; length(genes) - 100 ) ] ) )</code></pre>
<p><img src="figure/cv-adjusted-summary-pois.Rmd/venn-top-100-means-1.png" title="" alt="" width="672" style="display: block; margin: auto;" /></p>
</div>
<div id="go-analysis---individual-top-genes" class="section level3">
<h3>GO analysis - individual top genes</h3>
<p>Analyze top 1,000 genes in DM values.</p>
<pre class="r"><code># output top 1000 genes in DM values
top1000DM_genes &lt;-
  data.frame(NA19098 = genes[ which( rank(ENSG_cv_adj$NA19098$log10cv2_adj) 
                               &gt; length(genes) - 1000 ) ],
       NA19101 = genes[ which( rank(ENSG_cv_adj$NA19101$log10cv2_adj) 
                               &gt; length(genes) - 1000 ) ],
       NA19239 = genes[ which( rank(ENSG_cv_adj$NA19239$log10cv2_adj) 
                               &gt; length(genes) - 1000 ) ],
       stringsAsFactors = FALSE)

# write the gene names out to a text file,
# then copy and paste to GO Consortium interface
for (ind in c(&quot;NA19098&quot;, &quot;NA19101&quot;, &quot;NA19239&quot;)) {
  write.table(top100DM_genes[[ind]],
              file = paste0(&quot;../data/top-100-dm-&quot;,ind,&quot;.txt&quot;),
              sep = &quot;/t&quot;, quote = FALSE,
              col.names = FALSE, row.names = FALSE)
}
for (ind in c(&quot;NA19098&quot;, &quot;NA19101&quot;, &quot;NA19239&quot;)) {
  write.table(top1000DM_genes[[ind]],
              file = paste0(&quot;../data/top-1000-dm-&quot;,ind,&quot;.txt&quot;),
              sep = &quot;/t&quot;, quote = FALSE,
              col.names = FALSE, row.names = FALSE)
}
write.table(rownames(molecules_final),
            file = &quot;../data/gene-names.txt&quot;,
            sep = &quot;/t&quot;, quote = FALSE,
            col.names = FALSE, row.names = FALSE)</code></pre>
<div id="go-consortium" class="section level4">
<h4>GO consortium</h4>
<p>We found no significant GO terms in PANTHER database.</p>
</div>
<div id="gostats" class="section level4">
<h4>GOStats</h4>
<p>Not yet figured out how to get genes associated with each sig. GO terms.</p>
<p><em>[Chunk not evaluated]</em></p>
<pre class="r"><code>require(Humanzee)
# Find GO terms
go_top1000_results &lt;- lapply(c(&quot;NA19098&quot;, &quot;NA19101&quot;, &quot;NA19239&quot;),
    function(individual) {
        go_list &lt;- lapply(c(&quot;BP&quot;, &quot;MF&quot;, &quot;CC&quot;), function(term) {
              sig_gene_go &lt;- GOtest(rownames(molecules_final),
                                    top1000DM_genes[[individual]],
                                    ontology = term,
                                    conditional.method = F)
              
              
              go_data &lt;- goHeatmapData(list(summary(sig_gene_go$GO[[term]], 
                                                    pvalue = .01)))
              return(go_data)
        })
        names(go_list) &lt;- c(&quot;BP&quot;, &quot;MF&quot;, &quot;CC&quot;)
        return(go_list)      
})
names(go_top1000_results) &lt;- c(&quot;NA19098&quot;, &quot;NA19101&quot;, &quot;NA19239&quot;)

go_gene_list &lt;- sig_gene_go[[&quot;GO&quot;]][[1]]@goDag@nodeData@data
go_gene_list_names &lt;- names(go_gene_list)
go_sig &lt;- which(go_gene_list_names %in% go_data[,1])
go_gene_list_sig &lt;- go_gene_list[go_sig]
length(go_gene_list_sig)
go_gene_list_sig
head(go_data)
# Plot heatmap
go_heatmap &lt;- plotHeatmap(go_data, 
                          labCol = &quot;&quot;)</code></pre>
</div>
<div id="cpdb" class="section level4">
<h4>CPDB</h4>
<p>ConsensusPATHDB-Human was used to perform GO over-representation analysis.</p>
<p><em>NA19098</em></p>
<pre class="r"><code>go_NA19098 &lt;- 
  read.table(&quot;figure/cv-adjusted-summary-pois.Rmd/go-cpdb-top-1000-dm-NA19098.tab&quot;,
               sep = &quot;\t&quot;,
               header = TRUE)
go_NA19098_sig &lt;- go_NA19098[go_NA19098$q.value &lt; .1, ]
go_NA19098_sig$term_name</code></pre>
<pre><code> [1] response to oxygen levels                              
 [2] response to decreased oxygen levels                    
 [3] response to hypoxia                                    
 [4] sarcoplasmic reticulum                                 
 [5] very-low-density lipoprotein particle receptor activity
 [6] mononuclear cell migration                             
 [7] response to hyperoxia                                  
 [8] condensed nuclear chromosome, centromeric region       
 [9] condensed nuclear chromosome kinetochore               
[10] calcium ion binding                                    
[11] carbohydrate derivative transporter activity           
59 Levels: base-excision repair, AP site formation ...</code></pre>
<p><em>NA19101</em></p>
<pre class="r"><code>go_NA19101 &lt;- read.table(&quot;figure/cv-adjusted-summary-pois.Rmd/go-cpdb-top-1000-dm-NA19101.tab&quot;,
                         sep = &quot;\t&quot;,
                         header = TRUE)
head(go_NA19101$q.value)</code></pre>
<pre><code>[1] 0.008862498 0.015732966 0.033066707 0.142366905 0.142366905 0.138059112</code></pre>
<pre class="r"><code>go_NA19101_sig &lt;- go_NA19101[go_NA19101$q.value &lt; .1, ]
go_NA19101_sig$term_name</code></pre>
<pre><code>[1] mononuclear cell migration                 
[2] regulation of mononuclear cell migration   
[3] multicellular organismal response to stress
[4] glutathione derivative metabolic process   
56 Levels: acid-amino acid ligase activity ... xenobiotic metabolic process</code></pre>
<p><em>NA19239</em></p>
<pre class="r"><code>go_NA19239 &lt;- read.table(&quot;figure/cv-adjusted-summary-pois.Rmd/go-cpdb-top-1000-dm-NA19239.tab&quot;,
           sep = &quot;\t&quot;,
           header = TRUE)
go_NA19239_sig &lt;- go_NA19239[go_NA19239$q.value &lt; .1, ]

go_NA19239_sig$term_name</code></pre>
<pre><code> [1] extracellular space                                              
 [2] fibronectin binding                                              
 [3] calcium ion binding                                              
 [4] heparin binding                                                  
 [5] basement membrane                                                
 [6] proteinaceous extracellular matrix                               
 [7] sulfur compound binding                                          
 [8] response to zinc ion                                             
 [9] plasma membrane region                                           
[10] extracellular matrix disassembly                                 
[11] cell adhesion                                                    
[12] glycosaminoglycan binding                                        
[13] cell projection membrane                                         
[14] regulation of growth                                             
[15] intrinsic component of membrane                                  
[16] regulation of multicellular organismal process                   
[17] response to transition metal nanoparticle                        
[18] cytoskeletal protein binding                                     
[19] actin binding                                                    
[20] anatomical structure morphogenesis                               
[21] integral component of membrane                                   
[22] organ development                                                
[23] cartilage development                                            
[24] hormone metabolic process                                        
[25] system process                                                   
[26] wound healing                                                    
[27] multi-multicellular organism process                             
[28] developmental cell growth                                        
[29] negative regulation of growth                                    
[30] collagen catabolic process                                       
[31] multicellular organismal catabolic process                       
[32] laminin-10 complex                                               
[33] female pregnancy                                                 
[34] retinoic acid binding                                            
[35] connective tissue development                                    
[36] tissue development                                               
[37] plasma membrane                                                  
[38] lipid binding                                                    
[39] cell periphery                                                   
[40] extracellular structure organization                             
[41] membrane region                                                  
[42] developmental growth                                             
[43] isoprenoid metabolic process                                     
[44] apical plasma membrane                                           
[45] embryo implantation                                              
[46] response to alcohol                                              
[47] single-multicellular organism process                            
[48] sodium:potassium-exchanging ATPase complex                       
[49] carboxylic acid binding                                          
[50] cell growth                                                      
[51] plasma membrane part                                             
[52] hormone activity                                                 
[53] organic acid binding                                             
[54] developmental growth involved in morphogenesis                   
[55] detection of external stimulus                                   
[56] cell-cell junction                                               
[57] response to wounding                                             
[58] stereocilium                                                     
[59] proteoglycan binding                                             
[60] fibrillar collagen trimer                                        
[61] retinoid binding                                                 
[62] banded collagen fibril                                           
[63] actin filament-based process                                     
[64] detection of abiotic stimulus                                    
[65] regulation of developmental process                              
[66] basal lamina                                                     
[67] extracellular matrix binding                                     
[68] cartilage development involved in endochondral bone morphogenesis
[69] transcription factor activity, sequence-specific DNA binding     
[70] metal ion binding                                                
[71] cell migration                                                   
[72] cation binding                                                   
[73] growth factor binding                                            
[74] retinol binding                                                  
[75] glycosphingolipid binding                                        
[76] apical part of cell                                              
[77] collagen binding                                                 
[78] phosphatidylserine binding                                       
[79] insulin-like growth factor binding                               
[80] isoprenoid binding                                               
[81] insulin-like growth factor binding protein complex               
[82] phospholipid binding                                             
[83] extracellular matrix structural constituent                      
[84] single-organism developmental process                            
[85] cell motility                                                    
[86] localization of cell                                             
[87] laminin complex                                                  
[88] protein-lipid complex binding                                    
[89] apolipoprotein binding                                           
[90] laminin binding                                                  
[91] regulation of biological quality                                 
166 Levels: actin binding ... wound healing</code></pre>
<p>*Why is there so many more GO terms in NA19239 than the other two individuals?</p>
<p><em>[Chunk not evaluated]</em></p>
<pre class="r"><code>top1000DM_genes &lt;-
  data.frame(NA19098 = genes[ which( rank(ENSG_cv_adj$NA19098$log10cv2_adj) 
                               &gt; length(genes) - 1000 ) ],
       NA19101 = genes[ which( rank(ENSG_cv_adj$NA19101$log10cv2_adj) 
                               &gt; length(genes) - 1000 ) ],
       NA19239 = genes[ which( rank(ENSG_cv_adj$NA19239$log10cv2_adj) 
                               &gt; length(genes) - 1000 ) ],
       stringsAsFactors = FALSE)

par(mfrow = c(2,2))
boxplot(cbind(ENSG_cv_adj$NA19101$log10cv2_adj[which( rank(ENSG_cv_adj$NA19101$log10cv2_adj) 
                               &gt; length(genes) - 1000 )],
              ENSG_cv_adj$NA19239$log10cv2_adj[which( rank(ENSG_cv_adj$NA19239$log10cv2_adj) 
                               &gt; length(genes) - 1000 )]) )
title(main = &quot;adjusted CV&quot;)

boxplot(
  cbind(ENSG_cv_adj$NA19101$mean[ which( rank(ENSG_cv_adj$NA19101$log10cv2_adj)
                                        &gt; length(genes) - 1000 )],
        ENSG_cv_adj$NA19239$mean[which( rank(ENSG_cv_adj$NA19239$log10cv2_adj) 
                               &gt; length(genes) - 1000 )]) )</code></pre>
</div>
</div>
<div id="go-analysis---shared-noisy-genes" class="section level3">
<h3>GO analysis - shared noisy genes</h3>
<pre class="r"><code>shared_venn &lt;- gplots::venn(
  list(NA19098 = genes[ which( rank(ENSG_cv_adj$NA19098$log10cv2_adj) 
                               &gt; length(genes) - 1000 ) ],
       NA19101 = genes[ which( rank(ENSG_cv_adj$NA19101$log10cv2_adj) 
                               &gt; length(genes) - 1000 ) ],
       NA19239 = genes[ which( rank(ENSG_cv_adj$NA19239$log10cv2_adj) 
                               &gt; length(genes) - 1000 ) ] ) )

shared_genes &lt;- attr(shared_venn, &quot;intersections&quot;)[[&#39;111&#39;]]

# write.table(shared_genes,
#             file = (&quot;../data/top-1000-dm-shared.txt&quot;),
#             sep = &quot;/t&quot;, quote = FALSE,
#            col.names = FALSE, row.names = FALSE)</code></pre>
<p>ConsensusPATHDB-Human was used to perform GO over-representation analysis.</p>
<p>Of the 120 noisy genes that are common to all three individual cell lines, 38 of the genes are over-represented in the following functions:</p>
<pre class="r"><code>go_top_shared &lt;- read.table(&quot;figure/cv-adjusted-summary-pois.Rmd/go-cpdb-top-1000-shared.tab&quot;,
                         sep = &quot;\t&quot;,
                         header = TRUE)
go_top_shared &lt;- go_top_shared[go_top_shared$q.value &lt; .1, ]
go_top_shared$term_name</code></pre>
<pre><code> [1] inner mitochondrial membrane protein complex                                                                    
 [2] respiratory electron transport chain                                                                            
 [3] electron transport chain                                                                                        
 [4] nucleoside triphosphate metabolic process                                                                       
 [5] nucleoside monophosphate metabolic process                                                                      
 [6] mitochondrial protein complex                                                                                   
 [7] respiratory chain complex                                                                                       
 [8] condensed nuclear chromosome, centromeric region                                                                
 [9] condensed nuclear chromosome outer kinetochore                                                                  
[10] response to erythropoietin                                                                                      
[11] mitochondrial respiratory chain                                                                                 
[12] response to transition metal nanoparticle                                                                       
[13] NADH dehydrogenase (quinone) activity                                                                           
[14] respiratory chain                                                                                               
[15] response to hyperoxia                                                                                           
[16] response to increased oxygen levels                                                                             
[17] NADH dehydrogenase activity                                                                                     
[18] mitochondrial respiratory chain complex I                                                                       
[19] NADH dehydrogenase complex                                                                                      
[20] respiratory chain complex I                                                                                     
[21] oxidative phosphorylation                                                                                       
[22] purine nucleoside metabolic process                                                                             
[23] mitochondrial membrane part                                                                                     
[24] response to zinc ion                                                                                            
[25] cellular respiration                                                                                            
[26] oxidoreductase complex                                                                                          
[27] ribonucleoside metabolic process                                                                                
[28] oxidoreductase activity, acting on NAD(P)H, quinone or similar compound as acceptor                             
[29] vascular smooth muscle cell development                                                                         
[30] condensed nuclear chromosome kinetochore                                                                        
[31] condensed nuclear chromosome                                                                                    
[32] phospholipase inhibitor activity                                                                                
[33] DNA binding, bending                                                                                            
[34] cellular response to lipid                                                                                      
[35] cardiac muscle cell development                                                                                 
[36] response to cadmium ion                                                                                         
[37] lipase inhibitor activity                                                                                       
[38] cardiocyte differentiation                                                                                      
[39] histone-serine phosphorylation                                                                                  
[40] positive regulation of transcription from RNA polymerase II promoter in response to endoplasmic reticulum stress
[41] ribonucleotide metabolic process                                                                                
[42] purine nucleotide metabolic process                                                                             
[43] regulation of mitochondrial depolarization                                                                      
[44] glutathione biosynthetic process                                                                                
[45] synaptonemal complex organization                                                                               
[46] RNA replication                                                                                                 
[47] negative regulation of catabolic process                                                                        
[48] midbody                                                                                                         
[49] cellular response to metal ion                                                                                  
[50] meiotic chromosome segregation                                                                                  
[51] actin-mediated cell contraction                                                                                 
[52] condensed chromosome outer kinetochore                                                                          
[53] proton-transporting ATP synthase complex, coupling factor F(o)                                                  
[54] acid-amino acid ligase activity                                                                                 
[55] regulation of glial cell proliferation                                                                          
[56] NLS-bearing protein import into nucleus                                                                         
[57] mitochondrial depolarization                                                                                    
[58] negative regulation of cellular component organization                                                          
[59] positive regulation of chromosome segregation                                                                   
78 Levels: acid-amino acid ligase activity ...</code></pre>
<pre class="r"><code>go_top_shared_genes &lt;- getGenes(gsub(&quot;;&quot;, &quot;,&quot;,
                    as.character(go_top_shared$members_input_overlap_geneids)))
go_top_shared_genes &lt;- go_top_shared_genes[!duplicated(go_top_shared_genes[ , &quot;symbol&quot;]), ]
kable(data.frame(symbol = go_top_shared_genes[ ,&quot;symbol&quot;],
                  name = go_top_shared_genes[,&quot;name&quot;]) )</code></pre>
<table>
<thead>
<tr class="header">
<th align="left">symbol</th>
<th align="left">name</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td align="left">ND3</td>
<td align="left">NADH dehydrogenase, subunit 3 (complex I)</td>
</tr>
<tr class="even">
<td align="left">CYTB</td>
<td align="left">cytochrome b</td>
</tr>
<tr class="odd">
<td align="left">ND5</td>
<td align="left">NADH dehydrogenase, subunit 5 (complex I)</td>
</tr>
<tr class="even">
<td align="left">ND4L</td>
<td align="left">NADH dehydrogenase, subunit 4L (complex I)</td>
</tr>
<tr class="odd">
<td align="left">ATP8</td>
<td align="left">ATP synthase F0 subunit 8</td>
</tr>
<tr class="even">
<td align="left">ND2</td>
<td align="left">MTND2</td>
</tr>
<tr class="odd">
<td align="left">ATP6</td>
<td align="left">ATP synthase F0 subunit 6</td>
</tr>
<tr class="even">
<td align="left">ATP1A2</td>
<td align="left">ATPase Na+/K+ transporting subunit alpha 2</td>
</tr>
<tr class="odd">
<td align="left">DDIT4</td>
<td align="left">DNA damage inducible transcript 4</td>
</tr>
<tr class="even">
<td align="left">PLK1</td>
<td align="left">polo like kinase 1</td>
</tr>
<tr class="odd">
<td align="left">AURKA</td>
<td align="left">aurora kinase A</td>
</tr>
<tr class="even">
<td align="left">CCNB1</td>
<td align="left">cyclin B1</td>
</tr>
<tr class="odd">
<td align="left">MT1X</td>
<td align="left">metallothionein 1X</td>
</tr>
<tr class="even">
<td align="left">MT2A</td>
<td align="left">metallothionein 2A</td>
</tr>
<tr class="odd">
<td align="left">TSPO</td>
<td align="left">translocator protein</td>
</tr>
<tr class="even">
<td align="left">ATG7</td>
<td align="left">autophagy related 7</td>
</tr>
<tr class="odd">
<td align="left">HES1</td>
<td align="left">hes family bHLH transcription factor 1</td>
</tr>
<tr class="even">
<td align="left">SGCB</td>
<td align="left">sarcoglycan beta</td>
</tr>
<tr class="odd">
<td align="left">P3H4</td>
<td align="left">prolyl 3-hydroxylase family member 4 (non-enzymatic)</td>
</tr>
<tr class="even">
<td align="left">ANXA3</td>
<td align="left">annexin A3</td>
</tr>
<tr class="odd">
<td align="left">ANXA2</td>
<td align="left">annexin A2</td>
</tr>
<tr class="even">
<td align="left">TRERF1</td>
<td align="left">transcriptional regulating factor 1</td>
</tr>
<tr class="odd">
<td align="left">TOP2A</td>
<td align="left">topoisomerase (DNA) II alpha</td>
</tr>
<tr class="even">
<td align="left">STC1</td>
<td align="left">stanniocalcin 1</td>
</tr>
<tr class="odd">
<td align="left">LITAF</td>
<td align="left">lipopolysaccharide induced TNF factor</td>
</tr>
<tr class="even">
<td align="left">LRP8</td>
<td align="left">LDL receptor related protein 8</td>
</tr>
<tr class="odd">
<td align="left">GSS</td>
<td align="left">glutathione synthetase</td>
</tr>
<tr class="even">
<td align="left">ATF6B</td>
<td align="left">activating transcription factor 6 beta</td>
</tr>
<tr class="odd">
<td align="left">HSPA5</td>
<td align="left">heat shock protein family A (Hsp70) member 5</td>
</tr>
<tr class="even">
<td align="left">GCLM</td>
<td align="left">glutamate-cysteine ligase modifier subunit</td>
</tr>
<tr class="odd">
<td align="left">UBE2C</td>
<td align="left">ubiquitin conjugating enzyme E2 C</td>
</tr>
<tr class="even">
<td align="left">CCDC124</td>
<td align="left">coiled-coil domain containing 124</td>
</tr>
<tr class="odd">
<td align="left">VIM</td>
<td align="left">vimentin</td>
</tr>
<tr class="even">
<td align="left">KPNA5</td>
<td align="left">karyopherin subunit alpha 5</td>
</tr>
<tr class="odd">
<td align="left">KPNA2</td>
<td align="left">karyopherin subunit alpha 2</td>
</tr>
<tr class="even">
<td align="left">PMEPA1</td>
<td align="left">prostate transmembrane protein, androgen induced 1</td>
</tr>
<tr class="odd">
<td align="left">LGALS3</td>
<td align="left">lectin, galactoside binding soluble 3</td>
</tr>
<tr class="even">
<td align="left">PFN1</td>
<td align="left">profilin 1</td>
</tr>
</tbody>
</table>
</div>
</div>
<div id="differential-dms" class="section level2">
<h2>Differential DMs</h2>
<p>Compute median of absolute deviations (MAD) to quantify dissimilarity of the individual DM meausres.</p>
<pre class="r"><code>library(matrixStats)
dm_matrix &lt;- as.matrix(
                data.frame(NA19098 = ENSG_cv_adj$NA19098$log10cv2_adj,
                           NA19101 = ENSG_cv_adj$NA19101$log10cv2_adj,
                           NA19239 = ENSG_cv_adj$NA19239$log10cv2_adj) )
mad &lt;- rowMedians( abs( dm_matrix - rowMedians(dm_matrix) ) )</code></pre>
<div id="top-100-ranked-genes-by-mad." class="section level3">
<h3>Top 100 ranked genes by MAD.</h3>
<p>ConsensusPATHDB-Human was used to perform GO over-representation analysis.</p>
<p>Look up top 100 genes in MAD values</p>
<pre class="r"><code>mad_genes &lt;- rownames(molecules_final)[rank(mad) &gt; 
                                             (length(mad) - 100) ]
#write.table(mad_genes,
#            file = &quot;../data/mad-genes.txt&quot;,
#            sep = &quot;\t&quot;, quote = FALSE,
#            col.names = FALSE, row.names = FALSE)</code></pre>
<pre class="r"><code>library(mygene)
go_top &lt;- read.table(&quot;figure/cv-adjusted-summary-pois.Rmd/go-cpdb-all-top.tab&quot;,
                         sep = &quot;\t&quot;,
                         header = TRUE)
go_top &lt;- go_top[go_top$q.value &lt; .1, ]
as.character(go_top$term_name)</code></pre>
<pre><code>[1] &quot;extracellular space&quot;           &quot;ErbB-2 class receptor binding&quot;</code></pre>
<pre class="r"><code># go_top_genes &lt;- getGenes(gsub(&quot;;&quot;, &quot;,&quot;,
#                     as.character(go_top$members_input_overlap_geneids)))
# go_top_genes &lt;- go_top_genes[!duplicated(go_top_genes[ , &quot;symbol&quot;]), ]
# kable(data.frame(symbol =go_top_genes[ ,&quot;symbol&quot;],
#                  name = go_top_genes[,&quot;name&quot;]) )</code></pre>
</div>
</div>
<div id="session-information" class="section level2">
<h2>Session information</h2>
<pre class="r"><code>sessionInfo()</code></pre>
<pre><code>R version 3.2.0 (2015-04-16)
Platform: x86_64-unknown-linux-gnu (64-bit)

locale:
 [1] LC_CTYPE=en_US.UTF-8       LC_NUMERIC=C              
 [3] LC_TIME=en_US.UTF-8        LC_COLLATE=en_US.UTF-8    
 [5] LC_MONETARY=en_US.UTF-8    LC_MESSAGES=en_US.UTF-8   
 [7] LC_PAPER=en_US.UTF-8       LC_NAME=C                 
 [9] LC_ADDRESS=C               LC_TELEPHONE=C            
[11] LC_MEASUREMENT=en_US.UTF-8 LC_IDENTIFICATION=C       

attached base packages:
 [1] stats4    parallel  grid      stats     graphics  grDevices utils    
 [8] datasets  methods   base     

other attached packages:
 [1] gplots_2.17.0          mygene_1.2.3           GenomicFeatures_1.20.1
 [4] AnnotationDbi_1.30.1   Biobase_2.28.0         GenomicRanges_1.20.5  
 [7] GenomeInfoDb_1.4.0     IRanges_2.2.4          S4Vectors_0.6.0       
[10] BiocGenerics_0.14.0    matrixStats_0.14.0     MASS_7.3-40           
[13] cowplot_0.3.1          Humanzee_0.1.0         ggplot2_1.0.1         
[16] edgeR_3.10.2           limma_3.24.9           dplyr_0.4.2           
[19] data.table_1.9.4       knitr_1.10.5          

loaded via a namespace (and not attached):
 [1] httr_0.6.1              jsonlite_0.9.16        
 [3] splines_3.2.0           gsubfn_0.6-6           
 [5] gtools_3.5.0            Formula_1.2-1          
 [7] assertthat_0.1          highr_0.5              
 [9] latticeExtra_0.6-26     Rsamtools_1.20.4       
[11] yaml_2.1.13             RSQLite_1.0.0          
[13] lattice_0.20-31         chron_2.3-45           
[15] digest_0.6.8            RColorBrewer_1.1-2     
[17] XVector_0.8.0           colorspace_1.2-6       
[19] htmltools_0.2.6         plyr_1.8.3             
[21] XML_3.98-1.2            biomaRt_2.24.0         
[23] zlibbioc_1.14.0         scales_0.4.0           
[25] gdata_2.16.1            BiocParallel_1.2.2     
[27] sqldf_0.4-10            nnet_7.3-9             
[29] proto_0.3-10            survival_2.38-1        
[31] magrittr_1.5            evaluate_0.7           
[33] foreign_0.8-63          tools_3.2.0            
[35] formatR_1.2             stringr_1.0.0          
[37] munsell_0.4.3           cluster_2.0.1          
[39] lambda.r_1.1.7          Biostrings_2.36.1      
[41] caTools_1.17.1          futile.logger_1.4.1    
[43] RCurl_1.95-4.6          bitops_1.0-6           
[45] labeling_0.3            rmarkdown_0.6.1        
[47] gtable_0.1.2            DBI_0.3.1              
[49] reshape2_1.4.1          R6_2.1.1               
[51] GenomicAlignments_1.4.1 gridExtra_2.0.0        
[53] rtracklayer_1.28.4      Hmisc_3.16-0           
[55] futile.options_1.0.0    KernSmooth_2.23-14     
[57] stringi_1.0-1           Rcpp_0.12.4            
[59] rpart_4.1-9             acepack_1.3-3.3        </code></pre>
</div>


<!-- some extra javascript for older browsers -->
<script type="text/javascript" src="libs/polyfill.js"></script>

<script>

// manage active state of menu based on current page
$(document).ready(function () {

    // active menu
    href = window.location.pathname
    href = href.substr(href.lastIndexOf('/') + 1)
    $('a[href="' + href + '"]').parent().addClass('active');

    // manage active menu header
    if (href.startsWith('authoring_'))
      $('a[href="' + 'authoring' + '"]').parent().addClass('active');
    else if (href.endsWith('_format.html'))
      $('a[href="' + 'formats' + '"]').parent().addClass('active');
    else if (href.startsWith('developer_'))
      $('a[href="' + 'developer' + '"]').parent().addClass('active');

});

</script>

</div>

<script>

// add bootstrap table styles to pandoc tables
$(document).ready(function () {
  $('tr.header').parent('thead').parent('table').addClass('table table-condensed');
});

</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://cdn.mathjax.org/mathjax/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
