<!DOCTYPE html>

<html xmlns="http://www.w3.org/1999/xhtml">

<head>

<meta charset="utf-8">
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="generator" content="pandoc" />

<meta name="author" content="Joyce Hsiao" />

<meta name="date" content="2015-10-25" />

<title>Compare transcriptome-wide profile of CVs</title>

<script src="libs/jquery-1.11.0/jquery.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="libs/bootstrap-3.3.1/css/united.min.css" rel="stylesheet" />
<script src="libs/bootstrap-3.3.1/js/bootstrap.min.js"></script>
<script src="libs/bootstrap-3.3.1/shim/html5shiv.min.js"></script>
<script src="libs/bootstrap-3.3.1/shim/respond.min.js"></script>

<style type="text/css">

/* padding for bootstrap navbar */
body {
  padding-top: 50px;
  padding-bottom: 40px;
}


/* offset scroll position for anchor links (for fixed navbar)  */
.section h2 {
  padding-top: 55px;
  margin-top: -55px;
}
.section h3 {
  padding-top: 55px;
  margin-top: -55px;
}



/* don't use link color in navbar */
.dropdown-menu>li>a {
  color: black;
}

/* some padding for disqus */
#disqus_thread {
  margin-top: 45px;
}

</style>

<link rel="stylesheet" href="libs/font-awesome-4.1.0/css/font-awesome.min.css"/>

<style type="text/css">code{white-space: pre;}</style>
<link rel="stylesheet"
      href="libs/highlight/textmate.css"
      type="text/css" />
<script src="libs/highlight/highlight.js"></script>
<style type="text/css">
  pre:not([class]) {
    background-color: white;
  }
</style>
<script type="text/javascript">
if (window.hljs && document.readyState && document.readyState === "complete") {
   window.setTimeout(function() {
      hljs.initHighlighting();
   }, 0);
}
</script>



</head>

<body>

<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
code {
  color: inherit;
  background-color: rgba(0, 0, 0, 0.04);
}
img { 
  max-width:100%; 
  height: auto; 
}
</style>
<div class="container-fluid main-container">


<div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">singleCellSeq</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li><a href="index.html">Home</a></li>
        <li><a href="about.html">About</a></li>
        <li><a href="license.html">License</a></li>
        <li><a href="https://github.com/jdblischak/singleCellSeq">GitHub</a></li>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<div id="header">
<h1 class="title">Compare transcriptome-wide profile of CVs</h1>
<h4 class="author"><em>Joyce Hsiao</em></h4>
<h4 class="date"><em>2015-10-25</em></h4>
</div>

<div id="TOC">
<ul>
<li><a href="#objective">Objective</a></li>
<li><a href="#set-up">Set up</a></li>
<li><a href="#prepare-data">Prepare data</a></li>
<li><a href="#compute-normalized-cv-for-each-batch">Compute normalized CV for each batch</a></li>
<li><a href="#compare-transcriptome-profile-within-individuals">Compare transcriptome profile within individuals</a><ul>
<li><a href="#within-each-individual">Within each individual</a></li>
</ul></li>
<li><a href="#compare-transcriptome-profile-between-individual-adjusted-cv">Compare transcriptome profile between individual, adjusted CV</a><ul>
<li><a href="#between-individual-unadjusted-cv">Between individual, unadjusted CV</a></li>
</ul></li>
<li><a href="#session-information">Session information</a></li>
</ul>
</div>

<p><strong>Last updated:</strong> 2015-12-11</p>
<p><strong>Code version:</strong> a795f9edad85aa99750bd8425fbe4d89346cfd42</p>
<div id="objective" class="section level2">
<h2>Objective</h2>
<p>Compare transcriptome profile of cell-to-cell heterogeneity within individuals betwen batches and between individuals.</p>
<p>Here we compute CV for each of the batches and then follow the algorithm described in <a href="link1">Kolodziejczk et al. (2015)</a> to remove mean dependency of the batch-specific CVs. Then, in order to compare CVs between individuals, we need to account for the multiple CVs for each gene for each individual cell line. The statistical test of choice would be a nonparametric equivalent of two-way ANOVA, which turns out to be difficult to compute.</p>
</div>
<div id="set-up" class="section level2">
<h2>Set up</h2>
<pre class="r"><code>library(&quot;data.table&quot;)
library(&quot;dplyr&quot;)
library(&quot;limma&quot;)
library(&quot;edgeR&quot;)
library(&quot;ggplot2&quot;)
library(&quot;grid&quot;)
theme_set(theme_bw(base_size = 12))
source(&quot;functions.R&quot;)</code></pre>
</div>
<div id="prepare-data" class="section level2">
<h2>Prepare data</h2>
<p>Input annotation of only QC-filtered single cells. Remove NA19098.r2</p>
<pre class="r"><code>anno_qc_filter &lt;- read.table(&quot;../data/annotation-filter.txt&quot;, header = TRUE,
                   stringsAsFactors = FALSE)</code></pre>
<p>Import endogeneous gene molecule counts that are QC-filtered, CPM-normalized, ERCC-normalized, and also processed to remove unwanted variation from batch effet. ERCC genes are removed from this file.</p>
<pre class="r"><code>molecules_ENSG &lt;- read.table(&quot;../data/molecules-final.txt&quot;, header = TRUE, stringsAsFactors = FALSE)</code></pre>
<p>Input moleclule counts before log2 CPM transformation. This file is used to compute percent zero-count cells per sample.</p>
<pre class="r"><code>molecules_sparse &lt;- read.table(&quot;../data/molecules-filter.txt&quot;, header = TRUE, stringsAsFactors = FALSE)

molecules_sparse &lt;- molecules_sparse[grep(&quot;ENSG&quot;, rownames(molecules_sparse)), ]
stopifnot( all.equal(rownames(molecules_ENSG), rownames(molecules_sparse)) )</code></pre>
</div>
<div id="compute-normalized-cv-for-each-batch" class="section level2">
<h2>Compute normalized CV for each batch</h2>
<pre class="r"><code>source(&quot;../code/cv-functions.r&quot;)

# compute_cv() takes log2 counts as input
ENSG_cv &lt;- compute_cv(log2counts = molecules_ENSG,
                      grouping_vector = anno_qc_filter$batch)

ENSG_cv_adj &lt;- normalize_cv(group_cv = ENSG_cv, 
                            log2counts = molecules_ENSG, 
                            anno = anno_qc_filter)</code></pre>
</div>
<div id="compare-transcriptome-profile-within-individuals" class="section level2">
<h2>Compare transcriptome profile within individuals</h2>
<div id="within-each-individual" class="section level3">
<h3>Within each individual</h3>
<p>The CV normalization method employed here scales each sample individual with respect to data-wide coefficients of variation. Therefore, we would expect that the transcriptome-wide profile relationship between samples (with or between individuals) to be consistent before versus after normalization.</p>
<ul>
<li>Before normalization</li>
</ul>
<pre class="r"><code>print(friedman.test( cbind(ENSG_cv[[1]]$cv, ENSG_cv[[2]]$cv) ) )</code></pre>
<pre><code>
    Friedman rank sum test

data:  cbind(ENSG_cv[[1]]$cv, ENSG_cv[[2]]$cv)
Friedman chi-squared = 454, df = 1, p-value &lt; 2.2e-16</code></pre>
<pre class="r"><code>print(friedman.test( cbind(ENSG_cv[[3]]$cv, ENSG_cv[[4]]$cv, ENSG_cv[[5]]$cv) ) )</code></pre>
<pre><code>
    Friedman rank sum test

data:  cbind(ENSG_cv[[3]]$cv, ENSG_cv[[4]]$cv, ENSG_cv[[5]]$cv)
Friedman chi-squared = 285.15, df = 2, p-value &lt; 2.2e-16</code></pre>
<pre class="r"><code>print(friedman.test( cbind(ENSG_cv[[6]]$cv, ENSG_cv[[7]]$cv, ENSG_cv[[8]]$cv) ) )</code></pre>
<pre><code>
    Friedman rank sum test

data:  cbind(ENSG_cv[[6]]$cv, ENSG_cv[[7]]$cv, ENSG_cv[[8]]$cv)
Friedman chi-squared = 4623.9, df = 2, p-value &lt; 2.2e-16</code></pre>
<ul>
<li>After normalization</li>
</ul>
<pre class="r"><code>print(friedman.test( cbind(ENSG_cv_adj[[1]]$log10cv2_adj, ENSG_cv_adj[[2]]$log10cv2_adj) ) )</code></pre>
<pre><code>
    Friedman rank sum test

data:  cbind(ENSG_cv_adj[[1]]$log10cv2_adj, ENSG_cv_adj[[2]]$log10cv2_adj)
Friedman chi-squared = 454, df = 1, p-value &lt; 2.2e-16</code></pre>
<pre class="r"><code>print(friedman.test( cbind(ENSG_cv_adj[[3]]$log10cv2_adj, 
                           ENSG_cv_adj[[4]]$log10cv2_adj, 
                           ENSG_cv_adj[[5]]$log10cv2_adj) ) )</code></pre>
<pre><code>
    Friedman rank sum test

data:  cbind(ENSG_cv_adj[[3]]$log10cv2_adj, ENSG_cv_adj[[4]]$log10cv2_adj,     ENSG_cv_adj[[5]]$log10cv2_adj)
Friedman chi-squared = 285.15, df = 2, p-value &lt; 2.2e-16</code></pre>
<pre class="r"><code>print(ENSG_cv[[6]]$individual[6])</code></pre>
<pre><code>NULL</code></pre>
<pre class="r"><code>print(friedman.test( cbind(ENSG_cv_adj[[6]]$log10cv2_adj, 
                           ENSG_cv_adj[[7]]$log10cv2_adj, ENSG_cv_adj[[8]]$log10cv2_adj) ) )</code></pre>
<pre><code>
    Friedman rank sum test

data:  cbind(ENSG_cv_adj[[6]]$log10cv2_adj, ENSG_cv_adj[[7]]$log10cv2_adj,     ENSG_cv_adj[[8]]$log10cv2_adj)
Friedman chi-squared = 4623.9, df = 2, p-value &lt; 2.2e-16</code></pre>
<p>*Plot CV before/after normalization</p>
<p>The normalization method we used is a non-linear transformation. The method aims to remove heterogeneity of variances across mean gene expression levels.</p>
<pre class="r"><code>plot(x = rank(log10((ENSG_cv_adj[[6]]$cv)^2)),
     y = rank(ENSG_cv_adj[[6]]$log10cv2_adj),
     xlab = &quot;Before normalization&quot;, ylab = &quot;After normalization&quot;,
     cex = .4, pch = 16, main = &quot;Ranks of CV&quot;)
abline(0, 1, col = &quot;red&quot;)</code></pre>
<p><img src="figure/cv-adjusted-transcriptome-wide.Rmd/unnamed-chunk-8-1.png" title="" alt="" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>plot(x = (log10((ENSG_cv_adj[[6]]$cv)^2)),
     y = (ENSG_cv_adj[[6]]$log10cv2_adj),
     xlab = &quot;Before normalization&quot;, ylab = &quot;After normalization&quot;,
     cex = .4, pch = 16, main = &quot;log10 squared-CV&quot;)
abline(0, 1, col = &quot;red&quot;)</code></pre>
<p><img src="figure/cv-adjusted-transcriptome-wide.Rmd/unnamed-chunk-8-2.png" title="" alt="" width="672" style="display: block; margin: auto;" /></p>
</div>
</div>
<div id="compare-transcriptome-profile-between-individual-adjusted-cv" class="section level2">
<h2>Compare transcriptome profile between individual, adjusted CV</h2>
<ul>
<li>PCA</li>
</ul>
<p>Perform PCA for each individual. Consider replicates. Then compare PCA gene loadings between individuals.</p>
<p>Use <em>prcomp</em> to perform PCA: * output<span class="math inline"><em>x</em> : <em>l</em><em>o</em><em>a</em><em>d</em><em>i</em><em>n</em><em>g</em><em>s</em> * <em>o</em><em>u</em><em>t</em><em>p</em><em>u</em><em>t</em></span>rotation: loadings * output<span class="math inline"><em>c</em><em>e</em><em>n</em><em>t</em><em>e</em><em>r</em> : <em>v</em><em>a</em><em>r</em><em>i</em><em>a</em><em>b</em><em>l</em><em>e</em><em>m</em><em>e</em><em>a</em><em>n</em><em>s</em> * <em>o</em><em>u</em><em>t</em><em>p</em><em>u</em><em>t</em></span>scale: variable standard deviation</p>
<pre class="r"><code>library(matrixStats)
pca_individuals &lt;- lapply(1:3, function(ii_individual) {
  if (ii_individual == 1) {
  df_ind &lt;- cbind(ENSG_cv_adj$NA19098.r1$log10cv2_adj,
                  ENSG_cv_adj$NA19098.r3$log10cv2_adj ) 
  df_ind &lt;- (df_ind - rowMeans(df_ind))/sqrt(rowVars(df_ind))

  }
  if (ii_individual == 2) {
  df_ind &lt;- cbind(ENSG_cv_adj$NA19101.r1$log10cv2_adj,
                  ENSG_cv_adj$NA19101.r2$log10cv2_adj,
                  ENSG_cv_adj$NA19101.r3$log10cv2_adj )
  df_ind &lt;- (df_ind - rowMeans(df_ind))/sqrt(rowVars(df_ind))

  }
  if (ii_individual == 3) {
  df_ind &lt;- cbind(ENSG_cv_adj$NA19239.r1$log10cv2_adj,
                  ENSG_cv_adj$NA19239.r2$log10cv2_adj,
                  ENSG_cv_adj$NA19239.r3$log10cv2_adj )
  df_ind &lt;- (df_ind - rowMeans(df_ind))/sqrt(rowVars(df_ind))
  }
  pca_results &lt;- prcomp( t(df_ind), retx = TRUE, center = TRUE, scale = TRUE )
  prop_var &lt;- (pca_results$sdev^2) / sum(pca_results$sdev^2)
  
  list(loadings = pca_results$x,
       prop_var = prop_var)
})


sapply(pca_individuals, function(xx) round(xx$prop_var,2) )</code></pre>
<pre><code>[[1]]
[1] 1 0

[[2]]
[1] 0.51 0.49 0.00

[[3]]
[1] 0.59 0.41 0.00</code></pre>
<pre class="r"><code>pca_all &lt;- cbind(pca_individuals[[1]]$loadings[ , 1],
                 pca_individuals[[2]]$loadings[ , 1],
                 pca_individuals[[3]]$loadings[ , 1] )

friedman_results &lt;- friedman.test(pca_all)
friedman_results</code></pre>
<pre><code>
    Friedman rank sum test

data:  pca_all
Friedman chi-squared = 0.66667, df = 2, p-value = 0.7165</code></pre>
<div id="between-individual-unadjusted-cv" class="section level3">
<h3>Between individual, unadjusted CV</h3>
<ul>
<li>PCA for each individual</li>
</ul>
<pre class="r"><code>pca_individuals &lt;- lapply(1:3, function(ii_individual) {
  if (ii_individual == 1) {
  df_ind &lt;- log10(cbind(ENSG_cv_adj$NA19098.r1$cv,
                        ENSG_cv_adj$NA19098.r3$cv )^2 ) 
  }
  if (ii_individual == 2) {
  df_ind &lt;- log10(cbind(ENSG_cv_adj$NA19101.r1$cv,
                        ENSG_cv_adj$NA19101.r2$cv,
                        ENSG_cv_adj$NA19101.r3$cv )^2)
  }
  if (ii_individual == 3) {
  df_ind &lt;- log10(cbind(ENSG_cv_adj$NA19239.r1$cv,
                        ENSG_cv_adj$NA19239.r2$cv,
                        ENSG_cv_adj$NA19239.r3$cv )^2 )
  }
  pca_results &lt;- prcomp( df_ind, retx = TRUE, center = TRUE, scale = TRUE)
  prop_var &lt;- (pca_results$sdev^2) / sum(pca_results$sdev^2)
  
  list(scores = pca_results,
       loadings = pca_results$x,
       prop_var = prop_var)
})

sapply(pca_individuals, function(xx) xx$prop_var)</code></pre>
<pre><code>[[1]]
[1] 0.95561426 0.04438574

[[2]]
[1] 0.94399168 0.02924917 0.02675915

[[3]]
[1] 0.94613908 0.02872685 0.02513406</code></pre>
<pre class="r"><code>pca_all &lt;- cbind(pca_individuals[[1]]$loadings[ , 1],
                 pca_individuals[[2]]$loadings[ , 1],
                 pca_individuals[[3]]$loadings[ , 1] )
                 
friedman_results &lt;- friedman.test(pca_all)
friedman_results</code></pre>
<pre><code>
    Friedman rank sum test

data:  pca_all
Friedman chi-squared = 8.6708, df = 2, p-value = 0.0131</code></pre>
<ul>
<li>Compute CV across replicates</li>
</ul>
<p>First we make PCA of each individual samples to evaluate whether replicate is a main source of variation. If replicate is not a main source of variation, we will combine the replicates to perform between indvidual comparisons.</p>
<pre class="r"><code>temp &lt;- molecules_ENSG[, anno_qc_filter$individual == &quot;NA19098&quot;]
anno_temp &lt;- anno_qc_filter[anno_qc_filter$individual == &quot;NA19098&quot;,]
pca_temp &lt;- prcomp( t(temp - rowMeans(temp)) )
(pca_temp$sdev^2)[1:3]/sum(pca_temp$sdev^2)</code></pre>
<pre><code>[1] 0.16112091 0.01273397 0.01082798</code></pre>
<pre class="r"><code>xy &lt;- cbind(pca_temp$x[,1], pca_temp$x[,2] )

ggplot(data.frame(pc1 = pca_temp$x[,1],
                  pc2 = pca_temp$x[,2],
                  replicate = anno_temp$replicate),
       aes(x = pc1, y = pc2, col = replicate)) +
  geom_point() +
  ggtitle(&quot;NA19098&quot;)</code></pre>
<p><img src="figure/cv-adjusted-transcriptome-wide.Rmd/unnamed-chunk-11-1.png" title="" alt="" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>par(mfrow = c(1,2))
plot(x = rowMeans(temp), y = pca_temp$rotation[ ,1],
     xlab = &quot;mean molecule counts&quot;, ylab = &quot;first PC loadings&quot;)
plot(x = rowMeans(temp), y = pca_temp$rotation[ ,2],
     xlab = &quot;mean molecule counts&quot;, ylab = &quot;second PC loadings&quot;)
title(main = &quot;NA19098&quot;, outer = TRUE, line = -1)</code></pre>
<p><img src="figure/cv-adjusted-transcriptome-wide.Rmd/unnamed-chunk-11-2.png" title="" alt="" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>cor(rowMeans(temp), pca_temp$rotation[ ,1], method = &quot;spearman&quot;)</code></pre>
<pre><code>[1] -0.6703884</code></pre>
<pre class="r"><code>temp &lt;- molecules_ENSG[, anno_qc_filter$individual == &quot;NA19101&quot;]
anno_temp &lt;- anno_qc_filter[anno_qc_filter$individual == &quot;NA19101&quot;,]
pca_temp &lt;- prcomp( t(temp - rowMeans(temp)))
(pca_temp$sdev^2)[1:3]/sum(pca_temp$sdev^2)</code></pre>
<pre><code>[1] 0.123570628 0.012287068 0.008106379</code></pre>
<pre class="r"><code>xy &lt;- cbind(pca_temp$x[,1], pca_temp$x[,2] )

ggplot(data.frame(pc1 = pca_temp$x[,1],
                  pc2 = pca_temp$x[,2],
                  replicate = anno_temp$replicate),
       aes(x = pc1, y = pc2, col = replicate)) +
  geom_point() +
  ggtitle(&quot;NA19101&quot;)</code></pre>
<p><img src="figure/cv-adjusted-transcriptome-wide.Rmd/unnamed-chunk-12-1.png" title="" alt="" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>par(mfrow = c(1,2))
plot(x = rowMeans(temp), y = pca_temp$rotation[ ,1],
     xlab = &quot;mean molecule counts&quot;, ylab = &quot;first PC loadings&quot;)
plot(x = rowMeans(temp), y = pca_temp$rotation[ ,2],
     xlab = &quot;mean molecule counts&quot;, ylab = &quot;second PC loadings&quot;)
title(main = &quot;NA19101&quot;, outer = TRUE)</code></pre>
<p><img src="figure/cv-adjusted-transcriptome-wide.Rmd/unnamed-chunk-12-2.png" title="" alt="" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>cor(rowMeans(temp), pca_temp$rotation[ ,1], method = &quot;spearman&quot;)</code></pre>
<pre><code>[1] -0.6761292</code></pre>
<pre class="r"><code>temp &lt;- molecules_ENSG[, anno_qc_filter$individual == &quot;NA19239&quot;]
anno_temp &lt;- anno_qc_filter[anno_qc_filter$individual == &quot;NA19239&quot;,]
pca_temp &lt;- prcomp( t(temp - rowMeans(temp)), retx = TRUE, center = TRUE, scale = TRUE )
(pca_temp$sdev^2)[1:3]/sum(pca_temp$sdev^2)</code></pre>
<pre><code>[1] 0.190501391 0.011386141 0.007254053</code></pre>
<pre class="r"><code>xy &lt;- cbind(pca_temp$x[,1], pca_temp$x[,2] )

ggplot(data.frame(pc1 = pca_temp$x[,1],
                  pc2 = pca_temp$x[,2],
                  replicate = anno_temp$replicate),
       aes(x = pc1, y = pc2, col = replicate)) +
  geom_point() +
  ggtitle(&quot;NA19239&quot;)</code></pre>
<p><img src="figure/cv-adjusted-transcriptome-wide.Rmd/unnamed-chunk-13-1.png" title="" alt="" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>par(mfrow = c(1,2))
plot(x = rowMeans(temp), y = pca_temp$rotation[ ,1],
     xlab = &quot;mean molecule counts&quot;, ylab = &quot;first PC loadings&quot;)
plot(x = rowMeans(temp), y = pca_temp$rotation[ ,2],
     xlab = &quot;mean molecule counts&quot;, ylab = &quot;second PC loadings&quot;)
title(main = &quot;NA19239&quot;, outer = TRUE)</code></pre>
<p><img src="figure/cv-adjusted-transcriptome-wide.Rmd/unnamed-chunk-13-2.png" title="" alt="" width="672" style="display: block; margin: auto;" /></p>
<p>Compute CV across replicates.</p>
<pre class="r"><code>cv_individual &lt;- lapply(1:3, function(ii_individual) {
    individuals &lt;- unique(anno_qc_filter$individual)
    counts &lt;- molecules_ENSG[ , anno_qc_filter$individual == individuals [ii_individual]]
    means &lt;- apply(counts, 1, mean)
    sds &lt;- apply(counts, 1, sd)
    cv &lt;- sds/means
    return(cv)
})
names(cv_individual) &lt;- unique(anno_qc_filter$individual)
cv_individual &lt;- do.call(cbind, cv_individual)
friedman.test(cv_individual)</code></pre>
<pre><code>
    Friedman rank sum test

data:  cv_individual
Friedman chi-squared = 3524.3, df = 2, p-value &lt; 2.2e-16</code></pre>
</div>
</div>
<div id="session-information" class="section level2">
<h2>Session information</h2>
<pre class="r"><code>sessionInfo()</code></pre>
<pre><code>R version 3.2.1 (2015-06-18)
Platform: x86_64-apple-darwin13.4.0 (64-bit)
Running under: OS X 10.10.5 (Yosemite)

locale:
[1] en_US.UTF-8/en_US.UTF-8/en_US.UTF-8/C/en_US.UTF-8/en_US.UTF-8

attached base packages:
[1] grid      stats     graphics  grDevices utils     datasets  methods  
[8] base     

other attached packages:
[1] matrixStats_0.15.0 zoo_1.7-12         ggplot2_1.0.1     
[4] edgeR_3.10.5       limma_3.24.15      dplyr_0.4.3       
[7] data.table_1.9.6   knitr_1.11        

loaded via a namespace (and not attached):
 [1] Rcpp_0.12.2      magrittr_1.5     MASS_7.3-45      munsell_0.4.2   
 [5] lattice_0.20-33  colorspace_1.2-6 R6_2.1.1         stringr_1.0.0   
 [9] plyr_1.8.3       tools_3.2.1      parallel_3.2.1   gtable_0.1.2    
[13] DBI_0.3.1        htmltools_0.2.6  yaml_2.1.13      assertthat_0.1  
[17] digest_0.6.8     reshape2_1.4.1   formatR_1.2.1    evaluate_0.8    
[21] rmarkdown_0.8.1  labeling_0.3     stringi_1.0-1    scales_0.3.0    
[25] chron_2.3-47     proto_0.3-10    </code></pre>
</div>


<!-- some extra javascript for older browsers -->
<script type="text/javascript" src="libs/polyfill.js"></script>

<script>

// manage active state of menu based on current page
$(document).ready(function () {

    // active menu
    href = window.location.pathname
    href = href.substr(href.lastIndexOf('/') + 1)
    $('a[href="' + href + '"]').parent().addClass('active');

    // manage active menu header
    if (href.startsWith('authoring_'))
      $('a[href="' + 'authoring' + '"]').parent().addClass('active');
    else if (href.endsWith('_format.html'))
      $('a[href="' + 'formats' + '"]').parent().addClass('active');
    else if (href.startsWith('developer_'))
      $('a[href="' + 'developer' + '"]').parent().addClass('active');

});

</script>

</div>

<script>

// add bootstrap table styles to pandoc tables
$(document).ready(function () {
  $('tr.header').parent('thead').parent('table').addClass('table table-condensed');
});

</script>


</body>
</html>
