---
title: "Cell-to-cell variation analysis: expressed cells"
author: "Joyce Hsiao"
date: 2016-07-01  
---

**Last updated:** `r Sys.Date()`

**Code version:** `r system("git log -1 --format='%H'", intern = TRUE)`

```{r chunk-options, include=FALSE}
source("chunk-options.R")

library(knitr)
opts_chunk$set(message = FALSE, warning = FALSE, eval = TRUE, 
               echo = TRUE)
```



## Background and some observations

Compare expression variability among the detected/expressed cells between individuals. Our observations made in this document are summarized below.

1. __CV-mean relationship__: This pattern in the expressed cells is different from the usual concave function we observe in bulk RNA-seq and scRNA-seq. Gene CVs increase as a function of gene abundance for genes with more than 50 percent of undetected cells; while, gene CVs decrease as a function of mean abundance, a pattern similar to previous studies for genes with 50 percent or less undetected cells.

2. __Overlaps of individual top mean and top CV genes__: Similar to when including non-expressed cells, there are few common top CV genes across individuals (~100 genes) and many more common top genes across individuals ( > 800 genes). This suggests possible individual differences in expression variablity. 

3. __Compare CV of the expressed cells__: We found 680 genes with differential variation across the expressed cell. 

4. __Compare mean abundance of the expressed cells__: Due to the large number of cells in each individual cell lines, more than 95% of the genes were found to have statistically significant differences between all three individuals. We identified differential expression genes between pairs of individuals under the conditions of q-value less than .01 in the test and also log2 fold change greater than 2: 5 genes in NA19098-NA19101, 6 in NA19098-NA19239, and 2 genes in NA19101-NA19239. *Note: The criterion for differential expression genes may be stringent, but the goal of this analysis is not to have a final say on the biological differences between the cell line, but rather to begin a conversatoin about the relationship between percent of undeteced cells and mean abundance*


## Set up

```{r, message=FALSE, warning=FALSE}
library(knitr)
library("ggplot2")
library("grid")
theme_set(theme_bw(base_size = 12))
source("functions.R")
library("Humanzee")
library("cowplot")
library("MASS")
library("matrixStats")
source("../code/cv-functions.r")
source("../code/plotting-functions.R")
library("mygene")
```



## Prepare data

We import molecule counts before standardizing and transformation and also log2-transformed counts after batch-correction. Biological variation analysis of the individuals is performed on the batch-corrected and log2-transformed counts. 


```{r import-data}
# Import filtered annotations
anno_filter <- read.table("../data/annotation-filter.txt", 
                      header = TRUE,
                      stringsAsFactors = FALSE)

# Import filtered molecule counts
molecules_filter <- read.table("../data/molecules-filter.txt",
                               header = TRUE, stringsAsFactors = FALSE)
stopifnot(NROW(anno_filter) == NCOL(molecules_filter))

# Import final processed molecule counts of endogeneous genes
molecules_final <- read.table("../data/molecules-final.txt", 
                             header = TRUE, stringsAsFactors = FALSE)
stopifnot(NROW(anno_filter) == NCOL(molecules_final))

# Import gene symbols
gene_symbols <- read.table(file = "../data/gene-info.txt", sep = "\t",
                           header = TRUE, stringsAsFactors = FALSE, quote = "")

# Import cell-cycle gene list
cell_cycle_genes <- read.table("../data/cellcyclegenes.txt",
                               header = TRUE, sep = "\t",
                               stringsAsFactors = FALSE)

# Import pluripotency gene list
pluripotency_genes <- read.table("../data/pluripotency-genes.txt",
                               header = TRUE, sep = "\t",
                               stringsAsFactors = FALSE)$To
```



Load CV results of all cells from [previous analysis][link]

[link]: http://jdblischak.github.io/singleCellSeq/analysis/cv-adjusted-summary-pois.html

```{r}
load("../data/cv-all-cells.rda")
```



## Compute gene mean and variance in the expressed cells

```{r}
file_name <- "../data/cv-expressed-cells.rda"
if (file.exists(file_name)) {
  load(file_name)
} else {
expressed_cv <- compute_expressed_cv(molecules_filter, 
                                     molecules_final, 
                                     anno_filter$individual)

expressed_dm <- normalize_cv_input(expressed_cv,
                                   anno_filter$individual)

save(expressed_cv, expressed_dm, file = file_name)
}
str(expressed_cv)
```


13,043 genes were found to have valid CV and normalized CV (dm) measures in our computations that use a subset of cells with more than 1 UMI. This set of genes is smaller than the 13,058 genes in the final dataset (`molecules-final.txt`). The difference of 15 genes is the result of filtering out genes that do not have valid CV measures in all of the individuals; these are also genes that we did not detect a UMI in at least one of the individuals.

```{r}
diff_set <- setdiff(as.character(rownames(molecules_final)),
                    as.character(rownames(expressed_cv$all)) )
for (i in 1:length(diff_set)) {
print(
  table(unlist(molecules_filter[rownames(molecules_filter) == diff_set[i], ]) > 1, anno_filter$individual) )
}
```


We subset the expression matrices of the 13,058 genes to include only the 13,043 genes with valid CV measures.


```{r match-genes}
# get gene names from the cv data of the expessed cells
valid_genes_cv_expressed <- rownames(expressed_cv[[1]])

# make subset data for later analysis involving expressed cells
molecules_filter_subset <- molecules_filter[
  which(rownames(molecules_filter) %in% valid_genes_cv_expressed), ]

molecules_final_subset <- molecules_final[
  which(rownames(molecules_final) %in% valid_genes_cv_expressed), ]

# subset cv for all cells to include only the 13,043 genes with valid measures of CV among the expressed cells
ENSG_cv_subset <- lapply(ENSG_cv, function(x) {
  x[which(rownames(x) %in% valid_genes_cv_expressed), ]      
})
names(ENSG_cv_subset) <- names(ENSG_cv)  

# subset adjusted cv for all cells to include only the 13,043 genes with valid measures of CV among the expressed cells
ENSG_cv_adj_subset <- lapply(ENSG_cv_adj, function(x) {
  x[which(rownames(x) %in% valid_genes_cv_expressed), ]      
})
names(ENSG_cv_adj_subset) <- names(ENSG_cv_adj) 
```


Compute a matrix of 0's and 1's indicating non-detected and detected cells, respectively.

```{r}
molecules_expressed_subset <- molecules_filter_subset
molecules_expressed_subset[which(molecules_filter_subset > 0 , arr.ind = TRUE)] <- 1
molecules_expressed_subset <- as.matrix((molecules_expressed_subset))

# make a batch-corrected data set in which the non-detected cells are 
# code as NA
molecules_final_expressed_subset <- molecules_final_subset
molecules_final_expressed_subset[which(molecules_filter_subset == 0, arr.ind= TRUE)] <- NA
```



## CV-mean plots


```{r}
theme_set(theme_bw(base_size = 8))
cowplot::plot_grid(
    plot_poisson_cv_expressed(
          expr_mean = expressed_cv$all$expr_mean, 
          exprs_cv = expressed_cv$all$expr_cv, 
          ylab = "Coefficient of variation (CV)", 
          main = "All individauls, expressed cells") +
          theme(legend.position = "none"),
    plot_poisson_cv_expressed(
          expr_mean = expressed_cv$NA19098$expr_mean, 
          exprs_cv = expressed_cv$NA19098$expr_cv, 
          ylab = "Coefficient of variation (CV)", 
          main = "NA19098 expressed cells") +
          theme(legend.position = "none"),
    plot_poisson_cv_expressed(
          expr_mean = expressed_cv$NA19101$expr_mean, 
          exprs_cv = expressed_cv$NA19101$expr_cv, 
          ylab = "Coefficient of variation (CV)", 
          main = "NA19101 expressed cells") +
          theme(legend.position = "none"),
    plot_poisson_cv_expressed(
          expr_mean = expressed_cv$NA19239$expr_mean, 
          exprs_cv = expressed_cv$NA19239$expr_cv, 
          ylab = "Coefficient of variation (CV)", 
          main = "NA19239 expressed cells") +
          theme(legend.position = "none"),
  ncol = 2,
  labels = LETTERS[1:4])
```



CV all cells vs. expressed cells


```{r}
require(matrixStats)

xlabs <- "CV of all cells"
ylabs <- "CV of expressed cells"
plot_title <- names(expressed_cv)
par(mfrow = c(2,2))
# plot(x = ENSG_cv_subset$all$cv,
#      y = expressed_cv$all$expr_cv)
for (ind in names(expressed_cv)[1:3]) {
  which_ind <- which(names(ENSG_cv_subset) %in% ind)
  plot(x = ENSG_cv_subset[[ind]]$cv,
       y = expressed_cv[[ind]]$expr_cv, 
       cex = .7, pch = 16, col = scales::alpha("grey20", .7),
       xlab = xlabs,
       ylab = ylabs,
       main = plot_title[which_ind])
}
title("CV before adjustment")
```


Adjusted CV values are orthogonal between individuals.

```{r}
par(mfrow = c(2,2))
for (i in 1:2) {
  for (j in (i+1):3) {
    plot(expressed_cv[[i]]$expr_cv,
         expressed_cv[[j]]$expr_cv,
         xlab = names(expressed_cv)[i],
         ylab = names(expressed_cv)[j],
          cex = .7, pch = 16, col = scales::alpha("grey20", .7))
  }
}
title(main = "Between individual CVs",
      outer = TRUE, line = -1)


par(mfrow = c(2,2))
for (i in 1:2) {
  for (j in (i+1):3) {
    plot(expressed_dm[[i]],
         expressed_dm[[j]],
         xlab = names(expressed_cv)[i],
         ylab = names(expressed_cv)[j],
          cex = .7, pch = 16, col = scales::alpha("grey20", .7))
  }
}
title(main = "Between individual adjusted CVs",
      outer = TRUE, line = -1)
```



## Extreme CV genes - top 1000

CV before correction

```{r, fig.width=8, fig.height=4}
library(VennDiagram)
library(gridExtra)
genes <- rownames(molecules_final_subset)
overlap_list_expressed <- list(
   NA19098 = genes[ which( rank(expressed_cv$NA19098$expr_cv) 
                               > length(genes) - 1000 ) ],
   NA19101 = genes[ which( rank(expressed_cv$NA19101$expr_cv) 
                               > length(genes) - 1000 ) ],
   NA19239 = genes[ which( rank(expressed_cv$NA19239$expr_cv) 
                               > length(genes) - 1000 ) ] )  

overlap_list_all <- list(
   NA19098 = genes[ which( rank(ENSG_cv_subset$NA19098$cv) 
                               > length(genes) - 1000 ) ],
   NA19101 = genes[ which( rank(ENSG_cv_subset$NA19101$cv)
                           > length(genes) - 1000 ) ],
   NA19239 = genes[ which( rank(ENSG_cv_subset$NA19239$cv)
                           > length(genes) - 1000 ) ] ) 

grid.arrange(gTree(children = venn.diagram(overlap_list_all,filename = NULL,
                          category.names = names(overlap_list_all),
                          name = "All cells")),
             gTree(children = venn.diagram(overlap_list_expressed,filename = NULL,
                          category.names = names(overlap_list_expressed),
                          name = "Expressed cells")),
             ncol = 2)
```



Adjusted CV (CV after correction for mean abundance)

```{r, fig.width=8, fig.height=4}
genes <- rownames(molecules_final_subset)
overlap_list_expressed <- list(
   NA19098 = genes[ which( rank(expressed_dm$NA19098) 
                               > length(genes) - 1000 ) ],
   NA19101 = genes[ which( rank(expressed_dm$NA19101) 
                               > length(genes) - 1000 ) ],
   NA19239 = genes[ which( rank(expressed_dm$NA19239) 
                               > length(genes) - 1000 ) ] )  

overlap_list_all <- list(
   NA19098 = genes[ which( rank(ENSG_cv_adj_subset$NA19098$log10cv2_adj) 
                               > length(genes) - 1000 ) ],
   NA19101 = genes[ which( rank(ENSG_cv_adj_subset$NA19101$log10cv2_adj)
                           > length(genes) - 1000 ) ],
   NA19239 = genes[ which( rank(ENSG_cv_adj_subset$NA19239$log10cv2_adj)
                           > length(genes) - 1000 ) ] ) 

grid.arrange(gTree(children = venn.diagram(overlap_list_all,filename = NULL,
                          category.names = names(overlap_list_all),
                          name = "All cells")),
             gTree(children = venn.diagram(overlap_list_expressed,filename = NULL,
                          category.names = names(overlap_list_expressed),
                          name = "Expressed cells")),
             ncol = 2)
```




Mean and CV

```{r, fig.width=8, fig.height=4}
genes <- rownames(molecules_final_subset)
overlap_list_expressed <- list(
   NA19098 = genes[ which( rank(expressed_dm$NA19098) 
                               > length(genes) - 1000 ) ],
   NA19101 = genes[ which( rank(expressed_dm$NA19101) 
                               > length(genes) - 1000 ) ],
   NA19239 = genes[ which( rank(expressed_dm$NA19239) 
                               > length(genes) - 1000 ) ] )  

overlap_list_mn <- list(
   NA19098 = genes[ which( rank(expressed_cv$NA19098$expr_mean) 
                               > length(genes) - 1000 ) ],
   NA19101 = genes[ which( rank(expressed_cv$NA19101$expr_mean) 
                           > length(genes) - 1000 ) ],
   NA19239 = genes[ which( rank(expressed_cv$NA19239$expr_mean) 
                           > length(genes) - 1000 ) ] ) 

grid.arrange(gTree(children = venn.diagram(overlap_list_expressed,filename = NULL,
                          category.names = names(overlap_list_expressed),
                          name = "Adjusted CV")),
             gTree(children = venn.diagram(overlap_list_mn,filename = NULL,
                          category.names = names(overlap_list_mn),
                          name = "Abundance")),
             ncol = 2)
```




## Differential testing of adjusted CV - permutation

Compute MAD values.

```{r, eval = TRUE}
library(matrixStats)
mad_expressed <- rowMedians( abs( as.matrix(expressed_dm) - rowMedians(as.matrix(expressed_dm)) ) )
#save(mad_expressed, file = "../data/mad-expressed.rda")
```

Load empirical p-values of the MAD values. The permutations were done in midway.

Method: After obtaining the number of permutated statistics greatert than each observed test statistic, we follow the method by Philson and Smyth (2010) and compute the emprical p-value as $p = (b+1)/(m+1)$ where $b$ is the number of the permutation-based test statistics that are more significant than the observed test statistics, and $m$ is the number of permutations performed. 

*permuted-pval-expressed-set1.rda: 300,000 permutations*

```{r}
load("../data/permuted-pval-expressed-set1.rda")
load("../data/mad-expressed.rda")
par(mfrow = c(1,1))
hist(perm_pval_set1,
     main = paste(m_perm_set1, "permutations"),
     xlab = "Empirical p-value")
```

*[chunk not evaluated]*
```{r, eval = FALSE}
# This is an example script used to compute permutation-based MAD statistic.
library(Humanzee)
perm_set <- do.call(cbind, lapply(1:100, function(i) {
  perms <- permute_cv_test(log2counts = molecules_final_subset, 
                subset_matrix = molecules_expressed_subset,
                grouping_vector = anno_filter$individual, 
                anno = anno_filter, 
                number_permute = 10,                            
                output_rda = FALSE,
                do_parallel = TRUE,
                number_cores = 8) 
  return(perms)
})  )
save(perm_set, 
     file = "rda/cv-adjusted-summary-pois-expressed/mad-perm-values.rda")
  
# table(c(rowSums(perm_data > mad_expressed)) > 0)
# perm_pval <- (rowSums(perm_set > mad_expressed) + 1)/( NCOL(perm_set) + 1)
# hist(perm_pval)
# perm_fdr <- p.adjust(perm_pval, method = "fdr")
# summary(perm_fdr)
# sum(perm_fdr < .01, na.rm = TRUE)
```


Pluripotency genes

*P < .0001

```{r}
sig_genes <- names(perm_pval_set1)[which(perm_pval_set1 < .0001)]
sig_genes[which(sig_genes %in% pluripotency_genes)]
gene_symbols[which(gene_symbols$ensembl_gene_id %in% sig_genes[which(sig_genes %in% pluripotency_genes)]), ]
```



Print out pluripotent genes

```{r, fig.width=7, fig.height=4}
sig_pluri_ensg <- gene_symbols[which(gene_symbols$ensembl_gene_id %in% sig_genes[which(sig_genes %in% pluripotency_genes)]), ]

source("../code/plotting-functions.R")
par(mfrow = c(1,2))
for (i in 1:length(sig_pluri_ensg$ensembl_gene_id)) {
  plot_density_overlay(
    molecules = molecules_final_expressed_subset,
    annotation = anno_filter,
    which_gene = sig_pluri_ensg$ensembl_gene_id[i], 
    labels = "",
#    xlims = c(8,15), 
#    ylims = c(0,1.5),
    cex.lab = 1.2,
    cex.axis = 1.2,
    gene_symbols = gene_symbols)
}

pvals <- perm_pval_set1[which(names(perm_pval_set1) %in% sig_pluri_ensg$ensembl_gene_id)]
cbind(pvals, 
      mad_expressed[match(names(pvals), names(perm_pval_set1))],
      gene_symbols$external_gene_name[match(names(pvals), gene_symbols$ensembl_gene_id)])
```


```{r}
for (i in 1:2) {
  print(sig_pluri_ensg$external_gene_name[i])
  print(table(molecules_expressed_subset[rownames(molecules_expressed_subset) %in% sig_pluri_ensg$ensembl_gene_id[i]], anno_filter$individual) )
}
```


## Enrichment analysis of differential CV genes

I used CPDB (http://cpdb.molgen.mpg.de/) for over-representation gene set enrichment analysis. The enriched GO term are represented in WordCloud as follows.

![wordcloud](figure/cv-adjusted-summary-pois-expressed.Rmd/go-expressed-sig-wordcloud.png)



## Supplemental tables for the manuscript

Supplemental Table 2: Significant inter-individual variation genes

Output signficant genes

```{r, eval = FALSE}
sig_genes <- names(perm_pval_set1)[which(perm_pval_set1 < .0001)]
sig_genes_output <-
  data.frame(ensg = sig_genes,
             symbol = gene_symbols$external_gene_name[which(gene_symbols$ensembl_gene_id %in% sig_genes)],
             permute_pval = perm_pval_set1[which(perm_pval_set1 < .0001)])

write.table(sig_genes_output[order(sig_genes_output$permute_pval),],
            quote = FALSE,
            col.names = FALSE,
            row.names = FALSE,
            sep = "\t",
            file = "../data/sig-expressed-genes.txt")
```


Supplemental Table 3: Gene ontology over-representation analysis for significant genes

```
"figure/cv-adjusted-summary-pois-expressed.Rmd/go-expressed-sig-xls"
```


## Session information

```{r info}
sessionInfo()
```
