---
title: "Compare transcriptome-wide profile of CVs"
author: "Joyce Hsiao"
date: 2015-10-25
---

**Last updated:** `r Sys.Date()`

**Code version:** `r system("git log -1 --format='%H'", intern = TRUE)`

```{r chunk-options, include=FALSE}
source("chunk-options.R")

library(knitr)
opts_chunk$set(message = FALSE, warning = FALSE, eval = TRUE, 
               echo = TRUE)
```


## Objective

Compare transcriptome profile of cell-to-cell heterogeneity within individuals betwen batches and between individuals. 

[link]: http://jdblischak.github.io/singleCellSeq/analysis/cv-adjusted.html

For both within and between individuals comparisons, we will use the coefficients of variation normalized across all samples. 


## Set up

```{r, message=FALSE, warning=FALSE}
library("data.table")
library("dplyr")
library("limma")
library("edgeR")
library("ggplot2")
library("grid")
theme_set(theme_bw(base_size = 12))
source("functions.R")
```


## Prepare data

Input annotation of only QC-filtered single cells. Remove NA19098.r2

```{r}
anno_qc <- read.table("../data/annotation-filter.txt", header = TRUE,
                   stringsAsFactors = FALSE)
is_include <- anno_qc$batch != "NA19098.r2"
anno_qc_filter <- anno_qc[which(is_include), ]
```


Import endogeneous gene molecule counts that are QC-filtered, CPM-normalized, ERCC-normalized, and also processed to remove unwanted variation from batch effet. ERCC genes are removed from this file.

```{r}
molecules_ENSG <- read.table("../data/molecules-final.txt", header = TRUE, stringsAsFactors = FALSE)
molecules_ENSG <- molecules_ENSG[ , is_include]
```

Input moleclule counts before log2 CPM transformation. This file is used to compute percent zero-count cells per sample.

```{r}
molecules_sparse <- read.table("../data/molecules-filter.txt", header = TRUE, stringsAsFactors = FALSE)

molecules_sparse <- molecules_sparse[grep("ENSG", rownames(molecules_sparse)), ]
stopifnot( all.equal(rownames(molecules_ENSG), rownames(molecules_sparse)) )
```


## Compute normalized CV 

```{r}
source("../code/cv-functions.r")

# compute_cv() takes log2 counts as input
ENSG_cv <- compute_cv(counts = molecules_ENSG,
                           anno = anno_qc_filter)

ENSG_cv_adj <- normalize_cv(batch_cv = ENSG_cv, 
                                 counts = molecules_ENSG, 
                                 anno = anno_qc_filter)
```


## Compare transcriptome profile within individuals


### Within each individual

The CV normalization method employed here scales each sample individual with respect to 
data-wide coefficients of variation. Therefore, we would expect that the transcriptome-wide
profile relationship between samples (with or between individuals) to be consistent before 
versus after normalization.


* Before normalization

```{r}
print(ENSG_cv[[1]]$individual[1])
print(friedman.test( cbind(ENSG_cv[[1]]$cv, ENSG_cv[[2]]$cv) ) )

print(ENSG_cv[[3]]$individual[3])
print(friedman.test( cbind(ENSG_cv[[3]]$cv, ENSG_cv[[4]]$cv, ENSG_cv[[5]]$cv) ) )

print(ENSG_cv[[6]]$individual[6])
print(friedman.test( cbind(ENSG_cv[[6]]$cv, ENSG_cv[[7]]$cv, ENSG_cv[[8]]$cv) ) )
```

* After normalization

```{r}
print(ENSG_cv_adj[[1]]$individual[1])
print(friedman.test( cbind(ENSG_cv_adj[[1]]$log10cv2_adj, ENSG_cv_adj[[2]]$log10cv2_adj) ) )

print(ENSG_cv[[3]]$individual[3])
print(friedman.test( cbind(ENSG_cv_adj[[3]]$log10cv2_adj, 
                           ENSG_cv_adj[[4]]$log10cv2_adj, ENSG_cv_adj[[5]]$log10cv2_adj) ) )

print(ENSG_cv[[6]]$individual[6])
print(friedman.test( cbind(ENSG_cv_adj[[6]]$log10cv2_adj, 
                           ENSG_cv_adj[[7]]$log10cv2_adj, ENSG_cv_adj[[8]]$log10cv2_adj) ) )
```

*Plot CV before/after normalization

The normalization method we used is a non-linear transformation. The method aims to remove heterogeneity of variances across mean gene expression levels. 

```{r}
print(ENSG_cv[[6]]$individual[6])
plot(x = rank(log10((ENSG_cv_adj[[6]]$cv)^2)),
     y = rank(ENSG_cv_adj[[6]]$log10cv2_adj),
     xlab = "Before normalization", ylab = "After normalization",
     cex = .4, pch = 16, main = "Ranks of CV")
abline(0, 1, col = "red")

print(ENSG_cv[[6]]$individual[6])
plot(x = (log10((ENSG_cv_adj[[6]]$cv)^2)),
     y = (ENSG_cv_adj[[6]]$log10cv2_adj),
     xlab = "Before normalization", ylab = "After normalization",
     cex = .4, pch = 16, main = "log10 squared-CV")
abline(0, 1, col = "red")
```


### Between individual, adjusted CV

* PCA

Perform PCA for each individual. Then compare PCA gene loadings between individuals.

Use *prcomp* to perform PCA:
* output$x: loadings
* output$rotation: loadings
* output$center: variable means
* output$scale: variable standard deviation

```{r}
pca_individuals <- lapply(1:3, function(ii_individual) {
  if (ii_individual == 1) {
  df_ind <- cbind(ENSG_cv_adj$NA19098.r1$log10cv2_adj,
                  ENSG_cv_adj$NA19098.r3$log10cv2_adj ) 
  }
  if (ii_individual == 2) {
  df_ind <- cbind(ENSG_cv_adj$NA19101.r1$log10cv2_adj,
                  ENSG_cv_adj$NA19101.r2$log10cv2_adj,
                  ENSG_cv_adj$NA19101.r3$log10cv2_adj )
  }
  if (ii_individual == 3) {
  df_ind <- cbind(ENSG_cv_adj$NA19239.r1$log10cv2_adj,
                  ENSG_cv_adj$NA19239.r2$log10cv2_adj,
                  ENSG_cv_adj$NA19239.r3$log10cv2_adj )
  }
  pca_results <- prcomp( df_ind, retx = TRUE, center = TRUE, scale = TRUE)
  prop_var <- (pca_results$sdev^2) / sum(pca_results$sdev^2)
  
  list(loadings = pca_results$x,
       prop_var = prop_var)
})

sapply(pca_individuals, function(xx) xx$prop_var)

pca_all <- cbind(pca_individuals[[1]]$loadings[ , 1],
                 pca_individuals[[2]]$loadings[ , 1],
                 pca_individuals[[3]]$loadings[ , 1] )
                 
friedman_results <- friedman.test(pca_all)
friedman_results
```


* Mean of sample CV

Compute the mean of CVs of the samples.

```{r}
df_mean <- with(ENSG_cv_adj,
                cbind(rowMeans(cbind(NA19098.r1$log10cv2_adj,
                               NA19098.r3$log10cv2_adj) ),
                      rowMeans(cbind(ENSG_cv_adj$NA19101.r1$log10cv2_adj,
                               ENSG_cv_adj$NA19101.r2$log10cv2_adj,
                               ENSG_cv_adj$NA19101.r3$log10cv2_adj ) ),
                      rowMeans(cbind(ENSG_cv_adj$NA19239.r1$log10cv2_adj,
                               ENSG_cv_adj$NA19239.r2$log10cv2_adj,
                               ENSG_cv_adj$NA19239.r3$log10cv2_adj ) ) ) )

friedman_results <- friedman.test(df_mean)
friedman_results
```



### Between individual, unadjusted CV

* PCA


```{r}
pca_individuals <- lapply(1:3, function(ii_individual) {
  if (ii_individual == 1) {
  df_ind <- log10(cbind(ENSG_cv_adj$NA19098.r1$cv,
                        ENSG_cv_adj$NA19098.r3$cv )^2 ) 
  }
  if (ii_individual == 2) {
  df_ind <- log10(cbind(ENSG_cv_adj$NA19101.r1$cv,
                        ENSG_cv_adj$NA19101.r2$cv,
                        ENSG_cv_adj$NA19101.r3$cv )^2)
  }
  if (ii_individual == 3) {
  df_ind <- log10(cbind(ENSG_cv_adj$NA19239.r1$cv,
                        ENSG_cv_adj$NA19239.r2$cv,
                        ENSG_cv_adj$NA19239.r3$cv )^2 )
  }
  pca_results <- prcomp( df_ind, retx = TRUE, center = TRUE, scale = TRUE)
  prop_var <- (pca_results$sdev^2) / sum(pca_results$sdev^2)
  
  list(loadings = pca_results$x,
       prop_var = prop_var)
})

sapply(pca_individuals, function(xx) xx$prop_var)

pca_all <- cbind(pca_individuals[[1]]$loadings[ , 1],
                 pca_individuals[[2]]$loadings[ , 1],
                 pca_individuals[[3]]$loadings[ , 1] )
                 
friedman_results <- friedman.test(pca_all)
friedman_results
```


* Mean of sample CV

Compute the mean of CVs of the samples.

```{r}
df_mean <- with(ENSG_cv_adj,
              cbind(rowMeans(log10(cbind(NA19098.r1$cv,
                                        NA19098.r3$cv)^2) ),
                    rowMeans(log10(cbind(ENSG_cv_adj$NA19101.r1$cv,
                                         ENSG_cv_adj$NA19101.r2$cv,
                                         ENSG_cv_adj$NA19101.r3$cv )^2) ),
                    rowMeans(log10(cbind(ENSG_cv_adj$NA19239.r1$cv,
                                         ENSG_cv_adj$NA19239.r2$cv,
                                         ENSG_cv_adj$NA19239.r3$cv )^2) ) ) )

friedman_results <- friedman.test(df_mean)
friedman_results
```





## Session information

```{r info}
sessionInfo()
```
