<!DOCTYPE html>

<html xmlns="http://www.w3.org/1999/xhtml">

<head>

<meta charset="utf-8">
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="generator" content="pandoc" />

<meta name="author" content="Joyce Hsiao" />

<meta name="date" content="2015-10-29" />

<title>Profile individual adjusted CVs</title>

<script src="libs/jquery-1.11.0/jquery.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="libs/bootstrap-3.3.1/css/united.min.css" rel="stylesheet" />
<script src="libs/bootstrap-3.3.1/js/bootstrap.min.js"></script>
<script src="libs/bootstrap-3.3.1/shim/html5shiv.min.js"></script>
<script src="libs/bootstrap-3.3.1/shim/respond.min.js"></script>

<style type="text/css">

/* padding for bootstrap navbar */
body {
  padding-top: 50px;
  padding-bottom: 40px;
}


/* offset scroll position for anchor links (for fixed navbar)  */
.section h2 {
  padding-top: 55px;
  margin-top: -55px;
}
.section h3 {
  padding-top: 55px;
  margin-top: -55px;
}



/* don't use link color in navbar */
.dropdown-menu>li>a {
  color: black;
}

/* some padding for disqus */
#disqus_thread {
  margin-top: 45px;
}

</style>

<link rel="stylesheet" href="libs/font-awesome-4.1.0/css/font-awesome.min.css"/>

<style type="text/css">code{white-space: pre;}</style>
<link rel="stylesheet"
      href="libs/highlight/textmate.css"
      type="text/css" />
<script src="libs/highlight/highlight.js"></script>
<style type="text/css">
  pre:not([class]) {
    background-color: white;
  }
</style>
<script type="text/javascript">
if (window.hljs && document.readyState && document.readyState === "complete") {
   window.setTimeout(function() {
      hljs.initHighlighting();
   }, 0);
}
</script>



</head>

<body>

<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
code {
  color: inherit;
  background-color: rgba(0, 0, 0, 0.04);
}
img { 
  max-width:100%; 
  height: auto; 
}
</style>
<div class="container-fluid main-container">


<div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">singleCellSeq</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li><a href="index.html">Home</a></li>
        <li><a href="about.html">About</a></li>
        <li><a href="license.html">License</a></li>
        <li><a href="https://github.com/jdblischak/singleCellSeq">GitHub</a></li>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<div id="header">
<h1 class="title">Profile individual adjusted CVs</h1>
<h4 class="author"><em>Joyce Hsiao</em></h4>
<h4 class="date"><em>2015-10-29</em></h4>
</div>

<div id="TOC">
<ul>
<li><a href="#objective">Objective</a></li>
<li><a href="#set-up">Set up</a></li>
<li><a href="#prepare-data">Prepare data</a></li>
<li><a href="#compute-normalized-cv">Compute normalized CV</a></li>
<li><a href="#extreme-genes">Extreme genes</a></li>
<li><a href="#annotation">Annotation</a><ul>
<li><a href="#gostats">GOstats</a></li>
<li><a href="#extract-go-terms">Extract GO terms</a></li>
</ul></li>
<li><a href="#export-files">Export files</a></li>
<li><a href="#session-information">Session information</a></li>
</ul>
</div>

<p><strong>Last updated:</strong> 2015-12-11</p>
<p><strong>Code version:</strong> 7a25b2aef5c8329d4b5394cce595bbfc2b5b0c3e</p>
<div id="objective" class="section level2">
<h2>Objective</h2>
<p>We rank each individual’s adjusted CV (Deviation-from-the-Median; DM) and compare genes’ ranks in adjusted CV across individuals.</p>
<p>Method: Compute CV for each individual and adjust the CV to remove its dependency on mean gene expression. We will follow [Kolodziejczyk et al., 2015] to compute the adjusted coefficient of variation (referred to as Deviation-from-the-Median in [Kolodziejczyk et al., 2015] and also in our related documentations).</p>
<p><a href="http://dx.doi.org/10.1016/j.stem.2015.09.011">Kolodziejczyk et al., 2015</a></p>
</div>
<div id="set-up" class="section level2">
<h2>Set up</h2>
<pre class="r"><code>library(&quot;data.table&quot;)
library(&quot;dplyr&quot;)
library(&quot;limma&quot;)
library(&quot;edgeR&quot;)
library(&quot;ggplot2&quot;)
library(&quot;grid&quot;)
theme_set(theme_bw(base_size = 12))
source(&quot;functions.R&quot;)</code></pre>
</div>
<div id="prepare-data" class="section level2">
<h2>Prepare data</h2>
<p>Input annotation of only QC-filtered single cells. Remove NA19098.r2</p>
<pre class="r"><code>anno_qc &lt;- read.table(&quot;../data/annotation-filter.txt&quot;, header = TRUE,
                   stringsAsFactors = FALSE)
is_include &lt;- anno_qc$batch != &quot;NA19098.r2&quot;
anno_qc_filter &lt;- anno_qc[which(is_include), ]</code></pre>
<p>Import endogeneous gene molecule counts that are QC-filtered, CPM-normalized, ERCC-normalized, and also processed to remove unwanted variation from batch effet. ERCC genes are removed from this file.</p>
<pre class="r"><code>molecules_ENSG &lt;- read.table(&quot;../data/molecules-final.txt&quot;, header = TRUE, stringsAsFactors = FALSE)
molecules_ENSG &lt;- molecules_ENSG[ , is_include]</code></pre>
<p>Input moleclule counts before log2 CPM transformation. This file is used to compute percent zero-count cells per sample.</p>
<pre class="r"><code>molecules_sparse &lt;- read.table(&quot;../data/molecules-filter.txt&quot;, header = TRUE, stringsAsFactors = FALSE)

molecules_sparse &lt;- molecules_sparse[grep(&quot;ENSG&quot;, rownames(molecules_sparse)), ]
stopifnot( all.equal(rownames(molecules_ENSG), rownames(molecules_sparse)) )</code></pre>
</div>
<div id="compute-normalized-cv" class="section level2">
<h2>Compute normalized CV</h2>
<p>We compute squared CV across cells for each individual and then for each individual CV profile, account for mean dependency by computing distance with respect to the data-wide coefficient variation on the log10 scale.</p>
<pre class="r"><code>source(&quot;../code/cv-functions.r&quot;)

ENSG_cv &lt;- compute_cv(log2counts = molecules_ENSG,
                      grouping_vector = anno_qc_filter$individual)

ENSG_cv_adj &lt;- normalize_cv(group_cv = ENSG_cv, 
                            log2counts = molecules_ENSG, 
                            anno = anno_qc_filter)</code></pre>
</div>
<div id="extreme-genes" class="section level2">
<h2>Extreme genes</h2>
<p>*Top/bottom 50</p>
<pre class="r"><code>extreme_genes_50 &lt;- lapply(ENSG_cv_adj, function(xx) {
  low_mean &lt;- rank(xx$mean) &lt; 50
  high_mean &lt;- rank(xx$mean) &gt; (length(xx$mean) - 50)
  low_cv &lt;-  rank(xx$log10cv2_adj) &lt; 50
  high_cv &lt;- rank(xx$log10cv2_adj) &gt; (length(xx$log10cv2_adj) - 50)
  res &lt;- data.frame(high_mean = high_mean,
                    low_mean = low_mean,
                    high_cv = high_cv,
                    low_cv = low_cv)
  rownames(res) &lt;- rownames(ENSG_cv_adj[[1]])
  res
})</code></pre>
<p>*Top/bottom 200</p>
<pre class="r"><code>extreme_genes_200 &lt;- lapply(ENSG_cv_adj, function(xx) {
  low_mean &lt;- rank(xx$mean) &lt; 200
  high_mean &lt;- rank(xx$mean) &gt; (length(xx$mean) - 200)
  low_cv &lt;-  rank(xx$log10cv2_adj) &lt; 200
  high_cv &lt;- rank(xx$log10cv2_adj) &gt; (length(xx$log10cv2_adj) - 200)
  res &lt;- data.frame(high_mean = high_mean,
                    low_mean = low_mean,
                    high_cv = high_cv,
                    low_cv = low_cv)
  rownames(res) &lt;- rownames(ENSG_cv_adj[[1]])
  res
})</code></pre>
<p>*Output gene symbols.</p>
<pre class="r"><code>gene_info &lt;- read.table(&quot;../data/gene-info.txt&quot;, header = TRUE, sep = &quot;\t&quot;, 
                        stringsAsFactors = FALSE, quote = &quot;&quot;)
str(gene_info)</code></pre>
<pre><code>&#39;data.frame&#39;:   10483 obs. of  5 variables:
 $ ensembl_gene_id   : chr  &quot;ENSG00000000003&quot; &quot;ENSG00000000005&quot; &quot;ENSG00000000419&quot; &quot;ENSG00000000457&quot; ...
 $ chromosome_name   : chr  &quot;X&quot; &quot;X&quot; &quot;20&quot; &quot;1&quot; ...
 $ external_gene_name: chr  &quot;TSPAN6&quot; &quot;TNMD&quot; &quot;DPM1&quot; &quot;SCYL3&quot; ...
 $ transcript_count  : int  3 2 7 5 10 4 12 2 13 8 ...
 $ description       : chr  &quot;tetraspanin 6 [Source:HGNC Symbol;Acc:11858]&quot; &quot;tenomodulin [Source:HGNC Symbol;Acc:17757]&quot; &quot;dolichyl-phosphate mannosyltransferase polypeptide 1, catalytic subunit [Source:HGNC Symbol;Acc:3005]&quot; &quot;SCY1-like 3 (S. cerevisiae) [Source:HGNC Symbol;Acc:19285]&quot; ...</code></pre>
<p>Gene symbols ordered by CV</p>
<pre class="r"><code>adj_cv_order &lt;- with(gene_info, data.frame(
                          NA19098 = external_gene_name[
                            order(ENSG_cv_adj$NA19098$log10cv2_adj, decreasing = TRUE)],
                          NA19101 = external_gene_name[
                            order(ENSG_cv_adj$NA19101$log10cv2_adj, decreasing = TRUE)],
                          NA19239 = external_gene_name[
                            order(ENSG_cv_adj$NA19239$log10cv2_adj, decreasing = TRUE)], 
                          stringsAsFactors = FALSE) )

head(adj_cv_order)</code></pre>
<pre><code>   NA19098 NA19101 NA19239
1    HSPA4 GNPNAT1   VTI1A
2    TTC26   ADCY1 GNPNAT1
3  GNPNAT1   ADNP2   ADCY1
4    ADCY1   TTC26  ZNF350
5 KIAA1033    CBY1   ADNP2
6    ADNP2   TAF1D PITPNM1</code></pre>
<pre class="r"><code>if(!file.exists(&quot;../data/gene-order-high-to-low-cv.txt&quot;)) {
write.table(adj_cv_order, file = &quot;../data/gene-order-high-to-low-cv.txt&quot;,
            sep = &quot;\t&quot;,
            col.names = TRUE, row.names = FALSE, quote = FALSE)
}</code></pre>
</div>
<div id="annotation" class="section level2">
<h2>Annotation</h2>
<div id="gostats" class="section level3">
<h3>GOstats</h3>
<pre class="r"><code>require(Humanzee)
if (file.exists(&quot;rda/cv-adjusted-profile/go-cv-NA19098.rda&quot;)) {
  load(&quot;rda/cv-adjusted-profile/go-cv-NA19098.rda&quot;)  
} else {
go_cv_high &lt;- GOtest(my_ensembl_gene_universe = rownames(molecules_ENSG),
                      my_ensembl_gene_test = rownames(molecules_ENSG)[extreme_genes_50[[1]]$high_cv],
                      pval_cutoff = 1, ontology=c(&quot;BP&quot;) )
go_cv_low &lt;- GOtest(my_ensembl_gene_universe = rownames(molecules_ENSG),
                      my_ensembl_gene_test = rownames(molecules_ENSG)[extreme_genes_50[[1]]$low_cv],
                      pval_cutoff = 1, ontology=c(&quot;BP&quot;) )

go_mean_high &lt;- GOtest(my_ensembl_gene_universe = rownames(molecules_ENSG),
                      my_ensembl_gene_test = rownames(molecules_ENSG)[extreme_genes_50[[1]]$high_mean],
                      pval_cutoff = 1, ontology=c(&quot;BP&quot;) )
go_mean_low &lt;- GOtest(my_ensembl_gene_universe = rownames(molecules_ENSG),
                      my_ensembl_gene_test = rownames(molecules_ENSG)[extreme_genes_50[[1]]$low_mean],
                      pval_cutoff = 1, ontology=c(&quot;BP&quot;) )

save(ENSG_cv_adj, extreme_genes_50,
     go_cv_high, go_cv_low, go_mean_high, go_mean_low,
     file = &quot;rda/cv-adjusted-profile/go-cv-NA19098.rda&quot;)
}

if (file.exists(&quot;rda/cv-adjusted-profile/go-cv-NA19101.rda&quot;)) {
  load(&quot;rda/cv-adjusted-profile/go-cv-NA19101.rda&quot;)  
} else {
go_cv_high &lt;- GOtest(my_ensembl_gene_universe = rownames(molecules_ENSG),
                      my_ensembl_gene_test = rownames(molecules_ENSG)[extreme_genes_50[[2]]$high_cv],
                      pval_cutoff = 1, ontology=c(&quot;BP&quot;) )
go_cv_low &lt;- GOtest(my_ensembl_gene_universe = rownames(molecules_ENSG),
                      my_ensembl_gene_test = rownames(molecules_ENSG)[extreme_genes_50[[2]]$low_cv],
                      pval_cutoff = 1, ontology=c(&quot;BP&quot;) )

go_mean_high &lt;- GOtest(my_ensembl_gene_universe = rownames(molecules_ENSG),
                      my_ensembl_gene_test = rownames(molecules_ENSG)[extreme_genes_50[[2]]$high_mean],
                      pval_cutoff = 1, ontology=c(&quot;BP&quot;) )
go_mean_low &lt;- GOtest(my_ensembl_gene_universe = rownames(molecules_ENSG),
                      my_ensembl_gene_test = rownames(molecules_ENSG)[extreme_genes_50[[2]]$low_mean],
                      pval_cutoff = 1, ontology=c(&quot;BP&quot;) )
save(ENSG_cv_adj, extreme_genes_50,
     go_cv_high, go_cv_low, go_mean_high, go_mean_low,
     file = &quot;rda/cv-adjusted-profile/go-cv-NA19101.rda&quot;)
}


if (file.exists(&quot;rda/cv-adjusted-profile/go-cv-NA19239.rda&quot;)) {
  load(&quot;rda/cv-adjusted-profile/go-cv-NA19239.rda&quot;)  
} else {
go_cv_high &lt;- GOtest(my_ensembl_gene_universe = rownames(molecules_ENSG),
                      my_ensembl_gene_test = rownames(molecules_ENSG)[extreme_genes_50[[3]]$high_cv],
                      pval_cutoff = 1, ontology=c(&quot;BP&quot;) )
go_cv_low &lt;- GOtest(my_ensembl_gene_universe = rownames(molecules_ENSG),
                      my_ensembl_gene_test = rownames(molecules_ENSG)[extreme_genes_50[[3]]$low_cv],
                      pval_cutoff = 1, ontology=c(&quot;BP&quot;) )

go_mean_high &lt;- GOtest(my_ensembl_gene_universe = rownames(molecules_ENSG),
                      my_ensembl_gene_test = rownames(molecules_ENSG)[extreme_genes_50[[3]]$high_mean],
                      pval_cutoff = 1, ontology=c(&quot;BP&quot;) )
go_mean_low &lt;- GOtest(my_ensembl_gene_universe = rownames(molecules_ENSG),
                      my_ensembl_gene_test = rownames(molecules_ENSG)[extreme_genes_50[[3]]$low_mean],
                      pval_cutoff = 1, ontology=c(&quot;BP&quot;) )
save(ENSG_cv_adj, extreme_genes_50,
     go_cv_low, go_cv_high,
     go_mean_high, go_mean_low, file = &quot;rda/cv-adjusted-profile/go-cv-NA19239.rda&quot;)
}</code></pre>
</div>
<div id="extract-go-terms" class="section level3">
<h3>Extract GO terms</h3>
<ul>
<li>NA19098</li>
</ul>
<pre class="r"><code>load(file = &quot;rda/cv-adjusted-profile/go-cv-NA19098.rda&quot;)
goterms_cv_high &lt;- summary(go_cv_high$GO$BP, pvalue = .05)
goterms_cv_high &lt;- data.frame(ID = goterms_cv_high[[1]],
                             Pvalue = goterms_cv_high[[2]],
                             Terms = goterms_cv_high[[7]])
goterms_cv_high &lt;- goterms_cv_high[order(goterms_cv_high$Pvalue), ]
head(goterms_cv_high)

goterms_cv_low &lt;- summary(go_cv_low$GO$BP, pvalue = .05)
goterms_cv_low &lt;- data.frame(ID = goterms_cv_low[[1]],
                             Pvalue = goterms_cv_low[[2]],
                             Terms = goterms_cv_low[[7]])
goterms_cv_low &lt;- goterms_cv_high[order(goterms_cv_low$Pvalue), ]
head(goterms_cv_low)</code></pre>
<ul>
<li>NA19101</li>
</ul>
<pre class="r"><code>load(file = &quot;rda/cv-adjusted-profile/go-cv-NA19101.rda&quot;)
goterms_cv_high &lt;- summary(go_cv_high$GO$BP, pvalue = .05)
goterms_cv_high &lt;- data.frame(ID = goterms_cv_high[[1]],
                             Pvalue = goterms_cv_high[[2]],
                             Terms = goterms_cv_high[[7]])
goterms_cv_high &lt;- goterms_cv_high[order(goterms_cv_high$Pvalue), ]
head(goterms_cv_high)

goterms_cv_low &lt;- summary(go_cv_low$GO$BP, pvalue = .05)
goterms_cv_low &lt;- data.frame(ID = goterms_cv_low[[1]],
                             Pvalue = goterms_cv_low[[2]],
                             Terms = goterms_cv_low[[7]])
goterms_cv_low &lt;- goterms_cv_high[order(goterms_cv_low$Pvalue), ]
head(goterms_cv_low)</code></pre>
<ul>
<li>NA19239</li>
</ul>
<pre class="r"><code>load(file = &quot;rda/cv-adjusted-profile/go-cv-NA19239.rda&quot;)
goterms_cv_high &lt;- summary(go_cv_high$GO$BP, pvalue = .05)
goterms_cv_high &lt;- data.frame(ID = goterms_cv_high[[1]],
                             Pvalue = goterms_cv_high[[2]],
                             Terms = goterms_cv_high[[7]])
goterms_cv_high &lt;- goterms_cv_high[order(goterms_cv_high$Pvalue), ]
head(goterms_cv_high)

goterms_cv_low &lt;- summary(go_cv_low$GO$BP, pvalue = .05)
goterms_cv_low &lt;- data.frame(ID = goterms_cv_low[[1]],
                             Pvalue = goterms_cv_low[[2]],
                             Terms = goterms_cv_low[[7]])
goterms_cv_low &lt;- goterms_cv_high[order(goterms_cv_low$Pvalue), ]
head(goterms_cv_low)</code></pre>
</div>
</div>
<div id="export-files" class="section level2">
<h2>Export files</h2>
<pre class="r"><code>library(&quot;biomaRt&quot;)
ensembl &lt;- useMart(host = &quot;grch37.ensembl.org&quot;,
                   biomart = &quot;ENSEMBL_MART_ENSEMBL&quot;,
                   dataset = &quot;hsapiens_gene_ensembl&quot;)

info &lt;- getBM(attributes = c(&quot;ensembl_gene_id&quot;, &quot;chromosome_name&quot;,
                                                   &quot;external_gene_name&quot;, &quot;transcript_count&quot;,
                                                   &quot;description&quot;),
                                    filters = &quot;ensembl_gene_id&quot;,
                                    values = rownames(molecules_ENSG[sig_cv, ]),
                                    mart = ensembl)

write.table(extreme_genes_50$NA19098$high_cv, quote = FALSE)</code></pre>
</div>
<div id="session-information" class="section level2">
<h2>Session information</h2>
<pre class="r"><code>sessionInfo()</code></pre>
<pre><code>R version 3.2.1 (2015-06-18)
Platform: x86_64-apple-darwin13.4.0 (64-bit)
Running under: OS X 10.10.5 (Yosemite)

locale:
[1] en_US.UTF-8/en_US.UTF-8/en_US.UTF-8/C/en_US.UTF-8/en_US.UTF-8

attached base packages:
[1] grid      stats     graphics  grDevices utils     datasets  methods  
[8] base     

other attached packages:
[1] zoo_1.7-12       ggplot2_1.0.1    edgeR_3.10.5     limma_3.24.15   
[5] dplyr_0.4.3      data.table_1.9.6 knitr_1.11      

loaded via a namespace (and not attached):
 [1] Rcpp_0.12.2      magrittr_1.5     MASS_7.3-45      munsell_0.4.2   
 [5] lattice_0.20-33  colorspace_1.2-6 R6_2.1.1         stringr_1.0.0   
 [9] plyr_1.8.3       tools_3.2.1      parallel_3.2.1   gtable_0.1.2    
[13] DBI_0.3.1        htmltools_0.2.6  yaml_2.1.13      assertthat_0.1  
[17] digest_0.6.8     reshape2_1.4.1   formatR_1.2.1    evaluate_0.8    
[21] rmarkdown_0.8.1  stringi_1.0-1    scales_0.3.0     chron_2.3-47    
[25] proto_0.3-10    </code></pre>
</div>


<!-- some extra javascript for older browsers -->
<script type="text/javascript" src="libs/polyfill.js"></script>

<script>

// manage active state of menu based on current page
$(document).ready(function () {

    // active menu
    href = window.location.pathname
    href = href.substr(href.lastIndexOf('/') + 1)
    $('a[href="' + href + '"]').parent().addClass('active');

    // manage active menu header
    if (href.startsWith('authoring_'))
      $('a[href="' + 'authoring' + '"]').parent().addClass('active');
    else if (href.endsWith('_format.html'))
      $('a[href="' + 'formats' + '"]').parent().addClass('active');
    else if (href.startsWith('developer_'))
      $('a[href="' + 'developer' + '"]').parent().addClass('active');

});

</script>

</div>

<script>

// add bootstrap table styles to pandoc tables
$(document).ready(function () {
  $('tr.header').parent('thead').parent('table').addClass('table table-condensed');
});

</script>


</body>
</html>
