<!DOCTYPE html>

<html xmlns="http://www.w3.org/1999/xhtml">

<head>

<meta charset="utf-8">
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="generator" content="pandoc" />

<meta name="author" content="Joyce Hsiao" />

<meta name="date" content="2015-10-16" />

<title>Compare Coefficient of variation between individuals</title>

<script src="libs/jquery-1.11.0/jquery.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="libs/bootstrap-3.3.1/css/united.min.css" rel="stylesheet" />
<script src="libs/bootstrap-3.3.1/js/bootstrap.min.js"></script>
<script src="libs/bootstrap-3.3.1/shim/html5shiv.min.js"></script>
<script src="libs/bootstrap-3.3.1/shim/respond.min.js"></script>

<style type="text/css">

/* padding for bootstrap navbar */
body {
  padding-top: 50px;
  padding-bottom: 40px;
}


/* offset scroll position for anchor links (for fixed navbar)  */
.section h2 {
  padding-top: 55px;
  margin-top: -55px;
}
.section h3 {
  padding-top: 55px;
  margin-top: -55px;
}



/* don't use link color in navbar */
.dropdown-menu>li>a {
  color: black;
}

/* some padding for disqus */
#disqus_thread {
  margin-top: 45px;
}

</style>

<link rel="stylesheet" href="libs/font-awesome-4.1.0/css/font-awesome.min.css"/>

<style type="text/css">code{white-space: pre;}</style>
<link rel="stylesheet"
      href="libs/highlight/textmate.css"
      type="text/css" />
<script src="libs/highlight/highlight.js"></script>
<style type="text/css">
  pre:not([class]) {
    background-color: white;
  }
</style>
<script type="text/javascript">
if (window.hljs && document.readyState && document.readyState === "complete") {
   window.setTimeout(function() {
      hljs.initHighlighting();
   }, 0);
}
</script>



</head>

<body>

<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
code {
  color: inherit;
  background-color: rgba(0, 0, 0, 0.04);
}
img { 
  max-width:100%; 
  height: auto; 
}
</style>
<div class="container-fluid main-container">


<div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">singleCellSeq</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li><a href="index.html">Home</a></li>
        <li><a href="about.html">About</a></li>
        <li><a href="license.html">License</a></li>
        <li><a href="https://github.com/jdblischak/singleCellSeq">GitHub</a></li>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<div id="header">
<h1 class="title">Compare Coefficient of variation between individuals</h1>
<h4 class="author"><em>Joyce Hsiao</em></h4>
<h4 class="date"><em>2015-10-16</em></h4>
</div>

<div id="TOC">
<ul>
<li><a href="#objective">Objective</a></li>
<li><a href="#set-up">Set up</a></li>
<li><a href="#prepare-data">Prepare data</a></li>
<li><a href="#normalize-coefficient-of-variation">Normalize coefficient of variation</a></li>
<li><a href="#limma">limma</a></li>
<li><a href="#overlapy-with-pluripotency-genes">Overlapy with pluripotency genes</a></li>
<li><a href="#overlap-with-cell-cycle-genes">Overlap with cell-cycle genes</a><ul>
<li><a href="#gene-annotations">Gene annotations</a></li>
</ul></li>
<li><a href="#session-information">Session information</a></li>
</ul>
</div>

<p><strong>Last updated:</strong> 2015-10-21</p>
<p><strong>Code version:</strong> 78afeddbcf288d337c42915e95f0d9a578030a08</p>
<div id="objective" class="section level2">
<h2>Objective</h2>
<p>Previously, we used the entire set of data including all individuals and all batches to perform normalization of coefficient of variations (<a href="http://jdblischak.github.io/singleCellSeq/analysis/cv-adjusted.html">link1</a>) and performed</p>
<p>In this document, we wil perform per gene differential analysis of coefficients of variations using only coefficients of variation from ENSG genes and also from QC-approved batches (no NA19098.r2).</p>
</div>
<div id="set-up" class="section level2">
<h2>Set up</h2>
<pre class="r"><code>library(&quot;data.table&quot;)
library(&quot;dplyr&quot;)
library(&quot;limma&quot;)
library(&quot;edgeR&quot;)
library(&quot;ggplot2&quot;)
library(&quot;grid&quot;)
library(&quot;zoo&quot;)
theme_set(theme_bw(base_size = 12))
source(&quot;functions.R&quot;)</code></pre>
</div>
<div id="prepare-data" class="section level2">
<h2>Prepare data</h2>
<p>Input annotation of only QC-filtered single cells</p>
<pre class="r"><code>anno_qc &lt;- read.table(&quot;../data/annotation-filter.txt&quot;, header = TRUE,
                   stringsAsFactors = FALSE)
head(anno_qc)</code></pre>
<pre><code>  individual replicate well      batch      sample_id
1    NA19098        r1  A01 NA19098.r1 NA19098.r1.A01
2    NA19098        r1  A02 NA19098.r1 NA19098.r1.A02
3    NA19098        r1  A04 NA19098.r1 NA19098.r1.A04
4    NA19098        r1  A05 NA19098.r1 NA19098.r1.A05
5    NA19098        r1  A06 NA19098.r1 NA19098.r1.A06
6    NA19098        r1  A07 NA19098.r1 NA19098.r1.A07</code></pre>
<p>Remove NA19098.r2</p>
<pre class="r"><code>is_include &lt;- anno_qc$batch != &quot;NA19098.r2&quot;
anno_qc_filter &lt;- anno_qc[which(is_include), ]</code></pre>
<p>Import endogeneous gene molecule counts that are QC-filtered, CPM-normalized, ERCC-normalized, and also processed to remove unwanted variation from batch effet. ERCC genes are removed from this file.</p>
<pre class="r"><code>molecules_ENSG &lt;- read.table(&quot;../data/molecules-final.txt&quot;, header = TRUE, stringsAsFactors = FALSE)
molecules_ENSG &lt;- molecules_ENSG[ , is_include]</code></pre>
<p>Input moleclule counts before log2 CPM transformation. This file is used to compute percent zero-count cells per sample.</p>
<pre class="r"><code>molecules_sparse &lt;- read.table(&quot;../data/molecules-filter.txt&quot;, header = TRUE, stringsAsFactors = FALSE)

molecules_sparse &lt;- molecules_sparse[grep(&quot;ENSG&quot;, rownames(molecules_sparse)), ]
stopifnot( all.equal(rownames(molecules_ENSG), rownames(molecules_sparse)) )</code></pre>
</div>
<div id="normalize-coefficient-of-variation" class="section level2">
<h2>Normalize coefficient of variation</h2>
<p>Compute coefficient of variation.</p>
<pre class="r"><code># Compute CV and mean of normalized molecule counts (take 2^(log2-normalized count))

molecules_cv_batch_ENSG &lt;- 
  lapply(1:length(unique(anno_qc_filter$batch)), function(per_batch) {
      molecules_per_batch &lt;- 2^molecules_ENSG[ , unique(anno_qc_filter$batch) == unique(anno_qc_filter$batch)[per_batch] ]
      mean_per_gene &lt;- apply(molecules_per_batch, 1, mean, na.rm = TRUE)
      sd_per_gene &lt;- apply(molecules_per_batch, 1, sd, na.rm = TRUE)
      cv_per_gene &lt;- data.frame(mean = mean_per_gene,
                                sd = sd_per_gene,
                                cv = sd_per_gene/mean_per_gene)
      rownames(cv_per_gene) &lt;- rownames(molecules_ENSG)
  
      # cv_per_gene &lt;- cv_per_gene[rowSums(is.na(cv_per_gene)) == 0, ]
      cv_per_gene$batch &lt;- unique(anno_qc_filter$batch)[per_batch]
      
      # Add sparsity percent
      molecules_sparse_per_batch &lt;- molecules_sparse[ , unique(anno_qc_filter$batch) == unique(anno_qc_filter$batch)[per_batch]]
      cv_per_gene$sparse &lt;- rowMeans(as.matrix(molecules_sparse_per_batch) == 0)
        
      return(cv_per_gene)
      }) 
names(molecules_cv_batch_ENSG) &lt;- unique(anno_qc_filter$batch)

sapply(molecules_cv_batch_ENSG, dim)</code></pre>
<pre><code>     NA19098.r1 NA19098.r3 NA19101.r1 NA19101.r2 NA19101.r3 NA19239.r1
[1,]      10483      10483      10483      10483      10483      10483
[2,]          5          5          5          5          5          5
     NA19239.r2 NA19239.r3
[1,]      10483      10483
[2,]          5          5</code></pre>
<p>Merge summary data.frames.</p>
<pre class="r"><code>df_ENSG &lt;- do.call(rbind, molecules_cv_batch_ENSG)</code></pre>
<p>Compute rolling medians across all samples.</p>
<pre class="r"><code># Compute a data-wide coefficient of variation on CPM normalized counts.
data_cv_ENSG &lt;- apply(2^molecules_ENSG, 1, sd)/apply(2^molecules_ENSG, 1, mean)

# Order of genes by mean expression levels
order_gene &lt;- order(apply(2^molecules_ENSG, 1, mean))

# Rolling medians of log10 squared CV by mean expression levels
roll_medians &lt;- rollapply(log10(data_cv_ENSG^2)[order_gene], width = 50, by = 25,
                         FUN = median, fill = list(&quot;extend&quot;, &quot;extend&quot;, &quot;NA&quot;) )
ii_na &lt;- which( is.na(roll_medians) )
roll_medians[ii_na] &lt;- median( log10(data_cv_ENSG^2)[order_gene][ii_na] )

names(roll_medians) &lt;- rownames(molecules_ENSG)[order_gene]


# re-order rolling medians
reorder_gene &lt;- match(rownames(molecules_ENSG), names(roll_medians) )
roll_medians &lt;- roll_medians[ reorder_gene ]

stopifnot( all.equal(names(roll_medians), rownames(molecules_ENSG) ) )</code></pre>
<p>Compute adjusted coefficient of variation.</p>
<pre class="r"><code># adjusted coefficient of variation on log10 scale
log10cv2_adj_ENSG &lt;- 
  lapply(1:length(molecules_cv_batch_ENSG), function(per_batch) {
    foo &lt;- log10(molecules_cv_batch_ENSG[[per_batch]]$cv^2) - roll_medians
    return(foo)
})
df_ENSG$log10cv2_adj_ENSG &lt;- do.call(c, log10cv2_adj_ENSG)</code></pre>
</div>
<div id="limma" class="section level2">
<h2>limma</h2>
<p>We use the <em>lmFit</em> function in the limma package to fit linear models comparing coefficients of variations between individuals for all genes. <em>lmFit</em> provides a fast algorithm for least-square estimation, when design matrix is the same for all genes.</p>
<pre class="r"><code>library(limma)

df_limma &lt;- matrix(df_ENSG$log10cv2_adj_ENSG, 
                      nrow = nrow(molecules_ENSG), ncol = 8, byrow = FALSE)

design &lt;- data.frame(individual = factor(rep(unique(anno_qc_filter$individual), each = 3) ),
                     rep = factor(rep(c(1:3), times = 3)) )
design &lt;- design[ with(design, !(individual == &quot;NA19098&quot; &amp; rep == &quot;2&quot;)), ]

colnames(df_limma) &lt;- with(design, paste0(individual, rep))

fit_limma &lt;- lmFit(df_limma, design = model.matrix( ~ individual, data = design))                      
fit_limma &lt;- eBayes(fit_limma)</code></pre>
<p>False discover control adjutment.</p>
<pre class="r"><code>F.p.adj &lt;- p.adjust(fit_limma$F.p.value, method = &quot;fdr&quot;)</code></pre>
<p>Cutoffs</p>
<pre class="r"><code>df_cuts &lt;- data.frame(cuts = c(.001, .01, .05, .1, .15, .2))
df_cuts$sig_count &lt;- sapply(1:6, function(per_cut) {
                    sum(F.p.adj &lt; df_cuts$cuts[per_cut] )
                    })
df_cuts</code></pre>
<pre><code>   cuts sig_count
1 0.001        18
2 0.010        42
3 0.050       103
4 0.100       177
5 0.150       328
6 0.200       464</code></pre>
</div>
<div id="overlapy-with-pluripotency-genes" class="section level2">
<h2>Overlapy with pluripotency genes</h2>
<p>Input pluripotency genes. A list of 27 pluripotency genes used to demonstrate iPSC heterogeneity in Narshinh2011 Gene ID conversion was done by using the DAVID <a href="http://david.abcc.ncifcrf.gov" class="uri">http://david.abcc.ncifcrf.gov</a></p>
<pre class="r"><code>pluripotency_genes &lt;- read.table(&quot;../data/pluripotency-genes.txt&quot;, header = TRUE, sep=&quot;\t&quot;)</code></pre>
<pre class="r"><code>pluripotent_count &lt;- lapply(1:nrow(df_cuts), function(per_cut) {
    is_pluripotent &lt;- rownames(molecules_ENSG)[which(F.p.adj&lt; df_cuts$cuts[per_cut])] %in% pluripotency_genes$To
    sum(is_pluripotent)
  })
pluripotent_count &lt;- do.call(rbind, pluripotent_count)

rownames(pluripotent_count) &lt;- df_cuts$cuts
kable(pluripotent_count)</code></pre>
<table>
<tbody>
<tr class="odd">
<td align="left">0.001</td>
<td align="right">0</td>
</tr>
<tr class="even">
<td align="left">0.01</td>
<td align="right">0</td>
</tr>
<tr class="odd">
<td align="left">0.05</td>
<td align="right">0</td>
</tr>
<tr class="even">
<td align="left">0.1</td>
<td align="right">0</td>
</tr>
<tr class="odd">
<td align="left">0.15</td>
<td align="right">0</td>
</tr>
<tr class="even">
<td align="left">0.2</td>
<td align="right">1</td>
</tr>
</tbody>
</table>
</div>
<div id="overlap-with-cell-cycle-genes" class="section level2">
<h2>Overlap with cell-cycle genes</h2>
<p>Input cell cycle gene Gene sets reflecting 5 cell cycle phases were taken from Table S2 of Macosko2015 Gene ID conversion was done by using the DAVID <a href="http://david.abcc.ncifcrf.gov" class="uri">http://david.abcc.ncifcrf.gov</a></p>
<pre class="r"><code>cell_cycle_genes &lt;- read.table(&quot;../data/cellcyclegenes.txt&quot;, header = TRUE, sep=&quot;\t&quot;)</code></pre>
<pre class="r"><code>cell_cycle_count &lt;- lapply(1:nrow(df_cuts), function(per_cut) {
  per_cycle &lt;- sapply(1:ncol(cell_cycle_genes), function(per_cycle) {
        is_cell_cycle &lt;- rownames(molecules_ENSG)[which(F.p.adj&lt; df_cuts$cuts[per_cut])] %in% cell_cycle_genes[[per_cycle]]
        sum(is_cell_cycle)
  })
})
cell_cycle_count &lt;- do.call(rbind, cell_cycle_count)

colnames(cell_cycle_count) &lt;- colnames(cell_cycle_genes)
rownames(cell_cycle_count) &lt;- df_cuts$cuts
cell_cycle_count &lt;- cbind(df_cuts, cell_cycle_count)

kable(cell_cycle_count)</code></pre>
<table>
<thead>
<tr class="header">
<th align="left"></th>
<th align="right">cuts</th>
<th align="right">sig_count</th>
<th align="right">G1.S</th>
<th align="right">S</th>
<th align="right">G2.M</th>
<th align="right">M</th>
<th align="right">M.G1</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td align="left">0.001</td>
<td align="right">0.001</td>
<td align="right">18</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">1</td>
<td align="right">0</td>
<td align="right">0</td>
</tr>
<tr class="even">
<td align="left">0.01</td>
<td align="right">0.010</td>
<td align="right">42</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">2</td>
<td align="right">1</td>
<td align="right">1</td>
</tr>
<tr class="odd">
<td align="left">0.05</td>
<td align="right">0.050</td>
<td align="right">103</td>
<td align="right">0</td>
<td align="right">2</td>
<td align="right">3</td>
<td align="right">1</td>
<td align="right">1</td>
</tr>
<tr class="even">
<td align="left">0.1</td>
<td align="right">0.100</td>
<td align="right">177</td>
<td align="right">0</td>
<td align="right">3</td>
<td align="right">3</td>
<td align="right">1</td>
<td align="right">2</td>
</tr>
<tr class="odd">
<td align="left">0.15</td>
<td align="right">0.150</td>
<td align="right">328</td>
<td align="right">1</td>
<td align="right">4</td>
<td align="right">5</td>
<td align="right">1</td>
<td align="right">3</td>
</tr>
<tr class="even">
<td align="left">0.2</td>
<td align="right">0.200</td>
<td align="right">464</td>
<td align="right">2</td>
<td align="right">5</td>
<td align="right">6</td>
<td align="right">3</td>
<td align="right">3</td>
</tr>
</tbody>
</table>
<div id="gene-annotations" class="section level3">
<h3>Gene annotations</h3>
<p>For genes that we found to have FDR &lt; .001 (18 genes) in our analysis comparing coefficient of variations between individuals, we look up the gene annotations as follows</p>
<pre class="r"><code>library(knitr)
library(biomaRt)

ensembl &lt;- useMart(host = &quot;grch37.ensembl.org&quot;,
                   biomart = &quot;ENSEMBL_MART_ENSEMBL&quot;,
                   dataset = &quot;hsapiens_gene_ensembl&quot;)

diff_cv_genes &lt;- rownames(molecules_ENSG)[which(F.p.adj&lt; .001)]

diff_cv_genes_info &lt;- getBM(attributes = c(&quot;ensembl_gene_id&quot;, &quot;chromosome_name&quot;,
                                           &quot;external_gene_name&quot;, &quot;transcript_count&quot;,
                                           &quot;description&quot;),
                            filters = &quot;ensembl_gene_id&quot;,
                            values = diff_cv_genes[grep(&quot;ENSG&quot;, diff_cv_genes)],
                            mart = ensembl)
kable(diff_cv_genes_info)</code></pre>
<table>
<thead>
<tr class="header">
<th align="left">ensembl_gene_id</th>
<th align="right">chromosome_name</th>
<th align="left">external_gene_name</th>
<th align="right">transcript_count</th>
<th align="left">description</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td align="left">ENSG00000003436</td>
<td align="right">2</td>
<td align="left">TFPI</td>
<td align="right">13</td>
<td align="left">tissue factor pathway inhibitor (lipoprotein-associated coagulation inhibitor) [Source:HGNC Symbol;Acc:11760]</td>
</tr>
<tr class="even">
<td align="left">ENSG00000022556</td>
<td align="right">19</td>
<td align="left">NLRP2</td>
<td align="right">20</td>
<td align="left">NLR family, pyrin domain containing 2 [Source:HGNC Symbol;Acc:22948]</td>
</tr>
<tr class="odd">
<td align="left">ENSG00000026025</td>
<td align="right">10</td>
<td align="left">VIM</td>
<td align="right">10</td>
<td align="left">vimentin [Source:HGNC Symbol;Acc:12692]</td>
</tr>
<tr class="even">
<td align="left">ENSG00000106153</td>
<td align="right">7</td>
<td align="left">CHCHD2</td>
<td align="right">2</td>
<td align="left">coiled-coil-helix-coiled-coil-helix domain containing 2 [Source:HGNC Symbol;Acc:21645]</td>
</tr>
<tr class="odd">
<td align="left">ENSG00000108518</td>
<td align="right">17</td>
<td align="left">PFN1</td>
<td align="right">3</td>
<td align="left">profilin 1 [Source:HGNC Symbol;Acc:8881]</td>
</tr>
<tr class="even">
<td align="left">ENSG00000135549</td>
<td align="right">6</td>
<td align="left">PKIB</td>
<td align="right">9</td>
<td align="left">protein kinase (cAMP-dependent, catalytic) inhibitor beta [Source:HGNC Symbol;Acc:9018]</td>
</tr>
<tr class="odd">
<td align="left">ENSG00000143248</td>
<td align="right">1</td>
<td align="left">RGS5</td>
<td align="right">11</td>
<td align="left">regulator of G-protein signaling 5 [Source:HGNC Symbol;Acc:10001]</td>
</tr>
<tr class="even">
<td align="left">ENSG00000164434</td>
<td align="right">6</td>
<td align="left">FABP7</td>
<td align="right">2</td>
<td align="left">fatty acid binding protein 7, brain [Source:HGNC Symbol;Acc:3562]</td>
</tr>
<tr class="odd">
<td align="left">ENSG00000166450</td>
<td align="right">15</td>
<td align="left">PRTG</td>
<td align="right">5</td>
<td align="left">protogenin [Source:HGNC Symbol;Acc:26373]</td>
</tr>
<tr class="even">
<td align="left">ENSG00000167996</td>
<td align="right">11</td>
<td align="left">FTH1</td>
<td align="right">11</td>
<td align="left">ferritin, heavy polypeptide 1 [Source:HGNC Symbol;Acc:3976]</td>
</tr>
<tr class="odd">
<td align="left">ENSG00000175063</td>
<td align="right">20</td>
<td align="left">UBE2C</td>
<td align="right">7</td>
<td align="left">ubiquitin-conjugating enzyme E2C [Source:HGNC Symbol;Acc:15937]</td>
</tr>
<tr class="even">
<td align="left">ENSG00000185088</td>
<td align="right">15</td>
<td align="left">RPS27L</td>
<td align="right">7</td>
<td align="left">ribosomal protein S27-like [Source:HGNC Symbol;Acc:18476]</td>
</tr>
<tr class="odd">
<td align="left">ENSG00000187193</td>
<td align="right">16</td>
<td align="left">MT1X</td>
<td align="right">4</td>
<td align="left">metallothionein 1X [Source:HGNC Symbol;Acc:7405]</td>
</tr>
<tr class="even">
<td align="left">ENSG00000198825</td>
<td align="right">10</td>
<td align="left">INPP5F</td>
<td align="right">5</td>
<td align="left">inositol polyphosphate-5-phosphatase F [Source:HGNC Symbol;Acc:17054]</td>
</tr>
<tr class="odd">
<td align="left">ENSG00000198865</td>
<td align="right">5</td>
<td align="left">CCDC152</td>
<td align="right">2</td>
<td align="left">coiled-coil domain containing 152 [Source:HGNC Symbol;Acc:34438]</td>
</tr>
<tr class="even">
<td align="left">ENSG00000255823</td>
<td align="right">11</td>
<td align="left">MTRNR2L8</td>
<td align="right">1</td>
<td align="left">MT-RNR2-like 8 [Source:HGNC Symbol;Acc:37165]</td>
</tr>
<tr class="odd">
<td align="left">ENSG00000256618</td>
<td align="right">17</td>
<td align="left">MTRNR2L1</td>
<td align="right">1</td>
<td align="left">MT-RNR2-like 1 [Source:HGNC Symbol;Acc:37155]</td>
</tr>
<tr class="even">
<td align="left">ENSG00000269028</td>
<td align="right">3</td>
<td align="left">MTRNR2L12</td>
<td align="right">1</td>
<td align="left">MT-RNR2-like 12 (pseudogene) [Source:HGNC Symbol;Acc:37169]</td>
</tr>
</tbody>
</table>
</div>
</div>
<div id="session-information" class="section level2">
<h2>Session information</h2>
<pre class="r"><code>sessionInfo()</code></pre>
<pre><code>R version 3.2.0 (2015-04-16)
Platform: x86_64-unknown-linux-gnu (64-bit)

locale:
 [1] LC_CTYPE=en_US.UTF-8       LC_NUMERIC=C              
 [3] LC_TIME=en_US.UTF-8        LC_COLLATE=en_US.UTF-8    
 [5] LC_MONETARY=en_US.UTF-8    LC_MESSAGES=en_US.UTF-8   
 [7] LC_PAPER=en_US.UTF-8       LC_NAME=C                 
 [9] LC_ADDRESS=C               LC_TELEPHONE=C            
[11] LC_MEASUREMENT=en_US.UTF-8 LC_IDENTIFICATION=C       

attached base packages:
[1] grid      stats     graphics  grDevices utils     datasets  methods  
[8] base     

other attached packages:
[1] biomaRt_2.24.0   zoo_1.7-12       ggplot2_1.0.1    edgeR_3.10.2    
[5] limma_3.24.9     dplyr_0.4.2      data.table_1.9.4 knitr_1.10.5    

loaded via a namespace (and not attached):
 [1] Rcpp_0.12.0          GenomeInfoDb_1.4.0   formatR_1.2         
 [4] plyr_1.8.3           highr_0.5            bitops_1.0-6        
 [7] tools_3.2.0          digest_0.6.8         RSQLite_1.0.0       
[10] evaluate_0.7         gtable_0.1.2         lattice_0.20-31     
[13] DBI_0.3.1            yaml_2.1.13          parallel_3.2.0      
[16] proto_0.3-10         httr_0.6.1           stringr_1.0.0       
[19] IRanges_2.2.4        S4Vectors_0.6.0      stats4_3.2.0        
[22] Biobase_2.28.0       R6_2.1.1             AnnotationDbi_1.30.1
[25] XML_3.98-1.2         rmarkdown_0.6.1      reshape2_1.4.1      
[28] magrittr_1.5         BiocGenerics_0.14.0  scales_0.2.4        
[31] htmltools_0.2.6      MASS_7.3-40          assertthat_0.1      
[34] colorspace_1.2-6     stringi_0.4-1        RCurl_1.95-4.6      
[37] munsell_0.4.2        chron_2.3-45        </code></pre>
</div>


<!-- some extra javascript for older browsers -->
<script type="text/javascript" src="libs/polyfill.js"></script>

<script>

// manage active state of menu based on current page
$(document).ready(function () {

    // active menu
    href = window.location.pathname
    href = href.substr(href.lastIndexOf('/') + 1)
    $('a[href="' + href + '"]').parent().addClass('active');

    // manage active menu header
    if (href.startsWith('authoring_'))
      $('a[href="' + 'authoring' + '"]').parent().addClass('active');
    else if (href.endsWith('_format.html'))
      $('a[href="' + 'formats' + '"]').parent().addClass('active');
    else if (href.startsWith('developer_'))
      $('a[href="' + 'developer' + '"]').parent().addClass('active');

});

</script>

</div>

<script>

// add bootstrap table styles to pandoc tables
$(document).ready(function () {
  $('tr.header').parent('thead').parent('table').addClass('table table-condensed');
});

</script>


</body>
</html>
