<!DOCTYPE html>

<html xmlns="http://www.w3.org/1999/xhtml">

<head>

<meta charset="utf-8">
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="generator" content="pandoc" />

<meta name="author" content="Joyce Hsiao" />

<meta name="date" content="2015-10-31" />

<title>Quantify similarity between CVs</title>

<script src="libs/jquery-1.11.0/jquery.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="libs/bootstrap-3.3.1/css/united.min.css" rel="stylesheet" />
<script src="libs/bootstrap-3.3.1/js/bootstrap.min.js"></script>
<script src="libs/bootstrap-3.3.1/shim/html5shiv.min.js"></script>
<script src="libs/bootstrap-3.3.1/shim/respond.min.js"></script>

<style type="text/css">

/* padding for bootstrap navbar */
body {
  padding-top: 50px;
  padding-bottom: 40px;
}


/* offset scroll position for anchor links (for fixed navbar)  */
.section h2 {
  padding-top: 55px;
  margin-top: -55px;
}
.section h3 {
  padding-top: 55px;
  margin-top: -55px;
}



/* don't use link color in navbar */
.dropdown-menu>li>a {
  color: black;
}

/* some padding for disqus */
#disqus_thread {
  margin-top: 45px;
}

</style>

<link rel="stylesheet" href="libs/font-awesome-4.1.0/css/font-awesome.min.css"/>

<style type="text/css">code{white-space: pre;}</style>
<link rel="stylesheet"
      href="libs/highlight/textmate.css"
      type="text/css" />
<script src="libs/highlight/highlight.js"></script>
<style type="text/css">
  pre:not([class]) {
    background-color: white;
  }
</style>
<script type="text/javascript">
if (window.hljs && document.readyState && document.readyState === "complete") {
   window.setTimeout(function() {
      hljs.initHighlighting();
   }, 0);
}
</script>



</head>

<body>

<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
code {
  color: inherit;
  background-color: rgba(0, 0, 0, 0.04);
}
img { 
  max-width:100%; 
  height: auto; 
}
</style>
<div class="container-fluid main-container">


<div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">singleCellSeq</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li><a href="index.html">Home</a></li>
        <li><a href="about.html">About</a></li>
        <li><a href="license.html">License</a></li>
        <li><a href="https://github.com/jdblischak/singleCellSeq">GitHub</a></li>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<div id="header">
<h1 class="title">Quantify similarity between CVs</h1>
<h4 class="author"><em>Joyce Hsiao</em></h4>
<h4 class="date"><em>2015-10-31</em></h4>
</div>

<div id="TOC">
<ul>
<li><a href="#objective">Objective</a></li>
<li><a href="#set-up">Set up</a></li>
<li><a href="#prepare-data">Prepare data</a></li>
<li><a href="#import-gene-symbols">Import gene symbols</a></li>
<li><a href="#compute-normalized-cv">Compute normalized CV</a></li>
<li><a href="#compute-summary-measure-of-deviation">Compute summary measure of deviation</a></li>
<li><a href="#compare-the-two-metrics">Compare the two metrics</a></li>
<li><a href="#session-information">Session information</a></li>
</ul>
</div>

<p><strong>Last updated:</strong> 2015-12-11</p>
<p><strong>Code version:</strong> 884b335605607bd2cf5c9a1ada1e345466388866</p>
<div id="objective" class="section level2">
<h2>Objective</h2>
<p>We computed similarity metrics to quantify similarity between individuals in coefficients of variation (after accounting for mean dependenc): 1) Sum-of-Squared-Deviation-from-the-Meidan (SSM), and 2) Sum-of-Absolute-Deviation-from-the-Median (SAM). We ordered genes by these metrics and described genes with high and low similarity between individual adjusted-CV.</p>
</div>
<div id="set-up" class="section level2">
<h2>Set up</h2>
<pre class="r"><code>library(&quot;data.table&quot;)
library(&quot;dplyr&quot;)
library(&quot;limma&quot;)
library(&quot;edgeR&quot;)
library(&quot;ggplot2&quot;)
library(&quot;grid&quot;)
library(&quot;Humanzee&quot;)
theme_set(theme_bw(base_size = 12))
source(&quot;functions.R&quot;)</code></pre>
</div>
<div id="prepare-data" class="section level2">
<h2>Prepare data</h2>
<p>Input annotation of only QC-filtered single cells. Remove NA19098.r2</p>
<pre class="r"><code>anno_filter &lt;- read.table(&quot;../data/annotation-filter.txt&quot;, 
                    header = TRUE,
                    stringsAsFactors = FALSE)
dim(anno_filter)</code></pre>
<pre><code>[1] 568   5</code></pre>
<p>Import endogeneous gene molecule counts that are QC-filtered, CPM-normalized, ERCC-normalized, and also processed to remove unwanted variation from batch effet. ERCC genes are removed from this file.</p>
<pre class="r"><code>molecules_ENSG &lt;- read.table(&quot;../data/molecules-final.txt&quot;, 
                             header = TRUE, stringsAsFactors = FALSE)
stopifnot(NCOL(molecules_ENSG) == NROW(anno_filter))</code></pre>
</div>
<div id="import-gene-symbols" class="section level2">
<h2>Import gene symbols</h2>
<pre class="r"><code>gene_info &lt;- read.table(&quot;../data/gene-info.txt&quot;, sep = &quot;\t&quot;,
                        header = TRUE, stringsAsFactors = FALSE)
str(gene_info)</code></pre>
<pre><code>&#39;data.frame&#39;:   6152 obs. of  5 variables:
 $ ensembl_gene_id   : chr  &quot;ENSG00000000003&quot; &quot;ENSG00000000005&quot; &quot;ENSG00000000419&quot; &quot;ENSG00000000457&quot; ...
 $ chromosome_name   : chr  &quot;X&quot; &quot;X&quot; &quot;20&quot; &quot;1&quot; ...
 $ external_gene_name: chr  &quot;TSPAN6&quot; &quot;TNMD&quot; &quot;DPM1&quot; &quot;SCYL3&quot; ...
 $ transcript_count  : int  3 2 7 5 10 4 12 2 13 8 ...
 $ description       : chr  &quot;tetraspanin 6 [Source:HGNC Symbol;Acc:11858]&quot; &quot;tenomodulin [Source:HGNC Symbol;Acc:17757]&quot; &quot;dolichyl-phosphate mannosyltransferase polypeptide 1, catalytic subunit [Source:HGNC Symbol;Acc:3005]&quot; &quot;SCY1-like 3 (S. cerevisiae) [Source:HGNC Symbol;Acc:19285]&quot; ...</code></pre>
</div>
<div id="compute-normalized-cv" class="section level2">
<h2>Compute normalized CV</h2>
<p>We compute squared CV across cells for each individual and then for each individual CV profile, account for mean dependency by computing distance with respect to the data-wide coefficient variation on the log10 scale.</p>
<pre class="r"><code>ENSG_cv &lt;- Humanzee::compute_cv(log2counts = molecules_ENSG,
                      grouping_vector = anno_filter$individual)

ENSG_cv_adj &lt;- Humanzee::normalize_cv(group_cv = ENSG_cv, 
                            log2counts = molecules_ENSG, 
                            anno = anno_filter)</code></pre>
</div>
<div id="compute-summary-measure-of-deviation" class="section level2">
<h2>Compute summary measure of deviation</h2>
<ul>
<li>Standardize the each CV vectors</li>
</ul>
<p>Individual CV vectors are standarized for individual CV mean and coefficients of variation across genes.</p>
<pre class="r"><code>df_cv &lt;- data.frame(NA19098 = ENSG_cv_adj[[1]]$log10cv2_adj,
                    NA19101 = ENSG_cv_adj[[2]]$log10cv2_adj,
                    NA19239 = ENSG_cv_adj[[3]]$log10cv2_adj)

library(matrixStats)
df_norm &lt;- sweep(df_cv, MARGIN = 2, STATS = colMeans(as.matrix(df_cv)), FUN = &quot;-&quot;)
df_norm &lt;- sweep(df_norm, MARGIN = 2, STATS = sqrt(colVars(as.matrix(df_cv))), FUN = &quot;/&quot;)
colnames(df_norm) &lt;- names(ENSG_cv_adj)

#save(df_norm, file=&quot;./rda/cv-adjusted-comparison/adj-cv.rda&quot;)</code></pre>
<ul>
<li>Adj-CVs before/after standardization.</li>
</ul>
<p>CVs are similar between individuals.</p>
<pre class="r"><code>ggplot(data.frame(cv = c(t(rbind( ENSG_cv_adj[[1]]$cv,
                                  ENSG_cv_adj[[2]]$cv,
                                  ENSG_cv_adj[[3]]$cv) )),
                  source = factor(rep(colnames(df_cv), each = dim(df_cv)[1]) )),
        aes(x = source, y = log10(cv^2), fill = source) ) +
  geom_violin(alpha = .5) + geom_boxplot(alpha = .2, width = .2) +
  xlab (&quot;Cell line&quot;) + ylab(&quot;log10 squared CV&quot;) +
  ggtitle(&quot;log10 CV^2&quot;)</code></pre>
<p><img src="figure/cv-adjusted-comparison.Rmd/unnamed-chunk-7-1.png" title="" alt="" width="672" style="display: block; margin: auto;" /></p>
<p>The distributions of the adjusted squared CVs are not the same in their medians. This may cause an unfair comparison of the adj. CV^2 scores between individuals.</p>
<pre class="r"><code>ggplot(data.frame(adj_cv = c(t(rbind( ENSG_cv_adj[[1]]$log10cv2_adj,
                                  ENSG_cv_adj[[2]]$log10cv2_adj,
                                  ENSG_cv_adj[[3]]$log10cv2_adj) )),
                  source = factor(rep(colnames(df_cv), each = dim(df_cv)[1]) )),
        aes(x = source, y = adj_cv, fill = source) ) +
  geom_violin(alpha = .5) + geom_boxplot(alpha = .2, width = .2) +
  xlab (&quot;Cell line&quot;) + ylab(&quot;log10 adjusted coefficient of variation&quot;) +
  ggtitle(&quot;log10 adjusted CV&quot;)</code></pre>
<p><img src="figure/cv-adjusted-comparison.Rmd/unnamed-chunk-8-1.png" title="" alt="" width="672" style="display: block; margin: auto;" /></p>
<p>We normlized the log10 adj. CV^2 by standardizing indivdual CVs over the mean and standard deviation across genes. Note the change in CV distribution from log10 adj CV^2 to normalized log10 adj CV^2. After accounting for differences in variability of the log10 adj. CV62, we observed that NA19098 spans an obviously smaller range of normalized adj CV^2 values than the other two cell lines, probably a consequence of the small number of cells in NA19098 (n = 141) compared to the others (NA19101: 201 cells; NA19239: 225 cells).</p>
<pre class="r"><code>ggplot(data.frame(norm_cv = c(t( df_norm )),
                  source = factor(rep(colnames(df_cv), each = dim(df_cv)[1]) )),
        aes(x = source, y = norm_cv, fill = source) ) +
  geom_violin(alpha = .5) + geom_boxplot(alpha = .2, width = .2) +
  xlab (&quot;Cell line&quot;) + ylab(&quot;log10 standardized adj. CV^2&quot;) +
  ggtitle(&quot;log10 standardized adj. CV^2&quot;)</code></pre>
<p><img src="figure/cv-adjusted-comparison.Rmd/unnamed-chunk-9-1.png" title="" alt="" width="672" style="display: block; margin: auto;" /></p>
<ul>
<li>Compute metrics for quantifying similarity between the three individual coefficients of variation.</li>
</ul>
<pre class="r"><code>library(matrixStats)
df_norm &lt;- as.data.frame(df_norm)
df_norm$squared_dev &lt;- rowSums( ( df_norm - rowMedians(as.matrix(df_norm)) )^2 )
df_norm$abs_dev &lt;- rowSums(abs( df_norm - rowMedians(as.matrix(df_norm)) ))</code></pre>
</div>
<div id="compare-the-two-metrics" class="section level2">
<h2>Compare the two metrics</h2>
<p>Box plots</p>
<pre class="r"><code>ggplot(data.frame(dist = c(unlist(df_norm$squared_dev),
                           unlist(df_norm$abs_dev)),
                  source = factor(rep(c(1:2), each = dim(df_norm)[1]),
                                  labels = c(&quot;log10 SSM&quot;, &quot;log10 SAM&quot;))),
       aes(x = source, y = log10(dist), fill = source)) +
  geom_violin(alpha = .5) + geom_boxplot(alpha = .2, width = .2) +
  xlab(&quot;Similarity metric&quot;) + ylab(&quot;Distance&quot;) +
  ggtitle(&quot;Similarity between individual standardized log10 adj. CV^2&quot;)</code></pre>
<p><img src="figure/cv-adjusted-comparison.Rmd/unnamed-chunk-11-1.png" title="" alt="" width="672" style="display: block; margin: auto;" /></p>
<p>Density plots</p>
<pre class="r"><code>library(broman)
crayon &lt;- brocolors(&quot;crayon&quot;)
library(scales)

dens &lt;- list( density(log10(df_norm$squared_dev)),
              density(log10(df_norm$abs_dev)) )
names(dens) &lt;- c(&quot;SSM&quot;, &quot;SAM&quot;)
xlims &lt;- c(range( sapply(dens, &quot;[[&quot;, 1) ) )
ylims &lt;- c(0, max( sapply(dens, &quot;[[&quot;, 2) ) )
  
par(mfrow = c(1,1))
plot(0, pch = &quot;&quot;, 
     xlab = &quot;log10 distance&quot;, main = &quot;&quot;,
     ylab = &quot;Density&quot;, axes = F, xlim = xlims, ylim = ylims)
axis(1); axis(2)
polygon(dens[[1]], col = alpha( crayon[&quot;Blue Bell&quot;], .3), lwd = .5 )
polygon(dens[[2]], col = alpha( crayon[&quot;Fern&quot;], .3), lwd = .5)
legend(&quot;topleft&quot;, legend = c(&quot;SSM&quot;, &quot;SAM&quot;), bty=&quot;n&quot;,
       pch = 19, cex = 1, 
       col = c(alpha( crayon[&quot;Blue Bell&quot;], .8),
               alpha( crayon[&quot;Fern&quot;], .8) ) )  </code></pre>
<p><img src="figure/cv-adjusted-comparison.Rmd/unnamed-chunk-12-1.png" title="" alt="" width="672" style="display: block; margin: auto;" /></p>
<p>Gene rankings based on SSM (Sum-of-Squared-Deviation-from-the-Median) and SAM (Sum-of-Absolute-Deviation-from-the-Median).</p>
<pre class="r"><code>ggplot(df_norm,
       aes(x = rank(squared_dev), y = rank(abs_dev)) ) +
  geom_point() + geom_abline(col = &quot;blue&quot;) +
  ggtitle(&quot;Gene rankings&quot;) +
  xlab(&quot;Rank order based on SSM&quot;) +
  ylab(&quot;Rank order based on SAM&quot;)</code></pre>
<p><img src="figure/cv-adjusted-comparison.Rmd/unnamed-chunk-13-1.png" title="" alt="" width="672" style="display: block; margin: auto;" /></p>
<p>*Look at the gene with low ranking in SSM (small difference) but high ranking in SAM (large difference).</p>
<pre class="r"><code>library(broman)
crayon &lt;- brocolors(&quot;crayon&quot;)
library(scales)

index &lt;- rank(df_norm$squared_dev) &lt; 1000 &amp; rank(df_norm$abs_dev) &gt; 8000
sum(index)</code></pre>
<pre><code>[1] 2</code></pre>
<pre class="r"><code>ggplot(df_norm,
       aes(x = rank(squared_dev), y = rank(abs_dev)) ) +
  geom_point() + geom_abline(col = &quot;blue&quot;) +
#   ggtitle(&quot;Gene rankings&quot;) +
#   xlab(&quot;Rank order based on SSM&quot;) +
#   ylab(&quot;Rank order based on SAM&quot;) +
  geom_point(data = data.frame(x = rank(df_norm$squared_dev[which(index)]),
                               y = rank(df_norm$abs_dev)[which(index)]), 
             aes(x = x, y = y, col = &quot;red&quot;, cex = 1.5) ) </code></pre>
<p><img src="figure/cv-adjusted-comparison.Rmd/unnamed-chunk-14-1.png" title="" alt="" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code># Find gene symbol
gene_ENSG &lt;- rownames(molecules_ENSG)[which(index)]
gene_name &lt;- sapply(gene_ENSG, function(per_gene) {
    if ( sum(gene_info$ensembl_gene_id == per_gene) == 0 ) {
      return(NA)
    } 
    if ( sum(gene_info$ensembl_gene_id == per_gene) &gt; 0  ) {
      gene_info$external_gene_name[which(gene_info$ensembl_gene_id == per_gene) ]
    }
})
gene_name</code></pre>
<pre><code>ENSG00000269028 ENSG00000198840 
    &quot;MTRNR2L12&quot;        &quot;MT-ND3&quot; </code></pre>
<pre class="r"><code># Plot density for individual gene expression 
for (i in 1: length(gene_name)) {
  
  dens &lt;- lapply(1:3, function(ii_individual) {
     density( unlist(molecules_ENSG[ which(index)[i],
          anno_filter$individual == unique(anno_filter$individual)[ii_individual]]) )
    })
  names(dens) &lt;- unique(anno_filter$individual)
  xlims &lt;- c(range( sapply(dens, &quot;[[&quot;, 1) ) )
  ylims &lt;- c(0, max( sapply(dens, &quot;[[&quot;, 2) ) )
  
  par(mfrow = c(1,1))
  plot(0, pch = &quot;&quot;, 
       xlab = &quot;log2 gene expression&quot;, main = &quot;&quot;,
       ylab = &quot;Density&quot;, axes = F, xlim = xlims, ylim = ylims)
  axis(1); axis(2)
  polygon(dens[[1]], col = alpha( crayon[&quot;Blue Bell&quot;], .3), lwd = .5 )
  polygon(dens[[2]], col = alpha( crayon[&quot;Fern&quot;], .3), lwd = .5)
  polygon(dens[[3]], col = alpha( crayon[&quot;Sunglow&quot;], .2), lwd = .5 )
  title( ifelse(is.na(gene_name[i]), names(gene_name)[i], gene_name[i] ) )
}</code></pre>
<p><img src="figure/cv-adjusted-comparison.Rmd/unnamed-chunk-14-2.png" title="" alt="" width="672" style="display: block; margin: auto;" /><img src="figure/cv-adjusted-comparison.Rmd/unnamed-chunk-14-3.png" title="" alt="" width="672" style="display: block; margin: auto;" /></p>
<p>Print out CV values. Recall that these are standardized values (z-scores); hence, vlaues outside of mean +/- standard deviation are quite large.</p>
<pre class="r"><code>df_norm[which(index), ]</code></pre>
<pre><code>       NA19098  NA19101  NA19239 squared_dev  abs_dev
7156  8.491161 8.221735 8.174445  0.07482687 8.463623
10152 5.657274 5.402757 5.292184  0.07700510 5.690841</code></pre>
<p>Ranks of the three individuals are consistent for the three genes selected here.</p>
<pre class="r"><code>library(matrixStats)
matplot(t(df_norm[which(index), 1:3]), type = &quot;p&quot;, axes = F,
        xlab = &quot;Cell line&quot;, ylab = &quot;&quot;)
axis(1, at = c(1:3), labels = colnames(df_norm)[1:3] )
axis(2)</code></pre>
<p><img src="figure/cv-adjusted-comparison.Rmd/unnamed-chunk-16-1.png" title="" alt="" width="672" style="display: block; margin: auto;" /></p>
</div>
<div id="session-information" class="section level2">
<h2>Session information</h2>
<pre class="r"><code>sessionInfo()</code></pre>
<pre><code>R version 3.2.1 (2015-06-18)
Platform: x86_64-apple-darwin13.4.0 (64-bit)
Running under: OS X 10.10.5 (Yosemite)

locale:
[1] en_US.UTF-8/en_US.UTF-8/en_US.UTF-8/C/en_US.UTF-8/en_US.UTF-8

attached base packages:
[1] grid      stats     graphics  grDevices utils     datasets  methods  
[8] base     

other attached packages:
 [1] scales_0.3.0       broman_0.59-5      matrixStats_0.15.0
 [4] zoo_1.7-12         Humanzee_0.1.0     ggplot2_1.0.1     
 [7] edgeR_3.10.5       limma_3.24.15      dplyr_0.4.3       
[10] data.table_1.9.6   knitr_1.11        

loaded via a namespace (and not attached):
 [1] Rcpp_0.12.2      magrittr_1.5     MASS_7.3-45      munsell_0.4.2   
 [5] lattice_0.20-33  colorspace_1.2-6 R6_2.1.1         stringr_1.0.0   
 [9] plyr_1.8.3       tools_3.2.1      parallel_3.2.1   gtable_0.1.2    
[13] DBI_0.3.1        htmltools_0.2.6  yaml_2.1.13      assertthat_0.1  
[17] digest_0.6.8     reshape2_1.4.1   formatR_1.2.1    evaluate_0.8    
[21] rmarkdown_0.8.1  labeling_0.3     stringi_1.0-1    chron_2.3-47    
[25] proto_0.3-10    </code></pre>
</div>


<!-- some extra javascript for older browsers -->
<script type="text/javascript" src="libs/polyfill.js"></script>

<script>

// manage active state of menu based on current page
$(document).ready(function () {

    // active menu
    href = window.location.pathname
    href = href.substr(href.lastIndexOf('/') + 1)
    $('a[href="' + href + '"]').parent().addClass('active');

    // manage active menu header
    if (href.startsWith('authoring_'))
      $('a[href="' + 'authoring' + '"]').parent().addClass('active');
    else if (href.endsWith('_format.html'))
      $('a[href="' + 'formats' + '"]').parent().addClass('active');
    else if (href.startsWith('developer_'))
      $('a[href="' + 'developer' + '"]').parent().addClass('active');

});

</script>

</div>

<script>

// add bootstrap table styles to pandoc tables
$(document).ready(function () {
  $('tr.header').parent('thead').parent('table').addClass('table table-condensed');
});

</script>


</body>
</html>
