<!DOCTYPE html>

<html xmlns="http://www.w3.org/1999/xhtml">

<head>

<meta charset="utf-8">
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="generator" content="pandoc" />

<meta name="author" content="Joyce Hsiao" />

<meta name="date" content="2015-10-13" />

<title>Individual differences in CV and mean</title>

<script src="libs/jquery-1.11.0/jquery.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="libs/bootstrap-3.3.1/css/united.min.css" rel="stylesheet" />
<script src="libs/bootstrap-3.3.1/js/bootstrap.min.js"></script>
<script src="libs/bootstrap-3.3.1/shim/html5shiv.min.js"></script>
<script src="libs/bootstrap-3.3.1/shim/respond.min.js"></script>

<style type="text/css">

/* padding for bootstrap navbar */
body {
  padding-top: 50px;
  padding-bottom: 40px;
}


/* offset scroll position for anchor links (for fixed navbar)  */
.section h2 {
  padding-top: 55px;
  margin-top: -55px;
}
.section h3 {
  padding-top: 55px;
  margin-top: -55px;
}



/* don't use link color in navbar */
.dropdown-menu>li>a {
  color: black;
}

/* some padding for disqus */
#disqus_thread {
  margin-top: 45px;
}

</style>

<link rel="stylesheet" href="libs/font-awesome-4.1.0/css/font-awesome.min.css"/>

<style type="text/css">code{white-space: pre;}</style>
<link rel="stylesheet"
      href="libs/highlight/textmate.css"
      type="text/css" />
<script src="libs/highlight/highlight.js"></script>
<style type="text/css">
  pre:not([class]) {
    background-color: white;
  }
</style>
<script type="text/javascript">
if (window.hljs && document.readyState && document.readyState === "complete") {
   window.setTimeout(function() {
      hljs.initHighlighting();
   }, 0);
}
</script>



</head>

<body>

<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
code {
  color: inherit;
  background-color: rgba(0, 0, 0, 0.04);
}
img { 
  max-width:100%; 
  height: auto; 
}
</style>
<div class="container-fluid main-container">


<div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">singleCellSeq</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li><a href="index.html">Home</a></li>
        <li><a href="about.html">About</a></li>
        <li><a href="license.html">License</a></li>
        <li><a href="https://github.com/jdblischak/singleCellSeq">GitHub</a></li>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<div id="header">
<h1 class="title">Individual differences in CV and mean</h1>
<h4 class="author"><em>Joyce Hsiao</em></h4>
<h4 class="date"><em>2015-10-13</em></h4>
</div>

<div id="TOC">
<ul>
<li><a href="#objective">Objective</a></li>
<li><a href="#set-up">Set up</a></li>
<li><a href="#prepare-data">Prepare data</a></li>
<li><a href="#compare-coefficient-of-variation">Compare coefficient of variation</a></li>
<li><a href="#compare-mean-expression-levels.">Compare mean expression levels.</a><ul>
<li><a href="#single-cell-sequencing-data">Single-cell sequencing data</a></li>
</ul></li>
<li><a href="#session-information">Session information</a></li>
</ul>
</div>

<p><strong>Last updated:</strong> 2015-10-19</p>
<p><strong>Code version:</strong> 6b16d0b7ff64ea109c14ebf7c3bfbb5cebd4dec8</p>
<div id="objective" class="section level2">
<h2>Objective</h2>
<p>Previously, we compared normalized coefficient of variations across individuals. Here, we will also compare the mean gene expression across individuals, using single cell sequencing data and bulk RNA-seq data.</p>
<p>It would be interesting to learn the possible overlap or non-overlap between the genes that we observed significant individual differences in coefficient of variations versus those different in mean gene expression levels across cells.</p>
</div>
<div id="set-up" class="section level2">
<h2>Set up</h2>
<pre class="r"><code>library(&quot;data.table&quot;)
library(&quot;dplyr&quot;)
library(&quot;limma&quot;)
library(&quot;edgeR&quot;)
library(&quot;ggplot2&quot;)
library(&quot;grid&quot;)
theme_set(theme_bw(base_size = 12))
source(&quot;functions.R&quot;)</code></pre>
</div>
<div id="prepare-data" class="section level2">
<h2>Prepare data</h2>
<p>Input annotation of only QC-filtered single cells</p>
<pre class="r"><code>anno_qc &lt;- read.table(&quot;../data/annotation-filter.txt&quot;, header = TRUE,
                   stringsAsFactors = FALSE)
head(anno_qc)</code></pre>
<pre><code>  individual replicate well      batch      sample_id
1    NA19098        r1  A01 NA19098.r1 NA19098.r1.A01
2    NA19098        r1  A02 NA19098.r1 NA19098.r1.A02
3    NA19098        r1  A04 NA19098.r1 NA19098.r1.A04
4    NA19098        r1  A05 NA19098.r1 NA19098.r1.A05
5    NA19098        r1  A06 NA19098.r1 NA19098.r1.A06
6    NA19098        r1  A07 NA19098.r1 NA19098.r1.A07</code></pre>
<p>Input molecule counts that are filtered, transformed, and also processed to remove unwanted variation from batch effet. ERCC genes are also removed.</p>
<pre class="r"><code>molecules_qc &lt;- read.table(&quot;../data/molecules-final.txt&quot;, header = TRUE, stringsAsFactors = FALSE)</code></pre>
<p>Input moleclule counts before log2 CPM transformation.</p>
<pre class="r"><code>molecules_filter &lt;- read.table(&quot;../data/molecules-filter.txt&quot;, header = TRUE, stringsAsFactors = FALSE)
molecules_filter &lt;- molecules_filter[which(rownames(molecules_filter) %in% rownames(molecules_qc)), ]

stopifnot(dim(molecules_filter) == dim(molecules_qc))</code></pre>
</div>
<div id="compare-coefficient-of-variation" class="section level2">
<h2>Compare coefficient of variation</h2>
<p>Compute per batch coefficient of variation.</p>
<p>Include only genes with positive coefficient of variation. Some genes in this data may have zero coefficient of variation, because we include gene with more than 0 count across all cells.</p>
<pre class="r"><code>molecules_cv_batch &lt;- 
  lapply(1:length(unique(anno_qc$batch)), function(per_batch) {
      molecules_per_batch &lt;- 2^molecules_qc[ , unique(anno_qc$batch) == unique(anno_qc$batch)[per_batch] ]
      mean_per_gene &lt;- apply(molecules_per_batch, 1, mean, na.rm = TRUE)
      sd_per_gene &lt;- apply(molecules_per_batch, 1, sd, na.rm = TRUE)
      cv_per_gene &lt;- data.frame(mean = mean_per_gene,
                                sd = sd_per_gene,
                                cv = sd_per_gene/mean_per_gene)
      rownames(cv_per_gene) &lt;- rownames(molecules_qc)
  
      cv_per_gene &lt;- cv_per_gene[rowSums(is.na(cv_per_gene)) == 0, ]
      cv_per_gene$batch &lt;- unique(anno_qc$batch)[per_batch]
      
      # Add sparsity percent
      molecules_count &lt;- molecules_filter[ , unique(anno_qc$batch) == unique(anno_qc$batch)[per_batch]]
      cv_per_gene$sparse &lt;- rowMeans(as.matrix(molecules_count) == 0)
        
      return(cv_per_gene)
      }) 
names(molecules_cv_batch) &lt;- unique(anno_qc$batch)

dim(molecules_cv_batch[[1]])</code></pre>
<pre><code>[1] 10483     5</code></pre>
<p>Merge summary data.frames.</p>
<pre class="r"><code>df_plot &lt;- do.call(rbind, molecules_cv_batch)</code></pre>
<p>Compute rolling medians.</p>
<pre class="r"><code>library(zoo)
# Compute a data-wide coefficient of variation on CPM normalized counts.
data_cv &lt;- apply(2^molecules_qc, 1, sd)/apply(2^molecules_qc, 1, mean)

# Order of genes by mean expression levels
order_gene &lt;- order(apply(2^molecules_qc, 1, mean))

# Rolling medians of log10 squared CV by mean expression levels
roll_medians &lt;- rollapply(log10(data_cv^2)[order_gene], width = 50, by = 25,
                         FUN = median, fill = list(&quot;extend&quot;, &quot;extend&quot;, &quot;NA&quot;) )
ii_na &lt;- which( is.na(roll_medians) )
roll_medians[ii_na] &lt;- median( log10(data_cv^2)[order_gene][ii_na] )

names(roll_medians) &lt;- rownames(molecules_qc)[order_gene]

# re-order rolling medians
reorder_gene &lt;- match(rownames(molecules_qc), names(roll_medians) )
head(reorder_gene)</code></pre>
<pre><code>[1]  225 3650 4564 6158 3352 1963</code></pre>
<pre class="r"><code>roll_medians &lt;- roll_medians[ reorder_gene ]

head(names(roll_medians))</code></pre>
<pre><code>[1] &quot;ENSG00000237683&quot; &quot;ENSG00000188976&quot; &quot;ENSG00000187608&quot; &quot;ENSG00000188157&quot;
[5] &quot;ENSG00000078808&quot; &quot;ENSG00000176022&quot;</code></pre>
<pre class="r"><code>head(rownames(molecules_qc))</code></pre>
<pre><code>[1] &quot;ENSG00000237683&quot; &quot;ENSG00000188976&quot; &quot;ENSG00000187608&quot; &quot;ENSG00000188157&quot;
[5] &quot;ENSG00000078808&quot; &quot;ENSG00000176022&quot;</code></pre>
<p>Compute adjusted coefficient of variation.</p>
<pre class="r"><code># adjusted coefficient of variation on log10 scale
log10cv2_adj &lt;- 
  lapply(1:length(molecules_cv_batch), function(per_batch) {
    foo &lt;- log10(molecules_cv_batch[[per_batch]]$cv^2) - roll_medians
    return(foo)
})
df_plot$log10cv2_adj &lt;- do.call(c, log10cv2_adj)</code></pre>
<p>Compare adjusted CVs</p>
<pre class="r"><code>library(limma)

df_limma &lt;- matrix(df_plot$log10cv2_adj, 
                      nrow = nrow(molecules_qc), ncol = 9, byrow = FALSE)

design &lt;- data.frame(individual = factor(rep(unique(anno_qc$individual), each = 3) ),
                     rep = factor(rep(c(1:3), times = 3)) )

colnames(df_limma) &lt;- with(design, paste0(individual, rep))

fit_limma &lt;- lmFit(df_limma, design = model.matrix( ~ individual, data = design))                      
fit_limma &lt;- eBayes(fit_limma)</code></pre>
<p>False discover control adjustment.</p>
<pre class="r"><code>F.p.adj &lt;- p.adjust(fit_limma$F.p.value, method = &quot;fdr&quot;)</code></pre>
<p>Cutoffs</p>
<pre class="r"><code>df_cuts &lt;- data.frame(cuts = c(.001, .01, .05, .1, .15, .2))
df_cuts$sig_count &lt;- sapply(1:6, function(per_cut) {
                    sum(F.p.adj &lt; df_cuts$cuts[per_cut] )
                    })
df_cuts</code></pre>
<pre><code>   cuts sig_count
1 0.001        24
2 0.010        54
3 0.050       115
4 0.100       179
5 0.150       290
6 0.200       423</code></pre>
</div>
<div id="compare-mean-expression-levels." class="section level2">
<h2>Compare mean expression levels.</h2>
<div id="single-cell-sequencing-data" class="section level3">
<h3>Single-cell sequencing data</h3>
<p>Almost all of the genes came out to be statistically significant due the large sample size (about ~70 sample points per replicate, per individual).</p>
<pre class="r"><code>fit_limma_mean &lt;- lmFit(molecules_qc, 
                        design = model.matrix(~factor(individual), data = anno_qc) )

fit_limma_mean &lt;- eBayes(fit_limma_mean)

summary(fit_limma_mean$F.p.value)</code></pre>
<pre><code>      Min.    1st Qu.     Median       Mean    3rd Qu.       Max. 
 0.000e+00  0.000e+00  0.000e+00 1.985e-194  0.000e+00 1.180e-190 </code></pre>
<p>Histogram of p-values.</p>
<pre class="r"><code>hist(-log10(fit_limma_mean$F.p.value), 
     main = &quot;F-test p-values comparing mean expression&quot;, breaks = 100)</code></pre>
<p><img src="figure/cv-mean.Rmd/unnamed-chunk-13-1.png" title="" alt="" width="672" style="display: block; margin: auto;" /></p>
<p>FDR</p>
<pre class="r"><code>padj_mean_single &lt;- p.adjust(fit_limma_mean$F.p.value, method = &quot;fdr&quot;)
summary(padj_mean_single)</code></pre>
<pre><code>      Min.    1st Qu.     Median       Mean    3rd Qu.       Max. 
 0.000e+00  0.000e+00  0.000e+00 1.985e-194  0.000e+00 1.180e-190 </code></pre>
<p>Mean versus coefficient of variation comparisons</p>
<pre class="r"><code>plot(x = -log10(fit_limma_mean$F.p.value), 
     y = -log10(fit_limma$F.p.value),
     xlab = &quot;-log10 F-test p-value comparing means&quot;,
     ylab = &quot;-log10 F-test p-value compare adjusted CV&quot;,
     main = &quot;Unadjusted F-test p-values&quot;)
abline( h = -log10(.01), col = &quot;red&quot;)</code></pre>
<p><img src="figure/cv-mean.Rmd/unnamed-chunk-15-1.png" title="" alt="" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>plot(x = -log10(padj_mean_single), 
     y = -log10(F.p.adj),
     xlab = &quot;-log10 FDR comparing means&quot;,
     ylab = &quot;-log10 FDSR compare adjusted CV&quot;,
     main = &quot;FDR&quot;)
abline( h = -log10(.01), col = &quot;red&quot;)
abline( h = -log10(.05), col = &quot;red&quot;)</code></pre>
<p><img src="figure/cv-mean.Rmd/unnamed-chunk-15-2.png" title="" alt="" width="672" style="display: block; margin: auto;" /></p>
<p>Overlap of the significant genes</p>
<p>*Adjusted p-value cut-offs</p>
<pre class="r"><code>cuts &lt;- c(.001, .01, .05, .1, .15, .2)
both_table &lt;- lapply(1:6, function(per_cut) {
                  table(F.p.adj &lt; df_cuts$cuts[per_cut],
                        padj_mean_single &lt; df_cuts$cut[per_cut])
              })
names(both_table) &lt;- cuts

both_table</code></pre>
<pre><code>$`0.001`
       
         TRUE
  FALSE 10459
  TRUE     24

$`0.01`
       
         TRUE
  FALSE 10429
  TRUE     54

$`0.05`
       
         TRUE
  FALSE 10368
  TRUE    115

$`0.1`
       
         TRUE
  FALSE 10304
  TRUE    179

$`0.15`
       
         TRUE
  FALSE 10193
  TRUE    290

$`0.2`
       
         TRUE
  FALSE 10060
  TRUE    423</code></pre>
<p>*Ranks of adjusted p-values</p>
<pre class="r"><code>plot(x = rank(-log10(padj_mean_single)), 
     y = rank(-log10(F.p.adj)),
     xlab = &quot;rank of -log10 FDR comparing means&quot;,
     ylab = &quot;rank of -log10 FDR compare adjusted CV&quot;,
     main = &quot;FDR&quot;)</code></pre>
<p><img src="figure/cv-mean.Rmd/unnamed-chunk-17-1.png" title="" alt="" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>df_compare &lt;- 
  data.frame(mean = rowMeans( as.matrix( 
                      do.call(cbind, lapply(molecules_cv_batch, &quot;[[&quot;, 1) ) ) ),
             cv2 = rowMeans( as.matrix( 
                      do.call(cbind, lapply(molecules_cv_batch, &quot;[[&quot;, 3) ) )^2 ),
             adj_cv2 = rowMeans( 10^as.matrix( 
                      do.call(cbind, log10cv2_adj) ) ) )
             
library(broman)
crayon &lt;- brocolors(&quot;crayons&quot;)
with(df_compare, plot(x = log10(mean), y = cv2, pch = 1, cex = 1, col = &quot;grey50&quot;,
                      lwd = .5,
                      ylab = &quot;average squared coeffcient of variations \n across samples&quot;,
                      xlab = &quot;log10 average of mean CPM across samples&quot;) )
with(df_compare[F.p.adj &lt; .2, ], points(x = log10(mean), y = cv2, pch = 16, cex = .6,
                                        col = crayon[&quot;Tumbleweed&quot;]) )
with(df_compare[F.p.adj &lt; .1, ], points(x = log10(mean), y = cv2, pch = 16, cex = .6,
                                        col = crayon[&quot;Orange&quot;]) )
with(df_compare[F.p.adj &lt; .01, ], points(x = log10(mean), y = cv2, pch = 16, cex = .6,
                                        col = crayon[&quot;Scarlet&quot;]) )
title(main = &quot;Avg. Squred CV vs. log10(Avg. mean count)&quot;)
legend(&quot;topright&quot;, pch = c(1, 16, 16, 16), 
       legend = c(&quot;All genes&quot;, &quot;Adj p-value &lt; .2&quot;, 
                  &quot;Adj p-value &lt; .1&quot;, &quot;Adj p-value &lt; .01&quot;),
       col = c(&quot;grey50&quot;, crayon[c(&quot;Tumbleweed&quot;, &quot;Orange&quot;, &quot;Scarlet&quot;)]),
       bty = &quot;n&quot;)</code></pre>
<p><img src="figure/cv-mean.Rmd/unnamed-chunk-18-1.png" title="" alt="" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>with(df_compare, plot(x = log10(mean), y = adj_cv2, pch = 1, cex = 1, col = &quot;grey50&quot;,
                      lwd = .5,
                      ylab = &quot;average squared coeffcient of variations \n across samples&quot;,
                      xlab = &quot;log10 average of mean CPM across samples&quot;) )
with(df_compare[F.p.adj &lt; .2, ], points(x = log10(mean), y = adj_cv2, pch = 16, cex = .6,
                                        col = crayon[&quot;Tumbleweed&quot;]) )
with(df_compare[F.p.adj &lt; .1, ], points(x = log10(mean), y = adj_cv2, pch = 16, cex = .6,
                                        col = crayon[&quot;Orange&quot;]) )
with(df_compare[F.p.adj &lt; .01, ], points(x = log10(mean), y = adj_cv2, pch = 16, cex = .6,
                                        col = crayon[&quot;Scarlet&quot;]) )
title(main = &quot;Avg. Adjusted Squred CV vs. log10(Avg. mean count)&quot;)
legend(&quot;topright&quot;, pch = c(1, 16, 16, 16), 
       legend = c(&quot;All genes&quot;, &quot;Adj p-value &lt; .2&quot;, 
                  &quot;Adj p-value &lt; .1&quot;, &quot;Adj p-value &lt; .01&quot;),
       col = c(&quot;grey50&quot;, crayon[c(&quot;Tumbleweed&quot;, &quot;Orange&quot;, &quot;Scarlet&quot;)]),
       bty = &quot;n&quot;)</code></pre>
<p><img src="figure/cv-mean.Rmd/unnamed-chunk-18-2.png" title="" alt="" width="672" style="display: block; margin: auto;" /></p>
<p>Heatmap</p>
<pre class="r"><code>library(mygene)
gene_symbols &lt;- getGenes(rownames(molecules_qc), 
                         fields = c(&quot;name&quot;, &quot;symbol&quot;) ) [ , c(&quot;query&quot;, &quot;symbol&quot;)]
head(gene_symbols)</code></pre>
<pre><code>DataFrame with 6 rows and 2 columns
            query      symbol
      &lt;character&gt; &lt;character&gt;
1 ENSG00000237683          NA
2 ENSG00000188976       NOC2L
3 ENSG00000187608       ISG15
4 ENSG00000188157        AGRN
5 ENSG00000078808        SDF4
6 ENSG00000176022     B3GALT6</code></pre>
<pre class="r"><code>library(gplots)

order_cv &lt;- order(F.p.adj)[F.p.adj &lt; .01]

my_palette &lt;- colorRampPalette(c(&quot;red&quot;, &quot;yellow&quot;, &quot;blue&quot;))(n = 299)

heatmap.2(as.matrix(molecules_qc[order_cv, ]), 
          symm = FALSE, col = my_palette,
          Colv = &quot;NA&quot;, trace = &quot;none&quot;, labRow = gene_symbols$symbol[ order_cv ])</code></pre>
<p><img src="figure/cv-mean.Rmd/unnamed-chunk-20-1.png" title="" alt="" width="1440" style="display: block; margin: auto;" /></p>
<p>Use a different method to get gene ID.</p>
<pre class="r"><code>library(&quot;biomaRt&quot;)
ensembl &lt;- useMart(host = &quot;grch37.ensembl.org&quot;,
                   biomart = &quot;ENSEMBL_MART_ENSEMBL&quot;,
                   dataset = &quot;hsapiens_gene_ensembl&quot;)

differential_CV_genes_info &lt;- getBM(attributes = c(&quot;ensembl_gene_id&quot;, &quot;chromosome_name&quot;,
                                                 &quot;external_gene_name&quot;, &quot;transcript_count&quot;,
                                                 &quot;description&quot;),
                                  filters = &quot;ensembl_gene_id&quot;,
                                  values = rownames(molecules_qc[order_cv, ]),
                                  mart = ensembl)

differential_CV_genes_info</code></pre>
<pre><code>   ensembl_gene_id chromosome_name external_gene_name transcript_count
1  ENSG00000014216              11              CAPN1               28
2  ENSG00000025800               1              KPNA6                6
3  ENSG00000060069              18              CTDP1                7
4  ENSG00000065135               1              GNAI3                1
5  ENSG00000074695              18              LMAN1                5
6  ENSG00000076053              11               RBM7                8
7  ENSG00000077097               3              TOP2B                9
8  ENSG00000086712               X              TXLNG                3
9  ENSG00000100266              22            PACSIN2               11
10 ENSG00000100504              14               PYGL                7
11 ENSG00000101115              20              SALL4                5
12 ENSG00000105270              19              CLIP3                6
13 ENSG00000113273               5               ARSB                6
14 ENSG00000114331               3              ACAP2               14
15 ENSG00000114742               3              WDR48               13
16 ENSG00000119965              10           C10orf88                4
17 ENSG00000123268              12               ATF1                6
18 ENSG00000124164              20               VAPB                6
19 ENSG00000124613               6             ZNF391                4
20 ENSG00000129521              14              EGLN3               11
21 ENSG00000130032               X              PRRG3                5
22 ENSG00000130164              19               LDLR               14
23 ENSG00000130305               7              NSUN5                7
24 ENSG00000134779              18              TPGS2               19
25 ENSG00000134901              13             KDELC1                4
26 ENSG00000135775               1               COG2               11
27 ENSG00000135913               2              USP37               11
28 ENSG00000136141              13              LRCH1                6
29 ENSG00000136830               9            FAM129B                7
30 ENSG00000142208              14               AKT1               18
31 ENSG00000142875               1             PRKACB               16
32 ENSG00000155252              10             PI4K2A                2
33 ENSG00000156603              11              MED19                4
34 ENSG00000160953              19               MUM1               18
35 ENSG00000161980              16             POLR3K                2
36 ENSG00000164031               4            DNAJB14               11
37 ENSG00000169018              15              FEM1B                4
38 ENSG00000169627              16             BOLA2B                4
39 ENSG00000170145              11               SIK2                2
40 ENSG00000171202              11           TMEM126A                6
41 ENSG00000172057              17             ORMDL3                8
42 ENSG00000172071               2            EIF2AK3                6
43 ENSG00000172936               3              MYD88               11
44 ENSG00000173786              17                CNP               13
45 ENSG00000177045              19               SIX5                3
46 ENSG00000180035              16              ZNF48                4
47 ENSG00000186665              17           C17orf58                3
48 ENSG00000196374               6          HIST1H2BM                1
49 ENSG00000197713               2                RPE               17
50 ENSG00000205572               5             SERF1B                8
51 ENSG00000213341              10               CHUK                4
52 ENSG00000236287              11              ZBED5               10
53 ENSG00000253710              13              ALG11                3
54 ENSG00000254901              19            MEF2BNB                9
                                                                                                             description
1                                                          calpain 1, (mu/I) large subunit [Source:HGNC Symbol;Acc:1476]
2                                                   karyopherin alpha 6 (importin alpha 7) [Source:HGNC Symbol;Acc:6399]
3   CTD (carboxy-terminal domain, RNA polymerase II, polypeptide A) phosphatase, subunit 1 [Source:HGNC Symbol;Acc:2498]
4  guanine nucleotide binding protein (G protein), alpha inhibiting activity polypeptide 3 [Source:HGNC Symbol;Acc:4387]
5                                                               lectin, mannose-binding, 1 [Source:HGNC Symbol;Acc:6631]
6                                                              RNA binding motif protein 7 [Source:HGNC Symbol;Acc:9904]
7                                                      topoisomerase (DNA) II beta 180kDa [Source:HGNC Symbol;Acc:11990]
8                                                                           taxilin gamma [Source:HGNC Symbol;Acc:18578]
9                                protein kinase C and casein kinase substrate in neurons 2 [Source:HGNC Symbol;Acc:8571]
10                                                          phosphorylase, glycogen, liver [Source:HGNC Symbol;Acc:9725]
11                                                      spalt-like transcription factor 4 [Source:HGNC Symbol;Acc:15924]
12                                             CAP-GLY domain containing linker protein 3 [Source:HGNC Symbol;Acc:24314]
13                                                                          arylsulfatase B [Source:HGNC Symbol;Acc:714]
14                               ArfGAP with coiled-coil, ankyrin repeat and PH domains 2 [Source:HGNC Symbol;Acc:16469]
15                                                                    WD repeat domain 48 [Source:HGNC Symbol;Acc:30914]
16                                                    chromosome 10 open reading frame 88 [Source:HGNC Symbol;Acc:25822]
17                                                        activating transcription factor 1 [Source:HGNC Symbol;Acc:783]
18                  VAMP (vesicle-associated membrane protein)-associated protein B and C [Source:HGNC Symbol;Acc:12649]
19                                                                zinc finger protein 391 [Source:HGNC Symbol;Acc:18779]
20                                                egl-9 family hypoxia-inducible factor 3 [Source:HGNC Symbol;Acc:14661]
21                            proline rich Gla (G-carboxyglutamic acid) 3 (transmembrane) [Source:HGNC Symbol;Acc:30798]
22                                                        low density lipoprotein receptor [Source:HGNC Symbol;Acc:6547]
23                                                       NOP2/Sun domain family, member 5 [Source:HGNC Symbol;Acc:16385]
24                                              tubulin polyglutamylase complex subunit 2 [Source:HGNC Symbol;Acc:24561]
25                                                    KDEL (Lys-Asp-Glu-Leu) containing 1 [Source:HGNC Symbol;Acc:19350]
26                                                 component of oligomeric golgi complex 2 [Source:HGNC Symbol;Acc:6546]
27                                                        ubiquitin specific peptidase 37 [Source:HGNC Symbol;Acc:20063]
28                    leucine-rich repeats and calponin homology (CH) domain containing 1 [Source:HGNC Symbol;Acc:20309]
29                                          family with sequence similarity 129, member B [Source:HGNC Symbol;Acc:25282]
30                                            v-akt murine thymoma viral oncogene homolog 1 [Source:HGNC Symbol;Acc:391]
31                                         protein kinase, cAMP-dependent, catalytic, beta [Source:HGNC Symbol;Acc:9381]
32                                             phosphatidylinositol 4-kinase type 2 alpha [Source:HGNC Symbol;Acc:30031]
33                                                            mediator complex subunit 19 [Source:HGNC Symbol;Acc:29600]
34                                                melanoma associated antigen (mutated) 1 [Source:HGNC Symbol;Acc:29641]
35                            polymerase (RNA) III (DNA directed) polypeptide K, 12.3 kDa [Source:HGNC Symbol;Acc:14121]
36                                           DnaJ (Hsp40) homolog, subfamily B, member 14 [Source:HGNC Symbol;Acc:25881]
37                                                            fem-1 homolog b (C. elegans) [Source:HGNC Symbol;Acc:3649]
38                                                                  bolA family member 2B [Source:HGNC Symbol;Acc:32479]
39                                                                salt-inducible kinase 2 [Source:HGNC Symbol;Acc:21680]
40                                                             transmembrane protein 126A [Source:HGNC Symbol;Acc:25382]
41                                                            ORM1-like 3 (S. cerevisiae) [Source:HGNC Symbol;Acc:16038]
42                               eukaryotic translation initiation factor 2-alpha kinase 3 [Source:HGNC Symbol;Acc:3255]
43                                             myeloid differentiation primary response 88 [Source:HGNC Symbol;Acc:7562]
44                                            2&#39;,3&#39;-cyclic nucleotide 3&#39; phosphodiesterase [Source:HGNC Symbol;Acc:2158]
45                                                                         SIX homeobox 5 [Source:HGNC Symbol;Acc:10891]
46                                                                 zinc finger protein 48 [Source:HGNC Symbol;Acc:13114]
47                                                    chromosome 17 open reading frame 58 [Source:HGNC Symbol;Acc:27568]
48                                                                 histone cluster 1, H2bm [Source:HGNC Symbol;Acc:4750]
49                                                       ribulose-5-phosphate-3-epimerase [Source:HGNC Symbol;Acc:10293]
50                                                small EDRK-rich factor 1B (centromeric) [Source:HGNC Symbol;Acc:10756]
51                                            conserved helix-loop-helix ubiquitous kinase [Source:HGNC Symbol;Acc:1974]
52                                                     zinc finger, BED-type containing 5 [Source:HGNC Symbol;Acc:30803]
53                                                   ALG11, alpha-1,2-mannosyltransferase [Source:HGNC Symbol;Acc:32456]
54                                                                         MEF2B neighbor [Source:HGNC Symbol;Acc:37247]</code></pre>
<p>Use <a href="http://string-db.org">string analysis</a> data base to look for protein protein interaction of these differential CV genes.</p>
<p>This is the confidence view. Stronger associations are represented by thicker lines. <img src="figure/cv-mean.Rmd/string1.png" alt="This is the confidence view" /> This is the evidence view. Different line colors represent the types of evidence for the association. <img src="figure/cv-mean.Rmd/string2.png" alt="This is the evidence view" /></p>
</div>
</div>
<div id="session-information" class="section level2">
<h2>Session information</h2>
<pre class="r"><code>sessionInfo()</code></pre>
<pre><code>R version 3.2.0 (2015-04-16)
Platform: x86_64-unknown-linux-gnu (64-bit)

locale:
 [1] LC_CTYPE=en_US.UTF-8       LC_NUMERIC=C              
 [3] LC_TIME=en_US.UTF-8        LC_COLLATE=en_US.UTF-8    
 [5] LC_MONETARY=en_US.UTF-8    LC_MESSAGES=en_US.UTF-8   
 [7] LC_PAPER=en_US.UTF-8       LC_NAME=C                 
 [9] LC_ADDRESS=C               LC_TELEPHONE=C            
[11] LC_MEASUREMENT=en_US.UTF-8 LC_IDENTIFICATION=C       

attached base packages:
 [1] stats4    parallel  grid      stats     graphics  grDevices utils    
 [8] datasets  methods   base     

other attached packages:
 [1] biomaRt_2.24.0         gplots_2.17.0          mygene_1.2.3          
 [4] GenomicFeatures_1.20.1 AnnotationDbi_1.30.1   Biobase_2.28.0        
 [7] GenomicRanges_1.20.5   GenomeInfoDb_1.4.0     IRanges_2.2.4         
[10] S4Vectors_0.6.0        BiocGenerics_0.14.0    broman_0.59-5         
[13] zoo_1.7-12             ggplot2_1.0.1          edgeR_3.10.2          
[16] limma_3.24.9           dplyr_0.4.2            data.table_1.9.4      
[19] knitr_1.10.5          

loaded via a namespace (and not attached):
 [1] Rcpp_0.12.0             lattice_0.20-31        
 [3] gtools_3.5.0            Rsamtools_1.20.4       
 [5] Biostrings_2.36.1       assertthat_0.1         
 [7] digest_0.6.8            R6_2.1.1               
 [9] plyr_1.8.3              chron_2.3-45           
[11] futile.options_1.0.0    acepack_1.3-3.3        
[13] RSQLite_1.0.0           evaluate_0.7           
[15] sqldf_0.4-10            httr_0.6.1             
[17] zlibbioc_1.14.0         gdata_2.16.1           
[19] rpart_4.1-9             gsubfn_0.6-6           
[21] rmarkdown_0.6.1         proto_0.3-10           
[23] splines_3.2.0           BiocParallel_1.2.2     
[25] foreign_0.8-63          stringr_1.0.0          
[27] RCurl_1.95-4.6          munsell_0.4.2          
[29] rtracklayer_1.28.4      htmltools_0.2.6        
[31] nnet_7.3-9              gridExtra_0.9.1        
[33] Hmisc_3.16-0            XML_3.98-1.2           
[35] GenomicAlignments_1.4.1 MASS_7.3-40            
[37] bitops_1.0-6            jsonlite_0.9.16        
[39] gtable_0.1.2            DBI_0.3.1              
[41] magrittr_1.5            formatR_1.2            
[43] scales_0.2.4            KernSmooth_2.23-14     
[45] stringi_0.4-1           XVector_0.8.0          
[47] reshape2_1.4.1          latticeExtra_0.6-26    
[49] futile.logger_1.4.1     Formula_1.2-1          
[51] RColorBrewer_1.1-2      lambda.r_1.1.7         
[53] tools_3.2.0             survival_2.38-1        
[55] yaml_2.1.13             colorspace_1.2-6       
[57] cluster_2.0.1           caTools_1.17.1         </code></pre>
</div>


<!-- some extra javascript for older browsers -->
<script type="text/javascript" src="libs/polyfill.js"></script>

<script>

// manage active state of menu based on current page
$(document).ready(function () {

    // active menu
    href = window.location.pathname
    href = href.substr(href.lastIndexOf('/') + 1)
    $('a[href="' + href + '"]').parent().addClass('active');

    // manage active menu header
    if (href.startsWith('authoring_'))
      $('a[href="' + 'authoring' + '"]').parent().addClass('active');
    else if (href.endsWith('_format.html'))
      $('a[href="' + 'formats' + '"]').parent().addClass('active');
    else if (href.startsWith('developer_'))
      $('a[href="' + 'developer' + '"]').parent().addClass('active');

});

</script>

</div>

<script>

// add bootstrap table styles to pandoc tables
$(document).ready(function () {
  $('tr.header').parent('thead').parent('table').addClass('table table-condensed');
});

</script>


</body>
</html>
