<!DOCTYPE html>

<html xmlns="http://www.w3.org/1999/xhtml">

<head>

<meta charset="utf-8">
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="generator" content="pandoc" />

<meta name="author" content="Joyce Hsiao" />

<meta name="date" content="2015-10-13" />

<title>Individual differences in CV and mean</title>

<script src="libs/jquery-1.11.0/jquery.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="libs/bootstrap-3.3.1/css/united.min.css" rel="stylesheet" />
<script src="libs/bootstrap-3.3.1/js/bootstrap.min.js"></script>
<script src="libs/bootstrap-3.3.1/shim/html5shiv.min.js"></script>
<script src="libs/bootstrap-3.3.1/shim/respond.min.js"></script>

<style type="text/css">

/* padding for bootstrap navbar */
body {
  padding-top: 50px;
  padding-bottom: 40px;
}


/* offset scroll position for anchor links (for fixed navbar)  */
.section h2 {
  padding-top: 55px;
  margin-top: -55px;
}
.section h3 {
  padding-top: 55px;
  margin-top: -55px;
}



/* don't use link color in navbar */
.dropdown-menu>li>a {
  color: black;
}

/* some padding for disqus */
#disqus_thread {
  margin-top: 45px;
}

</style>

<link rel="stylesheet" href="libs/font-awesome-4.1.0/css/font-awesome.min.css"/>

<style type="text/css">code{white-space: pre;}</style>
<link rel="stylesheet"
      href="libs/highlight/textmate.css"
      type="text/css" />
<script src="libs/highlight/highlight.js"></script>
<style type="text/css">
  pre:not([class]) {
    background-color: white;
  }
</style>
<script type="text/javascript">
if (window.hljs && document.readyState && document.readyState === "complete") {
   window.setTimeout(function() {
      hljs.initHighlighting();
   }, 0);
}
</script>



</head>

<body>

<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
code {
  color: inherit;
  background-color: rgba(0, 0, 0, 0.04);
}
img { 
  max-width:100%; 
  height: auto; 
}
</style>
<div class="container-fluid main-container">


<div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">singleCellSeq</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li><a href="index.html">Home</a></li>
        <li><a href="about.html">About</a></li>
        <li><a href="license.html">License</a></li>
        <li><a href="https://github.com/jdblischak/singleCellSeq">GitHub</a></li>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<div id="header">
<h1 class="title">Individual differences in CV and mean</h1>
<h4 class="author"><em>Joyce Hsiao</em></h4>
<h4 class="date"><em>2015-10-13</em></h4>
</div>

<div id="TOC">
<ul>
<li><a href="#objective">Objective</a></li>
<li><a href="#set-up">Set up</a></li>
<li><a href="#prepare-data">Prepare data</a></li>
<li><a href="#compare-coefficient-of-variation">Compare coefficient of variation</a></li>
<li><a href="#compare-mean-expression-levels.">Compare mean expression levels.</a><ul>
<li><a href="#single-cell-sequencing-data">Single-cell sequencing data</a></li>
</ul></li>
<li><a href="#session-information">Session information</a></li>
</ul>
</div>

<p><strong>Last updated:</strong> 2015-10-16</p>
<p><strong>Code version:</strong> 62ced37788d540a4808e1a509096c32704067abe</p>
<div id="objective" class="section level2">
<h2>Objective</h2>
<p>Previously, we compared normalized coefficient of variations across individuals. Here, we will also compare the mean gene expression across individuals, using single cell sequencing data and bulk RNA-seq data.</p>
<p>It would be interesting to learn the possible overlap or non-overlap between the genes that we observed significant individual differences in coefficient of variations versus those different in mean gene expression levels across cells.</p>
</div>
<div id="set-up" class="section level2">
<h2>Set up</h2>
<pre class="r"><code>library(&quot;data.table&quot;)
library(&quot;dplyr&quot;)
library(&quot;limma&quot;)
library(&quot;edgeR&quot;)
library(&quot;ggplot2&quot;)
library(&quot;grid&quot;)
theme_set(theme_bw(base_size = 12))
source(&quot;functions.R&quot;)</code></pre>
</div>
<div id="prepare-data" class="section level2">
<h2>Prepare data</h2>
<p>Input annotation of only QC-filtered single cells</p>
<pre class="r"><code>anno_qc &lt;- read.table(&quot;../data/annotation-filter.txt&quot;, header = TRUE,
                   stringsAsFactors = FALSE)
head(anno_qc)</code></pre>
<pre><code>  individual replicate well      batch      sample_id
1    NA19098        r1  A01 NA19098.r1 NA19098.r1.A01
2    NA19098        r1  A02 NA19098.r1 NA19098.r1.A02
3    NA19098        r1  A04 NA19098.r1 NA19098.r1.A04
4    NA19098        r1  A05 NA19098.r1 NA19098.r1.A05
5    NA19098        r1  A06 NA19098.r1 NA19098.r1.A06
6    NA19098        r1  A07 NA19098.r1 NA19098.r1.A07</code></pre>
<p>Input molecule counts that are filtered, transformed, and also processed to remove unwanted variation from batch effet. ERCC genes are also removed.</p>
<pre class="r"><code>molecules_qc &lt;- read.table(&quot;../data/molecules-final.txt&quot;, header = TRUE, stringsAsFactors = FALSE)</code></pre>
<p>Input moleclule counts before log2 CPM transformation.</p>
<pre class="r"><code>molecules_filter &lt;- read.table(&quot;../data/molecules-filter.txt&quot;, header = TRUE, stringsAsFactors = FALSE)
molecules_filter &lt;- molecules_filter[which(rownames(molecules_filter) %in% rownames(molecules_qc)), ]

stopifnot(dim(molecules_filter) == dim(molecules_qc))</code></pre>
</div>
<div id="compare-coefficient-of-variation" class="section level2">
<h2>Compare coefficient of variation</h2>
<p>Compute per batch coefficient of variation.</p>
<p>Include only genes with positive coefficient of variation. Some genes in this data may have zero coefficient of variation, because we include gene with more than 0 count across all cells.</p>
<pre class="r"><code>molecules_cv_batch &lt;- 
  lapply(1:length(unique(anno_qc$batch)), function(per_batch) {
      molecules_per_batch &lt;- 2^molecules_qc[ , unique(anno_qc$batch) == unique(anno_qc$batch)[per_batch] ]
      mean_per_gene &lt;- apply(molecules_per_batch, 1, mean, na.rm = TRUE)
      sd_per_gene &lt;- apply(molecules_per_batch, 1, sd, na.rm = TRUE)
      cv_per_gene &lt;- data.frame(mean = mean_per_gene,
                                sd = sd_per_gene,
                                cv = sd_per_gene/mean_per_gene)
      rownames(cv_per_gene) &lt;- rownames(molecules_qc)
  
      cv_per_gene &lt;- cv_per_gene[rowSums(is.na(cv_per_gene)) == 0, ]
      cv_per_gene$batch &lt;- unique(anno_qc$batch)[per_batch]
      
      # Add sparsity percent
      molecules_count &lt;- molecules_filter[ , unique(anno_qc$batch) == unique(anno_qc$batch)[per_batch]]
      cv_per_gene$sparse &lt;- rowMeans(as.matrix(molecules_count) == 0)
        
      return(cv_per_gene)
      }) 
names(molecules_cv_batch) &lt;- unique(anno_qc$batch)

dim(molecules_cv_batch[[1]])</code></pre>
<pre><code>[1] 10483     5</code></pre>
<p>Merge summary data.frames.</p>
<pre class="r"><code>df_plot &lt;- do.call(rbind, molecules_cv_batch)</code></pre>
<p>Compute rolling medians.</p>
<pre class="r"><code>library(zoo)
# Compute a data-wide coefficient of variation on CPM normalized counts.
data_cv &lt;- apply(2^molecules_qc, 1, sd)/apply(2^molecules_qc, 1, mean)

# Order of genes by mean expression levels
order_gene &lt;- order(apply(2^molecules_qc, 1, mean))

# Rolling medians of log10 squared CV by mean expression levels
roll_medians &lt;- rollapply(log10(data_cv^2)[order_gene], width = 50, by = 25,
                         FUN = median, fill = list(&quot;extend&quot;, &quot;extend&quot;, &quot;NA&quot;) )
ii_na &lt;- which( is.na(roll_medians) )
roll_medians[ii_na] &lt;- median( log10(data_cv^2)[order_gene][ii_na] )

names(roll_medians) &lt;- rownames(molecules_qc)[order_gene]

# re-order rolling medians
reorder_gene &lt;- match(rownames(molecules_qc), names(roll_medians) )
head(reorder_gene)</code></pre>
<pre><code>[1]  225 3650 4564 6158 3352 1963</code></pre>
<pre class="r"><code>roll_medians &lt;- roll_medians[ reorder_gene ]

head(names(roll_medians))</code></pre>
<pre><code>[1] &quot;ENSG00000237683&quot; &quot;ENSG00000188976&quot; &quot;ENSG00000187608&quot; &quot;ENSG00000188157&quot;
[5] &quot;ENSG00000078808&quot; &quot;ENSG00000176022&quot;</code></pre>
<pre class="r"><code>head(rownames(molecules_qc))</code></pre>
<pre><code>[1] &quot;ENSG00000237683&quot; &quot;ENSG00000188976&quot; &quot;ENSG00000187608&quot; &quot;ENSG00000188157&quot;
[5] &quot;ENSG00000078808&quot; &quot;ENSG00000176022&quot;</code></pre>
<p>Compute adjusted coefficient of variation.</p>
<pre class="r"><code># adjusted coefficient of variation on log10 scale
log10cv2_adj &lt;- 
  lapply(1:length(molecules_cv_batch), function(per_batch) {
    foo &lt;- log10(molecules_cv_batch[[per_batch]]$cv^2) - roll_medians
    return(foo)
})
df_plot$log10cv2_adj &lt;- do.call(c, log10cv2_adj)</code></pre>
<p>Compare adjusted CVs</p>
<pre class="r"><code>library(limma)

df_limma &lt;- matrix(df_plot$log10cv2_adj, 
                      nrow = nrow(molecules_qc), ncol = 9, byrow = FALSE)

design &lt;- data.frame(individual = factor(rep(unique(anno_qc$individual), each = 3) ),
                     rep = factor(rep(c(1:3), times = 3)) )

colnames(df_limma) &lt;- with(design, paste0(individual, rep))

fit_limma &lt;- lmFit(df_limma, design = model.matrix( ~ individual, data = design))                      
fit_limma &lt;- eBayes(fit_limma)</code></pre>
<p>False discover control adjustment.</p>
<pre class="r"><code>F.p.adj &lt;- p.adjust(fit_limma$F.p.value, method = &quot;fdr&quot;)</code></pre>
<p>Cutoffs</p>
<pre class="r"><code>df_cuts &lt;- data.frame(cuts = c(.001, .01, .05, .1, .15, .2))
df_cuts$sig_count &lt;- sapply(1:6, function(per_cut) {
                    sum(F.p.adj &lt; df_cuts$cuts[per_cut] )
                    })
df_cuts</code></pre>
<pre><code>   cuts sig_count
1 0.001        24
2 0.010        54
3 0.050       115
4 0.100       179
5 0.150       290
6 0.200       423</code></pre>
</div>
<div id="compare-mean-expression-levels." class="section level2">
<h2>Compare mean expression levels.</h2>
<div id="single-cell-sequencing-data" class="section level3">
<h3>Single-cell sequencing data</h3>
<p>Almost all of the genes came out to be statistically significant due the large sample size (about ~70 sample points per replicate, per individual).</p>
<pre class="r"><code>fit_limma_mean &lt;- lmFit(molecules_qc, 
                        design = model.matrix(~factor(individual), data = anno_qc) )

fit_limma_mean &lt;- eBayes(fit_limma_mean)

summary(fit_limma_mean$F.p.value)</code></pre>
<pre><code>      Min.    1st Qu.     Median       Mean    3rd Qu.       Max. 
 0.000e+00  0.000e+00  0.000e+00 1.985e-194  0.000e+00 1.180e-190 </code></pre>
<p>Histogram of p-values.</p>
<pre class="r"><code>hist(-log10(fit_limma_mean$F.p.value), 
     main = &quot;F-test p-values comparing mean expression&quot;, breaks = 100)</code></pre>
<p><img src="figure/cv-mean.Rmd/unnamed-chunk-13-1.png" title="" alt="" width="672" style="display: block; margin: auto;" /></p>
<p>FDR</p>
<pre class="r"><code>padj_mean_single &lt;- p.adjust(fit_limma_mean$F.p.value, method = &quot;fdr&quot;)
summary(padj_mean_single)</code></pre>
<pre><code>      Min.    1st Qu.     Median       Mean    3rd Qu.       Max. 
 0.000e+00  0.000e+00  0.000e+00 1.985e-194  0.000e+00 1.180e-190 </code></pre>
<p>Mean versus coefficient of variation comparisons</p>
<pre class="r"><code>plot(x = -log10(fit_limma_mean$F.p.value), 
     y = -log10(fit_limma$F.p.value),
     xlab = &quot;-log10 F-test p-value comparing means&quot;,
     ylab = &quot;-log10 F-test p-value compare adjusted CV&quot;,
     main = &quot;Unadjusted F-test p-values&quot;)
abline( h = -log10(.01), col = &quot;red&quot;)</code></pre>
<p><img src="figure/cv-mean.Rmd/unnamed-chunk-15-1.png" title="" alt="" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>plot(x = -log10(padj_mean_single), 
     y = -log10(F.p.adj),
     xlab = &quot;-log10 FDR comparing means&quot;,
     ylab = &quot;-log10 FDSR compare adjusted CV&quot;,
     main = &quot;FDR&quot;)
abline( h = -log10(.01), col = &quot;red&quot;)
abline( h = -log10(.05), col = &quot;red&quot;)</code></pre>
<p><img src="figure/cv-mean.Rmd/unnamed-chunk-15-2.png" title="" alt="" width="672" style="display: block; margin: auto;" /></p>
<p>Overlap of the significant genes</p>
<p>*Adjusted p-value cut-offs</p>
<pre class="r"><code>cuts &lt;- c(.001, .01, .05, .1, .15, .2)
both_table &lt;- lapply(1:6, function(per_cut) {
                  table(F.p.adj &lt; df_cuts$cuts[per_cut],
                        padj_mean_single &lt; df_cuts$cut[per_cut])
              })
names(both_table) &lt;- cuts

both_table</code></pre>
<pre><code>$`0.001`
       
         TRUE
  FALSE 10459
  TRUE     24

$`0.01`
       
         TRUE
  FALSE 10429
  TRUE     54

$`0.05`
       
         TRUE
  FALSE 10368
  TRUE    115

$`0.1`
       
         TRUE
  FALSE 10304
  TRUE    179

$`0.15`
       
         TRUE
  FALSE 10193
  TRUE    290

$`0.2`
       
         TRUE
  FALSE 10060
  TRUE    423</code></pre>
<p>*Ranks of adjusted p-values</p>
<pre class="r"><code>plot(x = rank(-log10(padj_mean_single)), 
     y = rank(-log10(F.p.adj)),
     xlab = &quot;rank of -log10 FDR comparing means&quot;,
     ylab = &quot;rank of -log10 FDR compare adjusted CV&quot;,
     main = &quot;FDR&quot;)</code></pre>
<p><img src="figure/cv-mean.Rmd/unnamed-chunk-17-1.png" title="" alt="" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>df_compare &lt;- 
  data.frame(mean = rowMeans( as.matrix( 
                      do.call(cbind, lapply(molecules_cv_batch, &quot;[[&quot;, 1) ) ) ),
             cv2 = rowMeans( as.matrix( 
                      do.call(cbind, lapply(molecules_cv_batch, &quot;[[&quot;, 3) ) )^2 ),
             adj_cv2 = rowMeans( 10^as.matrix( 
                      do.call(cbind, log10cv2_adj) ) ) )
             
library(broman)
crayon &lt;- brocolors(&quot;crayons&quot;)
with(df_compare, plot(x = log10(mean), y = cv2, pch = 1, cex = 1, col = &quot;grey50&quot;,
                      lwd = .5,
                      ylab = &quot;average squared coeffcient of variations \n across samples&quot;,
                      xlab = &quot;log10 average of mean CPM across samples&quot;) )
with(df_compare[F.p.adj &lt; .2, ], points(x = log10(mean), y = cv2, pch = 16, cex = .6,
                                        col = crayon[&quot;Tumbleweed&quot;]) )
with(df_compare[F.p.adj &lt; .1, ], points(x = log10(mean), y = cv2, pch = 16, cex = .6,
                                        col = crayon[&quot;Orange&quot;]) )
with(df_compare[F.p.adj &lt; .01, ], points(x = log10(mean), y = cv2, pch = 16, cex = .6,
                                        col = crayon[&quot;Scarlet&quot;]) )
title(main = &quot;Avg. Squred CV vs. log10(Avg. mean count)&quot;)
legend(&quot;topright&quot;, pch = c(1, 16, 16, 16), 
       legend = c(&quot;All genes&quot;, &quot;Adj p-value &lt; .2&quot;, 
                  &quot;Adj p-value &lt; .1&quot;, &quot;Adj p-value &lt; .01&quot;),
       col = c(&quot;grey50&quot;, crayon[c(&quot;Tumbleweed&quot;, &quot;Orange&quot;, &quot;Scarlet&quot;)]),
       bty = &quot;n&quot;)</code></pre>
<p><img src="figure/cv-mean.Rmd/unnamed-chunk-18-1.png" title="" alt="" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>with(df_compare, plot(x = log10(mean), y = adj_cv2, pch = 1, cex = 1, col = &quot;grey50&quot;,
                      lwd = .5,
                      ylab = &quot;average squared coeffcient of variations \n across samples&quot;,
                      xlab = &quot;log10 average of mean CPM across samples&quot;) )
with(df_compare[F.p.adj &lt; .2, ], points(x = log10(mean), y = adj_cv2, pch = 16, cex = .6,
                                        col = crayon[&quot;Tumbleweed&quot;]) )
with(df_compare[F.p.adj &lt; .1, ], points(x = log10(mean), y = adj_cv2, pch = 16, cex = .6,
                                        col = crayon[&quot;Orange&quot;]) )
with(df_compare[F.p.adj &lt; .01, ], points(x = log10(mean), y = adj_cv2, pch = 16, cex = .6,
                                        col = crayon[&quot;Scarlet&quot;]) )
title(main = &quot;Avg. Adjusted Squred CV vs. log10(Avg. mean count)&quot;)
legend(&quot;topright&quot;, pch = c(1, 16, 16, 16), 
       legend = c(&quot;All genes&quot;, &quot;Adj p-value &lt; .2&quot;, 
                  &quot;Adj p-value &lt; .1&quot;, &quot;Adj p-value &lt; .01&quot;),
       col = c(&quot;grey50&quot;, crayon[c(&quot;Tumbleweed&quot;, &quot;Orange&quot;, &quot;Scarlet&quot;)]),
       bty = &quot;n&quot;)</code></pre>
<p><img src="figure/cv-mean.Rmd/unnamed-chunk-18-2.png" title="" alt="" width="672" style="display: block; margin: auto;" /></p>
<p>Heatmap</p>
<pre class="r"><code>library(mygene)
gene_symbols &lt;- getGenes(rownames(molecules_qc), 
                         fields = c(&quot;name&quot;, &quot;symbol&quot;) ) [ , c(&quot;query&quot;, &quot;symbol&quot;)]
head(gene_symbols)</code></pre>
<pre><code>DataFrame with 6 rows and 2 columns
            query      symbol
      &lt;character&gt; &lt;character&gt;
1 ENSG00000237683          NA
2 ENSG00000188976       NOC2L
3 ENSG00000187608       ISG15
4 ENSG00000188157        AGRN
5 ENSG00000078808        SDF4
6 ENSG00000176022     B3GALT6</code></pre>
<pre class="r"><code>library(gplots)

order_cv &lt;- order(F.p.adj)[F.p.adj &lt; .01]

my_palette &lt;- colorRampPalette(c(&quot;red&quot;, &quot;yellow&quot;, &quot;blue&quot;))(n = 299)

heatmap.2(as.matrix(molecules_qc[order_cv, ]), 
          symm = FALSE, col = my_palette,
          Colv = &quot;NA&quot;, trace = &quot;none&quot;, labRow = gene_symbols$symbol[ order_cv ])</code></pre>
<p><img src="figure/cv-mean.Rmd/unnamed-chunk-20-1.png" title="" alt="" width="1440" style="display: block; margin: auto;" /></p>
<pre class="r"><code>kable(queryMany(rownames(molecules_qc)[which(F.p.adj&lt; .01)], scopes = &quot;ensembl.gene&quot;, 
                 fields=c(&quot;summary&quot;), species=&quot;human&quot;) )</code></pre>
<pre><code>Finished
Pass returnall=TRUE to return lists of duplicate or missing query terms.</code></pre>
<table>
<thead>
<tr class="header">
<th align="left">summary</th>
<th align="left">query</th>
<th align="left">_id</th>
<th align="left">notfound</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td align="left">The cannabinoid delta-9-tetrahydrocannabinol is the principal psychoactive ingredient of marijuana. The proteins encoded by this gene and the cannabinoid receptor 1 (brain) (CNR1) gene have the characteristics of a guanine nucleotide-binding protein (G-protein)-coupled receptor for cannabinoids. They inhibit adenylate cyclase activity in a dose-dependent, stereoselective, and pertussis toxin-sensitive manner. These proteins have been found to be involved in the cannabinoid-induced CNS effects (including alterations in mood and cognition) experienced by users of marijuana. The cannabinoid receptors are members of family 1 of the G-protein-coupled receptors.</td>
<td align="left">ENSG00000188822</td>
<td align="left">1269</td>
<td align="left">NA</td>
</tr>
<tr class="even">
<td align="left">The secreted protein encoded by this gene is growth factor-inducible and promotes the adhesion of endothelial cells. The encoded protein interacts with several integrins and with heparan sulfate proteoglycan. This protein also plays a role in cell proliferation, differentiation, angiogenesis, apoptosis, and extracellular matrix formation.</td>
<td align="left">ENSG00000142871</td>
<td align="left">3491</td>
<td align="left">NA</td>
</tr>
<tr class="odd">
<td align="left">NA</td>
<td align="left">ENSG00000236334</td>
<td align="left">644591</td>
<td align="left">NA</td>
</tr>
<tr class="even">
<td align="left">NA</td>
<td align="left">ENSG00000168614</td>
<td align="left">NA</td>
<td align="left">TRUE</td>
</tr>
<tr class="odd">
<td align="left">The protein encoded by this gene is a member of the ADAM (a disintegrin and metalloproteinase) protein family. ADAM family members are type I transmembrane glycoproteins known to be involved in cell adhesion and proteolytic ectodomain processing of cytokines and adhesion molecules. This protein contains multiple functional domains including a zinc-binding metalloprotease domain, a disintegrin-like domain, as well as a EGF-like domain. Through its disintegrin-like domain, this protein specifically interacts with the integrin beta chain, beta 3. It also interacts with Src family protein-tyrosine kinases in a phosphorylation-dependent manner, suggesting that this protein may function in cell-cell adhesion as well as in cellular signaling. Multiple alternatively spliced transcript variants encoding distinct isoforms have been observed.</td>
<td align="left">ENSG00000143537</td>
<td align="left">8751</td>
<td align="left">NA</td>
</tr>
<tr class="even">
<td align="left">This gene encodes a member of the regulators of G protein signaling (RGS) family. The RGS proteins are signal transduction molecules which are involved in the regulation of heterotrimeric G proteins by acting as GTPase activators. This gene is a hypoxia-inducible factor-1 dependent, hypoxia-induced gene which is involved in the induction of endothelial apoptosis. This gene is also one of three genes on chromosome 1q contributing to elevated blood pressure. Alternatively spliced transcript variants have been identified.</td>
<td align="left">ENSG00000143248</td>
<td align="left">8490</td>
<td align="left">NA</td>
</tr>
<tr class="odd">
<td align="left">NA</td>
<td align="left">ENSG00000117266</td>
<td align="left">5129</td>
<td align="left">NA</td>
</tr>
<tr class="even">
<td align="left">NA</td>
<td align="left">ENSG00000175137</td>
<td align="left">80851</td>
<td align="left">NA</td>
</tr>
<tr class="odd">
<td align="left">This gene encodes a member of the intermediate filament family. Intermediate filamentents, along with microtubules and actin microfilaments, make up the cytoskeleton. The protein encoded by this gene is responsible for maintaining cell shape, integrity of the cytoplasm, and stabilizing cytoskeletal interactions. It is also involved in the immune response, and controls the transport of low-density lipoprotein (LDL)-derived cholesterol from a lysosome to the site of esterification. It functions as an organizer of a number of critical proteins involved in attachment, migration, and cell signaling. Mutations in this gene causes a dominant, pulverulent cataract.</td>
<td align="left">ENSG00000026025</td>
<td align="left">7431</td>
<td align="left">NA</td>
</tr>
<tr class="even">
<td align="left">The protein encoded by this gene is an inositol 1,4,5-trisphosphate (InsP3) 5-phosphatase and contains a Sac domain. The activity of this protein is specific for phosphatidylinositol 4,5-bisphosphate and phosphatidylinositol 3,4,5-trisphosphate. Alternatively spliced transcript variants encoding multiple isoforms have been observed for this gene.</td>
<td align="left">ENSG00000198825</td>
<td align="left">22876</td>
<td align="left">NA</td>
</tr>
<tr class="odd">
<td align="left">NA</td>
<td align="left">ENSG00000185885</td>
<td align="left">8519</td>
<td align="left">NA</td>
</tr>
<tr class="even">
<td align="left">NA</td>
<td align="left">ENSG00000255823</td>
<td align="left">100463486</td>
<td align="left">NA</td>
</tr>
<tr class="odd">
<td align="left">This gene encodes the heavy subunit of ferritin, the major intracellular iron storage protein in prokaryotes and eukaryotes. It is composed of 24 subunits of the heavy and light ferritin chains. Variation in ferritin subunit composition may affect the rates of iron uptake and release in different tissues. A major function of ferritin is the storage of iron in a soluble and nontoxic state. Defects in ferritin proteins are associated with several neurodegenerative diseases. This gene has multiple pseudogenes. Several alternatively spliced transcript variants have been observed, but their biological validity has not been determined.</td>
<td align="left">ENSG00000167996</td>
<td align="left">2495</td>
<td align="left">NA</td>
</tr>
<tr class="even">
<td align="left">NA</td>
<td align="left">ENSG00000133316</td>
<td align="left">54663</td>
<td align="left">NA</td>
</tr>
<tr class="odd">
<td align="left">SLC38A4 is found predominantly in liver and transports both cationic and neutral amino acids. The transport of cationic amino acids by SLC38A4 is Na(+) and pH independent, while the transport of neutral amino acids is Na(+) and pH dependent (Hatanaka et al., 2001 [PubMed 11342143]).</td>
<td align="left">ENSG00000139209</td>
<td align="left">55089</td>
<td align="left">NA</td>
</tr>
<tr class="even">
<td align="left">This gene encodes a protein belonging to a large protein family, members of which carry the LIM domain, a unique cysteine-rich zinc-binding domain. The encoded protein may function as a transcriptional regulator and be involved in the control of differentiation and development of the forebrain. In mice, this protein is essential for the regulation of precursor cell proliferation and the control of neuronal differentiation and migration during hippocampal development. This protein is involved in learning and motor functions in adult mice.</td>
<td align="left">ENSG00000089116</td>
<td align="left">64211</td>
<td align="left">NA</td>
</tr>
<tr class="odd">
<td align="left">NA</td>
<td align="left">ENSG00000172717</td>
<td align="left">161142</td>
<td align="left">NA</td>
</tr>
<tr class="even">
<td align="left">NA</td>
<td align="left">ENSG00000187951</td>
<td align="left">89839</td>
<td align="left">NA</td>
</tr>
<tr class="odd">
<td align="left">NA</td>
<td align="left">ENSG00000187951</td>
<td align="left">100288637</td>
<td align="left">NA</td>
</tr>
<tr class="even">
<td align="left">This gene encodes a protein sharing 96% amino acid similarity with ribosomal protein S27, which suggests the encoded protein may be a component of the 40S ribosomal subunit.</td>
<td align="left">ENSG00000185088</td>
<td align="left">51065</td>
<td align="left">NA</td>
</tr>
<tr class="odd">
<td align="left">NA</td>
<td align="left">ENSG00000125148</td>
<td align="left">4502</td>
<td align="left">NA</td>
</tr>
<tr class="even">
<td align="left">NA</td>
<td align="left">ENSG00000169715</td>
<td align="left">4493</td>
<td align="left">NA</td>
</tr>
<tr class="odd">
<td align="left">NA</td>
<td align="left">ENSG00000198417</td>
<td align="left">4494</td>
<td align="left">NA</td>
</tr>
<tr class="even">
<td align="left">NA</td>
<td align="left">ENSG00000125144</td>
<td align="left">4495</td>
<td align="left">NA</td>
</tr>
<tr class="odd">
<td align="left">NA</td>
<td align="left">ENSG00000205358</td>
<td align="left">4496</td>
<td align="left">NA</td>
</tr>
<tr class="even">
<td align="left">NA</td>
<td align="left">ENSG00000187193</td>
<td align="left">4501</td>
<td align="left">NA</td>
</tr>
<tr class="odd">
<td align="left">This gene encodes a member of the profilin family of small actin-binding proteins. The encoded protein plays an important role in actin dynamics by regulating actin polymerization in response to extracellular signals. Deletion of this gene is associated with Miller-Dieker syndrome, and the encoded protein may also play a role in Huntington disease. Multiple pseudogenes of this gene are located on chromosome 1.</td>
<td align="left">ENSG00000108518</td>
<td align="left">5216</td>
<td align="left">NA</td>
</tr>
<tr class="even">
<td align="left">NA</td>
<td align="left">ENSG00000175106</td>
<td align="left">201158</td>
<td align="left">NA</td>
</tr>
<tr class="odd">
<td align="left">NA</td>
<td align="left">ENSG00000256618</td>
<td align="left">100462977</td>
<td align="left">NA</td>
</tr>
<tr class="even">
<td align="left">NALP proteins, such as NALP2, are characterized by an N-terminal pyrin (MIM 608107) domain (PYD) and are involved in the activation of caspase-1 (CASP1; MIM 147678) by Toll-like receptors (see TLR4; MIM 603030). They may also be involved in protein complexes that activate proinflammatory caspases (Tschopp et al., 2003 [PubMed 12563287]).</td>
<td align="left">ENSG00000022556</td>
<td align="left">55655</td>
<td align="left">NA</td>
</tr>
<tr class="odd">
<td align="left">NA</td>
<td align="left">ENSG00000196862</td>
<td align="left">285190</td>
<td align="left">NA</td>
</tr>
<tr class="even">
<td align="left">This gene encodes a protease inhibitor that regulates the tissue factor (TF)-dependent pathway of blood coagulation. The coagulation process initiates with the formation of a factor VIIa-TF complex, which proteolytically activates additional proteases (factors IX and X) and ultimately leads to the formation of a fibrin clot. The product of this gene inhibits the activated factor X and VIIa-TF proteases in an autoregulatory loop. The encoded protein is glycosylated and predominantly found in the vascular endothelium and plasma in both free forms and complexed with plasma lipoproteins. Several alternatively spliced transcript variants of this gene have been described, but the full-length nature of some of these variants has not been confirmed.</td>
<td align="left">ENSG00000003436</td>
<td align="left">7035</td>
<td align="left">NA</td>
</tr>
<tr class="odd">
<td align="left">This gene encodes a subnuclear organelle and major component of the PML (promyelocytic leukemia)-SP100 nuclear bodies. PML and SP100 are covalently modified by the SUMO-1 modifier, which is considered crucial to nuclear body interactions. The encoded protein binds heterochromatin proteins and is thought to play a role in tumorigenesis, immunity, and gene regulation. Alternatively spliced variants have been identified for this gene; one of which encodes a high-mobility group protein.</td>
<td align="left">ENSG00000067066</td>
<td align="left">6672</td>
<td align="left">NA</td>
</tr>
<tr class="even">
<td align="left">The protein encoded by this gene is a proteolipid that may be involved in the regulation of ion channels during brain development. The encoded protein may also play a role in forming and maintaining the structure of the nervous system. This gene is found within an intron of the BLCAP gene, but on the opposite strand. This gene is imprinted and is expressed only from the paternal allele. Two transcript variants encoding two different isoforms have been found for this gene.</td>
<td align="left">ENSG00000053438</td>
<td align="left">4826</td>
<td align="left">NA</td>
</tr>
<tr class="odd">
<td align="left">The modification of proteins with ubiquitin is an important cellular mechanism for targeting abnormal or short-lived proteins for degradation. Ubiquitination involves at least three classes of enzymes: ubiquitin-activating enzymes, ubiquitin-conjugating enzymes, and ubiquitin-protein ligases. This gene encodes a member of the E2 ubiquitin-conjugating enzyme family. The encoded protein is required for the destruction of mitotic cyclins and for cell cycle progression, and may be involved in cancer progression. Multiple transcript variants encoding different isoforms have been found for this gene. Pseudogenes of this gene have been defined on chromosomes 4, 14, 15, 18, and 19.</td>
<td align="left">ENSG00000175063</td>
<td align="left">11065</td>
<td align="left">NA</td>
</tr>
<tr class="even">
<td align="left">NA</td>
<td align="left">ENSG00000196700</td>
<td align="left">57473</td>
<td align="left">NA</td>
</tr>
<tr class="odd">
<td align="left">NA</td>
<td align="left">ENSG00000184674</td>
<td align="left">NA</td>
<td align="left">TRUE</td>
</tr>
<tr class="even">
<td align="left">This gene is a member of the cytidine deaminase gene family. It is one of a group of related genes found in a cluster, thought to result from gene duplication, on chromosome 22. Members of the cluster encode proteins that are structurally and functionally related to the C to U RNA-editing cytidine deaminase APOBEC1 and inhibit retroviruses, such as HIV, by deaminating cytosine residues in nascent retroviral cDNA.</td>
<td align="left">ENSG00000243811</td>
<td align="left">140564</td>
<td align="left">NA</td>
</tr>
<tr class="odd">
<td align="left">NA</td>
<td align="left">ENSG00000249222</td>
<td align="left">267020</td>
<td align="left">NA</td>
</tr>
<tr class="even">
<td align="left">This gene encodes a single-pass type I membrane protein that shares similarity with the interleukin-17 receptor (IL-17RA). Unlike IL-17RA, which is predominantly expressed in hemopoietic cells, and binds with high affinity to only IL-17A, this protein is expressed in nonhemopoietic tissues, and binds both IL-17A and IL-17F with similar affinities. The proinflammatory cytokines, IL-17A and IL-17F, have been implicated in the progression of inflammatory and autoimmune diseases. Multiple alternatively spliced transcript variants encoding different isoforms have been detected for this gene, and it has been proposed that soluble, secreted proteins lacking transmembrane and intracellular domains may function as extracellular antagonists to cytokine signaling.</td>
<td align="left">ENSG00000163702</td>
<td align="left">84818</td>
<td align="left">NA</td>
</tr>
<tr class="odd">
<td align="left">NA</td>
<td align="left">ENSG00000269028</td>
<td align="left">100462981</td>
<td align="left">NA</td>
</tr>
<tr class="even">
<td align="left">NA</td>
<td align="left">ENSG00000250254</td>
<td align="left">10744</td>
<td align="left">NA</td>
</tr>
<tr class="odd">
<td align="left">IRX2 is a member of the Iroquois homeobox gene family. Members of this family appear to play multiple roles during pattern formation of vertebrate embryos.</td>
<td align="left">ENSG00000170561</td>
<td align="left">153572</td>
<td align="left">NA</td>
</tr>
<tr class="even">
<td align="left">NA</td>
<td align="left">ENSG00000198865</td>
<td align="left">100129792</td>
<td align="left">NA</td>
</tr>
<tr class="odd">
<td align="left">NA</td>
<td align="left">ENSG00000271043</td>
<td align="left">ENSG00000271043</td>
<td align="left">NA</td>
</tr>
<tr class="even">
<td align="left">This gene encodes a member of the late embryogenesis abundant motif-containing protein family. The encoded protein is localized to mitochondria and may function as a cytoprotectant by regulating cell death and differentiation. Alternative splicing results in multiple transcript variants encoding different isoforms. Several related pseudogenes have been identified.</td>
<td align="left">ENSG00000169230</td>
<td align="left">27166</td>
<td align="left">NA</td>
</tr>
<tr class="odd">
<td align="left">This gene encodes a member of the receptor protein tyrosine kinase family of proteins that transduce extracellular signals across the cell membrane. The encoded protein lacks detectable catalytic tyrosine kinase activity, is involved in the Wnt signaling pathway and plays a role in multiple cellular processes including polarity and adhesion. Alternatively spliced transcript variants encoding multiple isoforms have been observed for this gene.</td>
<td align="left">ENSG00000112655</td>
<td align="left">5754</td>
<td align="left">NA</td>
</tr>
<tr class="even">
<td align="left">This gene encodes a member of the cAMP-dependent protein kinase inhibitor family. The encoded protein may play a role in the protein kinase A (PKA) pathway by interacting with the catalytic subunit of PKA, and overexpression of this gene may play a role in prostate cancer. Alternatively spliced transcript variants encoding multiple isoforms have been observed for this gene.</td>
<td align="left">ENSG00000135549</td>
<td align="left">5570</td>
<td align="left">NA</td>
</tr>
<tr class="odd">
<td align="left">The protein encoded by this gene is a brain fatty acid binding protein. Fatty acid binding proteins (FABPs) are a family of small, highly conserved, cytoplasmic proteins that bind long-chain fatty acids and other hydrophobic ligands. FABPs are thought to play roles in fatty acid uptake, transport, and metabolism.</td>
<td align="left">ENSG00000164434</td>
<td align="left">2173</td>
<td align="left">NA</td>
</tr>
<tr class="even">
<td align="left">The protein encoded by this gene is a mitogen that is secreted by vascular endothelial cells. The encoded protein plays a role in chondrocyte proliferation and differentiation, cell adhesion in many cell types, and is related to platelet-derived growth factor. Certain polymorphisms in this gene have been linked with a higher incidence of systemic sclerosis.</td>
<td align="left">ENSG00000118523</td>
<td align="left">1490</td>
<td align="left">NA</td>
</tr>
<tr class="odd">
<td align="left">NA</td>
<td align="left">ENSG00000106153</td>
<td align="left">51142</td>
<td align="left">NA</td>
</tr>
<tr class="even">
<td align="left">NA</td>
<td align="left">ENSG00000164897</td>
<td align="left">83590</td>
<td align="left">NA</td>
</tr>
<tr class="odd">
<td align="left">NA</td>
<td align="left">ENSG00000198840</td>
<td align="left">4537</td>
<td align="left">NA</td>
</tr>
<tr class="even">
<td align="left">NA</td>
<td align="left">ENSG00000198695</td>
<td align="left">4541</td>
<td align="left">NA</td>
</tr>
<tr class="odd">
<td align="left">NA</td>
<td align="left">ENSG00000204071</td>
<td align="left">158931</td>
<td align="left">NA</td>
</tr>
</tbody>
</table>
</div>
</div>
<div id="session-information" class="section level2">
<h2>Session information</h2>
<pre class="r"><code>sessionInfo()</code></pre>
<pre><code>R version 3.2.0 (2015-04-16)
Platform: x86_64-unknown-linux-gnu (64-bit)

locale:
 [1] LC_CTYPE=en_US.UTF-8       LC_NUMERIC=C              
 [3] LC_TIME=en_US.UTF-8        LC_COLLATE=en_US.UTF-8    
 [5] LC_MONETARY=en_US.UTF-8    LC_MESSAGES=en_US.UTF-8   
 [7] LC_PAPER=en_US.UTF-8       LC_NAME=C                 
 [9] LC_ADDRESS=C               LC_TELEPHONE=C            
[11] LC_MEASUREMENT=en_US.UTF-8 LC_IDENTIFICATION=C       

attached base packages:
 [1] stats4    parallel  grid      stats     graphics  grDevices utils    
 [8] datasets  methods   base     

other attached packages:
 [1] gplots_2.17.0          mygene_1.2.3           GenomicFeatures_1.20.1
 [4] AnnotationDbi_1.30.1   Biobase_2.28.0         GenomicRanges_1.20.5  
 [7] GenomeInfoDb_1.4.0     IRanges_2.2.4          S4Vectors_0.6.0       
[10] BiocGenerics_0.14.0    broman_0.59-5          zoo_1.7-12            
[13] ggplot2_1.0.1          edgeR_3.10.2           limma_3.24.9          
[16] dplyr_0.4.2            data.table_1.9.4       knitr_1.10.5          

loaded via a namespace (and not attached):
 [1] httr_0.6.1              jsonlite_0.9.16        
 [3] splines_3.2.0           gsubfn_0.6-6           
 [5] gtools_3.5.0            Formula_1.2-1          
 [7] assertthat_0.1          highr_0.5              
 [9] latticeExtra_0.6-26     Rsamtools_1.20.4       
[11] yaml_2.1.13             RSQLite_1.0.0          
[13] lattice_0.20-31         chron_2.3-45           
[15] digest_0.6.8            RColorBrewer_1.1-2     
[17] XVector_0.8.0           colorspace_1.2-6       
[19] htmltools_0.2.6         plyr_1.8.3             
[21] XML_3.98-1.2            biomaRt_2.24.0         
[23] zlibbioc_1.14.0         scales_0.2.4           
[25] gdata_2.16.1            BiocParallel_1.2.2     
[27] sqldf_0.4-10            nnet_7.3-9             
[29] proto_0.3-10            survival_2.38-1        
[31] magrittr_1.5            evaluate_0.7           
[33] MASS_7.3-40             foreign_0.8-63         
[35] tools_3.2.0             formatR_1.2            
[37] stringr_1.0.0           munsell_0.4.2          
[39] cluster_2.0.1           lambda.r_1.1.7         
[41] Biostrings_2.36.1       caTools_1.17.1         
[43] futile.logger_1.4.1     RCurl_1.95-4.6         
[45] bitops_1.0-6            rmarkdown_0.6.1        
[47] gtable_0.1.2            DBI_0.3.1              
[49] reshape2_1.4.1          R6_2.1.1               
[51] GenomicAlignments_1.4.1 gridExtra_0.9.1        
[53] rtracklayer_1.28.4      Hmisc_3.16-0           
[55] futile.options_1.0.0    KernSmooth_2.23-14     
[57] stringi_0.4-1           Rcpp_0.12.0            
[59] rpart_4.1-9             acepack_1.3-3.3        </code></pre>
</div>


<!-- some extra javascript for older browsers -->
<script type="text/javascript" src="libs/polyfill.js"></script>

<script>

// manage active state of menu based on current page
$(document).ready(function () {

    // active menu
    href = window.location.pathname
    href = href.substr(href.lastIndexOf('/') + 1)
    $('a[href="' + href + '"]').parent().addClass('active');

    // manage active menu header
    if (href.startsWith('authoring_'))
      $('a[href="' + 'authoring' + '"]').parent().addClass('active');
    else if (href.endsWith('_format.html'))
      $('a[href="' + 'formats' + '"]').parent().addClass('active');
    else if (href.startsWith('developer_'))
      $('a[href="' + 'developer' + '"]').parent().addClass('active');

});

</script>

</div>

<script>

// add bootstrap table styles to pandoc tables
$(document).ready(function () {
  $('tr.header').parent('thead').parent('table').addClass('table table-condensed');
});

</script>


</body>
</html>
