---
title: "Sequencing depth and cellular RNA content"
date: 2015-09-30
---

**Last updated:** `r Sys.Date()`

**Code version:** `r system("git log -1 --format='%H'", intern = TRUE)`

```{r chunk-options, include=FALSE}
source("chunk-options.R")
opts_chunk$set(fig.width = 8, fig.height = 8)
```

## Setup

```{r packages, message=FALSE}
library("dplyr")
library("edgeR")
library("ggplot2")
library("cowplot")
theme_set(theme_bw(base_size = 12))
source("functions.R")
```

Start with the annotation file.
It contains the bulk samples as well, so remove them.

```{r input-annotation}
anno <- read.table("../data/annotation.txt", header = TRUE,
                   stringsAsFactors = FALSE)
anno_single <- anno %>% filter(well != "bulk")
head(anno_single)
```

Import the read counts per gene.
Per cell calculate the total number of reads that map to endogenous protein-coding genes,
the total number of reads that map to ERCC spike-ins,
and the total number of reads that map to both.

```{r input-read-counts}
reads <- read.table("../data/reads.txt", header = TRUE,
                    stringsAsFactors = FALSE)
stopifnot(colnames(reads) == anno_single$sample_id)
ercc_index <- grepl("ERCC", rownames(reads))
anno_single$total_reads_gene = colSums(reads[!ercc_index, ])
anno_single$total_reads_ercc = colSums(reads[ercc_index, ])
anno_single$total_reads = colSums(reads)
```

Import the molecule counts per gene.
Per cell calculate the total number of molecules that map to endogenous protein-coding genes,
the total number of molecules that map to ERCC spike-ins,
and the total number of molecules that map to both.

```{r input-molecule-counts}
molecules <- read.table("../data/molecules.txt", header = TRUE,
                    stringsAsFactors = FALSE)
stopifnot(colnames(molecules) == anno_single$sample_id,
          rownames(molecules) == rownames(reads))
anno_single$total_molecules_gene = colSums(molecules[!ercc_index, ])
anno_single$total_molecules_ercc = colSums(molecules[ercc_index, ])
anno_single$total_molecules = colSums(molecules)
```

Import the summary counts from featureCounts.
These were gather from the featureCounts summary files with [gather-summary-counts.py](https://github.com/jdblischak/singleCellSeq/blob/master/code/gather-summary-counts.py).
The data come from the sickle-trimmed sequences [combined across all lanes for each single cell][combined].

[combined]: https://jdblischak.github.io/singleCellSeq/analysis/process-samples.html#process-at-the-sample-level

```{r input-summary-counts}
summary_counts <- read.table("../data/summary-counts.txt", header = TRUE,
                             stringsAsFactors = FALSE)
```

Clean up the data.

```{r clean-summary-counts}
summary_per_sample <- summary_counts %>%
  filter(sickle == "quality-trimmed", well != "bulk") %>%
  select(-sickle) %>%
  arrange(rmdup, individual, batch, well) %>%
  as.data.frame
stopifnot(paste0("NA", summary_per_sample$individual) == anno_single$individual,
          paste0("r", summary_per_sample$batch) == anno_single$replicate,
          summary_per_sample$well == anno_single$well,
          summary_per_sample$Assigned == c(anno_single$total_molecules,
                                           anno_single$total_reads))
```

Calculate the total number of mapped reads, unmapped reads, and sequencing depth (mapped + unmapped reads).

```{r}
colnames(summary_per_sample)
summary_per_sample <- summary_per_sample %>%
  mutate(total_mapped = Assigned + Unassigned_Ambiguity + Unassigned_NoFeatures,
         total_unmapped = Unassigned_Unmapped,
         depth = total_mapped + total_unmapped)
anno_single$total_mapped <- summary_per_sample[summary_per_sample$rmdup == "reads",
                                               "total_mapped"]
anno_single$total_unmapped <- summary_per_sample[summary_per_sample$rmdup == "reads",
                                                 "total_unmapped"]
anno_single$depth <- summary_per_sample[summary_per_sample$rmdup == "reads",
                                        "depth"]
```

Input single cell observational quality control data.

```{r input-qc}
qc <- read.table("../data/qc-ipsc.txt", header = TRUE,
                 stringsAsFactors = FALSE)
qc <- qc %>% arrange(individual, batch, well)
stopifnot(paste0("NA", qc$individual) == anno_single$individual,
          paste0("r", qc$batch) == anno_single$replicate,
          qc$well == anno_single$well)
head(qc)
```

Incorporate informatin on cell number, concentration, and TRA1-60 status.

```{r add-qc}
anno_single$cell_number <- qc$cell_number
anno_single$concentration <- qc$concentration
anno_single$tra1.60 <- qc$tra1.60
```

Calculate percentage of ERCC reads (out of all reads assigned to a feature), ERCC molecules, and unmapped reads.

```{r percentages}
anno_single <- anno_single %>%
  mutate(perc_ercc_reads = total_reads_ercc / total_reads * 100,
         perc_ercc_molecules = total_molecules_ercc / total_molecules * 100,
         perc_unmapped = total_unmapped / depth * 100)
anno_single$num_genes = apply(reads[!ercc_index, ], 2, function(x) sum(x > 0))
```

## Filtering

As we try to understand the general relationships between sequencing results and cellular mRNA content, we remove outlier batches.
The quantification of the concentration of the single cells in replicate 1 of NA19098 failed.
The number of ERCC molecules in replicate 2 of NA19098 are abnormally high.

```{r filter-outliers}
anno_single <- anno_single %>% filter(batch != "NA19098.r1",
                                      batch != "NA19098.r2")
```

## Is concentration a good proxy for cellular mRNA content?

```{r cell-number-concentration}
ggplot(anno_single,
       aes(x = as.factor(cell_number), y = concentration)) +
  geom_boxplot()
```

## Does concentration affect sequencing depth?

```{r concentration-depth}
ggplot(anno_single,
       aes(x = concentration, y = depth)) +
  geom_point()
```

```{r concentration-depth-one-cell}
ggplot(anno_single[anno_single$cell_number == 1, ],
       aes(x = concentration, y = depth)) +
  geom_point()
```

## Is percentage ERCC a good proxy for RNA content?

```{r concentration-perc-ercc-reads}
ggplot(anno_single,
       aes(x = concentration, y = perc_ercc_reads)) +
  geom_point()
```

```{r cell-number-perc-ercc-reads}
ggplot(anno_single,
       aes(x = as.factor(cell_number), y = perc_ercc_reads)) +
  geom_boxplot()
```

```{r cell-number-perc-ercc-reads-1-3-cells}
ggplot(anno_single[anno_single$cell_number %in% 1:3, ],
       aes(x = as.factor(cell_number), y = perc_ercc_reads)) +
  geom_boxplot()
```

```{r cell-number-perc-ercc-barplot}
perc_ercc_bar <- anno_single %>% 
  filter(anno_single$batch != "NA19098.r1",
         anno_single$batch != "NA19098.r2") %>%
  group_by(cell_number) %>%
  summarize(mean_perc_ercc = mean(perc_ercc_reads),
            sem_perc_ercc = sd(perc_ercc_reads) / sqrt(n())) %>%
  filter(cell_number %in% 1:3)
ggplot(perc_ercc_bar,
       aes(x = as.factor(cell_number), y = mean_perc_ercc)) +
  geom_bar(stat = "identity") +
  geom_errorbar(aes(ymin = mean_perc_ercc - sem_perc_ercc,
                    ymax = mean_perc_ercc + sem_perc_ercc))
```

## Session information

```{r info}
sessionInfo()
```
