<!DOCTYPE html>

<html xmlns="http://www.w3.org/1999/xhtml">

<head>

<meta charset="utf-8">
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="generator" content="pandoc" />


<meta name="date" content="2015-09-30" />

<title>Sequencing depth and cellular RNA content</title>

<script src="libs/jquery-1.11.0/jquery.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="libs/bootstrap-3.3.1/css/united.min.css" rel="stylesheet" />
<script src="libs/bootstrap-3.3.1/js/bootstrap.min.js"></script>
<script src="libs/bootstrap-3.3.1/shim/html5shiv.min.js"></script>
<script src="libs/bootstrap-3.3.1/shim/respond.min.js"></script>

<style type="text/css">

/* padding for bootstrap navbar */
body {
  padding-top: 50px;
  padding-bottom: 40px;
}


/* offset scroll position for anchor links (for fixed navbar)  */
.section h2 {
  padding-top: 55px;
  margin-top: -55px;
}
.section h3 {
  padding-top: 55px;
  margin-top: -55px;
}



/* don't use link color in navbar */
.dropdown-menu>li>a {
  color: black;
}

/* some padding for disqus */
#disqus_thread {
  margin-top: 45px;
}

</style>

<link rel="stylesheet" href="libs/font-awesome-4.1.0/css/font-awesome.min.css"/>

<style type="text/css">code{white-space: pre;}</style>
<link rel="stylesheet"
      href="libs/highlight/textmate.css"
      type="text/css" />
<script src="libs/highlight/highlight.js"></script>
<style type="text/css">
  pre:not([class]) {
    background-color: white;
  }
</style>
<script type="text/javascript">
if (window.hljs && document.readyState && document.readyState === "complete") {
   window.setTimeout(function() {
      hljs.initHighlighting();
   }, 0);
}
</script>



</head>

<body>

<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
code {
  color: inherit;
  background-color: rgba(0, 0, 0, 0.04);
}
img { 
  max-width:100%; 
  height: auto; 
}
</style>
<div class="container-fluid main-container">


<div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">singleCellSeq</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li><a href="index.html">Home</a></li>
        <li><a href="about.html">About</a></li>
        <li><a href="license.html">License</a></li>
        <li><a href="https://github.com/jdblischak/singleCellSeq">GitHub</a></li>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<div id="header">
<h1 class="title">Sequencing depth and cellular RNA content</h1>
<h4 class="date"><em>2015-09-30</em></h4>
</div>

<div id="TOC">
<ul>
<li><a href="#setup">Setup</a></li>
<li><a href="#filtering">Filtering</a></li>
<li><a href="#is-concentration-a-good-proxy-for-cellular-mrna-content">Is concentration a good proxy for cellular mRNA content?</a></li>
<li><a href="#does-concentration-affect-sequencing-depth">Does concentration affect sequencing depth?</a></li>
<li><a href="#is-percentage-ercc-a-good-proxy-for-rna-content">Is percentage ERCC a good proxy for RNA content?</a></li>
<li><a href="#session-information">Session information</a></li>
</ul>
</div>

<p><strong>Last updated:</strong> 2015-10-11</p>
<p><strong>Code version:</strong> 172e9a3d594b1c72037445ad9dae00c7d3c81297</p>
<div id="setup" class="section level2">
<h2>Setup</h2>
<pre class="r"><code>library(&quot;dplyr&quot;)
library(&quot;edgeR&quot;)
library(&quot;ggplot2&quot;)
library(&quot;cowplot&quot;)
theme_set(theme_bw(base_size = 12))
source(&quot;functions.R&quot;)</code></pre>
<p>Start with the annotation file. It contains the bulk samples as well, so remove them.</p>
<pre class="r"><code>anno &lt;- read.table(&quot;../data/annotation.txt&quot;, header = TRUE,
                   stringsAsFactors = FALSE)
anno_single &lt;- anno %&gt;% filter(well != &quot;bulk&quot;)
head(anno_single)</code></pre>
<pre><code>  individual replicate well      batch      sample_id
1    NA19098        r1  A01 NA19098.r1 NA19098.r1.A01
2    NA19098        r1  A02 NA19098.r1 NA19098.r1.A02
3    NA19098        r1  A03 NA19098.r1 NA19098.r1.A03
4    NA19098        r1  A04 NA19098.r1 NA19098.r1.A04
5    NA19098        r1  A05 NA19098.r1 NA19098.r1.A05
6    NA19098        r1  A06 NA19098.r1 NA19098.r1.A06</code></pre>
<p>Import the read counts per gene. Per cell calculate the total number of reads that map to endogenous protein-coding genes, the total number of reads that map to ERCC spike-ins, and the total number of reads that map to both.</p>
<pre class="r"><code>reads &lt;- read.table(&quot;../data/reads.txt&quot;, header = TRUE,
                    stringsAsFactors = FALSE)
stopifnot(colnames(reads) == anno_single$sample_id)
ercc_index &lt;- grepl(&quot;ERCC&quot;, rownames(reads))
anno_single$total_reads_gene = colSums(reads[!ercc_index, ])
anno_single$total_reads_ercc = colSums(reads[ercc_index, ])
anno_single$total_reads = colSums(reads)</code></pre>
<p>Import the molecule counts per gene. Per cell calculate the total number of molecules that map to endogenous protein-coding genes, the total number of molecules that map to ERCC spike-ins, and the total number of molecules that map to both.</p>
<pre class="r"><code>molecules &lt;- read.table(&quot;../data/molecules.txt&quot;, header = TRUE,
                    stringsAsFactors = FALSE)
stopifnot(colnames(molecules) == anno_single$sample_id,
          rownames(molecules) == rownames(reads))
anno_single$total_molecules_gene = colSums(molecules[!ercc_index, ])
anno_single$total_molecules_ercc = colSums(molecules[ercc_index, ])
anno_single$total_molecules = colSums(molecules)</code></pre>
<p>Import the summary counts from featureCounts. These were gather from the featureCounts summary files with <a href="https://github.com/jdblischak/singleCellSeq/blob/master/code/gather-summary-counts.py">gather-summary-counts.py</a>. The data come from the sickle-trimmed sequences <a href="https://jdblischak.github.io/singleCellSeq/analysis/process-samples.html#process-at-the-sample-level">combined across all lanes for each single cell</a>.</p>
<pre class="r"><code>summary_counts &lt;- read.table(&quot;../data/summary-counts.txt&quot;, header = TRUE,
                             stringsAsFactors = FALSE)</code></pre>
<p>Clean up the data.</p>
<pre class="r"><code>summary_per_sample &lt;- summary_counts %&gt;%
  filter(sickle == &quot;quality-trimmed&quot;, well != &quot;bulk&quot;) %&gt;%
  select(-sickle) %&gt;%
  arrange(rmdup, individual, batch, well) %&gt;%
  as.data.frame
stopifnot(paste0(&quot;NA&quot;, summary_per_sample$individual) == anno_single$individual,
          paste0(&quot;r&quot;, summary_per_sample$batch) == anno_single$replicate,
          summary_per_sample$well == anno_single$well,
          summary_per_sample$Assigned == c(anno_single$total_molecules,
                                           anno_single$total_reads))</code></pre>
<p>Calculate the total number of mapped reads, unmapped reads, and sequencing depth (mapped + unmapped reads).</p>
<pre class="r"><code>colnames(summary_per_sample)</code></pre>
<pre><code> [1] &quot;individual&quot;                 &quot;batch&quot;                     
 [3] &quot;well&quot;                       &quot;rmdup&quot;                     
 [5] &quot;Assigned&quot;                   &quot;Unassigned_Ambiguity&quot;      
 [7] &quot;Unassigned_MultiMapping&quot;    &quot;Unassigned_NoFeatures&quot;     
 [9] &quot;Unassigned_Unmapped&quot;        &quot;Unassigned_MappingQuality&quot; 
[11] &quot;Unassigned_FragementLength&quot; &quot;Unassigned_Chimera&quot;        
[13] &quot;Unassigned_Secondary&quot;      </code></pre>
<pre class="r"><code>summary_per_sample &lt;- summary_per_sample %&gt;%
  mutate(total_mapped = Assigned + Unassigned_Ambiguity + Unassigned_NoFeatures,
         total_unmapped = Unassigned_Unmapped,
         depth = total_mapped + total_unmapped)
anno_single$total_mapped &lt;- summary_per_sample[summary_per_sample$rmdup == &quot;reads&quot;,
                                               &quot;total_mapped&quot;]
anno_single$total_unmapped &lt;- summary_per_sample[summary_per_sample$rmdup == &quot;reads&quot;,
                                                 &quot;total_unmapped&quot;]
anno_single$depth &lt;- summary_per_sample[summary_per_sample$rmdup == &quot;reads&quot;,
                                        &quot;depth&quot;]</code></pre>
<p>Input single cell observational quality control data.</p>
<pre class="r"><code>qc &lt;- read.table(&quot;../data/qc-ipsc.txt&quot;, header = TRUE,
                 stringsAsFactors = FALSE)
qc &lt;- qc %&gt;% arrange(individual, batch, well)
stopifnot(paste0(&quot;NA&quot;, qc$individual) == anno_single$individual,
          paste0(&quot;r&quot;, qc$batch) == anno_single$replicate,
          qc$well == anno_single$well)
head(qc)</code></pre>
<pre><code>  individual batch well cell_number concentration tra1.60
1      19098     1  A01           1      1.734785       1
2      19098     1  A02           1      1.723038       1
3      19098     1  A03           1      1.512786       1
4      19098     1  A04           1      1.347492       1
5      19098     1  A05           1      2.313047       1
6      19098     1  A06           1      2.056803       1</code></pre>
<p>Incorporate informatin on cell number, concentration, and TRA1-60 status.</p>
<pre class="r"><code>anno_single$cell_number &lt;- qc$cell_number
anno_single$concentration &lt;- qc$concentration
anno_single$tra1.60 &lt;- qc$tra1.60</code></pre>
<p>Calculate percentage of ERCC reads (out of all reads assigned to a feature), ERCC molecules, and unmapped reads.</p>
<pre class="r"><code>anno_single &lt;- anno_single %&gt;%
  mutate(perc_ercc_reads = total_reads_ercc / total_reads * 100,
         perc_ercc_molecules = total_molecules_ercc / total_molecules * 100,
         perc_unmapped = total_unmapped / depth * 100)
anno_single$num_genes = apply(reads[!ercc_index, ], 2, function(x) sum(x &gt; 0))</code></pre>
</div>
<div id="filtering" class="section level2">
<h2>Filtering</h2>
<p>As we try to understand the general relationships between sequencing results and cellular mRNA content, we remove outlier batches. The quantification of the concentration of the single cells in replicate 1 of NA19098 failed. The number of ERCC molecules in replicate 2 of NA19098 are abnormally high.</p>
<pre class="r"><code>anno_single &lt;- anno_single %&gt;% filter(batch != &quot;NA19098.r1&quot;,
                                      batch != &quot;NA19098.r2&quot;)</code></pre>
</div>
<div id="is-concentration-a-good-proxy-for-cellular-mrna-content" class="section level2">
<h2>Is concentration a good proxy for cellular mRNA content?</h2>
<pre class="r"><code>ggplot(anno_single,
       aes(x = as.factor(cell_number), y = concentration)) +
  geom_boxplot()</code></pre>
<p><img src="figure/depth.Rmd/cell-number-concentration-1.png" title="" alt="" width="768" style="display: block; margin: auto;" /></p>
</div>
<div id="does-concentration-affect-sequencing-depth" class="section level2">
<h2>Does concentration affect sequencing depth?</h2>
<pre class="r"><code>ggplot(anno_single,
       aes(x = concentration, y = depth)) +
  geom_point()</code></pre>
<p><img src="figure/depth.Rmd/concentration-depth-1.png" title="" alt="" width="768" style="display: block; margin: auto;" /></p>
<pre class="r"><code>ggplot(anno_single[anno_single$cell_number == 1, ],
       aes(x = concentration, y = depth)) +
  geom_point()</code></pre>
<p><img src="figure/depth.Rmd/concentration-depth-one-cell-1.png" title="" alt="" width="768" style="display: block; margin: auto;" /></p>
</div>
<div id="is-percentage-ercc-a-good-proxy-for-rna-content" class="section level2">
<h2>Is percentage ERCC a good proxy for RNA content?</h2>
<pre class="r"><code>ggplot(anno_single,
       aes(x = concentration, y = perc_ercc_reads)) +
  geom_point()</code></pre>
<p><img src="figure/depth.Rmd/concentration-perc-ercc-reads-1.png" title="" alt="" width="768" style="display: block; margin: auto;" /></p>
<pre class="r"><code>ggplot(anno_single,
       aes(x = as.factor(cell_number), y = perc_ercc_reads)) +
  geom_boxplot()</code></pre>
<p><img src="figure/depth.Rmd/cell-number-perc-ercc-reads-1.png" title="" alt="" width="768" style="display: block; margin: auto;" /></p>
<pre class="r"><code>ggplot(anno_single[anno_single$cell_number %in% 1:3, ],
       aes(x = as.factor(cell_number), y = perc_ercc_reads)) +
  geom_boxplot()</code></pre>
<p><img src="figure/depth.Rmd/cell-number-perc-ercc-reads-1-3-cells-1.png" title="" alt="" width="768" style="display: block; margin: auto;" /></p>
<pre class="r"><code>perc_ercc_bar &lt;- anno_single %&gt;% 
  filter(anno_single$batch != &quot;NA19098.r1&quot;,
         anno_single$batch != &quot;NA19098.r2&quot;) %&gt;%
  group_by(cell_number) %&gt;%
  summarize(mean_perc_ercc = mean(perc_ercc_reads),
            sem_perc_ercc = sd(perc_ercc_reads) / sqrt(n())) %&gt;%
  filter(cell_number %in% 1:3)
ggplot(perc_ercc_bar,
       aes(x = as.factor(cell_number), y = mean_perc_ercc)) +
  geom_bar(stat = &quot;identity&quot;) +
  geom_errorbar(aes(ymin = mean_perc_ercc - sem_perc_ercc,
                    ymax = mean_perc_ercc + sem_perc_ercc))</code></pre>
<p><img src="figure/depth.Rmd/cell-number-perc-ercc-barplot-1.png" title="" alt="" width="768" style="display: block; margin: auto;" /></p>
</div>
<div id="session-information" class="section level2">
<h2>Session information</h2>
<pre class="r"><code>sessionInfo()</code></pre>
<pre><code>R version 3.2.0 (2015-04-16)
Platform: x86_64-unknown-linux-gnu (64-bit)

locale:
 [1] LC_CTYPE=en_US.UTF-8       LC_NUMERIC=C              
 [3] LC_TIME=en_US.UTF-8        LC_COLLATE=en_US.UTF-8    
 [5] LC_MONETARY=en_US.UTF-8    LC_MESSAGES=en_US.UTF-8   
 [7] LC_PAPER=en_US.UTF-8       LC_NAME=C                 
 [9] LC_ADDRESS=C               LC_TELEPHONE=C            
[11] LC_MEASUREMENT=en_US.UTF-8 LC_IDENTIFICATION=C       

attached base packages:
[1] stats     graphics  grDevices utils     datasets  methods   base     

other attached packages:
[1] cowplot_0.3.1 ggplot2_1.0.1 edgeR_3.10.2  limma_3.24.9  dplyr_0.4.2  
[6] knitr_1.10.5 

loaded via a namespace (and not attached):
 [1] Rcpp_0.12.0      magrittr_1.5     MASS_7.3-40      munsell_0.4.2   
 [5] colorspace_1.2-6 R6_2.1.1         stringr_1.0.0    httr_0.6.1      
 [9] plyr_1.8.3       tools_3.2.0      parallel_3.2.0   grid_3.2.0      
[13] gtable_0.1.2     DBI_0.3.1        htmltools_0.2.6  lazyeval_0.1.10 
[17] yaml_2.1.13      assertthat_0.1   digest_0.6.8     reshape2_1.4.1  
[21] formatR_1.2      bitops_1.0-6     RCurl_1.95-4.6   evaluate_0.7    
[25] rmarkdown_0.6.1  labeling_0.3     stringi_0.4-1    scales_0.2.4    
[29] proto_0.3-10    </code></pre>
</div>


<!-- some extra javascript for older browsers -->
<script type="text/javascript" src="libs/polyfill.js"></script>

<script>

// manage active state of menu based on current page
$(document).ready(function () {

    // active menu
    href = window.location.pathname
    href = href.substr(href.lastIndexOf('/') + 1)
    $('a[href="' + href + '"]').parent().addClass('active');

    // manage active menu header
    if (href.startsWith('authoring_'))
      $('a[href="' + 'authoring' + '"]').parent().addClass('active');
    else if (href.endsWith('_format.html'))
      $('a[href="' + 'formats' + '"]').parent().addClass('active');
    else if (href.startsWith('developer_'))
      $('a[href="' + 'developer' + '"]').parent().addClass('active');

});

</script>

</div>

<script>

// add bootstrap table styles to pandoc tables
$(document).ready(function () {
  $('tr.header').parent('thead').parent('table').addClass('table table-condensed');
});

</script>


</body>
</html>
