<!DOCTYPE html>

<html xmlns="http://www.w3.org/1999/xhtml">

<head>

<meta charset="utf-8">
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="generator" content="pandoc" />

<meta name="author" content="John Blischak" />

<meta name="date" content="2015-03-30" />

<title>Download flow cell 01</title>

<script src="libs/jquery-1.11.0/jquery.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="libs/bootstrap-3.3.1/css/united.min.css" rel="stylesheet" />
<script src="libs/bootstrap-3.3.1/js/bootstrap.min.js"></script>
<script src="libs/bootstrap-3.3.1/shim/html5shiv.min.js"></script>
<script src="libs/bootstrap-3.3.1/shim/respond.min.js"></script>

<style type="text/css">

/* padding for bootstrap navbar */
body {
  padding-top: 50px;
  padding-bottom: 40px;
}


/* offset scroll position for anchor links (for fixed navbar)  */
.section h2 {
  padding-top: 55px;
  margin-top: -55px;
}
.section h3 {
  padding-top: 55px;
  margin-top: -55px;
}



/* don't use link color in navbar */
.dropdown-menu>li>a {
  color: black;
}

/* some padding for disqus */
#disqus_thread {
  margin-top: 45px;
}

</style>

<link rel="stylesheet" href="libs/font-awesome-4.1.0/css/font-awesome.min.css"/>

<style type="text/css">code{white-space: pre;}</style>
<link rel="stylesheet"
      href="libs/highlight/textmate.css"
      type="text/css" />
<script src="libs/highlight/highlight.js"></script>
<style type="text/css">
  pre:not([class]) {
    background-color: white;
  }
</style>
<script type="text/javascript">
if (window.hljs && document.readyState && document.readyState === "complete") {
   window.setTimeout(function() {
      hljs.initHighlighting();
   }, 0);
}
</script>



</head>

<body>

<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
code {
  color: inherit;
  background-color: rgba(0, 0, 0, 0.04);
}
img { 
  max-width:100%; 
  height: auto; 
}
</style>
<div class="container-fluid main-container">


<div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">singleCellSeq</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li><a href="index.html">Home</a></li>
        <li><a href="about.html">About</a></li>
        <li><a href="license.html">License</a></li>
        <li><a href="https://github.com/jdblischak/singleCellSeq">GitHub</a></li>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<div id="header">
<h1 class="title">Download flow cell 01</h1>
<h4 class="author"><em>John Blischak</em></h4>
<h4 class="date"><em>2015-03-30</em></h4>
</div>


<p><strong>Last updated:</strong> 2015-04-11</p>
<p><strong>Code version:</strong> 3275cbc6932e4def9f737a536299aa2022c5c7a1</p>
<p>The first of four flow cells was sequenced at the Functional Genomics Facility (FGF). Here is the message from Pieter Faber:</p>
<blockquote>
<p>We finished one Flowcell of your Illumina sequencing experiment (4 flowcells SR100). No technical problem were encountered. I attached several QC files in support. Declining sequencing quality at higher bp-length is (in part) caused by presence of adaptor only sequences (up to 40% at bp 100 according to FastQC).</p>
<p>I have uploaded the data in fastq format to the Genomics Core ftp server data server (gilad Lab folder, folder = /NGS/150320_700819F_0303_AC6WYKACXX-YG-SR100-FC-1). For access a username and password have been provided to the PI previously (not your CNET ID!!).</p>
<p>Let me know if other files and / or explanations are needed.</p>
<p>IMPORTANT: While the Core has looked at your data and found no technical irregularities we urge you to perform data analysis/QC in a timely manner and contact us with any questions/comments you may have within 3 months after the data were made available to you!!</p>
</blockquote>
<p>Nick has recently downloaded data from the FGF FTP site. In order to download the data from the command-line, you have to modify the path that Pieter sends. There does not appear to be a “Gilad lab folder”. I checked with Filezilla. Instead, you need to preface <code>/NGS</code> with <code>/Genomics_Data</code>.</p>
<p>I am storing the data in <code>internal_supp</code> so that it is easier for PoYuan and me to collaborate.</p>
<pre class="bash"><code>cd /mnt/gluster/data/internal_supp/singleCellSeq/
mkdir raw
cd raw
nohup wget --user=gilad --password=&#39;&lt;password&gt;&#39; -r ftp://fgfftp.uchicago.edu/Genomics_Data/NGS/150320_700819F_0303_AC6WYKACXX-YG-SR100-FC-1/ &amp;
mv nohup.out 150320_700819F_0303_AC6WYKACXX-YG-SR100-FC-1.log</code></pre>
<p>This recrursively downloads all the files. Unfortunately it also includes all the directories above the final listed directory. This is really annoying because the data is so nested. I tried running <code>wget</code> with the option <code>-np</code> (“no parent”), but this did not make a difference. The download took ~10 hours. It started at 12:05:18 and ended at 21:54:33. However, <code>wget</code> reported that it only took ~5 hours, so I’m not sure how it is calculated. Here’s the final line of output:</p>
<blockquote>
<p>Downloaded: 1375 files, 129G in 4h 59m 1s (7.37 MB/s)</p>
</blockquote>
<p>To remove the unnecessary directories from the FGF FTP site, I moved the files.</p>
<pre class="bash"><code>mv fgfftp.uchicago.edu/Genomics_Data/NGS/150320_700819F_0303_AC6WYKACXX-YG-SR100-FC-1/ 150320_700819F_0303_AC6WYKACXX-YG-SR100-FC-1/
rmdir -p fgfftp.uchicago.edu/Genomics_Data/NGS/</code></pre>
<p>Next I removed the extraneous CASAVA directories and added the flow cell name to the filename.</p>
<pre><code>cd -
mkdir fastq</code></pre>
<p>I did this with the following Python code:</p>
<pre class="python"><code>import glob
import shutil

files = glob.glob(&#39;raw/150320_700819F_0303_AC6WYKACXX-YG-SR100-FC-1/FastQ/Project_YG-SR100-1/Sample*/*fastq.gz&#39;)

target_dir = &#39;fastq/&#39;
log = open(&#39;rearrange_C6WYKACXX.log&#39;, &#39;w&#39;)
log.write(&#39;original\tnew\n&#39;)

for f in files:
    path = f.strip(&#39;fastq.gz&#39;).split(&#39;/&#39;)
    flow_cell = path[1].split(&#39;_&#39;)[-1][1:10]
    file_parts = path[-1].split(&#39;_&#39;)[:-1]
    new_name = target_dir + &#39;.&#39;.join(file_parts + [flow_cell]) + &#39;.fastq.gz&#39;
    log.write(f + &#39;\t&#39; + new_name + &#39;\n&#39;)
    shutil.move(f, new_name)

log.close()</code></pre>


<!-- some extra javascript for older browsers -->
<script type="text/javascript" src="libs/polyfill.js"></script>

<script>

// manage active state of menu based on current page
$(document).ready(function () {

    // active menu
    href = window.location.pathname
    href = href.substr(href.lastIndexOf('/') + 1)
    $('a[href="' + href + '"]').parent().addClass('active');

    // manage active menu header
    if (href.startsWith('authoring_'))
      $('a[href="' + 'authoring' + '"]').parent().addClass('active');
    else if (href.endsWith('_format.html'))
      $('a[href="' + 'formats' + '"]').parent().addClass('active');
    else if (href.startsWith('developer_'))
      $('a[href="' + 'developer' + '"]').parent().addClass('active');

});

</script>

</div>

<script>

// add bootstrap table styles to pandoc tables
$(document).ready(function () {
  $('tr.header').parent('thead').parent('table').addClass('table table-condensed');
});

</script>


</body>
</html>
