---
title: "Draft draft draft"
author: "Po-Yuan Tung"
date: 2015-09-21
---

**Last updated:** `r Sys.Date()`

**Code version:** `r system("git log -1 --format='%H'", intern = TRUE)`

```{r chunk-options, include=FALSE}
source("chunk-options.R")
```

A draft for a fabulous paper.

## Introduction

* challenges of single cell RNA-seq
* molecule and read:UMI
* normalization and techinical noise:ERCC
* sample size: number of reads, number of cells
* novelty of this study: multiple c1 bathes per individual and multiple individuals

## M and M
1. cell culture (charaterization?)
2. single cell collection, library prep, sequencing (sup fig. 4)
3. mapping, featurecounts, umi tool
4. linear shift and input counts of each ERCC gene 
5. mix effect model

## Figures
1. study design and QC
2. standardization: cmp ERCC genes and endogenous genes separately
3. subsampling: a. gene expression b. number of genes c. variance
4. normalization: pca plots showing before and after linear shift and mix effect model
5. biological findings: differential CV genes

## Sub Figures
1. data collection: sequencing 
2. reads to molecule: individual effect and batch effect
3. cell cycle: more reads, more molecule, although G2 have more 
4. 4 cells with high sequencing depth
5. ERCC not good: RUV and BASiCs
6. noisy genes: based on non standardized counts


## Results 
1. study design and data collection (Fig.1, sup fig.1 and 2)
2. cpm (Fig.2, sup fig.3 and 4)
3. subsample (fig.3)
4. remove batch effect (fig.4 and sup fig.5)
5. gene expression noise (fig.5 and sup fig.6)


## Session information

```{r info}
sessionInfo()
```
