<!DOCTYPE html>

<html xmlns="http://www.w3.org/1999/xhtml">

<head>

<meta charset="utf-8">
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="generator" content="pandoc" />

<meta name="author" content="PoYuan Tung" />

<meta name="date" content="2015-09-07" />

<title>ERCC cpm?</title>

<script src="libs/jquery-1.11.0/jquery.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="libs/bootstrap-3.3.1/css/united.min.css" rel="stylesheet" />
<script src="libs/bootstrap-3.3.1/js/bootstrap.min.js"></script>
<script src="libs/bootstrap-3.3.1/shim/html5shiv.min.js"></script>
<script src="libs/bootstrap-3.3.1/shim/respond.min.js"></script>

<style type="text/css">

/* padding for bootstrap navbar */
body {
  padding-top: 50px;
  padding-bottom: 40px;
}


/* offset scroll position for anchor links (for fixed navbar)  */
.section h2 {
  padding-top: 55px;
  margin-top: -55px;
}
.section h3 {
  padding-top: 55px;
  margin-top: -55px;
}



/* don't use link color in navbar */
.dropdown-menu>li>a {
  color: black;
}

/* some padding for disqus */
#disqus_thread {
  margin-top: 45px;
}

</style>

<link rel="stylesheet" href="libs/font-awesome-4.1.0/css/font-awesome.min.css"/>

<style type="text/css">code{white-space: pre;}</style>
<link rel="stylesheet"
      href="libs/highlight/textmate.css"
      type="text/css" />
<script src="libs/highlight/highlight.js"></script>
<style type="text/css">
  pre:not([class]) {
    background-color: white;
  }
</style>
<script type="text/javascript">
if (window.hljs && document.readyState && document.readyState === "complete") {
   window.setTimeout(function() {
      hljs.initHighlighting();
   }, 0);
}
</script>



</head>

<body>

<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
code {
  color: inherit;
  background-color: rgba(0, 0, 0, 0.04);
}
img { 
  max-width:100%; 
  height: auto; 
}
</style>
<div class="container-fluid main-container">


<div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">singleCellSeq</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li><a href="index.html">Home</a></li>
        <li><a href="about.html">About</a></li>
        <li><a href="license.html">License</a></li>
        <li><a href="https://github.com/jdblischak/singleCellSeq">GitHub</a></li>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<div id="header">
<h1 class="title">ERCC cpm?</h1>
<h4 class="author"><em>PoYuan Tung</em></h4>
<h4 class="date"><em>2015-09-07</em></h4>
</div>

<div id="TOC">
<ul>
<li><a href="#input">Input</a></li>
<li><a href="#prepare-single-cell-molecule-data">Prepare single cell molecule data</a></li>
<li><a href="#total-ercc-vs-total-molecule-and-total-reads-counts">total ERCC vs total molecule and total reads (counts)</a></li>
<li><a href="#cpm-transformation-based-on-total-molecule-counts-wrong">cpm transformation based on total molecule counts (WRONG!)</a></li>
<li><a href="#cpm-transformation-based-on-ercc-correct">cpm transformation based on ERCC (CORRECT!)</a></li>
<li><a href="#linear-shift-normalization">Linear Shift Normalization</a></li>
<li><a href="#session-information">Session information</a></li>
</ul>
</div>

<p><strong>Last updated:</strong> 2015-09-11</p>
<p><strong>Code version:</strong> 37f57b1c3b81f30e213726dfcc3da532197ac224</p>
<p>We standardize the molecule counts to account for differences in sequencing depth. This is necessary because the <a href="compare-reads-v-molecules.html#effect-of-sequencing-depth-on-molecule-count">sequencing depth affects the total molecule counts</a>.</p>
<p>However, according to our study design. Each cell within the same C1 patch, independent of total number of transcripts, should have equal amount of ERCC molecule. As a result, cpm (no log transforamtion) will introduce bias to ERCC, where cells with more molecules will end up with fewer ERCC molecules. Therefore, cpm should not be performed on ERCC genes, but only on endogenous genes.</p>
<div id="input" class="section level2">
<h2>Input</h2>
<p>Input annotation.</p>
<pre class="r"><code>anno &lt;- read.table(&quot;../data/annotation.txt&quot;, header = TRUE,
                   stringsAsFactors = FALSE)
head(anno)</code></pre>
<pre><code>  individual batch well     sample_id
1      19098     1  A01 NA19098.1.A01
2      19098     1  A02 NA19098.1.A02
3      19098     1  A03 NA19098.1.A03
4      19098     1  A04 NA19098.1.A04
5      19098     1  A05 NA19098.1.A05
6      19098     1  A06 NA19098.1.A06</code></pre>
<p>Input molecule counts.</p>
<pre class="r"><code>molecules &lt;- read.table(&quot;../data/molecules.txt&quot;, header = TRUE,
                    stringsAsFactors = FALSE)</code></pre>
<p>Input read counts.</p>
<pre class="r"><code>reads &lt;- read.table(&quot;../data/reads.txt&quot;, header = TRUE,
                    stringsAsFactors = FALSE)</code></pre>
<p>Input list of quality single cells.</p>
<pre class="r"><code>quality_single_cells &lt;- scan(&quot;../data/quality-single-cells.txt&quot;,
                             what = &quot;character&quot;)</code></pre>
<p>Input ERCC concentration</p>
<pre class="r"><code>ercc &lt;- read.table(&quot;../data/ercc-info.txt&quot;, header = TRUE, sep = &quot;\t&quot;,
                   stringsAsFactors = FALSE)
colnames(ercc) &lt;- c(&quot;num&quot;, &quot;id&quot;, &quot;subgroup&quot;, &quot;conc_mix1&quot;, &quot;conc_mix2&quot;,
                    &quot;expected_fc&quot;, &quot;log2_mix1_mix2&quot;)
head(ercc)</code></pre>
<pre><code>  num         id subgroup conc_mix1  conc_mix2 expected_fc log2_mix1_mix2
1   1 ERCC-00130        A 30000.000 7500.00000           4              2
2   2 ERCC-00004        A  7500.000 1875.00000           4              2
3   3 ERCC-00136        A  1875.000  468.75000           4              2
4   4 ERCC-00108        A   937.500  234.37500           4              2
5   5 ERCC-00116        A   468.750  117.18750           4              2
6   6 ERCC-00092        A   234.375   58.59375           4              2</code></pre>
<pre class="r"><code>stopifnot(nrow(ercc) == 92)

ercc &lt;- ercc[order(ercc$id), ]</code></pre>
</div>
<div id="prepare-single-cell-molecule-data" class="section level2">
<h2>Prepare single cell molecule data</h2>
<p>Keep only the single cells that passed the <a href="qc-cell-ipsc.html">QC filters</a> and the bulk samples.</p>
<pre class="r"><code>molecules &lt;- molecules[, grepl(&quot;bulk&quot;, colnames(molecules)) |
                         colnames(molecules) %in% quality_single_cells]
anno &lt;- anno[anno$well == &quot;bulk&quot; | anno$sample_id %in% quality_single_cells, ]
stopifnot(ncol(molecules) == nrow(anno),
          colnames(molecules) == anno$sample_id)

reads &lt;- reads[, grepl(&quot;bulk&quot;, colnames(reads)) |
                         colnames(reads) %in% quality_single_cells]
stopifnot(ncol(reads) == nrow(anno),
          colnames(reads) == anno$sample_id)</code></pre>
<p>Also remove batch 2 of individual 19098.</p>
<pre class="r"><code>molecules &lt;- molecules[, !(anno$individual == 19098 &amp; anno$batch == 2)]
reads &lt;- reads[, !(anno$individual == 19098 &amp; anno$batch == 2)]
anno &lt;- anno[!(anno$individual == 19098 &amp; anno$batch == 2), ]
stopifnot(ncol(molecules) == nrow(anno))</code></pre>
<p>Remove genes with zero read counts in the single cells or bulk samples.</p>
<pre class="r"><code>expressed &lt;- rowSums(molecules[, anno$well == &quot;bulk&quot;]) &gt; 0 &amp;
             rowSums(molecules[, anno$well != &quot;bulk&quot;]) &gt; 0
molecules &lt;- molecules[expressed, ]
dim(molecules)</code></pre>
<pre><code>[1] 17084   571</code></pre>
<pre class="r"><code>expressed &lt;- rowSums(reads[, anno$well == &quot;bulk&quot;]) &gt; 0 &amp;
             rowSums(reads[, anno$well != &quot;bulk&quot;]) &gt; 0
reads &lt;- reads[expressed, ]
dim(reads)</code></pre>
<pre><code>[1] 17093   571</code></pre>
<p>Split the bulk and single samples.</p>
<pre class="r"><code>molecules_bulk &lt;- molecules[, anno$well == &quot;bulk&quot;]
molecules_single &lt;- molecules[, anno$well != &quot;bulk&quot;]
reads_bulk &lt;- reads[, anno$well == &quot;bulk&quot;]
reads_single &lt;- reads[, anno$well != &quot;bulk&quot;]
anno_single &lt;- anno[anno$well != &quot;bulk&quot;,]</code></pre>
<p>How many genes have greater than or equal to 1,024 molecules in at least one of the cells?</p>
<pre class="r"><code>overexpressed_genes &lt;- rownames(molecules_single)[apply(molecules_single, 1,
                                                        function(x) any(x &gt;= 1024))]</code></pre>
<p>15 have greater than or equal to 1,024 molecules. Remove them.</p>
<pre class="r"><code>molecules_single &lt;- molecules_single[!(rownames(molecules_single) %in% overexpressed_genes), ]
reads_single &lt;- reads_single[!(rownames(reads_single) %in% overexpressed_genes), ]</code></pre>
<p>Correct for collision probability. See <a href="http://www.nature.com/nmeth/journal/v11/n6/full/nmeth.2930.html#methods">Grun et al. 2014</a> for details.</p>
<pre class="r"><code>molecules_single_collision &lt;- -1024 * log(1 - molecules_single / 1024)</code></pre>
<p>Standardization without log transformation (in order to calculate CV for gene expression noise)</p>
<pre class="r"><code>molecules_single_cpm &lt;- cpm(molecules_single_collision, log = FALSE)
reads_single_cpm &lt;- cpm(reads_single, log = FALSE)</code></pre>
</div>
<div id="total-ercc-vs-total-molecule-and-total-reads-counts" class="section level2">
<h2>total ERCC vs total molecule and total reads (counts)</h2>
<p>Number of total molecule correlates with total read number.</p>
<pre class="r"><code>anno_single$total_reads &lt;- apply(reads_single, 2, sum)
anno_single$total_molecules &lt;- apply(molecules_single_collision, 2, sum)

ggplot(anno_single, aes(x= total_reads, y= total_molecules, col=as.factor(individual), shape=as.factor(batch))) + geom_point() + labs(x = &quot;total reads&quot;, y = &quot;total molecule&quot;)</code></pre>
<p><img src="figure/ercc-cpm.Rmd/molecule-reads-1.png" title="" alt="" width="672" style="display: block; margin: auto;" /></p>
<p>Look at ERCC and endogenous gene separately. * Total ERCC molecule counts is independent on total reads, total moleceuls, or total gene molecule counts * Total ERCC molecule counts show (surprisngly) individual effect, where 19098 have more and 19239 have fewer.</p>
<pre class="r"><code>anno_single$total_ERCC_molecule &lt;- apply(molecules_single_collision[grep(&quot;ERCC&quot;, rownames(molecules_single)), ],2,sum)

anno_single$total_gene_molecule &lt;- apply(molecules_single_collision[grep(&quot;ENSG&quot;, rownames(molecules_single)), ],2,sum)

ggplot(anno_single, aes(x= total_reads, y= total_ERCC_molecule, col=as.factor(individual), shape=as.factor(batch))) + geom_point() + labs(x = &quot;total reads&quot;, y = &quot;total ERCC molecule&quot;)</code></pre>
<p><img src="figure/ercc-cpm.Rmd/ERCC-molecule-1.png" title="" alt="" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>ggplot(anno_single, aes(x= total_molecules, y= total_ERCC_molecule, col=as.factor(individual), shape=as.factor(batch))) + geom_point() + labs(x = &quot;total molecules&quot;, y = &quot;total ERCC molecule&quot;)</code></pre>
<p><img src="figure/ercc-cpm.Rmd/ERCC-molecule-2.png" title="" alt="" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>ggplot(anno_single, aes(x= total_gene_molecule, y= total_ERCC_molecule, col=as.factor(individual), shape=as.factor(batch))) + geom_point() + labs(x = &quot;total endogenous gene molecules&quot;, y = &quot;total ERCC molecule&quot;)</code></pre>
<p><img src="figure/ercc-cpm.Rmd/ERCC-molecule-3.png" title="" alt="" width="672" style="display: block; margin: auto;" /></p>
<p>PC1 of ERCC expression is explained by total ERCC molecule counts, but not total reads or total molecules</p>
<pre class="r"><code>## without cpm standardization
molecules_single_ERCC &lt;- molecules_single_collision[grep(&quot;ERCC&quot;, rownames(molecules_single_collision)), ]

pca_ERCC &lt;- run_pca(molecules_single_ERCC)
pca_ERCC_plot &lt;- plot_pca(pca_ERCC$PCs, explained = pca_ERCC$explained,
         metadata = anno_single, color = &quot;individual&quot;,
         shape = &quot;batch&quot;, factors = c(&quot;individual&quot;, &quot;batch&quot;))

pca_ERCC_plot + labs(title=&quot;ERCC spike-in (no-cpm)&quot;)</code></pre>
<p><img src="figure/ercc-cpm.Rmd/ERCC-pca-1.png" title="" alt="" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>## PC1 of ERCC expression is explained by total ERCC molecule number
pca_ERCC &lt;- prcomp(t(molecules_single_ERCC), retx = TRUE, scale. = TRUE, center = TRUE)
pca_ERCC_anno &lt;- cbind(anno_single, pca_ERCC$x)

ggplot(pca_ERCC_anno, aes(x = PC1, y = total_ERCC_molecule, col = as.factor(individual), shape = as.factor(batch))) +geom_point() + labs(title=&quot;ERCC spike-in (no-cpm)&quot;)</code></pre>
<p><img src="figure/ercc-cpm.Rmd/ERCC-pca-2.png" title="" alt="" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>ggplot(pca_ERCC_anno, aes(x = PC1, y = total_molecules, col = as.factor(individual), shape = as.factor(batch))) +geom_point() + labs(title=&quot;ERCC spike-in (no-cpm)&quot;)</code></pre>
<p><img src="figure/ercc-cpm.Rmd/ERCC-pca-3.png" title="" alt="" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>ggplot(pca_ERCC_anno, aes(x = PC1, y = total_reads, col = as.factor(individual), shape = as.factor(batch))) +geom_point() + labs(title=&quot;ERCC spike-in (no-cpm)&quot;)</code></pre>
<p><img src="figure/ercc-cpm.Rmd/ERCC-pca-4.png" title="" alt="" width="672" style="display: block; margin: auto;" /></p>
</div>
<div id="cpm-transformation-based-on-total-molecule-counts-wrong" class="section level2">
<h2>cpm transformation based on total molecule counts (WRONG!)</h2>
<p>After cpm, total ERCC molecule number is no longer independent on total molecule counts.</p>
<pre class="r"><code>anno_single$total_molecules_cpm &lt;- apply(molecules_single_cpm, 2, sum)
anno_single$total_ERCC_molecule_cpm &lt;- apply(molecules_single_cpm[grep(&quot;ERCC&quot;, rownames(molecules_single)), ],2,sum)

anno_single$total_gene_molecule_cpm &lt;- apply(molecules_single_cpm[grep(&quot;ENSG&quot;, rownames(molecules_single)), ],2,sum)

## before and after cpm
ggplot(anno_single, aes(x= total_ERCC_molecule_cpm, y= total_ERCC_molecule, col=as.factor(individual), shape=as.factor(batch))) + geom_point() + labs(x = &quot;total ERCC molecule cpm (no log)&quot;, y = &quot;total ERCC molecule&quot;, title =&quot;wrong cpm&quot;)</code></pre>
<p><img src="figure/ercc-cpm.Rmd/ercc-cpm-total-molecule-1.png" title="" alt="" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>ggplot(anno_single, aes(x= total_molecules, y= total_ERCC_molecule_cpm, col=as.factor(individual), shape=as.factor(batch))) + geom_point() + labs(x = &quot;total molecule counts&quot;, y = &quot;total ERCC molecule cpm&quot;, title =&quot;wrong cpm&quot;)</code></pre>
<p><img src="figure/ercc-cpm.Rmd/ercc-cpm-total-molecule-2.png" title="" alt="" width="672" style="display: block; margin: auto;" /></p>
</div>
<div id="cpm-transformation-based-on-ercc-correct" class="section level2">
<h2>cpm transformation based on ERCC (CORRECT!)</h2>
<p>cpm of ERCC should be perfomed seperated.</p>
<pre class="r"><code>## read counts of ERCC
reads_single_ERCC &lt;- reads_single[grep(&quot;ERCC&quot;, rownames(reads_single)), ]

## sum of ERCC readds per cell
anno_single$total_ERCC_reads &lt;- apply (reads_single_ERCC, 2, sum)

ggplot(anno_single, aes(x= total_ERCC_reads, y= total_ERCC_molecule, col=as.factor(individual), shape=as.factor(batch))) + geom_point() + labs(x = &quot;total ERCC molecule reads (counts)&quot;, y = &quot;total ERCC molecule (counts)&quot;)</code></pre>
<p><img src="figure/ercc-cpm.Rmd/ercc-cpm-reads-1.png" title="" alt="" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>## cpm of ERCC (seperate from endogenous genes)
molecules_single_ERCC_cpm &lt;- cpm(molecules_single_ERCC, log = TRUE)

## pca
pca_ERCC_cpm &lt;- run_pca(molecules_single_ERCC_cpm)
pca_ERCC_plot_cpm &lt;- plot_pca(pca_ERCC_cpm$PCs, explained = pca_ERCC_cpm$explained,
         metadata = anno_single, color = &quot;individual&quot;,
         shape = &quot;batch&quot;, factors = c(&quot;individual&quot;, &quot;batch&quot;))

pca_ERCC_plot_cpm + labs(title=&quot;ERCC spike-in (cpm ERCC)&quot;)</code></pre>
<p><img src="figure/ercc-cpm.Rmd/ercc-cpm-reads-2.png" title="" alt="" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code># compared the two different kinds of cpm
anno_single$total_ERCC_cpm &lt;- apply(molecules_single_ERCC_cpm, 2, sum)

ggplot(anno_single, aes(x= total_ERCC_molecule_cpm, y= total_ERCC_cpm, col=as.factor(individual), shape=as.factor(batch))) + geom_point() + labs(x = &quot;total ERCC molecule cpm all molecules (no log)&quot;, y = &quot;total ERCC cpm (log)&quot;, title =&quot;correct cpm&quot;)</code></pre>
<p><img src="figure/ercc-cpm.Rmd/ercc-cpm-reads-3.png" title="" alt="" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>ggplot(anno_single, aes(x= total_molecules, y= total_ERCC_cpm, col=as.factor(individual), shape=as.factor(batch))) + geom_point() + labs(x = &quot;total molecule counts&quot;, y = &quot;total ERCC cpm (log)&quot;, title =&quot;correct cpm&quot;)</code></pre>
<p><img src="figure/ercc-cpm.Rmd/ercc-cpm-reads-4.png" title="" alt="" width="672" style="display: block; margin: auto;" /></p>
</div>
<div id="linear-shift-normalization" class="section level2">
<h2>Linear Shift Normalization</h2>
<p>Try the linear shift normalization after cpm ERCC separately and then use the method <a href="ercc-normalization.html#linear-shift-normalization">linear shift normalization</a>. There is error in the original r code. It shoud be Y = mX + b -&gt; X = (Y - b) / m instead of Y = mX + b -&gt; X = (Y + b) / m. After the shift, take exponential to get non log transformed value.</p>
<pre class="r"><code>## prepare ERCC concentration information 
ercc_single &lt;- ercc[ercc$id %in% rownames(molecules_single), ]
stopifnot(rownames(molecules_single_ERCC_cpm) == ercc_single$id)

## cpm log transformed endogenous genes
molecules_single_genes &lt;- molecules_single_collision[grep(&quot;ENSG&quot;, rownames(molecules_single)), ]
molecules_single_genes_cpm &lt;- cpm(molecules_single_genes, log = TRUE)

## linear shift 
single_norm &lt;- molecules_single_genes_cpm
single_norm[, ] &lt;- NA
for (i in 1:ncol(single_norm)) {
  single_fit &lt;- lm(molecules_single_ERCC_cpm[, i] ~ log2(ercc_single$conc_mix1))
  # Y = mX + b -&gt; X = (Y - b) / m
  single_norm[, i] &lt;- (molecules_single_genes_cpm[, i] - single_fit$coefficients[1]) /
                       single_fit$coefficients[2]
}
stopifnot(!is.na(single_norm))


pca_single_norm &lt;- run_pca(single_norm)
plot_pca(pca_single_norm$PCs, explained = pca_single_norm$explained,
         metadata = anno_single, color = &quot;individual&quot;,
         shape = &quot;batch&quot;, factors = c(&quot;individual&quot;, &quot;batch&quot;)) + labs(title = &quot;Linear shift normalization (log2 cpm)&quot;)</code></pre>
<p><img src="figure/ercc-cpm.Rmd/ercc-norm-1.png" title="" alt="" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>## exponential
single_norm_expt &lt;- 2^(single_norm)
pca_single_norm_expt &lt;- run_pca(single_norm_expt)
plot_pca(pca_single_norm_expt$PCs, explained = pca_single_norm_expt$explained,
         metadata = anno_single, color = &quot;individual&quot;,
         shape = &quot;batch&quot;, factors = c(&quot;individual&quot;, &quot;batch&quot;)) + labs(title = &quot;Linear shift normalization (expt cpm)&quot;)</code></pre>
<p><img src="figure/ercc-cpm.Rmd/ercc-norm-2.png" title="" alt="" width="672" style="display: block; margin: auto;" /></p>
</div>
<div id="session-information" class="section level2">
<h2>Session information</h2>
<pre class="r"><code>sessionInfo()</code></pre>
<pre><code>R version 3.2.0 (2015-04-16)
Platform: x86_64-unknown-linux-gnu (64-bit)

locale:
 [1] LC_CTYPE=en_US.UTF-8       LC_NUMERIC=C              
 [3] LC_TIME=en_US.UTF-8        LC_COLLATE=en_US.UTF-8    
 [5] LC_MONETARY=en_US.UTF-8    LC_MESSAGES=en_US.UTF-8   
 [7] LC_PAPER=en_US.UTF-8       LC_NAME=C                 
 [9] LC_ADDRESS=C               LC_TELEPHONE=C            
[11] LC_MEASUREMENT=en_US.UTF-8 LC_IDENTIFICATION=C       

attached base packages:
[1] stats     graphics  grDevices utils     datasets  methods   base     

other attached packages:
[1] testit_0.4    ggplot2_1.0.1 edgeR_3.10.2  limma_3.24.9  knitr_1.10.5 

loaded via a namespace (and not attached):
 [1] Rcpp_0.12.0      digest_0.6.8     MASS_7.3-40      grid_3.2.0      
 [5] plyr_1.8.3       gtable_0.1.2     formatR_1.2      magrittr_1.5    
 [9] scales_0.2.4     evaluate_0.7     stringi_0.4-1    reshape2_1.4.1  
[13] rmarkdown_0.6.1  labeling_0.3     proto_0.3-10     tools_3.2.0     
[17] stringr_1.0.0    munsell_0.4.2    yaml_2.1.13      colorspace_1.2-6
[21] htmltools_0.2.6 </code></pre>
</div>


<!-- some extra javascript for older browsers -->
<script type="text/javascript" src="libs/polyfill.js"></script>

<script>

// manage active state of menu based on current page
$(document).ready(function () {

    // active menu
    href = window.location.pathname
    href = href.substr(href.lastIndexOf('/') + 1)
    $('a[href="' + href + '"]').parent().addClass('active');

    // manage active menu header
    if (href.startsWith('authoring_'))
      $('a[href="' + 'authoring' + '"]').parent().addClass('active');
    else if (href.endsWith('_format.html'))
      $('a[href="' + 'formats' + '"]').parent().addClass('active');
    else if (href.startsWith('developer_'))
      $('a[href="' + 'developer' + '"]').parent().addClass('active');

});

</script>

</div>

<script>

// add bootstrap table styles to pandoc tables
$(document).ready(function () {
  $('tr.header').parent('thead').parent('table').addClass('table table-condensed');
});

</script>


</body>
</html>
