<!DOCTYPE html>

<html xmlns="http://www.w3.org/1999/xhtml">

<head>

<meta charset="utf-8">
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="generator" content="pandoc" />

<meta name="author" content="Po-Yuan Tung" />

<meta name="date" content="2015-02-25" />

<title>Deteced genes and total molecule counts per cell</title>

<script src="libs/jquery-1.11.0/jquery.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="libs/bootstrap-3.3.1/css/united.min.css" rel="stylesheet" />
<script src="libs/bootstrap-3.3.1/js/bootstrap.min.js"></script>
<script src="libs/bootstrap-3.3.1/shim/html5shiv.min.js"></script>
<script src="libs/bootstrap-3.3.1/shim/respond.min.js"></script>

<style type="text/css">

/* padding for bootstrap navbar */
body {
  padding-top: 50px;
  padding-bottom: 40px;
}


/* offset scroll position for anchor links (for fixed navbar)  */
.section h2 {
  padding-top: 55px;
  margin-top: -55px;
}
.section h3 {
  padding-top: 55px;
  margin-top: -55px;
}



/* don't use link color in navbar */
.dropdown-menu>li>a {
  color: black;
}

/* some padding for disqus */
#disqus_thread {
  margin-top: 45px;
}

</style>

<link rel="stylesheet" href="libs/font-awesome-4.1.0/css/font-awesome.min.css"/>

<style type="text/css">code{white-space: pre;}</style>
<link rel="stylesheet"
      href="libs/highlight/textmate.css"
      type="text/css" />
<script src="libs/highlight/highlight.js"></script>
<style type="text/css">
  pre:not([class]) {
    background-color: white;
  }
</style>
<script type="text/javascript">
if (window.hljs && document.readyState && document.readyState === "complete") {
   window.setTimeout(function() {
      hljs.initHighlighting();
   }, 0);
}
</script>



</head>

<body>

<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
code {
  color: inherit;
  background-color: rgba(0, 0, 0, 0.04);
}
img { 
  max-width:100%; 
  height: auto; 
}
</style>
<div class="container-fluid main-container">


<div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">singleCellSeq</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li><a href="index.html">Home</a></li>
        <li><a href="about.html">About</a></li>
        <li><a href="license.html">License</a></li>
        <li><a href="https://github.com/jdblischak/singleCellSeq">GitHub</a></li>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<div id="header">
<h1 class="title">Deteced genes and total molecule counts per cell</h1>
<h4 class="author"><em>Po-Yuan Tung</em></h4>
<h4 class="date"><em>2015-02-25</em></h4>
</div>

<div id="TOC">
<ul>
<li><a href="#prepare-single-cell-molecule-data">Prepare single cell molecule data</a></li>
<li><a href="#distribution-of-molecule-counts-per-gene-in-each-individual">Distribution of molecule counts per gene in each individual</a></li>
<li><a href="#number-of-genes-detected-and-total-molecule-counts">number of genes detected and total molecule counts</a></li>
<li><a href="#session-information">Session information</a></li>
</ul>
</div>

<p><strong>Last updated:</strong> 2015-09-14</p>
<p><strong>Code version:</strong> 11a61e12229c07d54ae50b9a5db91ff78b59c4e4</p>
<pre class="r"><code>source(&quot;functions.R&quot;)
library(&quot;limma&quot;)
library(&quot;edgeR&quot;)
library(ggplot2)
theme_set(theme_bw(base_size = 16))</code></pre>
<p>From previous analysis, we found that <a href="compare-reads-v-molecules.html#effect-of-sequencing-depth-on-molecule-count">19098 has fewer reads but more molecules</a>. One of the possible cause is that there are more lowly expressed genes being detected in 19098.</p>
<div id="prepare-single-cell-molecule-data" class="section level2">
<h2>Prepare single cell molecule data</h2>
<p>Input annotation</p>
<pre class="r"><code>anno &lt;- read.table(&quot;../data/annotation.txt&quot;, header = TRUE,
                   stringsAsFactors = FALSE)</code></pre>
<p>Input molecule counts</p>
<pre class="r"><code>molecules &lt;- read.table(&quot;../data/molecules.txt&quot;, header = TRUE,
           stringsAsFactors = FALSE)</code></pre>
<p>Input read count</p>
<pre class="r"><code>reads &lt;- read.table(&quot;../data/reads.txt&quot;, header = TRUE,
                    stringsAsFactors = FALSE)</code></pre>
<p>Input list of quality single cells.</p>
<pre class="r"><code>quality_single_cells &lt;- scan(&quot;../data/quality-single-cells.txt&quot;,
                             what = &quot;character&quot;)</code></pre>
<p>Keep only the single cells that passed the <a href="qc-cell-ipsc.html">QC filters</a> and the bulk samples.</p>
<pre class="r"><code>reads &lt;- reads[, grepl(&quot;bulk&quot;, colnames(reads)) |
                 colnames(reads) %in% quality_single_cells]
molecules &lt;- molecules[, grepl(&quot;bulk&quot;, colnames(molecules)) |
                         colnames(molecules) %in% quality_single_cells]
anno &lt;- anno[anno$well == &quot;bulk&quot; | anno$sample_id %in% quality_single_cells, ]
stopifnot(dim(reads) == dim(molecules),
          nrow(anno) == ncol(molecules))</code></pre>
<p>Remove batch 2 of individual 19098.</p>
<pre class="r"><code>molecules_no &lt;- molecules[, !(anno$individual == 19098 &amp; anno$batch == 2)]
reads_single &lt;- reads[, !(anno$individual == 19098 &amp; anno$batch == 2)]
anno_no &lt;- anno[!(anno$individual == 19098 &amp; anno$batch == 2), ]
stopifnot(ncol(molecules_no) == nrow(anno_no))</code></pre>
<p>Remove bulk samples.</p>
<pre class="r"><code>molecules_single &lt;- molecules_no[, anno_no$well != &quot;bulk&quot;]
anno_single &lt;- anno_no[anno_no$well != &quot;bulk&quot;, ]
stopifnot(ncol(molecules_single) == nrow(anno_single))</code></pre>
<p>Remove genes with zero count in the single cells</p>
<pre class="r"><code>expressed_single &lt;- rowSums(molecules_single) &gt; 0
molecules_single &lt;- molecules_single[which(expressed_single), ]
reads_single &lt;- reads_single[expressed_single, ]</code></pre>
<p>remove gene with molecule count larger than 1024 (15 if them)</p>
<pre class="r"><code>overexpressed_genes &lt;- rownames(molecules_single)[apply(molecules_single, 1,
                                                        function(x) any(x &gt;= 1024))]
molecules_single &lt;- molecules_single[!(rownames(molecules_single) %in% overexpressed_genes), ]</code></pre>
<p>collision probability and cpm molecule counts</p>
<pre class="r"><code>molecules_single_collision &lt;- -1024 * log(1 - molecules_single / 1024)
molecules_single_cpm &lt;- cpm(molecules_single_collision, log = TRUE)</code></pre>
</div>
<div id="distribution-of-molecule-counts-per-gene-in-each-individual" class="section level2">
<h2>Distribution of molecule counts per gene in each individual</h2>
<pre class="r"><code>## 19098 (total 142 cells)
molecules_single_cpm_19098 &lt;- molecules_single_cpm[,grep(&quot;19098&quot;, colnames(molecules_single_cpm))]
hist(molecules_single_cpm_19098, breaks=100)</code></pre>
<p><img src="figure/gene-and-total-molecule.Rmd/unnamed-chunk-11-1.png" title="" alt="" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>plot(density(molecules_single_cpm_19098, na.rm = TRUE))</code></pre>
<p><img src="figure/gene-and-total-molecule.Rmd/unnamed-chunk-11-2.png" title="" alt="" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>## 19101 (total 197 cells)
molecules_single_cpm_19101 &lt;- molecules_single_cpm[,grep(&quot;19101&quot;, colnames(molecules_single_cpm))]
hist(molecules_single_cpm_19101, breaks=100)</code></pre>
<p><img src="figure/gene-and-total-molecule.Rmd/unnamed-chunk-11-3.png" title="" alt="" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>plot(density(molecules_single_cpm_19101, na.rm = TRUE))</code></pre>
<p><img src="figure/gene-and-total-molecule.Rmd/unnamed-chunk-11-4.png" title="" alt="" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>## 19239 (total 197 cells)
molecules_single_cpm_19239 &lt;- molecules_single_cpm[,grep(&quot;19239&quot;, colnames(molecules_single_cpm))]
hist(molecules_single_cpm_19239, breaks=100)</code></pre>
<p><img src="figure/gene-and-total-molecule.Rmd/unnamed-chunk-11-5.png" title="" alt="" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>plot(density(molecules_single_cpm_19239, na.rm = TRUE))</code></pre>
<p><img src="figure/gene-and-total-molecule.Rmd/unnamed-chunk-11-6.png" title="" alt="" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>## plot all together
plot_multi_dens &lt;- function(s)
{
    junk.x = NULL
    junk.y = NULL
    for(i in 1:length(s)) {
        junk.x = c(junk.x, density(s[[i]])$x)
        junk.y = c(junk.y, density(s[[i]])$y)
    }
    xr &lt;- range(junk.x)
    yr &lt;- range(junk.y)
    plot(density(s[[1]]), xlim = xr, ylim = yr, main = &quot;&quot;)
    for(i in 1:length(s)) {
        lines(density(s[[i]]), xlim = xr, ylim = yr, col = i)
    }
}

## the input of the following function MUST be a numeric list
plot_multi_dens(list(molecules_single_cpm_19098, molecules_single_cpm_19101, molecules_single_cpm_19239))
legend(12,1.8, c(&quot;19098&quot;,&quot;19101&quot;, &quot;19239&quot;), lwd=c(2.5,2.5),col=c(&quot;black&quot;, &quot;red&quot;, &quot;green&quot;))</code></pre>
<p><img src="figure/gene-and-total-molecule.Rmd/unnamed-chunk-11-7.png" title="" alt="" width="672" style="display: block; margin: auto;" /></p>
</div>
<div id="number-of-genes-detected-and-total-molecule-counts" class="section level2">
<h2>number of genes detected and total molecule counts</h2>
<pre class="r"><code>## number of genes detected in each cell
anno_single$number_genes &lt;- colSums(molecules_single &gt; 0)

## number of total molecule counts in each cell
anno_single$total_molecule &lt;- colSums(molecules_single) 

## plot 
ggplot(anno_single, aes(x = total_molecule, y = number_genes, col = as.factor(individual), shape = as.factor(batch))) + geom_point(size = 3, alpha = 0.5)</code></pre>
<p><img src="figure/gene-and-total-molecule.Rmd/unnamed-chunk-12-1.png" title="" alt="" width="672" style="display: block; margin: auto;" /></p>
</div>
<div id="session-information" class="section level2">
<h2>Session information</h2>
<pre class="r"><code>sessionInfo()</code></pre>
<pre><code>R version 3.2.0 (2015-04-16)
Platform: x86_64-unknown-linux-gnu (64-bit)

locale:
 [1] LC_CTYPE=en_US.UTF-8       LC_NUMERIC=C              
 [3] LC_TIME=en_US.UTF-8        LC_COLLATE=en_US.UTF-8    
 [5] LC_MONETARY=en_US.UTF-8    LC_MESSAGES=en_US.UTF-8   
 [7] LC_PAPER=en_US.UTF-8       LC_NAME=C                 
 [9] LC_ADDRESS=C               LC_TELEPHONE=C            
[11] LC_MEASUREMENT=en_US.UTF-8 LC_IDENTIFICATION=C       

attached base packages:
[1] stats     graphics  grDevices utils     datasets  methods   base     

other attached packages:
[1] ggplot2_1.0.1 edgeR_3.10.2  limma_3.24.9  knitr_1.10.5 

loaded via a namespace (and not attached):
 [1] Rcpp_0.12.0      digest_0.6.8     MASS_7.3-40      grid_3.2.0      
 [5] plyr_1.8.3       gtable_0.1.2     formatR_1.2      magrittr_1.5    
 [9] scales_0.2.4     evaluate_0.7     stringi_0.4-1    reshape2_1.4.1  
[13] rmarkdown_0.6.1  labeling_0.3     proto_0.3-10     tools_3.2.0     
[17] stringr_1.0.0    munsell_0.4.2    yaml_2.1.13      colorspace_1.2-6
[21] htmltools_0.2.6 </code></pre>
</div>


<!-- some extra javascript for older browsers -->
<script type="text/javascript" src="libs/polyfill.js"></script>

<script>

// manage active state of menu based on current page
$(document).ready(function () {

    // active menu
    href = window.location.pathname
    href = href.substr(href.lastIndexOf('/') + 1)
    $('a[href="' + href + '"]').parent().addClass('active');

    // manage active menu header
    if (href.startsWith('authoring_'))
      $('a[href="' + 'authoring' + '"]').parent().addClass('active');
    else if (href.endsWith('_format.html'))
      $('a[href="' + 'formats' + '"]').parent().addClass('active');
    else if (href.startsWith('developer_'))
      $('a[href="' + 'developer' + '"]').parent().addClass('active');

});

</script>

</div>

<script>

// add bootstrap table styles to pandoc tables
$(document).ready(function () {
  $('tr.header').parent('thead').parent('table').addClass('table table-condensed');
});

</script>


</body>
</html>
