<!DOCTYPE html>

<html xmlns="http://www.w3.org/1999/xhtml">

<head>

<meta charset="utf-8">
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="generator" content="pandoc" />

<meta name="author" content="Po-Yuan Tung" />

<meta name="date" content="2015-09-15" />

<title>Detected genes and total molecule counts per cell</title>

<script src="libs/jquery-1.11.0/jquery.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="libs/bootstrap-3.3.1/css/united.min.css" rel="stylesheet" />
<script src="libs/bootstrap-3.3.1/js/bootstrap.min.js"></script>
<script src="libs/bootstrap-3.3.1/shim/html5shiv.min.js"></script>
<script src="libs/bootstrap-3.3.1/shim/respond.min.js"></script>

<style type="text/css">

/* padding for bootstrap navbar */
body {
  padding-top: 50px;
  padding-bottom: 40px;
}


/* offset scroll position for anchor links (for fixed navbar)  */
.section h2 {
  padding-top: 55px;
  margin-top: -55px;
}
.section h3 {
  padding-top: 55px;
  margin-top: -55px;
}



/* don't use link color in navbar */
.dropdown-menu>li>a {
  color: black;
}

/* some padding for disqus */
#disqus_thread {
  margin-top: 45px;
}

</style>

<link rel="stylesheet" href="libs/font-awesome-4.1.0/css/font-awesome.min.css"/>

<style type="text/css">code{white-space: pre;}</style>
<link rel="stylesheet"
      href="libs/highlight/textmate.css"
      type="text/css" />
<script src="libs/highlight/highlight.js"></script>
<style type="text/css">
  pre:not([class]) {
    background-color: white;
  }
</style>
<script type="text/javascript">
if (window.hljs && document.readyState && document.readyState === "complete") {
   window.setTimeout(function() {
      hljs.initHighlighting();
   }, 0);
}
</script>



</head>

<body>

<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
code {
  color: inherit;
  background-color: rgba(0, 0, 0, 0.04);
}
img { 
  max-width:100%; 
  height: auto; 
}
</style>
<div class="container-fluid main-container">


<div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">singleCellSeq</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li><a href="index.html">Home</a></li>
        <li><a href="about.html">About</a></li>
        <li><a href="license.html">License</a></li>
        <li><a href="https://github.com/jdblischak/singleCellSeq">GitHub</a></li>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<div id="header">
<h1 class="title">Detected genes and total molecule counts per cell</h1>
<h4 class="author"><em>Po-Yuan Tung</em></h4>
<h4 class="date"><em>2015-09-15</em></h4>
</div>

<div id="TOC">
<ul>
<li><a href="#prepare-single-cell-molecule-data">Prepare single cell molecule data</a></li>
<li><a href="#number-of-genes-detected-and-total-molecule-counts-in-single-cells">Number of genes detected and total molecule counts in single cells</a></li>
<li><a href="#distribution-of-molecule-counts-per-gene-in-each-individual">Distribution of molecule counts per gene in each individual</a><ul>
<li><a href="#all-endogenous-genes">All endogenous genes</a></li>
<li><a href="#top-5000-genes">Top 5000 genes</a></li>
<li><a href="#bottom-5000-genes">Bottom 5000 genes</a></li>
</ul></li>
<li><a href="#median-molecule-counts-per-gene-of-the-same-individual">Median molecule counts per gene of the same individual</a></li>
<li><a href="#session-information">Session information</a></li>
</ul>
</div>

<p><strong>Last updated:</strong> 2016-02-17</p>
<p><strong>Code version:</strong> 3db3cc8e2a02964e7e219887bc3a35d6f8c9d2b9</p>
<pre class="r"><code>source(&quot;functions.R&quot;)
library(&quot;limma&quot;)
library(&quot;edgeR&quot;)
library(&quot;dplyr&quot;)
library(ggplot2)
theme_set(theme_bw(base_size = 16))</code></pre>
<p>From previous analysis, we found that <a href="compare-reads-v-molecules.html#effect-of-sequencing-depth-on-molecule-count">19098 has fewer reads but more molecules</a>. One of the possible cause is that there are more lowly expressed genes being detected in 19098.</p>
<div id="prepare-single-cell-molecule-data" class="section level2">
<h2>Prepare single cell molecule data</h2>
<p>Input annotation</p>
<pre class="r"><code>anno_filter &lt;- read.table(&quot;../data/annotation-filter.txt&quot;, header = TRUE,
                   stringsAsFactors = FALSE)</code></pre>
<p>Input molecule counts</p>
<pre class="r"><code>molecules_filter &lt;- read.table(&quot;../data/molecules-filter.txt&quot;, header = TRUE,
                               stringsAsFactors = FALSE)</code></pre>
<p>Remove ERCC, keep only the endogenous genes</p>
<pre class="r"><code>molecules_filter_ENSG &lt;- molecules_filter[grep(&quot;ENSG&quot;, rownames(molecules_filter)),]
molecules_filter_ERCC &lt;- molecules_filter[grep(&quot;ERCC&quot;, rownames(molecules_filter)),]
stopifnot(nrow(molecules_filter_ERCC) + nrow(molecules_filter_ENSG) == nrow(molecules_filter))</code></pre>
<p>standardization by cpm molecule counts</p>
<pre class="r"><code>molecules_filter_ENSG_cpm &lt;- cpm(molecules_filter_ENSG, log = TRUE)
molecules_filter_ERCC_cpm &lt;- cpm(molecules_filter_ERCC, log = TRUE)</code></pre>
</div>
<div id="number-of-genes-detected-and-total-molecule-counts-in-single-cells" class="section level2">
<h2>Number of genes detected and total molecule counts in single cells</h2>
<p>The number of genes (both endogenous genes and ERCC genes) per cell corelated with the the total molecule numbers. More total molecules, more genes. However, interestingly, 19098 have not only more total gene molecules and more detected genes but also more ERCC total molecules and more ERCC genes detected.</p>
<pre class="r"><code>## number of genes detected in each cell
anno_filter$number_gene &lt;- colSums(molecules_filter_ENSG &gt; 0)
anno_filter$number_ERCC &lt;- colSums(molecules_filter_ERCC &gt; 0)

## number of total molecule counts in each cell
anno_filter$total_gene_molecule &lt;- colSums(molecules_filter_ENSG) 
anno_filter$total_ERCC_molecule &lt;- colSums(molecules_filter_ERCC) 

## plot 
ggplot(anno_filter, aes(x = total_gene_molecule, y = number_gene, col = as.factor(individual), shape = as.factor(batch))) + geom_point(size = 3, alpha = 0.5) </code></pre>
<pre><code>Warning: The shape palette can deal with a maximum of 6 discrete values
because more than 6 becomes difficult to discriminate; you have 8.
Consider specifying shapes manually. if you must have them.</code></pre>
<pre><code>Warning: Removed 147 rows containing missing values (geom_point).</code></pre>
<pre><code>Warning: The shape palette can deal with a maximum of 6 discrete values
because more than 6 becomes difficult to discriminate; you have 8.
Consider specifying shapes manually. if you must have them.</code></pre>
<p><img src="figure/gene-and-total-molecule.Rmd/gene-detected-1.png" title="" alt="" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>ggplot(anno_filter, aes(x = as.factor(individual), y = total_gene_molecule, fill = as.factor(batch))) + geom_violin(alpha = 0.5) + geom_boxplot(alpha = 0.01, width = 0.1, position = position_dodge(width = 0.9)) + labs( title = &quot;total number of gene molecule counts&quot;)</code></pre>
<p><img src="figure/gene-and-total-molecule.Rmd/gene-detected-2.png" title="" alt="" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>ggplot(anno_filter, aes(x = as.factor(individual), y = total_ERCC_molecule, fill = as.factor(batch))) + geom_violin(alpha = 0.5) + geom_boxplot(alpha = 0.01, width = 0.1, position = position_dodge(width = 0.9)) + labs( title = &quot;total numer of ERCC molecule counts&quot;)</code></pre>
<p><img src="figure/gene-and-total-molecule.Rmd/gene-detected-3.png" title="" alt="" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>ggplot(anno_filter, aes(x = as.factor(individual), y = number_gene, fill = as.factor(batch))) + geom_violin(alpha = 0.5) + geom_boxplot(alpha = 0.1, width = 0.1, position = position_dodge(width = 0.9)) + labs( title = &quot;number of endogenous genes detected&quot;)</code></pre>
<p><img src="figure/gene-and-total-molecule.Rmd/gene-detected-4.png" title="" alt="" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>ggplot(anno_filter, aes(x = as.factor(individual), y = number_ERCC, fill = as.factor(batch))) + geom_violin(alpha = 0.5) + geom_boxplot(alpha = 0.1, width = 0.1, position = position_dodge(width = 0.9)) + labs( title = &quot;number of ERCC genes detected&quot;)</code></pre>
<p><img src="figure/gene-and-total-molecule.Rmd/gene-detected-5.png" title="" alt="" width="672" style="display: block; margin: auto;" /></p>
<p>To standardized within each cell, cpm was done separately for endogenous genes and ERCCs. cpm does not seem to affect the endogenous genes, whereas cpm make the ERCC more similar across batch and also across individual, suggesting that the distribution of ERCC gene expression is similar across batch and across indiviual.</p>
<pre class="r"><code>## number of total molecule counts in each cell AFTER cpm
anno_filter$total_gene_molecule_cpm &lt;- colSums(molecules_filter_ENSG_cpm) 
anno_filter$total_ERCC_molecule_cpm &lt;- colSums(molecules_filter_ERCC_cpm)

ggplot(anno_filter, aes(x = as.factor(individual), y = total_gene_molecule_cpm, fill = as.factor(batch))) + geom_violin(alpha = 0.5) + geom_boxplot(alpha = 0.01, width = 0.1, position = position_dodge(width = 0.9)) + labs( title = &quot;total gene molecule (cpm log)&quot;)</code></pre>
<p><img src="figure/gene-and-total-molecule.Rmd/gene-detected-cpm-1.png" title="" alt="" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>ggplot(anno_filter, aes(x = as.factor(individual), y = total_ERCC_molecule_cpm, fill = as.factor(batch))) + geom_violin(alpha = 0.5) + geom_boxplot(alpha = 0.01, width = 0.1, position = position_dodge(width = 0.9)) + labs( title = &quot;total ERCC molecule (cpm log)&quot;)</code></pre>
<p><img src="figure/gene-and-total-molecule.Rmd/gene-detected-cpm-2.png" title="" alt="" width="672" style="display: block; margin: auto;" /></p>
</div>
<div id="distribution-of-molecule-counts-per-gene-in-each-individual" class="section level2">
<h2>Distribution of molecule counts per gene in each individual</h2>
<div id="all-endogenous-genes" class="section level3">
<h3>All endogenous genes</h3>
<p>Do a density plot of molecule counts per gene per cell to see if 19098 have more lowly expressed genes. However, it’s very hard to visualize it.</p>
<pre class="r"><code>## 19098 (total 142 cells)
molecules_filter_ENSG_19098 &lt;- as.matrix(log(molecules_filter_ENSG[,grep(&quot;19098&quot;, colnames(molecules_filter_ENSG))]))

## 19101 (total 197 cells)
molecules_filter_ENSG_19101 &lt;- as.matrix(log(molecules_filter_ENSG[,grep(&quot;19101&quot;, colnames(molecules_filter_ENSG))]))

## 19239 (total 197 cells)
molecules_filter_ENSG_19239 &lt;- as.matrix(log(molecules_filter_ENSG[,grep(&quot;19239&quot;, colnames(molecules_filter_ENSG))]))

## plot all together
plot_multi_dens &lt;- function(s)
{
    junk.x = NULL
    junk.y = NULL
    for(i in 1:length(s)) {
        junk.x = c(junk.x, density(s[[i]])$x)
        junk.y = c(junk.y, density(s[[i]])$y)
    }
    xr &lt;- range(junk.x)
    yr &lt;- range(junk.y)
    plot(density(s[[1]]), xlim = xr, ylim = yr, main = &quot;&quot;)
    for(i in 1:length(s)) {
        lines(density(s[[i]]), xlim = xr, ylim = yr, col = i)
    }
}

plot_multi_dens(list(molecules_filter_ENSG_19098, molecules_filter_ENSG_19101, molecules_filter_ENSG_19239))
legend(5, 0.7, c(&quot;19098&quot;,&quot;19101&quot;, &quot;19239&quot;), lwd=c(2.5,2.5),col=c(&quot;black&quot;, &quot;red&quot;, &quot;green&quot;))</code></pre>
<p><img src="figure/gene-and-total-molecule.Rmd/distribution-all-gene-1.png" title="" alt="" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code># Not sure how to fix the following code. These vectors do not exist. -John 
#
# ## ggplot
# v_19098 &lt;- as.vector(molecules_filter_cpm_19098)
# v_19239 &lt;- as.vector(molecules_filter_cpm_19239)
# v_19101 &lt;- as.vector(molecules_filter_cpm_19101)
# 
# data_cpm &lt;- as.data.frame(cbind(counts=c(v_19098,v_19239,v_19101),individual=rep(c(&quot;19098&quot;,&quot;19239&quot;,&quot;19101&quot;),c(length(v_19098),length(v_19239),length(v_19101)))))
# 
# data_cpm$counts &lt;- as.numeric(as.character(data_cpm$counts))
# 
# ggplot(data_cpm, aes(x = counts,  col = individual)) + geom_density() + labs(title=&quot;molecule per gene per cell&quot;)</code></pre>
</div>
<div id="top-5000-genes" class="section level3">
<h3>Top 5000 genes</h3>
<p>There is some difference before cpm, but the distribution matched perfectly after cpm. However, surprisingly, not all top 5000 genes are expressed in cells.</p>
<pre class="r"><code>## calculate total and select the top 5000
molecules_filter_ENSG$total &lt;- apply(molecules_filter_ENSG, 1, sum)
molecules_filter_ENSG_top &lt;- molecules_filter_ENSG %&gt;% arrange(desc(total)) %&gt;%
  slice(1:5000) %&gt;% select(-total)

## number of gene detected 
anno_filter$number_gene_top &lt;- colSums(molecules_filter_ENSG_top &gt; 0)

ggplot(anno_filter, aes(x = as.factor(individual), y = number_gene_top, fill = as.factor(batch))) + geom_violin(alpha = 0.5) + geom_boxplot(alpha = 0.01, width = 0.1, position = position_dodge(width = 0.9)) + labs( title = &quot;number of gene detected (top 5000)&quot;)</code></pre>
<p><img src="figure/gene-and-total-molecule.Rmd/top-5000-1.png" title="" alt="" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>## divided by individual
molecules_filter_ENSG_top_19098 &lt;- as.matrix(log(molecules_filter_ENSG_top[,grep(&quot;19098&quot;, colnames(molecules_filter_ENSG_top))]))
molecules_filter_ENSG_top_19101 &lt;- as.matrix(log(molecules_filter_ENSG_top[,grep(&quot;19101&quot;, colnames(molecules_filter_ENSG_top))]))
molecules_filter_ENSG_top_19239 &lt;- as.matrix(log(molecules_filter_ENSG_top[,grep(&quot;19239&quot;, colnames(molecules_filter_ENSG_top))]))

plot_multi_dens(list(molecules_filter_ENSG_top_19098, molecules_filter_ENSG_top_19101, molecules_filter_ENSG_top_19239))
legend(5.5 ,0.4, c(&quot;19098&quot;,&quot;19101&quot;, &quot;19239&quot;), lwd=c(2.5,2.5),col=c(&quot;black&quot;, &quot;red&quot;, &quot;green&quot;))
title(&quot;molecule per gene top 5000 genes (counts)&quot;)</code></pre>
<p><img src="figure/gene-and-total-molecule.Rmd/top-5000-2.png" title="" alt="" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>## calculate total and select the top 5000 with cpm
molecules_filter_ENSG_cpm &lt;- as.data.frame(molecules_filter_ENSG_cpm)
molecules_filter_ENSG_cpm$total &lt;- apply(molecules_filter_ENSG_cpm, 1, sum)
molecules_filter_ENSG_cpm_top &lt;- molecules_filter_ENSG_cpm %&gt;% arrange(desc(total)) %&gt;%
  slice(1:5000) %&gt;% select(-total)
stopifnot(rownames(molecules_filter_ENSG_top) == rownames(molecules_filter_ENSG_cpm_top))

## divided by individual
molecules_filter_ENSG_cpm_top_19098 &lt;- as.matrix(molecules_filter_ENSG_cpm_top[,grep(&quot;19098&quot;, colnames(molecules_filter_ENSG_cpm_top))])
molecules_filter_ENSG_cpm_top_19101 &lt;- as.matrix(molecules_filter_ENSG_cpm_top[,grep(&quot;19101&quot;, colnames(molecules_filter_ENSG_cpm_top))])
molecules_filter_ENSG_cpm_top_19239 &lt;- as.matrix(molecules_filter_ENSG_cpm_top[,grep(&quot;19239&quot;, colnames(molecules_filter_ENSG_cpm_top))])

plot_multi_dens(list(molecules_filter_ENSG_cpm_top_19098, molecules_filter_ENSG_cpm_top_19098, molecules_filter_ENSG_cpm_top_19098))
legend(9, 0.25, c(&quot;19098&quot;,&quot;19101&quot;, &quot;19239&quot;), lwd=c(2.5,2.5),col=c(&quot;black&quot;, &quot;red&quot;, &quot;green&quot;))
title(&quot;molecule per gene top 5000 genes (cpm log)&quot;)</code></pre>
<p><img src="figure/gene-and-total-molecule.Rmd/top-5000-3.png" title="" alt="" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>## total molecule number
anno_filter$total_gene_molecule_top &lt;- colSums(molecules_filter_ENSG_top) 
ggplot(anno_filter, aes(x = as.factor(individual), y = total_gene_molecule_top, fill = as.factor(batch))) + geom_violin(alpha = 0.5) + geom_boxplot(alpha = 0.01, width = 0.1, position = position_dodge(width = 0.9)) + labs( title = &quot;total gene molecule top 5000&quot;)</code></pre>
<p><img src="figure/gene-and-total-molecule.Rmd/top-5000-4.png" title="" alt="" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>anno_filter$total_gene_molecule_cpm_top &lt;- colSums(molecules_filter_ENSG_cpm_top) 
ggplot(anno_filter, aes(x = as.factor(individual), y = total_gene_molecule_cpm_top, fill = as.factor(batch))) + geom_violin(alpha = 0.5) + geom_boxplot(alpha = 0.01, width = 0.1, position = position_dodge(width = 0.9)) + labs( title = &quot;total gene molecule top 5000 (cpm log)&quot;)</code></pre>
<p><img src="figure/gene-and-total-molecule.Rmd/top-5000-5.png" title="" alt="" width="672" style="display: block; margin: auto;" /></p>
</div>
<div id="bottom-5000-genes" class="section level3">
<h3>Bottom 5000 genes</h3>
<p>As exprected, more genes have only 1 molecule in 19098 than the other 2 individuals.</p>
<pre class="r"><code>## select the bottom 5000
molecules_filter_ENSG_bottom &lt;- molecules_filter_ENSG %&gt;% arrange(total) %&gt;%
  slice(1:5000) %&gt;% select(-total)

## number of gene detected 
anno_filter$number_gene_bottom &lt;- colSums(molecules_filter_ENSG_bottom &gt; 0)

ggplot(anno_filter, aes(x = as.factor(individual), y = number_gene_bottom, fill = as.factor(batch))) + geom_violin(alpha = 0.5) + geom_boxplot(alpha = 0.01, width = 0.1, position = position_dodge(width = 0.9)) + labs( title = &quot;number of gene detected (bottom 5000)&quot;)</code></pre>
<p><img src="figure/gene-and-total-molecule.Rmd/bottom-5000-1.png" title="" alt="" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>## divided by individual
molecules_filter_ENSG_bottom_19098 &lt;- as.matrix(log(molecules_filter_ENSG_bottom[,grep(&quot;19098&quot;, colnames(molecules_filter_ENSG_bottom))]))
molecules_filter_ENSG_bottom_19101 &lt;- as.matrix(log(molecules_filter_ENSG_bottom[,grep(&quot;19101&quot;, colnames(molecules_filter_ENSG_bottom))]))
molecules_filter_ENSG_bottom_19239 &lt;- as.matrix(log(molecules_filter_ENSG_bottom[,grep(&quot;19239&quot;, colnames(molecules_filter_ENSG_bottom))]))

plot_multi_dens(list(molecules_filter_ENSG_bottom_19098, molecules_filter_ENSG_bottom_19101, molecules_filter_ENSG_bottom_19239))
legend(2.5, 0.2, c(&quot;19098&quot;,&quot;19101&quot;, &quot;19239&quot;), lwd=c(2.5,2.5),col=c(&quot;black&quot;, &quot;red&quot;, &quot;green&quot;))
title(&quot;molecule counts per gene&quot;)</code></pre>
<p><img src="figure/gene-and-total-molecule.Rmd/bottom-5000-2.png" title="" alt="" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>## the bottom 5000 with cpm
molecules_filter_ENSG_cpm_bottom &lt;- molecules_filter_ENSG_cpm %&gt;% arrange(total) %&gt;%
  slice(1:5000) %&gt;% select(-total)
stopifnot(rownames(molecules_filter_ENSG_bottom) == rownames(molecules_filter_ENSG_cpm_bottom))

## divided by individual
molecules_filter_ENSG_cpm_bottom_19098 &lt;- as.matrix(molecules_filter_ENSG_cpm_bottom[,grep(&quot;19098&quot;, colnames(molecules_filter_ENSG_cpm_bottom))])
molecules_filter_ENSG_cpm_bottom_19101 &lt;- as.matrix(molecules_filter_ENSG_cpm_bottom[,grep(&quot;19101&quot;, colnames(molecules_filter_ENSG_cpm_bottom))])
molecules_filter_ENSG_cpm_bottom_19239 &lt;- as.matrix(molecules_filter_ENSG_cpm_bottom[,grep(&quot;19239&quot;, colnames(molecules_filter_ENSG_cpm_bottom))])

plot_multi_dens(list(molecules_filter_ENSG_cpm_bottom_19098, molecules_filter_ENSG_cpm_bottom_19098, molecules_filter_ENSG_cpm_bottom_19098))</code></pre>
<p><img src="figure/gene-and-total-molecule.Rmd/bottom-5000-3.png" title="" alt="" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>## total molecule number
anno_filter$total_gene_molecule_bottom &lt;- colSums(molecules_filter_ENSG_bottom) 
ggplot(anno_filter, aes(x = as.factor(individual), y = total_gene_molecule_bottom, fill = as.factor(batch))) + geom_violin(alpha = 0.5) + geom_boxplot(alpha = 0.01, width = 0.1, position = position_dodge(width = 0.9)) + labs( title = &quot;total gene molecule bottom 5000&quot;)</code></pre>
<p><img src="figure/gene-and-total-molecule.Rmd/bottom-5000-4.png" title="" alt="" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>anno_filter$total_gene_molecule_cpm_bottom &lt;- colSums(molecules_filter_ENSG_cpm_bottom) 
ggplot(anno_filter, aes(x = as.factor(individual), y = total_gene_molecule_cpm_bottom, fill = as.factor(batch))) + geom_violin(alpha = 0.5) + geom_boxplot(alpha = 0.01, width = 0.1, position = position_dodge(width = 0.9)) + labs( title = &quot;total gene molecule bottom 5000 (cpm log)&quot;)</code></pre>
<p><img src="figure/gene-and-total-molecule.Rmd/bottom-5000-5.png" title="" alt="" width="672" style="display: block; margin: auto;" /></p>
</div>
</div>
<div id="median-molecule-counts-per-gene-of-the-same-individual" class="section level2">
<h2>Median molecule counts per gene of the same individual</h2>
<pre class="r"><code>## with cpm 
molecules_counts_per_gene_19098_cpm &lt;- apply(molecules_filter_ENSG_19098, 1, median)
molecules_counts_per_gene_19101_cpm &lt;- apply(molecules_filter_ENSG_19101, 1, median)
molecules_counts_per_gene_19239_cpm &lt;- apply(molecules_filter_ENSG_19239, 1, median)

## plot
plot_multi_dens(list(molecules_counts_per_gene_19098_cpm, molecules_counts_per_gene_19101_cpm, molecules_counts_per_gene_19239_cpm))
legend(5, 0.25 , c(&quot;19098&quot;,&quot;19101&quot;, &quot;19239&quot;), lwd=c(2.5,2.5),col=c(&quot;black&quot;, &quot;red&quot;, &quot;green&quot;))</code></pre>
<p><img src="figure/gene-and-total-molecule.Rmd/total-per-individual-1.png" title="" alt="" width="672" style="display: block; margin: auto;" /></p>
</div>
<div id="session-information" class="section level2">
<h2>Session information</h2>
<pre class="r"><code>sessionInfo()</code></pre>
<pre><code>R version 3.2.0 (2015-04-16)
Platform: x86_64-unknown-linux-gnu (64-bit)

locale:
 [1] LC_CTYPE=en_US.UTF-8       LC_NUMERIC=C              
 [3] LC_TIME=en_US.UTF-8        LC_COLLATE=en_US.UTF-8    
 [5] LC_MONETARY=en_US.UTF-8    LC_MESSAGES=en_US.UTF-8   
 [7] LC_PAPER=en_US.UTF-8       LC_NAME=C                 
 [9] LC_ADDRESS=C               LC_TELEPHONE=C            
[11] LC_MEASUREMENT=en_US.UTF-8 LC_IDENTIFICATION=C       

attached base packages:
[1] stats     graphics  grDevices utils     datasets  methods   base     

other attached packages:
[1] ggplot2_1.0.1 dplyr_0.4.2   edgeR_3.10.2  limma_3.24.9  knitr_1.10.5 

loaded via a namespace (and not attached):
 [1] Rcpp_0.12.0      magrittr_1.5     MASS_7.3-40      munsell_0.4.2   
 [5] colorspace_1.2-6 R6_2.1.1         stringr_1.0.0    httr_0.6.1      
 [9] plyr_1.8.3       tools_3.2.0      parallel_3.2.0   grid_3.2.0      
[13] gtable_0.1.2     DBI_0.3.1        htmltools_0.2.6  lazyeval_0.1.10 
[17] yaml_2.1.13      digest_0.6.8     assertthat_0.1   reshape2_1.4.1  
[21] formatR_1.2      bitops_1.0-6     RCurl_1.95-4.6   evaluate_0.7    
[25] rmarkdown_0.6.1  labeling_0.3     stringi_0.4-1    scales_0.2.4    
[29] proto_0.3-10    </code></pre>
</div>


<!-- some extra javascript for older browsers -->
<script type="text/javascript" src="libs/polyfill.js"></script>

<script>

// manage active state of menu based on current page
$(document).ready(function () {

    // active menu
    href = window.location.pathname
    href = href.substr(href.lastIndexOf('/') + 1)
    $('a[href="' + href + '"]').parent().addClass('active');

    // manage active menu header
    if (href.startsWith('authoring_'))
      $('a[href="' + 'authoring' + '"]').parent().addClass('active');
    else if (href.endsWith('_format.html'))
      $('a[href="' + 'formats' + '"]').parent().addClass('active');
    else if (href.startsWith('developer_'))
      $('a[href="' + 'developer' + '"]').parent().addClass('active');

});

</script>

</div>

<script>

// add bootstrap table styles to pandoc tables
$(document).ready(function () {
  $('tr.header').parent('thead').parent('table').addClass('table table-condensed');
});

</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://cdn.mathjax.org/mathjax/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
