<!DOCTYPE html>

<html xmlns="http://www.w3.org/1999/xhtml">

<head>

<meta charset="utf-8">
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="generator" content="pandoc" />

<meta name="author" content="Joyce Hsiao" />

<meta name="date" content="2016-07-01" />

<title>Generate gene symbols</title>

<script src="libs/jquery-1.11.0/jquery.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="libs/bootstrap-3.3.1/css/united.min.css" rel="stylesheet" />
<script src="libs/bootstrap-3.3.1/js/bootstrap.min.js"></script>
<script src="libs/bootstrap-3.3.1/shim/html5shiv.min.js"></script>
<script src="libs/bootstrap-3.3.1/shim/respond.min.js"></script>

<style type="text/css">

/* padding for bootstrap navbar */
body {
  padding-top: 50px;
  padding-bottom: 40px;
}


/* offset scroll position for anchor links (for fixed navbar)  */
.section h2 {
  padding-top: 55px;
  margin-top: -55px;
}
.section h3 {
  padding-top: 55px;
  margin-top: -55px;
}



/* don't use link color in navbar */
.dropdown-menu>li>a {
  color: black;
}

/* some padding for disqus */
#disqus_thread {
  margin-top: 45px;
}

</style>

<link rel="stylesheet" href="libs/font-awesome-4.1.0/css/font-awesome.min.css"/>

<style type="text/css">code{white-space: pre;}</style>
<link rel="stylesheet"
      href="libs/highlight/textmate.css"
      type="text/css" />
<script src="libs/highlight/highlight.js"></script>
<style type="text/css">
  pre:not([class]) {
    background-color: white;
  }
</style>
<script type="text/javascript">
if (window.hljs && document.readyState && document.readyState === "complete") {
   window.setTimeout(function() {
      hljs.initHighlighting();
   }, 0);
}
</script>



</head>

<body>

<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
code {
  color: inherit;
  background-color: rgba(0, 0, 0, 0.04);
}
img { 
  max-width:100%; 
  height: auto; 
}
</style>
<div class="container-fluid main-container">


<div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">singleCellSeq</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li><a href="index.html">Home</a></li>
        <li><a href="about.html">About</a></li>
        <li><a href="license.html">License</a></li>
        <li><a href="https://github.com/jdblischak/singleCellSeq">GitHub</a></li>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<div id="header">
<h1 class="title">Generate gene symbols</h1>
<h4 class="author"><em>Joyce Hsiao</em></h4>
<h4 class="date"><em>2016-07-01</em></h4>
</div>

<div id="TOC">
<ul>
<li><a href="#objective">Objective</a></li>
<li><a href="#set-up">Set up</a></li>
<li><a href="#generate-gene-ids">Generate Gene IDs</a></li>
<li><a href="#runtime">Runtime</a></li>
<li><a href="#session-information">Session information</a></li>
</ul>
</div>

<p><strong>Last updated:</strong> 2016-07-08</p>
<p><strong>Code version:</strong> 641fc1222ff5c5bc6b4766ad24ea2f5420334427</p>
<div id="objective" class="section level2">
<h2>Objective</h2>
<p>Get gene symbols for the final data set and save them in a text file for annotation references.</p>
</div>
<div id="set-up" class="section level2">
<h2>Set up</h2>
<pre class="r"><code>library(&quot;data.table&quot;)
library(&quot;dplyr&quot;)
library(&quot;limma&quot;)
library(&quot;edgeR&quot;)
library(&quot;ggplot2&quot;)
library(&quot;grid&quot;)
theme_set(theme_bw(base_size = 12))
source(&quot;functions.R&quot;)</code></pre>
<p>Input annotation of QC-filtered data.</p>
<pre class="r"><code>anno_filter &lt;- read.table(&quot;../data/annotation-filter.txt&quot;, 
                          header = TRUE,
                          stringsAsFactors = FALSE)</code></pre>
<p>Import endogeneous genes in the final file.</p>
<pre class="r"><code>molecules_final &lt;- read.table(&quot;../data/molecules-final.txt&quot;, 
                             header = TRUE, 
                             stringsAsFactors = FALSE)</code></pre>
</div>
<div id="generate-gene-ids" class="section level2">
<h2>Generate Gene IDs</h2>
<pre class="r"><code>if(file.exists(&quot;../data/gene-info.txt&quot;)) {
   gene_info &lt;- read.table(file = &quot;../data/gene-info.txt&quot;, sep = &quot;\t&quot;,
                          header = TRUE, stringsAsFactors = FALSE, quote = &quot;&quot;)
   head(gene_info)
} else {
library(&quot;biomaRt&quot;)
  ensembl &lt;- useMart(host = &quot;grch37.ensembl.org&quot;,
                     biomart = &quot;ENSEMBL_MART_ENSEMBL&quot;,
                     dataset = &quot;hsapiens_gene_ensembl&quot;)

  gene_info &lt;- getBM(attributes = c(&quot;ensembl_gene_id&quot;, 
                               &quot;chromosome_name&quot;,
                               &quot;external_gene_name&quot;, 
                               &quot;transcript_count&quot;,
                               &quot;description&quot;),
                      filters = &quot;ensembl_gene_id&quot;,
                      values = rownames(molecules_final),
                      mart = ensembl)
  write.table(gene_info, row.names = FALSE,
              file = &quot;../data/gene-info.txt&quot;, quote = FALSE, sep = &quot;\t&quot;) 

  write.table(gene_info$external_gene_name, row.names = FALSE, col.names = F,
            file = &quot;../data/gene-info-symbol-only.txt&quot;, quote = FALSE, sep = &quot;\t&quot;) 
  
  write.table(gene_info$ensembl_gene_id, row.names = FALSE, col.names = F,
          file = &quot;../data/gene-info-ensg-only.txt&quot;, quote = FALSE, sep = &quot;\t&quot;) 
}</code></pre>
<pre><code>  ensembl_gene_id chromosome_name external_gene_name transcript_count
1 ENSG00000000003               X             TSPAN6                3
2 ENSG00000000005               X               TNMD                2
3 ENSG00000000419              20               DPM1                7
4 ENSG00000000457               1              SCYL3                5
5 ENSG00000000460               1           C1orf112               10
6 ENSG00000001036               6              FUCA2                4
                                                                                            description
1                                                          tetraspanin 6 [Source:HGNC Symbol;Acc:11858]
2                                                            tenomodulin [Source:HGNC Symbol;Acc:17757]
3 dolichyl-phosphate mannosyltransferase polypeptide 1, catalytic subunit [Source:HGNC Symbol;Acc:3005]
4                                            SCY1-like 3 (S. cerevisiae) [Source:HGNC Symbol;Acc:19285]
5                                    chromosome 1 open reading frame 112 [Source:HGNC Symbol;Acc:25565]
6                                          fucosidase, alpha-L- 2, plasma [Source:HGNC Symbol;Acc:4008]</code></pre>
</div>
<div id="runtime" class="section level2">
<h2>Runtime</h2>
<pre class="r"><code>t1 &lt;- Sys.time()
library(&quot;biomaRt&quot;)
  ensembl &lt;- useMart(host = &quot;grch37.ensembl.org&quot;,
                     biomart = &quot;ENSEMBL_MART_ENSEMBL&quot;,
                     dataset = &quot;hsapiens_gene_ensembl&quot;)

  gene_info &lt;- getBM(attributes = c(&quot;ensembl_gene_id&quot;, 
                               &quot;chromosome_name&quot;,
                               &quot;external_gene_name&quot;, 
                               &quot;transcript_count&quot;,
                               &quot;description&quot;),
                      filters = &quot;ensembl_gene_id&quot;,
                      values = rownames(molecules_final),
                      mart = ensembl)
t2 &lt;- Sys.time()</code></pre>
<p>Takes 21.0294459 to processs 13058.</p>
</div>
<div id="session-information" class="section level2">
<h2>Session information</h2>
<pre class="r"><code>sessionInfo()</code></pre>
<pre><code>R version 3.2.0 (2015-04-16)
Platform: x86_64-unknown-linux-gnu (64-bit)

locale:
 [1] LC_CTYPE=en_US.UTF-8       LC_NUMERIC=C              
 [3] LC_TIME=en_US.UTF-8        LC_COLLATE=en_US.UTF-8    
 [5] LC_MONETARY=en_US.UTF-8    LC_MESSAGES=en_US.UTF-8   
 [7] LC_PAPER=en_US.UTF-8       LC_NAME=C                 
 [9] LC_ADDRESS=C               LC_TELEPHONE=C            
[11] LC_MEASUREMENT=en_US.UTF-8 LC_IDENTIFICATION=C       

attached base packages:
[1] grid      stats     graphics  grDevices utils     datasets  methods  
[8] base     

other attached packages:
[1] biomaRt_2.24.0   ggplot2_1.0.1    edgeR_3.10.2     limma_3.24.9    
[5] dplyr_0.4.2      data.table_1.9.4 knitr_1.10.5    

loaded via a namespace (and not attached):
 [1] Rcpp_0.12.4          GenomeInfoDb_1.4.0   formatR_1.2         
 [4] plyr_1.8.3           bitops_1.0-6         tools_3.2.0         
 [7] digest_0.6.8         RSQLite_1.0.0        evaluate_0.7        
[10] gtable_0.1.2         DBI_0.3.1            yaml_2.1.13         
[13] parallel_3.2.0       proto_0.3-10         httr_0.6.1          
[16] stringr_1.0.0        IRanges_2.2.4        S4Vectors_0.6.0     
[19] stats4_3.2.0         Biobase_2.28.0       R6_2.1.1            
[22] AnnotationDbi_1.30.1 XML_3.98-1.2         rmarkdown_0.6.1     
[25] reshape2_1.4.1       magrittr_1.5         scales_0.4.0        
[28] htmltools_0.2.6      MASS_7.3-40          BiocGenerics_0.14.0 
[31] assertthat_0.1       colorspace_1.2-6     stringi_1.0-1       
[34] RCurl_1.95-4.6       munsell_0.4.3        chron_2.3-45        </code></pre>
</div>


<!-- some extra javascript for older browsers -->
<script type="text/javascript" src="libs/polyfill.js"></script>

<script>

// manage active state of menu based on current page
$(document).ready(function () {

    // active menu
    href = window.location.pathname
    href = href.substr(href.lastIndexOf('/') + 1)
    $('a[href="' + href + '"]').parent().addClass('active');

    // manage active menu header
    if (href.startsWith('authoring_'))
      $('a[href="' + 'authoring' + '"]').parent().addClass('active');
    else if (href.endsWith('_format.html'))
      $('a[href="' + 'formats' + '"]').parent().addClass('active');
    else if (href.startsWith('developer_'))
      $('a[href="' + 'developer' + '"]').parent().addClass('active');

});

</script>

</div>

<script>

// add bootstrap table styles to pandoc tables
$(document).ready(function () {
  $('tr.header').parent('thead').parent('table').addClass('table table-condensed');
});

</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://cdn.mathjax.org/mathjax/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
