---
title: "Deteced genes and total molecule counts per cell"
author: "Po-Yuan Tung"
date: 2015-02-25
---

**Last updated:** `r Sys.Date()`

**Code version:** `r system("git log -1 --format='%H'", intern = TRUE)`

```{r chunk-options, include=FALSE}
source("chunk-options.R")
```

```{r}
source("functions.R")
library("limma")
library("edgeR")
library(ggplot2)
theme_set(theme_bw(base_size = 16))
```

From previous analysis, we found that [19098 has fewer reads but more molecules](compare-reads-v-molecules.html#effect-of-sequencing-depth-on-molecule-count). One of the possible cause is that there are more lowly expressed genes being detected in 19098.

## Prepare single cell molecule data

Input annotation
```{r}
anno <- read.table("../data/annotation.txt", header = TRUE,
                   stringsAsFactors = FALSE)
```

Input molecule counts
```{r}
molecules <- read.table("../data/molecules.txt", header = TRUE,
           stringsAsFactors = FALSE)
```

Input read count
```{r}
reads <- read.table("../data/reads.txt", header = TRUE,
                    stringsAsFactors = FALSE)
```

Input list of quality single cells.

```{r input-quality-single-cells}
quality_single_cells <- scan("../data/quality-single-cells.txt",
                             what = "character")
```

Keep only the single cells that passed the [QC filters](qc-cell-ipsc.html) and the bulk samples.

```{r}
reads <- reads[, grepl("bulk", colnames(reads)) |
                 colnames(reads) %in% quality_single_cells]
molecules <- molecules[, grepl("bulk", colnames(molecules)) |
                         colnames(molecules) %in% quality_single_cells]
anno <- anno[anno$well == "bulk" | anno$sample_id %in% quality_single_cells, ]
stopifnot(dim(reads) == dim(molecules),
          nrow(anno) == ncol(molecules))
```

Remove batch 2 of individual 19098.
```{r}
molecules_no <- molecules[, !(anno$individual == 19098 & anno$batch == 2)]
reads_single <- reads[, !(anno$individual == 19098 & anno$batch == 2)]
anno_no <- anno[!(anno$individual == 19098 & anno$batch == 2), ]
stopifnot(ncol(molecules_no) == nrow(anno_no))
```

Remove bulk samples.
```{r}
molecules_single <- molecules_no[, anno_no$well != "bulk"]
anno_single <- anno_no[anno_no$well != "bulk", ]
stopifnot(ncol(molecules_single) == nrow(anno_single))
```

Remove genes with zero count in the single cells
```{r}
expressed_single <- rowSums(molecules_single) > 0
molecules_single <- molecules_single[which(expressed_single), ]
reads_single <- reads_single[expressed_single, ]
```

remove gene with molecule count larger than 1024 (15 if them)
```{r}
overexpressed_genes <- rownames(molecules_single)[apply(molecules_single, 1,
                                                        function(x) any(x >= 1024))]
molecules_single <- molecules_single[!(rownames(molecules_single) %in% overexpressed_genes), ]
```

collision probability and cpm molecule counts
```{r}
molecules_single_collision <- -1024 * log(1 - molecules_single / 1024)
molecules_single_cpm <- cpm(molecules_single_collision, log = TRUE)
```

## Distribution of molecule counts per gene in each individual
```{r}
## 19098 (total 142 cells)
molecules_single_cpm_19098 <- molecules_single_cpm[,grep("19098", colnames(molecules_single_cpm))]

## 19101 (total 197 cells)
molecules_single_cpm_19101 <- molecules_single_cpm[,grep("19101", colnames(molecules_single_cpm))]

## 19239 (total 197 cells)
molecules_single_cpm_19239 <- molecules_single_cpm[,grep("19239", colnames(molecules_single_cpm))]

## plot all together
plot_multi_dens <- function(s)
{
    junk.x = NULL
    junk.y = NULL
    for(i in 1:length(s)) {
        junk.x = c(junk.x, density(s[[i]])$x)
        junk.y = c(junk.y, density(s[[i]])$y)
    }
    xr <- range(junk.x)
    yr <- range(junk.y)
    plot(density(s[[1]]), xlim = xr, ylim = yr, main = "")
    for(i in 1:length(s)) {
        lines(density(s[[i]]), xlim = xr, ylim = yr, col = i)
    }
}

## the input of the following function MUST be a numeric list
plot_multi_dens(list(molecules_single_cpm_19098, molecules_single_cpm_19101, molecules_single_cpm_19239))
legend(12,1.8, c("19098","19101", "19239"), lwd=c(2.5,2.5),col=c("black", "red", "green"))
```

## total molecule counts per gene of the same individual
```{r}
## without cpm standardization
molecules_counts_per_gene_19098 <- log(rowSums(molecules_single[,grep("19098", colnames(molecules_single))])+1)
molecules_counts_per_gene_19101 <- log(rowSums(molecules_single[,grep("19101", colnames(molecules_single))])+1)
molecules_counts_per_gene_19239 <- log(rowSums(molecules_single[,grep("19239", colnames(molecules_single))])+1)

## plot
plot_multi_dens(list(molecules_counts_per_gene_19098, molecules_counts_per_gene_19101, molecules_counts_per_gene_19239))
legend(10, 0.15, c("19098","19101", "19239"), lwd=c(2.5,2.5),col=c("black", "red", "green"))

## with cpm 
molecules_counts_per_gene_19098_cpm <- rowSums(molecules_single_cpm_19098)
molecules_counts_per_gene_19101_cpm <- rowSums(molecules_single_cpm_19101)
molecules_counts_per_gene_19239_cpm <- rowSums(molecules_single_cpm_19239)

## plot
plot_multi_dens(list(molecules_counts_per_gene_19098_cpm, molecules_counts_per_gene_19101_cpm, molecules_counts_per_gene_19239_cpm))
legend(2500, 0.003 , c("19098","19101", "19239"), lwd=c(2.5,2.5),col=c("black", "red", "green"))
```

## number of genes detected and total molecule counts
```{r}
## number of genes detected in each cell
anno_single$number_genes <- colSums(molecules_single > 0)

## number of total molecule counts in each cell
anno_single$total_molecule <- colSums(molecules_single) 

## plot 
ggplot(anno_single, aes(x = total_molecule, y = number_genes, col = as.factor(individual), shape = as.factor(batch))) + geom_point(size = 3, alpha = 0.5)
```

## Session information

```{r info}
sessionInfo()
```
