---
title: "materials and methods"
author: "PoYuan Tung"
date: 2015-05-19
---

**Last updated:** `r Sys.Date()`

**Code version:** `r system("git log -1 --format='%H'", intern = TRUE)`

```{r chunk-options, include=FALSE}
source("chunk-options.R")
```

This is the materials and methods section of the single cell RNA-seq. 

## Important features

1. UMIs (unique molecular identifiers) were added to each RNA transcript during the reverse transcript step, which provides the power to calcuate the absolute numbers of the transcripts. 5 random nucleotides (NNNNN) were used as UMIs. Therefore, there are 1024 different UMIs.

2. ERCC spike-ins were added duing the cell lysis step as the control.

3. C1 fluidigm system is used to collect single cell 
![Text to display when hovering over fig](figure/material-and-method.Rmd/method.png) 
## LCL

* cell line: 19239
* sequencing data:
  1. twocell: two cells were sequenced in one rapid lane (index failed)
  2. 4 individual cells: each sequenced in 1 full flowcell lane, 2 made by in-house Tn2 and the other 2 by Epicentre
  3. 96 cells: cells collected by 1 C1 plate. Due to financial limitation, only 24 indices were used. Each pool of 24 libraries were sequenced in 1 full flowcell lane. These and the 4 individual cells were sequenced on the same flowcell.

![Text to display when hovering over fig](figure/material-and-method.Rmd/LCL.png)  
  
## iPSC
* cell line: 19239, 19101, 19098
* number of replicates: 3, each collected using different passages of cells
* bulk vs single cell: each C1 plate has a bulk sample collected at the same time using the same chemicals frechly prepared each time. For the bulk, 2 libraries using 2 different indices were generated from each bulk sample (total 2*9=18). All the bulk samples were pooled as the "bulk" for sequencing. For single cell, the 96 libraries from the same C1 plates were pooled and sequenced.
* sequencing: SE 100+8. total 31 lanes are required. therefore, 4 full flowcells were used. "bulk" sample was sequenced in 1 lane of all 4 flowcells. samples were balanced as much as possible. 
* goals: ~ 3 millions good reads per cell and average 200,000 moleculaes per cell 

![Text to display when hovering over fig](figure/material-and-method.Rmd/sort.png)
![Text to display when hovering over fig](figure/material-and-method.Rmd/seq.png)

## Session information

```{r info}
sessionInfo()
```
