<!DOCTYPE html>

<html xmlns="http://www.w3.org/1999/xhtml">

<head>

<meta charset="utf-8">
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="generator" content="pandoc" />

<meta name="author" content="PoYuan Tung" />

<meta name="date" content="2015-05-19" />

<title>materials and methods</title>

<script src="libs/jquery-1.11.0/jquery.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="libs/bootstrap-3.3.1/css/united.min.css" rel="stylesheet" />
<script src="libs/bootstrap-3.3.1/js/bootstrap.min.js"></script>
<script src="libs/bootstrap-3.3.1/shim/html5shiv.min.js"></script>
<script src="libs/bootstrap-3.3.1/shim/respond.min.js"></script>

<style type="text/css">

/* padding for bootstrap navbar */
body {
  padding-top: 50px;
  padding-bottom: 40px;
}


/* offset scroll position for anchor links (for fixed navbar)  */
.section h2 {
  padding-top: 55px;
  margin-top: -55px;
}
.section h3 {
  padding-top: 55px;
  margin-top: -55px;
}



/* don't use link color in navbar */
.dropdown-menu>li>a {
  color: black;
}

/* some padding for disqus */
#disqus_thread {
  margin-top: 45px;
}

</style>

<link rel="stylesheet" href="libs/font-awesome-4.1.0/css/font-awesome.min.css"/>

<style type="text/css">code{white-space: pre;}</style>
<link rel="stylesheet"
      href="libs/highlight/textmate.css"
      type="text/css" />
<script src="libs/highlight/highlight.js"></script>
<style type="text/css">
  pre:not([class]) {
    background-color: white;
  }
</style>
<script type="text/javascript">
if (window.hljs && document.readyState && document.readyState === "complete") {
   window.setTimeout(function() {
      hljs.initHighlighting();
   }, 0);
}
</script>



</head>

<body>

<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
code {
  color: inherit;
  background-color: rgba(0, 0, 0, 0.04);
}
img { 
  max-width:100%; 
  height: auto; 
}
</style>
<div class="container-fluid main-container">


<div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">singleCellSeq</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li><a href="index.html">Home</a></li>
        <li><a href="about.html">About</a></li>
        <li><a href="license.html">License</a></li>
        <li><a href="https://github.com/jdblischak/singleCellSeq">GitHub</a></li>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<div id="header">
<h1 class="title">materials and methods</h1>
<h4 class="author"><em>PoYuan Tung</em></h4>
<h4 class="date"><em>2015-05-19</em></h4>
</div>

<div id="TOC">
<ul>
<li><a href="#important-features">Important features</a></li>
<li><a href="#lcl">LCL</a></li>
<li><a href="#ipsc">iPSC</a></li>
<li><a href="#session-information">Session information</a></li>
</ul>
</div>

<p><strong>Last updated:</strong> 2015-10-08</p>
<p><strong>Code version:</strong> 1bbbd2621b9eebf02cef69dd34b322c5207f2435</p>
<p>This is the materials and methods section of the single cell RNA-seq.</p>
<div id="important-features" class="section level2">
<h2>Important features</h2>
<ol style="list-style-type: decimal">
<li><p>UMIs (unique molecular identifiers) were added to each RNA transcript during the reverse transcript step, which provides the power to calcuate the absolute numbers of the transcripts. 5 random nucleotides (NNNNN) were used as UMIs. Therefore, there are 1024 different UMIs.</p></li>
<li><p>ERCC spike-ins were added duing the cell lysis step as the control.</p></li>
<li><p>C1 fluidigm system is used to collect single cell <img src="figure/material-and-method.Rmd/method.png" alt="Text to display when hovering over fig" /></p></li>
</ol>
</div>
<div id="lcl" class="section level2">
<h2>LCL</h2>
<ul>
<li>cell line: 19239</li>
<li>sequencing data:</li>
</ul>
<ol style="list-style-type: decimal">
<li>twocell: two cells were sequenced in one rapid lane (index failed)</li>
<li>4 individual cells: each sequenced in 1 full flowcell lane, 2 made by in-house Tn2 and the other 2 by Epicentre</li>
<li>96 cells: cells collected by 1 C1 plate. Due to financial limitation, only 24 indices were used. Each pool of 24 libraries were sequenced in 1 full flowcell lane. These and the 4 individual cells were sequenced on the same flowcell.</li>
</ol>
<p><img src="figure/material-and-method.Rmd/LCL.png" alt="Text to display when hovering over fig" /></p>
</div>
<div id="ipsc" class="section level2">
<h2>iPSC</h2>
<ul>
<li>cell line: 19239, 19101, 19098</li>
<li>number of replicates: 3, each collected using different passages of cells</li>
<li>bulk vs single cell: each C1 plate has a bulk sample collected at the same time using the same chemicals frechly prepared each time. For the bulk, 2 libraries using 2 different indices were generated from each bulk sample (total 2*9=18). All the bulk samples were pooled as the “bulk” for sequencing. For single cell, the 96 libraries from the same C1 plates were pooled and sequenced.</li>
<li>sequencing: SE 100+8. total 31 lanes are required. therefore, 4 full flowcells were used. “bulk” sample was sequenced in 1 lane of all 4 flowcells. samples were balanced as much as possible.</li>
<li>goals: ~ 3 millions good reads per cell and average 200,000 moleculaes per cell</li>
</ul>
<p><img src="figure/material-and-method.Rmd/sort.png" alt="Text to display when hovering over fig" /> <img src="figure/material-and-method.Rmd/seq.png" alt="Text to display when hovering over fig" /></p>
</div>
<div id="session-information" class="section level2">
<h2>Session information</h2>
<pre class="r"><code>sessionInfo()</code></pre>
<pre><code>R version 3.2.0 (2015-04-16)
Platform: x86_64-unknown-linux-gnu (64-bit)

locale:
 [1] LC_CTYPE=en_US.UTF-8       LC_NUMERIC=C              
 [3] LC_TIME=en_US.UTF-8        LC_COLLATE=en_US.UTF-8    
 [5] LC_MONETARY=en_US.UTF-8    LC_MESSAGES=en_US.UTF-8   
 [7] LC_PAPER=en_US.UTF-8       LC_NAME=C                 
 [9] LC_ADDRESS=C               LC_TELEPHONE=C            
[11] LC_MEASUREMENT=en_US.UTF-8 LC_IDENTIFICATION=C       

attached base packages:
[1] stats     graphics  grDevices utils     datasets  methods   base     

other attached packages:
[1] knitr_1.10.5

loaded via a namespace (and not attached):
 [1] httr_0.6.1      magrittr_1.5    formatR_1.2     htmltools_0.2.6
 [5] tools_3.2.0     RCurl_1.95-4.6  yaml_2.1.13     rmarkdown_0.6.1
 [9] stringi_0.4-1   digest_0.6.8    stringr_1.0.0   bitops_1.0-6   
[13] evaluate_0.7   </code></pre>
</div>


<!-- some extra javascript for older browsers -->
<script type="text/javascript" src="libs/polyfill.js"></script>

<script>

// manage active state of menu based on current page
$(document).ready(function () {

    // active menu
    href = window.location.pathname
    href = href.substr(href.lastIndexOf('/') + 1)
    $('a[href="' + href + '"]').parent().addClass('active');

    // manage active menu header
    if (href.startsWith('authoring_'))
      $('a[href="' + 'authoring' + '"]').parent().addClass('active');
    else if (href.endsWith('_format.html'))
      $('a[href="' + 'formats' + '"]').parent().addClass('active');
    else if (href.startsWith('developer_'))
      $('a[href="' + 'developer' + '"]').parent().addClass('active');

});

</script>

</div>

<script>

// add bootstrap table styles to pandoc tables
$(document).ready(function () {
  $('tr.header').parent('thead').parent('table').addClass('table table-condensed');
});

</script>


</body>
</html>
