---
title: "Identification of noisy genes"
author: "Po-Yuan Tung"
date: 2015-06-03
---

**Last updated:** `r Sys.Date()`

**Code version:** `r system("git log -1 --format='%H'", intern = TRUE)`

```{r chunk-options, include=FALSE}
source("chunk-options.R")
opts_chunk$set(fig.width = 8, fig.height = 8)
```

## Input

```{r packages, message=FALSE}
library("dplyr")
library("ggplot2")
theme_set(theme_bw(base_size = 16))
library("gplots")
library("edgeR")
```

Summary counts from featureCounts.
Created with [gather-summary-counts.py](https://github.com/jdblischak/singleCellSeq/blob/master/code/gather-summary-counts.py).

```{r input-summary-counts}
summary_counts <- read.table("../data/summary-counts.txt", header = TRUE,
                             stringsAsFactors = FALSE)
summary_per_sample <- summary_counts %>%
  filter(sickle == "quality-trimmed") %>%
  select(-sickle) %>%
  arrange(individual, batch, well, rmdup) %>%
  as.data.frame
```

Input annotation.

```{r input-annotation}
anno <- read.table("../data/annotation.txt", header = TRUE,
                   stringsAsFactors = FALSE)
head(anno)
```

Input read counts.

```{r input-read-counts}
reads <- read.table("../data/reads.txt", header = TRUE,
                    stringsAsFactors = FALSE)
```

Input molecule counts.

```{r input-molecule-counts}
molecules <- read.table("../data/molecules.txt", header = TRUE,
                    stringsAsFactors = FALSE)
```

Input single cell observational quality control data.

```{r input-qc}
qc <- read.table("../data/qc-ipsc.txt", header = TRUE,
                 stringsAsFactors = FALSE)
head(qc)
```

Input list of single cells to keep based on qc.
```{r input-goodcell}
goodcell <- read.table("../data/quality-single-cells.txt", header = TRUE,
                 stringsAsFactors = FALSE)
head(goodcell)
```


## Remove bad quality cells
### Remove cells with total reads < 2 millons
```{r remove}
#reads per sample
summary_per_sample_reads <- summary_per_sample %>% filter(rmdup == "reads")

#create unmapped ratios
summary_per_sample_reads$unmapped.ratios <- summary_per_sample_reads[,9]/apply(summary_per_sample_reads[,5:13],1,sum)

#create total mapped reads
summary_per_sample_reads$total.mapped <- apply(summary_per_sample_reads[,5:8],1,sum)

#creat ERCC ratios
summary_per_sample_reads$ERCC.ratios <- apply(reads[grep("ERCC", rownames(reads)), ],2,sum)/apply(summary_per_sample_reads[,5:8],1,sum)

#remove bulk keep single cell
summary_per_sample_reads_single <- summary_per_sample_reads[summary_per_sample_reads$well!="bulk",]

#add cell number per well by merging qc file
summary_per_sample_reads_single_qc <- merge(summary_per_sample_reads_single,qc,by=c("individual","batch","well"))

#qc filter
summary_per_sample_reads_single_qc$qc_filter <- summary_per_sample_reads_single_qc$cell_number == 1 & summary_per_sample_reads_single_qc$total.mapped > 2 * 10^6

sum(summary_per_sample_reads_single_qc$qc_filter)

ggplot(summary_per_sample_reads_single_qc, aes(x = total.mapped , y = unmapped.ratios, col = qc_filter)) + geom_text(aes(label = cell_number))

ggplot(summary_per_sample_reads_single_qc, aes(x = total.mapped , y = ERCC.ratios, col = qc_filter)) + geom_text(aes(label = cell_number))

```

##  Total molecule number of ERCC

```{r ERCC}
# molecules per sample
summary_per_sample_molecules <- summary_per_sample %>% filter(rmdup == "molecules")

# total ERCC molecule
summary_per_sample_molecules$total.ERCC.mol <- apply(molecules[grep("ERCC", rownames(reads)), ],2,sum)

# ERCC molecule ratio
summary_per_sample_molecules$ERCC.ratio.mol <- summary_per_sample_molecules$total.ERCC.mol/summary_per_sample_molecules$Assigned

# remove bulk keep single cell
summary_per_sample_molecules_single <- summary_per_sample_molecules[summary_per_sample_molecules$well!="bulk",]

plot(summary_per_sample_molecules_single$total.ERCC.mol)

# adjust total ERCC molecules of 19098 batch2
summary_per_sample_molecules_single$index_19098_2 <- (summary_per_sample_molecules_single$individual == "19098" & summary_per_sample_molecules_single$batch == "2")

# calculating the ratio of 19098 batch 2 to the rest
adjusted_ratio.mol <- mean(summary_per_sample_molecules_single$total.ERCC.mol[summary_per_sample_molecules_single$index_19098_2])/mean(summary_per_sample_molecules_single$total.ERCC[!summary_per_sample_molecules_single$index_19098_2])

adjusted_ratio.mol 

# adjusted total ERCC reads
summary_per_sample_molecules_single$adj.total.ERCC.mol <- summary_per_sample_molecules_single$total.ERCC.mol  

summary_per_sample_molecules_single$adj.total.ERCC.mol[summary_per_sample_molecules_single$index_19098_2] <- summary_per_sample_molecules_single$adj.total.ERCC.mol[summary_per_sample_molecules_single$index_19098_2]/adjusted_ratio.mol

# adjusted ERCC ratios
summary_per_sample_molecules_single$adj.ERCC.ratios.mol <-
summary_per_sample_molecules_single$adj.total.ERCC.mol/summary_per_sample_molecules_single$Assigned

# add qc filter and cell number
summary_per_sample_molecules_single$qc_filter <- summary_per_sample_reads_single_qc$qc_filter
summary_per_sample_molecules_single$cell_number <- summary_per_sample_reads_single_qc$cell_number

ggplot(summary_per_sample_molecules_single, aes(x = Assigned, y = ERCC.ratio.mol, col = qc_filter)) + geom_text(aes(label = cell_number)) + facet_grid(individual ~ batch) + theme(axis.text.x = element_text(angle = 90, hjust = 0.9, vjust = 0.5))

ggplot(summary_per_sample_molecules_single, aes(x = Assigned, y = adj.ERCC.ratios.mol, col = qc_filter)) + geom_text(aes(label = cell_number)) + facet_grid(individual ~ batch) + theme(axis.text.x = element_text(angle = 90, hjust = 0.9, vjust = 0.5))
```

## CV and mean

### Looking at molecule
```{r CV-mean-molecule}
# remove molecules that are larger than 1024
rownames(molecules)[rowMeans(molecules) > 1024]
molecules.new <- molecules [rowMeans(molecules) < 1024,]
dim(molecules)
dim(molecules.new)

# remove bulk
molecules_single <- molecules %>% select(-contains("bulk"))

# remove 1024 and greater
molecules_single <- molecules_single[apply(molecules_single,1,max) < 1024,]


# apply qc_filter
molecules_single_qc <- molecules_single[,summary_per_sample_reads_single_qc$qc_filter]
dim(molecules_single_qc)

sample_name <- names(molecules_single_qc)
	

### create a function to compute the cv and mean of molecules
### input data is molecules_single_qc
### two parameters  
### 1. filter: sellect for specifc individaul or batch
### 2. adj.19098.batch: flag to control if we want to adj the 19098 batch2 molecules numberes. default to not adj, meaning using the raw numbers
prep_molecules.cv.mean <- function(filter,adj.19098.batch2=0){
   ### generate the data of interest 
   
   data.in <- molecules_single_qc[,grepl(filter,sample_name)]
  
	if(adj.19098.batch2 == 1){
		#### find out which columns belong 19098 batch2
		target.column <- sample_name[grep("19098.2",sample_name)]
		
		#### find out ERCC rows
		g <- rownames(data.in)
		target.row <- g[grep("ERCC",g)]
		
		#### replace the molecules numbers via dividing by adjusted_ratio.mol 
		data.in[target.row,target.column] <- (data.in[target.row,target.column])/adjusted_ratio.mol
		
	}
  
  if(adj.19098.batch2 == 2){
		#### find out which columns belong 19098 batch2
		#### remove 19098.2
		data.in<- data.in[,!grepl("19098.2",sample_name)]
		
		#### also need to take care of summary_per_sample_reads_single_qc
		summary_per_sample_reads_single_qc <- summary_per_sample_reads_single_qc[!((summary_per_sample_reads_single_qc$individual==19098)&(summary_per_sample_reads_single_qc$batch==2)),]
		
  }
	
  	if(adj.19098.batch2 == 3){
		#### find out which columns belong 19098 batch2
		target.column <- sample_name[grep("19098.2",sample_name)]
		
		#### find out ERCC rows
		g <- rownames(data.in)
		target.row <- g[grep("ERCC",g)]
		
		#### replace the molecules numbers via dividing by NA 
		data.in[target.row,target.column] <- NA
		
	}
	
	#correct for collision probability
	molecules.crt <- -1024*log(1-data.in/1024)

	# create a new dataset
	molecules_single_qc_w_mean_cv <- molecules.crt

	# add mean
	molecules_single_qc_w_mean_cv$mean <- apply(molecules.crt, 1, function(x) mean(x,na.rm=TRUE) )
	
	# add CV
	molecules_single_qc_w_mean_cv$CV <- apply(molecules.crt, 1, function(x) sd(x,na.rm=TRUE) )/ apply(molecules.crt, 1, function(x) mean(x,na.rm=TRUE))

	# add variance
	molecules_single_qc_w_mean_cv$var <- apply(molecules.crt, 1, function(x) var(x,na.rm=TRUE) )
	
	# remove non-expressed 
	molecules_single_qc_expressed <- molecules_single_qc_w_mean_cv[molecules_single_qc_w_mean_cv$mean >0,]
	dim(molecules_single_qc_expressed)

  # create a flag to ERCC
	molecules_single_qc_expressed$ERCC <- grepl("ERCC",rownames(molecules_single_qc_expressed))
	
	# add gene_name 
  molecules_single_qc_expressed$gene_name <- rownames(molecules_single_qc_expressed)

	return(molecules_single_qc_expressed)
###  end of prep.molecules.cv.mean function	
}


molecules_single_qc_expressed 		<- prep_molecules.cv.mean(filter="19",adj.19098.batch2=0)
molecules_single_qc_expressed_adj		<- prep_molecules.cv.mean(filter="19",adj.19098.batch2=1)
molecules_single_qc_expressed_rm 		<- prep_molecules.cv.mean(filter="19",adj.19098.batch2=2)
molecules_single_qc_expressed_rm_ERCC 		<- prep_molecules.cv.mean(filter="19",adj.19098.batch2=3)

# plot with color-blind-friendly palettes
cbPalette <- c("#999999", "#0000FF", "#56B4E9", "#009E73", "#F0E442", "#0072B2", "#D55E00", "#CC79A7")

ggplot(molecules_single_qc_expressed, aes(x = mean, y = CV, col = ERCC))  + geom_point(size = 2, alpha = 0.5) + scale_x_log10() + scale_colour_manual(values=cbPalette)

ggplot(molecules_single_qc_expressed_rm_ERCC, aes(x = mean, y = CV, col = ERCC))  + geom_point(size = 2, alpha = 0.5) + scale_x_log10() + scale_colour_manual(values=cbPalette)

### create molecule data by each individaul using the molecules_single_qc_expressed_rm_ERCC
## 19098
individual_19098_mean_CV <- prep_molecules.cv.mean(filter="19098",adj.19098.batch2 = 0)

ggplot(individual_19098_mean_CV, aes(x = mean, y = CV, col = ERCC))  + geom_point(size = 2, alpha = 0.5) + scale_x_log10() + scale_colour_manual(values=cbPalette)

# 19098 by batch
individual_19098.1_mean_CV <- prep_molecules.cv.mean(filter="19098.1",adj.19098.batch2 = 0)
individual_19098.2_mean_CV <- prep_molecules.cv.mean(filter="19098.2",adj.19098.batch2 = 1)
individual_19098.3_mean_CV <- prep_molecules.cv.mean(filter="19098.3",adj.19098.batch2 = 0)

## 19101
individual_19101_mean_CV <- prep_molecules.cv.mean(filter="19101",adj.19098.batch2 = 0)

ggplot(individual_19101_mean_CV, aes(x = mean, y = CV, col = ERCC))  + geom_point(size = 2, alpha = 0.5) + scale_x_log10() + scale_colour_manual(values=cbPalette)

# 19101 by batch
individual_19101.1_mean_CV <- prep_molecules.cv.mean(filter="19101.1",adj.19098.batch2 = 0)
individual_19101.2_mean_CV <- prep_molecules.cv.mean(filter="19101.2",adj.19098.batch2 = 0)
individual_19101.3_mean_CV <- prep_molecules.cv.mean(filter="19101.3",adj.19098.batch2 = 0)

## 19239
individual_19239_mean_CV <- prep_molecules.cv.mean(filter="19239",adj.19098.batch2 = 0)

ggplot(individual_19239_mean_CV, aes(x = mean, y = CV, col = ERCC))  + geom_point(size = 2, alpha = 0.5) + scale_x_log10() + scale_colour_manual(values=cbPalette)

# 19239 by batch
individual_19239.1_mean_CV <- prep_molecules.cv.mean(filter="19239.1",adj.19098.batch2 = 0)
individual_19239.2_mean_CV <- prep_molecules.cv.mean(filter="19239.2",adj.19098.batch2 = 0)
individual_19239.3_mean_CV <- prep_molecules.cv.mean(filter="19239.3",adj.19098.batch2 = 0)

```



### Looking at reads
```{r CV-mean-reads}
# remove bulk
reads_single <- reads %>% select(-contains("bulk"))

# apply qc_filter
reads_single_qc <- reads_single[,summary_per_sample_reads_single_qc$qc_filter]
dim(reads_single_qc)
sum(summary_per_sample_reads_single_qc$qc_filter)

# normalization
reads_single_qc_cpm <- cpm(reads_single_qc)

# create a new dataset
reads_single_qc_w_mean_cv <- data.frame(reads_single_qc_cpm)
sum(reads_single_qc_cpm!=reads_single_qc_w_mean_cv)

# add mean
reads_single_qc_w_mean_cv$mean <- apply(reads_single_qc_cpm, 1, mean)

# add CV
reads_single_qc_w_mean_cv$CV <- apply(reads_single_qc_cpm, 1, sd)/ apply(reads_single_qc_cpm, 1, mean)

# remove non-expressed 
reads_single_qc_expressed <- reads_single_qc_w_mean_cv[reads_single_qc_w_mean_cv$mean >0,]
dim(reads_single_qc_expressed)

# sellect ERCC
reads_single_qc_expressed$ERCC <- grepl("ERCC",rownames(reads_single_qc_expressed))

# plot with color-blind-friendly palettes
cbPalette <- c("#999999", "#0000FF", "#56B4E9", "#009E73", "#F0E442", "#0072B2", "#D55E00", "#CC79A7")

ggplot(reads_single_qc_expressed, aes(x = mean, y = CV, col = ERCC))  + geom_point(size = 2, alpha = 0.5) + scale_x_log10() + scale_colour_manual(values=cbPalette)

```


## Poisson sucks!
```{r poisson}
### this function will plot the mean vs cv based on the ERCC molecules counts
### only need to specify the input dataset
### the inpute dataset needs to have mean, cv and ERCC flag
### make minipar global
plot.cv.and.mean <- function(data.in){
	# model based on ERCC only
	# need to have a ERCC flag on the data.in
	molecules_single_qc_expressed_ERCC <- data.in[data.in$ERCC,]	
	
	# defnine poisson function on a log x scale
	poisson.c <- function (x) {
		(10^x)^(0.5)/(10^x)
	}

	# compute the lossy factor based on ERCC
	####   use LS: first define the function of f, then find the minimum
	####   dont use the points from ERCC.mol.mean < 0.1 to fit. 

	ERCC.mol.mean <- molecules_single_qc_expressed_ERCC$mean
	ERCC.mol.CV   <- molecules_single_qc_expressed_ERCC$CV

	# compute the sum of square errors
	target.fun <- function(f){
		sum((ERCC.mol.CV[ERCC.mol.mean>0.1]- sqrt(1/(f*ERCC.mol.mean[ERCC.mol.mean>0.1])))^2)
	}
	# find out the minimum
	ans <- nlminb(0.05,target.fun,lower=0.0000001,upper=1)
	minipar <- ans$par
	# use the minimum to create the lossy poisson
	lossy.posson <- function (x) {
		1/sqrt((10^x)*minipar)
	}

	# 4 s.d. 
	four.sd <- function (x) {
		4*(10^x)^(0.5)/(10^x)
	}

	# 3.7 sd + 0.3
	three.sd <- function (x) {
		3.7*(10^x)^(0.5)/(10^(x))+0.3
	}

	ggplot(data.in, aes(x = mean, y = CV, col = ERCC))  + geom_point(size = 2, alpha = 0.5) +  stat_function(fun= poisson.c, col= "#CC79A7") + stat_function(fun= four.sd, col= "#F0E442") + stat_function(fun= lossy.posson, col= "#56B4E9") + scale_x_log10() + ylim(0, max(data.in$CV)*1.1) + scale_colour_manual(values=cbPalette) + xlab("Average number of molecules") + ylab ("coefficient of variation (CV)")

}
plot.cv.and.mean(data.in=molecules_single_qc_expressed)
plot.cv.and.mean(data.in=molecules_single_qc_expressed_adj)
plot.cv.and.mean(data.in=molecules_single_qc_expressed_rm)
plot.cv.and.mean(data.in=molecules_single_qc_expressed_rm_ERCC)

plot.cv.and.mean(data.in=individual_19098_mean_CV)
plot.cv.and.mean(data.in=individual_19101_mean_CV)
plot.cv.and.mean(data.in=individual_19239_mean_CV)


### ignore the following code
 ignore <- function(xxx){
	# log plot
  plot(log(molecules_single_qc_expressed$mean,base=2),log(molecules_single_qc_expressed$CV,base=2), col= "#999999", xlab="log2  Average number of molecules",ylab= "log2 coefficient of variation",ylim=c(-2,5),xlim=c(-10,10),pch=20)
  points(log(molecules_single_qc_expressed_ERCC$mean,base=2),log(molecules_single_qc_expressed_ERCC$CV,base=2),col= "#0000FF" ,pch=20)
  # add lossy poison
  curve(-0.5*x-0.5*log(minipar,base=2),-100,6,add=TRUE,col="#56B4E9")
  # add poisson
  curve(-0.5*x,add=TRUE,col= "#CC79A7")
}
```

## noisy genes
```{r noisy-gene}
### this function will identify the noisy gene based on 3.7 sd
### only need to specify the input dataset
### the inpute dataset needs to have mean and CV
noisy_gene <- function(data.in){

 #  larger than 4 sd
 count.index  <- (!is.na(data.in$mean))&(data.in$mean>1)
 condi.index <- (data.in$CV > 4*(data.in$mean^(0.5))/data.in$mean)

 sum(count.index&condi.index)

 rownames(molecules_single_qc_expressed)[count.index&condi.index]
}

# noisy genes of all
noisy_gene_all <- noisy_gene(data.in=molecules_single_qc_expressed)

# noisy genes of each individaul
noisy_gene_19098 <- noisy_gene(data.in = individual_19098_mean_CV)
noisy_gene_19101 <- noisy_gene(data.in = individual_19101_mean_CV)
noisy_gene_19239 <- noisy_gene(data.in = individual_19239_mean_CV)

```

## variation between individuals
```{r var-individual}
# overall variation is the sum of inter- and intra- individual variation
# creat a table with all the variation
table_variance <- molecules_single_qc_expressed[,c("gene_name","mean","var","ERCC")]
names(table_variance) <- c("gene_name","mean_all","variance_all","ERCC")

merge_variance <- function(data.base,data.merge,post.name){
  data.merge <- data.merge[,c("gene_name","mean","var")]
  names(data.merge) <- c("gene_name",paste(c("mean","var"),post.name,sep=""))
  merge(data.base,data.merge,by="gene_name",all.x=TRUE)
}
table_variance <- merge_variance(data.base=table_variance,data.merge=individual_19098_mean_CV,post.name="_19098")
table_variance <- merge_variance(data.base=table_variance,data.merge=individual_19101_mean_CV,post.name="_19101")
table_variance <- merge_variance(data.base=table_variance,data.merge=individual_19239_mean_CV,post.name="_19239")

# keep non-missing across the table
table_variance <- table_variance[apply(table_variance,1,function(x)  sum(is.na(x)))==0,]

# number of cell
number.of.cell.all   <- sum(grepl("19",sample_name))
number.of.cell.19098 <- sum(grepl("19098",sample_name))
number.of.cell.19101 <- sum(grepl("19101",sample_name))
number.of.cell.19239 <- sum(grepl("19239",sample_name))

# compute inter individual variance
table_variance$between_indi_var <- (table_variance$variance_all*(number.of.cell.all   -1)  - 
 table_variance$var_19098   *(number.of.cell.19098 -1)  -      
 table_variance$var_19101   *(number.of.cell.19101 -1)  - 
 table_variance$var_19239   *(number.of.cell.19239 -1)  ) / 
 (number.of.cell.all -1)

# plot overall variance vs inter-individaul variance
plot(table_variance$variance_all,table_variance$between_indi_var)

# plot ratio of inter-individual variance
plot(table_variance$between_indi_var/table_variance$variance_all)
abline(h=0.5,col=2)

# identify genes that are noisy across all cells and also with a certain level of inter-individual variance 
whatever_list <- table_variance[(table_variance$between_indi_var/table_variance$variance_all) > 0.35,][,1]

whatever_list[whatever_list %in% noisy_gene_all]

```

## variance of batches by molecules
```{r variance}
# re create table using the molecule counts
group_list <- c("19","19098","19101","19239","19098.1","19098.2","19098.3","19101.1","19101.2","19101.3","19239.1","19239.2","19239.3")

### create a function to generate tables of mean,
create_info_table <- function(statistics="mean"){
  # first correct
  molecules.crt <- -1024*log(1-molecules_single_qc/1024)
  # loop all possible combinations
  big_table <- do.call(cbind,lapply(group_list,function(x){
      # subset data
      data_ref <- molecules.crt[grep(x,names(molecules.crt))]
      if(statistics=="mean"){
        ans <- apply(data_ref,1,function(xx) mean(xx,na.rm=TRUE))  
      }
      if(statistics=="var"){
        ans <- apply(data_ref,1,function(xx) var(xx,na.rm=TRUE))  
      }
      if(statistics=="CV"){
        ans <- apply(data_ref,1,function(xx)  sd(xx,na.rm=TRUE)) / apply(data_ref,1,function(xx)  mean(xx,na.rm=TRUE))   
      }
    ans
    })
  )
    big_table <- data.frame(big_table)
    names(big_table) <- paste(statistics,group_list,sep="_")
    big_table$gene_name <- rownames(molecules_single_qc)
    big_table
}

# create big table for mean, var and cv
table_mean <- create_info_table(statistics="mean")
table_var <- create_info_table(statistics="var")
table_cv <- create_info_table(statistics="CV")

# replace 0 as NA
table_mean[table_mean==0] <- NA
table_var[table_mean==0] <- NA
table_cv[table_mean==0] <- NA

# calculate the correlation
cor(table_mean[,2:13],use="pairwise.complete.obs")
cor(table_var[,2:13],use="pairwise.complete.obs")
cor(table_cv[,2:13],use="pairwise.complete.obs")

heatmap.2(cor(table_mean[,5:13],use="pairwise.complete.obs"), trace="none",cexRow=1,cexCol=1,margins=c(8,8))
heatmap.2(cor(table_var[,2:13],use="pairwise.complete.obs"), trace="none",cexRow=1,cexCol=1,margins=c(8,8))
heatmap.2(cor(table_cv[,2:13],use="pairwise.complete.obs"), trace="none",cexRow=1,cexCol=1,margins=c(8,8))

# create a table for boxplot of CV correlation 
corr_CV <- cor(table_cv[,2:13],use="pairwise.complete.obs")

corr_boxplot <- data.frame(
correlation=c(corr_CV[4,5:6],corr_CV[7,8:9],corr_CV[10,11:12],corr_CV[4:6,7:12],corr_CV[7:9,10:12]), cor_source=c(rep("within",6),rep("between",27)), individual=c(rep("19098",2),rep("19101",2),rep("19239",2),rep("between",27)))

# t test
t_test <- t.test(corr_boxplot[1:6,1],corr_boxplot[7:33,1],alternative = "greater")
t_test

ggplot(corr_boxplot, aes(cor_source,correlation)) + geom_boxplot() + geom_jitter(aes(colour = individual, size = 2, width = 0.5)) 
```

## Session information

```{r info}
sessionInfo()
```
