---
title: "Ordering effect of the capture sites"
author: "Joyce Hsiao"
date: 2015-09-09
---

**Last updated:** `r Sys.Date()`

**Code version:** `r system("git log -1 --format='%H'", intern = TRUE)`

```{r chunk-options, include=FALSE}
source("chunk-options.R")
```

## All genes, cells

We investigated potential ordering effect of wells on each place on gene
expression across the 9 plates (batches). [Leng](http://dx.doi.org/10.1038/nmeth.3549) et al. discussed ordering effect 
in single-cell RNA-seq experiments using Fluidigm C1 and implemented an algorithm that detects ordering effect of wells on gene expression in [OEFinder](http://dx.doi.org/10.1101/025437).

Note that OEFinder relies on a series of permutations. We had to run OEFinder on a cluster-based computing system.


## Setup
```{r}
source("functions.R")
library(edgeR)

# Packages required to start OEFinder
# library(shiny)
# library(gdata)
# library(shinyFiles)
# library(EBSeq)
```



## Prepare single cell data before filtering

Input annotation

```{r}
anno <- read.table("../data/annotation.txt", header = TRUE,
                   stringsAsFactors = FALSE)
head(anno)
```

Input read counts.

```{r}
reads <- read.table("../data/reads.txt", header = TRUE,
                    stringsAsFactors = FALSE)
```

Input molecule counts

```{r}
molecules <- read.table("../data/molecules.txt", header = TRUE, stringsAsFactors = FALSE)
```

Remove bulk samples

```{r}
single_samples <- anno$well != "bulk"
anno_single <- anno[ which(single_samples), ]
molecules_single <- molecules[ , which(single_samples)]
reads_single <- reads[ , which(single_samples)]
stopifnot(ncol(molecules_single) == nrow(anno_single),
          colnames(molecules_single) == anno_single$sample_id)
```


Output single cell samples to txt files.

```{r, eval = FALSE}
if (!file.exists("../data/molecules-single.txt")) {
  write.table(molecules_single, 
              file = "../data/molecules-single.txt",
              quote = FALSE,
              col.names = TRUE, row.names = TRUE)
}

if (!file.exists("../data/reads-single.txt")) {
  write.table(reads_single, 
              file = "../data/reads-single.txt",
              quote = FALSE,
              col.names = TRUE, row.names = TRUE)
}
```


Prepare capture site identification file. A txt file with one column of capture site ID (A, B, C, ..., H).

```{r}
require(stringr)
capture_site <- str_extract(anno_single$well, "[aA-zZ]+")
table(capture_site)
```


Save capture_site to a txt file.

```{r, eval = FALSE}
if (!file.exists("../data/capture-site.txt")) {
write.table(data.frame(site = capture_site), 
            file = "../data/capture-site.txt",
            quote = FALSE,
            col.names = FALSE, row.names = FALSE)
}
```


### Molecule count data

For unnormalized data, OEFinder defaults the DESeq normalization method. 

Upload *molecules.txt* and *capture-site.txt* to OEFinder Shiny GUI interface.

Output to *singleCellSeq/data/OEFinder*.


```{r}
# runGitHub("OEFinder", "lengning")

OE_raw <- read.csv("../data/OEFinder/all-genes-OEgenes.csv",
                     stringsAsFactors = FALSE,
                     quote = "\"", sep = ",", header = TRUE)
colnames(OE_raw) <- c("genes", "pvalue")
head(OE_raw)
str(OE_raw)
```


2 ERCC genes in the Overexpressed genes

```{r}
grep("ERCC", OE_raw$genes)
```


Create an indicator variable for the OE genes

```{r}
oefinder_raw <- rownames(molecules_single) %in% as.character(OE_raw$genes)
table(oefinder_raw)
```

Summarize individual, well, and batch information for the OE genes.

```{r}
anno_single_oe_raw <- anno_single[which(oefinder_raw), ]

with(anno_single_oe_raw, table(well, batch, individual))
```






## Session information

```{r info}
sessionInfo()
```
