---
title: "Potential batch effect in the ordering effect of the capture sites"
author: "Joyce Hsiao"
date: 2015-09-26
---

**Last updated:** `r Sys.Date()`

**Code version:** `r system("git log -1 --format='%H'", intern = TRUE)`

```{r chunk-options, include=FALSE}
source("chunk-options.R")
```

## Goal

We investigated potential ordering effect of wells on each place on gene
expression for each plate. [Leng](http://dx.doi.org/10.1038/nmeth.3549) et al. discussed ordering effect in single-cell RNA-seq experiments using Fluidigm C1 and implemented an algorithm that detects ordering effect of wells on gene expression in [OEFinder](http://dx.doi.org/10.1101/025437).

We inputted CPM-normalized data to OEFinder for one batch at a time.



## Setup

```{r}
source("functions.R")
library(edgeR)
library(ggplot2)
theme_set(theme_bw(base_size = 16))
```



## Prepare single cell data before filtering

Input annotation

```{r}
anno <- read.table("../data/annotation.txt", header = TRUE,
                   stringsAsFactors = FALSE)
head(anno)
```


Input read counts.

```{r}
reads <- read.table("../data/reads.txt", header = TRUE,
                    stringsAsFactors = FALSE)
```

Input molecule counts

```{r}
molecules <- read.table("../data/molecules.txt", header = TRUE, stringsAsFactors = FALSE)
```

Remove bulk samples

```{r}
single_samples <- anno$well != "bulk"
anno_single <- anno[ which(single_samples), ]
molecules_single <- molecules[ , which(single_samples)]
reads_single <- reads[ , which(single_samples)]
stopifnot(ncol(molecules_single) == nrow(anno_single),
          colnames(molecules_single) == anno_single$sample_id)
```

Remove ERCC genes.

```{r}
ii_nonERCC <- grep("ERCC", rownames(molecules_single), invert = TRUE)

molecules_single_ENSG <- molecules_single[ii_nonERCC, ]
```


Remove genes with no counts and also overly expressed genes.

```{r}
## remove gene with 0 counts
expressed_single_ENSG <- rowSums(molecules_single_ENSG) > 0
molecules_single_ENSG <- molecules_single[expressed_single_ENSG, ]
dim(molecules_single_ENSG)

## remove gene with molecule count larger than 1024 (15 if them)
overexpressed_genes <- rownames(molecules_single_ENSG)[apply(molecules_single_ENSG, 1,
                                                        function(x) any(x >= 1024))]
molecules_single_ENSG <- molecules_single_ENSG[!(rownames(molecules_single_ENSG) %in% overexpressed_genes), ]

## collision probability and cpm molecule counts
molecules_single_collision <- -1024 * log(1 - molecules_single_ENSG / 1024)
molecules_single_cpm <- cpm(molecules_single_collision, log = TRUE)
```


Output single cell CPM to txt files.

```{r, eval = FALSE}
if (!file.exists("../data/molecules-single-cpm.txt")) {
  write.table(molecules_single_cpm, 
              file = "../data/molecules-single-cpm.txt",
              quote = FALSE,
              col.names = TRUE, row.names = TRUE)
}
```


Prepare capture site identification file. A txt file with one column of capture site ID (A, B, C, ..., H).

```{r}
require(stringr)
capture_site <- str_extract(anno_single$well, "[aA-zZ]+")
table(capture_site)
```


Save capture_site to a txt file.

```{r}
if (!file.exists("../data/capture-site.txt")) {
write.table(data.frame(site = capture_site), 
            file = "../data/capture-site.txt",
            quote = FALSE,
            col.names = FALSE, row.names = FALSE)
}
```


## OEFinder

Upload *molecules-single-cpm.txt* and *capture-site.txt* to OEFinder Shiny GUI interface.

Output to *singleCellSeq/data/OEFinder*.


* Run OEFinder

```{r, eval = FALSE}
# Packages required to start OEFinder
library(shiny)
library(gdata)
library(shinyFiles)
library(EBSeq)

runGitHub("OEFinder", "lengning")
```

Load OEFinder outputted genes.

```{r}
OE_cpm <- read.csv("../data/OEFinder/cpm-OEgenes.csv",
                     stringsAsFactors = FALSE,
                     quote = "\"", sep = ",", header = TRUE)
colnames(OE_cpm) <- c("genes", "pvalue")
head(OE_cpm)
str(OE_cpm)
```


7 ERCC genes in the Overexpressed genes

```{r}
grep("ERCC", OE_cpm$genes)
```


## Distribution of OE genes

```{r}
cutoffs <- seq(1001, nrow(molecules_single_cpm), by = 1000)
cutoffs <- c(cutoffs, nrow(molecules_single_cpm))
top_genes_count <- lapply(1:length(cutoffs), function(cut) {
                        per_cutoff <- cutoffs[cut]
                        cell_across_order <- order(rowSums(molecules_single_cpm), decreasing = TRUE)
                        top_genes <- rownames(molecules_single_cpm)[cell_across_order < per_cutoff]
                        sum(OE_cpm$genes %in% top_genes)
                        })
top_genes_count <- do.call(c, top_genes_count)

ggplot(data.frame(top_count = top_genes_count,
                  cutoffs = cutoffs), 
       aes(x = as.factor(cutoffs), y = top_count)) + geom_point() +
       labs(x = "Top X genes", y = "Number of OEFinder OE genes")
```

OE genes identified by OEFinder were not limited to the top 1000 genes. On the contrary, we found
OE genes at all levels of gene expression (averaged acrosss cells).



## Gene annotation

```{r}
library(mygene)
ii_nonERCC <- grep("ERCC", OE_cpm$genes, invert = TRUE)

gene_query <- queryMany(OE_cpm$genes[ii_nonERCC], scopes="ensembl.gene", 
                 fields=c("name", "summary"), species="human")

kable(as.data.frame(gene_query)[ , 1:4])
```


## Session information

```{r info}
sessionInfo()
```
