<!DOCTYPE html>

<html xmlns="http://www.w3.org/1999/xhtml">

<head>

<meta charset="utf-8">
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="generator" content="pandoc" />

<meta name="author" content="Joyce Hsiao" />

<meta name="date" content="2015-09-26" />

<title>Potential batch effect in the ordering effect of the capture sites</title>

<script src="libs/jquery-1.11.0/jquery.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="libs/bootstrap-3.3.1/css/united.min.css" rel="stylesheet" />
<script src="libs/bootstrap-3.3.1/js/bootstrap.min.js"></script>
<script src="libs/bootstrap-3.3.1/shim/html5shiv.min.js"></script>
<script src="libs/bootstrap-3.3.1/shim/respond.min.js"></script>

<style type="text/css">

/* padding for bootstrap navbar */
body {
  padding-top: 50px;
  padding-bottom: 40px;
}


/* offset scroll position for anchor links (for fixed navbar)  */
.section h2 {
  padding-top: 55px;
  margin-top: -55px;
}
.section h3 {
  padding-top: 55px;
  margin-top: -55px;
}



/* don't use link color in navbar */
.dropdown-menu>li>a {
  color: black;
}

/* some padding for disqus */
#disqus_thread {
  margin-top: 45px;
}

</style>

<link rel="stylesheet" href="libs/font-awesome-4.1.0/css/font-awesome.min.css"/>

<style type="text/css">code{white-space: pre;}</style>
<link rel="stylesheet"
      href="libs/highlight/textmate.css"
      type="text/css" />
<script src="libs/highlight/highlight.js"></script>
<style type="text/css">
  pre:not([class]) {
    background-color: white;
  }
</style>
<script type="text/javascript">
if (window.hljs && document.readyState && document.readyState === "complete") {
   window.setTimeout(function() {
      hljs.initHighlighting();
   }, 0);
}
</script>



</head>

<body>

<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
code {
  color: inherit;
  background-color: rgba(0, 0, 0, 0.04);
}
img { 
  max-width:100%; 
  height: auto; 
}
</style>
<div class="container-fluid main-container">


<div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">singleCellSeq</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li><a href="index.html">Home</a></li>
        <li><a href="about.html">About</a></li>
        <li><a href="license.html">License</a></li>
        <li><a href="https://github.com/jdblischak/singleCellSeq">GitHub</a></li>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<div id="header">
<h1 class="title">Potential batch effect in the ordering effect of the capture sites</h1>
<h4 class="author"><em>Joyce Hsiao</em></h4>
<h4 class="date"><em>2015-09-26</em></h4>
</div>

<div id="TOC">
<ul>
<li><a href="#goal">Goal</a></li>
<li><a href="#setup">Setup</a></li>
<li><a href="#prepare-single-cell-data-before-filtering">Prepare single cell data before filtering</a></li>
<li><a href="#prepare-oefinder-input">Prepare OEFinder input</a><ul>
<li><a href="#per-batch-per-individual">Per batch, per individual</a></li>
<li><a href="#per-individual">Per individual</a></li>
</ul></li>
<li><a href="#oefinder">OEFinder</a></li>
<li><a href="#session-information">Session information</a></li>
</ul>
</div>

<p><strong>Last updated:</strong> 2015-09-28</p>
<p><strong>Code version:</strong> 3f442a23f405d3564d9ab3197ec337df22fe9383</p>
<div id="goal" class="section level2">
<h2>Goal</h2>
<p>We investigated potential ordering effect of wells on each place on gene expression for each plate. <a href="http://dx.doi.org/10.1038/nmeth.3549">Leng</a> et al. discussed ordering effect in single-cell RNA-seq experiments using Fluidigm C1 and implemented an algorithm that detects ordering effect of wells on gene expression in <a href="http://dx.doi.org/10.1101/025437">OEFinder</a>.</p>
<p>First, we applied OEFinder to each batch (within each individual) on the CPM-normalized data. The OEFinder returned a warning saying “essentially perfect fit: summary may be unreliable.”</p>
<p>We then applied OEFinder to one individual at a time, and the same error message returned…</p>
</div>
<div id="setup" class="section level2">
<h2>Setup</h2>
<pre class="r"><code>source(&quot;functions.R&quot;)
library(edgeR)</code></pre>
<pre><code>Loading required package: limma</code></pre>
<pre class="r"><code>library(ggplot2)
theme_set(theme_bw(base_size = 16))</code></pre>
</div>
<div id="prepare-single-cell-data-before-filtering" class="section level2">
<h2>Prepare single cell data before filtering</h2>
<p>Input annotation</p>
<pre class="r"><code>anno &lt;- read.table(&quot;../data/annotation.txt&quot;, header = TRUE,
                   stringsAsFactors = FALSE)
head(anno)</code></pre>
<pre><code>  individual batch well     sample_id
1      19098     1  A01 NA19098.1.A01
2      19098     1  A02 NA19098.1.A02
3      19098     1  A03 NA19098.1.A03
4      19098     1  A04 NA19098.1.A04
5      19098     1  A05 NA19098.1.A05
6      19098     1  A06 NA19098.1.A06</code></pre>
<p>Input read counts.</p>
<pre class="r"><code>reads &lt;- read.table(&quot;../data/reads.txt&quot;, header = TRUE,
                    stringsAsFactors = FALSE)</code></pre>
<p>Input molecule counts</p>
<pre class="r"><code>molecules &lt;- read.table(&quot;../data/molecules.txt&quot;, header = TRUE, stringsAsFactors = FALSE)</code></pre>
<p>Remove bulk samples</p>
<pre class="r"><code>single_samples &lt;- anno$well != &quot;bulk&quot;
anno_single &lt;- anno[ which(single_samples), ]
molecules_single &lt;- molecules[ , which(single_samples)]
reads_single &lt;- reads[ , which(single_samples)]
stopifnot(ncol(molecules_single) == nrow(anno_single),
          colnames(molecules_single) == anno_single$sample_id)</code></pre>
<p>Remove ERCC genes.</p>
<pre class="r"><code>ii_nonERCC &lt;- grep(&quot;ERCC&quot;, rownames(molecules_single), invert = TRUE)

molecules_single_ENSG &lt;- molecules_single[ii_nonERCC, ]</code></pre>
<p>Remove genes with no counts and also overly expressed genes.</p>
<pre class="r"><code>## remove gene with 0 counts
expressed_single_ENSG &lt;- rowSums(molecules_single_ENSG) &gt; 0
molecules_single_ENSG &lt;- molecules_single[expressed_single_ENSG, ]
dim(molecules_single_ENSG)</code></pre>
<pre><code>[1] 17769   864</code></pre>
<pre class="r"><code>## remove gene with molecule count larger than 1024 (15 if them)
overexpressed_genes &lt;- rownames(molecules_single_ENSG)[apply(molecules_single_ENSG, 1,
                                                        function(x) any(x &gt;= 1024))]
molecules_single_ENSG &lt;- molecules_single_ENSG[!(rownames(molecules_single_ENSG) %in% overexpressed_genes), ]

## collision probability and cpm molecule counts
molecules_single_collision &lt;- -1024 * log(1 - molecules_single_ENSG / 1024)
molecules_single_cpm &lt;- cpm(molecules_single_collision, log = TRUE)</code></pre>
</div>
<div id="prepare-oefinder-input" class="section level2">
<h2>Prepare OEFinder input</h2>
<div id="per-batch-per-individual" class="section level3">
<h3>Per batch, per individual</h3>
<p>Output single cell CPM to txt files.</p>
<pre class="r"><code>anno_single$unique_batch &lt;- with(anno_single, paste(individual, batch, sep = &quot;_&quot;))
for (per_batch in 1:length(unique(anno_single$unique_batch))) {
  ii_batch &lt;- anno_single$unique_batch == unique(anno_single$unique_batch)[per_batch]
  write.table(molecules_single_cpm[which(ii_batch), ], 
              file = paste(&quot;../data/molecules-single-cpm&quot;, &quot;-&quot;, 
                           unique(anno_single$unique_batch)[per_batch], &quot;.txt&quot;, sep = &quot;&quot;),
              quote = FALSE,
              col.names = TRUE, row.names = TRUE)
}</code></pre>
<p>Prepare capture site identification file. A txt file with one column of capture site ID (A, B, C, …, H).</p>
<pre class="r"><code>require(stringr)
capture_site &lt;- str_extract(anno_single$well, &quot;[aA-zZ]+&quot;)
table(capture_site)

# Save capture_site to a txt file.
for (per_batch in 1:length(unique(anno_single$unique_batch))) {
  ii_batch &lt;- anno_single$unique_batch == unique(anno_single$unique_batch)[per_batch]
  write.table(data.frame(site = capture_site[ii_batch]), 
              file = paste(&quot;../data/capture-site-&quot;,
                           unique(anno_single$unique_batch)[per_batch], &quot;.txt&quot;, sep = &quot;&quot;),
              quote = FALSE,
              col.names = FALSE, row.names = FALSE)
}</code></pre>
</div>
<div id="per-individual" class="section level3">
<h3>Per individual</h3>
<p>Output single cell CPM to txt files.</p>
<pre class="r"><code>for (per_person in 1:length(unique(anno_single$individual))) {
  ii_person &lt;- anno_single$individual == unique(anno_single$individual)[per_person]
  write.table(molecules_single_cpm[which(ii_person), ], 
              file = paste(&quot;../data/molecules-single-cpm&quot;, &quot;-&quot;, 
                           unique(anno_single$individual)[per_person], &quot;.txt&quot;, sep = &quot;&quot;),
              quote = FALSE,
              col.names = TRUE, row.names = TRUE)
}</code></pre>
<p>Prepare capture site identification file. A txt file with one column of capture site ID (A, B, C, …, H).</p>
<pre class="r"><code>require(stringr)
capture_site &lt;- str_extract(anno_single$well, &quot;[aA-zZ]+&quot;)
table(capture_site)

# Save capture_site to a txt file.
for (per_person in 1:length(unique(anno_single$individual))) {
  ii_person &lt;- anno_single$individual == unique(anno_single$individual)[per_person]
  write.table(data.frame(site = capture_site[ii_person]), 
              file = paste(&quot;../data/capture-site-&quot;,
                           unique(anno_single$individual)[per_person], &quot;.txt&quot;, sep = &quot;&quot;),
              quote = FALSE,
              col.names = FALSE, row.names = FALSE)
}</code></pre>
</div>
</div>
<div id="oefinder" class="section level2">
<h2>OEFinder</h2>
<p>Upload <em>molecules-single-cpm-(batch).txt</em> and <em>capture-site-(batch).txt</em> to OEFinder Shiny GUI interface.</p>
<p>Output to <em>singleCellSeq/data/OEFinder</em>.</p>
<ul>
<li>Run OEFinder</li>
</ul>
<pre class="r"><code># Packages required to start OEFinder
library(shiny)
library(gdata)
library(shinyFiles)
library(EBSeq)

runGitHub(&quot;OEFinder&quot;, &quot;lengning&quot;)</code></pre>
</div>
<div id="session-information" class="section level2">
<h2>Session information</h2>
<pre class="r"><code>sessionInfo()</code></pre>
<pre><code>R version 3.2.0 (2015-04-16)
Platform: x86_64-unknown-linux-gnu (64-bit)

locale:
 [1] LC_CTYPE=en_US.UTF-8       LC_NUMERIC=C              
 [3] LC_TIME=en_US.UTF-8        LC_COLLATE=en_US.UTF-8    
 [5] LC_MONETARY=en_US.UTF-8    LC_MESSAGES=en_US.UTF-8   
 [7] LC_PAPER=en_US.UTF-8       LC_NAME=C                 
 [9] LC_ADDRESS=C               LC_TELEPHONE=C            
[11] LC_MEASUREMENT=en_US.UTF-8 LC_IDENTIFICATION=C       

attached base packages:
[1] stats     graphics  grDevices utils     datasets  methods   base     

other attached packages:
[1] ggplot2_1.0.1 edgeR_3.10.2  limma_3.24.9  knitr_1.10.5 

loaded via a namespace (and not attached):
 [1] Rcpp_0.12.0      magrittr_1.5     MASS_7.3-40      munsell_0.4.2   
 [5] colorspace_1.2-6 stringr_1.0.0    httr_0.6.1       plyr_1.8.3      
 [9] tools_3.2.0      grid_3.2.0       gtable_0.1.2     htmltools_0.2.6 
[13] yaml_2.1.13      digest_0.6.8     reshape2_1.4.1   formatR_1.2     
[17] bitops_1.0-6     RCurl_1.95-4.6   evaluate_0.7     rmarkdown_0.6.1 
[21] stringi_0.4-1    scales_0.2.4     proto_0.3-10    </code></pre>
</div>


<!-- some extra javascript for older browsers -->
<script type="text/javascript" src="libs/polyfill.js"></script>

<script>

// manage active state of menu based on current page
$(document).ready(function () {

    // active menu
    href = window.location.pathname
    href = href.substr(href.lastIndexOf('/') + 1)
    $('a[href="' + href + '"]').parent().addClass('active');

    // manage active menu header
    if (href.startsWith('authoring_'))
      $('a[href="' + 'authoring' + '"]').parent().addClass('active');
    else if (href.endsWith('_format.html'))
      $('a[href="' + 'formats' + '"]').parent().addClass('active');
    else if (href.startsWith('developer_'))
      $('a[href="' + 'developer' + '"]').parent().addClass('active');

});

</script>

</div>

<script>

// add bootstrap table styles to pandoc tables
$(document).ready(function () {
  $('tr.header').parent('thead').parent('table').addClass('table table-condensed');
});

</script>


</body>
</html>
