<!DOCTYPE html>

<html xmlns="http://www.w3.org/1999/xhtml">

<head>

<meta charset="utf-8">
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="generator" content="pandoc" />


<meta name="date" content="2016-04-27" />

<title>Additional analysis for the manuscript</title>

<script src="libs/jquery-1.11.0/jquery.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="libs/bootstrap-3.3.1/css/united.min.css" rel="stylesheet" />
<script src="libs/bootstrap-3.3.1/js/bootstrap.min.js"></script>
<script src="libs/bootstrap-3.3.1/shim/html5shiv.min.js"></script>
<script src="libs/bootstrap-3.3.1/shim/respond.min.js"></script>

<style type="text/css">

/* padding for bootstrap navbar */
body {
  padding-top: 50px;
  padding-bottom: 40px;
}


/* offset scroll position for anchor links (for fixed navbar)  */
.section h2 {
  padding-top: 55px;
  margin-top: -55px;
}
.section h3 {
  padding-top: 55px;
  margin-top: -55px;
}



/* don't use link color in navbar */
.dropdown-menu>li>a {
  color: black;
}

/* some padding for disqus */
#disqus_thread {
  margin-top: 45px;
}

</style>

<link rel="stylesheet" href="libs/font-awesome-4.1.0/css/font-awesome.min.css"/>

<style type="text/css">code{white-space: pre;}</style>
<link rel="stylesheet"
      href="libs/highlight/textmate.css"
      type="text/css" />
<script src="libs/highlight/highlight.js"></script>
<style type="text/css">
  pre:not([class]) {
    background-color: white;
  }
</style>
<script type="text/javascript">
if (window.hljs && document.readyState && document.readyState === "complete") {
   window.setTimeout(function() {
      hljs.initHighlighting();
   }, 0);
}
</script>



</head>

<body>

<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
code {
  color: inherit;
  background-color: rgba(0, 0, 0, 0.04);
}
img { 
  max-width:100%; 
  height: auto; 
}
</style>
<div class="container-fluid main-container">


<div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">singleCellSeq</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li><a href="index.html">Home</a></li>
        <li><a href="about.html">About</a></li>
        <li><a href="license.html">License</a></li>
        <li><a href="https://github.com/jdblischak/singleCellSeq">GitHub</a></li>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<div id="header">
<h1 class="title">Additional analysis for the manuscript</h1>
<h4 class="date"><em>2016-04-27</em></h4>
</div>

<div id="TOC">
<ul>
<li><a href="#basic-framework">Basic framework</a></li>
<li><a href="#questions">Questions</a></li>
<li><a href="#input">Input</a></li>
<li><a href="#concentration-vs.total-molecule-count-ensg">Concentration vs. total molecule-count (ENSG)</a></li>
<li><a href="#reads-to-molecule-conversion-efficiency">Reads to molecule conversion efficiency</a></li>
<li><a href="#total-ensg-molecule-count-and-total-ercc-molecule-count">total ENSG molecule-count and total ERCC molecule-count</a></li>
<li><a href="#percent-variation-per-gene">Percent variation per gene</a></li>
<li><a href="#correlation-between-and-within-batches">Correlation between and within batches</a></li>
<li><a href="#session-information">Session information</a></li>
</ul>
</div>

<p><strong>Last updated:</strong> 2016-04-29</p>
<p><strong>Code version:</strong> 1e1b562e686b0c0a4bfc4c813f05de32b5077b40</p>
<pre class="r"><code>library(&quot;dplyr&quot;)
library(&quot;ggplot2&quot;)
library(&quot;cowplot&quot;)
library(&quot;lmtest&quot;)
library(&quot;lme4&quot;)
source(&quot;functions.R&quot;)</code></pre>
<div id="basic-framework" class="section level2">
<h2>Basic framework</h2>
</div>
<div id="questions" class="section level2">
<h2>Questions</h2>
<ol style="list-style-type: decimal">
<li>Concentration vs. total molecule-count (ENSG)</li>
</ol>
<p>(Our study design allows us to investigate…)</p>
<p>total molecule-count ~ Concentration</p>
<p><a href="http://jdblischak.github.io/singleCellSeq/analysis/compare-reads-v-molecules-per-batch-test.html">previous work</a></p>
<ol start="2" style="list-style-type: decimal">
<li>Molecule-to-read conversion rate (ENSG, ERCC)</li>
</ol>
<p>(We explored…)</p>
<p>total molecule-count ~ read</p>
<ol start="3" style="list-style-type: decimal">
<li>total ERCC molecule-count and total ENSG molecule-count</li>
</ol>
<p>(Could we account…)</p>
<p>total ENSG molecule-count ~ total ERCC molecule-count</p>
<ol start="4" style="list-style-type: decimal">
<li>Percent variation explained by individual and replicate effect in ENSG and ERCC</li>
</ol>
<p>(As a first step…)</p>
<p><a href="http://jdblischak.github.io/singleCellSeq/analysis/compare-reads-v-molecules-per-batch-test.html">previous work</a></p>
</div>
<div id="input" class="section level2">
<h2>Input</h2>
<p>Input filtered annotation.</p>
<pre class="r"><code>anno_filter &lt;- read.table(&quot;../data/annotation-filter.txt&quot;, header = TRUE,
                   stringsAsFactors = FALSE)
head(anno_filter)</code></pre>
<pre><code>  individual replicate well      batch      sample_id
1    NA19098        r1  A01 NA19098.r1 NA19098.r1.A01
2    NA19098        r1  A02 NA19098.r1 NA19098.r1.A02
3    NA19098        r1  A04 NA19098.r1 NA19098.r1.A04
4    NA19098        r1  A05 NA19098.r1 NA19098.r1.A05
5    NA19098        r1  A06 NA19098.r1 NA19098.r1.A06
6    NA19098        r1  A07 NA19098.r1 NA19098.r1.A07</code></pre>
<p>Input filtered molecule counts.</p>
<pre class="r"><code>molecules_filter &lt;- read.table(&quot;../data/molecules-filter.txt&quot;, header = TRUE,
                               stringsAsFactors = FALSE)
molecules_filter_ENSG &lt;- molecules_filter[grep(&quot;ERCC&quot;, rownames(molecules_filter), invert = TRUE), ]
  
stopifnot(ncol(molecules_filter) == nrow(anno_filter),
          colnames(molecules_filter) == anno_filter$sample_id)</code></pre>
<p>Input filtered read counts</p>
<pre class="r"><code>reads_filter &lt;- read.table(&quot;../data/reads-filter.txt&quot;, header = TRUE,
                    stringsAsFactors = FALSE)
reads_filter_ENSG &lt;- reads_filter[grep(&quot;ERCC&quot;, rownames(reads_filter), invert = TRUE), ]

stopifnot(all.equal(colnames(reads_filter_ENSG), 
                    colnames(molecules_filter_ENSG)))</code></pre>
<p>Input quality control file. Filter cells to match cells in molecules_filter.</p>
<pre class="r"><code>qc &lt;- read.table(&quot;../data/qc-ipsc.txt&quot;, header = TRUE,
                 stringsAsFactors = FALSE)
qc$sample_id &lt;- with(qc, paste0(individual, &quot;.&quot;, replicate, &quot;.&quot;, well))

qc_filter &lt;- qc[match(anno_filter$sample_id, qc$sample_id), ]

stopifnot(all.equal(qc_filter$sample_id, anno_filter$sample_id))</code></pre>
<p>Input standardized molecule counts.</p>
<pre class="r"><code>molecules_cpm &lt;- read.table(&quot;../data/molecules-cpm.txt&quot;, header = TRUE,
                            stringsAsFactors = FALSE)
stopifnot(ncol(molecules_cpm) == nrow(anno_filter),
          colnames(molecules_cpm) == anno_filter$sample_id)</code></pre>
<p>Input Poisson GLM transformed molecule counts per million.</p>
<pre class="r"><code>molecules_cpm_trans &lt;- read.table(&quot;../data/molecules-cpm-trans.txt&quot;, header = TRUE,
                               stringsAsFactors = FALSE)
stopifnot(ncol(molecules_cpm_trans) == nrow(anno_filter),
          colnames(molecules_cpm_trans) == anno_filter$sample_id)</code></pre>
<p>Input final batch-corrected molecule counts per million.</p>
<pre class="r"><code>molecules_final &lt;- read.table(&quot;../data/molecules-final.txt&quot;, header = TRUE,
                              stringsAsFactors = FALSE)
stopifnot(ncol(molecules_final) == nrow(anno_filter),
          colnames(molecules_final) == anno_filter$sample_id)</code></pre>
</div>
<div id="concentration-vs.total-molecule-count-ensg" class="section level2">
<h2>Concentration vs. total molecule-count (ENSG)</h2>
<p>As we try to understand the general relationships between sequencing results and cellular mRNA content, we remove outlier batches. NA19098 replicate 1 failed the quantification of the concentration of the single cells and was hence removed. Because NA19098 concentration is only quantified in one replicate, we removed NA19098 from analysis involving batch differences and concentration.</p>
<pre class="r"><code>anno_single &lt;- anno_filter
ercc_index &lt;- grepl(&quot;ERCC&quot;, rownames(molecules_filter))
anno_single$total_molecules_gene = colSums(molecules_filter[!ercc_index, ])
anno_single$total_molecules_ercc = colSums(molecules_filter[ercc_index, ])
anno_single$total_molecules = colSums(molecules_filter)
anno_single$num_genes = apply(molecules_filter[!ercc_index, ], 2, function(x) sum(x &gt; 0))
anno_single$concentration &lt;- qc_filter$concentration[match(anno_single$sample_id, qc_filter$sample_id)]

anno_single &lt;- anno_single %&gt;% filter(individual != &quot;NA19098&quot;)
anno_single$individual &lt;- as.factor(anno_single$individual)
anno_single$replicate &lt;- as.factor(anno_single$replicate)</code></pre>
<pre class="r"><code>fit &lt;- lmer(total_molecules_gene ~ concentration + individual + 
              (1|individual:replicate), 
            data = anno_single)
fit_1 &lt;- lm(total_molecules_gene ~ concentration + individual, 
            data = anno_single)
fit_2 &lt;- lmer(total_molecules_gene ~ concentration + (1|individual:replicate), 
            data = anno_single)

# significance of individual effect
lrtest(fit_2, fit)</code></pre>
<pre><code>Likelihood ratio test

Model 1: total_molecules_gene ~ concentration + (1 | individual:replicate)
Model 2: total_molecules_gene ~ concentration + individual + (1 | individual:replicate)
  #Df  LogLik Df  Chisq Pr(&gt;Chisq)    
1   4 -4537.6                         
2   5 -4526.1  1 23.086  1.549e-06 ***
---
Signif. codes:  0 &#39;***&#39; 0.001 &#39;**&#39; 0.01 &#39;*&#39; 0.05 &#39;.&#39; 0.1 &#39; &#39; 1</code></pre>
<pre class="r"><code># significance of replicate effect
anova(fit, fit_1)</code></pre>
<pre><code>refitting model(s) with ML (instead of REML)</code></pre>
<pre><code>Data: anno_single
Models:
fit_1: total_molecules_gene ~ concentration + individual
fit: total_molecules_gene ~ concentration + individual + (1 | individual:replicate)
      Df    AIC    BIC  logLik deviance  Chisq Chi Df Pr(&gt;Chisq)   
fit_1  4 9118.3 9134.4 -4555.1   9110.3                            
fit    5 9112.0 9132.2 -4551.0   9102.0 8.2633      1   0.004045 **
---
Signif. codes:  0 &#39;***&#39; 0.001 &#39;**&#39; 0.01 &#39;*&#39; 0.05 &#39;.&#39; 0.1 &#39; &#39; 1</code></pre>
</div>
<div id="reads-to-molecule-conversion-efficiency" class="section level2">
<h2>Reads to molecule conversion efficiency</h2>
<p>Prepare ERCC data</p>
<pre class="r"><code>reads_ERCC &lt;- reads_filter[grep(&quot;ERCC&quot;, rownames(reads_filter), 
                                invert = FALSE), ]
molecules_ERCC &lt;- molecules_filter[grep(&quot;ERCC&quot;, rownames(molecules_filter), 
                                invert = FALSE), ]

total_counts_ERCC &lt;- data.frame(total_reads = colSums(reads_ERCC),
                                total_molecules = colSums(molecules_ERCC))
total_counts_ERCC$conversion &lt;- with(total_counts_ERCC,
                                     total_molecules/total_reads)
total_counts_ERCC$individual &lt;- as.factor(anno_filter$individual[match(rownames(total_counts_ERCC),
                          anno_filter$sample_id)])
total_counts_ERCC$replicate &lt;- as.factor(anno_filter$replicate[match(rownames(total_counts_ERCC),
                          anno_filter$sample_id)])</code></pre>
<p>Prepare ENSG data</p>
<pre class="r"><code>reads_ENSG &lt;- reads_filter[grep(&quot;ERCC&quot;, rownames(reads_filter), 
                                invert = TRUE), ]
molecules_ENSG &lt;- molecules_filter[grep(&quot;ERCC&quot;, rownames(molecules_filter), 
                                invert = TRUE), ]

total_counts_ENSG &lt;- data.frame(total_reads = colSums(reads_ENSG),
                                total_molecules = colSums(molecules_ENSG))
total_counts_ENSG$conversion &lt;- with(total_counts_ENSG,
                                     total_molecules/total_reads)
total_counts_ENSG$individual &lt;- as.factor(anno_filter$individual[match(rownames(total_counts_ENSG),
                          anno_filter$sample_id)])
total_counts_ENSG$replicate &lt;- as.factor(anno_filter$replicate[match(rownames(total_counts_ENSG),
                          anno_filter$sample_id)])</code></pre>
<p>ENSG</p>
<pre class="r"><code>fit &lt;- lmer(log2(total_molecules) ~ log2(total_reads) + individual + 
              (1|individual:replicate), 
            data = total_counts_ENSG)
fit_1 &lt;- lm(log2(total_molecules) ~ log2(total_reads) + individual,
            data = total_counts_ENSG)
fit_2 &lt;- lmer(log2(total_molecules) ~ log2(total_reads) + 
              (1|individual:replicate), 
            data = total_counts_ENSG)

# significance of individual effect
lrtest(fit_2, fit)</code></pre>
<pre><code>Likelihood ratio test

Model 1: log2(total_molecules) ~ log2(total_reads) + (1 | individual:replicate)
Model 2: log2(total_molecules) ~ log2(total_reads) + individual + (1 | 
    individual:replicate)
  #Df LogLik Df Chisq Pr(&gt;Chisq)   
1   4 290.08                       
2   6 295.46  2 10.76   0.004608 **
---
Signif. codes:  0 &#39;***&#39; 0.001 &#39;**&#39; 0.01 &#39;*&#39; 0.05 &#39;.&#39; 0.1 &#39; &#39; 1</code></pre>
<pre class="r"><code># significance of replicate effect
anova(fit, fit_1)</code></pre>
<pre><code>refitting model(s) with ML (instead of REML)</code></pre>
<pre><code>Data: total_counts_ENSG
Models:
fit_1: log2(total_molecules) ~ log2(total_reads) + individual
fit: log2(total_molecules) ~ log2(total_reads) + individual + (1 | 
fit:     individual:replicate)
      Df     AIC     BIC logLik deviance  Chisq Chi Df Pr(&gt;Chisq)    
fit_1  5 -548.13 -526.45 279.06  -558.13                             
fit    6 -599.64 -573.63 305.82  -611.64 53.512      1   2.57e-13 ***
---
Signif. codes:  0 &#39;***&#39; 0.001 &#39;**&#39; 0.01 &#39;*&#39; 0.05 &#39;.&#39; 0.1 &#39; &#39; 1</code></pre>
<p>ERCC</p>
<pre class="r"><code>fit &lt;- lmer(log2(total_molecules) ~ log2(total_reads) + individual + 
              (1|individual:replicate), 
            data = total_counts_ERCC)
fit_1 &lt;- lm(log2(total_molecules) ~ log2(total_reads) + individual,
            data = total_counts_ERCC)
fit_2 &lt;- lmer(log2(total_molecules) ~ log2(total_reads) + 
              (1|individual:replicate), 
            data = total_counts_ERCC)

# significance of individual effect
lrtest(fit_2, fit)</code></pre>
<pre><code>Likelihood ratio test

Model 1: log2(total_molecules) ~ log2(total_reads) + (1 | individual:replicate)
Model 2: log2(total_molecules) ~ log2(total_reads) + individual + (1 | 
    individual:replicate)
  #Df LogLik Df  Chisq Pr(&gt;Chisq)
1   4 275.21                     
2   6 277.14  2 3.8665     0.1447</code></pre>
<pre class="r"><code># significance of replicate effect
anova(fit, fit_1)</code></pre>
<pre><code>refitting model(s) with ML (instead of REML)</code></pre>
<pre><code>Data: total_counts_ERCC
Models:
fit_1: log2(total_molecules) ~ log2(total_reads) + individual
fit: log2(total_molecules) ~ log2(total_reads) + individual + (1 | 
fit:     individual:replicate)
      Df     AIC     BIC logLik deviance  Chisq Chi Df Pr(&gt;Chisq)    
fit_1  5 -223.60 -201.92 116.80  -233.60                             
fit    6 -556.77 -530.76 284.38  -568.77 335.17      1  &lt; 2.2e-16 ***
---
Signif. codes:  0 &#39;***&#39; 0.001 &#39;**&#39; 0.01 &#39;*&#39; 0.05 &#39;.&#39; 0.1 &#39; &#39; 1</code></pre>
</div>
<div id="total-ensg-molecule-count-and-total-ercc-molecule-count" class="section level2">
<h2>total ENSG molecule-count and total ERCC molecule-count</h2>
<p>Prepare data</p>
<pre class="r"><code>anno_temp &lt;- anno_filter

anno_temp$ensg_total_count &lt;- 
  colSums(molecules_filter[grep(&quot;ERCC&quot;, 
                           rownames(molecules_filter), invert = FALSE), ])
anno_temp$ercc_total_count &lt;- 
  colSums(molecules_filter[grep(&quot;ERCC&quot;, 
                           rownames(molecules_filter), invert = TRUE), ])</code></pre>
<p>First, we assess total ERCC molecule-count beteween individuals and replicates.</p>
<pre class="r"><code>fit &lt;- lmer(log2(ercc_total_count) ~ individual + 
              (1|individual:replicate), 
            data = anno_temp)
fit_1 &lt;- lm(log2(ercc_total_count) ~ individual,
            data = anno_temp)
fit_2 &lt;- lmer(log2(ercc_total_count) ~ 1 + 
              (1|individual:replicate), 
            data = anno_temp)

# significance of individual effect
lrtest(fit_2, fit)</code></pre>
<pre><code>Likelihood ratio test

Model 1: log2(ercc_total_count) ~ 1 + (1 | individual:replicate)
Model 2: log2(ercc_total_count) ~ individual + (1 | individual:replicate)
  #Df  LogLik Df  Chisq Pr(&gt;Chisq)  
1   3 -138.78                       
2   5 -136.23  2 5.1014    0.07803 .
---
Signif. codes:  0 &#39;***&#39; 0.001 &#39;**&#39; 0.01 &#39;*&#39; 0.05 &#39;.&#39; 0.1 &#39; &#39; 1</code></pre>
<pre class="r"><code># significance of replicate effect
anova(fit, fit_1)</code></pre>
<pre><code>refitting model(s) with ML (instead of REML)</code></pre>
<pre><code>Data: anno_temp
Models:
fit_1: log2(ercc_total_count) ~ individual
fit: log2(ercc_total_count) ~ individual + (1 | individual:replicate)
      Df    AIC    BIC  logLik deviance  Chisq Chi Df Pr(&gt;Chisq)    
fit_1  4 281.21 298.55 -136.60   273.21                             
fit    5 270.03 291.70 -130.01   260.03 13.181      1  0.0002828 ***
---
Signif. codes:  0 &#39;***&#39; 0.001 &#39;**&#39; 0.01 &#39;*&#39; 0.05 &#39;.&#39; 0.1 &#39; &#39; 1</code></pre>
<p>Second, we perform the same analysis on total ENSG molecule-counts.</p>
<pre class="r"><code>fit &lt;- lmer(log2(ensg_total_count) ~ individual + 
              (1|individual:replicate), 
            data = anno_temp)
fit_1 &lt;- lm(log2(ensg_total_count) ~ individual,
            data = anno_temp)
fit_2 &lt;- lmer(log2(ensg_total_count) ~ 1 + 
              (1|individual:replicate), 
            data = anno_temp)

# significance of individual effect
lrtest(fit_2, fit)</code></pre>
<pre><code>Likelihood ratio test

Model 1: log2(ensg_total_count) ~ 1 + (1 | individual:replicate)
Model 2: log2(ensg_total_count) ~ individual + (1 | individual:replicate)
  #Df LogLik Df  Chisq Pr(&gt;Chisq)
1   3 254.98                     
2   5 256.73  2 3.5097     0.1729</code></pre>
<pre class="r"><code># significance of replicate effect
anova(fit, fit_1)</code></pre>
<pre><code>refitting model(s) with ML (instead of REML)</code></pre>
<pre><code>Data: anno_temp
Models:
fit_1: log2(ensg_total_count) ~ individual
fit: log2(ensg_total_count) ~ individual + (1 | individual:replicate)
      Df     AIC     BIC  logLik deviance  Chisq Chi Df Pr(&gt;Chisq)    
fit_1  4  -86.97  -69.63  47.485   -94.97                             
fit    5 -510.93 -489.25 260.463  -520.93 425.96      1  &lt; 2.2e-16 ***
---
Signif. codes:  0 &#39;***&#39; 0.001 &#39;**&#39; 0.01 &#39;*&#39; 0.05 &#39;.&#39; 0.1 &#39; &#39; 1</code></pre>
<p>Second, we include total ENSG molecule-count in the model of total ERCC molecule-count in addition to individual and replicate factors.</p>
<p>ERCC ~ ENSG</p>
<pre class="r"><code>fit &lt;- lmer(log2(ercc_total_count) ~ log2(ensg_total_count) + individual + 
              (1|individual:replicate), 
            data = anno_temp)
fit_1 &lt;- lm(log2(ercc_total_count) ~ log2(ensg_total_count) + individual,
            data = anno_temp)
fit_2 &lt;- lmer(log2(ercc_total_count) ~ log2(ensg_total_count) + 
              (1|individual:replicate), 
            data = anno_temp)

# significance of individual effect
lrtest(fit_2, fit)</code></pre>
<pre><code>Likelihood ratio test

Model 1: log2(ercc_total_count) ~ log2(ensg_total_count) + (1 | individual:replicate)
Model 2: log2(ercc_total_count) ~ log2(ensg_total_count) + individual + 
    (1 | individual:replicate)
  #Df  LogLik Df  Chisq Pr(&gt;Chisq)
1   4 -66.621                     
2   6 -66.770  2 0.2977     0.8617</code></pre>
<pre class="r"><code># significance of replicate effect
anova(fit, fit_1)</code></pre>
<pre><code>refitting model(s) with ML (instead of REML)</code></pre>
<pre><code>Data: anno_temp
Models:
fit_1: log2(ercc_total_count) ~ log2(ensg_total_count) + individual
fit: log2(ercc_total_count) ~ log2(ensg_total_count) + individual + 
fit:     (1 | individual:replicate)
      Df    AIC    BIC  logLik deviance  Chisq Chi Df Pr(&gt;Chisq)    
fit_1  5 224.79 246.46 -107.39   214.79                             
fit    6 134.98 160.99  -61.49   122.98 91.806      1  &lt; 2.2e-16 ***
---
Signif. codes:  0 &#39;***&#39; 0.001 &#39;**&#39; 0.01 &#39;*&#39; 0.05 &#39;.&#39; 0.1 &#39; &#39; 1</code></pre>
</div>
<div id="percent-variation-per-gene" class="section level2">
<h2>Percent variation per gene</h2>
<pre class="r"><code>individual &lt;- as.factor(anno_filter$individual)
replicate &lt;- as.factor(anno_filter$replicate)

library(blme)
## blme, Wishart prior, change coding
blme_varcomp &lt;- lapply( 1:NROW(molecules_filter_ENSG), function(i) {
    
    value &lt;- unlist(log2(molecules_filter_ENSG[i, ]+1))
    
    fit_try &lt;- tryCatch( fit &lt;- blmer(value ~ 1|individual/replicate, 
                                    cov.prior = wishart,
                                    resid.prior = invgamma),
                           condition = function(c) c)
    if(inherits(fit_try, &quot;condition&quot;)){
    var_foo &lt;- rep(NA, 3)
    return(var_foo)
    }
    if(!inherits(fit_try, &quot;condition&quot;)){
      var_foo &lt;- as.data.frame(VarCorr(fit_try))[,4]
      var_foo &lt;- var_foo[c(2,1,3)]
      var_foo
    }
  })
blme_varcomp &lt;- do.call(rbind, blme_varcomp)
rownames(blme_varcomp) &lt;- rownames(molecules_filter_ENSG)
colnames(blme_varcomp) &lt;- c(&quot;individual&quot;,&quot;replicate&quot;,&quot;residual&quot;)
save(blme_varcomp,
     file = &quot;../data/blme-variance.rda&quot;)</code></pre>
<p>Standardardize variance components such that for each gene the variance component estimates sum up to 1. Because variance of sequencing counts depend on the dynamic range, we avoid biased comparison of variance estimates when comparing across genes, by the action of standarizing per gene variance components by the total of variance estimates.</p>
<pre class="r"><code>load(&quot;../data/blme-variance.rda&quot;)
blme_varcomp_norm &lt;- (blme_varcomp/rowSums(blme_varcomp))

par(mfrow = c(2,2))
plot(blme_varcomp_norm[,1], blme_varcomp_norm[,2],
     xlab = &quot;Individual&quot;,
     ylab = &quot;Replicate&quot;,
     main = &quot;Variance component estimate&quot;)
plot(blme_varcomp_norm[,3], blme_varcomp_norm[,1],
     xlab = &quot;Residual&quot;,
     ylab = &quot;Individual&quot;,
     main = &quot;Variance component estimate&quot;)
plot(blme_varcomp_norm[,3], blme_varcomp_norm[,2],
     xlab = &quot;Residual&quot;,
     ylab = &quot;Replicate&quot;,
     main = &quot;Variance component estimate&quot;)
boxplot(blme_varcomp_norm)</code></pre>
<p><img src="figure/paper-additional-analysis.Rmd/unnamed-chunk-13-1.png" title="" alt="" width="768" style="display: block; margin: auto;" /></p>
<p>Compare individual versus replicate variance estimates.</p>
<pre class="r"><code>summary(blme_varcomp_norm)</code></pre>
<pre><code>   individual        replicate          residual      
 Min.   :0.02494   Min.   :0.00552   Min.   :0.01755  
 1st Qu.:0.12481   1st Qu.:0.03089   1st Qu.:0.57745  
 Median :0.21108   Median :0.05178   Median :0.72380  
 Mean   :0.25177   Mean   :0.06003   Mean   :0.68819  
 3rd Qu.:0.34422   3rd Qu.:0.08011   3rd Qu.:0.82784  
 Max.   :0.94753   Max.   :0.38313   Max.   :0.96808  
 NA&#39;s   :221       NA&#39;s   :221       NA&#39;s   :221      </code></pre>
<pre class="r"><code># Kruskal wallis rank sum test to compare
# proportions of variance explained due to individual
# versus due to replicate
kruskal.test(c(blme_varcomp_norm[,1], blme_varcomp_norm[,2]) ~
               rep(c(1,2), each = NROW(blme_varcomp_norm)) )</code></pre>
<pre><code>
    Kruskal-Wallis rank sum test

data:  c(blme_varcomp_norm[, 1], blme_varcomp_norm[, 2]) by rep(c(1, 2), each = NROW(blme_varcomp_norm))
Kruskal-Wallis chi-squared = 14185, df = 1, p-value &lt; 2.2e-16</code></pre>
</div>
<div id="correlation-between-and-within-batches" class="section level2">
<h2>Correlation between and within batches</h2>
<p>Code for computing correlation between cells within batches and between batches.</p>
<pre class="r"><code>#&#39; molecules_input &lt;- molecules_filter_ENSG
#&#39; annotation &lt;- anno_filter
compute_corr_batch &lt;- function(molecules_input, annotation) {

  cor_mat &lt;- cor(molecules_input, method = &quot;spearman&quot;)
  batch &lt;- unique(annotation$batch)
  individual &lt;- unique(annotation$individual)

  # same individual, within batch
  corr_same_ind_within_batch &lt;-
    lapply(1:length(individual), function(i) {
      batch &lt;-
        unique(annotation$batch[annotation$individual == individual[i]])

      corr_batch &lt;- lapply(1:length(batch), function(i) {
        df &lt;- cor_mat[annotation$batch == batch[i],
                      annotation$batch == batch[i]]
        df[upper.tri(df, diag = FALSE)]
      })
      unlist(corr_batch)
    })

  # same individual, between replicates
  corr_same_ind_between_batch &lt;-
    lapply(1:length(individual), function(i) {
      batch &lt;-
        unique(annotation$batch[annotation$individual == individual[i]])
      submat &lt;- lapply(1:(length(batch)-1), function(i) {
        submat0 &lt;- lapply(2:length(batch), function(j) {
          df &lt;- cor_mat[annotation$batch == batch[i],
                        annotation$batch == batch[j]]
          df[upper.tri(df, diag = FALSE)]
        })
        unlist(submat0)
      })
      unlist(submat)
    })

  # different individual
  corr_diff_ind_between_batch &lt;-
    lapply(1:(length(individual)-1), function(i) {
      if (i == 1) {
        batch &lt;-
          unique(annotation$batch[annotation$individual == individual[i]])
        batch_other &lt;-
          unique(annotation$batch[annotation$individual != individual[i+1]])
      }
      if (i == 2) {
        batch &lt;-
          unique(annotation$batch[annotation$individual == individual[i]])
        batch_other &lt;-
          unique(annotation$batch[annotation$individual == individual[i+1]])
      }

      submat &lt;- lapply(1:length(batch), function(i) {
        submat0 &lt;- lapply(1:length(batch_other), function(j) {
          df &lt;- cor_mat[annotation$batch == batch[i],
                        annotation$batch == batch_other[j]]
          df[upper.tri(df, diag = FALSE)]
        })
        unlist(submat0)
      })
      unlist(submat)
    })
  corr_diff_ind_between_batch &lt;- unlist(corr_diff_ind_between_batch)

  return( list(corr_same_ind_within_batch = corr_same_ind_within_batch,
               corr_same_ind_between_batch = corr_same_ind_between_batch,
               corr_diff_ind_between_batch = corr_diff_ind_between_batch) )

}</code></pre>
<p>Compute correlation for molecule-count data after filtering.</p>
<pre class="r"><code>corr_filter &lt;- compute_corr_batch(molecules_filter_ENSG, anno_filter)
par(mfrow = c(1,1))
boxplot(cbind(corr_filter[[1]][[1]],
              corr_filter[[1]][[2]],
              corr_filter[[1]][[3]],
              corr_filter[[2]][[1]],
              corr_filter[[2]][[2]],
              corr_filter[[2]][[3]],
              corr_filter[[3]]),
        main = &quot;Counts&quot;,
        axes = F)</code></pre>
<pre><code>Warning in cbind(corr_filter[[1]][[1]], corr_filter[[1]][[2]],
corr_filter[[1]][[3]], : number of rows of result is not a multiple of
vector length (arg 1)</code></pre>
<pre class="r"><code>axis(1, at = c(1:7),
     labels = c(&quot;within-batch-19098&quot;,
                      &quot;within-batch-19101&quot;,
                      &quot;within-batch-19239&quot;,
                      &quot;between-batch-19098&quot;,
                      &quot;between-batch-19101&quot;,
                      &quot;between-batch-19239&quot;,
                      &quot;across-batches&quot;))
axis(2)</code></pre>
<p><img src="figure/paper-additional-analysis.Rmd/unnamed-chunk-15-1.png" title="" alt="" width="768" style="display: block; margin: auto;" /></p>
<p>Kruskal wallis comparing all between-batch correlations with all within-batch correlations</p>
<pre class="r"><code>df &lt;- data.frame(corrs = c(unlist(corr_filter[[1]]),
                                unlist(corr_filter[[2]])),
                        label = c(rep(1, length(unlist(corr_filter[[1]]))),
                                  rep(2, length(unlist(corr_filter[[2]]))))) 
kruskal.test(df$corrs ~ df$label)</code></pre>
<pre><code>
    Kruskal-Wallis rank sum test

data:  df$corrs by df$label
Kruskal-Wallis chi-squared = 2210.8, df = 1, p-value &lt; 2.2e-16</code></pre>
<pre class="r"><code># summary statistics of correlations within-batches
# of all three individuals
summary(unlist(corr_filter[[1]]))</code></pre>
<pre><code>   Min. 1st Qu.  Median    Mean 3rd Qu.    Max. 
 0.6058  0.7222  0.7434  0.7412  0.7618  0.8319 </code></pre>
<pre class="r"><code># summary statistics of correlations between-batches
# of all three individuals
summary(unlist(corr_filter[[2]]))</code></pre>
<pre><code>   Min. 1st Qu.  Median    Mean 3rd Qu.    Max. 
 0.6113  0.7080  0.7294  0.7263  0.7475  0.8282 </code></pre>
</div>
<div id="session-information" class="section level2">
<h2>Session information</h2>
<pre class="r"><code>sessionInfo()</code></pre>
<pre><code>R version 3.2.0 (2015-04-16)
Platform: x86_64-unknown-linux-gnu (64-bit)

locale:
 [1] LC_CTYPE=en_US.UTF-8       LC_NUMERIC=C              
 [3] LC_TIME=en_US.UTF-8        LC_COLLATE=en_US.UTF-8    
 [5] LC_MONETARY=en_US.UTF-8    LC_MESSAGES=en_US.UTF-8   
 [7] LC_PAPER=en_US.UTF-8       LC_NAME=C                 
 [9] LC_ADDRESS=C               LC_TELEPHONE=C            
[11] LC_MEASUREMENT=en_US.UTF-8 LC_IDENTIFICATION=C       

attached base packages:
[1] stats     graphics  grDevices utils     datasets  methods   base     

other attached packages:
[1] lme4_1.1-10   Matrix_1.2-1  lmtest_0.9-34 zoo_1.7-12    cowplot_0.3.1
[6] ggplot2_1.0.1 dplyr_0.4.2   knitr_1.10.5 

loaded via a namespace (and not attached):
 [1] Rcpp_0.12.0      magrittr_1.5     splines_3.2.0    MASS_7.3-40     
 [5] munsell_0.4.2    lattice_0.20-31  colorspace_1.2-6 R6_2.1.1        
 [9] minqa_1.2.4      stringr_1.0.0    httr_0.6.1       plyr_1.8.3      
[13] tools_3.2.0      parallel_3.2.0   grid_3.2.0       nlme_3.1-120    
[17] gtable_0.1.2     DBI_0.3.1        htmltools_0.2.6  lazyeval_0.1.10 
[21] yaml_2.1.13      assertthat_0.1   digest_0.6.8     nloptr_1.0.4    
[25] reshape2_1.4.1   formatR_1.2      bitops_1.0-6     RCurl_1.95-4.6  
[29] evaluate_0.7     rmarkdown_0.6.1  stringi_0.4-1    scales_0.2.4    
[33] proto_0.3-10    </code></pre>
</div>


<!-- some extra javascript for older browsers -->
<script type="text/javascript" src="libs/polyfill.js"></script>

<script>

// manage active state of menu based on current page
$(document).ready(function () {

    // active menu
    href = window.location.pathname
    href = href.substr(href.lastIndexOf('/') + 1)
    $('a[href="' + href + '"]').parent().addClass('active');

    // manage active menu header
    if (href.startsWith('authoring_'))
      $('a[href="' + 'authoring' + '"]').parent().addClass('active');
    else if (href.endsWith('_format.html'))
      $('a[href="' + 'formats' + '"]').parent().addClass('active');
    else if (href.startsWith('developer_'))
      $('a[href="' + 'developer' + '"]').parent().addClass('active');

});

</script>

</div>

<script>

// add bootstrap table styles to pandoc tables
$(document).ready(function () {
  $('tr.header').parent('thead').parent('table').addClass('table table-condensed');
});

</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://cdn.mathjax.org/mathjax/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
