---
title: "Additional analysis for the manuscript"
date: 2016-04-27
---

**Last updated:** `r Sys.Date()`

**Code version:** `r system("git log -1 --format='%H'", intern = TRUE)`

```{r chunk-options, include=FALSE}
source("chunk-options.R")
opts_chunk$set(fig.width = 8, fig.height = 8)
```


```{r packages, message=FALSE}
library("dplyr")
library("ggplot2")
library("cowplot")
library("lmtest")
library("lme4")
source("functions.R")
```

## Basic framework




## Questions

1. Concentration vs. total molecule-count (ENSG)

(Our study design allows us to investigate...)

total molecule-count ~ Concentration 

[previous work](http://jdblischak.github.io/singleCellSeq/analysis/compare-reads-v-molecules-per-batch-test.html)

2. Molecule-to-read conversion rate (ENSG, ERCC)

(We explored...)

total molecule-count ~ read

3. total ERCC molecule-count and total ENSG molecule-count

(Could we account...)

total ENSG molecule-count ~ total ERCC molecule-count

4. Percent variation explained by individual and replicate effect in ENSG and ERCC

(As a first step...)

[previous work](http://jdblischak.github.io/singleCellSeq/analysis/compare-reads-v-molecules-per-batch-test.html)


## Input

Input filtered annotation.

```{r input-annotation-filter}
anno_filter <- read.table("../data/annotation-filter.txt", header = TRUE,
                   stringsAsFactors = FALSE)
head(anno_filter)
```

Input filtered molecule counts.

```{r input-molecule-counts-filter}
molecules_filter <- read.table("../data/molecules-filter.txt", header = TRUE,
                               stringsAsFactors = FALSE)
molecules_filter_ENSG <- molecules_filter[grep("ERCC", rownames(molecules_filter), invert = TRUE), ]
  
stopifnot(ncol(molecules_filter) == nrow(anno_filter),
          colnames(molecules_filter) == anno_filter$sample_id)
```

Input filtered read counts

```{r}
reads_filter <- read.table("../data/reads-filter.txt", header = TRUE,
                    stringsAsFactors = FALSE)
reads_filter_ENSG <- reads_filter[grep("ERCC", rownames(reads_filter), invert = TRUE), ]

stopifnot(all.equal(colnames(reads_filter_ENSG), 
                    colnames(molecules_filter_ENSG)))
```

Input quality control file. Filter cells to match cells in molecules_filter.

```{r}
qc <- read.table("../data/qc-ipsc.txt", header = TRUE,
                 stringsAsFactors = FALSE)
qc$sample_id <- with(qc, paste0(individual, ".", replicate, ".", well))

qc_filter <- qc[match(anno_filter$sample_id, qc$sample_id), ]

stopifnot(all.equal(qc_filter$sample_id, anno_filter$sample_id))
```

Input standardized molecule counts.

```{r input-molecule-counts-cpm}
molecules_cpm <- read.table("../data/molecules-cpm.txt", header = TRUE,
                            stringsAsFactors = FALSE)
stopifnot(ncol(molecules_cpm) == nrow(anno_filter),
          colnames(molecules_cpm) == anno_filter$sample_id)
```

Input Poisson GLM transformed molecule counts per million.

```{r input-molecule-counts-trans}
molecules_cpm_trans <- read.table("../data/molecules-cpm-trans.txt", header = TRUE,
                               stringsAsFactors = FALSE)
stopifnot(ncol(molecules_cpm_trans) == nrow(anno_filter),
          colnames(molecules_cpm_trans) == anno_filter$sample_id)
```

Input final batch-corrected molecule counts per million.

```{r input-molecule-counts-final}
molecules_final <- read.table("../data/molecules-final.txt", header = TRUE,
                              stringsAsFactors = FALSE)
stopifnot(ncol(molecules_final) == nrow(anno_filter),
          colnames(molecules_final) == anno_filter$sample_id)
```





## Concentration vs. total molecule-count (ENSG)

As we try to understand the general relationships between sequencing results and cellular mRNA content, we remove outlier batches. NA19098 replicate 1 failed the quantification of the concentration of the single cells and was hence removed. Because NA19098 concentration is only quantified in one replicate, we removed NA19098 from analysis involving batch differences and concentration.

```{r filter-outliers}
anno_single <- anno_filter
ercc_index <- grepl("ERCC", rownames(molecules_filter))
anno_single$total_molecules_gene = colSums(molecules_filter[!ercc_index, ])
anno_single$total_molecules_ercc = colSums(molecules_filter[ercc_index, ])
anno_single$total_molecules = colSums(molecules_filter)
anno_single$num_genes = apply(molecules_filter[!ercc_index, ], 2, function(x) sum(x > 0))
anno_single$concentration <- qc_filter$concentration[match(anno_single$sample_id, qc_filter$sample_id)]

anno_single <- anno_single %>% filter(individual != "NA19098")
anno_single$individual <- as.factor(anno_single$individual)
anno_single$replicate <- as.factor(anno_single$replicate)
```


```{r}
fit <- lmer(total_molecules_gene ~ concentration + individual + 
              (1|individual:replicate), 
            data = anno_single)
fit_1 <- lm(total_molecules_gene ~ concentration + individual, 
            data = anno_single)
fit_2 <- lmer(total_molecules_gene ~ concentration + (1|individual:replicate), 
            data = anno_single)

# significance of individual effect
lrtest(fit_2, fit)

# significance of replicate effect
anova(fit, fit_1)
```



## Reads to molecule conversion efficiency

Prepare ERCC data

```{r}
reads_ERCC <- reads_filter[grep("ERCC", rownames(reads_filter), 
                                invert = FALSE), ]
molecules_ERCC <- molecules_filter[grep("ERCC", rownames(molecules_filter), 
                                invert = FALSE), ]

total_counts_ERCC <- data.frame(total_reads = colSums(reads_ERCC),
                                total_molecules = colSums(molecules_ERCC))
total_counts_ERCC$conversion <- with(total_counts_ERCC,
                                     total_molecules/total_reads)
total_counts_ERCC$individual <- as.factor(anno_filter$individual[match(rownames(total_counts_ERCC),
                          anno_filter$sample_id)])
total_counts_ERCC$replicate <- as.factor(anno_filter$replicate[match(rownames(total_counts_ERCC),
                          anno_filter$sample_id)])
```


Prepare ENSG data

```{r}
reads_ENSG <- reads_filter[grep("ERCC", rownames(reads_filter), 
                                invert = TRUE), ]
molecules_ENSG <- molecules_filter[grep("ERCC", rownames(molecules_filter), 
                                invert = TRUE), ]

total_counts_ENSG <- data.frame(total_reads = colSums(reads_ENSG),
                                total_molecules = colSums(molecules_ENSG))
total_counts_ENSG$conversion <- with(total_counts_ENSG,
                                     total_molecules/total_reads)
total_counts_ENSG$individual <- as.factor(anno_filter$individual[match(rownames(total_counts_ENSG),
                          anno_filter$sample_id)])
total_counts_ENSG$replicate <- as.factor(anno_filter$replicate[match(rownames(total_counts_ENSG),
                          anno_filter$sample_id)])
```


ENSG

```{r}
fit <- lmer(log2(total_molecules) ~ log2(total_reads) + individual + 
              (1|individual:replicate), 
            data = total_counts_ENSG)
fit_1 <- lm(log2(total_molecules) ~ log2(total_reads) + individual,
            data = total_counts_ENSG)
fit_2 <- lmer(log2(total_molecules) ~ log2(total_reads) + 
              (1|individual:replicate), 
            data = total_counts_ENSG)

# significance of individual effect
lrtest(fit_2, fit)

# significance of replicate effect
anova(fit, fit_1)
```

ERCC

```{r}
fit <- lmer(log2(total_molecules) ~ log2(total_reads) + individual + 
              (1|individual:replicate), 
            data = total_counts_ERCC)
fit_1 <- lm(log2(total_molecules) ~ log2(total_reads) + individual,
            data = total_counts_ERCC)
fit_2 <- lmer(log2(total_molecules) ~ log2(total_reads) + 
              (1|individual:replicate), 
            data = total_counts_ERCC)

# significance of individual effect
lrtest(fit_2, fit)

# significance of replicate effect
anova(fit, fit_1)
```



## total ENSG molecule-count and total ERCC molecule-count

Prepare data

```{r}
anno_temp <- anno_filter

anno_temp$ensg_total_count <- 
  colSums(molecules_filter[grep("ERCC", 
                           rownames(molecules_filter), invert = FALSE), ])
anno_temp$ercc_total_count <- 
  colSums(molecules_filter[grep("ERCC", 
                           rownames(molecules_filter), invert = TRUE), ])
```

First, we assess total ERCC molecule-count beteween individuals and replicates.

```{r}
fit <- lmer(log2(ercc_total_count) ~ individual + 
              (1|individual:replicate), 
            data = anno_temp)
fit_1 <- lm(log2(ercc_total_count) ~ individual,
            data = anno_temp)
fit_2 <- lmer(log2(ercc_total_count) ~ 1 + 
              (1|individual:replicate), 
            data = anno_temp)

# significance of individual effect
lrtest(fit_2, fit)

# significance of replicate effect
anova(fit, fit_1)
```


Second, we perform the same analysis on total ENSG molecule-counts.


```{r}
fit <- lmer(log2(ensg_total_count) ~ individual + 
              (1|individual:replicate), 
            data = anno_temp)
fit_1 <- lm(log2(ensg_total_count) ~ individual,
            data = anno_temp)
fit_2 <- lmer(log2(ensg_total_count) ~ 1 + 
              (1|individual:replicate), 
            data = anno_temp)

# significance of individual effect
lrtest(fit_2, fit)

# significance of replicate effect
anova(fit, fit_1)
```


Second, we include total ENSG molecule-count in the model of total ERCC molecule-count in addition to individual and replicate factors.

ERCC ~ ENSG

```{r}
fit <- lmer(log2(ercc_total_count) ~ log2(ensg_total_count) + individual + 
              (1|individual:replicate), 
            data = anno_temp)
fit_1 <- lm(log2(ercc_total_count) ~ log2(ensg_total_count) + individual,
            data = anno_temp)
fit_2 <- lmer(log2(ercc_total_count) ~ log2(ensg_total_count) + 
              (1|individual:replicate), 
            data = anno_temp)

# significance of individual effect
lrtest(fit_2, fit)

# significance of replicate effect
anova(fit, fit_1)
```





## Percent variation per gene


```{r, eval = FALSE}
individual <- as.factor(anno_filter$individual)
replicate <- as.factor(anno_filter$replicate)

library(blme)
## blme, Wishart prior, change coding
blme_varcomp <- lapply( 1:NROW(molecules_filter_ENSG), function(i) {
    
    value <- unlist(log2(molecules_filter_ENSG[i, ]+1))
    
    fit_try <- tryCatch( fit <- blmer(value ~ 1|individual/replicate, 
                                    cov.prior = wishart,
                                    resid.prior = invgamma),
                           condition = function(c) c)
    if(inherits(fit_try, "condition")){
    var_foo <- rep(NA, 3)
    return(var_foo)
    }
    if(!inherits(fit_try, "condition")){
      var_foo <- as.data.frame(VarCorr(fit_try))[,4]
      var_foo <- var_foo[c(2,1,3)]
      var_foo
    }
  })
blme_varcomp <- do.call(rbind, blme_varcomp)
rownames(blme_varcomp) <- rownames(molecules_filter_ENSG)
colnames(blme_varcomp) <- c("individual","replicate","residual")
save(blme_varcomp,
     file = "../data/blme-variance.rda")
```

Standardardize variance components such that for each gene the variance 
component estimates sum up to 1. Because variance of sequencing counts depend on the dynamic range, we avoid biased comparison of variance estimates when comparing across genes, by the action of standarizing per gene variance components by the total of variance estimates.

```{r}
load("../data/blme-variance.rda")
blme_varcomp_norm <- (blme_varcomp/rowSums(blme_varcomp))

par(mfrow = c(2,2))
plot(blme_varcomp_norm[,1], blme_varcomp_norm[,2],
     xlab = "Individual",
     ylab = "Replicate",
     main = "Variance component estimate")
plot(blme_varcomp_norm[,3], blme_varcomp_norm[,1],
     xlab = "Residual",
     ylab = "Individual",
     main = "Variance component estimate")
plot(blme_varcomp_norm[,3], blme_varcomp_norm[,2],
     xlab = "Residual",
     ylab = "Replicate",
     main = "Variance component estimate")
boxplot(blme_varcomp_norm)
```

Compare individual versus replicate variance estimates.

```{r}
summary(blme_varcomp_norm)

# Kruskal wallis rank sum test to compare
# proportions of variance explained due to individual
# versus due to replicate
kruskal.test(c(blme_varcomp_norm[,1], blme_varcomp_norm[,2]) ~
               rep(c(1,2), each = NROW(blme_varcomp_norm)) )
```


## Correlation between and within batches

Code for computing correlation between cells within batches and between batches.

```{r compute-corr-batch, eval = TRUE}
#' molecules_input <- molecules_filter_ENSG
#' annotation <- anno_filter
compute_corr_batch <- function(molecules_input, annotation) {

  cor_mat <- cor(molecules_input, method = "spearman")
  batch <- unique(annotation$batch)
  individual <- unique(annotation$individual)

  # same individual, within batch
  corr_same_ind_within_batch <-
    lapply(1:length(individual), function(i) {
      batch <-
        unique(annotation$batch[annotation$individual == individual[i]])

      corr_batch <- lapply(1:length(batch), function(i) {
        df <- cor_mat[annotation$batch == batch[i],
                      annotation$batch == batch[i]]
        df[upper.tri(df, diag = FALSE)]
      })
      unlist(corr_batch)
    })

  # same individual, between replicates
  corr_same_ind_between_batch <-
    lapply(1:length(individual), function(i) {
      batch <-
        unique(annotation$batch[annotation$individual == individual[i]])
      submat <- lapply(1:(length(batch)-1), function(i) {
        submat0 <- lapply(2:length(batch), function(j) {
          df <- cor_mat[annotation$batch == batch[i],
                        annotation$batch == batch[j]]
          df[upper.tri(df, diag = FALSE)]
        })
        unlist(submat0)
      })
      unlist(submat)
    })

  # different individual
  corr_diff_ind_between_batch <-
    lapply(1:(length(individual)-1), function(i) {
      if (i == 1) {
        batch <-
          unique(annotation$batch[annotation$individual == individual[i]])
        batch_other <-
          unique(annotation$batch[annotation$individual != individual[i+1]])
      }
      if (i == 2) {
        batch <-
          unique(annotation$batch[annotation$individual == individual[i]])
        batch_other <-
          unique(annotation$batch[annotation$individual == individual[i+1]])
      }

      submat <- lapply(1:length(batch), function(i) {
        submat0 <- lapply(1:length(batch_other), function(j) {
          df <- cor_mat[annotation$batch == batch[i],
                        annotation$batch == batch_other[j]]
          df[upper.tri(df, diag = FALSE)]
        })
        unlist(submat0)
      })
      unlist(submat)
    })
  corr_diff_ind_between_batch <- unlist(corr_diff_ind_between_batch)

  return( list(corr_same_ind_within_batch = corr_same_ind_within_batch,
               corr_same_ind_between_batch = corr_same_ind_between_batch,
               corr_diff_ind_between_batch = corr_diff_ind_between_batch) )

}
```


Compute correlation for molecule-count data after filtering.

```{r}
corr_filter <- compute_corr_batch(molecules_filter_ENSG, anno_filter)
par(mfrow = c(1,1))
boxplot(cbind(corr_filter[[1]][[1]],
              corr_filter[[1]][[2]],
              corr_filter[[1]][[3]],
              corr_filter[[2]][[1]],
              corr_filter[[2]][[2]],
              corr_filter[[2]][[3]],
              corr_filter[[3]]),
        main = "Counts",
        axes = F)
axis(1, at = c(1:7),
     labels = c("within-batch-19098",
                      "within-batch-19101",
                      "within-batch-19239",
                      "between-batch-19098",
                      "between-batch-19101",
                      "between-batch-19239",
                      "across-batches"))
axis(2)
```

Kruskal wallis comparing all between-batch correlations with
all within-batch correlations


```{r}
df <- data.frame(corrs = c(unlist(corr_filter[[1]]),
                                unlist(corr_filter[[2]])),
                        label = c(rep(1, length(unlist(corr_filter[[1]]))),
                                  rep(2, length(unlist(corr_filter[[2]]))))) 
kruskal.test(df$corrs ~ df$label)

# summary statistics of correlations within-batches
# of all three individuals
summary(unlist(corr_filter[[1]]))

# summary statistics of correlations between-batches
# of all three individuals
summary(unlist(corr_filter[[2]]))
```


## Session information

```{r info}
sessionInfo()
```


