<!DOCTYPE html>

<html xmlns="http://www.w3.org/1999/xhtml">

<head>

<meta charset="utf-8">
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="generator" content="pandoc" />

<meta name="author" content="PoYuan Tung" />

<meta name="date" content="2015-10-23" />

<title>Proportion of gene detected after filter</title>

<script src="libs/jquery-1.11.0/jquery.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="libs/bootstrap-3.3.1/css/united.min.css" rel="stylesheet" />
<script src="libs/bootstrap-3.3.1/js/bootstrap.min.js"></script>
<script src="libs/bootstrap-3.3.1/shim/html5shiv.min.js"></script>
<script src="libs/bootstrap-3.3.1/shim/respond.min.js"></script>

<style type="text/css">

/* padding for bootstrap navbar */
body {
  padding-top: 50px;
  padding-bottom: 40px;
}


/* offset scroll position for anchor links (for fixed navbar)  */
.section h2 {
  padding-top: 55px;
  margin-top: -55px;
}
.section h3 {
  padding-top: 55px;
  margin-top: -55px;
}



/* don't use link color in navbar */
.dropdown-menu>li>a {
  color: black;
}

/* some padding for disqus */
#disqus_thread {
  margin-top: 45px;
}

</style>

<link rel="stylesheet" href="libs/font-awesome-4.1.0/css/font-awesome.min.css"/>

<style type="text/css">code{white-space: pre;}</style>
<link rel="stylesheet"
      href="libs/highlight/textmate.css"
      type="text/css" />
<script src="libs/highlight/highlight.js"></script>
<style type="text/css">
  pre:not([class]) {
    background-color: white;
  }
</style>
<script type="text/javascript">
if (window.hljs && document.readyState && document.readyState === "complete") {
   window.setTimeout(function() {
      hljs.initHighlighting();
   }, 0);
}
</script>



</head>

<body>

<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
code {
  color: inherit;
  background-color: rgba(0, 0, 0, 0.04);
}
img { 
  max-width:100%; 
  height: auto; 
}
</style>
<div class="container-fluid main-container">


<div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">singleCellSeq</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li><a href="index.html">Home</a></li>
        <li><a href="about.html">About</a></li>
        <li><a href="license.html">License</a></li>
        <li><a href="https://github.com/jdblischak/singleCellSeq">GitHub</a></li>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<div id="header">
<h1 class="title">Proportion of gene detected after filter</h1>
<h4 class="author"><em>PoYuan Tung</em></h4>
<h4 class="date"><em>2015-10-23</em></h4>
</div>

<div id="TOC">
<ul>
<li><a href="#setup">Setup</a></li>
<li><a href="#prepare-single-cell-molecule-data">Prepare single cell molecule data</a></li>
<li><a href="#molecule-count-data">Molecule count data</a></li>
<li><a href="#molecule-count-data-after-filtering">Molecule count data after filtering</a></li>
<li><a href="#molecule-count-data-final">Molecule count data final</a></li>
<li><a href="#summary-plots">Summary plots</a></li>
<li><a href="#session-information">Session information</a></li>
</ul>
</div>

<p><strong>Last updated:</strong> 2016-01-23</p>
<p><strong>Code version:</strong> 71bca2eedc500d1e5da32d526ba2127cb9aa7a63</p>
<p>The purpose is to see if removing the lowly expressed genes or using the final data (after correction of batch effect) would have any effect on the correltaion of PC1 and proportion of gene detected.</p>
<div id="setup" class="section level2">
<h2>Setup</h2>
<pre class="r"><code>source(&quot;functions.R&quot;)
library(&quot;edgeR&quot;)</code></pre>
<pre><code>Loading required package: limma</code></pre>
<pre class="r"><code>library(ggplot2)
library(&quot;cowplot&quot;)</code></pre>
<pre><code>
Attaching package: &#39;cowplot&#39;

The following object is masked from &#39;package:ggplot2&#39;:

    ggsave</code></pre>
<pre class="r"><code>theme_set(theme_bw(base_size = 16))</code></pre>
</div>
<div id="prepare-single-cell-molecule-data" class="section level2">
<h2>Prepare single cell molecule data</h2>
<p>Input annotation</p>
<pre class="r"><code>anno &lt;- read.table(&quot;../data/annotation.txt&quot;, header = TRUE,
                   stringsAsFactors = FALSE)

anno_filter &lt;- read.table(&quot;../data/annotation-filter.txt&quot;, header = TRUE,
                   stringsAsFactors = FALSE)</code></pre>
<p>Input molecule counts</p>
<pre class="r"><code>molecules &lt;- read.table(&quot;../data/molecules.txt&quot;, header = TRUE,
           stringsAsFactors = FALSE)

molecules_filter &lt;- read.table(&quot;../data/molecules-filter.txt&quot;, header = TRUE,
           stringsAsFactors = FALSE)

molecules_final &lt;- read.table(&quot;../data/molecules-final.txt&quot;, header = TRUE,
           stringsAsFactors = FALSE)

## qc cell
molecules_qc &lt;- molecules[,colnames(molecules_filter)]
stopifnot(anno_filter$sample_id == colnames(molecules_qc))</code></pre>
<p>Input read counts</p>
<pre class="r"><code>reads &lt;- read.table(&quot;../data/reads.txt&quot;, header = TRUE,
                    stringsAsFactors = FALSE)

reads_filter &lt;- read.table(&quot;../data/reads-filter.txt&quot;, header = TRUE,
                    stringsAsFactors = FALSE)
## qc cell
reads_qc &lt;- reads[,colnames(reads_filter)]
stopifnot(anno_filter$sample_id == colnames(reads_qc))</code></pre>
</div>
<div id="molecule-count-data" class="section level2">
<h2>Molecule count data</h2>
<p>First look at the data set before removing lowly expressed genes (genes with low counts)</p>
<p>Remove genes with zero count in the single cells</p>
<pre class="r"><code>expressed_single &lt;- rowSums(molecules_qc) &gt; 0
molecules_single &lt;- molecules_qc[which(expressed_single), ]
reads_single &lt;- reads_qc[expressed_single, ]</code></pre>
<pre class="r"><code>require(matrixStats)</code></pre>
<pre><code>Loading required package: matrixStats
matrixStats v0.14.0 (2015-02-13) successfully loaded. See ?matrixStats for help.</code></pre>
<pre class="r"><code>number_nonzero_cells &lt;- colSums(molecules_single != 0)
number_genes &lt;- dim(molecules_single)[1]
molecules_prop_genes_detected &lt;- 
    data.frame(prop = number_nonzero_cells/number_genes,
               individual = anno_filter$individual,
               individual.batch = anno_filter$batch)

## create a color palette with one color per individual and different shades for repplicates
great_color &lt;- c(&quot;#CC3300&quot;, &quot;#FF9966&quot;, &quot;#FFCC99&quot;, &quot;#006633&quot;, &quot;#009900&quot;, &quot;#99FF99&quot;, &quot;#3366FF&quot;, &quot;#6699FF&quot;, &quot;#66CCFF&quot;)

genes_detected_plot &lt;- ggplot(molecules_prop_genes_detected,
                       aes(y = prop, x = as.factor(individual.batch), fill = as.factor(individual.batch))) + 
                       geom_boxplot(alpha = .01, width = .2, position = position_dodge(width = .9)) +
                       geom_violin(alpha = .5) + 
                       scale_fill_manual(values = great_color) +
                       labs(x = &quot;Batch&quot;,
                       y = &quot;Proportion of detected genes&quot;,
                       title = &quot;Proportion of detected genes&quot;) +
                       theme(axis.text.x = element_text(hjust=1, angle = 45))

genes_detected_plot</code></pre>
<p><img src="figure/pca-correlation-rafa-filter.Rmd/mol-non-filtered-1.png" title="" alt="" width="672" style="display: block; margin: auto;" /></p>
<p>Principal component analysis on log2 transformed values. We avoid log of 0’s by add 1’s. In addition, our PCA analysis requires that every gene needs to be present in at least one of the cells.</p>
<pre class="r"><code>molecules_single_log2_pca &lt;- run_pca( log2( molecules_single + 1 ) )

pc1_plot &lt;- qplot(y = molecules_single_log2_pca$PCs[,1],
      x = molecules_prop_genes_detected$prop,
      shape = as.factor(anno_filter$replicate),
      colour = as.factor(anno_filter$individual),
      xlab = &quot;Proportion of genes detected&quot;,
      ylab = &quot;PC1&quot;,
      title = &quot;Proportion of genes detected&quot;) 

pc1_plot</code></pre>
<p><img src="figure/pca-correlation-rafa-filter.Rmd/pca-1.png" title="" alt="" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>qplot(y = molecules_single_log2_pca$PCs[,2],
      x = molecules_prop_genes_detected$prop,
      shape = as.factor(anno_filter$replicate),
      colour = as.factor(anno_filter$individual),
      xlab = &quot;Proportion of genes detected&quot;,
      ylab = &quot;PC2&quot;) </code></pre>
<p><img src="figure/pca-correlation-rafa-filter.Rmd/pca-2.png" title="" alt="" width="672" style="display: block; margin: auto;" /></p>
</div>
<div id="molecule-count-data-after-filtering" class="section level2">
<h2>Molecule count data after filtering</h2>
<pre class="r"><code>require(matrixStats)
number_nonzero_cells_filter &lt;- colSums(molecules_filter != 0)
number_genes_filter &lt;- dim(molecules_filter)[1]
molecules_prop_genes_detected_filter &lt;- 
    data.frame(prop = number_nonzero_cells_filter/number_genes_filter,
               individual = anno_filter$individual,
               individual.batch = anno_filter$batch)

genes_detected_filter_plot &lt;- ggplot(molecules_prop_genes_detected_filter,
                              aes(y = prop, x = as.factor(individual.batch), fill = as.factor(individual.batch))) + 
                              geom_boxplot(alpha = .01, width = .2, position = position_dodge(width = .9)) +
                              geom_violin(alpha = .5) + 
                              scale_fill_manual(values = great_color) +
                              labs(x = &quot;Batch&quot;,
                              y = &quot;Proportion of detected genes&quot;,
                              title = &quot;Proportion of detected genes (filtered)&quot;) +
                              theme(axis.text.x = element_text(hjust=1, angle = 45))

genes_detected_filter_plot</code></pre>
<p><img src="figure/pca-correlation-rafa-filter.Rmd/mol-filtered-1.png" title="" alt="" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>molecules_filter_log2_pca &lt;- run_pca( log2( molecules_filter + 1 ) )

pc1_filter_plot &lt;- qplot(y = molecules_filter_log2_pca$PCs[,1],
      x = molecules_prop_genes_detected_filter$prop,
      shape = as.factor(anno_filter$replicate),
      colour = as.factor(anno_filter$individual),
      xlab = &quot;Proportion of genes detected (filter)&quot;,
      ylab = &quot;PC1&quot;,
      title = &quot;Proportion of genes detected (filter)&quot;) 

pc1_filter_plot</code></pre>
<p><img src="figure/pca-correlation-rafa-filter.Rmd/pca-filter-1.png" title="" alt="" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>qplot(y = molecules_filter_log2_pca$PCs[,2],
      x = molecules_prop_genes_detected$prop,
      shape = as.factor(anno_filter$replicate),
      colour = as.factor(anno_filter$individual),
      xlab = &quot;Proportion of genes detected (filter)&quot;,
      ylab = &quot;PC2&quot;) </code></pre>
<p><img src="figure/pca-correlation-rafa-filter.Rmd/pca-filter-2.png" title="" alt="" width="672" style="display: block; margin: auto;" /></p>
</div>
<div id="molecule-count-data-final" class="section level2">
<h2>Molecule count data final</h2>
<pre class="r"><code>molecules_final_pca &lt;- run_pca(molecules_final)

pc1_final_plot &lt;- qplot(y = molecules_final_pca$PCs[,1],
      x = molecules_prop_genes_detected_filter$prop,
      shape = as.factor(anno_filter$replicate),
      colour = as.factor(anno_filter$individual),
      xlab = &quot;Proportion of genes detected (final)&quot;,
      ylab = &quot;PC1&quot;,
      title = &quot;Proportion of genes detected (final)&quot;) 

pc1_final_plot</code></pre>
<p><img src="figure/pca-correlation-rafa-filter.Rmd/pca-final-1.png" title="" alt="" width="672" style="display: block; margin: auto;" /></p>
</div>
<div id="summary-plots" class="section level2">
<h2>Summary plots</h2>
<pre class="r"><code>theme_set(theme_bw(base_size = 12))
plot_grid(genes_detected_plot + theme(legend.position = &quot;none&quot;),
          genes_detected_filter_plot + theme(legend.position = &quot;none&quot;),
          pc1_plot + theme(legend.position = c(.87,.35)) + labs(col = &quot;Individual&quot;, shape = &quot;Replicate&quot;),
          pc1_filter_plot + theme(legend.position = &quot;none&quot;),
          pc1_final_plot + theme(legend.position = &quot;none&quot;),
          labels = LETTERS[1:5],
          ncol = 2)</code></pre>
<p><img src="figure/pca-correlation-rafa-filter.Rmd/sum-plots-1.png" title="" alt="" width="1152" style="display: block; margin: auto;" /></p>
</div>
<div id="session-information" class="section level2">
<h2>Session information</h2>
<pre class="r"><code>sessionInfo()</code></pre>
<pre><code>R version 3.2.0 (2015-04-16)
Platform: x86_64-unknown-linux-gnu (64-bit)

locale:
 [1] LC_CTYPE=en_US.UTF-8       LC_NUMERIC=C              
 [3] LC_TIME=en_US.UTF-8        LC_COLLATE=en_US.UTF-8    
 [5] LC_MONETARY=en_US.UTF-8    LC_MESSAGES=en_US.UTF-8   
 [7] LC_PAPER=en_US.UTF-8       LC_NAME=C                 
 [9] LC_ADDRESS=C               LC_TELEPHONE=C            
[11] LC_MEASUREMENT=en_US.UTF-8 LC_IDENTIFICATION=C       

attached base packages:
[1] stats     graphics  grDevices utils     datasets  methods   base     

other attached packages:
[1] testit_0.4         matrixStats_0.14.0 cowplot_0.3.1     
[4] ggplot2_1.0.1      edgeR_3.10.2       limma_3.24.9      
[7] knitr_1.10.5      

loaded via a namespace (and not attached):
 [1] Rcpp_0.12.0      magrittr_1.5     MASS_7.3-40      munsell_0.4.2   
 [5] colorspace_1.2-6 stringr_1.0.0    httr_0.6.1       plyr_1.8.3      
 [9] tools_3.2.0      grid_3.2.0       gtable_0.1.2     htmltools_0.2.6 
[13] yaml_2.1.13      digest_0.6.8     reshape2_1.4.1   formatR_1.2     
[17] bitops_1.0-6     RCurl_1.95-4.6   evaluate_0.7     rmarkdown_0.6.1 
[21] labeling_0.3     stringi_0.4-1    scales_0.2.4     proto_0.3-10    </code></pre>
</div>


<!-- some extra javascript for older browsers -->
<script type="text/javascript" src="libs/polyfill.js"></script>

<script>

// manage active state of menu based on current page
$(document).ready(function () {

    // active menu
    href = window.location.pathname
    href = href.substr(href.lastIndexOf('/') + 1)
    $('a[href="' + href + '"]').parent().addClass('active');

    // manage active menu header
    if (href.startsWith('authoring_'))
      $('a[href="' + 'authoring' + '"]').parent().addClass('active');
    else if (href.endsWith('_format.html'))
      $('a[href="' + 'formats' + '"]').parent().addClass('active');
    else if (href.startsWith('developer_'))
      $('a[href="' + 'developer' + '"]').parent().addClass('active');

});

</script>

</div>

<script>

// add bootstrap table styles to pandoc tables
$(document).ready(function () {
  $('tr.header').parent('thead').parent('table').addClass('table table-condensed');
});

</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://cdn.mathjax.org/mathjax/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
