---
title: "Process sequence data"
author: "John Blischak"
date: 2015-04-11
---

**Last updated:** `r Sys.Date()`

**Code version:** `r system("git log -1 --format='%H'", intern = TRUE)`

All processing scripts were run from the data directory.

```bash
cd /mnt/gluster/data/internal_supp/singleCellSeq
```

The raw fastq files were write-protected to avoid accidental deletion.

```bash
chmod -x fastq/*fastq.gz
```

## Run FastQC

```bash
submit-array.sh run-fastqc.sh 2g fastq/*fastq.gz
```

```bash
grep -w success log/run-fastqc* | wc -l
```

## Trim UMI

```bash
submit-array.sh trim.sh 2g fastq/*fastq.gz
```

To confirm that the jobs ran successfully:

```bash
grep -w success log/trim* | wc -l
```

To re-run failed jobs, I re-ran the original command.
If the output file already exists, the code is not run and "success" is not echo'd to the log file.

## Map to genome

```bash
submit-array.sh map-subread.sh 8g trim/*fastq.gz
```

```bash
grep -w success log/map-subread* | wc -l
```

## Process bam files

Remove quality less than 10
Sort bam
index bam

```bash
submit-array.sh process-bam.sh 8g bam/*bam
```

```bash
grep -w success log/process-bam* | wc -l
```

Check for the presence of intermediate files output during sorting.

```bash
ls bam-processed/*sorted*0*bam
```

## Remove duplicate UMIs

```bash
submit-array.sh rmdup-umi.sh 2g bam-processed/*bam
```

```bash
grep -w success log/rmdup-umi* | wc -l
```

## Count reads per gene

```bash
submit-array.sh count-reads-per-gene.sh 2g bam-processed/*bam
```

```bash
submit-array.sh count-reads-per-gene.sh 2g bam-rmdup-umi/*bam
```

```bash
grep -w success log/count-reads-per-gene* | wc -l
```

