---
title: "Process sequence data"
author: "John Blischak"
date: 2015-04-11
---

**Last updated:** `r Sys.Date()`

**Code version:** `r system("git log -1 --format='%H'", intern = TRUE)`

All processing scripts were run from the data directory.

```bash
cd /mnt/gluster/data/internal_supp/singleCellSeq
```

The raw fastq files were write-protected to avoid accidental deletion.

```bash
chmod uga=r fastq/*fastq.gz
```

## Create genome for mapping

[create-genome.sh][] downloads the fasta files for human genome hg19 (chromosomes 1-22, X, Y, M) and the ERRC spike-ins.
It indexes the genome with `subread-buildindex`.

```bash
submit-array.sh create-genome.sh 8g genome
```

[create-genome.sh]: https://github.com/jdblischak/singleCellSeq/blob/master/code/create-genome.sh

## Run FastQC

```bash
submit-array.sh run-fastqc.sh 2g fastq/*fastq.gz
```

```bash
ls fastq/ | wc -l
grep -w success ~/log/run-fastqc.sh/* | wc -l
grep -w failure ~/log/run-fastqc.sh/* | wc -l
```

## Trim UMI

[trim.sh][] removes the 5 bp UMI at the 5' end of the read using the program `umitools trim`.
We used [umitools v2.1.1][umitools]. Output fastq files are written to the directory `trim`.
Reads without a valid UMI are written to the directory `invalid`.

```bash
submit-array.sh trim.sh 2g fastq/*fastq.gz
```

[trim.sh]: https://github.com/jdblischak/singleCellSeq/blob/master/code/trim.sh
[umitools]: https://github.com/brwnj/umitools/releases/tag/v2.1.1

To confirm that the jobs ran successfully:

```bash
ls trim/*fastq.gz | wc -l
ls invalid/*fastq.gz | wc -l
grep -w success ~/log/trim.sh/* | wc -l
grep -w failure ~/log/trim.sh/* | wc -l
```

To re-run failed jobs, I re-ran the original command.
If the output file already exists, the code is not run and "success" is not echo'd to the log file.

## Quality trim 3' end of reads

[sickle.sh][] performs quality trimming of the 3' end of the reads (flag `-x`).
We used the default quality thresholds.
We used [sickle version 1.33][sickle].
Output is written to the directory `sickle`.

```bash
submit-array.sh sickle.sh 2g trim/*fastq.gz
```

[sickle.sh]: https://github.com/jdblischak/singleCellSeq/blob/master/code/sickle.sh
[sickle]: https://github.com/najoshi/sickle/releases/tag/v1.33

To confirm that the jobs ran successfully:

```bash
ls sickle/*fastq.gz | wc -l
grep -w success ~/log/sickle.sh/* | wc -l
grep -w failure ~/log/sickle.sh/* | wc -l
```

## Assess sequence characteristics

```bash
submit-array.sh qc.sh 2g fastq/*fastq.gz trim/*fastq.gz sickle/*fastq.gz
```

```bash
ls seqqs/*_len.txt | wc -l
grep -w success ~/log/qc.sh/* | wc -l
grep -w failure ~/log/qc.sh/* | wc -l
```

## Map to genome

[map-subread.sh][] maps the reads to the combined genome described above.
We used [Subread version 1.5.0-p1][subread] with the default thresholds and reporting only uniquely mapping reads (flag `-u`).
Output BAM files are written to the directory `bam`.

```bash
submit-array.sh map-subread.sh 12g sickle/*fastq.gz
```

[map-subread.sh]: https://github.com/jdblischak/singleCellSeq/blob/master/code/map-subread.sh
[subread]: http://sourceforge.net/projects/subread/files/subread-1.5.0-p1/

```bash
ls bam/*bam | wc -l
grep -w success ~/log/map-subread.sh/* | wc -l
grep -w failure ~/log/map-subread.sh/* | wc -l
```

## Process bam files

[process-bam.sh][] uses [samtools version 0.1.18-dev (r982:313)][samtools] to sort and index each BAM file.
The output is written to the directory `bam-processed`.

```bash
submit-array.sh process-bam.sh 8g bam/*bam
```

[process-bam.sh]: https://github.com/jdblischak/singleCellSeq/blob/master/code/process-bam.sh
[samtools]: http://sourceforge.net/projects/samtools/files/samtools/0.1.18/

```bash
ls bam-processed/*bam | wc -l
grep -w success ~/log/process-bam.sh/* | wc -l
grep -w failure ~/log/process-bam.sh/* | wc -l
```

Check for the presence of intermediate files output during sorting.

```bash
ls bam-processed/*sorted*0*bam
```

## Combine single cell samples

Molecule counts are obtained by counting the number of unique UMIs observed at each start position.
In order for this to be accurate for a given single cell, we need to combine all the sequencing data for a given sample.
We merge the sorted BAM files using `samtools merge` and save the output BAM files in the directory `bam-combined`.

```bash
# From head node
mkdir -p bam-combined
mkdir -p ~/log/combine
for IND in 19098 19101 19239
do
  for REP in {1..3}
  do
    for ROW in {A..H}
    do
      for COL in {1..12}
      do
        ID=`printf "%s.%s.%s%02d\n" $IND $REP $ROW $COL`
        echo $ID
        echo "samtools merge bam-combined/$ID.trim.sickle.sorted.combined.bam bam-processed/$ID*trim.sickle.sorted.bam" | qsub -l h_vmem=4g -N $ROW$COL.$IND.$REP.combine -cwd -o ~/log/combine.sh -j y -V
      done
    done
  done
done
```

Confirm that it worked.
We only expect 864 samples (3 individuals * 3 replicates * 96 wells).
The bulk samples are not combined because they violate the assumptions of a UMI protocol.
In other words, they contain too many unique molecules for the 1,024 UMIs to properly tag them all.

```bash
ls bam-combined/*bam | wc -l
# There should be no output from samtools merge
cat ~/log/combine/* | head
```

Then index each merged sample:

```bash
# from head node
mkdir -p ~/log/index
for IND in 19098 19101 19239
do
  for REP in {1..3}
  do
    for ROW in {A..H}
    do
      for COL in {1..12}
      do
        ID=`printf "%s.%s.%s%02d\n" $IND $REP $ROW $COL`
        echo $ID
        echo "samtools index bam-combined/$ID.trim.sickle.sorted.combined.bam" | qsub -l h_vmem=2g -N $ROW$COL.$IND.$REP.index -cwd -o ~/log/index -j y -V -l 'hostname=!bigmem02'
      done
    done
  done
done
```

Confirm that it worked.

```bash
ls bam-combined/*bai | wc -l
# There should be no output from samtools index
cat ~/log/index/* | head
```

## Remove duplicate UMIs

```bash
submit-array.sh rmdup-umi.sh 2g bam-processed/*bam
```

```bash
ls bam-rmdup-umi/*bam | wc -l
grep -w success ~/log/rmdup-umi.sh/* | wc -l
grep -w failure ~/log/rmdup-umi.sh/* | wc -l
```

## Count reads per gene

```bash
submit-array.sh count-reads-per-gene.sh 2g bam-processed/*bam bam-rmdup-umi/*bam
```

```bash
ls counts/*genecounts.txt | wc -l
grep -w success ~/log/count-reads-per-gene.sh/* | wc -l
grep -w failure ~/log/count-reads-per-gene.sh/* | wc -l
```

## Gather total counts

The total number of reads at each stage of the processing pipeline.

```bash
gather-total-counts.py > total-counts.txt
```

## Gather summary counts

The classification of reads by featureCounts.

```bash
gather-summary-counts.py > summary-counts.txt
```

## Gather gene counts

The counts for each gene for each sequencing lane.

```bash
gather-gene-counts.py > gene-counts.txt
```
