<!DOCTYPE html>

<html xmlns="http://www.w3.org/1999/xhtml">

<head>

<meta charset="utf-8">
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="generator" content="pandoc" />

<meta name="author" content="John Blischak" />

<meta name="date" content="2015-04-11" />

<title>Process sequence data</title>

<script src="libs/jquery-1.11.0/jquery.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="libs/bootstrap-3.3.1/css/united.min.css" rel="stylesheet" />
<script src="libs/bootstrap-3.3.1/js/bootstrap.min.js"></script>
<script src="libs/bootstrap-3.3.1/shim/html5shiv.min.js"></script>
<script src="libs/bootstrap-3.3.1/shim/respond.min.js"></script>

<style type="text/css">

/* padding for bootstrap navbar */
body {
  padding-top: 50px;
  padding-bottom: 40px;
}


/* offset scroll position for anchor links (for fixed navbar)  */
.section h2 {
  padding-top: 55px;
  margin-top: -55px;
}
.section h3 {
  padding-top: 55px;
  margin-top: -55px;
}



/* don't use link color in navbar */
.dropdown-menu>li>a {
  color: black;
}

/* some padding for disqus */
#disqus_thread {
  margin-top: 45px;
}

</style>

<link rel="stylesheet" href="libs/font-awesome-4.1.0/css/font-awesome.min.css"/>

<style type="text/css">code{white-space: pre;}</style>
<link rel="stylesheet"
      href="libs/highlight/textmate.css"
      type="text/css" />
<script src="libs/highlight/highlight.js"></script>
<style type="text/css">
  pre:not([class]) {
    background-color: white;
  }
</style>
<script type="text/javascript">
if (window.hljs && document.readyState && document.readyState === "complete") {
   window.setTimeout(function() {
      hljs.initHighlighting();
   }, 0);
}
</script>



</head>

<body>

<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
code {
  color: inherit;
  background-color: rgba(0, 0, 0, 0.04);
}
img { 
  max-width:100%; 
  height: auto; 
}
</style>
<div class="container-fluid main-container">


<div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">singleCellSeq</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li><a href="index.html">Home</a></li>
        <li><a href="about.html">About</a></li>
        <li><a href="license.html">License</a></li>
        <li><a href="https://github.com/jdblischak/singleCellSeq">GitHub</a></li>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<div id="header">
<h1 class="title">Process sequence data</h1>
<h4 class="author"><em>John Blischak</em></h4>
<h4 class="date"><em>2015-04-11</em></h4>
</div>

<div id="TOC">
<ul>
<li><a href="#run-fastqc">Run FastQC</a></li>
<li><a href="#trim-umi">Trim UMI</a></li>
<li><a href="#quality-trim-3-end-of-reads">Quality trim 3’ end of reads</a></li>
<li><a href="#assess-sequence-characteristics">Assess sequence characteristics</a></li>
<li><a href="#map-to-genome">Map to genome</a></li>
<li><a href="#process-bam-files">Process bam files</a></li>
<li><a href="#remove-duplicate-umis">Remove duplicate UMIs</a></li>
<li><a href="#count-reads-per-gene">Count reads per gene</a></li>
<li><a href="#gather-total-counts">Gather total counts</a></li>
<li><a href="#gather-summary-counts">Gather summary counts</a></li>
<li><a href="#gather-gene-counts">Gather gene counts</a></li>
<li><a href="#process-at-the-sample-level">Process at the sample level</a></li>
</ul>
</div>

<p><strong>Last updated:</strong> 2015-06-11</p>
<p><strong>Code version:</strong> 44338a0a5f347c0c7a565729f2a86161eb3229d3</p>
<p>All processing scripts were run from the data directory.</p>
<pre class="bash"><code>cd /mnt/gluster/data/internal_supp/singleCellSeq</code></pre>
<p>The raw fastq files were write-protected to avoid accidental deletion.</p>
<pre class="bash"><code>chmod uga=r fastq/*fastq.gz</code></pre>
<div id="run-fastqc" class="section level2">
<h2>Run FastQC</h2>
<pre class="bash"><code>submit-array.sh run-fastqc.sh 2g fastq/*fastq.gz</code></pre>
<pre class="bash"><code>ls fastq/ | wc -l
grep -w success ~/log/run-fastqc.sh/* | wc -l
grep -w failure ~/log/run-fastqc.sh/* | wc -l</code></pre>
</div>
<div id="trim-umi" class="section level2">
<h2>Trim UMI</h2>
<pre class="bash"><code>submit-array.sh trim.sh 2g fastq/*fastq.gz</code></pre>
<p>To confirm that the jobs ran successfully:</p>
<pre class="bash"><code>ls trim/*fastq.gz | wc -l
grep -w success ~/log/trim.sh/* | wc -l
grep -w failure ~/log/trim.sh/* | wc -l</code></pre>
<p>To re-run failed jobs, I re-ran the original command. If the output file already exists, the code is not run and “success” is not echo’d to the log file.</p>
</div>
<div id="quality-trim-3-end-of-reads" class="section level2">
<h2>Quality trim 3’ end of reads</h2>
<pre class="bash"><code>submit-array.sh sickle.sh 2g trim/*fastq.gz</code></pre>
<p>To confirm that the jobs ran successfully:</p>
<pre class="bash"><code>ls sickle/*fastq.gz | wc -l
grep -w success ~/log/sickle.sh/* | wc -l
grep -w failure ~/log/sickle.sh/* | wc -l</code></pre>
</div>
<div id="assess-sequence-characteristics" class="section level2">
<h2>Assess sequence characteristics</h2>
<pre class="bash"><code>submit-array.sh qc.sh 2g fastq/*fastq.gz trim/*fastq.gz sickle/*fastq.gz</code></pre>
<pre class="bash"><code>ls seqqs/*_len.txt | wc -l
grep -w success ~/log/qc.sh/* | wc -l
grep -w failure ~/log/qc.sh/* | wc -l</code></pre>
</div>
<div id="map-to-genome" class="section level2">
<h2>Map to genome</h2>
<pre class="bash"><code>submit-array.sh map-subread.sh 8g trim/*fastq.gz sickle/*fastq.gz</code></pre>
<pre class="bash"><code>ls bam/*bam | wc -l
grep -w success ~/log/map-subread.sh/* | wc -l
grep -w failure ~/log/map-subread.sh/* | wc -l</code></pre>
</div>
<div id="process-bam-files" class="section level2">
<h2>Process bam files</h2>
<ul>
<li>Sort bam</li>
<li>Index bam</li>
</ul>
<pre class="bash"><code>submit-array.sh process-bam.sh 8g bam/*bam</code></pre>
<pre class="bash"><code>ls bam-processed/*bam | wc -l
grep -w success ~/log/process-bam.sh/* | wc -l
grep -w failure ~/log/process-bam.sh/* | wc -l</code></pre>
<p>Check for the presence of intermediate files output during sorting.</p>
<pre class="bash"><code>ls bam-processed/*sorted*0*bam</code></pre>
</div>
<div id="remove-duplicate-umis" class="section level2">
<h2>Remove duplicate UMIs</h2>
<pre class="bash"><code>submit-array.sh rmdup-umi.sh 2g bam-processed/*bam</code></pre>
<pre class="bash"><code>ls bam-rmdup-umi/*bam | wc -l
grep -w success ~/log/rmdup-umi.sh/* | wc -l
grep -w failure ~/log/rmdup-umi.sh/* | wc -l</code></pre>
</div>
<div id="count-reads-per-gene" class="section level2">
<h2>Count reads per gene</h2>
<pre class="bash"><code>submit-array.sh count-reads-per-gene.sh 2g bam-processed/*bam bam-rmdup-umi/*bam</code></pre>
<pre class="bash"><code>ls counts/*genecounts.txt | wc -l
grep -w success ~/log/count-reads-per-gene.sh/* | wc -l
grep -w failure ~/log/count-reads-per-gene.sh/* | wc -l</code></pre>
</div>
<div id="gather-total-counts" class="section level2">
<h2>Gather total counts</h2>
<p>The total number of reads at each stage of the processing pipeline.</p>
<pre class="bash"><code>gather-total-counts.py &gt; total-counts.txt</code></pre>
</div>
<div id="gather-summary-counts" class="section level2">
<h2>Gather summary counts</h2>
<p>The classification of reads by featureCounts.</p>
<pre class="bash"><code>gather-summary-counts.py &gt; summary-counts.txt</code></pre>
</div>
<div id="gather-gene-counts" class="section level2">
<h2>Gather gene counts</h2>
<p>The counts for each gene for each sequencing lane.</p>
<pre class="bash"><code>gather-gene-counts.py &gt; gene-counts.txt</code></pre>
</div>
<div id="process-at-the-sample-level" class="section level2">
<h2>Process at the sample level</h2>
<p>All the processing performed above is done at the level of the sequencing lane. Thus when removing reads that have the same UMI and start position, this is done at the level of lane instead of the whole sample.</p>
<p>To get true molecule counts, need to run <code>umitools rmdup</code> on all the reads for a given sample. Only using sickle trimmed data. Starting with the sorted bam files, merge the files per sample.</p>
<pre class="bash"><code># From head node
cd $ssd
mkdir -p bam-combined
mkdir -p ~/log/combine.sh
for IND in 19098 19101 19239
do
  for BATCH in {1..3}
  do
    for ROW in {A..H}
    do
      for COL in {1..12}
      do
        ID=`printf &quot;%s.%s.%s%02d\n&quot; $IND $BATCH $ROW $COL`
        echo $ID
        echo &quot;samtools merge bam-combined/$ID.trim.sickle.sorted.bam bam-processed/$ID*trim.sickle.sorted.bam&quot; | qsub -l h_vmem=4g -N $ROW$COL.$IND.$BATCH.combine -cwd -o ~/log/combine.sh -j y -V
      done
    done
  done
done</code></pre>
<p>Then index each merged sample:</p>
<pre class="bash"><code>mkdir -p ~/log/combine2.sh
for IND in 19098 19101 19239
do
  for BATCH in {1..3}
  do
    for ROW in {A..H}
    do
      for COL in {1..12}
      do
        ID=`printf &quot;%s.%s.%s%02d\n&quot; $IND $BATCH $ROW $COL`
        echo $ID
        echo &quot;samtools index bam-combined/$ID.trim.sickle.sorted.bam&quot; | qsub -l h_vmem=2g -N $ROW$COL.$IND.$BATCH.index -cwd -o ~/log/combine2.sh -j y -V
      done
    done
  done
done</code></pre>
<p>Now can run <code>rmdup umitools</code> and <code>featureCounts</code> as above:</p>
<pre class="bash"><code>submit-array.sh rmdup-umi.sh 2g bam-combined/*bam</code></pre>
<p>Since do not want to re-run all the samples, need to change the name so that it is clear it is a combined sample.</p>
<pre class="bash"><code>rename sorted.bam sorted.combined.bam *trim.sickle.sorted.bam
rename sorted.bam.bai sorted.combined.bam.bai *trim.sickle.sorted.bam.bai</code></pre>
<p>Now count the number of reads per gene for both the molecules and the reads:</p>
<pre class="bash"><code>submit-array.sh count-reads-per-gene.sh 2g bam-rmdup-umi/*combined.rmdup.bam bam-combined/*bam</code></pre>
<p>Now need to do the same for the bulk samples:</p>
<p>Merge bulk samples.</p>
<pre class="bash"><code># From head node
cd $ssd
mkdir ~/log/bulk
for IND in 19098 19101 19239
do
  for BATCH in {1..3}
  do
    ID=$IND.$BATCH.bulk
    echo $ID
    echo &quot;samtools merge bam-combined/$ID.trim.sickle.sorted.combined.bam bam-processed/$ID*trim.sickle.sorted.bam&quot; | qsub -l h_vmem=8g -N bulk.$IND.$BATCH.combine -cwd -o ~/log/bulk -j y -V
  done
done</code></pre>
<p>Index bulk samples.</p>
<pre class="bash"><code># From head node
cd $ssd
mkdir -p ~/log/bulk
for IND in 19098 19101 19239
do
  for BATCH in {1..3}
  do
    ID=$IND.$BATCH.bulk
    echo $ID
    echo &quot;samtools index bam-combined/$ID.trim.sickle.sorted.combined.bam&quot; | qsub -l h_vmem=8g -N bulk.$IND.$BATCH.index -cwd -o ~/log/bulk -j y -V
  done
done</code></pre>
<p>Remove duplicates and count reads per gene.</p>
<pre class="bash"><code>submit-array.sh rmdup-umi.sh 8g bam-combined/*bulk*bam</code></pre>
<pre class="bash"><code>submit-array.sh count-reads-per-gene.sh 2g bam-rmdup-umi/*bulk*combined.rmdup.bam bam-combined/*bulk*bam</code></pre>
</div>


<!-- some extra javascript for older browsers -->
<script type="text/javascript" src="libs/polyfill.js"></script>

<script>

// manage active state of menu based on current page
$(document).ready(function () {

    // active menu
    href = window.location.pathname
    href = href.substr(href.lastIndexOf('/') + 1)
    $('a[href="' + href + '"]').parent().addClass('active');

    // manage active menu header
    if (href.startsWith('authoring_'))
      $('a[href="' + 'authoring' + '"]').parent().addClass('active');
    else if (href.endsWith('_format.html'))
      $('a[href="' + 'formats' + '"]').parent().addClass('active');
    else if (href.startsWith('developer_'))
      $('a[href="' + 'developer' + '"]').parent().addClass('active');

});

</script>

</div>

<script>

// add bootstrap table styles to pandoc tables
$(document).ready(function () {
  $('tr.header').parent('thead').parent('table').addClass('table table-condensed');
});

</script>


</body>
</html>
