---
title: "Quality control at level of sequencing lane"
author: "John Blischak"
date: 2015-05-12
---

**Last updated:** `r Sys.Date()`

**Code version:** `r system("git log -1 --format='%H'", intern = TRUE)`

```{r chunk-options, include=FALSE}
source("chunk-options.R")
opts_chunk$set(fig.width = 8, fig.height = 8)
```

The script [gather-gene-counts.py][] compiles all the gene counts and extracts the relevant variables from the filename.
The result is the file [gene-counts.txt].
Each row is a given fastq file, and each column is a variable (including all the genes).

[gather-total-counts.py]: https://github.com/jdblischak/singleCellSeq/blob/master/code/gather-gene-counts.py
[gene-counts.txt]: https://github.com/jdblischak/singleCellSeq/blob/master/data/gene-counts.txt

## Analyzing gene counts

```{r packages}
library("data.table")
library("edgeR")
library("ggplot2")
library("gplots")
theme_set(theme_bw(base_size = 16))
```

```{r input}
# counts <- read.table("../data/gene-counts.txt", header = FALSE, sep = "\t", stringsAsFactors = FALSE)
counts <- fread("../data/gene-counts.txt")
```

```{r}
dim(counts)
head(counts[, 1:11, with = FALSE])
table(counts$lane, counts$flow_cell)
```

## Functions

```{r}
calc_pca <- function(dt, num_pcs = 1:6) {
  # Calculate pca.
  # dt - data.table
  # num_pcs: Number of PCs to return
  require("edgeR")
  df <- df[, colSums(df) > 0, with = FALSE]
  df <- t(df)
  cpm_mat <- cpm(df, log = TRUE)
  pca <- prcomp(t(cpm_mat), scale. = TRUE)
  # return(as.list(pca$x))
  return(pca$x[, num_pcs])
}
```

## Perform PCA on just the ERCC genes

```{r}
counts_ercc <- counts[, grep("ERCC", colnames(counts)), with = FALSE]
anno <- counts[, 1:8, with = FALSE]
anno$type <- ifelse(anno$well == "bulk", "bulk", "single")

pca_ercc <- data.frame()
for (rmdup_status in c("molecules", "reads")) {
  for (sickle_status in c("not-quality-trimmed", "quality-trimmed")) {
    dat <- counts_ercc[anno$rmdup == rmdup_status & anno$sickle == sickle_status, ]
    # print(dat[, .N])
    result <- calc_pca(dat)
    result_df <- data.frame(anno[rmdup == rmdup_status & sickle == sickle_status, ], result,
                            stringsAsFactors = FALSE)
    pca_ercc <- rbind(pca_ercc, result_df)
  }
}
```

```{r}
ggplot(pca_ercc, aes(x = PC1, y = PC2, col = as.factor(type))) +
  geom_point() +
  facet_grid(sickle ~ rmdup)

ggplot(pca_ercc, aes(x = PC1, y = PC2, col = as.factor(individual))) +
  geom_point() +
  facet_grid(sickle ~ rmdup)
```

## Correlate each factor with each PC

```{r}
factors <- colnames(pca_ercc)[c(1:6, 9)]
pcs <- colnames(pca_ercc)[10:ncol(pca_ercc)]
for (rmdup_status in c("molecules", "reads")) {
  for (sickle_status in c("not-quality-trimmed", "quality-trimmed")) {
    pca_ercc_corr <- matrix(NA, nrow = length(factors), ncol = length(pcs),
                            dimnames = list(factors, pcs))
    for (fac in factors) {
      for (pc in pcs) {
        result_lm <- lm(pca_ercc[pca_ercc$rmdup == rmdup_status &
                                 pca_ercc$sickle == sickle_status, pc] ~
                          as.factor(pca_ercc[pca_ercc$rmdup == rmdup_status &
                                             pca_ercc$sickle == sickle_status, fac]))
        result_r2 <- summary(result_lm)$adj.r.squared
        # result_df <- data.frame(fac, pc, result_r2, stringsAsFactors = FALSE)
        pca_ercc_corr[fac, pc] <- result_r2
      }
    }
    heatmap.2(pca_ercc_corr, trace = "none", main = paste(rmdup_status, sickle_status))
  }
}
```

## Find most variable genes

```{r}
cov <- counts[, lapply(.SD, function(x) sd(x) / (mean(x) + 1)),
              by = .(rmdup, sickle),
              .SDcols = grep("ENSG", colnames(counts))]
cov_reads_not_qual <- cov[rmdup == "reads" & sickle == "not-quality-trimmed",
                          grep("ENSG", colnames(cov)), with = FALSE]
cov_vec <- as.numeric(cov_reads_not_qual)
names(cov_vec) <- colnames(cov_reads_not_qual)
var_genes <- names(cov_vec)[cov_vec > 3]
```

## PCA using variable genes

```{r}
counts_ercc <- counts[, var_genes, with = FALSE]
```

## Session information

```{r info}
sessionInfo()
```
