<!DOCTYPE html>

<html xmlns="http://www.w3.org/1999/xhtml">

<head>

<meta charset="utf-8">
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="generator" content="pandoc" />


<meta name="date" content="2015-07-10" />

<title>Quality control for LCLs</title>

<script src="libs/jquery-1.11.0/jquery.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="libs/bootstrap-3.3.1/css/united.min.css" rel="stylesheet" />
<script src="libs/bootstrap-3.3.1/js/bootstrap.min.js"></script>
<script src="libs/bootstrap-3.3.1/shim/html5shiv.min.js"></script>
<script src="libs/bootstrap-3.3.1/shim/respond.min.js"></script>

<style type="text/css">

/* padding for bootstrap navbar */
body {
  padding-top: 50px;
  padding-bottom: 40px;
}


/* offset scroll position for anchor links (for fixed navbar)  */
.section h2 {
  padding-top: 55px;
  margin-top: -55px;
}
.section h3 {
  padding-top: 55px;
  margin-top: -55px;
}



/* don't use link color in navbar */
.dropdown-menu>li>a {
  color: black;
}

/* some padding for disqus */
#disqus_thread {
  margin-top: 45px;
}

</style>

<link rel="stylesheet" href="libs/font-awesome-4.1.0/css/font-awesome.min.css"/>

<style type="text/css">code{white-space: pre;}</style>
<link rel="stylesheet"
      href="libs/highlight/textmate.css"
      type="text/css" />
<script src="libs/highlight/highlight.js"></script>
<style type="text/css">
  pre:not([class]) {
    background-color: white;
  }
</style>
<script type="text/javascript">
if (window.hljs && document.readyState && document.readyState === "complete") {
   window.setTimeout(function() {
      hljs.initHighlighting();
   }, 0);
}
</script>



</head>

<body>

<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
code {
  color: inherit;
  background-color: rgba(0, 0, 0, 0.04);
}
img { 
  max-width:100%; 
  height: auto; 
}
</style>
<div class="container-fluid main-container">


<div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">singleCellSeq</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li><a href="index.html">Home</a></li>
        <li><a href="about.html">About</a></li>
        <li><a href="license.html">License</a></li>
        <li><a href="https://github.com/jdblischak/singleCellSeq">GitHub</a></li>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<div id="header">
<h1 class="title">Quality control for LCLs</h1>
<h4 class="date"><em>2015-07-10</em></h4>
</div>

<div id="TOC">
<ul>
<li><a href="#input">Input</a></li>
<li><a href="#total-mapped-reads-unmapped-ratios-and-ercc-ratios">Total mapped reads, unmapped ratios, and ERCC ratios</a></li>
<li><a href="#cell-number">Cell number</a></li>
<li><a href="#number-of-genes-detected">Number of genes detected</a></li>
<li><a href="#reads-mapped-to-mitochondrial-genes">Reads mapped to mitochondrial genes</a></li>
<li><a href="#pca">PCA</a></li>
<li><a href="#cutoffs">Cutoffs</a></li>
<li><a href="#session-information">Session information</a></li>
</ul>
</div>

<p><strong>Last updated:</strong> 2015-07-21</p>
<p><strong>Code version:</strong> c81003a9f8c7df2925a675f7e59f32b4a1c0833a</p>
<p>This is the quality control at single cell level of 19239 LCLs. 96 cells were collected on a C1 prep. I only purchased enough Tn5 from Epicentre for the generation of transposomes with 24 different barcodes. As a results, the 96 cells were divided into four groups, each have 24 cells, for library preps (4 batches) and for sequencing (4 lanes). Additionally, 4 individual cells collected from a different C1 prep were sujected to library preps using either the Epicentre Tn5 (A9E1 and B2E2) or the home-made Tn5 (B4H1 and D2H2).</p>
<div id="input" class="section level2">
<h2>Input</h2>
<pre class="r"><code>library(&quot;dplyr&quot;)
library(&quot;ggplot2&quot;)
theme_set(theme_bw(base_size = 16))
library(&quot;edgeR&quot;)</code></pre>
<pre class="r"><code>counts &lt;- read.table(&quot;/mnt/gluster/data/internal_supp/singleCellSeq/lcl/gene-counts-lcl.txt&quot;,
                     header = TRUE, sep = &quot;\t&quot;, stringsAsFactors = FALSE)</code></pre>
<pre class="r"><code>anno &lt;- counts %&gt;%
  filter(rmdup == &quot;molecules&quot;) %&gt;%
  select(individual:well) %&gt;%
  arrange(well)
anno &lt;- mutate(anno, sample_id = paste(paste0(&quot;NA&quot;, individual),
                                       batch, well, sep = &quot;.&quot;))
anno &lt;- mutate(anno, full_lane = well %in% c(&quot;A9E1&quot;, &quot;B2E2&quot;, &quot;B4H1&quot;, &quot;D2H2&quot;))
stopifnot(sum(anno$full_lane) == 4)
write.table(anno, &quot;../data/annotation-lcl.txt&quot;, quote = FALSE, sep = &quot;\t&quot;,
            row.names = FALSE)
head(anno)</code></pre>
<pre><code>  individual batch well     sample_id full_lane
1      19239     1  A01 NA19239.1.A01     FALSE
2      19239     1  A02 NA19239.1.A02     FALSE
3      19239     1  A03 NA19239.1.A03     FALSE
4      19239     1  A04 NA19239.1.A04     FALSE
5      19239     1  A05 NA19239.1.A05     FALSE
6      19239     1  A06 NA19239.1.A06     FALSE</code></pre>
<pre class="r"><code>molecules &lt;- counts %&gt;%
  arrange(well) %&gt;%
  filter(rmdup == &quot;molecules&quot;) %&gt;%
  select(-(individual:rmdup)) %&gt;%
  t
dim(molecules)</code></pre>
<pre><code>[1] 20419   100</code></pre>
<pre class="r"><code>colnames(molecules) &lt;- anno$sample_id
# Fix ERCC names. A data table can have dashes in column names, but data frame
# converts to period. Since the iPSC data was read with fread from the
# data.table package in sum-counts-per-sample.Rmd, this was not a problem
# before.
rownames(molecules) &lt;- sub(&quot;\\.&quot;, &quot;-&quot;, rownames(molecules))
write.table(molecules, &quot;../data/molecules-lcl.txt&quot;, quote = FALSE, sep = &quot;\t&quot;,
            col.names = NA)
molecules[1:10, 1:5]</code></pre>
<pre><code>                NA19239.1.A01 NA19239.1.A02 NA19239.1.A03 NA19239.1.A04
ENSG00000186092             0             0             0             0
ENSG00000237683             0             0             0             0
ENSG00000235249             0             0             0             0
ENSG00000185097             0             0             0             0
ENSG00000269831             0             0             0             0
ENSG00000269308             0             0             0             0
ENSG00000187634             0             0             0             0
ENSG00000268179             0             0             0             0
ENSG00000188976             0             0             0             0
ENSG00000187961             0             0             0             0
                NA19239.1.A05
ENSG00000186092             0
ENSG00000237683             0
ENSG00000235249             0
ENSG00000185097             0
ENSG00000269831             0
ENSG00000269308             0
ENSG00000187634             0
ENSG00000268179             0
ENSG00000188976             0
ENSG00000187961             0</code></pre>
<pre class="r"><code>reads &lt;- counts %&gt;%
  arrange(well) %&gt;%
  filter(rmdup == &quot;reads&quot;) %&gt;%
  select(-(individual:rmdup)) %&gt;%
  t
dim(reads)</code></pre>
<pre><code>[1] 20419   100</code></pre>
<pre class="r"><code>colnames(reads) &lt;- anno$sample_id
rownames(reads) &lt;- sub(&quot;\\.&quot;, &quot;-&quot;, rownames(reads))
write.table(reads, &quot;../data/reads-lcl.txt&quot;, quote = FALSE, sep = &quot;\t&quot;,
            col.names = NA)
reads[1:10, 1:5]</code></pre>
<pre><code>                NA19239.1.A01 NA19239.1.A02 NA19239.1.A03 NA19239.1.A04
ENSG00000186092             0             0             0             0
ENSG00000237683             0             0             0             0
ENSG00000235249             0             0             0             0
ENSG00000185097             0             0             0             0
ENSG00000269831             0             0             0             0
ENSG00000269308             0             0             0             0
ENSG00000187634             0             0             0             0
ENSG00000268179             0             0             0             0
ENSG00000188976             0             0             0             0
ENSG00000187961             0             0             0             0
                NA19239.1.A05
ENSG00000186092             0
ENSG00000237683             0
ENSG00000235249             0
ENSG00000185097             0
ENSG00000269831             0
ENSG00000269308             0
ENSG00000187634             0
ENSG00000268179             0
ENSG00000188976             0
ENSG00000187961             0</code></pre>
<p>Summary counts from featureCounts. Created with <a href="https://github.com/jdblischak/singleCellSeq/blob/master/code/gather-summary-counts.py">gather-summary-counts.py</a>. These data were collected from the summary files of the full combined samples.</p>
<pre class="r"><code>summary_counts &lt;- read.table(&quot;../data/summary-counts-lcl.txt&quot;, header = TRUE,
                             stringsAsFactors = FALSE)</code></pre>
<p>Currently this file only contains data from sickle-trimmed reads, so the code below simply ensures this and then removes the column.</p>
<pre class="r"><code>summary_per_sample &lt;- summary_counts %&gt;%
  filter(sickle == &quot;quality-trimmed&quot;) %&gt;%
  select(-sickle) %&gt;%
  arrange(individual, batch, well, rmdup)
stopifnot(summary_per_sample$well[c(TRUE, FALSE)] == anno$well)</code></pre>
<p>Input single cell observational quality control data.</p>
<pre class="r"><code># File needs to be created
qc &lt;- read.csv(&quot;../data/qc-lcl.csv&quot;, header = TRUE,
                 stringsAsFactors = FALSE)
head(qc)</code></pre>
<pre><code>  X ll_name cell.number.location cell.num
1 1     A01                1_C03        1
2 2     A02                1_C02        1
3 3     A03                0_C01        0
4 4     A04                1_C49        1
5 5     A05                1_C50        1
6 6     A06                1_C51        1</code></pre>
</div>
<div id="total-mapped-reads-unmapped-ratios-and-ercc-ratios" class="section level2">
<h2>Total mapped reads, unmapped ratios, and ERCC ratios</h2>
<p>Looking at the unmapped ratio and ERCC ratio of each cell based on number of reads.</p>
<pre class="r"><code># reads per sample
summary_per_sample_reads &lt;- summary_per_sample %&gt;% filter(rmdup == &quot;reads&quot;)

# create unmapped ratios
summary_per_sample_reads$unmapped.ratios &lt;- summary_per_sample_reads[,9]/apply(summary_per_sample_reads[,5:13],1,sum)

# create total mapped reads
summary_per_sample_reads$total.mapped &lt;- apply(summary_per_sample_reads[,5:8],1,sum)

# plot
ggplot(summary_per_sample_reads, aes(x = total.mapped, y = unmapped.ratios, col = as.factor(individual), shape = as.factor(batch), label = well)) + geom_point(size = 3, alpha = 0.5) + geom_text()</code></pre>
<p><img src="figure/qc-cell-lcl.Rmd/reads-1.png" title="" alt="" width="768" style="display: block; margin: auto;" /></p>
<pre class="r"><code># plot the sum of reads and &#39;Assigned&#39;
plot(apply(reads,2,sum),summary_per_sample_reads[,5])</code></pre>
<p><img src="figure/qc-cell-lcl.Rmd/reads-2.png" title="" alt="" width="768" style="display: block; margin: auto;" /></p>
<pre class="r"><code># total ERCC reads 
summary_per_sample_reads$total.ERCC &lt;- apply(reads[grep(&quot;ERCC&quot;, rownames(reads)), ],2,sum)

plot(summary_per_sample_reads$total.ERCC)</code></pre>
<p><img src="figure/qc-cell-lcl.Rmd/reads-3.png" title="" alt="" width="768" style="display: block; margin: auto;" /></p>
<pre class="r"><code># creat ERCC ratios
summary_per_sample_reads$ERCC.ratios &lt;- apply(reads[grep(&quot;ERCC&quot;, rownames(reads)), ],2,sum)/apply(summary_per_sample_reads[,5:8],1,sum)

# plot
ggplot(summary_per_sample_reads, aes(x = total.mapped, y = ERCC.ratios, col = as.factor(individual), shape = as.factor(batch), label = well)) + geom_point(size = 3, alpha = 0.5) + geom_text(angle = 45)</code></pre>
<p><img src="figure/qc-cell-lcl.Rmd/reads-4.png" title="" alt="" width="768" style="display: block; margin: auto;" /></p>
<p>Looking at only the multiplexed single cell libraries (96 samples total, 24 each in lanes 1-4):</p>
<pre class="r"><code># remove the 4 individual cells
summary_per_sample_reads_single &lt;- summary_per_sample_reads[!anno$full_lane, ]

# many plots
ggplot(summary_per_sample_reads_single, aes(x = total.mapped, y = unmapped.ratios, col = as.factor(individual), shape = as.factor(batch))) + geom_point(size = 3, alpha = 0.5) + xlab(&quot;Number of mapped reads&quot;) + ylab(&quot;Umapped reads ratio&quot;) + theme(axis.text.x = element_text(angle = 90, hjust = 0.9, vjust = 0.5))</code></pre>
<p><img src="figure/qc-cell-lcl.Rmd/reads-single-1.png" title="" alt="" width="768" style="display: block; margin: auto;" /></p>
<pre class="r"><code>ggplot(summary_per_sample_reads_single, aes(x = total.mapped, y = ERCC.ratios, col = as.factor(individual), shape = as.factor(batch))) + geom_point(size = 3, alpha = 0.5) + xlab(&quot;Number of mapped reads&quot;) + ylab(&quot;Spike-in reads ratio&quot;) + theme(axis.text.x = element_text(angle = 90, hjust = 0.9, vjust = 0.5))</code></pre>
<p><img src="figure/qc-cell-lcl.Rmd/reads-single-2.png" title="" alt="" width="768" style="display: block; margin: auto;" /></p>
</div>
<div id="cell-number" class="section level2">
<h2>Cell number</h2>
<p>The cell number of each capture site is recorded after the cell corting step and before the cells got lysed.</p>
<pre class="r"><code>#add cell number per well by merging qc file
summary_per_sample_reads_single_qc &lt;- summary_per_sample_reads_single
summary_per_sample_reads_single_qc$cell_number &lt;- qc$cell.num

ggplot(summary_per_sample_reads_single_qc, aes(x = total.mapped, y = unmapped.ratios, col = as.factor(individual), label = as.character(cell_number))) + geom_text(fontface=3)</code></pre>
<p><img src="figure/qc-cell-lcl.Rmd/cell-number-1.png" title="" alt="" width="768" style="display: block; margin: auto;" /></p>
<pre class="r"><code>ggplot(summary_per_sample_reads_single_qc, aes(x = total.mapped, y = ERCC.ratios, col = as.factor(individual), label = as.character(cell_number))) + geom_text(fontface=3)</code></pre>
<p><img src="figure/qc-cell-lcl.Rmd/cell-number-2.png" title="" alt="" width="768" style="display: block; margin: auto;" /></p>
<p>Based on the observation that these is a dinstint cell population with more than 2 million reads, we used it as a cutoff.</p>
<pre class="r"><code>#qc filter only keep cells with more than 2 million reads
summary_per_sample_reads_single_qc$qc_filter &lt;- summary_per_sample_reads_single_qc$cell_number == 1 &amp; summary_per_sample_reads_single_qc$total.mapped &gt; 2 * 10^6

ggplot(summary_per_sample_reads_single_qc, aes(x = total.mapped, y = unmapped.ratios, col = qc_filter, label = as.character(cell_number))) + geom_text(fontface=3) + facet_grid(individual ~ batch) + theme(axis.text.x = element_text(angle = 90, hjust = 0.9, vjust = 0.5)) + xlab(&quot;Number of mapped reads&quot;) + ylab(&quot;Umapped reads ratio&quot;)</code></pre>
<p><img src="figure/qc-cell-lcl.Rmd/retained-cell-1.png" title="" alt="" width="768" style="display: block; margin: auto;" /></p>
<pre class="r"><code>ggplot(summary_per_sample_reads_single_qc, aes(x = total.mapped, y = ERCC.ratios, col = qc_filter, label = as.character(cell_number))) + geom_text(fontface=3) + facet_grid(individual ~ batch) + theme(axis.text.x = element_text(angle = 90, hjust = 0.9, vjust = 0.5)) + xlab(&quot;Number of mapped reads&quot;) + ylab(&quot;Spike-in reads ratio&quot;)</code></pre>
<p><img src="figure/qc-cell-lcl.Rmd/retained-cell-2.png" title="" alt="" width="768" style="display: block; margin: auto;" /></p>
</div>
<div id="number-of-genes-detected" class="section level2">
<h2>Number of genes detected</h2>
<p>Number of genes deteced in LCLS are smaller than iPSCs!!!</p>
<pre class="r"><code>## remove genes with no read
expressed &lt;- rowSums(reads) &gt; 0
reads &lt;- reads[expressed, ]
dim(reads)</code></pre>
<pre><code>[1] 13721   100</code></pre>
<pre class="r"><code>## number of expressed gene in each cell
reads_single &lt;- reads[, anno$full_lane == &quot;FALSE&quot;]
reads_single_gene_number &lt;- colSums(reads_single &gt; 1)
summary_per_sample_reads_single_qc$reads_single_gene_number &lt;- reads_single_gene_number

ggplot(summary_per_sample_reads_single_qc, aes(x = total.mapped, y = reads_single_gene_number, col = qc_filter, label = as.character(cell_number))) + geom_text(fontface=3) + facet_grid(individual ~ batch) + theme(axis.text.x = element_text(angle = 90, hjust = 0.9, vjust = 0.5)) + xlab(&quot;Number of mapped reads&quot;) + ylab(&quot;Number of genes&quot;)</code></pre>
<p><img src="figure/qc-cell-lcl.Rmd/gene-number-1.png" title="" alt="" width="768" style="display: block; margin: auto;" /></p>
<pre class="r"><code>ggplot(summary_per_sample_reads_single_qc, aes(x = reads_single_gene_number, y = ERCC.ratios, col = qc_filter, label = as.character(cell_number))) + geom_text(fontface=3) + facet_grid(individual ~ batch) + theme(axis.text.x = element_text(angle = 90, hjust = 0.9, vjust = 0.5)) + xlab(&quot;Number of genes&quot;) + ylab(&quot;Spike-in reads ratio&quot;)</code></pre>
<p><img src="figure/qc-cell-lcl.Rmd/gene-number-2.png" title="" alt="" width="768" style="display: block; margin: auto;" /></p>
<p>What are the numbers of detected genes of the 4 heavily-sequenced individual cells?</p>
<pre class="r"><code>## the 4 individual cells
reads_4 &lt;- reads[, anno$full_lane == &quot;TRUE&quot;]
reads_4_gene_number &lt;- colSums(reads_4 &gt; 1)
reads_4_gene_number</code></pre>
<pre><code>NA19239.1.A9E1 NA19239.1.B2E2 NA19239.1.B4H1 NA19239.1.D2H2 
          5896           6911           6025           5783 </code></pre>
</div>
<div id="reads-mapped-to-mitochondrial-genes" class="section level2">
<h2>Reads mapped to mitochondrial genes</h2>
<pre class="r"><code>## create a list of mitochondrial genes (13 protein-coding genes)
## MT-ATP6, MT-CYB, MT-ND1, MT-ND4, MT-ND4L, MT-ND5, MT-ND6, MT-CO2, MT-CO1, MT-ND2, MT-ATP8, MT-CO3, MT-ND3
mtgene &lt;- c(&quot;ENSG00000198899&quot;, &quot;ENSG00000198727&quot;, &quot;ENSG00000198888&quot;, &quot;ENSG00000198886&quot;, &quot;ENSG00000212907&quot;, &quot;ENSG00000198786&quot;, &quot;ENSG00000198695&quot;, &quot;ENSG00000198712&quot;, &quot;ENSG00000198804&quot;, &quot;ENSG00000198763&quot;,&quot;ENSG00000228253&quot;, &quot;ENSG00000198938&quot;, &quot;ENSG00000198840&quot;)

## reads of mt genes in single cells
mt_reads &lt;- reads_single[mtgene,]
dim(mt_reads)</code></pre>
<pre><code>[1] 13 96</code></pre>
<pre class="r"><code>## mt ratio of single cell
mt_reads_total &lt;- apply(mt_reads, 2, sum)
summary_per_sample_reads_single_qc$mt_reads_total &lt;- mt_reads_total
summary_per_sample_reads_single_qc$mt_reads_ratio &lt;- summary_per_sample_reads_single_qc$mt_reads_total/summary_per_sample_reads_single_qc$total.mapped

ggplot(summary_per_sample_reads_single_qc, aes(x = total.mapped, y = mt_reads_ratio, col = qc_filter, label = as.character(cell_number))) + geom_text(fontface=3) + facet_grid(individual ~ batch) + theme(axis.text.x = element_text(angle = 90, hjust = 0.9, vjust = 0.5)) + xlab(&quot;Number of mapped reads&quot;) + ylab(&quot;Mitochondrial ratio&quot;)</code></pre>
<p><img src="figure/qc-cell-lcl.Rmd/mitochondrial-1.png" title="" alt="" width="768" style="display: block; margin: auto;" /></p>
<pre class="r"><code>ggplot(summary_per_sample_reads_single_qc, aes(x = mt_reads_ratio, y = ERCC.ratios, col = qc_filter, label = as.character(cell_number))) + geom_text(fontface=3) + facet_grid(individual ~ batch) + theme(axis.text.x = element_text(angle = 90, hjust = 0.9, vjust = 0.5)) + xlab(&quot;Mitochondrial ratio&quot;) + ylab(&quot;Spike-in reads ratio&quot;)</code></pre>
<p><img src="figure/qc-cell-lcl.Rmd/mitochondrial-2.png" title="" alt="" width="768" style="display: block; margin: auto;" /></p>
<pre class="r"><code>ggplot(summary_per_sample_reads_single_qc, aes(x = reads_single_gene_number, y = mt_reads_ratio, col = qc_filter, label = as.character(cell_number))) + geom_text(fontface=3) + facet_grid(individual ~ batch) + theme(axis.text.x = element_text(angle = 90, hjust = 0.9, vjust = 0.5)) + xlab(&quot;Number of genes&quot;) + ylab(&quot;Mitochondrial ratio&quot;)</code></pre>
<p><img src="figure/qc-cell-lcl.Rmd/mitochondrial-3.png" title="" alt="" width="768" style="display: block; margin: auto;" /></p>
</div>
<div id="pca" class="section level2">
<h2>PCA</h2>
<p>Calculate the counts per million for the singe cells.</p>
<pre class="r"><code>reads_cells &lt;- reads_single
reads_cells_cpm &lt;- cpm(reads_cells)</code></pre>
<p>Select the most variable genes.</p>
<pre class="r"><code>reads_cells_cpm_log_var &lt;- log(apply(reads_cells_cpm, 1, var))
hist(reads_cells_cpm_log_var)</code></pre>
<p><img src="figure/qc-cell-lcl.Rmd/variable-genes-1.png" title="" alt="" width="768" style="display: block; margin: auto;" /></p>
<pre class="r"><code>sum(reads_cells_cpm_log_var &gt; 8)</code></pre>
<pre><code>[1] 6473</code></pre>
<p>Using the 6473 most variable genes, perform PCA.</p>
<pre class="r"><code>reads_cells_cpm &lt;- reads_cells_cpm[reads_cells_cpm_log_var &gt; 8, ]
pca_reads_cells &lt;- prcomp(t(reads_cells_cpm), retx = TRUE, scale. = TRUE,
                          center = TRUE)</code></pre>
<pre class="r"><code>plot(pca_reads_cells)</code></pre>
<p><img src="figure/qc-cell-lcl.Rmd/perc-explained-1.png" title="" alt="" width="768" style="display: block; margin: auto;" /></p>
<pre class="r"><code>pca_reads_cells$perc_explained &lt;- pca_reads_cells$sdev^2 / sum(pca_reads_cells$sdev^2) * 100
plot(pca_reads_cells$perc_explained)</code></pre>
<p><img src="figure/qc-cell-lcl.Rmd/perc-explained-2.png" title="" alt="" width="768" style="display: block; margin: auto;" /></p>
<p>The first PC accounts for 5% of the variance and the second PC 3%.</p>
<pre class="r"><code>stopifnot(colnames(reads_cells) == 
            paste(paste0(&quot;NA&quot;, summary_per_sample_reads_single_qc$individual),
                   summary_per_sample_reads_single_qc$batch,
                   summary_per_sample_reads_single_qc$well, sep = &quot;.&quot;))
pca_reads_cells_anno &lt;- cbind(summary_per_sample_reads_single_qc, pca_reads_cells$x)</code></pre>
<pre class="r"><code>ggplot(pca_reads_cells_anno, aes(x = PC1, y = PC2, col = as.factor(individual),
                                 shape = as.factor(batch))) +
  geom_point()</code></pre>
<p><img src="figure/qc-cell-lcl.Rmd/pca-plot-1.png" title="" alt="" width="768" style="display: block; margin: auto;" /></p>
</div>
<div id="cutoffs" class="section level2">
<h2>Cutoffs</h2>
<p>Using various simple filtering cutoffs.</p>
<pre class="r"><code>pca_reads_cells_anno$cell_filter &lt;- pca_reads_cells_anno$cell_number == 1
ggplot(pca_reads_cells_anno, aes(x = PC1, y = PC2, col = cell_filter)) +
  geom_point()</code></pre>
<p><img src="figure/qc-cell-lcl.Rmd/filter-cell-num-1.png" title="" alt="" width="768" style="display: block; margin: auto;" /></p>
<pre class="r"><code>pca_reads_cells_anno$total_cutoff &lt;- pca_reads_cells_anno$total.mapped &gt; 1.5 * 10^6
ggplot(pca_reads_cells_anno, aes(x = PC1, y = PC2, col = total_cutoff)) +
  geom_point()</code></pre>
<p><img src="figure/qc-cell-lcl.Rmd/filter-total-1.png" title="" alt="" width="768" style="display: block; margin: auto;" /></p>
<pre class="r"><code>pca_reads_cells_anno$unmapped_cutoff &lt;- pca_reads_cells_anno$unmapped.ratios &lt; 0.4
ggplot(pca_reads_cells_anno, aes(x = PC1, y = PC2, col = unmapped_cutoff)) +
  geom_point()</code></pre>
<p><img src="figure/qc-cell-lcl.Rmd/filter-unmapped-1.png" title="" alt="" width="768" style="display: block; margin: auto;" /></p>
<pre class="r"><code>pca_reads_cells_anno$ERCC_cutoff &lt;- pca_reads_cells_anno$ERCC.ratios &lt; 0.05
ggplot(pca_reads_cells_anno, aes(x = PC1, y = PC2, col = ERCC_cutoff)) +
  geom_point()</code></pre>
<p><img src="figure/qc-cell-lcl.Rmd/filter-ERCC-1.png" title="" alt="" width="768" style="display: block; margin: auto;" /></p>
<pre class="r"><code>pca_reads_cells_anno$gene_filter &lt;- pca_reads_cells_anno$reads_single_gene_number  &gt; 3000
ggplot(pca_reads_cells_anno, aes(x = PC1, y = PC2, col = gene_filter)) +
  geom_point()</code></pre>
<p><img src="figure/qc-cell-lcl.Rmd/filter-gene-num-1.png" title="" alt="" width="768" style="display: block; margin: auto;" /></p>
<pre class="r"><code>pca_reads_cells_anno$mt_filter &lt;- pca_reads_cells_anno$mt_reads_ratio &lt; 0.15
ggplot(pca_reads_cells_anno, aes(x = PC1, y = PC2, col = mt_filter)) +
  geom_point()</code></pre>
<p><img src="figure/qc-cell-lcl.Rmd/filter-mt-genes-1.png" title="" alt="" width="768" style="display: block; margin: auto;" /></p>
<p>The two cutoffs, total reads and cell number, largely overlap.</p>
<pre class="r"><code>table(pca_reads_cells_anno$cell_filter, pca_reads_cells_anno$total_cutoff,
      dnn = c(&quot;Num cells == 1&quot;, &quot;Total reads &gt; 1.5e6&quot;))</code></pre>
<pre><code>              Total reads &gt; 1.5e6
Num cells == 1 FALSE TRUE
         FALSE     8    4
         TRUE      5   79</code></pre>
<p>Add the third cutoff, 3000 genes</p>
<pre class="r"><code>pca_reads_cells_anno$qc_filter &lt;- pca_reads_cells_anno$total_cutoff &amp;
                                  pca_reads_cells_anno$gene_filter &amp;
                                  pca_reads_cells_anno$cell_filter
  
ggplot(pca_reads_cells_anno, aes(x = PC1, y = PC2, col = qc_filter,
                                 label = as.character(cell_number))) +
  geom_text(fontface=3)</code></pre>
<p><img src="figure/qc-cell-lcl.Rmd/filter-total-cell-num-1.png" title="" alt="" width="768" style="display: block; margin: auto;" /></p>
<p>Apply all the cutoffs</p>
<pre class="r"><code>pca_reads_cells_anno$qc_filter_all &lt;- pca_reads_cells_anno$cell_filter &amp;
                                      pca_reads_cells_anno$total_cutoff &amp;
                                      pca_reads_cells_anno$unmapped_cutoff &amp;
                                      pca_reads_cells_anno$ERCC_cutoff &amp;
                                      pca_reads_cells_anno$gene_filter &amp;
                                      pca_reads_cells_anno$mt_filter
  
ggplot(pca_reads_cells_anno, aes(x = PC1, y = PC2, col = qc_filter_all,
                                 label = as.character(cell_number))) +
  geom_text(fontface=3)</code></pre>
<p><img src="figure/qc-cell-lcl.Rmd/filter-all-1.png" title="" alt="" width="768" style="display: block; margin: auto;" /></p>
<p>How many cells do we keep from each individual and batch using this filter?</p>
<pre class="r"><code>table(pca_reads_cells_anno[pca_reads_cells_anno$qc_filter,
                           c(&quot;individual&quot;, &quot;batch&quot;)])</code></pre>
<pre><code>          batch
individual  1
     19239 76</code></pre>
<pre class="r"><code>table(pca_reads_cells_anno[pca_reads_cells_anno$qc_filter_all,
                           c(&quot;individual&quot;, &quot;batch&quot;)])</code></pre>
<pre><code>          batch
individual  1
     19239 75</code></pre>
<p>Output list of single cells to keep.</p>
<pre class="r"><code>stopifnot(nrow(pca_reads_cells_anno) == nrow(anno[anno$full_lane == &quot;FALSE&quot;, ]))
quality_single_cells &lt;- anno %&gt;%
  filter(full_lane == &quot;FALSE&quot;) %&gt;%
  filter(pca_reads_cells_anno$qc_filter_all) %&gt;%
  select(sample_id)
stopifnot(!grepl(&quot;TRUE&quot;, quality_single_cells$sample_id))
write.table(quality_single_cells,
            file = &quot;../data/quality-single-cells-lcl.txt&quot;, quote = FALSE,
            sep = &quot;\t&quot;, row.names = FALSE, col.names = FALSE)</code></pre>
</div>
<div id="session-information" class="section level2">
<h2>Session information</h2>
<pre class="r"><code>sessionInfo()</code></pre>
<pre><code>R version 3.2.0 (2015-04-16)
Platform: x86_64-unknown-linux-gnu (64-bit)

locale:
 [1] LC_CTYPE=en_US.UTF-8       LC_NUMERIC=C              
 [3] LC_TIME=en_US.UTF-8        LC_COLLATE=en_US.UTF-8    
 [5] LC_MONETARY=en_US.UTF-8    LC_MESSAGES=en_US.UTF-8   
 [7] LC_PAPER=en_US.UTF-8       LC_NAME=C                 
 [9] LC_ADDRESS=C               LC_TELEPHONE=C            
[11] LC_MEASUREMENT=en_US.UTF-8 LC_IDENTIFICATION=C       

attached base packages:
[1] stats     graphics  grDevices utils     datasets  methods   base     

other attached packages:
[1] edgeR_3.10.2  limma_3.24.9  ggplot2_1.0.1 dplyr_0.4.1   knitr_1.10.5 

loaded via a namespace (and not attached):
 [1] Rcpp_0.11.6      magrittr_1.5     MASS_7.3-40      munsell_0.4.2   
 [5] colorspace_1.2-6 stringr_1.0.0    plyr_1.8.2       tools_3.2.0     
 [9] parallel_3.2.0   grid_3.2.0       gtable_0.1.2     DBI_0.3.1       
[13] htmltools_0.2.6  lazyeval_0.1.10  yaml_2.1.13      assertthat_0.1  
[17] digest_0.6.8     reshape2_1.4.1   formatR_1.2      evaluate_0.7    
[21] rmarkdown_0.6.1  labeling_0.3     stringi_0.4-1    scales_0.2.4    
[25] proto_0.3-10    </code></pre>
</div>


<!-- some extra javascript for older browsers -->
<script type="text/javascript" src="libs/polyfill.js"></script>

<script>

// manage active state of menu based on current page
$(document).ready(function () {

    // active menu
    href = window.location.pathname
    href = href.substr(href.lastIndexOf('/') + 1)
    $('a[href="' + href + '"]').parent().addClass('active');

    // manage active menu header
    if (href.startsWith('authoring_'))
      $('a[href="' + 'authoring' + '"]').parent().addClass('active');
    else if (href.endsWith('_format.html'))
      $('a[href="' + 'formats' + '"]').parent().addClass('active');
    else if (href.startsWith('developer_'))
      $('a[href="' + 'developer' + '"]').parent().addClass('active');

});

</script>

</div>

<script>

// add bootstrap table styles to pandoc tables
$(document).ready(function () {
  $('tr.header').parent('thead').parent('table').addClass('table table-condensed');
});

</script>


</body>
</html>
