---
title: "Quality control for iPS cells"
output: html_document
date: '2015-05-27'
---

**Last updated:** `r Sys.Date()`

**Code version:** `r system("git log -1 --format='%H'", intern = TRUE)`

```{r chunk-options, include=FALSE}
source("chunk-options.R")
opts_chunk$set(fig.width = 8, fig.height = 8)
```

## Input

```{r packages, message=FALSE}
library("dplyr")
library("ggplot2")
theme_set(theme_bw(base_size = 16))
library("edgeR")
```

Summary counts from featureCounts.
Created with [gather-summary-counts.py](https://github.com/jdblischak/singleCellSeq/blob/master/code/gather-summary-counts.py).
These data were collected from the summary files of the full combined samples.

```{r input-summary-counts}
summary_counts <- read.table("../data/summary-counts.txt", header = TRUE,
                             stringsAsFactors = FALSE)
```

Currently this file only contains data from sickle-trimmed reads, so the code below simply ensures this and then removes the column.

```{r clean-summary-counts}
summary_per_sample <- summary_counts %>%
  filter(sickle == "quality-trimmed") %>%
  select(-sickle) %>%
  arrange(individual, batch, well, rmdup) %>%
  as.data.frame
```

Input annotation.

```{r input-annotation}
anno <- read.table("../data/annotation.txt", header = TRUE,
                   stringsAsFactors = FALSE)
head(anno)
```

Input read counts.

```{r input-read-counts}
reads <- read.table("../data/reads.txt", header = TRUE,
                    stringsAsFactors = FALSE)
```

Input molecule counts.

```{r input-molecule-counts}
molecules <- read.table("../data/molecules.txt", header = TRUE,
                    stringsAsFactors = FALSE)
```

Input single cell observational quality control data.

```{r input-qc}
qc <- read.table("../data/qc-ipsc.txt", header = TRUE,
                 stringsAsFactors = FALSE)
head(qc)
```

## total mapped reads, unmapped ratios, and ERCC ratios

Looking at the unmapped ratio and ERCC ratio of each cell based on number of reads.

1. low number of mapped reads might be caused by cell apoptosis, RNA degradation, insufficient cell lysis, bad Tn5 transposase quality, or contamination
2. high proportion of ERCC spike-in indicates low RNA quantity from iPSCs which might cell death

```{r reads}
# reads per sample
summary_per_sample_reads <- summary_per_sample %>% filter(rmdup == "reads")

# create unmapped ratios
summary_per_sample_reads$unmapped.ratios <- summary_per_sample_reads[,9]/apply(summary_per_sample_reads[,5:13],1,sum)

# create total mapped reads
summary_per_sample_reads$total.mapped <- apply(summary_per_sample_reads[,5:8],1,sum)

# plot
ggplot(summary_per_sample_reads, aes(x = total.mapped, y = unmapped.ratios, col = as.factor(individual), shape = as.factor(batch))) + geom_point(size = 3, alpha = 0.5)

# plot the sum of reads and 'Assigned'
plot(apply(reads,2,sum),summary_per_sample_reads[,5])

# total ERCC reads 
summary_per_sample_reads$total.ERCC <- apply(reads[grep("ERCC", rownames(reads)), ],2,sum)

plot(summary_per_sample_reads$total.ERCC)

# creat ERCC ratios
summary_per_sample_reads$ERCC.ratios <- apply(reads[grep("ERCC", rownames(reads)), ],2,sum)/apply(summary_per_sample_reads[,5:8],1,sum)

# plot
ggplot(summary_per_sample_reads, aes(x = total.mapped, y = ERCC.ratios, col = as.factor(individual), shape = as.factor(batch))) + geom_point(size = 3, alpha = 0.5)
```

Looking at only the single cell libraries
```{r reads-single}
# remove bulk keep single cell
summary_per_sample_reads_single <- summary_per_sample_reads[summary_per_sample_reads$well!="bulk",]

# so mamy plots
ggplot(summary_per_sample_reads_single, aes(x = total.mapped, y = unmapped.ratios, col = as.factor(individual), shape = as.factor(batch))) + geom_point(size = 3, alpha = 0.5) + xlab("Number of mapped reads") + ylab("Umapped reads ratio")

ggplot(summary_per_sample_reads_single, aes(x = total.mapped, y = ERCC.ratios, col = as.factor(individual), shape = as.factor(batch))) + geom_point(size = 3, alpha = 0.5) + xlab("Number of mapped reads") + ylab("Spike-in reads ratio")

ggplot(summary_per_sample_reads_single, aes(x = total.mapped, y = unmapped.ratios, col = as.factor(individual), shape = as.factor(batch))) + geom_point(size = 3, alpha = 0.5) + facet_grid(individual ~ batch) + theme(axis.text.x = element_text(angle = 90, hjust = 0.9, vjust = 0.5))

ggplot(summary_per_sample_reads_single, aes(x = total.mapped, y = ERCC.ratios, col = as.factor(individual), shape = as.factor(batch))) + geom_point(size = 3, alpha = 0.5) + facet_grid(individual ~ batch) + theme(axis.text.x = element_text(angle = 90, hjust = 0.9, vjust = 0.5))

ggplot(summary_per_sample_reads_single, aes(x = total.mapped, y = total.ERCC, col = as.factor(individual), shape = as.factor(batch))) + geom_point(size = 3, alpha = 0.5) + facet_grid(individual ~ batch) + theme(axis.text.x = element_text(angle = 90, hjust = 0.9, vjust = 0.5))
```

### adjust the ERCC total reads of 19098 batch 2

It is obvious that the 19098 batch 2 have a lot more ERCC than all the other. Here is trying to fix it by adjusting the ERCC of the super sad batch.
```{r 19098-2}
summary_per_sample_reads_single$index_19098_2 <- (summary_per_sample_reads_single$individual=="19098" & summary_per_sample_reads_single$batch=="2")

#calculating the ratio of 19098 batch 2 to the rest
adjusted_ratio <- mean(summary_per_sample_reads_single$total.ERCC[summary_per_sample_reads_single$index_19098_2])/mean(summary_per_sample_reads_single$total.ERCC[!summary_per_sample_reads_single$index_19098_2])

#calculating the pippetting error rate before 21 cycle amplification
adjusted_ratio^(1/22)

#adjusted total ERCC reads
summary_per_sample_reads_single$adj.total.ERCC <- summary_per_sample_reads_single$total.ERCC  

summary_per_sample_reads_single$adj.total.ERCC[summary_per_sample_reads_single$index_19098_2] <- summary_per_sample_reads_single$adj.total.ERCC[summary_per_sample_reads_single$index_19098_2]/adjusted_ratio

#adjust ERCC ratio
summary_per_sample_reads_single$adj.ERCC.ratios <-
summary_per_sample_reads_single$adj.total.ERCC/summary_per_sample_reads_single$total.mapped

ggplot(summary_per_sample_reads_single, aes(x = total.mapped, y = adj.ERCC.ratios, col = as.factor(individual), shape = as.factor(batch))) + geom_point(size = 3, alpha = 0.5) + facet_grid(individual ~ batch) + theme(axis.text.x = element_text(angle = 90, hjust = 0.9, vjust = 0.5))
```

### QC by cell number

The cell number of each capture site is recorded after the cell corting step and before the cells got lysed.
```{r cell-number}
#add cell number per well by merging qc file
summary_per_sample_reads_single_qc <- merge(summary_per_sample_reads_single,qc,by=c("individual","batch","well"))

ggplot(summary_per_sample_reads_single_qc, aes(x = total.mapped, y = unmapped.ratios, col = as.factor(individual), label = as.character(cell_number))) + geom_text(fontface=3)

ggplot(summary_per_sample_reads_single_qc, aes(x = total.mapped, y = ERCC.ratios, col = as.factor(individual), label = as.character(cell_number))) + geom_text(fontface=3)
```

Based on the observation that these is a dinstint cell population with more than 2 million reads, we used it as a cutoff.
```{r retained-cell}
#qc filter only keep cells with more than 2 million reads
summary_per_sample_reads_single_qc$qc_filter <- summary_per_sample_reads_single_qc$cell_number == 1 & summary_per_sample_reads_single_qc$total.mapped > 2 * 10^6

ggplot(summary_per_sample_reads_single_qc, aes(x = total.mapped, y = unmapped.ratios, col = qc_filter, label = as.character(cell_number))) + geom_text(fontface=3) + facet_grid(individual ~ batch) + theme(axis.text.x = element_text(angle = 90, hjust = 0.9, vjust = 0.5)) + xlab("Number of mapped reads") + ylab("Umapped reads ratio")

ggplot(summary_per_sample_reads_single_qc, aes(x = total.mapped, y = ERCC.ratios, col = qc_filter, label = as.character(cell_number))) + geom_text(fontface=3) + facet_grid(individual ~ batch) + theme(axis.text.x = element_text(angle = 90, hjust = 0.9, vjust = 0.5)) + xlab("Number of mapped reads") + ylab("Spike-in reads ratio")
```

We decided to apply the same cutoff for all the C1 preps. This is just a test.
```{r by-C1}
#individual 19098 batch2
C1_19098_2 <- summary_per_sample_reads_single_qc[summary_per_sample_reads_single_qc$individual == "19098" & summary_per_sample_reads_single_qc$batch =="2",]

ggplot(C1_19098_2, aes(x = total.mapped, y = unmapped.ratios, col = as.factor(individual), label = as.character(cell_number))) + geom_text(fontface=3)

ggplot(C1_19098_2, aes(x = total.mapped, y = ERCC.ratios, col = as.factor(individual), label = as.character(cell_number))) + geom_text(fontface=3)

C1_19098_2_retained <- C1_19098_2[C1_19098_2$unmapped.ratios< 0.6 & C1_19098_2$ERCC.ratios< 0.35 & C1_19098_2$cell_number == 1,]

dim(C1_19098_2_retained)
```

## Number of genes deteced
```{r gene-number}
## remove genes with no read in both single cells and bulk
expressed <- rowSums(reads[, anno$well == "bulk"]) > 0 &
             rowSums(reads[, anno$well != "bulk"]) > 0
reads <- reads[expressed, ]
dim(reads)

## number of expressed gene in each cell
reads_single <- reads[, anno$well != "bulk"]
reads_single_gene_number <- colSums(reads_single > 1)
summary_per_sample_reads_single_qc$reads_single_gene_number <- reads_single_gene_number

ggplot(summary_per_sample_reads_single_qc, aes(x = total.mapped, y = reads_single_gene_number, col = qc_filter, label = as.character(cell_number))) + geom_text(fontface=3) + facet_grid(individual ~ batch) + theme(axis.text.x = element_text(angle = 90, hjust = 0.9, vjust = 0.5)) + xlab("Number of mapped reads") + ylab("Number of genes")

ggplot(summary_per_sample_reads_single_qc, aes(x = reads_single_gene_number, y = ERCC.ratios, col = qc_filter, label = as.character(cell_number))) + geom_text(fontface=3) + facet_grid(individual ~ batch) + theme(axis.text.x = element_text(angle = 90, hjust = 0.9, vjust = 0.5)) + xlab("Number of genes") + ylab("Spike-in reads ratio")
```

## Effect of cell QC data on sequencing results

### Cell number

```{r cell-num-counts}
table(summary_per_sample_reads_single_qc$cell_number)
```

```{r cell-num-ercc}
ggplot(summary_per_sample_reads_single_qc,
       aes(x = as.factor(cell_number), y = ERCC.ratios)) +
  geom_boxplot()
```

```{r cell-num-unmapped}
ggplot(summary_per_sample_reads_single_qc,
       aes(x = as.factor(cell_number), y = unmapped.ratios)) +
  geom_boxplot()
```

### Concentration

```{r concentration-sum}
summary(summary_per_sample_reads_single_qc$concentration)
```

```{r conc-ercc}
ggplot(summary_per_sample_reads_single_qc,
       aes(x = concentration, y = ERCC.ratios)) +
  geom_point() +
  geom_smooth()
```

```{r conc-unmapped}
ggplot(summary_per_sample_reads_single_qc,
       aes(x = concentration, y = unmapped.ratios)) +
  geom_point() +
  geom_smooth()
```

### Tra-1-60

Tra-1-60 is a [stem cell marker][marker].

[marker]: https://en.wikipedia.org/wiki/Stem_cell_marker

```{r tra-sum}
table(summary_per_sample_reads_single_qc$tra1.60)
```

```{r tra-ercc}
ggplot(summary_per_sample_reads_single_qc,
       aes(x = as.factor(tra1.60), y = ERCC.ratios)) +
  geom_boxplot()
```

```{r tra-unmapped}
ggplot(summary_per_sample_reads_single_qc,
       aes(x = as.factor(tra1.60), y = unmapped.ratios)) +
  geom_boxplot()
```

### Filtering by cell number and Tra-1-60

Simply removing all the cells with bad cell number or tra-1-60 does not remove all the low quality samples.

```{r cell-num-tra-sum}
table(summary_per_sample_reads_single_qc$cell_number,
      summary_per_sample_reads_single_qc$tra1.60)
```

```{r qc-filtered-by-cell-num-tra}
ggplot(subset(summary_per_sample_reads_single_qc, cell_number == 1 & tra1.60 == 1),
       aes(x = total.mapped, y = ERCC.ratios)) +
  geom_point()
ggplot(subset(summary_per_sample_reads_single_qc, cell_number == 1 & tra1.60 == 1),
       aes(x = total.mapped, y = unmapped.ratios)) +
  geom_point()
```

## PCA

### Reads

Calculate the counts per million for the singe cells.

```{r cpm}
reads_cells <- reads %>% select(-contains("bulk"))
reads_cells_cpm <- cpm(reads_cells)
```

Select the most variable genes.

```{r variable-genes}
reads_cells_cpm_log_var <- log(apply(reads_cells_cpm, 1, var))
hist(reads_cells_cpm_log_var)
sum(reads_cells_cpm_log_var > 8)
```

Using the `r sum(reads_cells_cpm_log_var > 8)` most variable genes, perform PCA.

```{r pca-reads}
reads_cells_cpm <- reads_cells_cpm[reads_cells_cpm_log_var > 8, ]
pca_reads_cells <- prcomp(t(reads_cells_cpm), retx = TRUE, scale. = TRUE,
                          center = TRUE)
```

```{r perc-explained}
plot(pca_reads_cells)
pca_reads_cells$perc_explained <- pca_reads_cells$sdev^2 / sum(pca_reads_cells$sdev^2) * 100
plot(pca_reads_cells$perc_explained)
```

The first PC accounts for `r round(pca_reads_cells$perc_explained[1])`% of the variance and the second PC `r round(pca_reads_cells$perc_explained[2])`%.

```{r cbind-anno-pca}
stopifnot(colnames(reads_cells) == 
            paste(paste0("NA", summary_per_sample_reads_single_qc$individual),
                   summary_per_sample_reads_single_qc$batch,
                   summary_per_sample_reads_single_qc$well, sep = "."))
pca_reads_cells_anno <- cbind(summary_per_sample_reads_single_qc, pca_reads_cells$x)
```

```{r pca-plot}
ggplot(pca_reads_cells_anno, aes(x = PC1, y = PC2, col = as.factor(individual),
                                 shape = as.factor(batch))) +
  geom_point()
```

What effect is PC1 capturing?

```{r pc1-ind}
ggplot(pca_reads_cells_anno, aes(x = as.factor(individual), y = PC1)) +
  geom_boxplot() + facet_wrap(~batch)
```

```{r pc1-batch}
ggplot(pca_reads_cells_anno, aes(x = as.factor(batch), y = PC1)) +
  geom_boxplot() + facet_wrap(~individual)
```

```{r pca-cell-number}
ggplot(pca_reads_cells_anno, aes(x = PC1, y = PC2)) +
  geom_text(aes(label = cell_number))
```

The outliers on PC2 are mainly empty wells.

```{r pc1-ercc}
ggplot(pca_reads_cells_anno, aes(x = ERCC.ratios, y = PC1)) +
  geom_point()
```

PC1 is negatively correlated with the ratio of unmapped reads.

```{r pc1-unmapped}
ggplot(pca_reads_cells_anno, aes(x = unmapped.ratios, y = PC1)) +
  geom_point() +
  geom_smooth()
```

PC1 is positively correlated with the total number of mapped reads.

```{r pc1-total-mapped}
ggplot(pca_reads_cells_anno, aes(x = total.mapped, y = PC1)) +
  geom_point() +
  geom_smooth()
```

PC1 is positively correlated with the detected gene number
```{r pc1-gene-number}
ggplot(pca_reads_cells_anno, aes(x = reads_single_gene_number, y = PC1)) +
  geom_point() +
  geom_smooth()
```

Using various simple filtering cutoffs, it is not possible to capture all the cells that have low values of PC1.

```{r filter-total}
pca_reads_cells_anno$total_cutoff <- pca_reads_cells_anno$total.mapped > 1.5 * 10^6
ggplot(pca_reads_cells_anno, aes(x = PC1, y = PC2, col = total_cutoff)) +
  geom_point()
```

```{r filter-unmapped}
pca_reads_cells_anno$unmapped_cutoff <- pca_reads_cells_anno$unmapped.ratios < 0.4
ggplot(pca_reads_cells_anno, aes(x = PC1, y = PC2, col = unmapped_cutoff)) +
  geom_point()
```

```{r filter-ERCC}
pca_reads_cells_anno$ERCC_cutoff <- pca_reads_cells_anno$ERCC.ratios < 0.05
ggplot(pca_reads_cells_anno, aes(x = PC1, y = PC2, col = ERCC_cutoff)) +
  geom_point()
```

```{r filter-gene-num}
pca_reads_cells_anno$gene_filter <- pca_reads_cells_anno$reads_single_gene_number  > 6000
ggplot(pca_reads_cells_anno, aes(x = PC1, y = PC2, col = gene_filter)) +
  geom_point()
```

```{r filter-cell-num}
pca_reads_cells_anno$cell_filter <- pca_reads_cells_anno$cell_number == 1
ggplot(pca_reads_cells_anno, aes(x = PC1, y = PC2, col = cell_filter)) +
  geom_point()
```

The two cutoffs largely overlap.

```{r cell-total-filter-overlap}
table(pca_reads_cells_anno$cell_filter, pca_reads_cells_anno$total_cutoff,
      dnn = c("Num cells == 1", "Total reads > 1e6"))
```

Add the third cutoff, 6000 genes detected, help a bit.
```{r filter-total-cell-num}
pca_reads_cells_anno$qc_filter <- pca_reads_cells_anno$total_cutoff &
                                  pca_reads_cells_anno$gene_filter &
                                  pca_reads_cells_anno$cell_filter
  
ggplot(pca_reads_cells_anno, aes(x = PC1, y = PC2, col = qc_filter,
                                 label = as.character(cell_number))) +
  geom_text(fontface=3)
```

Apply all the cutoffs

```{r filter-all}
pca_reads_cells_anno$qc_filter_all <- pca_reads_cells_anno$total_cutoff &
                                      pca_reads_cells_anno$unmapped_cutoff &
                                      pca_reads_cells_anno$ERCC_cutoff &
                                      pca_reads_cells_anno$gene_filter &
                                      pca_reads_cells_anno$cell_filter
  
ggplot(pca_reads_cells_anno, aes(x = PC1, y = PC2, col = qc_filter_all,
                                 label = as.character(cell_number))) +
  geom_text(fontface=3)
```

How many cells do we keep from each individual and batch using this filter?

```{r individuals-remaining}
table(pca_reads_cells_anno[pca_reads_cells_anno$qc_filter,
                           c("individual", "batch")])

table(pca_reads_cells_anno[pca_reads_cells_anno$qc_filter_all,
                           c("individual", "batch")])
```

Output list of single cells to keep.

```{r filter-keep-list}
stopifnot(nrow(pca_reads_cells_anno) == nrow(anno[anno$well != "bulk", ]))
quality_single_cells <- anno %>%
  filter(well != "bulk") %>%
  filter(pca_reads_cells_anno$qc_filter) %>%
  select(sample_id)
stopifnot(!grepl("bulk", quality_single_cells$sample_id))
write.table(quality_single_cells,
            file = "../data/quality-single-cells.txt", quote = FALSE,
            sep = "\t", row.names = FALSE, col.names = FALSE)
```

## Session information

```{r info}
sessionInfo()
```
