<!DOCTYPE html>

<html xmlns="http://www.w3.org/1999/xhtml">

<head>

<meta charset="utf-8">
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="generator" content="pandoc" />

<meta name="author" content="PoYuan Tung" />

<meta name="date" content="2015-10-23" />

<title>QC of single cell libraries</title>

<script src="libs/jquery-1.11.0/jquery.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="libs/bootstrap-3.3.1/css/united.min.css" rel="stylesheet" />
<script src="libs/bootstrap-3.3.1/js/bootstrap.min.js"></script>
<script src="libs/bootstrap-3.3.1/shim/html5shiv.min.js"></script>
<script src="libs/bootstrap-3.3.1/shim/respond.min.js"></script>

<style type="text/css">

/* padding for bootstrap navbar */
body {
  padding-top: 50px;
  padding-bottom: 40px;
}


/* offset scroll position for anchor links (for fixed navbar)  */
.section h2 {
  padding-top: 55px;
  margin-top: -55px;
}
.section h3 {
  padding-top: 55px;
  margin-top: -55px;
}



/* don't use link color in navbar */
.dropdown-menu>li>a {
  color: black;
}

/* some padding for disqus */
#disqus_thread {
  margin-top: 45px;
}

</style>

<link rel="stylesheet" href="libs/font-awesome-4.1.0/css/font-awesome.min.css"/>

<style type="text/css">code{white-space: pre;}</style>
<link rel="stylesheet"
      href="libs/highlight/textmate.css"
      type="text/css" />
<script src="libs/highlight/highlight.js"></script>
<style type="text/css">
  pre:not([class]) {
    background-color: white;
  }
</style>
<script type="text/javascript">
if (window.hljs && document.readyState && document.readyState === "complete") {
   window.setTimeout(function() {
      hljs.initHighlighting();
   }, 0);
}
</script>



</head>

<body>

<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
code {
  color: inherit;
  background-color: rgba(0, 0, 0, 0.04);
}
img { 
  max-width:100%; 
  height: auto; 
}
</style>
<div class="container-fluid main-container">


<div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">singleCellSeq</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li><a href="index.html">Home</a></li>
        <li><a href="about.html">About</a></li>
        <li><a href="license.html">License</a></li>
        <li><a href="https://github.com/jdblischak/singleCellSeq">GitHub</a></li>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<div id="header">
<h1 class="title">QC of single cell libraries</h1>
<h4 class="author"><em>PoYuan Tung</em></h4>
<h4 class="date"><em>2015-10-23</em></h4>
</div>

<div id="TOC">
<ul>
<li><a href="#input">Input</a></li>
<li><a href="#total-ercc-and-removal-of-na19098.r2">Total ERCC and removal of NA19098.r2</a></li>
<li><a href="#total-mapped-reads-reads">Total mapped reads reads</a></li>
<li><a href="#unmapped-ratios">Unmapped ratios</a></li>
<li><a href="#ercc-percentage">ERCC percentage</a></li>
<li><a href="#number-of-genes-detected">Number of genes detected</a></li>
<li><a href="#total-molecule-counts">Total molecule counts</a><ul>
<li><a href="#linear-discriminat-analysis">Linear Discriminat Analysis</a></li>
</ul></li>
<li><a href="#reads-to-molecule-conversion">Reads to molecule conversion</a></li>
<li><a href="#mitochondrial-genes">Mitochondrial genes</a></li>
<li><a href="#filter">Filter</a><ul>
<li><a href="#final-list">Final list</a></li>
<li><a href="#mito-ratios">Mito ratios</a></li>
<li><a href="#plots">plots</a></li>
</ul></li>
<li><a href="#session-information">Session information</a></li>
</ul>
</div>

<p><strong>Last updated:</strong> 2016-02-15</p>
<p><strong>Code version:</strong> afb6992553e07bb4a9f9437d4f2aebb9b973eb37</p>
<div id="input" class="section level2">
<h2>Input</h2>
<pre class="r"><code>library(&quot;dplyr&quot;)
library(&quot;edgeR&quot;)
library(&quot;ggplot2&quot;)
library(&quot;cowplot&quot;)
theme_set(theme_bw(base_size = 12))
theme_update(panel.grid.minor.x = element_blank(),
             panel.grid.minor.y = element_blank(),
             panel.grid.major.x = element_blank(),
             panel.grid.major.y = element_blank())
source(&quot;functions.R&quot;)</code></pre>
<p>Summary counts from featureCounts. Created with <a href="https://github.com/jdblischak/singleCellSeq/blob/master/code/gather-summary-counts.py">gather-summary-counts.py</a>. These data were collected from the summary files of the full combined samples.</p>
<pre class="r"><code>summary_per_sample &lt;- read.table(&quot;../data/summary-counts.txt&quot;, header = TRUE,
                             stringsAsFactors = FALSE)
stopifnot(summary_per_sample$well != &quot;bulk&quot;,
          sum(summary_per_sample$rmdup == &quot;reads&quot;) == 864,
          sum(summary_per_sample$rmdup == &quot;molecules&quot;) == 864)</code></pre>
<p>Remove featureCounts classifications with zero counts.</p>
<pre class="r"><code>stopifnot(colSums(summary_per_sample[, c(7, 10:15)]) == 0)
summary_per_sample &lt;- summary_per_sample[, c(-7, -10:-15)]
head(summary_per_sample)</code></pre>
<pre><code>  individual replicate well     rmdup Assigned Unassigned_Ambiguity
1    NA19098        r1  A01 molecules    67043                 2520
2    NA19098        r1  A01     reads  1953434                71505
3    NA19098        r1  A02 molecules    68222                 2598
4    NA19098        r1  A02     reads  2064909                78380
5    NA19098        r1  A03 molecules    45901                 1681
6    NA19098        r1  A03     reads  1019557                31762
  Unassigned_NoFeatures Unassigned_Unmapped
1                 44570                   0
2                810666             1116473
3                 39183                   0
4                643671             1175385
5                 27282                   0
6                380232              755641</code></pre>
<p>Input annotation.</p>
<pre class="r"><code>anno &lt;- read.table(&quot;../data/annotation.txt&quot;, header = TRUE,
                   stringsAsFactors = FALSE)
stopifnot(anno$well != &quot;bulk&quot;, nrow(anno) == 864,
          rep(anno$individual, each = 2) == summary_per_sample$individual,
          rep(anno$replicate, each = 2) == summary_per_sample$replicate,
          rep(anno$well, each = 2) == summary_per_sample$well)
head(anno)</code></pre>
<pre><code>  individual replicate well      batch      sample_id
1    NA19098        r1  A01 NA19098.r1 NA19098.r1.A01
2    NA19098        r1  A02 NA19098.r1 NA19098.r1.A02
3    NA19098        r1  A03 NA19098.r1 NA19098.r1.A03
4    NA19098        r1  A04 NA19098.r1 NA19098.r1.A04
5    NA19098        r1  A05 NA19098.r1 NA19098.r1.A05
6    NA19098        r1  A06 NA19098.r1 NA19098.r1.A06</code></pre>
<p>Input read counts.</p>
<pre class="r"><code>reads &lt;- read.table(&quot;../data/reads.txt&quot;, header = TRUE,
                    stringsAsFactors = FALSE)
stopifnot(ncol(reads) == nrow(anno),
          colnames(reads) == anno$sample_id)</code></pre>
<p>Input molecule counts.</p>
<pre class="r"><code>molecules &lt;- read.table(&quot;../data/molecules.txt&quot;, header = TRUE,
                    stringsAsFactors = FALSE)
stopifnot(ncol(molecules) == nrow(anno),
          colnames(reads) == anno$sample_id)</code></pre>
<p>Input single cell observational quality control data.</p>
<pre class="r"><code>qc &lt;- read.table(&quot;../data/qc-ipsc.txt&quot;, header = TRUE,
                 stringsAsFactors = FALSE)
stopifnot(qc$individual == anno$individual,
          qc$replicate == anno$replicate,
          qc$well == anno$well)
head(qc)</code></pre>
<pre><code>  individual replicate well cell_number concentration tra1.60
1    NA19098        r1  A01           1      1.734785       1
2    NA19098        r1  A02           1      1.723038       1
3    NA19098        r1  A03           1      1.512786       1
4    NA19098        r1  A04           1      1.347492       1
5    NA19098        r1  A05           1      2.313047       1
6    NA19098        r1  A06           1      2.056803       1</code></pre>
</div>
<div id="total-ercc-and-removal-of-na19098.r2" class="section level2">
<h2>Total ERCC and removal of NA19098.r2</h2>
<p>Show the evidence that removing NA19098 batch 2 is the first thing to do.</p>
<pre class="r"><code>summary_per_sample_reads &lt;- summary_per_sample[summary_per_sample$rmdup == &quot;reads&quot;,]

summary_per_sample_reads$sample_id &lt;- anno$sample_id
summary_per_sample_reads$batch &lt;- anno$batch
stopifnot(colnames(reads) == summary_per_sample_reads$sample_id )

summary_per_sample_reads$ERCC_reads &lt;- apply(reads[grep(&quot;ERCC&quot;, rownames(reads)), ],2,sum)
summary_per_sample_reads$ERCC_molecules &lt;- apply(molecules[grep(&quot;ERCC&quot;, rownames(molecules)), ],2,sum)</code></pre>
<pre class="r"><code>## create a color palette with one color per individual and different shades for repplicates
great_color &lt;- c(&quot;#CC3300&quot;, &quot;#FF9966&quot;, &quot;#FFCC99&quot;, &quot;#006633&quot;, &quot;#009900&quot;, &quot;#99FF99&quot;, &quot;#3366FF&quot;, &quot;#6699FF&quot;, &quot;#66CCFF&quot;)
great_color_8 &lt;- c(&quot;#CC3300&quot;, &quot;#FF9966&quot;, &quot;#006633&quot;, &quot;#009900&quot;, &quot;#99FF99&quot;, &quot;#3366FF&quot;, &quot;#6699FF&quot;, &quot;#66CCFF&quot;)

ercc_reads_plot &lt;- ggplot(summary_per_sample_reads,
                   aes(x = factor(batch), y = ERCC_reads,
                   fill = factor(batch)), height = 600, width = 2000) +
                   geom_violin(alpha = .5) + 
                   geom_boxplot(alpha = .01, width = .2, position = position_dodge(width = .9)) +
                   scale_fill_manual(values = great_color) + 
                   labs(x = &quot;&quot;, y = &quot;ERCC reads&quot;, title = &quot;Total ERCC read counts per cell&quot;) +
                   theme(axis.text.x = element_text(hjust=1, angle = 45))

ercc_molecule_plot &lt;- ggplot(summary_per_sample_reads,
                   aes(x = factor(batch), y = ERCC_molecules,
                   fill = factor(batch)), height = 600, width = 2000) +
                   geom_violin(alpha = .5) + 
                   geom_boxplot(alpha = .01, width = .2, position = position_dodge(width = .9)) +
                   scale_fill_manual(values = great_color) + 
                   labs(x = &quot;&quot;, y = &quot;ERCC molecules&quot;, title = &quot;Total ERCC molecule counts per cell&quot;) +
                   theme(axis.text.x = element_text(hjust=1, angle = 45))

plot_grid(ercc_reads_plot + theme(legend.position=c(.8,.7)),
          ercc_molecule_plot + theme(legend.position = &quot;none&quot;),
          labels = LETTERS[1:2])</code></pre>
<p><img src="figure/qc-filter-ipsc.Rmd/sad-ercc-plot-1.png" title="" alt="" width="1152" style="display: block; margin: auto;" /></p>
<p>Remove NA19098r2 for all the following analysis</p>
<pre class="r"><code>remove_19098r2 &lt;- anno$batch != &quot;NA19098.r2&quot;
anno_rm &lt;- anno[remove_19098r2,]
summary_per_sample_reads_rm &lt;- summary_per_sample_reads[remove_19098r2,]
reads_rm &lt;- reads[, remove_19098r2]
molecules_rm &lt;- molecules[, remove_19098r2]

stopifnot(summary_per_sample_reads_rm$sample_id == colnames(reads_rm))</code></pre>
</div>
<div id="total-mapped-reads-reads" class="section level2">
<h2>Total mapped reads reads</h2>
<pre class="r"><code>## add cell number per well by merging qc file
summary_per_sample_reads_qc &lt;- merge(summary_per_sample_reads_rm,qc,by=c(&quot;individual&quot;,&quot;replicate&quot;,&quot;well&quot;))

## calculate total mapped reads per sample
summary_per_sample_reads_qc$total_mapped &lt;- apply(summary_per_sample_reads_qc[,5:7],1,sum)

## cut off 
cut_off_reads &lt;- quantile(summary_per_sample_reads_qc[summary_per_sample_reads_qc$cell_number == 0,&quot;total_mapped&quot;], 0.95)

cut_off_reads</code></pre>
<pre><code>    95% 
1538579 </code></pre>
<pre class="r"><code>summary_per_sample_reads_qc$cut_off_reads &lt;- summary_per_sample_reads_qc$total_mapped &gt; cut_off_reads

## numbers of cells 
sum(summary_per_sample_reads_qc[summary_per_sample_reads_qc$cell_number == 1, &quot;total_mapped&quot;] &gt; cut_off_reads)</code></pre>
<pre><code>[1] 602</code></pre>
<pre class="r"><code>sum(summary_per_sample_reads_qc[summary_per_sample_reads_qc$cell_number == 1, &quot;total_mapped&quot;] &lt;= cut_off_reads)</code></pre>
<pre><code>[1] 97</code></pre>
<pre class="r"><code>## density plots
plot_reads &lt;- ggplot(summary_per_sample_reads_qc[summary_per_sample_reads_qc$cell_number == 0 |
                                           summary_per_sample_reads_qc$cell_number == 1 , ],
       aes(x = total_mapped, fill = as.factor(cell_number))) + 
       geom_density(alpha = 0.5) +
       geom_vline(xintercept = cut_off_reads, colour=&quot;grey&quot;, linetype = &quot;longdash&quot;) +
       labs(x = &quot;Total mapped reads&quot;, title = &quot;Number of Total mappred reads&quot;, fill = &quot;Cell number&quot;)</code></pre>
</div>
<div id="unmapped-ratios" class="section level2">
<h2>Unmapped ratios</h2>
<pre class="r"><code>## calculate unmapped ratios
summary_per_sample_reads_qc$unmapped_ratios &lt;- summary_per_sample_reads_qc[,8]/apply(summary_per_sample_reads_qc[,5:8],1,sum)

## cut off 
cut_off_unmapped &lt;- quantile(summary_per_sample_reads_qc[summary_per_sample_reads_qc$cell_number == 0,&quot;unmapped_ratios&quot;], 0.05)

cut_off_unmapped</code></pre>
<pre><code>       5% 
0.3715408 </code></pre>
<pre class="r"><code>summary_per_sample_reads_qc$cut_off_unmapped &lt;- summary_per_sample_reads_qc$unmapped_ratios &lt; cut_off_unmapped

## numbers of cells 
sum(summary_per_sample_reads_qc[summary_per_sample_reads_qc$cell_number == 1, &quot;unmapped_ratios&quot;] &gt;= cut_off_unmapped)</code></pre>
<pre><code>[1] 102</code></pre>
<pre class="r"><code>sum(summary_per_sample_reads_qc[summary_per_sample_reads_qc$cell_number == 1, &quot;unmapped_ratios&quot;] &lt; cut_off_unmapped)</code></pre>
<pre><code>[1] 597</code></pre>
<pre class="r"><code>## density plots
plot_unmapped &lt;- ggplot(summary_per_sample_reads_qc[summary_per_sample_reads_qc$cell_number == 0 |
                                           summary_per_sample_reads_qc$cell_number == 1 , ],
       aes(x = unmapped_ratios *100, fill = as.factor(cell_number))) + 
       geom_density(alpha = 0.5) +
       geom_vline(xintercept = cut_off_unmapped *100, colour=&quot;grey&quot;, linetype = &quot;longdash&quot;) +
       labs(x = &quot;Unmapped reads/ total reads&quot;, title = &quot;Unmapped reads percentage&quot;)</code></pre>
</div>
<div id="ercc-percentage" class="section level2">
<h2>ERCC percentage</h2>
<pre class="r"><code>## calculate ercc reads percentage
summary_per_sample_reads_qc$ercc_percentage &lt;- apply(reads_rm[grep(&quot;ERCC&quot;, rownames(reads_rm)), ],2,sum)/apply(summary_per_sample_reads_qc[,5:7],1,sum) 

## cut off 
cut_off_ercc &lt;- quantile(summary_per_sample_reads_qc[summary_per_sample_reads_qc$cell_number == 0,&quot;ercc_percentage&quot;], 0.05)

cut_off_ercc</code></pre>
<pre><code>        5% 
0.03245405 </code></pre>
<pre class="r"><code>summary_per_sample_reads_qc$cut_off_ercc &lt;- summary_per_sample_reads_qc$ercc_percentage &lt; cut_off_ercc

## numbers of cells 
sum(summary_per_sample_reads_qc[summary_per_sample_reads_qc$cell_number == 1, &quot;ercc_percentage&quot;] &gt;= cut_off_ercc)</code></pre>
<pre><code>[1] 92</code></pre>
<pre class="r"><code>sum(summary_per_sample_reads_qc[summary_per_sample_reads_qc$cell_number == 1, &quot;ercc_percentage&quot;] &lt; cut_off_ercc)</code></pre>
<pre><code>[1] 607</code></pre>
<pre class="r"><code>## density plots
plot_ercc &lt;- ggplot(summary_per_sample_reads_qc[summary_per_sample_reads_qc$cell_number == 0 |
                                           summary_per_sample_reads_qc$cell_number == 1 , ],
       aes(x = ercc_percentage *100, fill = as.factor(cell_number))) + 
       geom_density(alpha = 0.5) +
       geom_vline(xintercept = cut_off_ercc *100, colour=&quot;grey&quot;, linetype = &quot;longdash&quot;) +
       labs(x = &quot;ERCC reads / total mapped reads&quot;, title = &quot;ERCC reads percentage&quot;)</code></pre>
</div>
<div id="number-of-genes-detected" class="section level2">
<h2>Number of genes detected</h2>
<pre class="r"><code>## endogenous genes
reads_rm_gene &lt;- reads_rm[grep(&quot;ENSG&quot;, rownames(reads_rm)), ]

## number of genes detected 
summary_per_sample_reads_qc$gene_number &lt;- colSums(reads_rm_gene &gt;= 1)

## cut off 
cut_off_genes &lt;- quantile(summary_per_sample_reads_qc[summary_per_sample_reads_qc$cell_number == 0,&quot;gene_number&quot;], 0.95)

cut_off_genes</code></pre>
<pre><code>   95% 
6862.1 </code></pre>
<pre class="r"><code>summary_per_sample_reads_qc$cut_off_genes &lt;- summary_per_sample_reads_qc$gene_number &gt; cut_off_genes

## numbers of cells 
sum(summary_per_sample_reads_qc[summary_per_sample_reads_qc$cell_number == 1, &quot;gene_number&quot;] &gt; cut_off_genes)</code></pre>
<pre><code>[1] 625</code></pre>
<pre class="r"><code>sum(summary_per_sample_reads_qc[summary_per_sample_reads_qc$cell_number == 1, &quot;gene_number&quot;] &lt;= cut_off_genes)</code></pre>
<pre><code>[1] 74</code></pre>
<pre class="r"><code>## density plots
plot_gene &lt;- ggplot(summary_per_sample_reads_qc[summary_per_sample_reads_qc$cell_number == 0 |
                                           summary_per_sample_reads_qc$cell_number == 1 , ],
       aes(x = gene_number, fill = as.factor(cell_number))) + 
       geom_density(alpha = 0.5) +
       geom_vline(xintercept = cut_off_genes, colour=&quot;grey&quot;, linetype = &quot;longdash&quot;) +
       labs(x = &quot;Gene numbers&quot;, title = &quot;Numbers of detected genes&quot;)</code></pre>
<pre class="r"><code>plot_grid(plot_reads + theme(legend.position=c(.7,.7)),
          plot_unmapped + theme(legend.position = &quot;none&quot;),
          plot_ercc + theme(legend.position = &quot;none&quot;), 
          plot_gene + theme(legend.position = &quot;none&quot;),
          labels = LETTERS[3:6])</code></pre>
<p><img src="figure/qc-filter-ipsc.Rmd/density%20plots-1.png" title="" alt="" width="768" style="display: block; margin: auto;" /></p>
</div>
<div id="total-molecule-counts" class="section level2">
<h2>Total molecule counts</h2>
<pre class="r"><code>## calculate total gene molecule counts
summary_per_sample_reads_qc$total_gene_molecule &lt;- colSums(molecules_rm[grep(&quot;ENSG&quot;, rownames(molecules_rm)),])

## look for outiers
ggplot(summary_per_sample_reads_qc, aes(x = concentration, y = total_gene_molecule / 10^3,
  color = as.factor(cell_number))) +
  geom_text(aes(label = cell_number)) +
  labs(x = &quot;Concentration&quot;, y = &quot;Gene molecules (thousands)&quot;) +
  scale_color_brewer(palette = &quot;Dark2&quot;) +
  theme(legend.position = &quot;none&quot;)</code></pre>
<p><img src="figure/qc-filter-ipsc.Rmd/total-mol-1.png" title="" alt="" width="768" style="display: block; margin: auto;" /></p>
<pre class="r"><code>outliers &lt;- summary_per_sample_reads_qc %&gt;% filter(cell_number == 1, concentration &lt; 1.25, concentration &gt; .15,
                                   total_gene_molecule &gt; 100000)

outliers %&gt;% dplyr::select(sample_id)</code></pre>
<pre><code>        sample_id
1  NA19098.r3.B04
2  NA19098.r3.B11
3  NA19101.r1.B10
4  NA19101.r2.D07
5  NA19101.r3.C07
6  NA19101.r3.D08
7  NA19101.r3.D10
8  NA19101.r3.F05
9  NA19101.r3.F10
10 NA19239.r2.A12
11 NA19239.r2.B07</code></pre>
<pre class="r"><code>summary_per_sample_reads_qc$molecule_outlier &lt;- summary_per_sample_reads_qc$cell_number == 1 &amp;
  summary_per_sample_reads_qc$concentration &lt; 1.25 &amp;
  summary_per_sample_reads_qc$concentration &gt; .15 &amp;
  summary_per_sample_reads_qc$total_gene_molecule &gt; 100000</code></pre>
<div id="linear-discriminat-analysis" class="section level3">
<h3>Linear Discriminat Analysis</h3>
<pre class="r"><code>library(MASS)</code></pre>
<pre><code>
Attaching package: &#39;MASS&#39;

The following object is masked from &#39;package:dplyr&#39;:

    select</code></pre>
<pre class="r"><code>## create 3 groups according to cell number
group_3 &lt;- rep(&quot;two&quot;,dim(summary_per_sample_reads_qc)[1])
         group_3[grep(&quot;0&quot;, summary_per_sample_reads_qc$cell_number)] &lt;- &quot;no&quot;
         group_3[grep(&quot;1&quot;, summary_per_sample_reads_qc$cell_number)] &lt;- &quot;one&quot;

## create data frame
data_lda &lt;- data.frame(anno_rm,
                       cell_number = summary_per_sample_reads_qc$cell_number,
                       concentration = summary_per_sample_reads_qc$concentration,
                       total_gene_molecule = summary_per_sample_reads_qc$total_gene_molecule,
                       group = group_3)

## remove 19098.r1
data_con &lt;- data_lda %&gt;% filter(batch != &quot;NA19098.r1&quot;)
plot_before &lt;- ggplot(data_con, aes(x = concentration, y = total_gene_molecule / 10^3,
               color = as.factor(group))) +
               geom_text(aes(label = cell_number)) +
               labs(x = &quot;Concentration&quot;, y = &quot;Gene molecules (thousands)&quot;, title = &quot;Before&quot;) +
               scale_color_brewer(palette = &quot;Dark2&quot;) +
               theme(legend.position = &quot;none&quot;)

## perform lda
data_con_lda &lt;- lda(group ~ concentration + total_gene_molecule,
                    data = data_con)
data_con_lda_p &lt;- predict(data_con_lda, 
                          newdata = data_con[,c(&quot;concentration&quot;, &quot;total_gene_molecule&quot;)])$class

## determine how well the model fix
table(data_con_lda_p, data_con[, &quot;group&quot;])</code></pre>
<pre><code>              
data_con_lda_p  no one two
           no    1   0   0
           one  16 595  16
           two   0  11  33</code></pre>
<pre class="r"><code>data_con$data_con_lda_p &lt;- data_con_lda_p

plot_after &lt;- ggplot(data_con, aes(x = concentration, y = total_gene_molecule / 10^3,
               color = as.factor(data_con_lda_p))) +
               geom_text(aes(label = cell_number)) +
               labs(x = &quot;Concentration&quot;, y = &quot;Gene molecules (thousands)&quot;, title = &quot;After&quot;) +
               scale_color_brewer(palette = &quot;Dark2&quot;) +
               theme(legend.position = &quot;none&quot;)

## identify the outlier 
outliers_lda &lt;- data_con %&gt;% filter(cell_number == 1, data_con_lda_p == &quot;two&quot;)
outliers_lda$sample_id</code></pre>
<pre><code> [1] &quot;NA19098.r3.B04&quot; &quot;NA19098.r3.B11&quot; &quot;NA19101.r1.B10&quot; &quot;NA19101.r2.D07&quot;
 [5] &quot;NA19101.r3.C07&quot; &quot;NA19101.r3.D08&quot; &quot;NA19101.r3.D10&quot; &quot;NA19101.r3.F05&quot;
 [9] &quot;NA19101.r3.F10&quot; &quot;NA19239.r2.A12&quot; &quot;NA19239.r2.B07&quot;</code></pre>
<pre class="r"><code>## The lds method identifies outliers
stopifnot(outliers %&gt;% dplyr::select(sample_id) == outliers_lda$sample_id)

plot_grid(plot_before + theme(legend.position=c(.8,.85)), 
          plot_after + theme(legend.position = &quot;none&quot;),
          labels = LETTERS[1:2])</code></pre>
<p><img src="figure/qc-filter-ipsc.Rmd/lda-1.png" title="" alt="" width="1152" style="display: block; margin: auto;" /></p>
</div>
</div>
<div id="reads-to-molecule-conversion" class="section level2">
<h2>Reads to molecule conversion</h2>
<pre class="r"><code>## calculate convertion
summary_per_sample_reads_qc$ERCC_conversion &lt;- summary_per_sample_reads_qc$ERCC_molecules / summary_per_sample_reads_qc$ERCC_reads

summary_per_sample_reads_qc$conversion &lt;- summary_per_sample_reads_qc$total_gene_molecule /  colSums(reads_rm[grep(&quot;ENSG&quot;, rownames(reads_rm)),])

ggplot(summary_per_sample_reads_qc, aes(x = ERCC_conversion, y = conversion,
  color = as.factor(cell_number))) +
  geom_text(aes(label = cell_number)) +
  labs(x = &quot;Convertion of ERCC&quot;, y = &quot;Conversion of genes&quot;) +
  scale_color_brewer(palette = &quot;Dark2&quot;) +
  theme(legend.position = &quot;none&quot;)</code></pre>
<p><img src="figure/qc-filter-ipsc.Rmd/convertion-1.png" title="" alt="" width="1152" style="display: block; margin: auto;" /></p>
<pre class="r"><code>out_ercc_con &lt;- summary_per_sample_reads_qc %&gt;% filter(cell_number == &quot;1&quot;, ERCC_conversion &gt; .094)

## try lda
data_lda$conversion &lt;- summary_per_sample_reads_qc$conversion
data_lda$ERCC_conversion &lt;- summary_per_sample_reads_qc$ERCC_conversion

data_ercc_lda &lt;- lda(group ~ ERCC_conversion + conversion,
                    data = data_lda)
data_ercc_lda_p &lt;- predict(data_ercc_lda, 
                          newdata = data_lda[,c(&quot;ERCC_conversion&quot;, &quot;conversion&quot;)])$class
table(data_con_lda_p, data_con[, &quot;group&quot;])</code></pre>
<pre><code>              
data_con_lda_p  no one two
           no    1   0   0
           one  16 595  16
           two   0  11  33</code></pre>
<pre class="r"><code>data_lda$data_ercc_lda_p &lt;- data_ercc_lda_p

## identify the outlier 
outliers_ercc &lt;- data_lda %&gt;% filter(cell_number == 1, data_ercc_lda_p == &quot;two&quot;)
outliers_ercc$sample_id</code></pre>
<pre><code> [1] &quot;NA19098.r1.F01&quot; &quot;NA19098.r3.B04&quot; &quot;NA19098.r3.B11&quot; &quot;NA19101.r2.D07&quot;
 [5] &quot;NA19101.r3.C07&quot; &quot;NA19101.r3.D08&quot; &quot;NA19101.r3.F05&quot; &quot;NA19101.r3.F10&quot;
 [9] &quot;NA19239.r2.A12&quot; &quot;NA19239.r2.B07&quot; &quot;NA19239.r3.G02&quot;</code></pre>
<pre class="r"><code>## cutoff
out_ercc_con &lt;- summary_per_sample_reads_qc %&gt;% filter(cell_number == &quot;1&quot;, ERCC_conversion &gt; .08)
# The two methods no longer give identical results
stopifnot(out_ercc_con %&gt;% dplyr::select(sample_id) == outliers_ercc$sample_id)

summary_per_sample_reads_qc$conversion_outlier &lt;- summary_per_sample_reads_qc$cell_number == 1 &amp;
                                                         summary_per_sample_reads_qc$ERCC_conversion &gt; .08

plot_ercc_before &lt;- ggplot(data_lda, aes(x = ERCC_conversion, y = conversion,
               color = as.factor(group))) +
               geom_text(aes(label = cell_number)) +
               labs(x = &quot;Convertion of ERCC&quot;, y = &quot;Conversion of genes&quot;, title = &quot;Before&quot;) +
               scale_color_brewer(palette = &quot;Dark2&quot;) +
               theme(legend.position = &quot;none&quot;)

plot_ercc_after &lt;- ggplot(data_lda, aes(x = ERCC_conversion, y = conversion,
               color = as.factor(data_ercc_lda_p))) +
               geom_text(aes(label = cell_number)) +
               labs(x = &quot;Convertion of ERCC&quot;, y = &quot;Conversion of genes&quot;, title = &quot;After&quot;) +
               scale_color_brewer(palette = &quot;Dark2&quot;) +
               theme(legend.position = &quot;none&quot;)

plot_grid(plot_ercc_before, 
          plot_ercc_after,
          labels = LETTERS[3:4])</code></pre>
<p><img src="figure/qc-filter-ipsc.Rmd/convertion-2.png" title="" alt="" width="1152" style="display: block; margin: auto;" /></p>
</div>
<div id="mitochondrial-genes" class="section level2">
<h2>Mitochondrial genes</h2>
<pre class="r"><code>## create a list of mitochondrial genes (13 protein-coding genes)
## MT-ATP6, MT-CYB, MT-ND1, MT-ND4, MT-ND4L, MT-ND5, MT-ND6, MT-CO2, MT-CO1, MT-ND2, MT-ATP8, MT-CO3, MT-ND3
mtgene &lt;- c(&quot;ENSG00000198899&quot;, &quot;ENSG00000198727&quot;, &quot;ENSG00000198888&quot;, &quot;ENSG00000198886&quot;, &quot;ENSG00000212907&quot;, &quot;ENSG00000198786&quot;, &quot;ENSG00000198695&quot;, &quot;ENSG00000198712&quot;, &quot;ENSG00000198804&quot;, &quot;ENSG00000198763&quot;,&quot;ENSG00000228253&quot;, &quot;ENSG00000198938&quot;, &quot;ENSG00000198840&quot;)

## reads of mt genes in single cells
mt_reads &lt;- reads_rm_gene[mtgene,]
dim(mt_reads)</code></pre>
<pre><code>[1]  13 768</code></pre>
<pre class="r"><code>stopifnot(colnames(reads_rm) == rownames(summary_per_sample_reads_qc$sample_id))

## mt ratio of single cell
summary_per_sample_reads_qc$mt_reads &lt;- apply(mt_reads, 2, sum)
summary_per_sample_reads_qc$mt_reads_ratio &lt;- summary_per_sample_reads_qc$mt_reads /summary_per_sample_reads_qc$total_mapped

## vs. number of genes detected
ggplot(summary_per_sample_reads_qc,
       aes(x = gene_number, y = mt_reads_ratio, 
       color = as.factor(cell_number))) +
       geom_text(aes(label = cell_number)) +
       labs(x = &quot;Number of genes&quot;, y = &quot;Mitochondrial ratio&quot;) +
       scale_color_brewer(palette = &quot;Dark2&quot;) +
       theme(legend.position = &quot;none&quot;)</code></pre>
<p><img src="figure/qc-filter-ipsc.Rmd/mito-gene-1.png" title="" alt="" width="768" style="display: block; margin: auto;" /></p>
</div>
<div id="filter" class="section level2">
<h2>Filter</h2>
<div id="final-list" class="section level3">
<h3>Final list</h3>
<pre class="r"><code>## all filter
summary_per_sample_reads_qc$filter_all &lt;- summary_per_sample_reads_qc$cell_number == 1 &amp;
                                                 summary_per_sample_reads_qc$cut_off_reads &amp;
                                                 summary_per_sample_reads_qc$cut_off_unmapped &amp;
                                                 summary_per_sample_reads_qc$cut_off_ercc &amp;
                                                 summary_per_sample_reads_qc$cut_off_genes &amp;
                                                 summary_per_sample_reads_qc$molecule_outlier == &quot;FALSE&quot; &amp;
                                                 summary_per_sample_reads_qc$conversion_outlier == &quot;FALSE&quot;
                                                 
table(summary_per_sample_reads_qc[summary_per_sample_reads_qc$filter_all,
                           c(&quot;individual&quot;, &quot;replicate&quot;)])    </code></pre>
<pre><code>          replicate
individual r1 r2 r3
   NA19098 85  0 56
   NA19101 80 70 49
   NA19239 73 68 79</code></pre>
<pre class="r"><code>stopifnot(nrow(summary_per_sample_reads_qc) == nrow(anno_rm))
quality_single_cells &lt;- anno_rm %&gt;%
  filter(summary_per_sample_reads_qc$filter_all) %&gt;%
  dplyr :: select(sample_id)

write.table(quality_single_cells,
            file = &quot;../data/quality-single-cells.txt&quot;, quote = FALSE,
            sep = &quot;\t&quot;, row.names = FALSE, col.names = FALSE)</code></pre>
</div>
<div id="mito-ratios" class="section level3">
<h3>Mito ratios</h3>
<pre class="r"><code>ggplot(summary_per_sample_reads_qc,
       aes(x = gene_number, y = mt_reads_ratio, 
       color = as.factor(filter_all))) +
       geom_text(aes(label = cell_number)) +
       labs(x = &quot;Number of genes&quot;, y = &quot;Mitochondrial ratio&quot;) +
       theme(legend.position = &quot;none&quot;)</code></pre>
<p><img src="figure/qc-filter-ipsc.Rmd/mito-filter-1.png" title="" alt="" width="768" style="display: block; margin: auto;" /></p>
<pre class="r"><code>ggplot(summary_per_sample_reads_qc[summary_per_sample_reads_qc$cell_number == 1,],
       aes(x = factor(filter_all), y = mt_reads_ratio,
           fill = factor(filter_all)), height = 600, width = 2000) +
geom_violin(alpha = .5) + 
geom_boxplot(alpha = .01, width = .2, position = position_dodge(width = .9)) +
labs(x = &quot;Quality&quot;, y = &quot;Mitochonrial ratio&quot;, title = &quot;Mitochondrial ratio of libraries with 1 cell&quot;)</code></pre>
<p><img src="figure/qc-filter-ipsc.Rmd/mito-filter-2.png" title="" alt="" width="768" style="display: block; margin: auto;" /></p>
<pre class="r"><code>## check the batch of those outliers
mito_outliers &lt;- summary_per_sample_reads_qc %&gt;% filter(filter_all == &quot;TRUE&quot;, mt_reads_ratio &gt; .15)
mito_outliers  %&gt;% dplyr::select(sample_id, mt_reads_ratio)</code></pre>
<pre><code>       sample_id mt_reads_ratio
1 NA19098.r1.D07      0.1593539
2 NA19098.r3.G03      0.1625643
3 NA19098.r3.G04      0.1511902</code></pre>
<pre class="r"><code>## check if 19098 have high mt genes
ggplot(summary_per_sample_reads_qc[summary_per_sample_reads_qc$filter_all == &quot;TRUE&quot;,],
                   aes(x = factor(batch), y = mt_reads_ratio,
                   fill = factor(batch)), height = 600, width = 2000) +
                   geom_violin(alpha = .5) + 
                   geom_boxplot(alpha = .01, width = .2, position = position_dodge(width = .9)) +
                   scale_fill_manual(values = great_color_8) +  
                   labs(x = &quot;batch&quot;, y = &quot;Mitochondrial ratio&quot;) +
                   theme(axis.text.x = element_text(hjust=1, angle = 45))</code></pre>
<p><img src="figure/qc-filter-ipsc.Rmd/mito-filter-3.png" title="" alt="" width="768" style="display: block; margin: auto;" /></p>
</div>
<div id="plots" class="section level3">
<h3>plots</h3>
<pre class="r"><code>genes_unmapped &lt;-  ggplot(summary_per_sample_reads_qc,
                   aes(x = gene_number, y = unmapped_ratios * 100,
                   col = as.factor(individual), height = 600, width = 2000)) +
                   geom_point(size = 3, alpha = 0.3) + 
                   geom_vline(xintercept = cut_off_genes, colour=&quot;grey&quot;, linetype = &quot;longdash&quot;) +
                   geom_hline(yintercept = cut_off_unmapped * 100, colour=&quot;grey&quot;, linetype = &quot;longdash&quot;) +
                   labs(x = &quot;Number of genes&quot;, y = &quot;Unmapped reads percentage (%)&quot;) 

genes_ercc &lt;-  ggplot(summary_per_sample_reads_qc,
                   aes(x = gene_number, y = ercc_percentage * 100,
                   col = as.factor(individual), shape = as.factor(replicate), height = 600, width = 2000)) +
                   geom_point(size = 3, alpha = 0.3) +
                   geom_vline(xintercept = cut_off_genes, colour=&quot;grey&quot;, linetype = &quot;longdash&quot;) +
                   geom_hline(yintercept = cut_off_ercc * 100, colour=&quot;grey&quot;, linetype = &quot;longdash&quot;) +
                   labs(x = &quot;Number of genes&quot;, y = &quot;ERCC reads percentage (%)&quot;) 

reads_unmapped_num &lt;-  ggplot(summary_per_sample_reads_qc,
                   aes(x = total_mapped, y = unmapped_ratios * 100,
                   col = as.factor(individual), label = as.character(cell_number), height = 600, width = 2000)) +
                   geom_text(fontface = 3, alpha = 0.5) + 
                   geom_vline(xintercept = cut_off_reads, colour=&quot;grey&quot;, linetype = &quot;longdash&quot;) +
                   geom_hline(yintercept = cut_off_unmapped * 100, colour=&quot;grey&quot;, linetype = &quot;longdash&quot;) +
                   labs(x = &quot;Total mapped reads&quot;, y = &quot;Unmapped reads percentage (%)&quot;) 

reads_ercc_num &lt;-  ggplot(summary_per_sample_reads_qc,
                   aes(x = total_mapped, y = ercc_percentage * 100,
                   col = as.factor(individual), label = as.character(cell_number), height = 600, width = 2000)) +
                   geom_text(fontface = 3, alpha = 0.5) + 
                   geom_vline(xintercept = cut_off_reads, colour=&quot;grey&quot;, linetype = &quot;longdash&quot;) +
                   geom_hline(yintercept = cut_off_ercc * 100, colour=&quot;grey&quot;, linetype = &quot;longdash&quot;) +
                   labs(x = &quot;Total mapped reads&quot;, y = &quot;ERCC reads percentage (%)&quot;) 

plot_grid(genes_unmapped + theme(legend.position = &quot;none&quot;), 
          genes_ercc + theme(legend.position = &quot;none&quot;),
          reads_unmapped_num + theme(legend.position = &quot;none&quot;), 
          reads_ercc_num + theme(legend.position = &quot;none&quot;),
          labels = LETTERS[1:4])</code></pre>
<p><img src="figure/qc-filter-ipsc.Rmd/plots-1.png" title="" alt="" width="1152" style="display: block; margin: auto;" /></p>
<pre class="r"><code>plot_grid(genes_unmapped + theme(legend.position = c(.75,.9)) + labs(col = &quot;Individual&quot;), 
          reads_unmapped_num + theme(legend.position = &quot;none&quot;), 
          reads_ercc_num + theme(legend.position = &quot;none&quot;),
          labels = LETTERS[3:5],
          nrow = 1)</code></pre>
<p><img src="figure/qc-filter-ipsc.Rmd/paper-fig1-1.png" title="" alt="" width="1152" style="display: block; margin: auto;" /></p>
<pre class="r"><code>plot_grid(ercc_reads_plot + theme(legend.position = &quot;none&quot;),
          ercc_molecule_plot + theme(legend.position = &quot;none&quot;),
          plot_reads + theme(legend.position=c(.8,.85)),
          plot_unmapped + theme(legend.position = &quot;none&quot;),
          plot_ercc + theme(legend.position = &quot;none&quot;), 
          plot_gene + theme(legend.position = &quot;none&quot;),
          labels = LETTERS[1:6],
          ncol = 2)</code></pre>
<p><img src="figure/qc-filter-ipsc.Rmd/paper-fig-1.png" title="" alt="" width="1152" style="display: block; margin: auto;" /></p>
<pre class="r"><code>plot_grid(plot_before + theme(legend.position=c(.85,.85)) + labs(col = &quot;Cell number&quot;), 
          plot_after + theme(legend.position = &quot;none&quot;),
          plot_ercc_before, 
          plot_ercc_after,
          labels = LETTERS[1:4])</code></pre>
<p><img src="figure/qc-filter-ipsc.Rmd/paper-fig-2-1.png" title="" alt="" width="1152" style="display: block; margin: auto;" /></p>
</div>
</div>
<div id="session-information" class="section level2">
<h2>Session information</h2>
<pre class="r"><code>sessionInfo()</code></pre>
<pre><code>R version 3.2.0 (2015-04-16)
Platform: x86_64-unknown-linux-gnu (64-bit)

locale:
 [1] LC_CTYPE=en_US.UTF-8       LC_NUMERIC=C              
 [3] LC_TIME=en_US.UTF-8        LC_COLLATE=en_US.UTF-8    
 [5] LC_MONETARY=en_US.UTF-8    LC_MESSAGES=en_US.UTF-8   
 [7] LC_PAPER=en_US.UTF-8       LC_NAME=C                 
 [9] LC_ADDRESS=C               LC_TELEPHONE=C            
[11] LC_MEASUREMENT=en_US.UTF-8 LC_IDENTIFICATION=C       

attached base packages:
[1] stats     graphics  grDevices utils     datasets  methods   base     

other attached packages:
[1] MASS_7.3-40   cowplot_0.3.1 ggplot2_1.0.1 edgeR_3.10.2  limma_3.24.9 
[6] dplyr_0.4.2   knitr_1.10.5 

loaded via a namespace (and not attached):
 [1] Rcpp_0.12.0        magrittr_1.5       munsell_0.4.2     
 [4] colorspace_1.2-6   R6_2.1.1           stringr_1.0.0     
 [7] httr_0.6.1         plyr_1.8.3         tools_3.2.0       
[10] parallel_3.2.0     grid_3.2.0         gtable_0.1.2      
[13] DBI_0.3.1          htmltools_0.2.6    lazyeval_0.1.10   
[16] yaml_2.1.13        assertthat_0.1     digest_0.6.8      
[19] RColorBrewer_1.1-2 reshape2_1.4.1     formatR_1.2       
[22] bitops_1.0-6       RCurl_1.95-4.6     evaluate_0.7      
[25] rmarkdown_0.6.1    labeling_0.3       stringi_0.4-1     
[28] scales_0.2.4       proto_0.3-10      </code></pre>
</div>


<!-- some extra javascript for older browsers -->
<script type="text/javascript" src="libs/polyfill.js"></script>

<script>

// manage active state of menu based on current page
$(document).ready(function () {

    // active menu
    href = window.location.pathname
    href = href.substr(href.lastIndexOf('/') + 1)
    $('a[href="' + href + '"]').parent().addClass('active');

    // manage active menu header
    if (href.startsWith('authoring_'))
      $('a[href="' + 'authoring' + '"]').parent().addClass('active');
    else if (href.endsWith('_format.html'))
      $('a[href="' + 'formats' + '"]').parent().addClass('active');
    else if (href.startsWith('developer_'))
      $('a[href="' + 'developer' + '"]').parent().addClass('active');

});

</script>

</div>

<script>

// add bootstrap table styles to pandoc tables
$(document).ready(function () {
  $('tr.header').parent('thead').parent('table').addClass('table table-condensed');
});

</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://cdn.mathjax.org/mathjax/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
