<!DOCTYPE html>

<html xmlns="http://www.w3.org/1999/xhtml">

<head>

<meta charset="utf-8">
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="generator" content="pandoc" />

<meta name="author" content="Po-Yuan Tung" />

<meta name="date" content="2015-07-10" />

<title>stemness pluripotency</title>

<script src="libs/jquery-1.11.0/jquery.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="libs/bootstrap-3.3.1/css/united.min.css" rel="stylesheet" />
<script src="libs/bootstrap-3.3.1/js/bootstrap.min.js"></script>
<script src="libs/bootstrap-3.3.1/shim/html5shiv.min.js"></script>
<script src="libs/bootstrap-3.3.1/shim/respond.min.js"></script>

<style type="text/css">

/* padding for bootstrap navbar */
body {
  padding-top: 50px;
  padding-bottom: 40px;
}


/* offset scroll position for anchor links (for fixed navbar)  */
.section h2 {
  padding-top: 55px;
  margin-top: -55px;
}
.section h3 {
  padding-top: 55px;
  margin-top: -55px;
}



/* don't use link color in navbar */
.dropdown-menu>li>a {
  color: black;
}

/* some padding for disqus */
#disqus_thread {
  margin-top: 45px;
}

</style>

<link rel="stylesheet" href="libs/font-awesome-4.1.0/css/font-awesome.min.css"/>

<style type="text/css">code{white-space: pre;}</style>
<link rel="stylesheet"
      href="libs/highlight/textmate.css"
      type="text/css" />
<script src="libs/highlight/highlight.js"></script>
<style type="text/css">
  pre:not([class]) {
    background-color: white;
  }
</style>
<script type="text/javascript">
if (window.hljs && document.readyState && document.readyState === "complete") {
   window.setTimeout(function() {
      hljs.initHighlighting();
   }, 0);
}
</script>



</head>

<body>

<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
code {
  color: inherit;
  background-color: rgba(0, 0, 0, 0.04);
}
img { 
  max-width:100%; 
  height: auto; 
}
</style>
<div class="container-fluid main-container">


<div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">singleCellSeq</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li><a href="index.html">Home</a></li>
        <li><a href="about.html">About</a></li>
        <li><a href="license.html">License</a></li>
        <li><a href="https://github.com/jdblischak/singleCellSeq">GitHub</a></li>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<div id="header">
<h1 class="title">stemness pluripotency</h1>
<h4 class="author"><em>Po-Yuan Tung</em></h4>
<h4 class="date"><em>2015-07-10</em></h4>
</div>

<div id="TOC">
<ul>
<li><a href="#input">Input</a></li>
<li><a href="#expressions-of-pluripotency-genes-in-ipscs">Expressions of pluripotency genes in iPSCs</a></li>
<li><a href="#use-lcls-as-a-negative-control">Use LCLs as a negative control</a></li>
<li><a href="#the-cv-between-ipsc-and-lcl">The CV between iPSC and LCL</a></li>
</ul>
</div>

<p><strong>Last updated:</strong> 2015-08-28</p>
<p><strong>Code version:</strong> 4b60929b44b585789da8bbab66261d3d17e08b6c</p>
<p>Current model of reprogramming suggests that pluripotency occurs in two-phases: a prolonged stochastic phase followed by a rapid deterministic phase. <a href="http://journals.plos.org/plosone/article?id=10.1371/journal.pone.0095304">Chung2014</a> Our iPSCs were reprogrammed and cultured in mane different batches. In addition, some lines have been through different culture mediums. In order to make sure that the variance we observed at the single cell level is not cause by different levels of plupotency, we need to more rigorously estimate the cells status than just carrying out pluritest from the bulk.</p>
<div id="input" class="section level2">
<h2>Input</h2>
<pre class="r"><code>library(&quot;dplyr&quot;)
library(&quot;ggplot2&quot;)
theme_set(theme_bw(base_size = 16))
library(&quot;edgeR&quot;)
library(&quot;gplots&quot;)
library(&quot;RColorBrewer&quot;)</code></pre>
<p>Input annotation.</p>
<pre class="r"><code>anno &lt;- read.table(&quot;../data/annotation.txt&quot;, header = TRUE,
                   stringsAsFactors = FALSE)
head(anno)</code></pre>
<pre><code>  individual batch well     sample_id
1      19098     1  A01 NA19098.1.A01
2      19098     1  A02 NA19098.1.A02
3      19098     1  A03 NA19098.1.A03
4      19098     1  A04 NA19098.1.A04
5      19098     1  A05 NA19098.1.A05
6      19098     1  A06 NA19098.1.A06</code></pre>
<p>Input molecule counts.</p>
<pre class="r"><code>molecules &lt;- read.table(&quot;../data/molecules.txt&quot;, header = TRUE,
                    stringsAsFactors = FALSE)</code></pre>
<p>Input read counts.</p>
<pre class="r"><code>reads &lt;- read.table(&quot;../data/reads.txt&quot;, header = TRUE,
                    stringsAsFactors = FALSE)</code></pre>
<p>Input list of quality single cells.</p>
<pre class="r"><code>quality_single_cells &lt;- scan(&quot;../data/quality-single-cells.txt&quot;,
                             what = &quot;character&quot;)</code></pre>
<p>Keep only the single cells that passed the <a href="qc-cell-ipsc.html">QC filters</a> and the bulk samples.</p>
<pre class="r"><code>molecules &lt;- molecules[, grepl(&quot;bulk&quot;, colnames(molecules)) |
                         colnames(molecules) %in% quality_single_cells]
anno &lt;- anno[anno$well == &quot;bulk&quot; | anno$sample_id %in% quality_single_cells, ]
stopifnot(ncol(molecules) == nrow(anno),
          colnames(molecules) == anno$sample_id)

reads &lt;- reads[, grepl(&quot;bulk&quot;, colnames(reads)) |
                         colnames(reads) %in% quality_single_cells]
stopifnot(ncol(reads) == nrow(anno),
          colnames(reads) == anno$sample_id)</code></pre>
<p>Remove genes with zero read counts in the single cells or bulk samples.</p>
<pre class="r"><code>expressed &lt;- rowSums(molecules[, anno$well == &quot;bulk&quot;]) &gt; 0 &amp;
             rowSums(molecules[, anno$well != &quot;bulk&quot;]) &gt; 0
molecules &lt;- molecules[expressed, ]
dim(molecules)</code></pre>
<pre><code>[1] 17153   587</code></pre>
<pre class="r"><code>expressed &lt;- rowSums(reads[, anno$well == &quot;bulk&quot;]) &gt; 0 &amp;
             rowSums(reads[, anno$well != &quot;bulk&quot;]) &gt; 0
reads &lt;- reads[expressed, ]
dim(reads)</code></pre>
<pre><code>[1] 17162   587</code></pre>
<p>Split the bulk and single samples.</p>
<pre class="r"><code>molecules_bulk &lt;- molecules[, anno$well == &quot;bulk&quot;]
molecules_single &lt;- molecules[, anno$well != &quot;bulk&quot;]
reads_bulk &lt;- reads[, anno$well == &quot;bulk&quot;]
reads_single &lt;- reads[, anno$well != &quot;bulk&quot;]</code></pre>
<p>Remove genes with max molecule numer larger than 1024</p>
<pre class="r"><code>molecules_single &lt;- molecules_single[apply(molecules_single,1,max) &lt; 1024,]</code></pre>
<p>Correct for collision probability. See <a href="http://www.nature.com/nmeth/journal/v11/n6/full/nmeth.2930.html#methods">Grun et al. 2014</a> for details.</p>
<pre class="r"><code>molecules_single_collision &lt;- -1024 * log(1 - molecules_single / 1024)</code></pre>
<p>Standardization</p>
<pre class="r"><code>molecules_single_cpm &lt;- cpm(molecules_single_collision, log = TRUE)</code></pre>
<p>Calculate TMM-normalized read counts per million.</p>
<pre class="r"><code>norm_factors_bulk &lt;- calcNormFactors(reads_bulk, method = &quot;TMM&quot;)
reads_bulk_cpm &lt;- cpm(reads_bulk, log = TRUE,
                          lib.size = colSums(reads_bulk) * norm_factors_bulk)</code></pre>
</div>
<div id="expressions-of-pluripotency-genes-in-ipscs" class="section level2">
<h2>Expressions of pluripotency genes in iPSCs</h2>
<p>Input pluripotency genes. A list of 27 pluripotency genes used to demonstrate iPSC heterogeneity in <a href="http://www.jci.org/articles/view/44635">Narshinh2011</a> Gene ID conversion was done by using the DAVID <a href="http://david.abcc.ncifcrf.gov" class="uri">http://david.abcc.ncifcrf.gov</a></p>
<pre class="r"><code>pluripotency_genes &lt;- read.table(&quot;../data/pluripotency-genes.txt&quot;, header = TRUE, sep=&quot;\t&quot;)</code></pre>
<p>Expression of plurypotency genes in iPSCs</p>
<pre class="r"><code>molecules_single_pluripotency &lt;- molecules_single_cpm[rownames(molecules_single_cpm) %in% pluripotency_genes[,2],]

reads_bulk_pluripotency &lt;- reads_bulk_cpm[rownames(reads_bulk_cpm) %in% pluripotency_genes[,2],]

### heatmap 
individual &lt;- colnames(molecules_single_pluripotency)
color &lt;- rep(&quot;yellow&quot;,dim(molecules_single_pluripotency)[2])
color[grep(&quot;19098&quot;, individual)] &lt;- &quot;red&quot;
color[grep(&quot;19239&quot;, individual)] &lt;- &quot;blue&quot;

heatmap.2(molecules_single_pluripotency, trace=&quot;none&quot;, cexRow=1, cexCol=1, margins=c(8,8), xlab=&quot;single cells&quot;, ColSideColors=color)</code></pre>
<p><img src="figure/stemness.Rmd/expressed-pluripotency-gene-1.png" title="" alt="" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>colorbulk &lt;- rep(c(&quot;red&quot;,&quot;yellow&quot;,&quot;blue&quot;),each=3)
heatmap.2(reads_bulk_pluripotency, trace=&quot;none&quot;, cexRow=1, cexCol=1, margins=c(8,8),xlab=&quot;bulk&quot;, ColSideColors=colorbulk)</code></pre>
<p><img src="figure/stemness.Rmd/expressed-pluripotency-gene-2.png" title="" alt="" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>pca_single_cells &lt;- prcomp(t(molecules_single_pluripotency), retx = TRUE, scale. = TRUE, center = TRUE)
plot(pca_single_cells)</code></pre>
<p><img src="figure/stemness.Rmd/expressed-pluripotency-gene-pca-1.png" title="" alt="" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>pca_single_cells$perc_explained &lt;- pca_single_cells$sdev^2 / sum(pca_single_cells$sdev^2) * 100
plot(pca_single_cells$perc_explained)</code></pre>
<p><img src="figure/stemness.Rmd/expressed-pluripotency-gene-pca-2.png" title="" alt="" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>anno_single &lt;- anno[anno$well != &quot;bulk&quot;,]
stopifnot(colnames(pca_single_cells) == 
            paste(paste0(&quot;NA&quot;, anno_single$individual),
                   anno_single$batch,
                   anno_single$well, sep = &quot;.&quot;))

pca_single_cells_anno &lt;- cbind(anno_single, pca_single_cells$x)
ggplot(pca_single_cells_anno, aes(x = PC1, y = PC2, col = as.factor(individual), shape = as.factor(batch))) +geom_point()</code></pre>
<p><img src="figure/stemness.Rmd/expressed-pluripotency-gene-pca-3.png" title="" alt="" width="672" style="display: block; margin: auto;" /></p>
</div>
<div id="use-lcls-as-a-negative-control" class="section level2">
<h2>Use LCLs as a negative control</h2>
<p>Input LCL annotation.</p>
<pre class="r"><code>anno_lcl &lt;- read.table(&quot;../data/annotation-lcl.txt&quot;, header = TRUE,
                   stringsAsFactors = FALSE)
head(anno_lcl)</code></pre>
<pre><code>  individual batch well     sample_id full_lane
1      19239     1  A01 NA19239.1.A01     FALSE
2      19239     1  A02 NA19239.1.A02     FALSE
3      19239     1  A03 NA19239.1.A03     FALSE
4      19239     1  A04 NA19239.1.A04     FALSE
5      19239     1  A05 NA19239.1.A05     FALSE
6      19239     1  A06 NA19239.1.A06     FALSE</code></pre>
<p>Input LCL molecule counts.</p>
<pre class="r"><code>molecules_lcl &lt;- read.table(&quot;../data/molecules-lcl.txt&quot;, header = TRUE,
                    stringsAsFactors = FALSE)</code></pre>
<p>Input LCL read counts.</p>
<pre class="r"><code>reads_lcl &lt;- read.table(&quot;../data/reads-lcl.txt&quot;, header = TRUE,
                    stringsAsFactors = FALSE)</code></pre>
<p>Input list of quality single cells of LCL.</p>
<pre class="r"><code>quality_single_cells_lcl &lt;- scan(&quot;../data/quality-single-cells-lcl.txt&quot;,
                             what = &quot;character&quot;)</code></pre>
<p>Keep only the single cells that passed the <a href="qc-cell-lcl.html">QC filters</a> and the bulk samples.</p>
<pre class="r"><code>molecules_lcl &lt;- molecules_lcl[colnames(molecules_lcl) %in% quality_single_cells_lcl]
anno_lcl &lt;- anno_lcl[anno_lcl$full_lane == &quot;FALSE&quot; &amp; anno_lcl$sample_id %in% quality_single_cells_lcl, ]
stopifnot(ncol(molecules_lcl) == nrow(anno_lcl),
          colnames(molecules_lcl) == anno_lcl$sample_id)
molecules_single_lcl &lt;- molecules_lcl[, anno_lcl$full_lane == &quot;FALSE&quot;]</code></pre>
<pre class="r"><code>molecules_single_lcl &lt;- molecules_single_lcl[apply(molecules_single_lcl,1,max) &lt; 1024,]
molecules_single_collision_lcl &lt;- -1024 * log(1 - molecules_single_lcl / 1024)</code></pre>
<p>Standardization</p>
<pre class="r"><code>molecules_single_cpm_lcl &lt;- cpm(molecules_single_collision_lcl, log = TRUE)</code></pre>
<p>Expression of plurypotency genes in LCL</p>
<pre class="r"><code>molecules_single_pluripotency_lcl &lt;- molecules_single_cpm_lcl[rownames(molecules_single_cpm_lcl) %in% pluripotency_genes[,2],]

heatmap.2(molecules_single_pluripotency_lcl, trace=&quot;none&quot;, cexRow=1, cexCol=1, margins=c(8,8),xlab=&quot;single cells&quot;, ylab= &quot;pluripotency gene&quot;)</code></pre>
<p><img src="figure/stemness.Rmd/expressed-pluripotency-gene-lcl-1.png" title="" alt="" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>### merge two cell types
molecules_single_pluripotency_all &lt;- merge(molecules_single_pluripotency,molecules_single_pluripotency_lcl,by = &quot;row.names&quot;, all = TRUE)

### remove the column and na for data.matrix
molecules_single_pluripotency_all_matrix  &lt;- as.matrix(molecules_single_pluripotency_all[,2:654])
rownames(molecules_single_pluripotency_all_matrix) &lt;- molecules_single_pluripotency_all[,1]
molecules_single_pluripotency_all_matrix[is.na(molecules_single_pluripotency_all_matrix)] &lt;- 0

### heatmap
heatmap.2(molecules_single_pluripotency_all_matrix, trace=&quot;none&quot;, cexRow=1, cexCol=1, margins=c(8,8), xlab=&quot;single cells&quot;, ylab= &quot;pluripotency gene&quot;)</code></pre>
<p><img src="figure/stemness.Rmd/expressed-pluripotency-gene-all-1.png" title="" alt="" width="672" style="display: block; margin: auto;" /></p>
</div>
<div id="the-cv-between-ipsc-and-lcl" class="section level2">
<h2>The CV between iPSC and LCL</h2>
<p>The CV values are calculated based on standardized molecule numbers (cpm and log transformed)</p>
<pre class="r"><code>## select 19239 iPSC
molecules_single_cpm_19239 &lt;- molecules_single_cpm[,grep(19239,colnames(molecules_single_cpm))]

## keep the common genes iPSCs and LCLs data
molecules_single_cpm_lcl_19239 &lt;- molecules_single_cpm_lcl[rownames(molecules_single_cpm_lcl) %in% rownames(molecules_single_cpm_19239), ]

molecules_single_cpm_iPSC_19239 &lt;- molecules_single_cpm_19239[rownames(molecules_single_cpm_19239) %in% rownames(molecules_single_cpm_lcl_19239),]

## calculate CV
CV &lt;- function(x){apply(x,1,sd)/apply(x,1,mean)}
CV_iPSC &lt;- CV(molecules_single_cpm_iPSC_19239)
CV_LCL &lt;- CV(molecules_single_cpm_lcl_19239)

## combine the CV values from the 2 cell types
iPSC_LCL &lt;- as.data.frame(cbind(CV_iPSC,CV_LCL))

## select ERCC
iPSC_LCL$ERCC &lt;- grepl(&quot;ERCC&quot;, rownames(iPSC_LCL))

## color palette
cbPalette &lt;- c(&quot;#999999&quot;, &quot;#0000FF&quot;, &quot;#990033&quot;, &quot;#F0E442&quot;, &quot;#0072B2&quot;, &quot;#D55E00&quot;, &quot;#CC79A7&quot;, &quot;#009E73&quot;)

ggplot(iPSC_LCL, aes(x = CV_iPSC, y = CV_LCL, col = ERCC)) + geom_point(size = 3, alpha = 0.5) + scale_colour_manual(values=cbPalette) + stat_function(fun= function(x) {x}, col= &quot;#56B4E9&quot;) + labs(x = &quot;CV of iPSC&quot;, y =  &quot;CV of LCL&quot;, title = &quot;cpm with log&quot;)</code></pre>
<p><img src="figure/stemness.Rmd/CV-iPSC-LCL-1.png" title="" alt="" width="672" style="display: block; margin: auto;" /></p>
<p>Use the non-standardized molecule numbers (no cpm) or standardized but not log transformed (cpm no log)</p>
<pre class="r"><code>## select 19239 iPSC
molecules_single_collision_19239 &lt;- molecules_single_collision[,grep(19239,colnames(molecules_single_collision))]

## keep the common genes iPSCs and LCLs data
molecules_single_collision_lcl_19239 &lt;- molecules_single_collision_lcl[rownames(molecules_single_collision_lcl) %in% rownames(molecules_single_collision_19239), ]

molecules_single_collision_iPSC_19239 &lt;- molecules_single_collision_19239[rownames(molecules_single_collision_19239) %in% rownames(molecules_single_collision_lcl_19239),]

## calculate CV
iPSC_LCL$CV_iPSC_collision &lt;- CV(molecules_single_collision_iPSC_19239)
iPSC_LCL$CV_LCL_collision &lt;- CV(molecules_single_collision_lcl_19239)
iPSC_LCL$mean_iPSC_collision &lt;- apply(molecules_single_collision_iPSC_19239,1, mean)
iPSC_LCL$mean_LCL_collision &lt;- apply(molecules_single_collision_lcl_19239,1, mean)

ggplot(iPSC_LCL, aes(x = mean_iPSC_collision, y = CV_iPSC_collision, col = ERCC)) + geom_point(size = 3, alpha = 0.5) + scale_colour_manual(values=cbPalette) + scale_x_log10()</code></pre>
<pre><code>Warning: Removed 1073 rows containing missing values (geom_point).</code></pre>
<p><img src="figure/stemness.Rmd/CV-iPSC-LCL-non-1.png" title="" alt="" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>ggplot(iPSC_LCL, aes(x = mean_LCL_collision, y = CV_LCL_collision, col = ERCC)) + geom_point(size = 3, alpha = 0.5) + scale_colour_manual(values=cbPalette) + scale_x_log10()</code></pre>
<pre><code>Warning: Removed 4184 rows containing missing values (geom_point).</code></pre>
<p><img src="figure/stemness.Rmd/CV-iPSC-LCL-non-2.png" title="" alt="" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>ggplot(iPSC_LCL, aes(x = CV_iPSC_collision, y = CV_LCL_collision, col = ERCC)) + geom_point(size = 3, alpha = 0.5) + scale_colour_manual(values=cbPalette) + stat_function(fun= function(x) {x}, col= &quot;#56B4E9&quot;) + scale_x_log10() + scale_y_log10() + labs(x = &quot;CV of iPSC&quot;, y =  &quot;CV of LCL&quot;, title = &quot;non standarized&quot;)</code></pre>
<pre><code>Warning: Removed 4474 rows containing missing values (geom_point).</code></pre>
<p><img src="figure/stemness.Rmd/CV-iPSC-LCL-non-3.png" title="" alt="" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>## cpm without log 
molecules_single_cpmnolog_lcl_19239 &lt;- cpm(molecules_single_collision_lcl_19239, log = FALSE)
molecules_single_cpmnolog_iPSC_19239 &lt;- cpm(molecules_single_collision_iPSC_19239, log = FALSE)

## calculate CV of cpm no log
iPSC_LCL$CV_iPSC_cpmnolog &lt;- CV(molecules_single_cpmnolog_iPSC_19239)
iPSC_LCL$CV_LCL_cpmnolog &lt;- CV(molecules_single_cpmnolog_lcl_19239)

ggplot(iPSC_LCL, aes(x = CV_iPSC_cpmnolog, y = CV_LCL_cpmnolog, col = ERCC)) + geom_point(size = 3, alpha = 0.5) + scale_colour_manual(values=cbPalette) + stat_function(fun= function(x) {x}, col= &quot;#56B4E9&quot;) + scale_x_log10() + scale_y_log10() + labs(x = &quot;CV of iPSC&quot;, y =  &quot;CV of LCL&quot;, title = &quot;cpm no log&quot;)</code></pre>
<pre><code>Warning: Removed 4474 rows containing missing values (geom_point).</code></pre>
<p><img src="figure/stemness.Rmd/CV-iPSC-LCL-non-4.png" title="" alt="" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>## density plot
CV_cmpnolog &lt;- data.frame(CV = c(iPSC_LCL$CV_iPSC_cpmnolog,iPSC_LCL$CV_LCL_cpmnolog), cell_type = rep(c(&quot;iPSC&quot;, &quot;LCL&quot;), each = 17130))
ggplot(CV_cmpnolog, aes(x = CV, fill = cell_type)) + geom_density(alpha = 0.5) + labs(title = &quot;cpm no log&quot;)</code></pre>
<pre><code>Warning: Removed 1073 rows containing non-finite values (stat_density).</code></pre>
<pre><code>Warning: Removed 4184 rows containing non-finite values (stat_density).</code></pre>
<p><img src="figure/stemness.Rmd/CV-iPSC-LCL-non-5.png" title="" alt="" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>## MA plot
iPSC_LCL$fold_change_CV &lt;- log2(iPSC_LCL$CV_iPSC_cpmnolog / iPSC_LCL$CV_LCL_cpmnolog)
iPSC_LCL$mean_CV &lt;- 0.5 * log2((iPSC_LCL$CV_iPSC_cpmnolog + iPSC_LCL$CV_LCL_cpmnolog))

ggplot(iPSC_LCL, aes(x = mean_CV, y = fold_change_CV, col = ERCC)) + geom_point(size = 3, alpha = 0.5) + scale_colour_manual(values=cbPalette)  + labs(x = &quot;1/2(iPSC.CV + LCL.CV)&quot;, y =  &quot;log2(iPSC.CV/LCL.CV)&quot;, title = &quot;cpm no log&quot;)</code></pre>
<pre><code>Warning: Removed 4474 rows containing missing values (geom_point).</code></pre>
<p><img src="figure/stemness.Rmd/CV-iPSC-LCL-non-6.png" title="" alt="" width="672" style="display: block; margin: auto;" /></p>
<p>Look at only batch one of 19239 to see if sample size (cell numbers) matters.</p>
<pre class="r"><code>## look at only one batah of iPSC
molecules_single_cpmnolog_iPSC_19239.1 &lt;- molecules_single_cpmnolog_iPSC_19239[,grep(19239.1,colnames(molecules_single_cpmnolog_iPSC_19239))]
iPSC_LCL$CV_iPSC_cpmnolog_1 &lt;- CV(molecules_single_cpmnolog_iPSC_19239.1)

ggplot(iPSC_LCL, aes(x = CV_iPSC_cpmnolog_1, y = CV_LCL_cpmnolog, col = ERCC)) + geom_point(size = 3, alpha = 0.5) + scale_colour_manual(values=cbPalette) + stat_function(fun= function(x) {x}, col= &quot;#56B4E9&quot;) + scale_x_log10() + scale_y_log10() + labs(x = &quot;CV of iPSC batch 1&quot;, y =  &quot;CV of LCL&quot;, title = &quot;cpm no log&quot;)</code></pre>
<pre><code>Warning: Removed 5036 rows containing missing values (geom_point).</code></pre>
<p><img src="figure/stemness.Rmd/CV-iPSC-LCL-19293batch1-1.png" title="" alt="" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>CV_cmpnolog_1 &lt;- data.frame(CV = c(iPSC_LCL$CV_iPSC_cpmnolog_1,iPSC_LCL$CV_LCL_cpmnolog), cell_type = rep(c(&quot;iPSC&quot;, &quot;LCL&quot;), each = 17130))
ggplot(CV_cmpnolog_1, aes(x = CV, fill = cell_type)) + geom_density(alpha = 0.5) + labs(title = &quot;cpm no log, 19239 batch 1&quot;)</code></pre>
<pre><code>Warning: Removed 2477 rows containing non-finite values (stat_density).</code></pre>
<pre><code>Warning: Removed 4184 rows containing non-finite values (stat_density).</code></pre>
<p><img src="figure/stemness.Rmd/CV-iPSC-LCL-19293batch1-2.png" title="" alt="" width="672" style="display: block; margin: auto;" /> ## Session information</p>
<pre class="r"><code>sessionInfo()</code></pre>
<pre><code>R version 3.2.0 (2015-04-16)
Platform: x86_64-unknown-linux-gnu (64-bit)

locale:
 [1] LC_CTYPE=en_US.UTF-8       LC_NUMERIC=C              
 [3] LC_TIME=en_US.UTF-8        LC_COLLATE=en_US.UTF-8    
 [5] LC_MONETARY=en_US.UTF-8    LC_MESSAGES=en_US.UTF-8   
 [7] LC_PAPER=en_US.UTF-8       LC_NAME=C                 
 [9] LC_ADDRESS=C               LC_TELEPHONE=C            
[11] LC_MEASUREMENT=en_US.UTF-8 LC_IDENTIFICATION=C       

attached base packages:
[1] stats     graphics  grDevices utils     datasets  methods   base     

other attached packages:
[1] RColorBrewer_1.1-2 gplots_2.17.0      edgeR_3.10.2      
[4] limma_3.24.9       ggplot2_1.0.1      dplyr_0.4.2       
[7] knitr_1.10.5      

loaded via a namespace (and not attached):
 [1] Rcpp_0.12.0        magrittr_1.5       MASS_7.3-40       
 [4] munsell_0.4.2      colorspace_1.2-6   R6_2.1.1          
 [7] stringr_1.0.0      plyr_1.8.3         caTools_1.17.1    
[10] tools_3.2.0        parallel_3.2.0     grid_3.2.0        
[13] gtable_0.1.2       KernSmooth_2.23-14 DBI_0.3.1         
[16] gtools_3.5.0       htmltools_0.2.6    yaml_2.1.13       
[19] assertthat_0.1     digest_0.6.8       reshape2_1.4.1    
[22] formatR_1.2        bitops_1.0-6       evaluate_0.7      
[25] rmarkdown_0.6.1    labeling_0.3       gdata_2.16.1      
[28] stringi_0.4-1      scales_0.2.4       proto_0.3-10      </code></pre>
</div>


<!-- some extra javascript for older browsers -->
<script type="text/javascript" src="libs/polyfill.js"></script>

<script>

// manage active state of menu based on current page
$(document).ready(function () {

    // active menu
    href = window.location.pathname
    href = href.substr(href.lastIndexOf('/') + 1)
    $('a[href="' + href + '"]').parent().addClass('active');

    // manage active menu header
    if (href.startsWith('authoring_'))
      $('a[href="' + 'authoring' + '"]').parent().addClass('active');
    else if (href.endsWith('_format.html'))
      $('a[href="' + 'formats' + '"]').parent().addClass('active');
    else if (href.startsWith('developer_'))
      $('a[href="' + 'developer' + '"]').parent().addClass('active');

});

</script>

</div>

<script>

// add bootstrap table styles to pandoc tables
$(document).ready(function () {
  $('tr.header').parent('thead').parent('table').addClass('table table-condensed');
});

</script>


</body>
</html>
