---
title: "stemness pluripotency"
author: "Po-Yuan Tung"
date: 2015-07-10
---

**Last updated:** `r Sys.Date()`

**Code version:** `r system("git log -1 --format='%H'", intern = TRUE)`

```{r chunk-options, include=FALSE}
source("chunk-options.R")
```

Current model of reprogramming suggests that pluripotency occurs in two-phases: a prolonged stochastic phase followed by a rapid deterministic phase. [Chung2014] 
Our iPSCs were reprogrammed and cultured in mane different batches. In addition, some lines have been through different culture mediums. In order to make sure that the variance we observed at the single cell level is not cause by different levels of plupotency, we need to more rigorously estimate the cells status than just carrying out pluritest from the bulk.

[Chung2014]:http://journals.plos.org/plosone/article?id=10.1371/journal.pone.0095304

## Input

```{r packages, message=FALSE}
library("dplyr")
library("ggplot2")
theme_set(theme_bw(base_size = 16))
library("edgeR")
library("gplots")
library("RColorBrewer")
```

Input annotation.

```{r input-annotation}
anno <- read.table("../data/annotation.txt", header = TRUE,
                   stringsAsFactors = FALSE)
head(anno)
```

Input molecule counts.

```{r input-molecule-counts}
molecules <- read.table("../data/molecules.txt", header = TRUE,
                    stringsAsFactors = FALSE)
```

Input read counts.

```{r input-read-counts}
reads <- read.table("../data/reads.txt", header = TRUE,
                    stringsAsFactors = FALSE)
```

Input list of quality single cells.

```{r input-quality-single-cells}
quality_single_cells <- scan("../data/quality-single-cells.txt",
                             what = "character")
```

Keep only the single cells that passed the [QC filters](qc-cell-ipsc.html) and the bulk samples.

```{r qc-filter}
molecules <- molecules[, grepl("bulk", colnames(molecules)) |
                         colnames(molecules) %in% quality_single_cells]
anno <- anno[anno$well == "bulk" | anno$sample_id %in% quality_single_cells, ]
stopifnot(ncol(molecules) == nrow(anno),
          colnames(molecules) == anno$sample_id)

reads <- reads[, grepl("bulk", colnames(reads)) |
                         colnames(reads) %in% quality_single_cells]
stopifnot(ncol(reads) == nrow(anno),
          colnames(reads) == anno$sample_id)
```

Remove genes with zero read counts in the single cells or bulk samples.

```{r remove-non-expressed-genes}
expressed <- rowSums(molecules[, anno$well == "bulk"]) > 0 &
             rowSums(molecules[, anno$well != "bulk"]) > 0
molecules <- molecules[expressed, ]
dim(molecules)

expressed <- rowSums(reads[, anno$well == "bulk"]) > 0 &
             rowSums(reads[, anno$well != "bulk"]) > 0
reads <- reads[expressed, ]
dim(reads)
```

Split the bulk and single samples.

```{r split-bulk-single}
molecules_bulk <- molecules[, anno$well == "bulk"]
molecules_single <- molecules[, anno$well != "bulk"]
reads_bulk <- reads[, anno$well == "bulk"]
reads_single <- reads[, anno$well != "bulk"]
```

Remove genes with max molecule numer larger than 1024
```{r remove-1024}
molecules_single <- molecules_single[apply(molecules_single,1,max) < 1024,]
```

Correct for collision probability. See [Grun et al. 2014][Grun2014] for details.

[Grun2014]: http://www.nature.com/nmeth/journal/v11/n6/full/nmeth.2930.html#methods

```{r collision-probability}
molecules_single_collision <- -1024 * log(1 - molecules_single / 1024)
```

Standardization
```{r calc-cpm-single}
molecules_single_cpm <- cpm(molecules_single_collision, log = TRUE)
```

Calculate TMM-normalized read counts per million.

```{r calc-cpm-bulk-reads}
norm_factors_bulk <- calcNormFactors(reads_bulk, method = "TMM")
reads_bulk_cpm <- cpm(reads_bulk, log = TRUE,
                          lib.size = colSums(reads_bulk) * norm_factors_bulk)
```

## Expressions of pluripotency genes in iPSCs

Input pluripotency genes.
A list of 27 pluripotency genes used to demonstrate iPSC heterogeneity in [Narshinh2011]
Gene ID conversion was done by using the DAVID http://david.abcc.ncifcrf.gov

[Narshinh2011]: http://www.jci.org/articles/view/44635

```{r input-pluripotency-gene}
pluripotency_genes <- read.table("../data/pluripotency-genes.txt", header = TRUE, sep="\t")
```

Expression of plurypotency genes in iPSCs
```{r expressed-pluripotency-gene}
molecules_single_pluripotency <- molecules_single_cpm[rownames(molecules_single_cpm) %in% pluripotency_genes[,2],]

reads_bulk_pluripotency <- reads_bulk_cpm[rownames(reads_bulk_cpm) %in% pluripotency_genes[,2],]

### heatmap 
individual <- colnames(molecules_single_pluripotency)
color <- rep("yellow",dim(molecules_single_pluripotency)[2])
color[grep("19098", individual)] <- "red"
color[grep("19239", individual)] <- "blue"

heatmap.2(molecules_single_pluripotency, trace="none", cexRow=1, cexCol=1, margins=c(8,8), xlab="single cells", ColSideColors=color)

colorbulk <- rep(c("red","yellow","blue"),each=3)
heatmap.2(reads_bulk_pluripotency, trace="none", cexRow=1, cexCol=1, margins=c(8,8),xlab="bulk", ColSideColors=colorbulk)

```

```{r expressed-pluripotency-gene-pca}
pca_single_cells <- prcomp(t(molecules_single_pluripotency), retx = TRUE, scale. = TRUE, center = TRUE)

plot(pca_single_cells)
```

## Use LCLs as a negative control 
Input LCL annotation.

```{r input-annotation-lcl}
anno_lcl <- read.table("../data/annotation-lcl.txt", header = TRUE,
                   stringsAsFactors = FALSE)
head(anno_lcl)
```

Input LCL molecule counts.

```{r input-molecule-counts-lcl}
molecules_lcl <- read.table("../data/molecules-lcl.txt", header = TRUE,
                    stringsAsFactors = FALSE)
```

Input LCL read counts.

```{r input-read-counts-lcl}
reads_lcl <- read.table("../data/reads-lcl.txt", header = TRUE,
                    stringsAsFactors = FALSE)
```

Input list of quality single cells of LCL.

```{r input-quality-single-cells-lcl}
quality_single_cells_lcl <- scan("../data/quality-single-cells-lcl.txt",
                             what = "character")
```

Keep only the single cells that passed the [QC filters](qc-cell-lcl.html) and the bulk samples.

```{r qc-filter-lcl}
molecules_lcl <- molecules_lcl[colnames(molecules_lcl) %in% quality_single_cells_lcl]
anno_lcl <- anno_lcl[anno_lcl$full_lane == "FALSE" & anno_lcl$sample_id %in% quality_single_cells_lcl, ]
stopifnot(ncol(molecules_lcl) == nrow(anno_lcl),
          colnames(molecules_lcl) == anno_lcl$sample_id)
molecules_single_lcl <- molecules_lcl[, anno_lcl$full_lane == "FALSE"]
```

```{r collision-probability-lcl}
molecules_single_lcl <- molecules_single_lcl[apply(molecules_single_lcl,1,max) < 1024,]
molecules_single_collision_lcl <- -1024 * log(1 - molecules_single_lcl / 1024)
```

Standardization
```{r calc-cpm-single-lcl}
molecules_single_cpm_lcl <- cpm(molecules_single_collision_lcl, log = TRUE)
```


Expression of plurypotency genes in LCL
```{r expressed-pluripotency-gene-lcl}
molecules_single_pluripotency_lcl <- molecules_single_cpm_lcl[rownames(molecules_single_cpm_lcl) %in% pluripotency_genes[,2],]

heatmap.2(molecules_single_pluripotency_lcl, trace="none", cexRow=1, cexCol=1, margins=c(8,8),xlab="single cells", ylab= "pluripotency gene")
```


```{r expressed-pluripotency-gene-all}
### merge two cell types
molecules_single_pluripotency_all <- merge(molecules_single_pluripotency,molecules_single_pluripotency_lcl,by = "row.names", all = TRUE)

### remove the column and na for data.matrix
molecules_single_pluripotency_all_matrix  <- as.matrix(molecules_single_pluripotency_all[,2:654])
rownames(molecules_single_pluripotency_all_matrix) <- molecules_single_pluripotency_all[,1]
molecules_single_pluripotency_all_matrix[is.na(molecules_single_pluripotency_all_matrix)] <- 0

### heatmap
heatmap.2(molecules_single_pluripotency_all_matrix, trace="none", cexRow=1, cexCol=1, margins=c(8,8), xlab="single cells", ylab= "pluripotency gene")
```

## The CV between iPSC and LCL 
The CV values are calculated based on standardized molecule numbers
```{r CV-iPSC-LCL}
## select 19239 iPSC
molecules_single_cpm_19239 <- molecules_single_cpm[,grep(19239,colnames(molecules_single_cpm))]

## keep the common genes iPSCs and LCLs data
molecules_single_cpm_lcl_19239 <- molecules_single_cpm_lcl[rownames(molecules_single_cpm_lcl) %in% rownames(molecules_single_cpm_19239), ]

molecules_single_cpm_iPSC_19239 <- molecules_single_cpm_19239[rownames(molecules_single_cpm_19239) %in% rownames(molecules_single_cpm_lcl_19239),]

## calculate CV
CV <- function(x){apply(x,1,sd)/apply(x,1,mean)}
CV_iPSC <- CV(molecules_single_cpm_iPSC_19239)
CV_LCL <- CV(molecules_single_cpm_lcl_19239)

## combine the CV values from the 2 cell types
iPSC_LCL <- as.data.frame(cbind(CV_iPSC,CV_LCL))

## select ERCC
iPSC_LCL$ERCC <- grepl("ERCC", rownames(iPSC_LCL))

## color palette
cbPalette <- c("#999999", "#0000FF", "#990033", "#F0E442", "#0072B2", "#D55E00", "#CC79A7", "#009E73")

ggplot(iPSC_LCL, aes(x = CV_iPSC, y = CV_LCL, col = ERCC)) + geom_point(size = 3, alpha = 0.5) + scale_colour_manual(values=cbPalette) + stat_function(fun= function(x) {x}, col= "#56B4E9") + labs(x = "CV of iPSC", y =  "CV of LCL", title = "cpm with log")

iPSC_LCL$mean_iPSC <- apply(molecules_single_cpm_iPSC_19239,1, mean)
iPSC_LCL$mean_LCL <- apply(molecules_single_cpm_lcl_19239,1, mean)

ggplot(iPSC_LCL, aes(x = mean_iPSC, y = mean_LCL, col = ERCC)) + geom_point(size = 3, alpha = 0.5) + scale_colour_manual(values=cbPalette) + stat_function(fun= function(x) {x}, col= "#56B4E9") 
```

Use the non-standardized molecule numbers
```{r CV-iPSC-LCL-non}
## select 19239 iPSC
molecules_single_collision_19239 <- molecules_single_collision[,grep(19239,colnames(molecules_single_collision))]

## keep the common genes iPSCs and LCLs data
molecules_single_collision_lcl_19239 <- molecules_single_collision_lcl[rownames(molecules_single_collision_lcl) %in% rownames(molecules_single_collision_19239), ]

molecules_single_collision_iPSC_19239 <- molecules_single_collision_19239[rownames(molecules_single_collision_19239) %in% rownames(molecules_single_collision_lcl_19239),]

## calculate CV
iPSC_LCL$CV_iPSC_collision <- CV(molecules_single_collision_iPSC_19239)
iPSC_LCL$CV_LCL_collision <- CV(molecules_single_collision_lcl_19239)
iPSC_LCL$mean_iPSC_collision <- apply(molecules_single_collision_iPSC_19239,1, mean)
iPSC_LCL$mean_LCL_collision <- apply(molecules_single_collision_lcl_19239,1, mean)

ggplot(iPSC_LCL, aes(x = mean_iPSC_collision, y = CV_iPSC_collision, col = ERCC)) + geom_point(size = 3, alpha = 0.5) + scale_colour_manual(values=cbPalette) + scale_x_log10()

ggplot(iPSC_LCL, aes(x = mean_LCL_collision, y = CV_LCL_collision, col = ERCC)) + geom_point(size = 3, alpha = 0.5) + scale_colour_manual(values=cbPalette) + scale_x_log10()

ggplot(iPSC_LCL, aes(x = CV_iPSC_collision, y = CV_LCL_collision, col = ERCC)) + geom_point(size = 3, alpha = 0.5) + scale_colour_manual(values=cbPalette) + stat_function(fun= function(x) {x}, col= "#56B4E9") + scale_x_log10() + scale_y_log10() + labs(x = "CV of iPSC", y =  "CV of LCL", title = "non standarized")

## cpm without log 
molecules_single_cpmnolog_lcl_19239 <- cpm(molecules_single_collision_lcl_19239, log = FALSE)
molecules_single_cpmnolog_iPSC_19239 <- cpm(molecules_single_collision_iPSC_19239, log = FALSE)

iPSC_LCL$CV_iPSC_cpmnolog <- CV(molecules_single_cpmnolog_iPSC_19239)
iPSC_LCL$CV_LCL_cpmnolog <- CV(molecules_single_cpmnolog_lcl_19239)

ggplot(iPSC_LCL, aes(x = CV_iPSC_cpmnolog, y = CV_LCL_cpmnolog, col = ERCC)) + geom_point(size = 3, alpha = 0.5) + scale_colour_manual(values=cbPalette) + stat_function(fun= function(x) {x}, col= "#56B4E9") + scale_x_log10() + scale_y_log10() + labs(x = "CV of iPSC", y =  "CV of LCL", title = "cpm no log")

## density plot
CV_cmpnolog <- data.frame(CV = c(iPSC_LCL$CV_iPSC_cpmnolog,iPSC_LCL$CV_LCL_cpmnolog), cell_type = rep(c("iPSC", "LCL"), each = 17130))
ggplot(CV_cmpnolog, aes(x = CV, fill = cell_type)) + geom_density(alpha = 0.5) + labs(title = "cpm no log")

## look at only one batah of iPSC
molecules_single_cpmnolog_iPSC_19239.1 <- molecules_single_cpmnolog_iPSC_19239[,grep(19239.1,colnames(molecules_single_cpmnolog_iPSC_19239))]
iPSC_LCL$CV_iPSC_cpmnolog_1 <- CV(molecules_single_cpmnolog_iPSC_19239.1)

ggplot(iPSC_LCL, aes(x = CV_iPSC_cpmnolog_1, y = CV_LCL_cpmnolog, col = ERCC)) + geom_point(size = 3, alpha = 0.5) + scale_colour_manual(values=cbPalette) + stat_function(fun= function(x) {x}, col= "#56B4E9") + scale_x_log10() + scale_y_log10() + labs(x = "CV of iPSC batch 1", y =  "CV of LCL", title = "cpm no log")

CV_cmpnolog_1 <- data.frame(CV = c(iPSC_LCL$CV_iPSC_cpmnolog_1,iPSC_LCL$CV_LCL_cpmnolog), cell_type = rep(c("iPSC", "LCL"), each = 17130))
ggplot(CV_cmpnolog_1, aes(x = CV, fill = cell_type)) + geom_density(alpha = 0.5) + labs(title = "cpm no log, 19239 batch 1")
```
## Session information

```{r info}
sessionInfo()
```
