---
title: "Subsample - LCLs"
date: 2015-06-30
---

**Last updated:** `r Sys.Date()`

**Code version:** `r system("git log -1 --format='%H'", intern = TRUE)`

```{r chunk-options, include=FALSE}
source("chunk-options.R")
```

Subsample reads post-mapping.
Includes both mapped and unmapped reads.

## Full lane samples

Lanes 5-8 each contained just one single cell.
These are highly oversequenced and will be used to assess saturation.

```bash
cd $ssd/lcl
mkdir -p full-lane/bam-combined
mkdir -p ~/log/subsample-bam.py
for WELL in A9E1 B2E2 B4H1 D2H2
do
  FILE=bam-combined/19239.1.$WELL.trim.sickle.sorted.combined.bam
  for NUM in 200000 1000000 10000000 20000000 30000000 40000000 50000000
  do
    echo "subsample-bam.py 12345 $NUM full-lane/bam-combined $FILE" | qsub -l h_vmem=8g -V -cwd -N sub.$WELL.$NUM -j y -o ~/log/subsample-bam.py
  done
done
```

```bash
ls full-lane/bam-combined
cat ~/log/subsample-bam.py/* | grep success | wc -l
cat ~/log/subsample-bam.py/* | grep failure | wc -l
```

Switch to directory `full-lane`.
Symlink `exons.saf`.

```bash
cd full-lane
mkdir genome
ln -s /mnt/lustre/home/jdblischak/singleCellSeq/data/exons.saf genome/exons.saf
```

Remove duplicate UMIs.

```bash
submit-array.sh rmdup-umi.sh 2g bam-combined/*bam
```

```bash
ls bam-rmdup-umi/*bam | wc -l
grep -w success ~/log/rmdup-umi.sh/* | wc -l
grep -w failure ~/log/rmdup-umi.sh/* | wc -l
```

Count reads per gene.

```bash
submit-array.sh count-reads-per-gene.sh 2g bam-combined/*bam bam-rmdup-umi/*bam
```

```bash
ls counts/*genecounts.txt | wc -l
grep -w success ~/log/count-reads-per-gene.sh/* | wc -l
grep -w failure ~/log/count-reads-per-gene.sh/* | wc -l
```

Gather the gene counts.

```bash
mkdir -p ~/log/gather-gene-counts.py
for NUM in 200000 1000000 10000000 20000000 30000000 40000000 50000000
do
  echo "gather-gene-counts.py counts/*.$NUM.genecounts.txt > read-counts-$NUM.txt" | qsub -l h_vmem=8g -cwd -V -j y -o ~/log/gather-gene-counts.py -N read-counts-$NUM
  echo "gather-gene-counts.py counts/*.$NUM.rmdup.genecounts.txt > molecule-counts-$NUM.txt" | qsub -l h_vmem=8g -cwd -V -j y -o ~/log/gather-gene-counts.py -N molecule-counts-$NUM
done
```

## Mutliplexed samples

Lanes 1-4 each contained 24 single cells.
These multiplexed cells will be used to compare to the iPSC results.

```bash
cd $ssd/lcl
mkdir -p multiplexed/bam-combined
mkdir -p ~/log/subsample-bam.py
for ROW in {A..H}
do
  for COL in {1..12}
  do
    WELL=`printf "%s%02d\n" $ROW $COL`
    FILE=bam-combined/19239.1.$WELL.trim.sickle.sorted.combined.bam
    for NUM in 250000 500000 1000000 2000000 3000000 4000000
    do
      echo "subsample-bam.py 12345 $NUM multiplexed/bam-combined $FILE" | qsub -l h_vmem=2g -V -cwd -N sub.$WELL.$NUM -j y -o ~/log/subsample-bam.py
    done
  done
done
```

```bash
ls multiplexed/bam-combined
cat ~/log/subsample-bam.py/* | grep success | wc -l
cat ~/log/subsample-bam.py/* | grep failure | wc -l
# Many fail because they do not have sufficient reads for the subsampling
cat ~/log/subsample-bam.py/* | grep failure | grep -v less | wc -l
```

Switch to directory `multiplexed`.
Symlink `exons.saf`.

```bash
cd multiplexed
mkdir genome
ln -s /mnt/lustre/home/jdblischak/singleCellSeq/data/exons.saf genome/exons.saf
```

Remove duplicate UMIs.

```bash
submit-array.sh rmdup-umi.sh 2g bam-combined/*bam
```

```bash
ls bam-rmdup-umi/*bam | wc -l
grep -w success ~/log/rmdup-umi.sh/* | wc -l
grep -w failure ~/log/rmdup-umi.sh/* | wc -l
```

Count reads per gene.

```bash
submit-array.sh count-reads-per-gene.sh 2g bam-combined/*bam bam-rmdup-umi/*bam
```

```bash
ls counts/*genecounts.txt | wc -l
grep -w success ~/log/count-reads-per-gene.sh/* | wc -l
grep -w failure ~/log/count-reads-per-gene.sh/* | wc -l
```

Gather the gene counts.

```bash
mkdir -p ~/log/gather-gene-counts.py
for NUM in 250000 500000 1000000 2000000 3000000 4000000
do
  echo "gather-gene-counts.py counts/*.$NUM.genecounts.txt > read-counts-$NUM.txt" | qsub -l h_vmem=8g -cwd -V -j y -o ~/log/gather-gene-counts.py -N read-counts-$NUM
  echo "gather-gene-counts.py counts/*.$NUM.rmdup.genecounts.txt > molecule-counts-$NUM.txt" | qsub -l h_vmem=8g -cwd -V -j y -o ~/log/gather-gene-counts.py -N molecule-counts-$NUM
done
```

## Session information

```{r info}
sessionInfo()
```
