<!DOCTYPE html>

<html xmlns="http://www.w3.org/1999/xhtml">

<head>

<meta charset="utf-8">
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="generator" content="pandoc" />


<meta name="date" content="2016-06-06" />

<title>Final subsampling plots</title>

<script src="libs/jquery-1.11.0/jquery.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="libs/bootstrap-3.3.1/css/united.min.css" rel="stylesheet" />
<script src="libs/bootstrap-3.3.1/js/bootstrap.min.js"></script>
<script src="libs/bootstrap-3.3.1/shim/html5shiv.min.js"></script>
<script src="libs/bootstrap-3.3.1/shim/respond.min.js"></script>

<style type="text/css">

/* padding for bootstrap navbar */
body {
  padding-top: 50px;
  padding-bottom: 40px;
}


/* offset scroll position for anchor links (for fixed navbar)  */
.section h2 {
  padding-top: 55px;
  margin-top: -55px;
}
.section h3 {
  padding-top: 55px;
  margin-top: -55px;
}



/* don't use link color in navbar */
.dropdown-menu>li>a {
  color: black;
}

/* some padding for disqus */
#disqus_thread {
  margin-top: 45px;
}

</style>

<link rel="stylesheet" href="libs/font-awesome-4.1.0/css/font-awesome.min.css"/>

<style type="text/css">code{white-space: pre;}</style>
<link rel="stylesheet"
      href="libs/highlight/textmate.css"
      type="text/css" />
<script src="libs/highlight/highlight.js"></script>
<style type="text/css">
  pre:not([class]) {
    background-color: white;
  }
</style>
<script type="text/javascript">
if (window.hljs && document.readyState && document.readyState === "complete") {
   window.setTimeout(function() {
      hljs.initHighlighting();
   }, 0);
}
</script>



</head>

<body>

<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
code {
  color: inherit;
  background-color: rgba(0, 0, 0, 0.04);
}
img { 
  max-width:100%; 
  height: auto; 
}
</style>
<div class="container-fluid main-container">


<div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">singleCellSeq</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li><a href="index.html">Home</a></li>
        <li><a href="about.html">About</a></li>
        <li><a href="license.html">License</a></li>
        <li><a href="https://github.com/jdblischak/singleCellSeq">GitHub</a></li>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<div id="header">
<h1 class="title">Final subsampling plots</h1>
<h4 class="date"><em>2016-06-06</em></h4>
</div>

<div id="TOC">
<ul>
<li><a href="#input">Input</a></li>
<li><a href="#figures">Figures</a></li>
<li><a href="#number-of-molecules-per-sequencing-depth">Number of molecules per sequencing depth</a></li>
<li><a href="#session-information">Session information</a></li>
</ul>
</div>

<p><strong>Last updated:</strong> 2016-06-11</p>
<p><strong>Code version:</strong> 0242e47aaea33879b60143338a1f1f3232683dc6</p>
<p>The subsampled files were created using the pipeline described <a href="subsample.html">here</a>, which is identical to the <a href="process-samples.html">pipeline</a> used to process the full data files.</p>
<div id="input" class="section level2">
<h2>Input</h2>
<pre class="r"><code>library(&quot;dplyr&quot;)
library(&quot;tidyr&quot;)
library(&quot;ggplot2&quot;)
library(&quot;cowplot&quot;)
theme_set(theme_bw(base_size = 16))
theme_update(panel.grid.minor.x = element_blank(),
             panel.grid.minor.y = element_blank(),
             panel.grid.major.x = element_blank(),
             panel.grid.major.y = element_blank())</code></pre>
<pre class="r"><code>d &lt;- read.table(&quot;../data/subsampling-results.txt&quot;,
                header = TRUE, sep = &quot;\t&quot;, stringsAsFactors = FALSE)
str(d)</code></pre>
<pre><code>&#39;data.frame&#39;:   6240 obs. of  25 variables:
 $ type            : chr  &quot;reads&quot; &quot;molecules&quot; &quot;reads&quot; &quot;molecules&quot; ...
 $ depth           : int  50000 50000 50000 50000 50000 50000 250000 250000 250000 250000 ...
 $ gene_subset     : chr  &quot;all&quot; &quot;all&quot; &quot;lower&quot; &quot;lower&quot; ...
 $ seed            : int  1 1 1 1 1 1 1 1 1 1 ...
 $ subsampled_cells: int  5 5 5 5 5 5 5 5 5 5 ...
 $ individual      : chr  &quot;NA19098&quot; &quot;NA19098&quot; &quot;NA19098&quot; &quot;NA19098&quot; ...
 $ replicate       : logi  NA NA NA NA NA NA ...
 $ lower_q         : num  0 0 0 0 0.5 0.5 0 0 0 0 ...
 $ upper_q         : num  1 1 0.5 0.5 1 1 1 1 0.5 0.5 ...
 $ available_ensg  : int  12192 12192 12192 12192 12192 12192 12192 12192 12192 12192 ...
 $ used_ensg       : int  12192 12192 6096 6096 6096 6096 12192 12192 6096 6096 ...
 $ available_ercc  : int  43 43 43 43 43 43 43 43 43 43 ...
 $ used_ercc       : int  43 43 22 22 22 22 43 43 22 22 ...
 $ potential_cells : int  142 142 142 142 142 142 142 142 142 142 ...
 $ available_cells : int  142 142 142 142 142 142 142 142 142 142 ...
 $ pearson_ensg    : num  0.774 0.79 0.368 0.372 0.727 ...
 $ pearson_ercc    : num  0.893 0.907 0.456 0.406 0.914 ...
 $ spearman_ensg   : num  0.772 0.79 0.405 0.411 0.703 ...
 $ spearman_ercc   : num  0.854 0.853 0.456 0.383 0.955 ...
 $ detected_ensg   : int  7972 7972 2413 2413 5559 5559 9493 9493 3550 3550 ...
 $ detected_ercc   : int  29 29 10 10 20 20 35 35 14 14 ...
 $ mean_counts_ensg: num  22471 12126 1292 738 21180 ...
 $ mean_counts_ercc: num  568.8 269 8.2 3.2 561.4 ...
 $ var_pearson     : num  0.804 0.846 0.4 0.455 0.76 ...
 $ var_spearman    : num  0.762 0.792 0.451 0.467 0.693 ...</code></pre>
<pre class="r"><code>d_grouped &lt;- d %&gt;%
  group_by(type, depth, gene_subset, subsampled_cells,
           individual, potential_cells, available_cells,
           lower_q, upper_q, available_ensg, used_ensg,
           available_ercc, used_ercc) %&gt;%
  summarize(mean_detected = mean(detected_ensg),
            sem_detected = sd(detected_ensg) / sqrt(length(detected_ensg)),
            mean_bulk = mean(pearson_ensg),
            sem_bulk = sd(pearson_ensg) / sqrt(length(pearson_ensg)),
            mean_var = mean(var_pearson),
            sem_var = sd(var_pearson) / sqrt(length(var_pearson)))</code></pre>
<pre class="r"><code>d_filter &lt;- d_grouped %&gt;% filter(individual == &quot;NA19239&quot;,
                                type == &quot;molecules&quot;,
                                gene_subset %in% c(&quot;lower&quot;, &quot;upper&quot;))
d_filter$gene_subset &lt;- factor(d_filter$gene_subset,
                               levels = c(&quot;lower&quot;, &quot;upper&quot;),
                               labels = c(&quot;Lower 50% of expressed genes&quot;,
                                          &quot;Upper 50% of expressed genes&quot;))</code></pre>
</div>
<div id="figures" class="section level2">
<h2>Figures</h2>
<pre class="r"><code>plot_bulk &lt;- ggplot(d_filter,
                 aes(x = subsampled_cells, y = mean_bulk,
                     color = as.factor(depth))) +
  geom_line() +
  geom_errorbar(aes(ymin = mean_bulk - sem_bulk,
                    ymax = mean_bulk + sem_bulk),
                width = 1) +
  facet_wrap(~gene_subset) +
  scale_color_grey(start = 0.8, end = 0.2, name = &quot;Sequencing depth&quot;) +
  theme(legend.position = &quot;none&quot;) +
  labs(x = &quot;Number of subsampled cells&quot;,
       y = &quot;Pearson&#39;s r (+/- SEM)&quot;,
       title = &quot;Correlation with bulk sequencing&quot;)
plot_bulk</code></pre>
<p><img src="figure/subsample-final.Rmd/plot-bulk-1.png" title="" alt="" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>plot_detected &lt;- ggplot(d_filter,
                 aes(x = subsampled_cells, y = mean_detected,
                     color = as.factor(depth))) +
  geom_line() +
  geom_errorbar(aes(ymin = mean_detected - sem_detected,
                    ymax = mean_detected + sem_detected),
                width = 1) +
  facet_wrap(~gene_subset) +
  scale_color_grey(start = 0.8, end = 0.2, name = &quot;Sequencing depth&quot;) +
  theme(legend.position = c(0.75, 0.35)) +
  labs(x = &quot;Number of subsampled cells&quot;,
       y = &quot;Number of genes detected&quot;,
       title = &quot;Number of genes detected&quot;)
plot_detected</code></pre>
<p><img src="figure/subsample-final.Rmd/plot-detected-1.png" title="" alt="" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>plot_var &lt;- ggplot(d_filter,
                 aes(x = subsampled_cells, y = mean_var,
                     color = as.factor(depth))) +
  geom_line() +
  geom_errorbar(aes(ymin = mean_var - sem_var,
                    ymax = mean_var + sem_var),
                width = 1) +
  facet_wrap(~gene_subset) +
  scale_color_grey(start = 0.8, end = 0.2, name = &quot;Sequencing depth&quot;) +
  theme(legend.position = &quot;none&quot;) +
  labs(x = &quot;Number of subsampled cells&quot;,
       y = &quot;Pearson&#39;s r (+/- SEM)&quot;,
       title = &quot;Correlation with cell-to-cell variance of full data&quot;)
plot_var</code></pre>
<p><img src="figure/subsample-final.Rmd/plot-var-1.png" title="" alt="" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>plot_final &lt;- plot_grid(plot_bulk, plot_detected, plot_var,
                        ncol = 1, labels = LETTERS[1:3])
plot_final</code></pre>
<p><img src="figure/subsample-final.Rmd/plot-final-1.png" title="" alt="" width="1152" style="display: block; margin: auto;" /></p>
<pre class="r"><code>png(&quot;../paper/figure/fig-subsample.png&quot;, width = 8, height = 12, units = &quot;in&quot;, res = 300)
plot_final
dev.off()</code></pre>
<pre><code>png 
  2 </code></pre>
</div>
<div id="number-of-molecules-per-sequencing-depth" class="section level2">
<h2>Number of molecules per sequencing depth</h2>
<pre class="r"><code>molecules_per_depth &lt;- d %&gt;%
  filter(gene_subset == &quot;all&quot;, type == &quot;molecules&quot;)</code></pre>
<pre class="r"><code>plot_mol_depth_ensg &lt;- ggplot(molecules_per_depth,
                              aes(x = as.factor(depth),
                                  y = mean_counts_ensg)) +
  geom_boxplot() +
  facet_wrap(~individual) +
  scale_y_continuous(breaks = seq(0, 1e5, by = 1e4)) +
  theme(axis.text.x = element_text(angle = 90))
plot_mol_depth_ensg</code></pre>
<p><img src="figure/subsample-final.Rmd/molecules-per-depth-ensg-1.png" title="" alt="" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>plot_mol_depth_ercc &lt;- plot_mol_depth_ensg %+% aes(y = mean_counts_ercc)
plot_mol_depth_ercc</code></pre>
<p><img src="figure/subsample-final.Rmd/molecules-per-depth-ercc-1.png" title="" alt="" width="672" style="display: block; margin: auto;" /></p>
</div>
<div id="session-information" class="section level2">
<h2>Session information</h2>
<pre class="r"><code>sessionInfo()</code></pre>
<pre><code>R version 3.2.0 (2015-04-16)
Platform: x86_64-unknown-linux-gnu (64-bit)

locale:
 [1] LC_CTYPE=en_US.UTF-8       LC_NUMERIC=C              
 [3] LC_TIME=en_US.UTF-8        LC_COLLATE=en_US.UTF-8    
 [5] LC_MONETARY=en_US.UTF-8    LC_MESSAGES=en_US.UTF-8   
 [7] LC_PAPER=en_US.UTF-8       LC_NAME=C                 
 [9] LC_ADDRESS=C               LC_TELEPHONE=C            
[11] LC_MEASUREMENT=en_US.UTF-8 LC_IDENTIFICATION=C       

attached base packages:
[1] stats     graphics  grDevices utils     datasets  methods   base     

other attached packages:
[1] cowplot_0.3.1 ggplot2_1.0.1 tidyr_0.2.0   dplyr_0.4.2   knitr_1.10.5 

loaded via a namespace (and not attached):
 [1] Rcpp_0.12.4      magrittr_1.5     MASS_7.3-40      munsell_0.4.3   
 [5] colorspace_1.2-6 R6_2.1.1         stringr_1.0.0    httr_0.6.1      
 [9] plyr_1.8.3       tools_3.2.0      parallel_3.2.0   grid_3.2.0      
[13] gtable_0.1.2     DBI_0.3.1        htmltools_0.2.6  lazyeval_0.1.10 
[17] yaml_2.1.13      assertthat_0.1   digest_0.6.8     reshape2_1.4.1  
[21] formatR_1.2      bitops_1.0-6     RCurl_1.95-4.6   evaluate_0.7    
[25] rmarkdown_0.6.1  labeling_0.3     stringi_1.0-1    scales_0.4.0    
[29] proto_0.3-10    </code></pre>
</div>


<!-- some extra javascript for older browsers -->
<script type="text/javascript" src="libs/polyfill.js"></script>

<script>

// manage active state of menu based on current page
$(document).ready(function () {

    // active menu
    href = window.location.pathname
    href = href.substr(href.lastIndexOf('/') + 1)
    $('a[href="' + href + '"]').parent().addClass('active');

    // manage active menu header
    if (href.startsWith('authoring_'))
      $('a[href="' + 'authoring' + '"]').parent().addClass('active');
    else if (href.endsWith('_format.html'))
      $('a[href="' + 'formats' + '"]').parent().addClass('active');
    else if (href.startsWith('developer_'))
      $('a[href="' + 'developer' + '"]').parent().addClass('active');

});

</script>

</div>

<script>

// add bootstrap table styles to pandoc tables
$(document).ready(function () {
  $('tr.header').parent('thead').parent('table').addClass('table table-condensed');
});

</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://cdn.mathjax.org/mathjax/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
