---
title: "Subsample high coverage single cells"
date: 2015-07-13
---

**Last updated:** `r Sys.Date()`

**Code version:** `r system("git log -1 --format='%H'", intern = TRUE)`

```{r chunk-options, include=FALSE}
source("chunk-options.R")
```

```{r message=FALSE}
library("dplyr")
library("ggplot2")
library("cowplot")
theme_set(theme_bw(base_size = 14))
```

## Batch process each subsampled data set

Run for each single cell for each sequencing depth.
The analysis is performed by [detect-genes.R](https://github.com/jdblischak/singleCellSeq/blob/master/code/detect-genes.R).

```bash
cd $ssd/lcl/full-lane
mkdir -p genes-detected
mkdir -p ~/log/detect-genes.R
CELLS=1
SEED=1
for WELL in A9E1 B2E2 B4H1 D2H2
do
  for NUM in 200000 1000000 10000000 20000000 30000000 40000000 50000000
  do
    for GENE in ENSG ERCC
    do
      # Molecules
      CMD="detect-genes.R $CELLS $SEED molecule-counts-$NUM.txt --min_count=1 --min_cells=1 --wells=$WELL --gene=$GENE"
      DEST="genes-detected/molecule-$WELL-$CELLS-$SEED-$NUM-$GENE.txt"
      echo "$CMD > $DEST" | qsub -l h_vmem=2g -cwd -V -N detect-molecule-$WELL-$CELLS-$SEED-$NUM-$GENE -j y -o ~/log/detect-genes.R -l 'hostname=!bigmem01'
      sleep .01s
      # Reads
      CMD="detect-genes.R $CELLS $SEED read-counts-$NUM.txt --min_count=10 --min_cells=1 --wells=$WELL --gene=$GENE"
      DEST="genes-detected/read-$WELL-$CELLS-$SEED-$NUM-$GENE.txt"
      echo "$CMD > $DEST" | qsub -l h_vmem=2g -cwd -V -N detect-read-$WELL-$CELLS-$SEED-$NUM-$GENE -j y -o ~/log/detect-genes.R -l 'hostname=!bigmem01'
      sleep .01s
    done
  done
done
```

Convert to one file using Python.
Run from `$ssd/lcl/full-lane`.

```python
import os
import glob
files = glob.glob("genes-detected/*txt")
out = open("genes-detected.txt", "w")
out.write("type\twell\tdepth\tfeatures\tnum_cells\tseed\tgenes\tmean_counts\n")
for fname in files:
    fname_parts = os.path.basename(fname).rstrip(".txt").split("-")
    type = fname_parts[0]
    well = fname_parts[1]
    depth = fname_parts[4]
    features = fname_parts[5]
    f = open(fname, "r")
    out.write(type + "\t" + well + "\t" + depth + "\t" + features + "\t" + f.read())
    f.close()

out.close()
```

## Number of genes detected

```{r input}
genes_data <- read.table("/mnt/gluster/data/internal_supp/singleCellSeq/lcl/full-lane/genes-detected.txt",
                         header = TRUE, sep = "\t", stringsAsFactors = FALSE)
genes_data$depth <- as.factor(genes_data$depth)
levels(genes_data$depth) <- c("0.2", "1", "10", "20", "30", "40", "50")
```

For the analysis of read counts, a gene was detected if it had greater than 10 reads.
For the analysis of molecule counts, a gene was detected if it had greater than 1 molecule.

### Endogenous genes

```{r subsample-genes-detected-ensg, fig.width=10, fig.height=10}
genes <- ggplot(genes_data[genes_data$features == "ENSG", ],
            aes(x = as.factor(depth), y = genes)) +
  geom_boxplot() +
  geom_point(aes(color = as.factor(depth), shape = well)) +
  facet_wrap(~type) +
  labs(x = "Sequencing depth (millions)",
       y = "Number of genes detected",
       color = "Depth",
       title = "Subsample: Number of genes detected")
genes
```

### ERCC

```{r subsample-genes-detected-ercc, fig.width=10, fig.height=10}
ercc <- genes %+% genes_data[genes_data$features == "ERCC", ]
ercc
```

## Number of total counts

Explore the effect of subsampling sequencing depth on the total count.
Only includes counts of genes which had the minumum count (10 reads; 1 molecule).

### Endogenous genes - reads

```{r subsample-total-counts-ensg-reads}
p <- ggplot(genes_data[genes_data$features == "ENSG" &
                       genes_data$type == "read", ],
            aes(x = as.factor(depth), y = mean_counts)) +
  geom_boxplot() +
  geom_point(aes(color = as.factor(depth), shape = well)) +
  labs(x = "Sequencing depth (millions)",
       y = "Total count",
       color = "Depth",
       title = "Subsample: Endogenous genes, reads")
reads_gene <- p
reads_gene
```

### Endogenous genes - molecules

```{r subsample-total-counts-ensg-molecules}
molecule_gene <-  p %+% genes_data[genes_data$features == "ENSG" & 
                  genes_data$type == "molecule", ] +
                  labs(y = "Total count",
                  title = "Subsample: Endogenous genes, molecules")
molecule_gene
```

### ERCC - reads

```{r subsample-total-counts-ercc-reads}
read_ercc <- p %+% genes_data[genes_data$features == "ERCC" & 
                 genes_data$type == "read", ] +
  labs(y = "Total count",
       title = "Subsample: ERCC, reads")
read_ercc
```

### ERCC - molecules

```{r subsample-total-counts-ercc-molecules}
molecule_ercc <- p %+% genes_data[genes_data$features == "ERCC" & 
                 genes_data$type == "molecule", ] +
  labs(y = "Total count",
       title = "Subsample: ERCC, molecules")
molecule_ercc
```

## Sample concentrations

The cDNA concentrations are listed as followed:

*  A9E1: 1.73 ng/ul 
*  B2E2: 2.25 ng/ul
*  B4H1: 1.74 ng/ul
*  D2H2: 1.73 ng/ul

```{r paper-figure, fig.width=12, fig.height=6}
theme_set(theme_bw(base_size = 12))
plot_grid(genes + theme(legend.position = c(.4,.35)),
          ercc + labs(title = "Subsample: Number of ERCC detected") + theme(legend.position = "none"),
          labels = LETTERS[1:2])
```

```{r paper-figure2, fig.width=12, fig.height=4}
plot_grid(reads_gene + labs(title = "Genes-reads") + theme(legend.position = "none"),
          molecule_gene + labs(title = "Genes-molecules") + theme(legend.position = "none"),
          read_ercc + labs(title = "ERCC-reads") + theme(legend.position = "none"),
          molecule_ercc + labs(title = "ERCC-molecules") + theme(legend.position = "none"),
          nrow = 1,
          labels = LETTERS[3:6])
```


## Session information

```{r info}
sessionInfo()
```
