---
title: "Subsample - LCLs"
date: 2015-06-30
---

**Last updated:** `r Sys.Date()`

**Code version:** `r system("git log -1 --format='%H'", intern = TRUE)`

```{r chunk-options, include=FALSE}
source("chunk-options.R")
```

Subsample reads post-mapping.
Includes both mapped and unmapped reads.

## Full lane samples

Lanes 5-8 each contained just one single cell.
These are highly oversequenced and will be used to assess saturation.

```bash
cd /mnt/gluster/home/jdblischak/ssd/lcl
mkdir -p subsampled/bam-combined
mkdir -p ~/log/subsample-bam.py
for WELL in A9E1 B2E2 B4H1 D2H2
do
  FILE=bam-combined/19239.1.$WELL.trim.sickle.sorted.combined.bam
  for NUM in 1000000 10000000 20000000 30000000 40000000 50000000 60000000 70000000 80000000
  do
    echo "subsample-bam.py 12345 $NUM subsampled/bam-combined $FILE" | qsub -l h_vmem=8g -V -cwd -N sub.$WELL.$NUM -j y -o ~/log/subsample-bam.py
  done
done
```

```bash
ls subsampled/bam-combined/*bam | wc -l
cat ~/log/subsample-bam.py/* | grep success | wc -l
cat ~/log/subsample-bam.py/* | grep failure | wc -l
```

Switch to directory `subsampled`.
Symlink `exons.saf`.

```bash
cd subsampled
mkdir genome
ln -s /mnt/lustre/home/jdblischak/singleCellSeq/data/exons.saf genome/exons.saf
```

Remove duplicate UMIs.

```bash
submit-array.sh rmdup-umi.sh 12g bam-combined/*bam
```

```bash
ls bam-rmdup-umi/*bam | wc -l
grep -w success ~/log/rmdup-umi.sh/* | wc -l
grep -w failure ~/log/rmdup-umi.sh/* | wc -l
```

Count reads per gene.

```bash
submit-array.sh count-reads-per-gene.sh 12g bam-combined/*bam bam-rmdup-umi/*bam
```

```bash
ls counts/*genecounts.txt | wc -l
grep -w success ~/log/count-reads-per-gene.sh/* | wc -l
grep -w failure ~/log/count-reads-per-gene.sh/* | wc -l
```

Remove the `*.featureCounts` files created by the `-R` flag.
These report the assignment of each read, which is only useful for detailed diagnostics.
Because each file is data from a whole lane, these files are large.

```bash
rm counts/*.featureCounts
```

Gather the gene counts.

```bash
mkdir -p counts-matrix
mkdir -p ~/log/gather-gene-counts-subsample.py
for NUM in 1000000 10000000 20000000 30000000 40000000 50000000 60000000 70000000 80000000
do
  echo "gather-gene-counts-subsample.py counts-matrix/$NUM- counts/*.$NUM.*genecounts.txt" | qsub -l h_vmem=2g -cwd -V -j y -o ~/log/gather-gene-counts-subsample.py -N gene-counts-$NUM -q blades.q
done
```

```bash
# There should be no output
cat ~/log/gather-gene-counts-subsample.py/*
```

## Session information

```{r info}
sessionInfo()
```
