<!DOCTYPE html>

<html xmlns="http://www.w3.org/1999/xhtml">

<head>

<meta charset="utf-8">
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="generator" content="pandoc" />


<meta name="date" content="2016-06-15" />

<title>Final LCL subsampling plots</title>

<script src="libs/jquery-1.11.0/jquery.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="libs/bootstrap-3.3.1/css/united.min.css" rel="stylesheet" />
<script src="libs/bootstrap-3.3.1/js/bootstrap.min.js"></script>
<script src="libs/bootstrap-3.3.1/shim/html5shiv.min.js"></script>
<script src="libs/bootstrap-3.3.1/shim/respond.min.js"></script>

<style type="text/css">

/* padding for bootstrap navbar */
body {
  padding-top: 50px;
  padding-bottom: 40px;
}


/* offset scroll position for anchor links (for fixed navbar)  */
.section h2 {
  padding-top: 55px;
  margin-top: -55px;
}
.section h3 {
  padding-top: 55px;
  margin-top: -55px;
}



/* don't use link color in navbar */
.dropdown-menu>li>a {
  color: black;
}

/* some padding for disqus */
#disqus_thread {
  margin-top: 45px;
}

</style>

<link rel="stylesheet" href="libs/font-awesome-4.1.0/css/font-awesome.min.css"/>

<style type="text/css">code{white-space: pre;}</style>
<link rel="stylesheet"
      href="libs/highlight/textmate.css"
      type="text/css" />
<script src="libs/highlight/highlight.js"></script>
<style type="text/css">
  pre:not([class]) {
    background-color: white;
  }
</style>
<script type="text/javascript">
if (window.hljs && document.readyState && document.readyState === "complete") {
   window.setTimeout(function() {
      hljs.initHighlighting();
   }, 0);
}
</script>



</head>

<body>

<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
code {
  color: inherit;
  background-color: rgba(0, 0, 0, 0.04);
}
img { 
  max-width:100%; 
  height: auto; 
}
</style>
<div class="container-fluid main-container">


<div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">singleCellSeq</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li><a href="index.html">Home</a></li>
        <li><a href="about.html">About</a></li>
        <li><a href="license.html">License</a></li>
        <li><a href="https://github.com/jdblischak/singleCellSeq">GitHub</a></li>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<div id="header">
<h1 class="title">Final LCL subsampling plots</h1>
<h4 class="date"><em>2016-06-15</em></h4>
</div>

<div id="TOC">
<ul>
<li><a href="#input">Input</a></li>
<li><a href="#session-information">Session information</a></li>
</ul>
</div>

<p><strong>Last updated:</strong> 2016-06-29</p>
<p><strong>Code version:</strong> c59e3a495e7257caa3180d1dfdaf315cdd79d715</p>
<p>The LCL subsampled files were created using the pipeline described <a href="subsample-lcl.html">here</a>, which is similar to the <a href="process-samples.html">pipeline</a> used to process the full data files. The only difference is that the samples were processed in chunks of 4 million reads (the default output from CASAVA) and then merged pre-removal of duplicate UMIs.</p>
<div id="input" class="section level2">
<h2>Input</h2>
<pre class="r"><code>library(&quot;dplyr&quot;)
library(&quot;tidyr&quot;)
library(&quot;ggplot2&quot;)
library(&quot;cowplot&quot;)
theme_set(theme_bw(base_size = 16))
theme_update(panel.grid.minor.x = element_blank(),
             panel.grid.minor.y = element_blank(),
             panel.grid.major.x = element_blank(),
             panel.grid.major.y = element_blank())</code></pre>
<p>The subsampling statistics were generated with <a href="https://github.com/jdblischak/singleCellSeq/blob/master/code/subsample-pipeline.py">subsample-pipeline.py</a> and <a href="https://github.com/jdblischak/singleCellSeq/blob/master/code/detect-genes.R">detect-genes.R</a>.</p>
<pre class="r"><code>d &lt;- read.table(&quot;../data/subsampling-results-lcl.txt&quot;,
                header = TRUE, sep = &quot;\t&quot;, stringsAsFactors = FALSE)
d$depth_mil &lt;- d$depth / 10^6
d$counts_thous &lt;- d$counts / 10^3
d$counts_mil &lt;- d$counts / 10^6
str(d)</code></pre>
<pre><code>&#39;data.frame&#39;:   144 obs. of  11 variables:
 $ depth       : int  1000000 1000000 1000000 1000000 1000000 1000000 1000000 1000000 10000000 10000000 ...
 $ type        : chr  &quot;molecules&quot; &quot;reads&quot; &quot;molecules&quot; &quot;reads&quot; ...
 $ well        : chr  &quot;A9E1&quot; &quot;A9E1&quot; &quot;B2E2&quot; &quot;B2E2&quot; ...
 $ gene_subset : chr  &quot;ENSG&quot; &quot;ENSG&quot; &quot;ENSG&quot; &quot;ENSG&quot; ...
 $ num_cells   : int  1 1 1 1 1 1 1 1 1 1 ...
 $ seed        : int  1 1 1 1 1 1 1 1 1 1 ...
 $ genes       : int  5290 5290 6253 6254 5319 5319 4925 4925 6605 6606 ...
 $ counts      : num  21630 493812 30932 518453 19398 ...
 $ depth_mil   : num  1 1 1 1 1 1 1 1 10 10 ...
 $ counts_thous: num  21.6 493.8 30.9 518.5 19.4 ...
 $ counts_mil  : num  0.0216 0.4938 0.0309 0.5185 0.0194 ...</code></pre>
<p>Endogenous genes detected</p>
<pre class="r"><code>p_genes_ensg &lt;- ggplot(d[d$gene_subset == &quot;ENSG&quot; &amp;
                         d$type == &quot;molecules&quot;, ],
                       aes(x = as.factor(depth_mil), y = genes)) +
  geom_boxplot() + 
  geom_point(aes(color  = well)) +
  scale_color_brewer(palette = &quot;Dark2&quot;, name = &quot;Single cell&quot;) +
  theme(legend.position = &quot;none&quot;) +
  labs(x = expression(&quot;Sequencing depth (&quot; * 10^6 * &quot;)&quot;),
       y = &quot;Genes detected&quot;,
       title = &quot;Endogenous genes&quot;)
p_genes_ensg</code></pre>
<p><img src="figure/subsample-lcl-final.Rmd/plot-genes-ensg-1.png" title="" alt="" width="672" style="display: block; margin: auto;" /></p>
<p>ERCC control genes detected</p>
<pre class="r"><code>p_genes_ercc &lt;- p_genes_ensg %+% d[d$gene_subset == &quot;ERCC&quot; &amp;
                                   d$type == &quot;molecules&quot;, ] +
  labs(title = &quot;ERCC controls&quot;)
p_genes_ercc</code></pre>
<p><img src="figure/subsample-lcl-final.Rmd/plot-genes-ercc-1.png" title="" alt="" width="672" style="display: block; margin: auto;" /></p>
<p>Endogenous molecules</p>
<pre class="r"><code>p_molecules_ensg &lt;- p_genes_ensg %+% d[d$type == &quot;molecules&quot; &amp;
                                       d$gene_subset == &quot;ENSG&quot;, ] %+%
  aes(y = counts_thous) +
  labs(y = expression(&quot;Molecules (&quot; * 10^3 * &quot;)&quot;))
p_molecules_ensg</code></pre>
<p><img src="figure/subsample-lcl-final.Rmd/plot-molecules-ensg-1.png" title="" alt="" width="672" style="display: block; margin: auto;" /></p>
<p>ERCC molecules</p>
<pre class="r"><code>p_molecules_ercc &lt;- p_molecules_ensg %+% d[d$type == &quot;molecules&quot; &amp;
                                           d$gene_subset == &quot;ERCC&quot;, ] +
  labs(title = &quot;ERCC controls&quot;) +
  theme(legend.position = c(0.75, 0.35))
p_molecules_ercc</code></pre>
<p><img src="figure/subsample-lcl-final.Rmd/plot-molecules-ercc-1.png" title="" alt="" width="672" style="display: block; margin: auto;" /></p>
<p>Endogenous reads</p>
<pre class="r"><code>p_reads_ensg &lt;- p_molecules_ensg %+% d[d$type == &quot;reads&quot; &amp;
                                       d$gene_subset == &quot;ENSG&quot;, ] +
  aes(y = counts_mil) +
  labs(y = expression(&quot;Reads (&quot; * 10^6 * &quot;)&quot;))
p_reads_ensg</code></pre>
<p><img src="figure/subsample-lcl-final.Rmd/plot-reads-ensg-1.png" title="" alt="" width="672" style="display: block; margin: auto;" /></p>
<p>ERCC reads</p>
<pre class="r"><code>p_reads_ercc &lt;- p_reads_ensg %+% d[d$type == &quot;reads&quot; &amp;
                                   d$gene_subset == &quot;ERCC&quot;, ] +
  labs(title = &quot;ERCC controls&quot;)
p_reads_ercc</code></pre>
<p><img src="figure/subsample-lcl-final.Rmd/plot-reads-ercc-1.png" title="" alt="" width="672" style="display: block; margin: auto;" /></p>
<p>Final plot for supplementary figure</p>
<pre class="r"><code>plot_final &lt;- plot_grid(p_genes_ensg, p_genes_ercc,
                        p_molecules_ensg, p_molecules_ercc,
                        p_reads_ensg, p_reads_ercc,
                        ncol = 2, labels = LETTERS[1:6])
png(&quot;../paper/figure/fig-subsample-lcl.png&quot;, width = 8, height = 12,
    units = &quot;in&quot;, res = 300)
plot_final
dev.off()</code></pre>
<pre><code>png 
  2 </code></pre>
</div>
<div id="session-information" class="section level2">
<h2>Session information</h2>
<pre class="r"><code>sessionInfo()</code></pre>
<pre><code>R version 3.2.0 (2015-04-16)
Platform: x86_64-unknown-linux-gnu (64-bit)

locale:
 [1] LC_CTYPE=en_US.UTF-8       LC_NUMERIC=C              
 [3] LC_TIME=en_US.UTF-8        LC_COLLATE=en_US.UTF-8    
 [5] LC_MONETARY=en_US.UTF-8    LC_MESSAGES=en_US.UTF-8   
 [7] LC_PAPER=en_US.UTF-8       LC_NAME=C                 
 [9] LC_ADDRESS=C               LC_TELEPHONE=C            
[11] LC_MEASUREMENT=en_US.UTF-8 LC_IDENTIFICATION=C       

attached base packages:
[1] stats     graphics  grDevices utils     datasets  methods   base     

other attached packages:
[1] cowplot_0.3.1 ggplot2_1.0.1 tidyr_0.2.0   dplyr_0.4.2   knitr_1.10.5 

loaded via a namespace (and not attached):
 [1] Rcpp_0.12.4        magrittr_1.5       MASS_7.3-40       
 [4] munsell_0.4.3      colorspace_1.2-6   R6_2.1.1          
 [7] stringr_1.0.0      httr_0.6.1         plyr_1.8.3        
[10] tools_3.2.0        parallel_3.2.0     grid_3.2.0        
[13] gtable_0.1.2       DBI_0.3.1          htmltools_0.2.6   
[16] yaml_2.1.13        assertthat_0.1     digest_0.6.8      
[19] RColorBrewer_1.1-2 reshape2_1.4.1     formatR_1.2       
[22] bitops_1.0-6       RCurl_1.95-4.6     evaluate_0.7      
[25] rmarkdown_0.6.1    labeling_0.3       stringi_1.0-1     
[28] scales_0.4.0       proto_0.3-10      </code></pre>
</div>


<!-- some extra javascript for older browsers -->
<script type="text/javascript" src="libs/polyfill.js"></script>

<script>

// manage active state of menu based on current page
$(document).ready(function () {

    // active menu
    href = window.location.pathname
    href = href.substr(href.lastIndexOf('/') + 1)
    $('a[href="' + href + '"]').parent().addClass('active');

    // manage active menu header
    if (href.startsWith('authoring_'))
      $('a[href="' + 'authoring' + '"]').parent().addClass('active');
    else if (href.endsWith('_format.html'))
      $('a[href="' + 'formats' + '"]').parent().addClass('active');
    else if (href.startsWith('developer_'))
      $('a[href="' + 'developer' + '"]').parent().addClass('active');

});

</script>

</div>

<script>

// add bootstrap table styles to pandoc tables
$(document).ready(function () {
  $('tr.header').parent('thead').parent('table').addClass('table table-condensed');
});

</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://cdn.mathjax.org/mathjax/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
