<!DOCTYPE html>

<html xmlns="http://www.w3.org/1999/xhtml">

<head>

<meta charset="utf-8">
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="generator" content="pandoc" />

<meta name="author" content="John Blischak" />

<meta name="date" content="2015-05-12" />

<title>Sum counts per sample</title>

<script src="libs/jquery-1.11.0/jquery.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="libs/bootstrap-3.3.1/css/united.min.css" rel="stylesheet" />
<script src="libs/bootstrap-3.3.1/js/bootstrap.min.js"></script>
<script src="libs/bootstrap-3.3.1/shim/html5shiv.min.js"></script>
<script src="libs/bootstrap-3.3.1/shim/respond.min.js"></script>

<style type="text/css">

/* padding for bootstrap navbar */
body {
  padding-top: 50px;
  padding-bottom: 40px;
}


/* offset scroll position for anchor links (for fixed navbar)  */
.section h2 {
  padding-top: 55px;
  margin-top: -55px;
}
.section h3 {
  padding-top: 55px;
  margin-top: -55px;
}



/* don't use link color in navbar */
.dropdown-menu>li>a {
  color: black;
}

/* some padding for disqus */
#disqus_thread {
  margin-top: 45px;
}

</style>

<link rel="stylesheet" href="libs/font-awesome-4.1.0/css/font-awesome.min.css"/>

<style type="text/css">code{white-space: pre;}</style>
<link rel="stylesheet"
      href="libs/highlight/textmate.css"
      type="text/css" />
<script src="libs/highlight/highlight.js"></script>
<style type="text/css">
  pre:not([class]) {
    background-color: white;
  }
</style>
<script type="text/javascript">
if (window.hljs && document.readyState && document.readyState === "complete") {
   window.setTimeout(function() {
      hljs.initHighlighting();
   }, 0);
}
</script>



</head>

<body>

<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
code {
  color: inherit;
  background-color: rgba(0, 0, 0, 0.04);
}
img { 
  max-width:100%; 
  height: auto; 
}
</style>
<div class="container-fluid main-container">


<div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">singleCellSeq</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li><a href="index.html">Home</a></li>
        <li><a href="about.html">About</a></li>
        <li><a href="license.html">License</a></li>
        <li><a href="https://github.com/jdblischak/singleCellSeq">GitHub</a></li>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<div id="header">
<h1 class="title">Sum counts per sample</h1>
<h4 class="author"><em>John Blischak</em></h4>
<h4 class="date"><em>2015-05-12</em></h4>
</div>

<div id="TOC">
<ul>
<li><a href="#input">Input</a></li>
<li><a href="#sum-counts-by-sample">Sum counts by sample</a></li>
<li><a href="#transpose">Transpose</a></li>
<li><a href="#output">Output</a></li>
<li><a href="#session-information">Session information</a></li>
</ul>
</div>

<p><strong>Last updated:</strong> 2015-05-29</p>
<p><strong>Code version:</strong> bd50ee78f288e4be408987c8211ab0183400a853</p>
<p>For each sample, sum the number of reads or molecules for every gene across sequencing runs.</p>
<p>Creates the following files:</p>
<ul>
<li>anno.txt - Annotation file</li>
<li>molecules.txt - Molecule counts (sample x gene)</li>
<li>reads.txt - Read counts (sample x gene)</li>
</ul>
<div id="input" class="section level2">
<h2>Input</h2>
<pre class="r"><code>library(&quot;data.table&quot;)
library(&quot;dplyr&quot;)</code></pre>
<pre class="r"><code>counts &lt;- fread(&quot;/mnt/gluster/data/internal_supp/singleCellSeq/gene-counts.txt&quot;)</code></pre>
<pre><code>
Read 0.0% of 10656 rows
Read 93.8% of 10656 rows
Read 10656 rows and 20427 (of 20427) columns from 0.448 GB file in 00:00:13</code></pre>
</div>
<div id="sum-counts-by-sample" class="section level2">
<h2>Sum counts by sample</h2>
<pre class="r"><code>counts_by_sample &lt;- counts %&gt;%
  select(individual, batch, well, rmdup, starts_with(&quot;ENSG&quot;), starts_with(&quot;ERCC&quot;)) %&gt;%
  group_by(individual, batch, well, rmdup) %&gt;%
  summarise_each(funs(sum)) %&gt;%
  arrange(individual, batch, well, rmdup) %&gt;%
  ungroup</code></pre>
<pre class="r"><code>counts_by_sample %&gt;% select(1:8) %&gt;% slice(1:10)</code></pre>
<pre><code>Source: local data table [10 x 8]

   individual batch well     rmdup ENSG00000186092 ENSG00000237683
1       19098     1  A01 molecules               0               4
2       19098     1  A01     reads               0               4
3       19098     1  A02 molecules               0               0
4       19098     1  A02     reads               0               0
5       19098     1  A03 molecules               0               8
6       19098     1  A03     reads               0              12
7       19098     1  A04 molecules               0              12
8       19098     1  A04     reads               0              94
9       19098     1  A05 molecules               0               0
10      19098     1  A05     reads               0               0
Variables not shown: ENSG00000235249 (int), ENSG00000185097 (int)</code></pre>
</div>
<div id="transpose" class="section level2">
<h2>Transpose</h2>
<pre class="r"><code>anno &lt;- counts_by_sample %&gt;%
  filter(rmdup == &quot;molecules&quot;) %&gt;%
  select(individual:well) %&gt;%
  as.data.frame
anno$sample_id &lt;- paste(paste0(&quot;NA&quot;, anno$individual), anno$batch, anno$well,
                        sep = &quot;.&quot;)
head(anno)</code></pre>
<pre><code>  individual batch well     sample_id
1      19098     1  A01 NA19098.1.A01
2      19098     1  A02 NA19098.1.A02
3      19098     1  A03 NA19098.1.A03
4      19098     1  A04 NA19098.1.A04
5      19098     1  A05 NA19098.1.A05
6      19098     1  A06 NA19098.1.A06</code></pre>
<pre class="r"><code>molecules &lt;- counts_by_sample %&gt;%
  filter(rmdup == &quot;molecules&quot;) %&gt;%
  select(-(individual:rmdup)) %&gt;%
  t
dim(molecules)</code></pre>
<pre><code>[1] 20419   873</code></pre>
<pre class="r"><code>colnames(molecules) &lt;- anno$sample_id
molecules[1:10, 1:5]</code></pre>
<pre><code>                NA19098.1.A01 NA19098.1.A02 NA19098.1.A03 NA19098.1.A04
ENSG00000186092             0             0             0             0
ENSG00000237683             4             0             8            12
ENSG00000235249             0             0             0             0
ENSG00000185097             0             0             0             0
ENSG00000269831             0             0             0             0
ENSG00000269308             0             0             0             0
ENSG00000187634             0             0             0             0
ENSG00000268179             0             0             0             0
ENSG00000188976            10            12             0             5
ENSG00000187961             0             0             0             0
                NA19098.1.A05
ENSG00000186092             0
ENSG00000237683             0
ENSG00000235249             0
ENSG00000185097             0
ENSG00000269831             0
ENSG00000269308             0
ENSG00000187634             0
ENSG00000268179             0
ENSG00000188976            29
ENSG00000187961             0</code></pre>
<pre class="r"><code>reads &lt;- counts_by_sample %&gt;%
  filter(rmdup == &quot;reads&quot;) %&gt;%
  select(-(individual:rmdup)) %&gt;%
  t
dim(reads)</code></pre>
<pre><code>[1] 20419   873</code></pre>
<pre class="r"><code>colnames(reads) &lt;- anno$sample_id
reads[1:10, 1:5]</code></pre>
<pre><code>                NA19098.1.A01 NA19098.1.A02 NA19098.1.A03 NA19098.1.A04
ENSG00000186092             0             0             0             0
ENSG00000237683             4             0            12            94
ENSG00000235249             0             0             0             0
ENSG00000185097             0             0             0             0
ENSG00000269831             0             0             0             0
ENSG00000269308             0             0             0             0
ENSG00000187634             0             0             0             0
ENSG00000268179             0             0             0             0
ENSG00000188976           116            37             0             5
ENSG00000187961             0             0             0             0
                NA19098.1.A05
ENSG00000186092             0
ENSG00000237683             0
ENSG00000235249             0
ENSG00000185097             0
ENSG00000269831             0
ENSG00000269308             0
ENSG00000187634             0
ENSG00000268179             0
ENSG00000188976           336
ENSG00000187961             0</code></pre>
</div>
<div id="output" class="section level2">
<h2>Output</h2>
<p>Output annotation file.</p>
<pre class="r"><code>write.table(anno, &quot;../data/annotation.txt&quot;, quote = FALSE, sep = &quot;\t&quot;,
            row.names = FALSE)</code></pre>
<p>Output molecule counts.</p>
<pre class="r"><code>write.table(molecules, &quot;../data/molecules.txt&quot;, quote = FALSE, sep = &quot;\t&quot;,
            col.names = NA)</code></pre>
<p>Output read counts.</p>
<pre class="r"><code>write.table(reads, &quot;../data/reads.txt&quot;, quote = FALSE, sep = &quot;\t&quot;,
            col.names = NA)</code></pre>
</div>
<div id="session-information" class="section level2">
<h2>Session information</h2>
<pre class="r"><code>sessionInfo()</code></pre>
<pre><code>R version 3.1.1 (2014-07-10)
Platform: x86_64-unknown-linux-gnu (64-bit)

locale:
 [1] LC_CTYPE=en_US.UTF-8       LC_NUMERIC=C              
 [3] LC_TIME=en_US.UTF-8        LC_COLLATE=en_US.UTF-8    
 [5] LC_MONETARY=en_US.UTF-8    LC_MESSAGES=en_US.UTF-8   
 [7] LC_PAPER=en_US.UTF-8       LC_NAME=C                 
 [9] LC_ADDRESS=C               LC_TELEPHONE=C            
[11] LC_MEASUREMENT=en_US.UTF-8 LC_IDENTIFICATION=C       

attached base packages:
[1] stats     graphics  grDevices utils     datasets  methods   base     

other attached packages:
[1] dplyr_0.4.1      data.table_1.9.4 knitr_1.10.5    

loaded via a namespace (and not attached):
 [1] assertthat_0.1  chron_2.3-45    DBI_0.3.1       digest_0.6.4   
 [5] evaluate_0.7    formatR_0.10    htmltools_0.2.6 lazyeval_0.1.10
 [9] magrittr_1.0.1  parallel_3.1.1  plyr_1.8.1      Rcpp_0.11.6    
[13] reshape2_1.4    rmarkdown_0.5.1 stringr_0.6.2   tools_3.1.1    
[17] yaml_2.1.13    </code></pre>
</div>


<!-- some extra javascript for older browsers -->
<script type="text/javascript" src="libs/polyfill.js"></script>

<script>

// manage active state of menu based on current page
$(document).ready(function () {

    // active menu
    href = window.location.pathname
    href = href.substr(href.lastIndexOf('/') + 1)
    $('a[href="' + href + '"]').parent().addClass('active');

    // manage active menu header
    if (href.startsWith('authoring_'))
      $('a[href="' + 'authoring' + '"]').parent().addClass('active');
    else if (href.endsWith('_format.html'))
      $('a[href="' + 'formats' + '"]').parent().addClass('active');
    else if (href.startsWith('developer_'))
      $('a[href="' + 'developer' + '"]').parent().addClass('active');

});

</script>

</div>

<script>

// add bootstrap table styles to pandoc tables
$(document).ready(function () {
  $('tr.header').parent('thead').parent('table').addClass('table table-condensed');
});

</script>


</body>
</html>
