<!DOCTYPE html>

<html xmlns="http://www.w3.org/1999/xhtml">

<head>

<meta charset="utf-8">
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="generator" content="pandoc" />

<meta name="author" content="John Blischak" />

<meta name="date" content="2015-04-22" />

<title>Total counts</title>

<script src="libs/jquery-1.11.0/jquery.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="libs/bootstrap-3.3.1/css/united.min.css" rel="stylesheet" />
<script src="libs/bootstrap-3.3.1/js/bootstrap.min.js"></script>
<script src="libs/bootstrap-3.3.1/shim/html5shiv.min.js"></script>
<script src="libs/bootstrap-3.3.1/shim/respond.min.js"></script>

<style type="text/css">

/* padding for bootstrap navbar */
body {
  padding-top: 50px;
  padding-bottom: 40px;
}


/* offset scroll position for anchor links (for fixed navbar)  */
.section h2 {
  padding-top: 55px;
  margin-top: -55px;
}
.section h3 {
  padding-top: 55px;
  margin-top: -55px;
}



/* don't use link color in navbar */
.dropdown-menu>li>a {
  color: black;
}

/* some padding for disqus */
#disqus_thread {
  margin-top: 45px;
}

</style>

<link rel="stylesheet" href="libs/font-awesome-4.1.0/css/font-awesome.min.css"/>

<style type="text/css">code{white-space: pre;}</style>
<link rel="stylesheet"
      href="libs/highlight/textmate.css"
      type="text/css" />
<script src="libs/highlight/highlight.js"></script>
<style type="text/css">
  pre:not([class]) {
    background-color: white;
  }
</style>
<script type="text/javascript">
if (window.hljs && document.readyState && document.readyState === "complete") {
   window.setTimeout(function() {
      hljs.initHighlighting();
   }, 0);
}
</script>



</head>

<body>

<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
code {
  color: inherit;
  background-color: rgba(0, 0, 0, 0.04);
}
img { 
  max-width:100%; 
  height: auto; 
}
</style>
<div class="container-fluid main-container">


<div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">singleCellSeq</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li><a href="index.html">Home</a></li>
        <li><a href="about.html">About</a></li>
        <li><a href="license.html">License</a></li>
        <li><a href="https://github.com/jdblischak/singleCellSeq">GitHub</a></li>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<div id="header">
<h1 class="title">Total counts</h1>
<h4 class="author"><em>John Blischak</em></h4>
<h4 class="date"><em>2015-04-22</em></h4>
</div>

<div id="TOC">
<ul>
<li><a href="#analyzing-total-counts">Analyzing total counts</a></li>
<li><a href="#variation-in-counts">Variation in counts</a></li>
<li><a href="#session-information">Session information</a></li>
</ul>
</div>

<p><strong>Last updated:</strong> 2015-05-29</p>
<p><strong>Code version:</strong> bd50ee78f288e4be408987c8211ab0183400a853</p>
<p>During the <a href="process-samples.html">processing pipeline</a>, the number of reads in a file are counted and saved in a separate text files. The script <a href="https://github.com/jdblischak/singleCellSeq/blob/master/code/gather-total-counts.py">gather-total-counts.py</a> compiles all these counts and extracts the relevant variables from the filename.</p>
<div id="analyzing-total-counts" class="section level2">
<h2>Analyzing total counts</h2>
<pre class="r"><code>library(&quot;ggplot2&quot;)
theme_set(theme_bw(base_size = 16))</code></pre>
<pre class="r"><code>counts &lt;- read.table(&quot;../data/total-counts.txt&quot;, header = TRUE,
                     stringsAsFactors = FALSE)</code></pre>
<pre class="r"><code>head(counts)</code></pre>
<pre><code>  stage              sickle individual batch well    index lane flow_cell
1 fastq not-quality-trimmed      19098     1  H03 GGGAATAT L002 C6WYKACXX
2 fastq not-quality-trimmed      19098     2  B05 CTTGAAGG L005 C72JMACXX
3 fastq not-quality-trimmed      19098     2  G10 AGCAAGCG L003 C723YACXX
4 fastq not-quality-trimmed      19098     1  H11 TTCTCTGC L002 C6WYKACXX
5 fastq not-quality-trimmed      19098     2  E07 TGCCCAGA L003 C723YACXX
6 fastq not-quality-trimmed      19098     2  F12 AACCCATG L001 C6WURACXX
   counts
1 2114841
2 2394493
3  753351
4 3126886
5 1589238
6 2504377</code></pre>
<pre class="r"><code>str(counts)</code></pre>
<pre><code>&#39;data.frame&#39;:   29304 obs. of  9 variables:
 $ stage     : chr  &quot;fastq&quot; &quot;fastq&quot; &quot;fastq&quot; &quot;fastq&quot; ...
 $ sickle    : chr  &quot;not-quality-trimmed&quot; &quot;not-quality-trimmed&quot; &quot;not-quality-trimmed&quot; &quot;not-quality-trimmed&quot; ...
 $ individual: int  19098 19098 19098 19098 19098 19098 19098 19098 19098 19098 ...
 $ batch     : int  1 2 2 1 2 2 2 2 2 2 ...
 $ well      : chr  &quot;H03&quot; &quot;B05&quot; &quot;G10&quot; &quot;H11&quot; ...
 $ index     : chr  &quot;GGGAATAT&quot; &quot;CTTGAAGG&quot; &quot;AGCAAGCG&quot; &quot;TTCTCTGC&quot; ...
 $ lane      : chr  &quot;L002&quot; &quot;L005&quot; &quot;L003&quot; &quot;L002&quot; ...
 $ flow_cell : chr  &quot;C6WYKACXX&quot; &quot;C72JMACXX&quot; &quot;C723YACXX&quot; &quot;C6WYKACXX&quot; ...
 $ counts    : int  2114841 2394493 753351 3126886 1589238 2504377 2118488 2214717 1673986 1864989 ...</code></pre>
<pre class="r"><code># Order the processing steps
counts$stage &lt;- factor(counts$stage,
                       levels = c(&quot;fastq&quot;, &quot;trim&quot;, &quot;bam-processed&quot;,
                                  &quot;bam-rmdup-umi&quot;, &quot;counts&quot;, &quot;counts-rmdup&quot;))
# Make new variable to separate bulk and single cell samples
counts$type &lt;- ifelse(counts$well == &quot;bulk&quot;, &quot;bulk&quot;, &quot;single&quot;)
# Scale to millions of counts
counts$counts_mil &lt;- counts$counts / 10^6
# Convert numeric categorical variables to factors
counts$individual &lt;- factor(counts$individual)
counts$batch &lt;- factor(counts$batch)</code></pre>
<p>Plot the number of reads at each processing stage faceted by bulk versus single cell sequencing.</p>
<pre class="r"><code>ggplot(counts, aes(x = stage, y = counts_mil)) +
  geom_boxplot() +
  # facet_wrap(type ~ sickle, nrow = 1) +
  facet_grid(sickle ~ type) +
  scale_y_continuous(breaks = seq(0, 16, 2)) +
  labs(x = &quot;Processing stage&quot;, y = &quot;Number of reads (millions)&quot;) +
  theme(axis.text.x = element_text(angle = 90, hjust = 0.9, vjust = 0.5))</code></pre>
<p><img src="figure/total-counts.Rmd/counts-by-processing-stage-1.png" title="" alt="" width="768" style="display: block; margin: auto;" /></p>
<pre class="r"><code>counts_sum &lt;- tapply(counts$counts, list(counts$stage, counts$type, counts$sickle), mean)
counts_sum</code></pre>
<pre><code>, , not-quality-trimmed

                 bulk     single
fastq         9705088 2092200.84
trim          6536956 1478098.76
bam-processed 5265894 1051763.77
bam-rmdup-umi 3370583   92219.13
counts        3840849  699017.58
counts-rmdup  2571847   56229.42

, , quality-trimmed

                 bulk     single
fastq              NA         NA
trim          6436990 1460819.87
bam-processed 5186264 1040234.15
bam-rmdup-umi 3338797   92514.04
counts        3786297  692108.05
counts-rmdup  2546592   56208.94</code></pre>
<p>The bulk samples have an average of 2.546591810^{6} molecules that map to protein-coding exons (2.571846910^{6} before quality trimming). The single cell samples have an average of 5.620894110^{4} molecules that map to protein-coding exons (5.622941610^{4} before quality trimming). Since each sample is sequenced three times, the average total number of molecules will be 7.639775410^{6} and 1.686268210^{5} molecules for the bulk and single cell samples, respectively.</p>
</div>
<div id="variation-in-counts" class="section level2">
<h2>Variation in counts</h2>
<pre class="r"><code>variables &lt;- c(&quot;individual&quot;, &quot;batch&quot;, &quot;index&quot;, &quot;lane&quot;, &quot;flow_cell&quot;)
for (v in variables) {
  p &lt;- ggplot(counts, aes_string(x = v, y = &quot;counts_mil&quot;)) +
  geom_boxplot() +
  facet_grid(sickle ~ type) +
  scale_y_continuous(breaks = seq(0, 16, 2)) +
  labs(x = v, y = &quot;Number of reads (millions)&quot;) +
  theme(axis.text.x = element_text(angle = 90, hjust = 0.9, vjust = 0.5))
  print(p)
}</code></pre>
<p><img src="figure/total-counts.Rmd/counts-by-variables-1.png" title="" alt="" width="768" style="display: block; margin: auto;" /><img src="figure/total-counts.Rmd/counts-by-variables-2.png" title="" alt="" width="768" style="display: block; margin: auto;" /><img src="figure/total-counts.Rmd/counts-by-variables-3.png" title="" alt="" width="768" style="display: block; margin: auto;" /><img src="figure/total-counts.Rmd/counts-by-variables-4.png" title="" alt="" width="768" style="display: block; margin: auto;" /><img src="figure/total-counts.Rmd/counts-by-variables-5.png" title="" alt="" width="768" style="display: block; margin: auto;" /></p>
</div>
<div id="session-information" class="section level2">
<h2>Session information</h2>
<pre class="r"><code>sessionInfo()</code></pre>
<pre><code>R version 3.1.1 (2014-07-10)
Platform: x86_64-unknown-linux-gnu (64-bit)

locale:
 [1] LC_CTYPE=en_US.UTF-8       LC_NUMERIC=C              
 [3] LC_TIME=en_US.UTF-8        LC_COLLATE=en_US.UTF-8    
 [5] LC_MONETARY=en_US.UTF-8    LC_MESSAGES=en_US.UTF-8   
 [7] LC_PAPER=en_US.UTF-8       LC_NAME=C                 
 [9] LC_ADDRESS=C               LC_TELEPHONE=C            
[11] LC_MEASUREMENT=en_US.UTF-8 LC_IDENTIFICATION=C       

attached base packages:
[1] stats     graphics  grDevices utils     datasets  methods   base     

other attached packages:
[1] ggplot2_1.0.0 knitr_1.10.5 

loaded via a namespace (and not attached):
 [1] colorspace_1.2-4 digest_0.6.4     evaluate_0.7     formatR_0.10    
 [5] grid_3.1.1       gtable_0.1.2     htmltools_0.2.6  MASS_7.3-33     
 [9] munsell_0.4.2    plyr_1.8.1       proto_0.3-10     Rcpp_0.11.6     
[13] reshape2_1.4     rmarkdown_0.5.1  scales_0.2.4     stringr_0.6.2   
[17] tools_3.1.1      yaml_2.1.13     </code></pre>
</div>


<!-- some extra javascript for older browsers -->
<script type="text/javascript" src="libs/polyfill.js"></script>

<script>

// manage active state of menu based on current page
$(document).ready(function () {

    // active menu
    href = window.location.pathname
    href = href.substr(href.lastIndexOf('/') + 1)
    $('a[href="' + href + '"]').parent().addClass('active');

    // manage active menu header
    if (href.startsWith('authoring_'))
      $('a[href="' + 'authoring' + '"]').parent().addClass('active');
    else if (href.endsWith('_format.html'))
      $('a[href="' + 'formats' + '"]').parent().addClass('active');
    else if (href.startsWith('developer_'))
      $('a[href="' + 'developer' + '"]').parent().addClass('active');

});

</script>

</div>

<script>

// add bootstrap table styles to pandoc tables
$(document).ready(function () {
  $('tr.header').parent('thead').parent('table').addClass('table table-condensed');
});

</script>


</body>
</html>
