<!DOCTYPE html>

<html xmlns="http://www.w3.org/1999/xhtml">

<head>

<meta charset="utf-8">
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="generator" content="pandoc" />

<meta name="author" content="Po-Yuan Tung" />

<meta name="date" content="2016-04-18" />

<title>Total molecule counts</title>

<script src="libs/jquery-1.11.0/jquery.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="libs/bootstrap-3.3.1/css/united.min.css" rel="stylesheet" />
<script src="libs/bootstrap-3.3.1/js/bootstrap.min.js"></script>
<script src="libs/bootstrap-3.3.1/shim/html5shiv.min.js"></script>
<script src="libs/bootstrap-3.3.1/shim/respond.min.js"></script>

<style type="text/css">

/* padding for bootstrap navbar */
body {
  padding-top: 50px;
  padding-bottom: 40px;
}


/* offset scroll position for anchor links (for fixed navbar)  */
.section h2 {
  padding-top: 55px;
  margin-top: -55px;
}
.section h3 {
  padding-top: 55px;
  margin-top: -55px;
}



/* don't use link color in navbar */
.dropdown-menu>li>a {
  color: black;
}

/* some padding for disqus */
#disqus_thread {
  margin-top: 45px;
}

</style>

<link rel="stylesheet" href="libs/font-awesome-4.1.0/css/font-awesome.min.css"/>

<style type="text/css">code{white-space: pre;}</style>
<link rel="stylesheet"
      href="libs/highlight/textmate.css"
      type="text/css" />
<script src="libs/highlight/highlight.js"></script>
<style type="text/css">
  pre:not([class]) {
    background-color: white;
  }
</style>
<script type="text/javascript">
if (window.hljs && document.readyState && document.readyState === "complete") {
   window.setTimeout(function() {
      hljs.initHighlighting();
   }, 0);
}
</script>



</head>

<body>

<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
code {
  color: inherit;
  background-color: rgba(0, 0, 0, 0.04);
}
img { 
  max-width:100%; 
  height: auto; 
}
</style>
<div class="container-fluid main-container">


<div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">singleCellSeq</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li><a href="index.html">Home</a></li>
        <li><a href="about.html">About</a></li>
        <li><a href="license.html">License</a></li>
        <li><a href="https://github.com/jdblischak/singleCellSeq">GitHub</a></li>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<div id="header">
<h1 class="title">Total molecule counts</h1>
<h4 class="author"><em>Po-Yuan Tung</em></h4>
<h4 class="date"><em>2016-04-18</em></h4>
</div>

<div id="TOC">
<ul>
<li><a href="#setup">Setup</a></li>
<li><a href="#the-total-molecule-counts-not-including-na19098">The total molecule-counts (not including NA19098)</a></li>
<li><a href="#the-relationship-between-concentration-and-total-molecule-counts">The relationship between concentration and total molecule-counts</a></li>
<li><a href="#ercc-counts-and-total-molecule-counts">ERCC counts and total molecule counts</a></li>
<li><a href="#plots-for-paper">Plots for paper</a></li>
<li><a href="#session-information">Session information</a></li>
</ul>
</div>

<p><strong>Last updated:</strong> 2016-04-28</p>
<p><strong>Code version:</strong> 619e90fd099bbed4c78929c937abbdbdaad65b88</p>
<p>Recreate the <a href="depth.html">Sequencing depth and cellular RNA content</a> using updated data files to make figures for the paper.</p>
<div id="setup" class="section level2">
<h2>Setup</h2>
<pre class="r"><code>library(&quot;dplyr&quot;)
library(&quot;tidyr&quot;)
library(&quot;edgeR&quot;)
library(&quot;lmtest&quot;)
library(&quot;ggplot2&quot;)
library(&quot;cowplot&quot;)
theme_set(theme_bw(base_size = 12))
theme_update(panel.grid.minor.x = element_blank(),
             panel.grid.minor.y = element_blank(),
             panel.grid.major.x = element_blank(),
             panel.grid.major.y = element_blank())
source(&quot;functions.R&quot;)</code></pre>
<p>Input annotation</p>
<pre class="r"><code>anno_single &lt;- read.table(&quot;../data/annotation.txt&quot;, header = TRUE,
                   stringsAsFactors = FALSE)
head(anno_single)</code></pre>
<pre><code>  individual replicate well      batch      sample_id
1    NA19098        r1  A01 NA19098.r1 NA19098.r1.A01
2    NA19098        r1  A02 NA19098.r1 NA19098.r1.A02
3    NA19098        r1  A03 NA19098.r1 NA19098.r1.A03
4    NA19098        r1  A04 NA19098.r1 NA19098.r1.A04
5    NA19098        r1  A05 NA19098.r1 NA19098.r1.A05
6    NA19098        r1  A06 NA19098.r1 NA19098.r1.A06</code></pre>
<p>Input single cell observational quality control data.</p>
<pre class="r"><code>qc &lt;- read.table(&quot;../data/qc-ipsc.txt&quot;, header = TRUE,
                 stringsAsFactors = FALSE)
qc &lt;- qc %&gt;% arrange(individual, replicate, well)
stopifnot(qc$individual == anno_single$individual,
          qc$replicate == anno_single$replicate,
          qc$well == anno_single$well)
head(qc)</code></pre>
<pre><code>  individual replicate well cell_number concentration tra1.60
1    NA19098        r1  A01           1      1.734785       1
2    NA19098        r1  A02           1      1.723038       1
3    NA19098        r1  A03           1      1.512786       1
4    NA19098        r1  A04           1      1.347492       1
5    NA19098        r1  A05           1      2.313047       1
6    NA19098        r1  A06           1      2.056803       1</code></pre>
<p>Incorporate informatin on cell number, concentration, and TRA1-60 status.</p>
<pre class="r"><code>anno_single$cell_number &lt;- qc$cell_number
anno_single$concentration &lt;- qc$concentration
anno_single$tra1.60 &lt;- qc$tra1.60</code></pre>
<p>Keep only quality single cell</p>
<pre class="r"><code>quality_single_cells &lt;- scan(&quot;../data/quality-single-cells.txt&quot;,
                             what = &quot;character&quot;)
anno_single &lt;- anno_single[anno_single$sample_id %in% quality_single_cells,]</code></pre>
<p>Input molecule counts after filtering</p>
<pre class="r"><code>molecules &lt;- read.table(&quot;../data/molecules-filter.txt&quot;, header = TRUE,
                    stringsAsFactors = FALSE)
stopifnot(colnames(molecules) == anno_single$sample_id)
ercc_index &lt;- grepl(&quot;ERCC&quot;, rownames(molecules))
anno_single$total_molecules_gene = colSums(molecules[!ercc_index, ])
anno_single$total_molecules_ercc = colSums(molecules[ercc_index, ])
anno_single$total_molecules = colSums(molecules)
anno_single$num_genes = apply(molecules[!ercc_index, ], 2, function(x) sum(x &gt; 0))</code></pre>
</div>
<div id="the-total-molecule-counts-not-including-na19098" class="section level2">
<h2>The total molecule-counts (not including NA19098)</h2>
<p>For wells with a single cell observed, the total molecule-counts range from 30408 to 96315, the first quartile for the total number of gene molecules is 55561.25, and the third quartile is 78173.25.</p>
<pre class="r"><code>## create a color palette with one color per individual and different shades for repplicates
great_color_8 &lt;- c(&quot;#CC3300&quot;, &quot;#FF9966&quot;, &quot;#006633&quot;, &quot;#009900&quot;, &quot;#99FF99&quot;, &quot;#3366FF&quot;, &quot;#6699FF&quot;, &quot;#66CCFF&quot;)

plot_total_molecules_gene &lt;- ggplot(anno_single,
                       aes(x = as.factor(batch), y = total_molecules_gene / 10^3, fill = as.factor(batch))) + 
                       geom_boxplot(alpha = .01, width = .2) +
                       geom_violin(alpha = .5) + 
                       scale_fill_manual(values = great_color_8) +
                       labs(x = &quot;Batch&quot;, y = &quot;Total gene molecule-counts (10^3)&quot;) +
                       theme(axis.text.x = element_text(hjust=1, angle = 45))

plot_total_molecules_ercc &lt;-  plot_total_molecules_gene %+%
                                   aes(y = total_molecules_ercc / 10^3) +
                                   labs(y = &quot;Total ERCC molecule-counts (10^3)&quot;)

summary(aov(total_molecules_gene ~ individual, data = anno_single))</code></pre>
<pre><code>             Df    Sum Sq   Mean Sq F value Pr(&gt;F)    
individual    2 2.571e+10 1.286e+10   70.39 &lt;2e-16 ***
Residuals   561 1.025e+11 1.827e+08                   
---
Signif. codes:  0 &#39;***&#39; 0.001 &#39;**&#39; 0.01 &#39;*&#39; 0.05 &#39;.&#39; 0.1 &#39; &#39; 1</code></pre>
<pre class="r"><code>plot_grid(plot_total_molecules_gene + theme(legend.position = &quot;none&quot;),
          plot_total_molecules_ercc + theme(legend.position = &quot;none&quot;),
          labels = LETTERS[1:2])</code></pre>
<p><img src="figure/total-molecule.Rmd/total_molecules-1.png" title="" alt="" width="1152" style="display: block; margin: auto;" /></p>
</div>
<div id="the-relationship-between-concentration-and-total-molecule-counts" class="section level2">
<h2>The relationship between concentration and total molecule-counts</h2>
<p>As we try to understand the general relationships between sequencing results and cellular mRNA content, we remove outlier batches. NA19098 replicate 1 failed the quantification of the concentration of the single cells and was hence removed. Because NA19098 concentration is only quantified in one replicate, we removed NA19098 from analysis involving batch differences and concentration.</p>
<pre class="r"><code>anno_single_6 &lt;- anno_single %&gt;% filter(individual != &quot;NA19098&quot;)</code></pre>
<pre class="r"><code>## look at endogenous genes
plot_conc_molecules_gene_individual &lt;- 
  ggplot(anno_single_6,
         aes(x = concentration, 
             y = total_molecules_gene / 10^3, color = individual)) +
     geom_point(alpha = 0.8) +
     geom_smooth(method = &quot;lm&quot;, se = FALSE) +
     labs(x = &quot;Concentration&quot;, y = &quot;Total gene molecule-counts (10^3)&quot;)

## look at ERCC spike-ins
plot_conc_molecules_ercc_individual &lt;-
  plot_conc_molecules_gene_individual %+%
     aes(y = total_molecules_ercc / 10^3) +
     labs(y = &quot;Total ERCC molecule-counts (10^3)&quot;)

## plots
plot_grid(plot_conc_molecules_gene_individual + theme(legend.position = &quot;none&quot;), 
          plot_conc_molecules_ercc_individual + theme(legend.position = &quot;none&quot;),
          labels = LETTERS[1:2])</code></pre>
<p><img src="figure/total-molecule.Rmd/molecules-num-cells-batch-1.png" title="" alt="" width="1152" style="display: block; margin: auto;" /></p>
<pre class="r"><code>## Is there a difference across the three individuals
table(anno_single_6$individual, anno_single_6$replicate)</code></pre>
<pre><code>         
          r1 r2 r3
  NA19101 80 70 51
  NA19239 74 68 79</code></pre>
<pre class="r"><code>fit0 &lt;- lm(total_molecules_gene ~ concentration, 
   data = anno_single_6)
fit1 &lt;- lm(total_molecules_gene ~ concentration + as.factor(individual), 
   data = anno_single_6)
# use likelihood ratio test to detect individual differences
lrtest(fit1, fit0)</code></pre>
<pre><code>Likelihood ratio test

Model 1: total_molecules_gene ~ concentration + as.factor(individual)
Model 2: total_molecules_gene ~ concentration
  #Df  LogLik Df  Chisq Pr(&gt;Chisq)    
1   4 -4555.1                         
2   3 -4577.1 -1 43.956  3.358e-11 ***
---
Signif. codes:  0 &#39;***&#39; 0.001 &#39;**&#39; 0.01 &#39;*&#39; 0.05 &#39;.&#39; 0.1 &#39; &#39; 1</code></pre>
<pre class="r"><code>## Is there a difference across replicates of 19101
fit0 &lt;- lm(total_molecules_gene ~ concentration, 
   data = anno_single_6, subset = individual == &quot;NA19101&quot;)
fit1 &lt;- lm(total_molecules_gene ~ concentration + as.factor(replicate), 
   data = anno_single_6, subset = individual == &quot;NA19101&quot;)
# use likelihood ratio test to detect individual differences
lrtest(fit1, fit0)</code></pre>
<pre><code>Likelihood ratio test

Model 1: total_molecules_gene ~ concentration + as.factor(replicate)
Model 2: total_molecules_gene ~ concentration
  #Df  LogLik Df  Chisq Pr(&gt;Chisq)
1   5 -2180.5                     
2   3 -2180.8 -2 0.6238     0.7321</code></pre>
<pre class="r"><code># quick line plot
ggplot(anno_single_6[anno_single_6$individual == &quot;NA19101&quot;, ],
       aes(x = concentration, 
           y = total_molecules_gene / 10^3, color = replicate)) +
   geom_point(alpha = 0.8) +
   geom_smooth(method = &quot;lm&quot;, se = FALSE) +
   labs(x = &quot;Concentration&quot;, 
        y = &quot;Total gene molecule-counts (10^3)&quot;)</code></pre>
<p><img src="figure/total-molecule.Rmd/molecules-num-cells-batch-2.png" title="" alt="" width="1152" style="display: block; margin: auto;" /></p>
<pre class="r"><code>## Is there a difference across replicates of 19239
fit0 &lt;- lm(total_molecules_gene ~ concentration, 
   data = anno_single_6, subset = individual == &quot;NA19239&quot;)
fit1 &lt;- lm(total_molecules_gene ~ concentration + as.factor(replicate), 
   data = anno_single_6, subset = individual == &quot;NA19239&quot;)
ggplot(anno_single_6[anno_single_6$individual == &quot;NA19239&quot;, ],
       aes(x = concentration, 
           y = total_molecules_gene / 10^3, color = replicate)) +
   geom_point(alpha = 0.8) +
   geom_smooth(method = &quot;lm&quot;, se = FALSE) +
   labs(x = &quot;Concentration&quot;, 
        y = &quot;Total gene molecule-counts (10^3)&quot;)</code></pre>
<p><img src="figure/total-molecule.Rmd/molecules-num-cells-batch-3.png" title="" alt="" width="1152" style="display: block; margin: auto;" /></p>
<pre class="r"><code># use likelihood ratio test to detect individual differences
lrtest(fit1, fit0)</code></pre>
<pre><code>Likelihood ratio test

Model 1: total_molecules_gene ~ concentration + as.factor(replicate)
Model 2: total_molecules_gene ~ concentration
  #Df  LogLik Df  Chisq Pr(&gt;Chisq)    
1   5 -2351.6                         
2   3 -2365.0 -2 26.662  1.623e-06 ***
---
Signif. codes:  0 &#39;***&#39; 0.001 &#39;**&#39; 0.01 &#39;*&#39; 0.05 &#39;.&#39; 0.1 &#39; &#39; 1</code></pre>
<p>Calculate correlation of concentration and molecule counts</p>
<pre class="r"><code>## for each individual
for (i in 1:length(unique(anno_single_6$individual))) {
    print(unique(anno_single_6$individual)[i])
    select_individual &lt;- with(anno_single_6, individual == unique(individual)[i])
    print(cor(anno_single_6[select_individual,7],anno_single_6[select_individual,9]))
}</code></pre>
<pre><code>[1] &quot;NA19101&quot;
[1] 0.2767428
[1] &quot;NA19239&quot;
[1] 0.524656</code></pre>
<pre class="r"><code>## for each batch
for (i in 1:length(unique(anno_single_6$batch))) {
    print(unique(anno_single_6$batch)[i])
    select_replicate &lt;- with(anno_single_6, batch == unique(batch)[i])
    print(cor(anno_single_6[select_replicate,7],
              anno_single_6[select_replicate,9]))
}</code></pre>
<pre><code>[1] &quot;NA19101.r1&quot;
[1] 0.3607414
[1] &quot;NA19101.r2&quot;
[1] 0.5084765
[1] &quot;NA19101.r3&quot;
[1] -0.0631127
[1] &quot;NA19239.r1&quot;
[1] 0.7466212
[1] &quot;NA19239.r2&quot;
[1] 0.7784135
[1] &quot;NA19239.r3&quot;
[1] -0.03799184</code></pre>
</div>
<div id="ercc-counts-and-total-molecule-counts" class="section level2">
<h2>ERCC counts and total molecule counts</h2>
<pre class="r"><code>## calulate ERCC percentage
anno_single &lt;- anno_single %&gt;%
  mutate(perc_ercc_molecules = total_molecules_ercc / total_molecules * 100)

## ERCC molecule versus total molecule
plot_gene_mol_ercc_mol &lt;- ggplot(anno_single,
                                 aes(x = total_molecules_gene,
                                     y = total_molecules_ercc)) +
                         geom_point(aes(color = individual, shape = replicate, alpha = 0.8))+
                         labs(x = &quot;Total gene molecules&quot;, y = &quot;Total ERCC molecules&quot;) 

plot_gene_mol_perc_ercc &lt;- plot_gene_mol_ercc_mol %+%
                           aes(x = perc_ercc_molecules)+
                           labs(x = &quot;Percent ERCC molecules&quot;)

plot_grid(plot_gene_mol_ercc_mol + theme(legend.position = &quot;none&quot;),
          plot_gene_mol_perc_ercc + theme(legend.position= &quot;none&quot;),
          labels = LETTERS[1:2])</code></pre>
<p><img src="figure/total-molecule.Rmd/ercc-total-counts-1.png" title="" alt="" width="1152" style="display: block; margin: auto;" /></p>
<pre class="r"><code>## Is there a difference across the three individuals
table(anno_single$individual, anno_single$replicate)</code></pre>
<pre><code>         
          r1 r2 r3
  NA19098 85  0 57
  NA19101 80 70 51
  NA19239 74 68 79</code></pre>
<pre class="r"><code>fit0 &lt;- lm(total_molecules_ercc ~ 1,  data = anno_single)
fit1 &lt;- lm(total_molecules_ercc ~ 1 + as.factor(individual), data = anno_single)
anova(fit0, fit1)</code></pre>
<pre><code>Analysis of Variance Table

Model 1: total_molecules_ercc ~ 1
Model 2: total_molecules_ercc ~ 1 + as.factor(individual)
  Res.Df      RSS Df Sum of Sq      F    Pr(&gt;F)    
1    563 34614165                                  
2    561 17394891  2  17219274 277.67 &lt; 2.2e-16 ***
---
Signif. codes:  0 &#39;***&#39; 0.001 &#39;**&#39; 0.01 &#39;*&#39; 0.05 &#39;.&#39; 0.1 &#39; &#39; 1</code></pre>
<pre class="r"><code>## Is there a difference across replicates for all individuals
table(anno_single$batch, anno_single$individual)</code></pre>
<pre><code>            
             NA19098 NA19101 NA19239
  NA19098.r1      85       0       0
  NA19098.r3      57       0       0
  NA19101.r1       0      80       0
  NA19101.r2       0      70       0
  NA19101.r3       0      51       0
  NA19239.r1       0       0      74
  NA19239.r2       0       0      68
  NA19239.r3       0       0      79</code></pre>
<pre class="r"><code>fit2 &lt;- lm(total_molecules_gene ~ concentration + as.factor(individual) + batch, 
   data = anno_single)

summary(fit2)$coef</code></pre>
<pre><code>                               Estimate Std. Error    t value
(Intercept)                   59887.763   2024.402 29.5829414
concentration                  8493.346   1026.977  8.2702406
as.factor(individual)NA19101 -17606.151   2221.855 -7.9240756
as.factor(individual)NA19239  -4818.372   1983.119 -2.4296936
batchNA19098.r3                6981.848   2159.017  3.2338089
batchNA19101.r1                1385.609   2234.378  0.6201319
batchNA19101.r2                1776.442   2290.976  0.7754085
batchNA19239.r1               -3112.020   2007.401 -1.5502736
batchNA19239.r2               -8494.326   2050.918 -4.1417193
                                  Pr(&gt;|t|)
(Intercept)                  3.619347e-116
concentration                 9.941901e-16
as.factor(individual)NA19101  1.265325e-14
as.factor(individual)NA19239  1.542774e-02
batchNA19098.r3               1.294113e-03
batchNA19101.r1               5.354255e-01
batchNA19101.r2               4.384286e-01
batchNA19239.r1               1.216459e-01
batchNA19239.r2               3.984182e-05</code></pre>
<pre class="r"><code>lrtest(fit1,fit2)</code></pre>
<pre><code>Likelihood ratio test

Model 1: total_molecules_ercc ~ 1 + as.factor(individual)
Model 2: total_molecules_gene ~ concentration + as.factor(individual) + 
    batch
  #Df  LogLik Df  Chisq Pr(&gt;Chisq)    
1   4 -3715.2                         
2  10 -6111.6  6 4792.8  &lt; 2.2e-16 ***
---
Signif. codes:  0 &#39;***&#39; 0.001 &#39;**&#39; 0.01 &#39;*&#39; 0.05 &#39;.&#39; 0.1 &#39; &#39; 1</code></pre>
</div>
<div id="plots-for-paper" class="section level2">
<h2>Plots for paper</h2>
<pre class="r"><code>plot_grid(plot_total_molecules_gene + theme(legend.position = &quot;none&quot;),
          plot_conc_molecules_gene_individual + theme(legend.position= &quot;none&quot;),
          plot_gene_mol_ercc_mol + theme(legend.position = &quot;none&quot;),
          plot_gene_mol_perc_ercc + theme(legend.position= &quot;none&quot;),
          labels = LETTERS[1:4])</code></pre>
<p><img src="figure/total-molecule.Rmd/plot-1.png" title="" alt="" width="1152" style="display: block; margin: auto;" /></p>
</div>
<div id="session-information" class="section level2">
<h2>Session information</h2>
<pre class="r"><code>sessionInfo()</code></pre>
<pre><code>R version 3.2.0 (2015-04-16)
Platform: x86_64-unknown-linux-gnu (64-bit)

locale:
 [1] LC_CTYPE=en_US.UTF-8       LC_NUMERIC=C              
 [3] LC_TIME=en_US.UTF-8        LC_COLLATE=en_US.UTF-8    
 [5] LC_MONETARY=en_US.UTF-8    LC_MESSAGES=en_US.UTF-8   
 [7] LC_PAPER=en_US.UTF-8       LC_NAME=C                 
 [9] LC_ADDRESS=C               LC_TELEPHONE=C            
[11] LC_MEASUREMENT=en_US.UTF-8 LC_IDENTIFICATION=C       

attached base packages:
[1] stats     graphics  grDevices utils     datasets  methods   base     

other attached packages:
[1] cowplot_0.3.1 ggplot2_1.0.1 lmtest_0.9-34 zoo_1.7-12    edgeR_3.10.2 
[6] limma_3.24.9  tidyr_0.2.0   dplyr_0.4.2   knitr_1.10.5 

loaded via a namespace (and not attached):
 [1] Rcpp_0.12.0      magrittr_1.5     MASS_7.3-40      munsell_0.4.2   
 [5] colorspace_1.2-6 lattice_0.20-31  R6_2.1.1         plyr_1.8.3      
 [9] stringr_1.0.0    httr_0.6.1       tools_3.2.0      parallel_3.2.0  
[13] grid_3.2.0       gtable_0.1.2     DBI_0.3.1        htmltools_0.2.6 
[17] lazyeval_0.1.10  yaml_2.1.13      assertthat_0.1   digest_0.6.8    
[21] reshape2_1.4.1   formatR_1.2      bitops_1.0-6     RCurl_1.95-4.6  
[25] evaluate_0.7     rmarkdown_0.6.1  labeling_0.3     stringi_0.4-1   
[29] scales_0.2.4     proto_0.3-10    </code></pre>
</div>


<!-- some extra javascript for older browsers -->
<script type="text/javascript" src="libs/polyfill.js"></script>

<script>

// manage active state of menu based on current page
$(document).ready(function () {

    // active menu
    href = window.location.pathname
    href = href.substr(href.lastIndexOf('/') + 1)
    $('a[href="' + href + '"]').parent().addClass('active');

    // manage active menu header
    if (href.startsWith('authoring_'))
      $('a[href="' + 'authoring' + '"]').parent().addClass('active');
    else if (href.endsWith('_format.html'))
      $('a[href="' + 'formats' + '"]').parent().addClass('active');
    else if (href.startsWith('developer_'))
      $('a[href="' + 'developer' + '"]').parent().addClass('active');

});

</script>

</div>

<script>

// add bootstrap table styles to pandoc tables
$(document).ready(function () {
  $('tr.header').parent('thead').parent('table').addClass('table table-condensed');
});

</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://cdn.mathjax.org/mathjax/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
