<!DOCTYPE html>

<html xmlns="http://www.w3.org/1999/xhtml">

<head>

<meta charset="utf-8">
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="generator" content="pandoc" />

<meta name="author" content="John Blischak" />

<meta name="date" content="2015-04-22" />

<title>Total counts</title>

<script src="libs/jquery-1.11.0/jquery.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="libs/bootstrap-3.3.1/css/united.min.css" rel="stylesheet" />
<script src="libs/bootstrap-3.3.1/js/bootstrap.min.js"></script>
<script src="libs/bootstrap-3.3.1/shim/html5shiv.min.js"></script>
<script src="libs/bootstrap-3.3.1/shim/respond.min.js"></script>

<style type="text/css">

/* padding for bootstrap navbar */
body {
  padding-top: 50px;
  padding-bottom: 40px;
}


/* offset scroll position for anchor links (for fixed navbar)  */
.section h2 {
  padding-top: 55px;
  margin-top: -55px;
}
.section h3 {
  padding-top: 55px;
  margin-top: -55px;
}



/* don't use link color in navbar */
.dropdown-menu>li>a {
  color: black;
}

/* some padding for disqus */
#disqus_thread {
  margin-top: 45px;
}

</style>

<link rel="stylesheet" href="libs/font-awesome-4.1.0/css/font-awesome.min.css"/>

<style type="text/css">code{white-space: pre;}</style>
<link rel="stylesheet"
      href="libs/highlight/textmate.css"
      type="text/css" />
<script src="libs/highlight/highlight.js"></script>
<style type="text/css">
  pre:not([class]) {
    background-color: white;
  }
</style>
<script type="text/javascript">
if (window.hljs && document.readyState && document.readyState === "complete") {
   window.setTimeout(function() {
      hljs.initHighlighting();
   }, 0);
}
</script>



</head>

<body>

<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
code {
  color: inherit;
  background-color: rgba(0, 0, 0, 0.04);
}
img { 
  max-width:100%; 
  height: auto; 
}
</style>
<div class="container-fluid main-container">


<div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">singleCellSeq</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li><a href="index.html">Home</a></li>
        <li><a href="about.html">About</a></li>
        <li><a href="license.html">License</a></li>
        <li><a href="https://github.com/jdblischak/singleCellSeq">GitHub</a></li>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<div id="header">
<h1 class="title">Total counts</h1>
<h4 class="author"><em>John Blischak</em></h4>
<h4 class="date"><em>2015-04-22</em></h4>
</div>

<div id="TOC">
<ul>
<li><a href="#setup">Setup</a></li>
<li><a href="#read-counts-per-lane">Read counts per lane</a></li>
<li><a href="#read-counts-per-sample">Read counts per sample</a></li>
<li><a href="#molecule-counts-per-lane">Molecule counts per lane</a></li>
<li><a href="#molecule-counts-per-sample">Molecule counts per sample</a></li>
<li><a href="#summary-for-single-cells">Summary for single cells</a></li>
<li><a href="#session-information">Session information</a></li>
</ul>
</div>

<p><strong>Last updated:</strong> 2016-04-24</p>
<p><strong>Code version:</strong> 74f30e4e4b6c6fea3667336da91b01fb5603564f</p>
<p>During the <a href="process-samples.html">processing pipeline</a>, the number of reads in a file are counted and saved in a separate text files. The script <a href="https://github.com/jdblischak/singleCellSeq/blob/master/code/gather-total-counts.py">gather-total-counts.py</a> compiles all these counts and extracts the relevant variables from the filename.</p>
<div id="setup" class="section level2">
<h2>Setup</h2>
<pre class="r"><code>library(&quot;dplyr&quot;)
library(&quot;ggplot2&quot;)
theme_set(theme_bw(base_size = 16))
theme_update(panel.grid.minor.x = element_blank(),
             panel.grid.minor.y = element_blank(),
             panel.grid.major.x = element_blank(),
             panel.grid.major.y = element_blank())</code></pre>
<pre class="r"><code>counts &lt;- read.table(&quot;../data/total-counts.txt&quot;, header = TRUE, sep = &quot;\t&quot;,
                     stringsAsFactors = TRUE)</code></pre>
<pre class="r"><code>head(counts)</code></pre>
<pre><code>  stage sequences combined individual replicate well    index lane
1   raw     reads       NA    NA19098        r1  A01 ATTAGACG L002
2   raw     reads       NA    NA19098        r1  A01 ATTAGACG L003
3   raw     reads       NA    NA19098        r1  A01 ATTAGACG L006
4   raw     reads       NA    NA19098        r1  A02 ACGAGTCT L002
5   raw     reads       NA    NA19098        r1  A02 ACGAGTCT L003
6   raw     reads       NA    NA19098        r1  A02 ACGAGTCT L006
  flow_cell  counts
1 C6WYKACXX 1986825
2 C6WURACXX 1785837
3 C723YACXX 1739622
4 C6WYKACXX 2019671
5 C6WURACXX 1809239
6 C723YACXX 1762255</code></pre>
<pre class="r"><code>str(counts)</code></pre>
<pre><code>&#39;data.frame&#39;:   20232 obs. of  10 variables:
 $ stage     : Factor w/ 5 levels &quot;mapped to exons&quot;,..: 4 4 4 4 4 4 4 4 4 4 ...
 $ sequences : Factor w/ 2 levels &quot;molecules&quot;,&quot;reads&quot;: 2 2 2 2 2 2 2 2 2 2 ...
 $ combined  : logi  NA NA NA NA NA NA ...
 $ individual: Factor w/ 3 levels &quot;NA19098&quot;,&quot;NA19101&quot;,..: 1 1 1 1 1 1 1 1 1 1 ...
 $ replicate : Factor w/ 3 levels &quot;r1&quot;,&quot;r2&quot;,&quot;r3&quot;: 1 1 1 1 1 1 1 1 1 1 ...
 $ well      : Factor w/ 97 levels &quot;A01&quot;,&quot;A02&quot;,&quot;A03&quot;,..: 1 1 1 2 2 2 3 3 3 4 ...
 $ index     : Factor w/ 96 levels &quot;AAAGGTCC&quot;,&quot;AACCCATG&quot;,..: 23 23 23 13 13 13 85 85 85 68 ...
 $ lane      : Factor w/ 8 levels &quot;L001&quot;,&quot;L002&quot;,..: 2 3 6 2 3 6 2 3 6 2 ...
 $ flow_cell : Factor w/ 4 levels &quot;C6WURACXX&quot;,&quot;C6WYKACXX&quot;,..: 2 1 3 2 1 3 2 1 3 2 ...
 $ counts    : int  1986825 1785837 1739622 2019671 1809239 1762255 1191181 1056347 989324 1601993 ...</code></pre>
<pre class="r"><code># Order the processing steps
counts$stage &lt;- factor(counts$stage,
                       levels = c(&quot;raw&quot;, &quot;valid UMI&quot;, &quot;quality trimmed&quot;,
                                  &quot;mapped to genome&quot;, &quot;mapped to exons&quot;))
# Make new variable to separate bulk and single cell samples
counts$type &lt;- ifelse(counts$well == &quot;bulk&quot;, &quot;bulk&quot;, &quot;single&quot;)
counts$type &lt;- factor(counts$type, levels = c(&quot;bulk&quot;, &quot;single&quot;))
# Scale to thousands and millions of counts
counts$counts_thousands &lt;- counts$counts / 10^3
counts$counts_mil &lt;- counts$counts / 10^6
summary(counts)</code></pre>
<pre><code>              stage          sequences      combined         individual  
 raw             :2664   molecules: 6912   Mode :logical   NA19098:6744  
 valid UMI       :2664   reads    :13320   FALSE:7848      NA19101:6744  
 quality trimmed :2664                     TRUE :1728      NA19239:6744  
 mapped to genome:6120                     NA&#39;s :10656                   
 mapped to exons :6120                                                   
                                                                         
                                                                         
 replicate      well            index            lane     
 r1:6744   bulk   :  360   AAGTCTTC:  209   L002   :2688  
 r2:6744   A01    :  207   ACACAACT:  209   L003   :2688  
 r3:6744   A02    :  207   ACAGCGAA:  209   L004   :2688  
           A03    :  207   AGTCCACC:  209   L006   :2688  
           A04    :  207   CACGGTGT:  209   L001   :2106  
           A05    :  207   (Other) :17459   (Other):5646  
           (Other):18837   NA&#39;s    : 1728   NA&#39;s   :1728  
     flow_cell        counts             type       counts_thousands   
 C6WURACXX:4794   Min.   :    2495   bulk  :  360   Min.   :    2.495  
 C6WYKACXX:4794   1st Qu.:   84396   single:19872   1st Qu.:   84.396  
 C723YACXX:4794   Median :  822508                  Median :  822.509  
 C72JMACXX:4122   Mean   : 1002821                  Mean   : 1002.821  
 NA&#39;s     :1728   3rd Qu.: 1578909                  3rd Qu.: 1578.909  
                  Max.   :15126607                  Max.   :15126.607  
                                                                       
   counts_mil       
 Min.   : 0.002495  
 1st Qu.: 0.084396  
 Median : 0.822508  
 Mean   : 1.002821  
 3rd Qu.: 1.578909  
 Max.   :15.126607  
                    </code></pre>
</div>
<div id="read-counts-per-lane" class="section level2">
<h2>Read counts per lane</h2>
<pre class="r"><code>counts_per_lane &lt;- counts %&gt;% filter(!combined | is.na(combined),
                                     sequences == &quot;reads&quot;)
stopifnot(table(counts_per_lane$stage) == 2664)</code></pre>
<p>Plot the number of reads per lane at each processing stage faceted by bulk versus single cell sequencing.</p>
<pre class="r"><code>ggplot(counts_per_lane,
       aes(x = stage, y = counts_mil)) +
  geom_boxplot() +
  facet_wrap(~type) +
  scale_y_continuous(breaks = seq(0, 16, 2)) +
  labs(x = &quot;Processing stage&quot;, y = &quot;Number of reads (millions)&quot;,
       title = &quot;Total number of reads per lane at each stage&quot;) +
  theme(axis.text.x = element_text(angle = 45, hjust = 0.9, vjust = 0.75))</code></pre>
<p><img src="figure/total-counts.Rmd/counts-per-lane-by-processing-stage-1.png" title="" alt="" width="768" style="display: block; margin: auto;" /></p>
<p>Plot the number of raw reads per lane for each replicate of each individual faceted by bulk versus single cell sequencing.</p>
<pre class="r"><code>ggplot(counts_per_lane[counts$stage == &quot;raw&quot;, ],
       aes(x = individual, y = counts_mil, fill = replicate)) +
  geom_boxplot() +
  facet_wrap(~type) +
  scale_y_continuous(breaks = seq(0, 16, 2)) +
  labs(x = &quot;Individual&quot;, y = &quot;Number of raw reads (millions)&quot;,
       title = &quot;Total number of raw reads per lane for each replicate&quot;) +
  theme(axis.text.x = element_text(angle = 45, hjust = 0.9, vjust = 0.75))</code></pre>
<p><img src="figure/total-counts.Rmd/counts-per-lane-per-replicate-1.png" title="" alt="" width="768" style="display: block; margin: auto;" /></p>
<p>Plot the number of raw reads per lane for each lane of each flow cell faceted by bulk versus single cell sequencing.</p>
<pre class="r"><code>ggplot(counts_per_lane[counts$stage == &quot;raw&quot;, ],
       aes(x = flow_cell, y = counts_mil, fill = lane)) +
  geom_boxplot() +
  facet_wrap(~type) +
  scale_y_continuous(breaks = seq(0, 16, 2)) +
  labs(x = &quot;Flow cell&quot;, y = &quot;Number of raw reads (millions)&quot;,
       title = &quot;Total number of raw reads per lane for each lane&quot;) +
  theme(axis.text.x = element_text(angle = 45, hjust = 0.9, vjust = 0.75))</code></pre>
<p><img src="figure/total-counts.Rmd/counts-per-lane-per-lane-1.png" title="" alt="" width="768" style="display: block; margin: auto;" /></p>
<p>Plot the number of raw reads per lane for each index faceted by bulk versus single cell sequencing.</p>
<pre class="r"><code>ggplot(counts_per_lane[counts$stage == &quot;raw&quot;, ],
       aes(x = index, y = counts_mil)) +
  geom_boxplot() +
  facet_wrap(~type) +
  scale_y_continuous(breaks = seq(0, 16, 2)) +
  labs(x = &quot;Index&quot;, y = &quot;Number of raw reads (millions)&quot;,
       title = &quot;Total number of raw reads per lane for each index&quot;) +
  theme(axis.text.x = element_blank())</code></pre>
<p><img src="figure/total-counts.Rmd/counts-per-lane-per-index-1.png" title="" alt="" width="768" style="display: block; margin: auto;" /></p>
</div>
<div id="read-counts-per-sample" class="section level2">
<h2>Read counts per sample</h2>
<pre class="r"><code>counts_per_sample &lt;- counts_per_lane %&gt;%
  group_by(stage, individual, replicate, well, type) %&gt;%
  summarize(counts_mil = sum(counts) / 10^6)
stopifnot(table(counts_per_sample$stage) == 864 + 9)</code></pre>
<p>Plot the number of reads per sample at each processing stage faceted by bulk versus single cell sequencing.</p>
<pre class="r"><code>ggplot(counts_per_sample,
       aes(x = stage, y = counts_mil)) +
  geom_boxplot() +
  facet_wrap(~type) +
  scale_y_continuous(breaks = seq(0, 100, 10)) +
  labs(x = &quot;Processing stage&quot;, y = &quot;Number of reads (millions)&quot;,
       title = &quot;Total number of reads per sample at each stage&quot;) +
  theme(axis.text.x = element_text(angle = 45, hjust = 0.9, vjust = 0.75))</code></pre>
<p><img src="figure/total-counts.Rmd/counts-per-sample-by-processing-stage-1.png" title="" alt="" width="768" style="display: block; margin: auto;" /></p>
<p>Plot the number of raw reads per sample for each replicate of each individual faceted by bulk versus single cell sequencing.</p>
<pre class="r"><code>ggplot(counts_per_sample[counts$stage == &quot;raw&quot;, ],
       aes(x = individual, y = counts_mil, fill = replicate)) +
  geom_boxplot() +
  facet_wrap(~type) +
  scale_y_continuous(breaks = seq(0, 100, 10)) +
  labs(x = &quot;Individual&quot;, y = &quot;Number of raw reads (millions)&quot;,
       title = &quot;Total number of raw reads per sample for each replicate&quot;) +
  theme(axis.text.x = element_text(angle = 45, hjust = 0.9, vjust = 0.75))</code></pre>
<p><img src="figure/total-counts.Rmd/counts-per-sample-per-replicate-1.png" title="" alt="" width="768" style="display: block; margin: auto;" /></p>
</div>
<div id="molecule-counts-per-lane" class="section level2">
<h2>Molecule counts per lane</h2>
<pre class="r"><code>molecules_per_lane &lt;- counts %&gt;% filter(!combined | is.na(combined),
                                        sequences == &quot;molecules&quot;,
                                        stage %in% c(&quot;mapped to genome&quot;,
                                                     &quot;mapped to exons&quot;))
molecules_per_lane &lt;- droplevels(molecules_per_lane)
stopifnot(table(molecules_per_lane$stage) == 96 * 3 * 3 * 3)</code></pre>
<p>Plot the number of molecules per lane at each processing stage.</p>
<pre class="r"><code>ggplot(molecules_per_lane,
       aes(x = stage, y = counts_thousands)) +
  geom_boxplot() +
  scale_y_continuous(breaks = seq(0, 300, 50)) +
  labs(x = &quot;Processing stage&quot;, y = &quot;Number of molecules (thousands)&quot;,
       title = &quot;Total number of molecules per lane at each stage&quot;) +
  theme(axis.text.x = element_text(angle = 45, hjust = 0.9, vjust = 0.75))</code></pre>
<p><img src="figure/total-counts.Rmd/molecules-per-lane-by-processing-stage-1.png" title="" alt="" width="768" style="display: block; margin: auto;" /></p>
<p>Plot the number of mapped molecules per lane for each replicate of each individual.</p>
<pre class="r"><code>ggplot(molecules_per_lane[molecules_per_lane$stage == &quot;mapped to genome&quot;, ],
       aes(x = individual, y = counts_thousands, fill = replicate)) +
  geom_boxplot() +
  scale_y_continuous(breaks = seq(0, 300, 50)) +
  labs(x = &quot;Individual&quot;, y = &quot;Number of mapped molecules (thousands)&quot;,
       title = &quot;Total number of mapped molecules per lane for each replicate&quot;) +
  theme(axis.text.x = element_text(angle = 45, hjust = 0.9, vjust = 0.75))</code></pre>
<p><img src="figure/total-counts.Rmd/molecules-per-lane-per-replicate-1.png" title="" alt="" width="768" style="display: block; margin: auto;" /></p>
<p>Plot the number of mapped molecules per lane for each lane of each flow cell faceted by bulk versus single cell sequencing.</p>
<pre class="r"><code>ggplot(molecules_per_lane[molecules_per_lane$stage == &quot;mapped to genome&quot;, ],
       aes(x = flow_cell, y = counts_thousands, fill = lane)) +
  geom_boxplot() +
  scale_y_continuous(breaks = seq(0, 300, 50)) +
  labs(x = &quot;Flow cell&quot;, y = &quot;Number of mapped molecules (thousands)&quot;,
       title = &quot;Total number of mapped molecules per lane for each lane&quot;) +
  theme(axis.text.x = element_text(angle = 45, hjust = 0.9, vjust = 0.75))</code></pre>
<p><img src="figure/total-counts.Rmd/molecules-per-lane-per-lane-1.png" title="" alt="" width="768" style="display: block; margin: auto;" /></p>
<p>Plot the number of mapped molecules per lane for each index faceted by bulk versus single cell sequencing.</p>
<pre class="r"><code>ggplot(molecules_per_lane[molecules_per_lane$stage == &quot;mapped to genome&quot;, ],
       aes(x = index, y = counts_thousands)) +
  geom_boxplot() +
  scale_y_continuous(breaks = seq(0, 500, 50)) +
  labs(x = &quot;Index&quot;, y = &quot;Number of mapped molecules (thousands)&quot;,
       title = &quot;Total number of mapped molecules per lane for each index&quot;) +
  theme(axis.text.x = element_blank())</code></pre>
<p><img src="figure/total-counts.Rmd/molecules-per-lane-per-index-1.png" title="" alt="" width="768" style="display: block; margin: auto;" /></p>
</div>
<div id="molecule-counts-per-sample" class="section level2">
<h2>Molecule counts per sample</h2>
<pre class="r"><code>molecules_per_sample &lt;- counts %&gt;% filter(combined == TRUE,
                                          stage %in% c(&quot;mapped to genome&quot;,
                                                       &quot;mapped to exons&quot;))
molecules_per_sample &lt;- droplevels(molecules_per_sample)
stopifnot(molecules_per_sample$sequences == &quot;molecules&quot;,
          table(molecules_per_sample$stage) == 864)</code></pre>
<p>Plot the number of molecules per sample at each processing stage.</p>
<pre class="r"><code>ggplot(molecules_per_sample,
       aes(x = stage, y = counts_thousands)) +
  geom_boxplot() +
  scale_y_continuous(breaks = seq(0, 400, 50)) +
  labs(x = &quot;Processing stage&quot;, y = &quot;Number of molecules (thousands)&quot;,
       title = &quot;Total number of molecules per sample at each stage&quot;) +
  theme(axis.text.x = element_text(angle = 45, hjust = 0.9, vjust = 0.75))</code></pre>
<p><img src="figure/total-counts.Rmd/molecules-per-sample-by-processing-stage-1.png" title="" alt="" width="768" style="display: block; margin: auto;" /></p>
<p>Plot the number of mapped molecules per sample for each replicate of each individual.</p>
<pre class="r"><code>ggplot(molecules_per_sample[molecules_per_sample$stage == &quot;mapped to genome&quot;, ],
       aes(x = individual, y = counts_thousands, fill = replicate)) +
  geom_boxplot() +
  scale_y_continuous(breaks = seq(0, 400, 50)) +
  labs(x = &quot;Individual&quot;, y = &quot;Number of mapped molecules (thousands)&quot;,
       title = &quot;Total number of mapped molecules per sample for each replicate&quot;) +
  theme(axis.text.x = element_text(angle = 45, hjust = 0.9, vjust = 0.75))</code></pre>
<p><img src="figure/total-counts.Rmd/molecules-per-sample-per-replicate-1.png" title="" alt="" width="768" style="display: block; margin: auto;" /></p>
</div>
<div id="summary-for-single-cells" class="section level2">
<h2>Summary for single cells</h2>
<p>These summary statistics include low quality cells that are removed before downstream analyses.</p>
<pre class="r"><code>depth_stats &lt;- counts %&gt;%
  filter(stage == &quot;raw&quot;, well != &quot;bulk&quot;) %&gt;%
  group_by(individual, replicate, well) %&gt;%
  summarize(counts_per_sample = sum(counts)) %&gt;%
  ungroup() %&gt;%
  summarize(mean = mean(counts_per_sample), sd = sd(counts_per_sample),
            min = min(counts_per_sample), max = max(counts_per_sample))</code></pre>
<p>We obtained an average of 6.28 +/- 2.11 million raw sequencing reads per sample (range 0.361-11.2 million reads).</p>
<pre class="r"><code>exons_reads_stats &lt;- counts %&gt;%
  filter(stage == &quot;mapped to exons&quot;, well != &quot;bulk&quot;, sequences == &quot;reads&quot;) %&gt;%
  group_by(individual, replicate, well) %&gt;%
  summarize(counts_per_sample = sum(counts)) %&gt;%
  ungroup() %&gt;%
  summarize(mean = mean(counts_per_sample), sd = sd(counts_per_sample),
            min = min(counts_per_sample), max = max(counts_per_sample))</code></pre>
<p>We obtained an average of 2.1 +/- 0.879 million sequencing reads per sample that mapped to protein-coding exons (range 0.0552-4.33 million reads).</p>
<pre class="r"><code>exons_molecules_stats &lt;- counts %&gt;%
  filter(stage == &quot;mapped to exons&quot;, well != &quot;bulk&quot;, sequences == &quot;molecules&quot;,
         combined) %&gt;%
  group_by(individual, replicate, well) %&gt;%
  summarize(counts_per_sample = sum(counts)) %&gt;%
  ungroup() %&gt;%
  summarize(mean = mean(counts_per_sample), sd = sd(counts_per_sample),
            min = min(counts_per_sample), max = max(counts_per_sample))</code></pre>
<p>We obtained an average of 68.1 +/- 28.2 thousand molecules per sample that mapped to protein-coding exons (range 6.03-235 thousand molecules).</p>
</div>
<div id="session-information" class="section level2">
<h2>Session information</h2>
<pre class="r"><code>sessionInfo()</code></pre>
<pre><code>R version 3.2.0 (2015-04-16)
Platform: x86_64-unknown-linux-gnu (64-bit)

locale:
 [1] LC_CTYPE=en_US.UTF-8       LC_NUMERIC=C              
 [3] LC_TIME=en_US.UTF-8        LC_COLLATE=en_US.UTF-8    
 [5] LC_MONETARY=en_US.UTF-8    LC_MESSAGES=en_US.UTF-8   
 [7] LC_PAPER=en_US.UTF-8       LC_NAME=C                 
 [9] LC_ADDRESS=C               LC_TELEPHONE=C            
[11] LC_MEASUREMENT=en_US.UTF-8 LC_IDENTIFICATION=C       

attached base packages:
[1] stats     graphics  grDevices utils     datasets  methods   base     

other attached packages:
[1] ggplot2_1.0.1 dplyr_0.4.2   knitr_1.10.5 

loaded via a namespace (and not attached):
 [1] Rcpp_0.12.0      magrittr_1.5     MASS_7.3-40      munsell_0.4.2   
 [5] colorspace_1.2-6 R6_2.1.1         stringr_1.0.0    httr_0.6.1      
 [9] plyr_1.8.3       tools_3.2.0      parallel_3.2.0   grid_3.2.0      
[13] gtable_0.1.2     DBI_0.3.1        htmltools_0.2.6  lazyeval_0.1.10 
[17] yaml_2.1.13      assertthat_0.1   digest_0.6.8     reshape2_1.4.1  
[21] formatR_1.2      bitops_1.0-6     RCurl_1.95-4.6   evaluate_0.7    
[25] rmarkdown_0.6.1  stringi_0.4-1    scales_0.2.4     proto_0.3-10    </code></pre>
</div>


<!-- some extra javascript for older browsers -->
<script type="text/javascript" src="libs/polyfill.js"></script>

<script>

// manage active state of menu based on current page
$(document).ready(function () {

    // active menu
    href = window.location.pathname
    href = href.substr(href.lastIndexOf('/') + 1)
    $('a[href="' + href + '"]').parent().addClass('active');

    // manage active menu header
    if (href.startsWith('authoring_'))
      $('a[href="' + 'authoring' + '"]').parent().addClass('active');
    else if (href.endsWith('_format.html'))
      $('a[href="' + 'formats' + '"]').parent().addClass('active');
    else if (href.startsWith('developer_'))
      $('a[href="' + 'developer' + '"]').parent().addClass('active');

});

</script>

</div>

<script>

// add bootstrap table styles to pandoc tables
$(document).ready(function () {
  $('tr.header').parent('thead').parent('table').addClass('table table-condensed');
});

</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://cdn.mathjax.org/mathjax/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
