---
title: "Total counts"
author: "John Blischak"
date: 2015-04-22
---

**Last updated:** `r Sys.Date()`

**Code version:** `r system("git log -1 --format='%H'", intern = TRUE)`

```{r chunk-options, include=FALSE}
source("chunk-options.R")
opts_chunk$set(fig.width = 8, fig.height = 8)
```

During the [processing pipeline](process-samples.html), the number of reads in a file are counted and saved in a separate text files.
The script [gather-total-counts.py][] compiles all these counts and extracts the relevant variables from the filename.

[gather-total-counts.py]: https://github.com/jdblischak/singleCellSeq/blob/master/code/gather-total-counts.py

## Setup

```{r packages, message=FALSE}
library("dplyr")
library("ggplot2")
theme_set(theme_bw(base_size = 16))
theme_update(panel.grid.minor.x = element_blank(),
             panel.grid.minor.y = element_blank(),
             panel.grid.major.x = element_blank(),
             panel.grid.major.y = element_blank())
```

```{r input}
counts <- read.table("../data/total-counts.txt", header = TRUE, sep = "\t",
                     stringsAsFactors = TRUE)
```

```{r}
head(counts)
str(counts)
```

```{r}
# Order the processing steps
counts$stage <- factor(counts$stage,
                       levels = c("raw", "valid UMI", "quality trimmed",
                                  "mapped to genome", "mapped to exons"))
# Make new variable to separate bulk and single cell samples
counts$type <- ifelse(counts$well == "bulk", "bulk", "single")
counts$type <- factor(counts$type, levels = c("bulk", "single"))
# Scale to thousands and millions of counts
counts$counts_thousands <- counts$counts / 10^3
counts$counts_mil <- counts$counts / 10^6
summary(counts)
```

## Read counts per lane

```{r calc-reads-lane}
counts_per_lane <- counts %>% filter(!combined | is.na(combined),
                                     sequences == "reads")
stopifnot(table(counts_per_lane$stage) == 2664)
```

Plot the number of reads per lane at each processing stage faceted by bulk versus single cell sequencing.

```{r counts-per-lane-by-processing-stage}
ggplot(counts_per_lane,
       aes(x = stage, y = counts_mil)) +
  geom_boxplot() +
  facet_wrap(~type) +
  scale_y_continuous(breaks = seq(0, 16, 2)) +
  labs(x = "Processing stage", y = "Number of reads (millions)",
       title = "Total number of reads per lane at each stage") +
  theme(axis.text.x = element_text(angle = 45, hjust = 0.9, vjust = 0.75))
```

Plot the number of raw reads per lane for each replicate of each individual faceted by bulk versus single cell sequencing.

```{r counts-per-lane-per-replicate}
ggplot(counts_per_lane[counts$stage == "raw", ],
       aes(x = individual, y = counts_mil, fill = replicate)) +
  geom_boxplot() +
  facet_wrap(~type) +
  scale_y_continuous(breaks = seq(0, 16, 2)) +
  labs(x = "Individual", y = "Number of raw reads (millions)",
       title = "Total number of raw reads per lane for each replicate") +
  theme(axis.text.x = element_text(angle = 45, hjust = 0.9, vjust = 0.75))
```

Plot the number of raw reads per lane for each lane of each flow cell faceted by bulk versus single cell sequencing.

```{r counts-per-lane-per-lane}
ggplot(counts_per_lane[counts$stage == "raw", ],
       aes(x = flow_cell, y = counts_mil, fill = lane)) +
  geom_boxplot() +
  facet_wrap(~type) +
  scale_y_continuous(breaks = seq(0, 16, 2)) +
  labs(x = "Flow cell", y = "Number of raw reads (millions)",
       title = "Total number of raw reads per lane for each lane") +
  theme(axis.text.x = element_text(angle = 45, hjust = 0.9, vjust = 0.75))
```

Plot the number of raw reads per lane for each index faceted by bulk versus single cell sequencing.

```{r counts-per-lane-per-index}
ggplot(counts_per_lane[counts$stage == "raw", ],
       aes(x = index, y = counts_mil)) +
  geom_boxplot() +
  facet_wrap(~type) +
  scale_y_continuous(breaks = seq(0, 16, 2)) +
  labs(x = "Index", y = "Number of raw reads (millions)",
       title = "Total number of raw reads per lane for each index") +
  theme(axis.text.x = element_blank())
```

## Read counts per sample

```{r calc-reads-sample}
counts_per_sample <- counts_per_lane %>%
  group_by(stage, individual, replicate, well, type) %>%
  summarize(counts_mil = sum(counts) / 10^6)
stopifnot(table(counts_per_sample$stage) == 864 + 9)
```

Plot the number of reads per sample at each processing stage faceted by bulk versus single cell sequencing.

```{r counts-per-sample-by-processing-stage}
ggplot(counts_per_sample,
       aes(x = stage, y = counts_mil)) +
  geom_boxplot() +
  facet_wrap(~type) +
  scale_y_continuous(breaks = seq(0, 100, 10)) +
  labs(x = "Processing stage", y = "Number of reads (millions)",
       title = "Total number of reads per sample at each stage") +
  theme(axis.text.x = element_text(angle = 45, hjust = 0.9, vjust = 0.75))
```

Plot the number of raw reads per sample for each replicate of each individual faceted by bulk versus single cell sequencing.

```{r counts-per-sample-per-replicate}
ggplot(counts_per_sample[counts$stage == "raw", ],
       aes(x = individual, y = counts_mil, fill = replicate)) +
  geom_boxplot() +
  facet_wrap(~type) +
  scale_y_continuous(breaks = seq(0, 100, 10)) +
  labs(x = "Individual", y = "Number of raw reads (millions)",
       title = "Total number of raw reads per sample for each replicate") +
  theme(axis.text.x = element_text(angle = 45, hjust = 0.9, vjust = 0.75))
```

## Molecule counts per lane

```{r calc-molecules-lane}
molecules_per_lane <- counts %>% filter(!combined | is.na(combined),
                                        sequences == "molecules",
                                        stage %in% c("mapped to genome",
                                                     "mapped to exons"))
molecules_per_lane <- droplevels(molecules_per_lane)
stopifnot(table(molecules_per_lane$stage) == 96 * 3 * 3 * 3)
```

Plot the number of molecules per lane at each processing stage.

```{r molecules-per-lane-by-processing-stage}
ggplot(molecules_per_lane,
       aes(x = stage, y = counts_thousands)) +
  geom_boxplot() +
  scale_y_continuous(breaks = seq(0, 300, 50)) +
  labs(x = "Processing stage", y = "Number of molecules (thousands)",
       title = "Total number of molecules per lane at each stage") +
  theme(axis.text.x = element_text(angle = 45, hjust = 0.9, vjust = 0.75))
```

Plot the number of mapped molecules per lane for each replicate of each individual.

```{r molecules-per-lane-per-replicate}
ggplot(molecules_per_lane[molecules_per_lane$stage == "mapped to genome", ],
       aes(x = individual, y = counts_thousands, fill = replicate)) +
  geom_boxplot() +
  scale_y_continuous(breaks = seq(0, 300, 50)) +
  labs(x = "Individual", y = "Number of mapped molecules (thousands)",
       title = "Total number of mapped molecules per lane for each replicate") +
  theme(axis.text.x = element_text(angle = 45, hjust = 0.9, vjust = 0.75))
```

Plot the number of mapped molecules per lane for each lane of each flow cell faceted by bulk versus single cell sequencing.

```{r molecules-per-lane-per-lane}
ggplot(molecules_per_lane[molecules_per_lane$stage == "mapped to genome", ],
       aes(x = flow_cell, y = counts_thousands, fill = lane)) +
  geom_boxplot() +
  scale_y_continuous(breaks = seq(0, 300, 50)) +
  labs(x = "Flow cell", y = "Number of mapped molecules (thousands)",
       title = "Total number of mapped molecules per lane for each lane") +
  theme(axis.text.x = element_text(angle = 45, hjust = 0.9, vjust = 0.75))
```

Plot the number of mapped molecules per lane for each index faceted by bulk versus single cell sequencing.

```{r molecules-per-lane-per-index}
ggplot(molecules_per_lane[molecules_per_lane$stage == "mapped to genome", ],
       aes(x = index, y = counts_thousands)) +
  geom_boxplot() +
  scale_y_continuous(breaks = seq(0, 500, 50)) +
  labs(x = "Index", y = "Number of mapped molecules (thousands)",
       title = "Total number of mapped molecules per lane for each index") +
  theme(axis.text.x = element_blank())
```

## Molecule counts per sample

```{r calc-molecules-sample}
molecules_per_sample <- counts %>% filter(combined == TRUE,
                                          stage %in% c("mapped to genome",
                                                       "mapped to exons"))
molecules_per_sample <- droplevels(molecules_per_sample)
stopifnot(molecules_per_sample$sequences == "molecules",
          table(molecules_per_sample$stage) == 864)
```

Plot the number of molecules per sample at each processing stage.

```{r molecules-per-sample-by-processing-stage}
ggplot(molecules_per_sample,
       aes(x = stage, y = counts_thousands)) +
  geom_boxplot() +
  scale_y_continuous(breaks = seq(0, 400, 50)) +
  labs(x = "Processing stage", y = "Number of molecules (thousands)",
       title = "Total number of molecules per sample at each stage") +
  theme(axis.text.x = element_text(angle = 45, hjust = 0.9, vjust = 0.75))
```

Plot the number of mapped molecules per sample for each replicate of each individual.

```{r molecules-per-sample-per-replicate}
ggplot(molecules_per_sample[molecules_per_sample$stage == "mapped to genome", ],
       aes(x = individual, y = counts_thousands, fill = replicate)) +
  geom_boxplot() +
  scale_y_continuous(breaks = seq(0, 400, 50)) +
  labs(x = "Individual", y = "Number of mapped molecules (thousands)",
       title = "Total number of mapped molecules per sample for each replicate") +
  theme(axis.text.x = element_text(angle = 45, hjust = 0.9, vjust = 0.75))
```

## Summary for single cells

These summary statistics include low quality cells that are removed before downstream analyses.

```{r calculate-seq-depth}
depth_stats <- counts %>%
  filter(stage == "raw", well != "bulk") %>%
  group_by(individual, replicate, well) %>%
  summarize(counts_per_sample = sum(counts)) %>%
  ungroup() %>%
  summarize(mean = mean(counts_per_sample), sd = sd(counts_per_sample),
            min = min(counts_per_sample), max = max(counts_per_sample))
```

We obtained an average of `r format(depth_stats$mean / 10^6, digits = 3)` +/-
`r format(depth_stats$sd / 10^6, digits = 3)` million raw sequencing reads per sample
(range `r format(depth_stats$min / 10^6, digits = 3)`-`r format(depth_stats$max / 10^6, digits = 3)` million reads).

```{r calculate-exons-reads}
exons_reads_stats <- counts %>%
  filter(stage == "mapped to exons", well != "bulk", sequences == "reads") %>%
  group_by(individual, replicate, well) %>%
  summarize(counts_per_sample = sum(counts)) %>%
  ungroup() %>%
  summarize(mean = mean(counts_per_sample), sd = sd(counts_per_sample),
            min = min(counts_per_sample), max = max(counts_per_sample))
```

We obtained an average of `r format(exons_reads_stats$mean / 10^6, digits = 3)` +/-
`r format(exons_reads_stats$sd / 10^6, digits = 3)` million sequencing reads per sample
that mapped to protein-coding exons
(range `r format(exons_reads_stats$min / 10^6, digits = 3)`-`r format(exons_reads_stats$max / 10^6, digits = 3)` million reads).

```{r calculate-exons-molecules}
exons_molecules_stats <- counts %>%
  filter(stage == "mapped to exons", well != "bulk", sequences == "molecules",
         combined) %>%
  group_by(individual, replicate, well) %>%
  summarize(counts_per_sample = sum(counts)) %>%
  ungroup() %>%
  summarize(mean = mean(counts_per_sample), sd = sd(counts_per_sample),
            min = min(counts_per_sample), max = max(counts_per_sample))
```

We obtained an average of `r format(exons_molecules_stats$mean / 10^3, digits = 3)` +/-
`r format(exons_molecules_stats$sd / 10^3, digits = 3)` thousand molecules per sample
that mapped to protein-coding exons
(range `r format(exons_molecules_stats$min / 10^3, digits = 3)`-`r format(exons_molecules_stats$max / 10^3, digits = 3)` thousand molecules).

## Session information

```{r info}
sessionInfo()
```
