---
title: "Tracking down Unassigned_Ambiguity"
author: "John Blischak"
date: 2015-05-28
---

**Last updated:** `r Sys.Date()`

**Code version:** `r system("git log -1 --format='%H'", intern = TRUE)`

```{r chunk-options, include=FALSE}
source("chunk-options.R")
opts_chunk$set(engine = "bash", cache = TRUE)
```

According to the featureCounts [documentation][], Unassigned_Ambiguity is assigned when a read overlaps a feature from more than one meta-feature, i.e. more than one gene.
We have a decent amount of those.
What is causing them?

[documentation]: https://groups.google.com/forum/#!topic/subread/JPPw9lVfgpw

## Setup

Unlike most of the bash code on this site, these code chunks were executable

R adds the bash PATH variable to its list of system variables.
However, it does not include all the locally defined variables in the .bashrc file, so have to manually set this in R.
This allows all the knitr chunks using hte bash engine to run correctly.

```{r set-data-path, engine='R', cache = FALSE}
Sys.setenv(ssd = "/mnt/gluster/data/internal_supp/singleCellSeq")
```

## Diagnosing the problem

I ran featureCounts with the `-R` option (see [SO post][so] for explanation of piping stderr):

[so]: http://stackoverflow.com/a/2342841/2483477

```{r r-option}
featureCounts 2>&1 > /dev/null | grep -w 'R' -A 7
```

Annoyingly, the filename is automatically generated based on the input file and not the output file.

```{r}
grep -w Assigned $ssd/bam-processed/19098.1.A01.ATTAGACG.L002.R1.C6WYKACXX.trim.sickle.sorted.bam.featureCounts | wc -l
```

```{r}
grep -w Unassigned_Ambiguity $ssd/bam-processed/19098.1.A01.ATTAGACG.L002.R1.C6WYKACXX.trim.sickle.sorted.bam.featureCounts | wc -l
```

```{r}
grep -w Unassigned_Ambiguity $ssd/bam-processed/19098.1.A01.ATTAGACG.L002.R1.C6WYKACXX.trim.sickle.sorted.bam.featureCounts | head
```

I searched for the read HWI-700819F:303:C6WYKACXX:2:1107:14113:10344:UMI_GGAAAGGG in the corresponding bam file.

```{r}
samtools view $ssd/bam-processed/19098.1.A01.ATTAGACG.L002.R1.C6WYKACXX.trim.sickle.sorted.bam | grep HWI-700819F:303:C6WYKACXX:2:1107:14113:10344:UMI_GGAAAGGG
```

I Blat'd the sequence "CGTTCACAAGACACAACACGTGAACCTAACAATTCTCTTTTGCTCCAGGAGAGAGAAAAGAAGGAATGGTTTGGGGAGATTCACCGAAATTC" against hg19 and got the same coordinates.

SCORE START END QSIZE IDENTITY CHRO STRAND START END SPAN

92 1 92 92 100.0% 10 + 710206 710297 92

As predicted by featureCounts, it does overlap two protein-coding genes.
It overlaps an exon of [PRR26][] (ENSG00000180525) and an intron of [DIP2C][] (ENSG00000151240).
Is there really an intronic sequence in our exons file?

[PRR26]: http://www.genome.ucsc.edu/cgi-bin/hgc?hgsid=428631025_q1Cp0StMFpE8tmaOjnrKb25eh7Qm&c=chr10&o=696016&t=711109&g=ensGene&i=ENST00000381489
[DIP2C]: http://www.genome.ucsc.edu/cgi-bin/hgc?hgsid=428631025_q1Cp0StMFpE8tmaOjnrKb25eh7Qm&c=chr10&o=320129&t=735606&g=ensGene&i=ENST00000280886

```{r}
# PRR26
grep ENSG00000180525 $ssd/exons.saf | sort | uniq
```

```{r}
# DIP2C
grep ENSG00000151240 $ssd/exons.saf | sort | uniq
```

This is bad.
There is only one unique entry for each of these proteins even though they have multiple exons.
Sadly, all genes only have one entry!

```{r}
# Structure of SAF file
head $ssd/exons.saf
```

```{r}
# Total number of exons
wc -l $ssd/exons.saf
```

```{r}
# Total number of unique exons
sort $ssd/exons.saf | uniq | wc -l
```

```{r}
# Total number of unique genes
cut -f1 $ssd/exons.saf | sort | uniq | wc -l
```

There must be a bug in the script to create the exons file ([link to version 55e1420][exons])

[exons]: https://github.com/jdblischak/singleCellSeq/blob/55e1420c73df6add57d61a80f7c69ced21f322bc/code/create_exons.R

## Fixing bug

```{r biomaRt, engine='R'}
library("biomaRt")
ensembl <- useMart(host = "grch37.ensembl.org",
                   biomart = "ENSEMBL_MART_ENSEMBL",
                   dataset = "hsapiens_gene_ensembl")
```

The fix is straightforward.
I was using "start_position" and "end_position" from "feature_page", but these correspond to the start and stop positions for the entire gene.
So each exons receives the same coordinates.

```{r feature-page, engine='R'}
attributePages(ensembl)
atts <- listAttributes(ensembl, page = "feature_page")
atts[grep("start", atts$description, ignore.case = TRUE), ]
```

I needed to be using "exon_chrom_start" and "exon_chrom_end" from the "structure" page.

```{r structure, engine='R'}
atts_struct <- listAttributes(ensembl, page = "structure")
atts_struct[grep("start", atts_struct$description, ignore.case = TRUE), ]
```

Updating this fixed the issue.

Also, I do not need to worry about the code from my last project because I had used a different strategy.
My function [create_exon_data][] used `makeTranscriptDbFromBiomart` and `exonsBy` from the GenomicFeatures package.

[create_exon_data]: https://bitbucket.org/jdblischak/ngspipeline/src/5d7360a65ea447e911b292faced38057b3fdda85/R/setup.R?at=master#cl-216

## Session information

```{r featureCounts-version}
featureCounts -v
```

```{r info, engine='R'}
sessionInfo()
```
