---
title: "variance"
author: "PoYuan Tung"
date: 2015-06-15
---

**Last updated:** `r Sys.Date()`

**Code version:** `r system("git log -1 --format='%H'", intern = TRUE)`

```{r chunk-options, include=FALSE}
source("chunk-options.R")
```

## Input

```{r packages, message=FALSE}
library("dplyr")
library("ggplot2")
theme_set(theme_bw(base_size = 16))
library("edgeR")
library("gplots")
```

Input annotation.

```{r input-annotation}
anno <- read.table("../data/annotation.txt", header = TRUE,
                   stringsAsFactors = FALSE)
head(anno)
```

Input read counts.

```{r input-read-counts}
reads <- read.table("../data/reads.txt", header = TRUE,
                    stringsAsFactors = FALSE)
```

Input molecule counts.

```{r input-molecule-counts}
molecules <- read.table("../data/molecules.txt", header = TRUE,
                    stringsAsFactors = FALSE)
```

Input list of quality single cells.

```{r input-quality-single-cells}
quality_single_cells <- scan("../data/quality-single-cells.txt",
                             what = "character")
```

Keep only the single cells that passed the [QC filters](qc-cell-ipsc.html) and the bulk samples.

```{r qc-filter}
molecules <- molecules[, grepl("bulk", colnames(molecules)) |
                         colnames(molecules) %in% quality_single_cells]
anno <- anno[anno$well == "bulk" | anno$sample_id %in% quality_single_cells, ]
stopifnot(ncol(molecules) == nrow(anno),
          colnames(molecules) == anno$sample_id)

reads <- reads[, grepl("bulk", colnames(reads)) |
                         colnames(reads) %in% quality_single_cells]
stopifnot(ncol(reads) == nrow(anno),
          colnames(reads) == anno$sample_id)
```

Remove genes with zero read counts in the single cells or bulk samples.

```{r remove-non-expressed-genes}
expressed <- rowSums(molecules[, anno$well == "bulk"]) > 0 &
             rowSums(molecules[, anno$well != "bulk"]) > 0
molecules <- molecules[expressed, ]
dim(molecules)

expressed <- rowSums(reads[, anno$well == "bulk"]) > 0 &
             rowSums(reads[, anno$well != "bulk"]) > 0
reads <- reads[expressed, ]
dim(reads)
```

Split the bulk and single samples.

```{r split-bulk-single}
molecules_bulk <- molecules[, anno$well == "bulk"]
molecules_single <- molecules[, anno$well != "bulk"]
reads_bulk <- reads[, anno$well == "bulk"]
reads_single <- reads[, anno$well != "bulk"]
```

Remove genes with max molecule numer larger than 1024
```{r remove-1024}
molecules_single <- molecules_single[apply(molecules_single,1,max) < 1024,]
```

## variance between batches (C1 prep) by molecules

See if batches from the same individual cluster together
```{r batch-variance}
# create table using the molecule counts
group_list <- c("19","19098","19101","19239","19098.1","19098.2","19098.3","19101.1","19101.2","19101.3","19239.1","19239.2","19239.3")

### create a function to generate tables of mean,
create_info_table <- function(statistics="mean"){
  # first correct
  molecules.crt <- -1024*log(1-molecules_single/1024)
  # loop all possible combinations
  big_table <- do.call(cbind,lapply(group_list,function(x){
      # subset data
      data_ref <- molecules.crt[grep(x,names(molecules.crt))]
      if(statistics=="mean"){
        ans <- apply(data_ref,1,function(xx) mean(xx,na.rm=TRUE))  
      }
      if(statistics=="var"){
        ans <- apply(data_ref,1,function(xx) var(xx,na.rm=TRUE))  
      }
      if(statistics=="CV"){
        ans <- apply(data_ref,1,function(xx)  sd(xx,na.rm=TRUE)) / apply(data_ref,1,function(xx)  mean(xx,na.rm=TRUE))   
      }
    ans
    })
  )
    big_table <- data.frame(big_table)
    names(big_table) <- paste(statistics,group_list,sep="_")
    big_table$gene_name <- rownames(molecules_single)
    big_table
}

# create big table for mean, var and cv
table_mean <- create_info_table(statistics="mean")
table_var <- create_info_table(statistics="var")
table_cv <- create_info_table(statistics="CV")

# replace 0 as NA
table_mean[table_mean==0] <- NA
table_var[table_mean==0] <- NA
table_cv[table_mean==0] <- NA

# calculate the correlation
cor(table_mean[,2:13],use="pairwise.complete.obs")
cor(table_var[,2:13],use="pairwise.complete.obs")
cor(table_cv[,2:13],use="pairwise.complete.obs")

heatmap.2(cor(table_mean[,2:13],use="pairwise.complete.obs"), trace="none",cexRow=1,cexCol=1,margins=c(8,8))
heatmap.2(cor(table_var[,2:13],use="pairwise.complete.obs"), trace="none",cexRow=1,cexCol=1,margins=c(8,8))
heatmap.2(cor(table_cv[,2:13],use="pairwise.complete.obs"), trace="none",cexRow=1,cexCol=1,margins=c(8,8))
```

Look at the correlation of CV from batches 
```{r CV}
# create a table for boxplot of CV correlation 
corr_CV <- cor(table_cv[,2:13],use="pairwise.complete.obs")

corr_boxplot <- data.frame(
correlation=c(corr_CV[4,5:6],corr_CV[5,6],corr_CV[7,8:9],corr_CV[8,9],corr_CV[7,8:9],corr_CV[11,12],corr_CV[4:6,7:12],corr_CV[7:9,10:12]), cor_source=c(rep("within",9),rep("between",27)), individual=c(rep("19098",3),rep("19101",3),rep("19239",3),rep("19098.19101",9),rep("19098.19239",9),rep("19101.19239",9)))

# t test
t_test <- t.test(corr_boxplot[1:6,1],corr_boxplot[7:33,1],alternative = "greater")
t_test

ggplot(corr_boxplot, aes(cor_source,correlation)) + geom_boxplot() + geom_jitter(aes(colour = individual, size = 2, width = 0.5)) 

# create a table for boxplot of CV correlation without 19098.2
corr_boxplot_no <- data.frame(
correlation=c(corr_CV[4,6],corr_CV[7,8:9],corr_CV[8,9],corr_CV[7,8:9],corr_CV[11,12],corr_CV[4,7:12],corr_CV[6,7:12],corr_CV[7:9,10:12]), cor_source=c(rep("within",7),rep("between",21)), individual=c(rep("19098",1),rep("19101",3),rep("19239",3),rep("19098.19101",3),rep("19098.19239",3),rep("19098.19101",3),rep("19098.19239",3), rep("19101.19239",9)))

# t test
t_test <- t.test(corr_boxplot_no[1:6,1],corr_boxplot_no[7:33,1],alternative = "greater")
t_test

ggplot(corr_boxplot_no, aes(cor_source,correlation)) + geom_boxplot() + geom_jitter(aes(colour = individual, size = 2, width = 0.5))
```

## variance of gene expression between individuls

Calculate the variance betwwen individuls from the bulk samples
```{r variance-bulk}
# normalization
reads_bulk_cpm <- cpm(reads_bulk)

# create a new dataset
reads_var <- data.frame(reads_bulk_cpm)
sum(reads_var!=reads_bulk_cpm)

# add mean of each individauls
reads_var$mean19098 <- apply(reads_var[,grep("NA19098",names(reads_var))], 1, mean)
reads_var$mean19101 <- apply(reads_var[,grep("NA19101",names(reads_var))], 1, mean)
reads_var$mean19239 <- apply(reads_var[,grep("NA19239",names(reads_var))], 1, mean)

# add variance of bulk means
reads_var$bulk_variance <- apply(reads_var[,c("mean19098","mean19101","mean19239")],1,var)

```

Calculate the variance between individuals using the means from single cells

```{r variance-single-mean}
# normalization
reads_single_cpm <- data.frame(cpm(reads_single))

# remove the ERCC of 19098.2 
remove <-function(xxx){
  ## identify 19098.2
     sample_name <- names(reads_single_cpm)
     targeremove()mn <- sample_name[grep("19098.2",sample_name)]
  ## find out ERCC rows
		 g <- rownames(reads_single_cpm)
		 target.row <- g[grep("ERCC",g)]
	## replace the molecules numbers with NA 
		 reads_single_cpm[target.row,target.column] <- NA}
		
# means of single cells within individuals
reads_var$mean.single19098 <- apply(reads_single_cpm[,grep("NA19098",names(reads_single_cpm))], 1, mean, na.rm = TRUE)
reads_var$mean.single19101 <- apply(reads_single_cpm[,grep("NA19101",names(reads_single_cpm))], 1, mean, na.rm = TRUE)
reads_var$mean.single19239 <- apply(reads_single_cpm[,grep("NA19239",names(reads_single_cpm))], 1, mean, na.rm = TRUE)


## variance of means from single cells
reads_var$mean_single_variance <- apply(reads_var[,c("mean.single19098","mean.single19101","mean.single19239")],1,var)

# sellect ERCC
reads_var$ERCC <- grepl("ERCC",rownames(reads_var))

# plot with color-blind-friendly palettes
cbPalette <- c("#999999", "#0000FF", "#990033", "#F0E442", "#0072B2", "#D55E00", "#CC79A7", "#009E73")

ggplot(reads_var, aes(x = bulk_variance, y = mean_single_variance, col = ERCC)) + geom_point(size = 2, alpha = 0.5) + scale_colour_manual(values=cbPalette) + scale_x_log10() + scale_y_log10() + stat_function(fun= function(x) {x}, col= "#56B4E9")
```

Calculate the variance between and within individuals using the single cell data
```{r variance-single}
## variance within individual 
variance.single19098 <- apply(reads_single_cpm[,grep("NA19098",names(reads_single_cpm))], 1, var, na.rm = TRUE)
variance.single19101 <- apply(reads_single_cpm[,grep("NA19101",names(reads_single_cpm))], 1, var, na.rm = TRUE)
variance.single19239 <- apply(reads_single_cpm[,grep("NA19239",names(reads_single_cpm))], 1, var, na.rm = TRUE)

# number of cell
number.of.cell.all   <- sum(grepl("19",sample_name))
number.of.cell.19098 <- sum(grepl("19098",sample_name))
number.of.cell.19101 <- sum(grepl("19101",sample_name))
number.of.cell.19239 <- sum(grepl("19239",sample_name))

# total within individual variance 
reads_var$var_within_individual<-  
 (variance.single19098   *(number.of.cell.19098)  +      
  variance.single19101   *(number.of.cell.19101)  + 
  variance.single19239   *(number.of.cell.19239)  ) / 
 (number.of.cell.all)

ggplot(reads_var, aes(x = bulk_variance, y = var_within_individual, col = ERCC)) + geom_point(size = 2, alpha = 0.5) + scale_colour_manual(values=cbPalette) + scale_x_log10() + scale_y_log10() + stat_function(fun= function(x) {x}, col= "#56B4E9")

## variance between all single cells
var_all_single <- apply(reads_single_cpm, 1, var, na.rm = TRUE)
reads_var$var_all_single <- var_all_single

## keep non-missing across the table
reads_var <- reads_var[apply(reads_var,1,function(x)  sum(is.na(x)))==0,]

## variance between individauls
reads_var$var_between_individual<-( 
  var_all_single         *(number.of.cell.all-1)-
  variance.single19098   *(number.of.cell.19098-1)  -      
  variance.single19101   *(number.of.cell.19101-1)  - 
  variance.single19239   *(number.of.cell.19239-1)  ) / 
 (number.of.cell.all-1)

## the variaance contributed by between individual
reads_var$ratio_var_between_individual <- reads_var[,"var_between_individual"]/var_all_single

ggplot(reads_var, aes(x = var_all_single, y = ratio_var_between_individual, col = ERCC)) + geom_point(size = 2, alpha = 0.5) + scale_colour_manual(values=cbPalette) + scale_x_log10()

ggplot(reads_var, aes(x = var_all_single, y = var_between_individual, col = ERCC)) + geom_point(size = 2, alpha = 0.5) + scale_colour_manual(values=cbPalette) + scale_x_log10() + scale_y_log10() + stat_function(fun= function(x) {x}, col= "#56B4E9")

ggplot(reads_var, aes(x = bulk_variance, y = var_between_individual, col = ERCC)) + geom_point(size = 2, alpha = 0.5) + scale_colour_manual(values=cbPalette) + scale_x_log10() + scale_y_log10() + stat_function(fun= function(x) {x}, col= "#56B4E9")

```

## AVONA F-statistics

Pull the p-value
```{r p_of_f}
### create a function for f.test by anova
### compare the to fits:
### 1. lm from all single cells
### 2. lm from each individaul
f.test <- function(data.in){
  tt <- names(data.in)
  individual.id <- rep("19098",length(tt))
  individual.id[grep("19101",tt)] <- "19101"
  individual.id[grep("19239",tt)] <- "19239"
  
  dd <- data.frame(reads=unlist(data.in),individual.id=individual.id)
  fit1 <- lm(reads~1,data=dd)
  fit2 <- lm(reads~1 + individual.id,data=dd)
  anova(fit1,fit2)[2,"Pr(>F)"]
}

# creat the f test table
f.test.table <- do.call(rbind,lapply(rownames(reads_single_cpm),function(x){
  data.frame(gene_name=x,p_of_f=f.test(reads_single_cpm[x,]))  
}))

# sellect ERCC
f.test.table$ERCC <- grepl("ERCC",f.test.table[,1])

# sort 
f.test.table.sort <- f.test.table[order(f.test.table[,2]),]
head(f.test.table.sort)

plot(f.test.table.sort[,2], log = "y",col=as.numeric(f.test.table.sort$ERCC+1))

plot(f.test.table.sort[1:5000,2], log = "y",col=as.numeric(f.test.table.sort$ERCC[1:5000]+1))

plot(f.test.table.sort[1:5000,2],col=as.numeric(f.test.table.sort$ERCC[1:5000]+1),pch=20,cex=.3)

# plot the variance and show p 0f f
reads_var$p_of_f <- f.test.table$p_of_f

ggplot(reads_var, aes(x = bulk_variance, y = mean_single_variance, shape = as.factor(ERCC), col = p_of_f)) + geom_point(size = 2, alpha = 0.25) + scale_x_log10() + scale_y_log10() + scale_colour_gradient2(midpoint= 0.5, space="Lab") + stat_function(fun= function(x) {x}, col= "#56B4E9")

ggplot(reads_var, aes(x = bulk_variance, y = var_within_individual, shape = as.factor(ERCC), col = p_of_f)) + geom_point(size = 2, alpha = 0.5) + scale_x_log10() + scale_y_log10() + scale_colour_gradient2(midpoint= 0.5, space="Lab") + stat_function(fun= function(x) {x}, col= "#56B4E9")

ggplot(reads_var, aes(x = bulk_variance, y = var_between_individual, shape = as.factor(ERCC), col = p_of_f)) + geom_point(size = 2, alpha = 0.5) + scale_x_log10() + scale_y_log10() + scale_colour_gradient2(midpoint= 0.5, space="Lab") + stat_function(fun= function(x) {x}, col= "#56B4E9")

ggplot(reads_var, aes(x = var_all_single, y = var_between_individual, shape = as.factor(ERCC), col = p_of_f)) + geom_point(size = 2, alpha = 0.5) + scale_x_log10() + scale_y_log10() + scale_colour_gradient2(midpoint= 0.5, space="Lab") + stat_function(fun= function(x) {x}, col= "#56B4E9")
```

Pull the F value
```{r Fvalue}
# looking at the F value
f.test.F <- function(data.in){
  tt <- names(data.in)
  individual.id <- rep("19098",length(tt))
  individual.id[grep("19101",tt)] <- "19101"
  individual.id[grep("19239",tt)] <- "19239"
  
  dd <- data.frame(reads=unlist(data.in),individual.id=individual.id)
  fit1 <- lm(reads~1,data=dd)
  fit2 <- lm(reads~1 + individual.id,data=dd)
  anova(fit1,fit2)[2,"F"]
}
# creat the f test table of F value
f.test.F.table <- do.call(rbind,lapply(rownames(reads_single_cpm),function(x){
  data.frame(gene_name=x,F_value=f.test.F(reads_single_cpm[x,]))  
}))

# sellect ERCC
f.test.F.table$ERCC <- grepl("ERCC",f.test.F.table[,1])

# plot the variance and show F value
reads_var$F_value <- f.test.F.table$F_value

# calculate F value from p
# when p=0.01
qf(1-0.05,2,629)

# create color index
reads_var$F_color <- "1 < F < 3"
reads_var$F_color[reads_var$F_value >=3] <- "F >= 3"
reads_var$F_color[reads_var$F_value <=1] <- "F <= 1"

# use the F value from p to scale the gradient
ggplot(reads_var, aes(x = bulk_variance, y = mean_single_variance, shape = as.factor(ERCC), col = F_color)) + scale_colour_manual(values=cbPalette) + geom_point(size = 2, alpha = 0.5) + scale_x_log10() + scale_y_log10() + stat_function(fun= function(x) {x}, col= "#56B4E9")

ggplot(reads_var, aes(x = bulk_variance, y = var_within_individual, shape = as.factor(ERCC), col = F_color)) + scale_colour_manual(values=cbPalette) + geom_point(size = 2, alpha = 0.5) + scale_x_log10() + scale_y_log10() + stat_function(fun= function(x) {x}, col= "#56B4E9")

ggplot(reads_var, aes(x = bulk_variance, y = var_between_individual, shape = as.factor(ERCC), col = F_color)) + scale_colour_manual(values=cbPalette) + geom_point(size = 2, alpha = 0.5) + scale_x_log10() + scale_y_log10() + stat_function(fun= function(x) {x}, col= "#56B4E9")

ggplot(reads_var, aes(x = var_all_single, y = var_within_individual, shape = as.factor(ERCC), col = F_color)) + scale_colour_manual(values=cbPalette) + geom_point(size = 2, alpha = 0.5) + scale_x_log10() + scale_y_log10() + stat_function(fun= function(x) {x}, col= "#56B4E9")

ggplot(reads_var, aes(x = var_all_single, y = var_between_individual, shape = as.factor(ERCC), col = F_color)) + scale_colour_manual(values=cbPalette) + geom_point(size = 2, alpha = 0.2) + scale_x_log10() + scale_y_log10() + stat_function(fun= function(x) {x}, col= "#56B4E9")

```

## Session information

```{r info}
sessionInfo()
```
