<!DOCTYPE html>

<html xmlns="http://www.w3.org/1999/xhtml">

<head>

<meta charset="utf-8">
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="generator" content="pandoc" />

<meta name="author" content="PoYuan Tung" />

<meta name="date" content="2015-06-15" />

<title>variance</title>

<script src="libs/jquery-1.11.0/jquery.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="libs/bootstrap-3.3.1/css/united.min.css" rel="stylesheet" />
<script src="libs/bootstrap-3.3.1/js/bootstrap.min.js"></script>
<script src="libs/bootstrap-3.3.1/shim/html5shiv.min.js"></script>
<script src="libs/bootstrap-3.3.1/shim/respond.min.js"></script>

<style type="text/css">

/* padding for bootstrap navbar */
body {
  padding-top: 50px;
  padding-bottom: 40px;
}


/* offset scroll position for anchor links (for fixed navbar)  */
.section h2 {
  padding-top: 55px;
  margin-top: -55px;
}
.section h3 {
  padding-top: 55px;
  margin-top: -55px;
}



/* don't use link color in navbar */
.dropdown-menu>li>a {
  color: black;
}

/* some padding for disqus */
#disqus_thread {
  margin-top: 45px;
}

</style>

<link rel="stylesheet" href="libs/font-awesome-4.1.0/css/font-awesome.min.css"/>

<style type="text/css">code{white-space: pre;}</style>
<link rel="stylesheet"
      href="libs/highlight/textmate.css"
      type="text/css" />
<script src="libs/highlight/highlight.js"></script>
<style type="text/css">
  pre:not([class]) {
    background-color: white;
  }
</style>
<script type="text/javascript">
if (window.hljs && document.readyState && document.readyState === "complete") {
   window.setTimeout(function() {
      hljs.initHighlighting();
   }, 0);
}
</script>



</head>

<body>

<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
code {
  color: inherit;
  background-color: rgba(0, 0, 0, 0.04);
}
img { 
  max-width:100%; 
  height: auto; 
}
</style>
<div class="container-fluid main-container">


<div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">singleCellSeq</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li><a href="index.html">Home</a></li>
        <li><a href="about.html">About</a></li>
        <li><a href="license.html">License</a></li>
        <li><a href="https://github.com/jdblischak/singleCellSeq">GitHub</a></li>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<div id="header">
<h1 class="title">variance</h1>
<h4 class="author"><em>PoYuan Tung</em></h4>
<h4 class="date"><em>2015-06-15</em></h4>
</div>

<div id="TOC">
<ul>
<li><a href="#input">Input</a></li>
<li><a href="#remove-bad-quality-cells">Remove bad quality cells</a></li>
<li><a href="#variance-of-gene-expression-between-individuls">variance of gene expression between individuls</a></li>
<li><a href="#avona-f-statistics">AVONA F-statistics</a></li>
<li><a href="#session-information">Session information</a></li>
</ul>
</div>

<p><strong>Last updated:</strong> 2015-06-20</p>
<p><strong>Code version:</strong> 870a709fb088f0287b899424ce69db87cb64e477</p>
<div id="input" class="section level2">
<h2>Input</h2>
<pre class="r"><code>library(&quot;dplyr&quot;)
library(&quot;ggplot2&quot;)
theme_set(theme_bw(base_size = 16))
library(&quot;edgeR&quot;)</code></pre>
<p>Summary counts from featureCounts. Created with <a href="https://github.com/jdblischak/singleCellSeq/blob/master/code/gather-summary-counts.py">gather-summary-counts.py</a>.</p>
<pre class="r"><code>summary_counts &lt;- read.table(&quot;../data/summary-counts.txt&quot;, header = TRUE,
                             stringsAsFactors = FALSE)
summary_per_sample &lt;- summary_counts %&gt;%
  filter(sickle == &quot;quality-trimmed&quot;) %&gt;%
  select(-sickle) %&gt;%
  arrange(individual, batch, well, rmdup) %&gt;%
  as.data.frame</code></pre>
<p>Input annotation.</p>
<pre class="r"><code>anno &lt;- read.table(&quot;../data/annotation.txt&quot;, header = TRUE,
                   stringsAsFactors = FALSE)
head(anno)</code></pre>
<pre><code>  individual batch well     sample_id
1      19098     1  A01 NA19098.1.A01
2      19098     1  A02 NA19098.1.A02
3      19098     1  A03 NA19098.1.A03
4      19098     1  A04 NA19098.1.A04
5      19098     1  A05 NA19098.1.A05
6      19098     1  A06 NA19098.1.A06</code></pre>
<p>Input read counts.</p>
<pre class="r"><code>reads &lt;- read.table(&quot;../data/reads.txt&quot;, header = TRUE,
                    stringsAsFactors = FALSE)</code></pre>
<p>Input molecule counts.</p>
<pre class="r"><code>molecules &lt;- read.table(&quot;../data/molecules.txt&quot;, header = TRUE,
                    stringsAsFactors = FALSE)</code></pre>
<p>Input single cell observational quality control data.</p>
<pre class="r"><code>qc &lt;- read.table(&quot;../data/qc-ipsc.txt&quot;, header = TRUE,
                 stringsAsFactors = FALSE)
head(qc)</code></pre>
<pre><code>  individual batch well cell_number concentration tra1.60
1      19098     1  A01           1      1.734785       1
2      19098     1  A02           1      1.723038       1
3      19098     1  A03           1      1.512786       1
4      19098     1  A04           1      1.347492       1
5      19098     1  A05           1      2.313047       1
6      19098     1  A06           1      2.056803       1</code></pre>
</div>
<div id="remove-bad-quality-cells" class="section level2">
<h2>Remove bad quality cells</h2>
<div id="remove-cells-with-total-reads-2-millons" class="section level3">
<h3>Remove cells with total reads &lt; 2 millons</h3>
<pre class="r"><code>#reads per sample
summary_per_sample_reads &lt;- summary_per_sample %&gt;% filter(rmdup == &quot;reads&quot;)

#create unmapped ratios
summary_per_sample_reads$unmapped.ratios &lt;- summary_per_sample_reads[,9]/apply(summary_per_sample_reads[,5:13],1,sum)

#create total mapped reads
summary_per_sample_reads$total.mapped &lt;- apply(summary_per_sample_reads[,5:8],1,sum)

#creat ERCC ratios
summary_per_sample_reads$ERCC.ratios &lt;- apply(reads[grep(&quot;ERCC&quot;, rownames(reads)), ],2,sum)/apply(summary_per_sample_reads[,5:8],1,sum)

#remove bulk keep single cell
summary_per_sample_reads_single &lt;- summary_per_sample_reads[summary_per_sample_reads$well!=&quot;bulk&quot;,]

#add cell number per well by merging qc file
summary_per_sample_reads_single_qc &lt;- merge(summary_per_sample_reads_single,qc,by=c(&quot;individual&quot;,&quot;batch&quot;,&quot;well&quot;))

#qc filter
summary_per_sample_reads_single_qc$qc_filter &lt;- summary_per_sample_reads_single_qc$cell_number == 1 &amp; summary_per_sample_reads_single_qc$total.mapped &gt; 2 * 10^6

sum(summary_per_sample_reads_single_qc$qc_filter)</code></pre>
<pre><code>[1] 632</code></pre>
</div>
</div>
<div id="variance-of-gene-expression-between-individuls" class="section level2">
<h2>variance of gene expression between individuls</h2>
<p>Calculate the variance betwwen individuls from the bulk samples</p>
<pre class="r"><code># select bulk samples
reads_bulk&lt;- reads %&gt;% select(contains(&quot;bulk&quot;))

# normalization
reads_bulk_cpm &lt;- cpm(reads_bulk)

# create a new dataset
reads_var &lt;- data.frame(reads_bulk_cpm)
sum(reads_var!=reads_bulk_cpm)</code></pre>
<pre><code>[1] 0</code></pre>
<pre class="r"><code># add mean of each individauls
reads_var$mean19098 &lt;- apply(reads_var[,grep(&quot;NA19098&quot;,names(reads_var))], 1, mean)
reads_var$mean19101 &lt;- apply(reads_var[,grep(&quot;NA19101&quot;,names(reads_var))], 1, mean)
reads_var$mean19239 &lt;- apply(reads_var[,grep(&quot;NA19239&quot;,names(reads_var))], 1, mean)

# add variance of bulk means
reads_var$bulk_variance &lt;- apply(reads_var[,c(&quot;mean19098&quot;,&quot;mean19101&quot;,&quot;mean19239&quot;)],1,var)</code></pre>
<p>Calculate the variance between individuals using the means from single cells</p>
<pre class="r"><code># select single cell samples
reads_single &lt;- reads %&gt;% select(-contains(&quot;bulk&quot;))

# apply qc_filter
reads_single_qc &lt;- reads_single[,summary_per_sample_reads_single_qc$qc_filter]
dim(reads_single_qc)</code></pre>
<pre><code>[1] 20419   632</code></pre>
<pre class="r"><code># normalization
reads_single_cpm &lt;- data.frame(cpm(reads_single_qc))

# remove the ERCC of 19098.2 
  ## identify 19098.2
     sample_name &lt;- names(reads_single_cpm)
     target.column &lt;- sample_name[grep(&quot;19098.2&quot;,sample_name)]
  ## find out ERCC rows
         g &lt;- rownames(reads_single_cpm)
         target.row &lt;- g[grep(&quot;ERCC&quot;,g)]
    ## replace the molecules numbers with NA 
         reads_single_cpm[target.row,target.column] &lt;- NA
        
# means of single cells within individuals
reads_var$mean.single19098 &lt;- apply(reads_single_cpm[,grep(&quot;NA19098&quot;,names(reads_single_cpm))], 1, mean, na.rm = TRUE)
reads_var$mean.single19101 &lt;- apply(reads_single_cpm[,grep(&quot;NA19101&quot;,names(reads_single_cpm))], 1, mean, na.rm = TRUE)
reads_var$mean.single19239 &lt;- apply(reads_single_cpm[,grep(&quot;NA19239&quot;,names(reads_single_cpm))], 1, mean, na.rm = TRUE)


## variance of means from single cells
reads_var$mean_single_variance &lt;- apply(reads_var[,c(&quot;mean.single19098&quot;,&quot;mean.single19101&quot;,&quot;mean.single19239&quot;)],1,var)

# sellect ERCC
reads_var$ERCC &lt;- grepl(&quot;ERCC&quot;,rownames(reads_var))

# plot with color-blind-friendly palettes
cbPalette &lt;- c(&quot;#999999&quot;, &quot;#0000FF&quot;, &quot;#990033&quot;, &quot;#F0E442&quot;, &quot;#0072B2&quot;, &quot;#D55E00&quot;, &quot;#CC79A7&quot;, &quot;#009E73&quot;)

ggplot(reads_var, aes(x = bulk_variance, y = mean_single_variance, col = ERCC)) + geom_point(size = 2, alpha = 0.5) + scale_colour_manual(values=cbPalette) + scale_x_log10() + scale_y_log10() + stat_function(fun= function(x) {x}, col= &quot;#56B4E9&quot;)</code></pre>
<p><img src="figure/variance.Rmd/variance-single-mean-1.png" title="" alt="" width="672" style="display: block; margin: auto;" /></p>
<p>Calculate the variance between and within individuals using the single cell data</p>
<pre class="r"><code>## variance within individual 
variance.single19098 &lt;- apply(reads_single_cpm[,grep(&quot;NA19098&quot;,names(reads_single_cpm))], 1, var, na.rm = TRUE)
variance.single19101 &lt;- apply(reads_single_cpm[,grep(&quot;NA19101&quot;,names(reads_single_cpm))], 1, var, na.rm = TRUE)
variance.single19239 &lt;- apply(reads_single_cpm[,grep(&quot;NA19239&quot;,names(reads_single_cpm))], 1, var, na.rm = TRUE)

# number of cell
number.of.cell.all   &lt;- sum(grepl(&quot;19&quot;,sample_name))
number.of.cell.19098 &lt;- sum(grepl(&quot;19098&quot;,sample_name))
number.of.cell.19101 &lt;- sum(grepl(&quot;19101&quot;,sample_name))
number.of.cell.19239 &lt;- sum(grepl(&quot;19239&quot;,sample_name))

# total within individual variance 
reads_var$var_within_individual&lt;-  
 (variance.single19098   *(number.of.cell.19098)  +      
  variance.single19101   *(number.of.cell.19101)  + 
  variance.single19239   *(number.of.cell.19239)  ) / 
 (number.of.cell.all)

ggplot(reads_var, aes(x = bulk_variance, y = var_within_individual, col = ERCC)) + geom_point(size = 2, alpha = 0.5) + scale_colour_manual(values=cbPalette) + scale_x_log10() + scale_y_log10() + stat_function(fun= function(x) {x}, col= &quot;#56B4E9&quot;)</code></pre>
<p><img src="figure/variance.Rmd/variance-single-1.png" title="" alt="" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>## variance between all single cells
var_all_single &lt;- apply(reads_single_cpm, 1, var, na.rm = TRUE)
reads_var$var_all_single &lt;- var_all_single

## variance between individauls
reads_var$var_between_individual&lt;-( 
  var_all_single         *(number.of.cell.all-1)-
  variance.single19098   *(number.of.cell.19098-1)  -      
  variance.single19101   *(number.of.cell.19101-1)  - 
  variance.single19239   *(number.of.cell.19239-1)  ) / 
 (number.of.cell.all-1)

## the variaance contributed by between individual
reads_var$ratio_var_between_individual &lt;- reads_var[,&quot;var_between_individual&quot;]/var_all_single

ggplot(reads_var, aes(x = var_all_single, y = ratio_var_between_individual, col = ERCC)) + geom_point(size = 2, alpha = 0.5) + scale_colour_manual(values=cbPalette) + scale_x_log10()</code></pre>
<pre><code>Warning in loop_apply(n, do.ply): Removed 2812 rows containing missing
values (geom_point).</code></pre>
<p><img src="figure/variance.Rmd/variance-single-2.png" title="" alt="" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>ggplot(reads_var, aes(x = var_all_single, y = var_between_individual, col = ERCC)) + geom_point(size = 2, alpha = 0.5) + scale_colour_manual(values=cbPalette) + scale_x_log10() + scale_y_log10() + stat_function(fun= function(x) {x}, col= &quot;#56B4E9&quot;)</code></pre>
<pre><code>Warning in scale$trans$trans(x): NaNs produced</code></pre>
<pre><code>Warning in scale$trans$trans(x): NaNs produced</code></pre>
<pre><code>Warning in loop_apply(n, do.ply): Removed 28 rows containing missing values
(geom_point).</code></pre>
<p><img src="figure/variance.Rmd/variance-single-3.png" title="" alt="" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>ggplot(reads_var, aes(x = bulk_variance, y = var_between_individual, col = ERCC)) + geom_point(size = 2, alpha = 0.5) + scale_colour_manual(values=cbPalette) + scale_x_log10() + scale_y_log10() + stat_function(fun= function(x) {x}, col= &quot;#56B4E9&quot;)</code></pre>
<pre><code>Warning in scale$trans$trans(x): NaNs produced</code></pre>
<pre><code>Warning in scale$trans$trans(x): NaNs produced</code></pre>
<pre><code>Warning in loop_apply(n, do.ply): Removed 28 rows containing missing values
(geom_point).</code></pre>
<p><img src="figure/variance.Rmd/variance-single-4.png" title="" alt="" width="672" style="display: block; margin: auto;" /></p>
</div>
<div id="avona-f-statistics" class="section level2">
<h2>AVONA F-statistics</h2>
<p>Pull the p-value</p>
<pre class="r"><code>### create a function for f.test by anova
### compare the to fits:
### 1. lm from all single cells
### 2. lm from each individaul
f.test &lt;- function(data.in){
  tt &lt;- names(data.in)
  individual.id &lt;- rep(&quot;19098&quot;,length(tt))
  individual.id[grep(&quot;19101&quot;,tt)] &lt;- &quot;19101&quot;
  individual.id[grep(&quot;19239&quot;,tt)] &lt;- &quot;19239&quot;
  
  dd &lt;- data.frame(reads=unlist(data.in),individual.id=individual.id)
  fit1 &lt;- lm(reads~1,data=dd)
  fit2 &lt;- lm(reads~1 + individual.id,data=dd)
  anova(fit1,fit2)[2,&quot;Pr(&gt;F)&quot;]
}

# creat the f test table
f.test.table &lt;- do.call(rbind,lapply(rownames(reads_single_cpm),function(x){
  data.frame(gene_name=x,p_of_f=f.test(reads_single_cpm[x,]))  
}))

# sellect ERCC
f.test.table$ERCC &lt;- grepl(&quot;ERCC&quot;,f.test.table[,1])

# sort 
f.test.table.sort &lt;- f.test.table[order(f.test.table[,2]),]
head(f.test.table.sort)</code></pre>
<pre><code>            gene_name        p_of_f  ERCC
17307 ENSG00000106153 1.096914e-139 FALSE
8060  ENSG00000256618 5.541342e-115 FALSE
6079  ENSG00000183648 6.039660e-104 FALSE
10536 ENSG00000022556 1.795330e-102 FALSE
16832 ENSG00000112306  8.905248e-96 FALSE
19106 ENSG00000176386  2.422581e-88 FALSE</code></pre>
<pre class="r"><code>plot(f.test.table.sort[,2], log = &quot;y&quot;,col=as.numeric(f.test.table.sort$ERCC+1))</code></pre>
<p><img src="figure/variance.Rmd/p_of_f-1.png" title="" alt="" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>plot(f.test.table.sort[1:5000,2], log = &quot;y&quot;,col=as.numeric(f.test.table.sort$ERCC[1:5000]+1))</code></pre>
<p><img src="figure/variance.Rmd/p_of_f-2.png" title="" alt="" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>plot(f.test.table.sort[1:5000,2],col=as.numeric(f.test.table.sort$ERCC[1:5000]+1),pch=20,cex=.3)</code></pre>
<p><img src="figure/variance.Rmd/p_of_f-3.png" title="" alt="" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code># plot the variance and show p 0f f
reads_var$p_of_f &lt;- f.test.table$p_of_f

ggplot(reads_var, aes(x = bulk_variance, y = mean_single_variance, shape = as.factor(ERCC), col = p_of_f)) + geom_point(size = 2, alpha = 0.25) + scale_x_log10() + scale_y_log10() + scale_colour_gradient2(midpoint= 0.5, space=&quot;Lab&quot;) + stat_function(fun= function(x) {x}, col= &quot;#56B4E9&quot;)</code></pre>
<p><img src="figure/variance.Rmd/p_of_f-4.png" title="" alt="" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>ggplot(reads_var, aes(x = bulk_variance, y = var_within_individual, shape = as.factor(ERCC), col = p_of_f)) + geom_point(size = 2, alpha = 0.5) + scale_x_log10() + scale_y_log10() + scale_colour_gradient2(midpoint= 0.5, space=&quot;Lab&quot;) + stat_function(fun= function(x) {x}, col= &quot;#56B4E9&quot;)</code></pre>
<p><img src="figure/variance.Rmd/p_of_f-5.png" title="" alt="" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>ggplot(reads_var, aes(x = bulk_variance, y = var_between_individual, shape = as.factor(ERCC), col = p_of_f)) + geom_point(size = 2, alpha = 0.5) + scale_x_log10() + scale_y_log10() + scale_colour_gradient2(midpoint= 0.5, space=&quot;Lab&quot;) + stat_function(fun= function(x) {x}, col= &quot;#56B4E9&quot;)</code></pre>
<pre><code>Warning in scale$trans$trans(x): NaNs produced</code></pre>
<pre><code>Warning in scale$trans$trans(x): NaNs produced</code></pre>
<pre><code>Warning in loop_apply(n, do.ply): Removed 28 rows containing missing values
(geom_point).</code></pre>
<p><img src="figure/variance.Rmd/p_of_f-6.png" title="" alt="" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>ggplot(reads_var, aes(x = var_all_single, y = var_between_individual, shape = as.factor(ERCC), col = p_of_f)) + geom_point(size = 2, alpha = 0.5) + scale_x_log10() + scale_y_log10() + scale_colour_gradient2(midpoint= 0.5, space=&quot;Lab&quot;) + stat_function(fun= function(x) {x}, col= &quot;#56B4E9&quot;)</code></pre>
<pre><code>Warning in scale$trans$trans(x): NaNs produced</code></pre>
<pre><code>Warning in scale$trans$trans(x): NaNs produced</code></pre>
<pre><code>Warning in loop_apply(n, do.ply): Removed 28 rows containing missing values
(geom_point).</code></pre>
<p><img src="figure/variance.Rmd/p_of_f-7.png" title="" alt="" width="672" style="display: block; margin: auto;" /></p>
<p>Pull the F value</p>
<pre class="r"><code># looking at the F value
f.test.F &lt;- function(data.in){
  tt &lt;- names(data.in)
  individual.id &lt;- rep(&quot;19098&quot;,length(tt))
  individual.id[grep(&quot;19101&quot;,tt)] &lt;- &quot;19101&quot;
  individual.id[grep(&quot;19239&quot;,tt)] &lt;- &quot;19239&quot;
  
  dd &lt;- data.frame(reads=unlist(data.in),individual.id=individual.id)
  fit1 &lt;- lm(reads~1,data=dd)
  fit2 &lt;- lm(reads~1 + individual.id,data=dd)
  anova(fit1,fit2)[2,&quot;F&quot;]
}
# creat the f test table of F value
f.test.F.table &lt;- do.call(rbind,lapply(rownames(reads_single_cpm),function(x){
  data.frame(gene_name=x,F_value=f.test.F(reads_single_cpm[x,]))  
}))

# sellect ERCC
f.test.F.table$ERCC &lt;- grepl(&quot;ERCC&quot;,f.test.F.table[,1])

# plot the variance and show F value
reads_var$F_value &lt;- f.test.F.table$F_value

# calculate F value from p
# when p=0.01
qf(1-0.05,2,629)</code></pre>
<pre><code>[1] 3.010045</code></pre>
<pre class="r"><code># create color index
reads_var$F_color &lt;- &quot;1 &lt; F &lt; 3&quot;
reads_var$F_color[reads_var$F_value &gt;=3] &lt;- &quot;F &gt;= 3&quot;
reads_var$F_color[reads_var$F_value &lt;=1] &lt;- &quot;F &lt;= 1&quot;

# use the F value from p to scale the gradient
ggplot(reads_var, aes(x = bulk_variance, y = mean_single_variance, shape = as.factor(ERCC), col = F_color)) + scale_colour_manual(values=cbPalette) + geom_point(size = 2, alpha = 0.5) + scale_x_log10() + scale_y_log10() + stat_function(fun= function(x) {x}, col= &quot;#56B4E9&quot;)</code></pre>
<p><img src="figure/variance.Rmd/Fvalue-1.png" title="" alt="" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>ggplot(reads_var, aes(x = bulk_variance, y = var_within_individual, shape = as.factor(ERCC), col = F_color)) + scale_colour_manual(values=cbPalette) + geom_point(size = 2, alpha = 0.5) + scale_x_log10() + scale_y_log10() + stat_function(fun= function(x) {x}, col= &quot;#56B4E9&quot;)</code></pre>
<p><img src="figure/variance.Rmd/Fvalue-2.png" title="" alt="" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>ggplot(reads_var, aes(x = bulk_variance, y = var_between_individual, shape = as.factor(ERCC), col = F_color)) + scale_colour_manual(values=cbPalette) + geom_point(size = 2, alpha = 0.5) + scale_x_log10() + scale_y_log10() + stat_function(fun= function(x) {x}, col= &quot;#56B4E9&quot;)</code></pre>
<pre><code>Warning in scale$trans$trans(x): NaNs produced</code></pre>
<pre><code>Warning in scale$trans$trans(x): NaNs produced</code></pre>
<pre><code>Warning in loop_apply(n, do.ply): Removed 28 rows containing missing values
(geom_point).</code></pre>
<p><img src="figure/variance.Rmd/Fvalue-3.png" title="" alt="" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>ggplot(reads_var, aes(x = var_all_single, y = var_within_individual, shape = as.factor(ERCC), col = F_color)) + scale_colour_manual(values=cbPalette) + geom_point(size = 2, alpha = 0.5) + scale_x_log10() + scale_y_log10() + stat_function(fun= function(x) {x}, col= &quot;#56B4E9&quot;)</code></pre>
<p><img src="figure/variance.Rmd/Fvalue-4.png" title="" alt="" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>ggplot(reads_var, aes(x = var_all_single, y = var_between_individual, shape = as.factor(ERCC), col = F_color)) + scale_colour_manual(values=cbPalette) + geom_point(size = 2, alpha = 0.2) + scale_x_log10() + scale_y_log10() + stat_function(fun= function(x) {x}, col= &quot;#56B4E9&quot;)</code></pre>
<pre><code>Warning in scale$trans$trans(x): NaNs produced</code></pre>
<pre><code>Warning in scale$trans$trans(x): NaNs produced</code></pre>
<pre><code>Warning in loop_apply(n, do.ply): Removed 28 rows containing missing values
(geom_point).</code></pre>
<p><img src="figure/variance.Rmd/Fvalue-5.png" title="" alt="" width="672" style="display: block; margin: auto;" /></p>
</div>
<div id="session-information" class="section level2">
<h2>Session information</h2>
<pre class="r"><code>sessionInfo()</code></pre>
<pre><code>R version 3.2.0 (2015-04-16)
Platform: x86_64-unknown-linux-gnu (64-bit)

locale:
 [1] LC_CTYPE=en_US.UTF-8       LC_NUMERIC=C              
 [3] LC_TIME=en_US.UTF-8        LC_COLLATE=en_US.UTF-8    
 [5] LC_MONETARY=en_US.UTF-8    LC_MESSAGES=en_US.UTF-8   
 [7] LC_PAPER=en_US.UTF-8       LC_NAME=C                 
 [9] LC_ADDRESS=C               LC_TELEPHONE=C            
[11] LC_MEASUREMENT=en_US.UTF-8 LC_IDENTIFICATION=C       

attached base packages:
[1] stats     graphics  grDevices utils     datasets  methods   base     

other attached packages:
[1] edgeR_3.10.2  limma_3.24.9  ggplot2_1.0.1 dplyr_0.4.1   knitr_1.10.5 

loaded via a namespace (and not attached):
 [1] Rcpp_0.11.6      magrittr_1.5     MASS_7.3-40      munsell_0.4.2   
 [5] colorspace_1.2-6 stringr_1.0.0    plyr_1.8.2       tools_3.2.0     
 [9] parallel_3.2.0   grid_3.2.0       gtable_0.1.2     DBI_0.3.1       
[13] htmltools_0.2.6  lazyeval_0.1.10  yaml_2.1.13      assertthat_0.1  
[17] digest_0.6.8     reshape2_1.4.1   formatR_1.2      evaluate_0.7    
[21] rmarkdown_0.6.1  labeling_0.3     stringi_0.4-1    scales_0.2.4    
[25] proto_0.3-10    </code></pre>
</div>


<!-- some extra javascript for older browsers -->
<script type="text/javascript" src="libs/polyfill.js"></script>

<script>

// manage active state of menu based on current page
$(document).ready(function () {

    // active menu
    href = window.location.pathname
    href = href.substr(href.lastIndexOf('/') + 1)
    $('a[href="' + href + '"]').parent().addClass('active');

    // manage active menu header
    if (href.startsWith('authoring_'))
      $('a[href="' + 'authoring' + '"]').parent().addClass('active');
    else if (href.endsWith('_format.html'))
      $('a[href="' + 'formats' + '"]').parent().addClass('active');
    else if (href.startsWith('developer_'))
      $('a[href="' + 'developer' + '"]').parent().addClass('active');

});

</script>

</div>

<script>

// add bootstrap table styles to pandoc tables
$(document).ready(function () {
  $('tr.header').parent('thead').parent('table').addClass('table table-condensed');
});

</script>


</body>
</html>
