---
bibliography: refs.bib
---

```{r chunk-options, include=FALSE}
source("chunk-options-paper.R")
```

## Discussion

### Study design and sample size for scRNA-seq
Our nested study design allowed us to explicitly estimate technical batch effects associated with single cell sample processing on the C1 platform. 
We found previously unreported technical sources of variation associated with the C1 sample processing and the use of UMIs, including the read-to-molecule conversion efficiency. 
Our batch effect correction following proper standardization and normalization can successfully remove these technical effects while maintaining individual differences in gene expression level and variation across individual cells. 
However, to fully capture the batch effect across C1 plates and to retrieve the maximum amount of biological information, we recommend a balanced study design consisting of multiple individuals within a C1 plate and multiple C1 replicates (for example, Sup figure. `r fig_supp_design`). 
In such a scenario, single cells can be assigned to biological samples based on their genotypes, which can be characterized based on the RNA sequencing (see Sup figure. `r fig_supp_design`). 

We also addressed additional study design parameters, including the number of cells and the sequencing coverage and complexity. 
These parameters should be determined based on the goal of the study. 
In our case, we focused on recapitulating gene expression levels and inter-individual regulatory differences, estimated from the bulk samples, using the single cell gene expression data. A similar assessment using single cell data from the C1 platform was published previously, but it was based on non-UMI data [@Wu2014; @Pollen2014]. 
Our results, using UMI based molecule count data, suggest that a sample size of 75 high quality cells per individual, with sequencing depth of 2 million reads per cell, are sufficient not only to recapitulate gene expression estimates based on bulk populations, but also to faithfully represent transcriptional variability across single cells (Figure. `rfig_main_subsample`C).

### Effective use of UMIs and molecule counts
When UMIs were first incorporated in scRNA-seq protocols using the C1 system, it was assumed that total molecule counts directly relate to the absolute number of mRNA molecules in the cell [@Islam2014].
In other words, a difference in total number of observed molecules between cells was thought to indicate a corresponding difference in the actual number of mRNA molecules between cells [@Vallejos2015].
In the ideal situation, samples with more mRNA content, the ones with more than one cells or in G2 phase, would have more total molecules than those have only one cells or in G1 phase, respectively [@Vallejos2015; @Buettner2015].
Our work suggests that the correlation between sample concentration and total molecule counts is associated with considerable variation. 
A particular extreme example include the class of samples in which more than one cell were captured. 
In these cases, the RNA content is expected to be roughly doubled compared with samples in which a single cell was captured, yet the library concentration, as well as the total number of sequenced molecules, are not consistently higher (and in many cases, are lower than the single cell samples). 

In contrast to our observations, a few recent studies have reported that the cell cycle phase of single cells can be inferred based on total read or molecule counts, which allows one to distinguish between cells at larger (G2) and smaller (G1) phases. 
However, these studies sorted the single cells by cell cycle phase before using the C1 to process each sorted population separately [@Buettner2015; @Leng2015].
As such, the cell cycle phase of the pre-sorted single cells was completely confounded with the C1 preparation, which as we have shown, is associated with a technical batch effect. 

Another underlying assumption of the UMI-based single-cell approach is that the conversion of read to molecule counts is consistent across samples. 
Yet, we have shown that this is not the case. Read to molecule conversion rates differ not only among samples from different individuals, but also between samples from the same individual, which were processed in different C1 batches. 
Remarkably, this was true for both endogenous and ERCC spike-ins genes. There may be various potential causes for this observation, both biological or technical, including cell sizes, cell proliferation rate, response to the reverse transcription, UMI tagging efficiency and so on. 
Importantly, because of the inconsistency of conversion efficiency, we cannot assume that counting UMIs is a completely unbiased process. 
As a result, UMI based molecule counts need to be standardized, normalized, and corrected for batch effect.  

### The effectiveness of ERCC spike-in controls in scRNA-seq
ERCC spike-in controls have been systematically included in numerous scRNA-seq studies for multiple purposes: to filter for low quality cells, to model the technical variability across cells, and to infer the cellular mRNA content [@Brennecke2013; @Grun2014; @Ding2015; @Vallejos2015]. 
Using the percentage of ERCC spike-ins as a filter of potential dead or ruptured cells, we removed a small group of samples with higher percentages, but at the meantime we also observed a continuous distribution of the samples (Figure. `rfig_main_qc`E and Sup Figure. 7B), indicating that choosing a cutoff based on ERCC read percentage is not as obvious as we anticipated. 
While ERCC spike-in controls cover a wide range of concentration, the fact that they are relative short in length and have shorter polyA tails compared to mammalian mRNAs brings up the question of whether these differences will affect mRNA capture efficiency.
In our hands, only 43 out of the 92 ERCC spike-ins passed our expression filter and were used for normalization across samples (Figure. `rfig_main_qc`G).
For the estimation of cellular mRNA contents, theoretically, the total molecule counts of ERCCs should be the same across samples and the proportion of ERCC molecule can serve as an indicator of the amount of total endogenous gene molecules [@Vallejos2015].
Nonetheless, samples with multiple cells have not only lower ERCC percentage but also lower ERCC molecule counts than samples with one cells (data not shown).
Furthermore, the differences of ERCC percentages and ERCC molecule counts across individuals caused by unknown factors have never been reported nor taken into account while modeling the technical variability of scRNA-seq data.
Even though ERCC spike-in controls can provide the information required to create a standard baseline measurement of RNA abundance across samples, as well as to determine both the dynamic range and the lower detection limit [@Mortazavi2008], the strong library preparation effects have also been observed in bulk RNA-seq data [@Risso2014; @SEQC/MAQC-IIIConsortium2014].
A better alternative to the ERCC spike-in controls that can more faithfully mimic mammalian mRNAs is highly desired and would improve future scRNA-seq studies.

### Cell to cell variation of gene expression
As have been reported [@Shalek2013], we observed that the bimodal distribution of certain genes across single cells was a result of unobserved expression.
To more rigorously investigate the gene expression noise that is not caused by the non-expressed cells, we decided to look at all cells or only cells in which a given gene was detected.

