---
bibliography: refs.bib
---

```{r chunk-options, include=FALSE}
source("chunk-options-paper.R")
```

## Discussion

### Study design and sample size for scRNA-seq
Our nested study design allowed us to explicitly estimate technical batch effects associated with single cell sample processing on the C1 platform.
We found previously unreported technical sources of variation associated with the C1 sample processing and the use of UMIs, including a batch-specific read-to-molecule conversion efficiency. 
Our batch effect correction following standardization and normalization of the molecule count data can successfully remove these technical effects while maintaining individual differences in gene expression level and variation across individual cells. 
However, to fully capture the batch effect across C1 plates and to retrieve the maximum amount of biological information, we recommend a balanced study design consisting of multiple individuals within a C1 plate and multiple C1 replicates (for example, Figure `r fig_supp_design`A). 
In such a scenario, single cells can be assigned to biological samples based on their genotypes, which can be characterized based on the RNA sequencing (Figure `r fig_supp_design`B).

We also addressed additional study design parameters, including the number of cells and the sequencing coverage and complexity. 
Ultimately, these parameters should be determined based on the goal of the study. In our case, we focused on recapitulating the gene expression levels observed in bulk sequencing experiments and the cell-cell variance in gene expression levels across single cells.
A similar assessment using single cell data from the C1 platform was published previously, but it was based on non-UMI data [@Wu2014; @Pollen2014]. 
Our results, using UMI based molecule count data, suggest that a sample size of 75 high quality cells per individual, with sequencing depth of 2 million reads per cell, are sufficient not only to recapitulate gene expression estimates based on bulk populations, but also to faithfully represent transcriptional variability across single cells (Figure. `r fig_main_subsample`C).

### Effective use of UMIs and molecule counts
When UMIs were first incorporated in scRNA-seq protocols using the C1 system, it was assumed that total molecule counts directly relate to the absolute number of mRNA molecules in the cell [@Islam2014].
In other words, a difference in total number of observed molecules between cells was thought to indicate a corresponding difference in the actual number of mRNA molecules between cells [@Vallejos2015].
The inference was that samples with more mRNA content, for example - cells in G2 phase, would have higher numbers of total molecules than samples with less mRNA content, for example - cells in G1 phase [@Vallejos2015; @Buettner2015].
Our work suggests that the correlation between sample concentration and total molecule counts is associated with considerable variation.
A particular extreme example include the class of samples in which more than one cell were captured.
In these cases, the RNA content is expected to be roughly doubled compared with samples in which a single cell was captured, yet the library concentration, as well as the total number of sequenced molecules, are not consistently higher (and in many cases, are lower than the single cell samples).

In contrast to our observations, a few recent studies have reported that the cell cycle phase of single cells can be inferred based on total read or molecule counts.
However, these studies sorted the single cells by cell cycle phase before using the C1 to process each sorted population separately [@Buettner2015; @Leng2015].
As such, the cell cycle phase of the pre-sorted single cells was completely confounded with the C1 preparation, which as we have shown, is associated with a technical batch effect.

Another underlying assumption of the UMI-based single-cell approach is that the conversion of read to molecule counts is consistent across samples.
Yet, we have shown that this is not the case. Read to molecule conversion rates differ not only among samples from different individuals, but also between samples from the same individual, which were processed in different C1 batches.
Remarkably, this was true for both endogenous and ERCC spike-ins genes. 
There may be various potential causes for this observation, both biological or technical, including cell sizes, cell proliferation rate, response to the reverse transcription, UMI tagging efficiency and so on. 
Importantly, because of the inconsistency of conversion efficiency, we cannot assume that counting UMIs is a completely unbiased process.
As a result, UMI based molecule counts need to be standardized, normalized, and corrected for batch effect.  
