---
bibliography: refs.bib
---

```{r chunk-options, include=FALSE}
source("chunk-options-paper.R")
```

## Discussion
### Summary

### Design and QC
UMI, ERCC, technical and biological replicates, 
Our results from subsampling provide practical information for the number of cells and the sequencing depths required to recapitulate bulk population and also to represent variability across single cells.

### UMIs and molecule number 
absolute number, relative number, sequencing depth

### Read to molecule conversion
vary across samples, batch effect, individual effect

### ERCC spike-ins
filter low quality sample, techcnical variability across sample, and cellular content

Even though ERCC spike-in controls can provide the information required to create a standard baseline measurement of RNA abundance across samples, as well as to determine both the dynamic range and the lower detection limit [@Mortazavi2008], the strong library preparation effects have also been observed in bulk RNA-seq data [@Risso2014; @SEQC/MAQC-IIIConsortium2014].

### Pipeline
collision probability, standardization, normalization, and batch correction

### Cell to cell variation of gene expression
measure of noise level, identify different distribution patterns, pluripotency genes 
