---
bibliography: refs.bib
---

```{r chunk-options, include=FALSE}
source("chunk-options-paper.R")
```

## Discussion

### Study design and sample size for scRNA-seq

Our nested study design allowed us to explicitly estimate technical batch effects associated with single cell sample processing on the C1 platform.
We found previously unreported technical sources of variation associated with the C1 sample processing and the use of UMIs, including the property of batch-specific read-to-molecule conversion efficiency.
As we used a well-replicated nested study design, we were able to model, estimate, and account for the batch while maintaining individual differences in gene expression levels.
We believe that our observations indicate that future studies should avoid confounding C1 batch and individual source of single cell samples.
Instead, we recommend a balanced study design consisting of multiple individuals within a C1 plate and multiple C1 replicates (for example, Figure `r fig_supp_design`A).
The origin of each cell can then be identified using the RNA sequencing data.
Indeed, using a method originally developed for detecting sample swaps in DNA sequencing experiments [@Jun2012], we were able to correctly identify the correct YRI individual of origin for all the single cells from the current experiment by comparing the polymorphisms identified using the RNA-seq reads to the known genotypes for all 120 YRI individuals of the International HapMap Project (Figure `r fig_supp_design`B).
The mixed-individual-plate is an attractive study design because it allows one to account for the batch effect without the requirement to explicitly spend additional resources on purely technical replication (because the total number of cells assayed from each individual can be equal to a design in which one individual is being processed in using a single C1 plate).

We also addressed additional study design properties with respect to the desired number of single cells and the desired depth of sequencing (Figure `r fig_main_subsample`).
Similar assessments have been previously performed for single cell sequencing with the C1 platform without the use of UMIs [@Wu2014; @Pollen2014], but no previous study has investigated the effects of these parameters for single cells studies using UMIs.
We focused on recapitulating the gene expression levels observed in bulk sequencing experiments, detecting as many genes as possible, and accurately measuring the cell-to-cell variation in gene expression levels.
We recommend sequencing at least 75 high quality cells per biological condition with a minimum of 1.5 million raw reads per cell to obtain optimal performance of these three metrics.

### The limitations of the ERCC spike-in controls

The ERCC spike-in controls have been used in previous scRNA-seq studies to identify low quality single cell samples, infer the absolute total number of molecules per cell, and model the technical variability across cells [@Brennecke2013; @Grun2014; @Ding2015; @Vallejos2015].
In our experience, the ERCC controls are not particularly well-suited for any one of these tasks, much less all three.
With respect to identifying low quality samples, we indeed observed that samples with no visible cell had a higher percentage of reads mapping to the ERCC controls, as expected.
However, there was no clear difference between low and high quality samples in the percentage of ERCC reads or molecules, and thus any arbitrarily chosen cutoff would be associated with considerable error (Figure `r fig_main_qc`E).
With respect to inferring the absolute total number of molecules per cell, we observed that the biological covariate of interest (difference between the three YRI individuals), rather than batch, explained a large proportion of the variance in the ERCC counts (Figure `r fig_supp_variance`), and furthermore that the ERCC controls were also affected by the individual-specific effect on the read-to-molecule conversion rate (Figure `r fig_main_batch`D).
Thus ERCC-based corrected estimates of total number of molecules per cell, across technical or biological replicates, are expected to be biased.
Because the batch effects associated with the ERCC controls are driven by the biological covariate of interest, they will also impede the modeling of the technical variation in single cell experiments that confound batch and the biological source of the single cells.

More generally, it is inherently difficult to model unknown sources of technical variation using so few genes [@Risso2014] (only approximately half of the 92 ERCC controls are detected in typical single cell experiments), and the ERCC controls are also strongly impacted by technical sources of variation even in bulk RNA-seq experiments [@SEQC/MAQC-IIIConsortium2014].
Lastly, from a theoretical perspective, the ERCC controls have shorter polyA tails and are overall shorter than mammalian mRNAs.
For these reasons, we caution against the reliance of ERCC controls in scRNA-seq studies and highlight that an alternative set of controls that more faithfully mimics mammalian mRNAs and provides more detectable spike-in genes is desired.
Our recommendation is to include total RNA from a distant species, for example using RNA from *Drosophila* *melanogaster* in studies of single cells from humans.

### Outlook

Single cell experiments are ideally suited to study gene regulatory noise and robustness [@Borel2015; @Finak2015].
Yet, in order to study the biological noise in gene expression levels, it is imperative that one should be able to effectively estimate and account for the technical noise in single cell gene expression data.
Our results indicate that previous single cells gene expression studies may not have been able to distinguish between the technical and the biological components of variation, because single cell samples from each biological condition were processed on a single C1 batch.
When technical noise is properly accounted for, even in this small pilot study, our findings indicate pervasive inter-individual differences in gene regulatory noise, independently of gene abundance.
