---
bibliography: refs.bib
---

```{r chunk-options, include=FALSE}
source("chunk-options-paper.R")
```

## Discussion

### Study design and sample size for scRNA-seq

Our nested study design allowed us to explicitly estimate technical batch effects associated with single cell sample processing on the C1 platform.
We found previously unreported technical sources of variation associated with the C1 sample processing and the use of UMIs, including a batch-specific read-to-molecule conversion efficiency. 
In the context of the nested study design, our batch effect correction following standardization and normalization of the molecule count data can successfully remove these technical effects while maintaining individual differences in gene expression level and variation across single cells. 
However, to fully capture the batch effect across C1 plates and to retrieve the maximum amount of biological information, we recommend a balanced study design consisting of multiple individuals within a C1 plate and multiple C1 replicates (for example, Figure `r fig_supp_design`A).
The cell can then be identified using the RNA sequencing data.
Indeed, using a method originally developed for detecting sample swaps in DNA sequencing experiments [@Jun2012], we were able to correctly identify the correct YRI individual of origin for all the single cells from the current experiment by comparing the polymorphisms identified using the RNA-seq reads to the known genotypes for all 120 YRI individuals of the International HapMap Project (Figure `r fig_supp_design`B).
The mixed-individual-plate is an attractive study design because it allows one to account for the batch effect without the requirement to explicitly spend additional resources on technical replication (because the total number of cells assayed from each individual can be equal to a design in which one individual is being processed in using a single C1 plate).

We also addressed additional study design properties with respect to the desired number of single cells to collect and the desired depth of sequencing (Figure `r fig_main_subsample`).
Similar assessments have been previously performed for single cell sequencing with the C1 platform [@Wu2014; @Pollen2014], but no previous study has investigated the effects of these parameters for single cells studies using UMIs.
We focused on recapitulating the gene expression levels observed in bulk sequencing experiments, detecting as many genes as possible, and accurately measuring the cell-to-cell variation in gene expression levels.
We recommend sequencing at least 75 high quality cells per biological condition with a minumum of 1.5 million raw reads per cell to obtain optimal performance of these three metrics.

### Effective use of UMIs and molecule counts

When UMIs were first incorporated in scRNA-seq protocols using the C1 system, it was assumed that total molecule counts directly relate to the absolute number of mRNA molecules in the cell [@Islam2014].
In other words, a difference in total number of observed molecules between cells was thought to indicate a corresponding difference in the actual number of mRNA molecules between cells [@Vallejos2015].
The inference was that samples with more mRNA content, for example - cells in G2 phase, would have higher numbers of sequenced total molecules than samples with less mRNA content, for example - cells in G1 phase [@Vallejos2015; @Buettner2015].
Our work suggests that the correlation between sample concentration and total molecule counts sequenced from single cells is associated with considerable variation.
A particular extreme example include the class of samples in which more than one cell was captured.
In these cases, the RNA content is expected to be roughly doubled compared with samples in which a single cell was captured, yet the library concentration, as well as the total number of sequenced molecules, are not consistently higher (and in many cases, are lower than the single cell samples).

In contrast to our observations, a few recent studies have reported that the cell cycle phase of single cells can be inferred based on total read or molecule counts.
However, these studies sorted the single cells by cell cycle phase before using the C1 to process each sorted population separately [@Buettner2015; @Leng2015].
As such, the cell cycle phase of the pre-sorted single cells was completely confounded with the C1 preparation, which as we have shown, is associated with a technical batch effect.

Another underlying assumption of the UMI-based single-cell approach is that the conversion of read to molecule counts is consistent across samples.
Yet, we have shown that this is not the case. Read to molecule conversion rates differ not only among samples from different individuals, but also between samples from the same individual, which were processed in different C1 batches.
Remarkably, this was true for both endogenous and ERCC spike-ins genes. 
There may be various potential causes for this observation, both biological or technical, including cell sizes, cell proliferation rate, response to the reverse transcription, UMI tagging efficiency and so on. 
Importantly, because of the inconsistency of conversion efficiency, we cannot assume that counting UMIs is a completely unbiased process.
As a result, UMI based molecule counts need to be standardized, normalized, and corrected for batch effect.

### Are ERCC spike-in controls useful?

The ERCC spike-in controls have been used in previous scRNA-seq studies to identify low quality single cell samples, infer the absolute total number of molecules per cell, and model the technical variability across cells [@Brennecke2013; @Grun2014; @Ding2015; @Vallejos2015]. 
In our experience, the ERCC controls are not particularly well-suited for any one of these tasks, much less all three.
With respect to identifying low quality samples, we indeed observed that wells with no visible cell had a higher percentage of reads mapping to the ERCC controls, as expected.
However, there was no clear difference between low and high quality samples in the percentage of ERCC reads or molecules, and thus any arbitrarily chosen cutoff would be associated with considerable error (Figure `r fig_main_qc`E).
With respect to inferring the absolute total number of molecules per cell, we observed that the biological covariate of interest (difference between the three YRI individuals), rather than batch, explained a large proportion of the variance in the ERCC counts (Figure `r fig_supp_variance`), and furthermore that the ERCC controls were also affected by the individual-specific effect on the read-to-molecule conversion (Figure `r fig_main_batch`D,F).
Thus ERCC-based estimates of the total number of molecules per cell across technical or biological replicates will be strongly biased.
These batch effects on the ERCC controls driven by the biological covariate of interest will also impede the modeling of the technical variation in single cell experiments.

More generally, it is inherently difficult to model unknown sources of technical variation using so few genes [@Risso2014] (only approximately half of the 92 ERCC controls are detecetd in typical single cell experiments), and the ERCC controls are also strongly impacted by technical sources of variation even in bulk RNA-seq experiments [@SEQC/MAQC-IIIConsortium2014].
Lastly, from a theoretical perspective, the ERCC controls have shorter polyA tails and are overall shorter compared to mammalian mRNAs.
For these reasons, we caution against the reliance of ERCC controls in scRNA-seq studies and highlight that an alternative set of controls that more faithfully mimics mammalian mRNAs and provides more detectable spike-ins is highly desired for improving future scRNA-seq studies.
Our recommendation is to include total RNA from a distant species, for example using drosophila RNA in studies of single cells from humans.

### Outlook

Single cell experiments are ideally suited to study gene regulatory noise and robustness [@Borel2015; @Finak2015]. 
Yet, in order to study biological noise in gene expression levels it is imperative that one is able to effectively estimate and account for technical noise single cell gene expression data. 
Our results indicate that previous single cells gene expression studies may have not been able to distinguish between the technical and biological components of variation because single cell from each individual biological sample were processed using a single batch. 
When technical noise is properly accounted for, even in this small pilot study, our findings indicate pervasive inter-individual difference in gene regulatory noise, independently from gene abundance. 
