---
bibliography: refs.bib
---

```{r chunk-options, include=FALSE}
source("chunk-options-paper.R")
```

## Discussion
### Summary

### Design and QC
Our results from subsampling provide practical information for the number of cells and the sequencing depths required to recapitulate bulk population and also to represent variability across single cells.

### UMIs and molecule number 
When UMIs was first incoporated in the scRNA-seq protocal of the C1 system, a strong assumption that total molecule counts represent the absolute numbers of mRNA molecules within as cell was made [@Islam2014].
In other words, the relative differences in total number of observed molecules between cells are the relative difference in the absolate number of mRNA molecules in cells [@Vallejos2015].
In the deal situation, samples with more mRNA content, the ones with more than one cells or in G2 phase, would have more total molecules than those have only one cells or in G1 phase, respectively [@Vallejos2015; @Buettner2015].
While we were able to observe a possitive correlation between sample concentration and total molecule counts, the variance is quite substantials (Sup Figure. 5C), emphasizing the existance of non-neglible technical variability.
We also found similar results in cells with different cell cycle phases (Sup Figure. 6), suggesting that total molecule counts is a noisy estimate for interring the difference of mRNA content across cells.
Current single cell studies using the cell cycle information were done by first sorting the cells and then collecting data on separate C1 plates [@Buettner2015; @Leng2015], which could result in the observed difference in molecule counts across cell cycle phases due to merely batch effect.
More importantly, total molecule counts are highly correlated with total read counts of the sample (Figure. 3), indicating that the seqnencing depth of 2-3 million reads per sample is not reaching the satuation point yet (Figure. 2) and that the effect of sequencing depth is still dominant.
We therefore conclude that molecule counts per gene should be standardized by the total number of molecules per sample as has been done in traditional bulk RNA-seq.

### Read to molecule conversion
Another strong assumption of using UMI-based method is that 
vary across samples, batch effect, individual effect

### ERCC spike-ins
ERCC spike-in contols have been systematically included in numerous scRNA-seq studies for multiple purposes: to filter for low quality cells, to model the technical variability across cells, and to infer the cellular mRNA content [@Brennecke2013; @Grun2014; @Ding2015; @Vallejos2015]. 
Using the percentage of ERCC spike-ins as a filter of potential dead or raptured cells, we removed a small group of samples with higher percentages, but at the meantime we also observed a continuous distribution of the samples (Figure. 1E and Sup Figure. 7B), indicating that choosing a cutoff based on ERCC read percentage is not as obvious as we anticipated. 
While ERCC spike-in controls cover a wide range of concentration, the fact that they are realtive short in length and have shorter ployA tails compared to mammalian mRNAs brings up the quesion of whether these differences will affect mRNA capture efficiency.
In our hands, only 43 out of the 92 ERCC spike-ins passed our expression filter and were used for normalization across samples (Figure. 1G), which is a reason why we did not expliciltly model technical vaviabilities across samples.
For the estimation of cellular mRNA contents, theoratically, the total molecule counts of ERCCs should be the same across samples and the proportion of ERCC molecule can serve as an indicator of the amount of total endogenous gene molecules.
Nonetheless, we observed that individual 19098 have not only higher percentage of ERCCs but also higher ERCC molecule counts compared to individual 19239 (Figure. 3C and Sup Figure. 7A-C), suggesting that sampling bias was also involved during sequencing.
Furthermore, the differences of ERCC percentages and ERCC molecule counts across individuals have never been reported and taken into account.  
Even though ERCC spike-in controls can provide the information required to create a standard baseline measurement of RNA abundance across samples, as well as to determine both the dynamic range and the lower detection limit [@Mortazavi2008], the strong library preparation effects have also been observed in bulk RNA-seq data [@Risso2014; @SEQC/MAQC-IIIConsortium2014].
A better alternative to the ERCC spike-in controls is highly desired and somewhat required in future scRNA-seq studies.


### Cell to cell variation of gene expression
measure of noise level, identify different distribution patterns, pluripotency genes 
