---
bibliography: refs.bib
---

```{r chunk-options, include=FALSE}
source("chunk-options-paper.R")
```

## Introduction

Single-cell transcriptomics has recently emerged as a powerful technology to explore stochastic gene expression by assaying steady state transcriptomes from numerous individual cells with minimum losses in cDNA synthesis and high reproducibility [@Macaulay2014; @Saliba2014]. 
Given that it is still a field in its infancy, several challenges and problems that arose in parallel with the application of the technology have yet to be overcome [@Stegle2015].
A fundamental problem, for instance, is the presence of inevitable technical variability introduced during the processing steps, including but not limited to the reaction conditions of mRNA capture, amplification bias, sequencing depth, and pippetting errors, making the analysis of biological variability challenging. 

To overcome the challenge of efficiency and uniformity with which each mRNA molecule is copied and sequenced in single cells, the idea of unique molecule identifiers (UMIs; also called Random Molecule Tags) have been introduced to barcode each single cDNA molecule during in vitro reverse transcription prior to any amplification step [@Kivioja2011; @Fu2011; @Casbon2011; @Shiroguchi2012].
In single cell RNA-seq studies, UMIs have been used as an internal validation control [@Shalek2013] as well as direct quantitative measure of gene expression [@Jaitin2014; @Islam2014; @Grun2014].
To further increase the accuracy of UMI counts while collapsing reads of the same genomic coordinates, computational UMI-tools has also been developed to solve the problem of base-calling errors coming from either PCR during library preparation or sequencing  ([https://github.com/CGATOxford/UMI-tools][UMI-tools]).
Although the molecule counts, as opposed to read counts, using the UMI-based method can substantially reduce the level of technical variability caused by amplification bias, a non-negligible proportion remain unexplained [@Islam2014; @Vallejos2015].    

The motivation to account for unexplained technical variability in single cell RNA-seq experiments has led to the systematic addition of spike-in quantification standards of known abundance in samples [@Brennecke2013; @Grun2014]. 
Specifically, a common set of external RNA controls with various lengths and GC content covering a 220 concentration range as spike-in controls has been developed by the External RNA Controls Consortium (ERCC) [@Jiang2011]. 
With respect to single cell RNA-seq, a number of studies focusing on analyzing the technical variability based on ERCC spike-in controls have been reported using different expression cutoffs and various statistical models [@Brennecke2013; @Grun2014; @Ding2015; @Vallejos2015].
However, to what extent the ERCC spike-in controls can faithfully reflect the technical variability across single cell libraries, both within or between batches, remain inaccessible/uncharacterized.

The automated C1 microfluidic platform (Fluidigm) for single cell RNA-seq has been shown to confer several advantages over tube-based library preparation with higher reproducibility and reduced bias [@Wu2014]. 
Notably, the C1 system allows for direct confirmation of single cell capture events that other microfluidic methods, including the state of the art drop-seq technique [@Macosko2015; @Klein2015], cannot provide.
One of the biggest limitations of using the C1 system, however, is that single cell capture and preparation from different conditions are fully independent, leading to the consequences that batch effects are inherently confounded with the biological covariate of interest (Hicks et al.2015).
Consequently, multiple replicates of C1 collections from the same biological condition are necessary to facilitate estimation of technical variability even with the presence of ERCC spike-in controls [@Stegle2015].
To our knowledge, no study has been purposely conducted to access the technical variability across batches within and between different genetic backgrounds or biological conditions. 

In this study, single cell RNA-seq data were collected from three induced pluripotent stem cell (iPSC) lines from Yoruba individuals in Ibadan, Nigeria (abbreviation: YRI) using C1 microfluidic plates.
Specifically, three C1 collections were independently performed from each individual to disentangle batch effects from biological covariate of interest which, in this case, is the difference between individuals caused by genetic variations.
Both ERCC spike-in controls and UMIs were included into our single cell RNA-seq data for detection of unwanted technical variability, enabling the quantification of mRNA molecule numbers with exceptional accuracy.
Most importantly, a novel analysis pipeline that incorporates standardization, normalization, and batch effect correction was developed and further shown to be effective.
With these data, we were able to identify genes whose expression levels show differential variability across individuals.
This approach allowed us to elucidate technical variability both within and between C1 collections and provide a deep characterization of cell-to-cell variation in gene expression levels across individuals.
