---
bibliography: refs.bib
---

```{r chunk-options, include=FALSE}
source("chunk-options-paper.R")
```

## Introduction

Much of our current understanding of biological processes, especially in mammalian systems, has been based on bulk population-average measurements. 
However, as have been evident in bacteria and yeasts, the phenotypic heterogeneity in a population can result from the stochastic fluctuations in a single cell over time and also from the cell-to-cell variability of certain phenotypes.
The recent advents of single cell technology has revolutionized almost every area of the biological sciences, particularly in regulatory of gene expression, by producing unprecedentedly high resolution data from single cell RNA sequencing (RNA-seq) [@Macaulay]. 
Given that single-cell genomics is still a field in its infancy, several challenges and problems that arose in parallel with the application of this technology have yet to be overcome (Stegle et al., 2015).
A fundamental problem, for instance, is the presence of inevitable technical noise introduced during the processing steps, including but not limited to reaction conditions of mRNA capture, amplification bias, sequencing depth, and pippetting errors. 
The motivation to account for unexplained technical noise in single cell RNA-seq experiments has led to the systematic addition of spike-in quantification standards of known abundance in samples (Brennecke et al., 2013; Grun et al., 2014). 
Specifically, a common set of external RNA controls with various lengths and GC content covering a 220 concentration range as spike-in controls has been developed by the External RNA Controls Consortium (ERCC) (Jiang et al). 

In addition, efforts have been made to ensure the efficiency and the uniformity with which each mRNA molecule is unfaithfully copied and presented in the sequencing libraries by barcoding mRNA transcript with Unique Molecule Identifiers (UMIs) (Jaitin et al., 2014).
Nonetheless, 

Quantitative transcriptome measurement in individual cells with high sensitivity and accuracy are now routinely reported using various single cell RNA-seq approaches, including both traditional tube-based and recently developed microfluidic-based technologies (Saliba et al., 2014).
The automated C1 microfluidic platform (Fluidigm) for single cell RNA-seq has been shown to confer several advantages over tube-based library preparation with higher reproducibility and reduced bias (Wu et al., 2014). 
Notably, the C1 system allows for direct confirmation of single cell capture events that other microfluidic methods, including the state of the art drop-seq technique (Macosko et al., 2015), cannot provide. 



* challenges of single cell RNA-seq
* molecule and read:UMI
* normalization and techinical noise:ERCC
* sample size: number of reads, number of cells
* novelty of this study: multiple c1 bathes per individual and multiple individuals
