---
bibliography: refs.bib
---

```{r chunk-options, include=FALSE}
source("chunk-options-paper.R")
```

# Towards studies of gene regulatory noise using single-cell gene expression profiling

## Introduction

Single-cell genomic technologies can be used to study the regulation of gene expression at unprecedented resolution. 
Using single-cell gene expression data, we might be able to effectively characterize and classify individual cell types and cell states, develop a better understanding of gene regulatory threshold effects in response to treatments or stress, and address a large number of outstanding questions that pertain to the regulation of noise and robustness of gene expression. 

Single-cell technologies coupled with RNA sequencing enable the characterization of steady state gene expression levels from a large number of individual cells with high reproducibility [@Macaulay2014; @Saliba2014]. 
Recently, single cell gene expression data have been used to study and provide unique insight into a wide range of research topics, from differentiation and tissue development [@Macosko2015; @Handel2016], to immune response [@Jaitin2014; @Drissen2016], to pharmacogenomics [@Miyamoto2015; @Kim2015].

Yet, given that it is still a field in its infancy, there are several outstanding challenges and problems that arose in parallel with the application of single cell technology [@Stegle2015].
A fundamental problem, for instance, is the presence of inevitable technical variability introduced during sample processing steps, including but not limited to the conditions of mRNA capture from a single cell, amplification bias, sequencing depth, and pipetting errors. 
These (and other sources of error) may not be unique to single cell technologies, but in the context of studies where each sample (individual cell) is processed as a single batch, these technical considerations make the analysis of biological variability across individual cells particularly challenging.

The use of unique molecule identifiers (UMIs) was proposed to address challenges that are related to the efficiency and uniformity with which mRNA molecules are copied and sequenced in single cells [@Kivioja2011; @Fu2011; @Casbon2011; @Shiroguchi2012]. 
The rationale is that by counting molecules rather than the number of amplified sequencing reads, one can sidestep challenges related to amplification biases, and obtain a more accurate estimates of relative gene expression levels [@Jaitin2014; @Islam2014; @Grun2014]. 
Nevertheless, though molecule counts, as opposed to sequencing read counts, are associated with substantially reduced levels of technical variability, a non-negligible proportion remain unexplained [@Islam2014; @Vallejos2015].

The motivation to better account for unexplained technical variability in single cell RNA-seq experiments has led to the systematic addition of spike-in RNA standards of known abundance to the endogenous samples [@Brennecke2013; @Grun2014]. 
The most commonly used spike in was developed by the External RNA Controls Consortium (ERCC) [@Jiang2011]; it is comprised of a set of RNA controls with various lengths and GC content, spiked at 220 different yet known concentrations.  
A number of single cell studies focusing on analyzing the technical variability based on ERCC spike-in controls have been reported
[@Brennecke2013; @Grun2014; @Ding2015; @Vallejos2015]. 
However, one principle problem with spike-ins (in the context of all genomic technologies) is that they do not ‘experience’ all processing steps that the endogenous sample is subjected to. 
For that reason, it is unknown to what extent the spike-ins can faithfully reflect the error that is being accumulated during the entire sample processing procedures, either within or across batches.

There are a few common platforms in use for single cell RNA-seq. 
The automated C1 microfluidic platform (Fluidigm), while more expensive per sample, has been shown to confer several advantages over tube-based platforms [@Wu2014], including the ability to process smaller samples (when cell numbers are limiting), and a higher gene capture efficiency. 
Notably, in the context of this study, the C1 system also allows for direct confirmation of single cell capture events, in contrast to a few other microfluidic-based approaches [@Macosko2015; @Klein2015].
One of the biggest limitations of using the C1 system, however, is that single cell capture and preparation from different conditions are fully independent [@Hicks2015].
Consequently, multiple replicates of C1 collections from the same biological condition are necessary to facilitate estimation of technical variability even with the presence of ERCC spike-in controls [@Stegle2015]. 
To our knowledge, to date, no study has been purposely conducted to assess the technical variability across batches on the C1 platform.

To address this gap, we collected single cell RNA-seq data from induced pluripotent stem cell (iPSC) lines from three Yoruba individuals (abbreviation: YRI) using C1 microfluidic plates. 
Three C1 collections were independently performed from each individual to disentangle batch effects from biological covariate of interest, which, in this case, is the difference between individuals. Both ERCC spike-in controls and UMIs were included in our sample processing. 
With these data, we were able to elucidate technical variability both within and between C1 batches and thus provide a deep characterization of cell-to-cell variation in gene expression levels across individuals. 
We were also able to identify genes whose inter-cell variability in expression levels differs across individuals.
