---
bibliography: refs.bib
---

```{r chunk-options, include=FALSE}
source("chunk-options-paper.R")
```

## Introduction

Much of our current understanding of biological processes, especially in mammalian systems, has been based on bulk population-average measurements. 
However, as have been evident in bacteria and yeasts, the phenotypic heterogeneity in a population can result from the stochastic fluctuations in a single cell over time and also from the cell-to-cell variability of certain phenotypes (Raser 2004).
The recent advents of single cell technology has revolutionized almost every area of the biological sciences, particularly in regulatory of gene expression, by producing unprecedentedly high resolution data from single cell RNA sequencing (RNA-seq) [@Macaulay2014]. 
Given that single-cell genomics is still a field in its infancy, several challenges and problems that arose in parallel with the application of this technology have yet to be overcome (Stegle et al., 2015).
A fundamental problem, for instance, is the presence of inevitable technical variablity introduced during the processing steps, including but not limited to the reaction conditions of mRNA capture, amplification bias, sequencing depth, and pippetting errors, making the analysis of biological variability challenging. 

The motivation to account for unexplained technical variability in single cell RNA-seq experiments has led to the systematic addition of spike-in quantification standards of known abundance in samples (Brennecke et al., 2013; Grun et al., 2014). 
Specifically, a common set of external RNA controls with various lengths and GC content covering a 220 concentration range as spike-in controls has been developed by the External RNA Controls Consortium (ERCC) (Jiang et al, 2011). 
Even though ERCC spike-in controls can provide the information required to create a standard baseline measurement of RNA abundance across samples, as well as to determine both the dynamic range and the lower detection limit (Mortazavi et al., 2008), the strong library preparation effects have also been observed in bulk RNA-seq data (Risso et al., 2014). 
With the respect to single cell RNA-seq, a number of studies focusing on analyzing the technical variability based on ERCC spike-in controls have been reported using different expression cutoffs and various statistical models (Brennecke et al., 2013; Grun et al., 2014; Ding 2015; Vallejos 2015). 
However, to what extent the ERCC spike-in controls can faithfully refelct the technical variability across single cell libraries, both within or between batches, remain inaccessible/uncharacterized.
Additionally, to overcome the challenge of efficiency and uniformity with which each mRNA molecule is copied and sequenced, unique molecule identifiers (UMIs) were used to barcode each single cDNA molecule during in vitro reverse transcription prior to the amplification step (Kivioja, 2011; Jaitin et al., 2014).
Although the molecule counts of each gene, as opposed to read counts, using the UMI-based method can subtantially reduce the level of technical variability, a non-negligible proportion remain unexplained (Vallejos 2015).    

Quantitative transcriptome measurement in individual cells with high sensitivity and accuracy are now routinely reported using various single cell RNA-seq approaches, including both traditional tube-based and recently developed microfluidic-based technologies (Saliba et al., 2014).
The automated C1 microfluidic platform (Fluidigm) for single cell RNA-seq has been shown to confer several advantages over tube-based library preparation with higher reproducibility and reduced bias (Wu et al., 2014). 
Notably, the C1 system allows for direct confirmation of single cell capture events that other microfluidic methods, including the state of the art drop-seq technique (Macosko et al., 2015; Klein 2015), cannot provide.
One of the biggest limitations of using C1 sytem, however, is that single cell capture and preparation from different conditions or individuls are fully independent, leading to the result that batch effects are confounded with the biological covariate of interest.
As a result, multiple technical and biological replicates of C1 collections are neccesary to faciliate estimation of technical variability even with the presence of ERCC spike-in controls.
Furthermore, the quantitative assessment of C1 system based on non-UMI data has been examined in separate studies showing that as few as 10 single cells can accurately reflect gene expression from a pooled population (Pollen 2014), and that 2 million reads can already reach saturation point of gene detection (Wu 2014).
As for UMI-based methods, no systematic subsampling at the levels of sequencing depth, including both read counts and corresponding molecule counts, along with the consideration of cell number has been reported.

Besides being a promising new resource for regenerative medicine and also as a tool for disease modeling and studies of dynamic molecular phenotypes, our previous studies have demonstrated that human induced pluripotent stem cells (iPSCs) can be utilized as a powerful tool for studying the genetic association with variation in gene regulation (Thomas 2014). 
Importantly, the heterogeneous gene expression levels between individual cells observed in clonal populations of iPSCs has been proposed to reflect the alternate cell states that represent multiple differentiation potentials (Narsinh 2011). 
To understand the underlying mechanisms of such plasticity in iPSCs, quantification of gene expression at the single-cell level is required.
Therefore, the development of an iPSC system for single cell studies would have lasting effects in the field, as it will allow investigators, for the first time, to directly study cell-cell variability during differentiation in humans.

In this study, we generated single cell RNA-seq libraries from three Yoruba iPSC lines using C1 microfluidic plates followed by sequencing on the Illumina HiSeq system. 
With these data, we identified genes whose expression levels show high variability both within and between individuals.
This approach allowed us to elucidate technical variability between and withing each C1 collection and provide a deep characterization of cell-to-cell variation in gene expression levels across individuals.

