---
title: Legends
bibliography: refs.bib
---

```{r chunk-options, include=FALSE}
source("chunk-options-paper.R")
```

## Figures

### Figure 1. Experimental design and quality control of scRNA-seq.
\(A\) Three C1 96 well-integrated fluidic circuit (IFC) replicate were collected from each of the three Yoruba individuals.
Bulk sample was included each time using the exact same chemicals.
(B) Summary of the cutoffs used to filter out low quality cells that might be raptured or dead (See Figure `r fig_supp_qc` for details).
(C-E) To assess the quality of the single cell RNA-seq data, the capture efficiency of cells and the faithfulness of mRNA fraction amplification were determined by the unmapped reads ratio (numbers of unmapped reads divided by total reads), the number of detected genes, the numbers of total mapped reads, and the ERCC spike-in ratio (number of ERCC reads divided by total mapped reads) across cells.
The dash lines indicate the cutoffs summarized in (B).
The three colors represent the three individuals where NA19098 is in red, NA19101 in green, and NA19239 in blue, and the numbers indicate the cells observed in each capture site on C1 plate.
(F) Scatterplots in log scale showing the mean read counts and the mean molecule counts of each endogenous genes (grey) and ERCC spike-ins (blue) from quality single cell samples (`r nrow(anno_filter)` cells) before removal of genes with low expression.
(G) mRNA capture efficiency shown as observed molecule count versus number of spike-in molecules of each ERCC RNA transcript after removal of genes with low expression (`r sum(grepl("ERCC", rownames(molecules_filter)))` ERCC transcripts left).
Each red dot represents the mean of an ERCC transcript across `r nrow(anno_filter)` cells with error bar of SEM.

### Figure 2. The effect of and sequencing depth and cell number on single cell UMI estimates.
Single cell samples were subsampled to the indicated sequencing depth and cell number, and subsequently converted to molecules using the UMIs.
Each point represents the mean +/- sem of 10 random draws of the cell number.
The left panel displays the results for the 6,097 genes with lower expression and the right panel the results for the 6,097 genes with higher expression.
The results are for individual NA19239.
\(A\) The Pearson correlation of aggregated gene expression level estimates from single cells compared to the bulk sequencing samples.
(B) The total number of genes detected with at least one molecule in at least one of the single cells.
(C) The Pearson correlation of cell-to-cell gene expression level variance estimates from single cells compared to the full single cell data set.

### Figure 3. Batch effect of scRNA-seq data using the C1 platform.
(A and B) Violin plots showing the number of total molecule-counts in thousands in single cell samples by C1 replicate.
The endogenous genes and the ERCC spike-ins were shown separately in (A) and (B), respectively.
(C) Scatterplot showing the sample concentration of cDNA amplicons (ng/ul) and total gene molecule-counts.
(D) Scatterplot showing the total ERCC molecule-counts and total gene molecule-counts.
(E and F) Violin plots showing the reads to molecule conversion efficiency (total molecule-counts divided by total read-counts per single cells) by C1 replicate.
The endogenous genes and the ERCC spike-ins were shown separately in (E) and (F), respectively.
There is significant difference across individuals of both endogenous genes in (E; *P* < 0.001) and ERCC spike-ins in (F; *P* < 0.05).
The difference across C1 replicates per individual was also evaluated (*** indicates *P* < 0.001).
The colors represent the three individuals where NA19098 is in red, NA19101 in green, and NA19239 in blue.
Different shapes represent different C1 replicates of the individual.

### Figure 4. Normalization and removal of technical variability.
\(A\) PCA of gene expression in single cell sample using molecule counts before standardization and normalization.
(B) PCA of gene expression after cpm standardization followed by log2 transformation.
(C) PCA of gene expression using the Poisson- normalized molecule numbers.
(D) PCA of gene expression using the batch-corrected molecule numbers.
The colors represent the three individuals where NA19098 is in red, NA19101 in green, and NA19239 in blue.
Different shapes represent different C1 replicates of the individual.

### Figure 5. Cell-to-cell variation in gene expression.
(A and B) Adjusted CV (see Method) plotted against average molecule counts across all cells in (A) and across only the expressed cells in (B), including all three individuals. Each dot represents a gene, and the color indicates the corresponding gene-specific dropout rate (the proportion of undetected cells). (C and D) Venn diagrams showing the overlaps of top 1000 genes across individuals based on mean expression level in (C) and based on adjusted CV values in (D) across only the expressed cells. (E and F) Similarly, venn diagrams showing the overlaps of top 1000 genes across individuals based on mean expression level in (E) and based on adjusted CV values in (E) across all cells.

### Figure 6. Differential noise genes across individuals.
Selected genes are featured and plotted 1) in the upper panel, the log2 gene expression distribution across the detected cells for each individual, 2) in the middle panel, the proportion of detected cells (dark grey) and the proportion of undetected cells (light grey) for each individual, and 3) in the lower panel, the absolute value of adjusted CV for each individual, along with the corresponding gene-specific MAD (median of absolute deviation) value, and the *P* value.
The three colors in the upper and lower panel represent the individuals where NA19098 is in red, NA19101 in green, and NA19239 in blue.

## Tables
### Table 1. Quality single cell samples
List of high quality single cell samples. There are total `r nrow(anno_filter)` cells.

## Supplemental Figures

### Sup. Figure 1. Removal of low quality samples using cutoffs.
(A and B) Violin plots showing by C1 replicate the total number of ERCC spike-in controls in read counts in (A) or in molecule counts in (B) of each single cell samples.
The colors represent the three individuals where NA19098 is in red, NA19101 in green, and NA19239 in blue.
Different shades represent different C1 replicates of the individual.
(C-F) Density plots showing the distributions of the number total mapped reads in (C), the percentage of unmapped reads in (D), the percentage of ERCC reads in (E), and the number of detected genes in (F).
The dash lines represent the cutoffs based on the 95 percentile of samples with no cells (green).
Each cutoff was then applied to remove samples with one cell (red).

### Sup. Figure 2. Removal of potential multi-cell samples.
(A and B) Scatterplots showing the three groups of samples (no cell in green, one cells in orange, and two or more in purple) before (A) and after (B) linear discriminant analysis (LDA) based on sample concentration of cDNA amplicons (ng/ul) and number of detected genes.
(C and D) Similarly, LDA was performed to identify potential multi-cell sample using the read-to-molecule conversion efficiency, computed from total molecule counts divided by total read counts per sample, of endogenous genes and ERCC spike-in control.
Scatterplots showing before and after the LDA in (C) and (D), respectively.
Potential multi-cell samples, the ones that were one cells but identified as two or more by LDA, were then excluded.
The numbers indicate the cells observed in each capture site on C1 plate.

### Sup. Figure 3. Cell cycle phase and cellular mRNA content.
(A and B) Single cells were assigned a cell cycle phases (G1/S, S, G2, G2/M, and M/G1) according to the expressions of a subset of genes based on molecule counts before (A) or after (B) applying the correction (standardization, normalization, and batch correction, see Method).
Each dot represents the mean (+/- sem) of total gene molecule-counts per sample before the correction.
(C and D) Similarly, the mean (+/- sem) of total read-counts per sample were shown using the same classification of cell cycle phases as in (A) and (B).
(E and F) The total cell numbers of each cell cycle phase were shown based on the gene expression before correction in (E) or after correction in (F).
Different colors represent different C1 replicates of the individual.

### Sup. Figure 4. Subsample of high coverage samples.
Each of the four single cells (A9E1, B2E2, B4H1, and D2H2) were sequenced on one Hi-seq lane, providing the sequencing depth up to 50 millions.
(A and B) Corresponding number of genes detected were shown while subsampling for molecule counts (left panel) or for read counts (right panel).
Subsampling was done separately for endogenous genes in (A) and ERCC spike-ins in (B).
(C-F) The effect of different subsample depth was explored on the total read-counts (C and E) or total molecule-counts (D and F) by including counts of genes with minimum 10 reads or 1 molecule.

### Sup. Figure 5. Variance component analysis of per-gene expression profile.
(A and B) Box plots showing the proportion of variance due to individual effect or batch effect from different C1 replicates. Endogenous genes were shown in (A) and the ERCC spike-in controls in (B).

### Sup. Figure 6. The proportion of undetected cells of each gene.
The proportion of undetected cells of each gene and its relationship with log10 mean expression in (A), with log10 variance of expression in (B), and with the CV in (C) of the expressed cells (cells that were quantified by one or more molecule counts of the given gene).
Each dot represents a gene, and red lines display the predicted values using locally weighted scatterplot smooth (LOESS).

### Sup. Figure 7. Permutation-based *P*-value.
\(A\) Histogram of empirical *P*-values based on `r format(sum(grepl("ENSG", rownames(molecules_filter))), big.mark = ",")` permutations.
(B) -log10 empirical p-values are plotted against average gene expression levels. Blue line displays the predicted -log10 *P*-values using locally weighted scatterplot smooth (LOESS).
(C) Median of Absolute Deviation (MAD) of genes versus average gene expression levels.
LOESS was also used to depict predicted MAD values.

### Sup. Figure 8. Cell-to-cell variation of pluripotency genes.  
Density plots showing the distribution of log2 gene expression of key pluripotent genes across all single cells by individual.
The peaks with lower gene expression values (log2 around 4) represent the undetected cells.
The colors represent the three individuals where NA19098 is in red, NA19101 in green, and NA19239 in blue.

### Sup. Figure 9. Proposed study design for scRNA-seq on C1.
\(A\) A balanced study design consisting of multiple individuals within a C1 plate and multiple C1 replicates can fully capture the batch effect across C1 plates and further retrieve the maximum amount of biological information.
(B) The correct identity of each single cell sample was determined by examining the SNPs present in their RNA sequencing reads.

### Sup. Figure 10. Proportion of genes detected in single cell samples.
(A and B) Violin plots showing the proportion of genes detected, computed by the total number of detected genes per cell divided by the total number of genes being detected across all single cells (`r nrow(anno_filter)` cells), before (A) and after (B) the removal of genes with low expressions.
The colors represent the three individuals where NA19098 is in red, NA19101 in green, and NA19239 in blue.
Different shades represent different C1 replicates of the individual.

### Sup. Figure 11. Coefficients of variation (CV) before and after adjusting for gene mean abundance.
(A-C) CV plotted against average molecule counts across all cells (detected and non-detected) for each individual. Grey dots represent endogenous genes, and blue dots indicate ERCC spike-in control genes.
The curves represent the expected CV under three different scenarios. Red curve depicts the expected CV of the endogenous genes while assuming a Poisson distribution with no over-dispersion. Likewise, blue curve depicts the expected CVs of the ERCC spike-in controls under the Poisson assumption. Yellow curve predicts the expected CVs of an over-dispersed Poisson distribution for which standard deviation is three times the ERCC spike-in control genes. (D-F) Adjusted CV values of each gene including all cells are plotted against log10 of the average molecule counts for each individual.

## Supplemental Table

### Sup. Table 1. Data collection.
\(A\) IPSCs were sorted using the 10-17 μm IFC preps with the staining of the pluripotency marker, TRA1-60.
Single cell occupancy is the percentage of occupied capture sites containing one single cell.
The average cDNA concentration was measured by the HT DNA high sensitivity LabChip (Caliper).
(B) To obtain 3 million good quality raw reads and the average of 200,000 molecules per cells, the 96 single cell libraries from one C1 plate were pooled and sequenced in 3 HiSeq lanes.
The samples were balanced across the four 8-lane flowcells.

### Sup. Table 2. Differential CV genes across individuals
List of genes that we classified the estimates of regulatory noise as significantly different across individuals (empirical permutation *P* < $10^{-5}$). There are total 330 genes.

### Sup. Table 3. Gene ontology analysis of the differential CV genes
List of the GO with *P* < 0.05.
