---
title: Legends
bibliography: refs.bib
---

```{r chunk-options, include=FALSE}
source("chunk-options-paper.R")
```

## Figure Legends

### Figure 1. Experimental design and quality control of scRNA-seq.
\(A\) Three C1 96 well-integrated fluidic circuit (IFC) replicates were collected from each of the three Yoruba individuals.
A bulk sample was included in each batch.
(B) Summary of the cutoffs used to remove low quality cells that might be ruptured or dead (See Figure `r fig_supp_qc` for details).
(C-E) To assess the quality of the scRNA-seq data, the capture efficiency of cells and the faithfulness of mRNA fraction amplification were determined based on the proportion of unmapped reads, the number of detected genes, the numbers of total mapped reads, and the proportion of ERCC spike-in reads across cells.
The dash lines indicate the cutoffs summarized in (B).
The three colors represent the three individuals (NA19098 in red, NA19101 in green, and NA19239 in blue), and the numbers indicate the cell numbers observed in each capture site on C1 plate.
(F) Scatterplots in log scale showing the mean read counts and the mean molecule counts of each endogenous gene (grey) and ERCC spike-ins (blue) from the `r nrow(anno_filter)` high quality single cell samples before removal of genes with low expression.
(G) mRNA capture efficiency shown as observed molecule count versus number of molecules added to each sample, only including the `r sum(grepl("ERCC", rownames(molecules_filter)))` ERCC spike-in controls remaining after removal of genes with low abundance.
Each red dot represents the mean +/- SEM of an ERCC spike-in across the `r nrow(anno_filter)` high quality single cell samples.

### Figure 2. The effect of sequencing depth and cell number on single cell UMI estimates.
Sequencing reads from single cells were subsampled to the indicated sequencing depth and cell number, and subsequently converted to molecules using the UMIs. Each point represents the mean +/- SEM of 10 random draws of the indicated cell number.
The left panel displays the results for 6,097 (50% of detected) genes with lower expression levels  and the right panel the results for 6,097 genes with higher expression levels.
\(A\) Pearson correlation of aggregated gene expression level estimates from single cells compared to the bulk sequencing samples.
(B) Total number of genes detected with at least one molecule in at least one of the single cells.
(C) Pearson correlation of cell-to-cell gene expression level variance estimates from subsets of single cells compared to the full single cell data set.

### Figure 3. Batch effect of scRNA-seq data using the C1 platform.
\(A\) Violin plots of the number of total ERCC spike-in molecule-counts in single cell samples per C1 replicate.
(B) Scatterplot of the total ERCC molecule-counts and total gene molecule-counts.
The colors represent the three individuals (NA19098 is in red, NA19101 in green, and NA19239 in blue).
Data from different C1 replicates is plotted in different shapes.
(C and D) Violin plots of the reads to molecule conversion efficiency (total molecule-counts divided by total read-counts per single cells) by C1 replicate.
The endogenous genes and the ERCC spike-ins are shown separately in (C) and (D), respectively.
There is significant difference across individuals of both endogenous genes (*P* < 0.001) and ERCC spike-ins (*P* < 0.05).
The differences across C1 replicates per individual of endogenous genes and ERCC spike-ins were also evaluated (both *P* < 0.01).

### Figure 4. Normalization and removal of technical variability.
Principal component (PC) 1 versus PC2 of the (A) raw molecule counts, (B) log~2~ counts per million (cpm), (C) Poisson transformed expression levels (accounting for technical variability modeled by the ERCC spike-ins), and (D) batch-corrected expression levels.
The colors represent the three individuals (NA19098 in red, NA19101 in green, and NA19239 in blue).
Data from different C1 replicates is plotted in different shapes.

### Figure 5. Cell-to-cell variation in gene expression.
Adjusted CV plotted against average molecule counts across all cells in (A) and across only the cells in which the gene is expressed (B), including data from all three individuals.
Each dot represents a gene, and the color indicates the corresponding gene-specific dropout rate (the proportion of cells in which the gene is undetected).
(C and D) Venn diagrams showing the overlaps of top 1000 genes across individuals based on mean expression level in (C) and based on adjusted CV values in (D), considering only the cells in which the gene is expressed.
(E and F) Similarly, Venn diagrams showing the overlaps of top 1000 genes across individuals based on mean expression level in (E) and based on adjusted CV values in (F), across all cells.

## Tables
### Table 1. High quality single cell samples.
List of the `r nrow(anno_filter)` high quality single cell samples.

## Supplemental Figures

### Figure `r fig_supp_qc`. Removal of low quality samples.
Violin plots of the total read-counts of ERCC spike-in controls in (A) and the total molecule-counts in (B) in single cell samples.
The three colors represent the three individuals (NA19098 in red, NA19101 in green, and NA19239 in blue).
(C-F) Density plots of the distributions of the total mapped reads in (C), the percentage of unmapped reads in (D), the percentage of ERCC reads in (E), and the number of detected genes in (F).
The dash lines indicate the cutoffs based on the 95 percentile of the samples with no cells.

### Figure `r fig_supp_lda`. Removal of samples with multiple cells.
Scatterplots of the three groups of samples (no cell in green, single-cell in orange, and two or more cells in purple) before (A) and after (B) the linear discriminant analysis (LDA) using sample concentration of cDNA amplicons (ng/ul) and the number of detected genes.
(C and D) Similarly, LDA was performed to identify potential multi-cell samples using the read-to-molecule conversion efficiency (total molecule-counts divided by total read-counts per sample) of endogenous genes and ERCC spike-in controls.
Scatterplots of before and after the LDA in (C) and (D), respectively.
The numbers indicate the number of cells observed in each cell capture site.

### Figure `r fig_supp_variance`. Sources of cell-to-cell variance in per-gene expression profile.
Violin plots of the proportion of per-gene cell-to-cell variance that is due to individual sample of origin, different C1 replicates, and other single cell sample differences.
Data is based on molecule counts before normalization and batch correction. 
Endogenous genes are shown in (A) and the ERCC spike-in controls in (B).

### Figure `r fig_supp_dropout`. The gene-specific dropout rate.
The gene-specific dropout rate (the proportion of cells in which the gene is undetected) and its relationship with log~10~ mean expression in (A), with log~10~ variance of expression in (B), and with the CV in (C) of the cells in which the gene is expressed (cells that were quantified by one or more molecule counts of the given gene).
Each point represents a gene, and red lines indicate the predicted values using locally weighted scatterplot smooth (LOESS).

### Figure `r fig_supp_permutation`. Permutation-based *P*-value.
\(A\) Histogram of empirical *P*-values based on `r format(sum(grepl("ENSG", rownames(molecules_filter))), big.mark = ",")` permutations.
(B) -log10 empirical *P*-values are plotted against average gene expression levels.
Blue line indicates the predicted -log10 *P*-values using locally weighted scatterplot smooth (LOESS).
(C) Median of Absolute Deviation (MAD) of genes versus average gene expression levels.
LOESS was also used to depict predicted MAD values (green line).

### Figure `r fig_supp_noisygene`. Inter-individual differences in regulatory noise.
In the upper panel, the distribution of the log~2~ gene expression levels of selected genes are shown, considering only cells in which the gene is detected as expressed.
In the middle panel, the proportion of cells in which the gene is detected as expressed (dark grey) and the dropout rate (light grey) for each individual.
In the lower panel, the absolute value of adjusted CV for each individual, along with the corresponding gene-specific MAD (median of absolute deviation) value, and the *P* value.
The three colors in the upper and lower panel represent the individuals (NA19098 in red, NA19101 in green, and NA19239 in blue).

### Figure `r fig_supp_plurigene`. Cell-to-cell variation of pluripotency genes.
Density plots of the distribution of log2 gene expression of key pluripotent genes across all single cells by individual.
The peaks with lower gene expression values (log2 around 4) represent the cells in which the gene is undetected.
The three colors represent the three individuals (NA19098 is in red, NA19101 in green, and NA19239 in blue).

### Figure `r fig_supp_design`. Proposed study design for scRNA-seq using C1 platform.
\(A\) A balanced study design consisting of multiple individuals within a C1 plate and multiple C1 replicates to fully capture the batch effect across C1 plates and further retrieve the maximum amount of biological information.
(B) The correct identity of each single cell sample was determined by examining the SNPs present in their RNA sequencing reads.

### Figure `r fig_supp_proportion`. The proportion of genes detected in single cell samples.
Violin plots of the proportion of genes detected, computed by the total number of detected genes in each single cell divided by the total number of genes detected across all single cells, before in (A) and after in (B) the removal of genes with low expressions.
The three colors represent the three individuals (NA19098 is in red, NA19101 in green, and NA19239 in blue).

### Figure `r fig_supp_CV`. Coefficients of variation (CV) before and after adjusting for gene mean abundance.
(A-C) CV plotted against average molecule counts across all cells for each individual [@Islam2014].
Grey points represent endogenous genes, and blue points represent ERCC spike-in controls.
The curves indicate the expected CV under three different scenarios.
Red curve depicts the expected CV of the endogenous genes while assuming a Poisson distribution with no over-dispersion.
Likewise, blue curve depicts the expected CVs of the ERCC spike-in controls under the Poisson assumption.
Yellow curve depicts the expected CVs of an over-dispersed Poisson distribution for which standard deviation is three times the ERCC spike-in controls.
(D-F) Adjusted CV values of each gene including all cells are plotted against log~10~ of the average molecule counts for each individual.

## Supplemental Table

### Table `r table_supp_collection`. Data collection.
\(A\) iPSCs were sorted using the 10-17 μm IFC plates with the staining of the pluripotency marker, TRA1-60.
Single cell occupancy is the percentage of occupied capture sites containing one single cell.
The average cDNA concentration was measured by the HT DNA high sensitivity LabChip (Caliper).
(B) The 96 single cell libraries from one C1 plate were pooled and sequenced in three HiSeq lanes.
The pooled samples were assigned across the four 8-lane flowcells.

### Table `r table_noisygene`. Genes associated with inter-individual differences in regulatory noise.
List of genes that we classified the estimates of regulatory noise as significantly different across individuals (empirical permutation *P* < 10^-5^). There are total 330 genes.

### Table `r table_GO`. Gene ontology analysis of the genes associated with inter-individual differences in regulatory noise.
List of the GO with *P* < 0.05.
