---
bibliography: refs.bib
---

```{r chunk-options, include=FALSE}
source("chunk-options-paper.R")
```

## Materials and Methods

### Cell culture
Undifferentiated feeder-free iPSCs generated from Yoruba LCLs were grown in E8 medium (Life Tech) [@Chen2011] on Metrigel-coated tissue culture plates with daily media feeding at 37 °C with 5% (vol/col) CO2. 
For standard maintenance, cells were splited every 3-4 days using cell release solution (0.5 mM EDTA and NaCl in PBS) at the confluence of roughly 80%. 
For the single cell suspension, iPSC were individualized by Accutase Cell Detachment Solution (BD) for 5-7 minutes at 37 °C and washed twice with E8 media immediately before each experiment. 
Cell viability and cell counts were then measured by the Automated Cell Counter (Bio-Rad) to generate resuspension at densities of 2.5 X 105 cells/mL in E8 medium for C1 cell capture.   

### Single cell capture and single cell library preparation
Single cell loading and capture was performed following the fluidigm manual "Using C1 to Generate Single-Cell cDNA Libraries for mRNA Sequencing Protocol"(PN 100-7168). 
Briefly, 30 ul of C1 Suspension Reagent was added to a 70-ul aliquot of ~17,500 cells. 
Five ul of this cell mix were loaded onto 10-17 um C1 Single-Cell Auto Prep IFC microfluidic chip (Fluidigm), and the chip was then processed on a C1 instrument using the cell-loading script according to the manufacturer's instructions. 
Using the standard staining script, the iPSCs were stained with StainAlive TRA-1-60 Antibody (Stemgent, PN 09-0068). 
The capture efficiency and TRA-1-60 staining were then inspected using the EVOS FL Cell Imaging System (ThermoFisher)(supplemental Table X). 

Immediately after imaging, reverse transcription and cDNA amplification were performed in the C1 system using the SMARTer PCR cDNA Synthesis kit (Clontech) and the Advantage 2 PCR kit (Clontech) according to the instructions in the Fluidigm user manual with minor changes to incorporate UMI labeling [@Islam2014].
Specifically, the reverse transcription primer and the 1:50,000 Ambion® ERCC Spike-In Mix1 (Life Tech) were added to the lysis buffer, and the template-switching oligos which contain the UMI (5-bp random sequence) were included in the reverse transcription mix. 
When the run finished, full-length, amplified, single-cell cDNA libraries were harvested in a total of approximately 13 ul C1 Harvesting Reagents and quantified using DNA High Sensitivity LabChip (Caliper). 
A bulk sample, a 40 ul aliquot of ~10,000 cells, was collected in parallel with each C1 chip using the same reaction mixes following the C1 protocol of "Tube Controls with Purified RNA" (PN 100-7168, Appendix A).

For sequencing library preparation, tagmentation and isolation of 5' fragments were performed according to the UMI protocol [@Islam2014].
Instead of using commercial available Tn5 transposase, Tn5 protein stock was freshly purified in house using the IMPACT system (pTXB1, NEB) following the protocol previously described [@Picelli2014].
The activity of Tn5 was tested and shown to be comparable with the EZ-Tn5-Transposase (Epicentre). 
Importantly, all the libraries in this study were generated using the same batch of Tn5 protein purification. 
For each of the bulk samples, two libraries were generated using two different indices in order to get sufficient material. 
All of the 18 bulk libraries were then pooled and labelled as the "bulk" for sequencing.

### Illumina high-throughput sequencing
Single-cell RNA-seq libraries generated from 96 individual cells (namely, one C1 microfluidic prep) were pooled and then sequenced in three lanes on an Illumina Hiseq 2500 instrument using the PCR primer (C1-P1-PCR-2) as the read 1 primer and the Tn5 adapter (C1-Tn5-U) as the index read primer [@Islam2014]. 
Libraries, both from bulk and single cell samples, were assigned to lanes at random on the flowcells (Fig. `r fig_supp_flowcell`). 
Single-end reads of 100 bp were generated along with 8-bp index reads corresponding to the cell-specific barcodes. 

### Read mapping

We processed the sequencing data from the single cells and bulk samples through the same pipeline.
To assess read quality, we ran FastQC ([http://www.bioinformatics.babraham.ac.uk/projects/fastqc][fastqc]) and observed a decrease in base quality at the 3' end of the reads.
Thus we removed low quality bases from the 3' end of the reads using sickle with default settings [@Joshi2011] (Figure `r fig_supp_sickle`).
To handle the UMI sequences at the 5' end of each read, we used umitools ([https://brwnj.github.io/umitools][umitools]) to find all reads with a UMI of the pattern NNNNNGGG.
The UMI was cut from the 5' end of the read and added to the name of the sequence, and reads without UMIs were discarded.
We then mapped reads to human genome hg19 (only including chromosomes 1-22, X, and Y, plus the ERCC sequences) with Subread [@Liao2013], discarding non-uniquely mapped reads (option -u) and breaking ties for best mapping location using Hamming distance (option -H).
Post-mapping we combined all reads for a given single cell using samtools [@Li2009].
We converted read counts to molecule counts using umitools, which keeps only one read for each UMI and start position combination.
To obtain gene-level counts, we separately assigned reads or molecules to protein-coding genes (Ensembl GRCh37 release 82) and the ERCC spike-in genes using featureCounts [@Liao2014].

[fastqc]: http://www.bioinformatics.babraham.ac.uk/projects/fastqc
[umitools]: https://brwnj.github.io/umitools

### Filtering cells and genes

We performed multiple quality control analyses to detect and remove low quality cells.
In an initial analysis investigating the percentage of reads mapping to the ERCC spike-in controls, we observed that replicate 2 of individual 19098 was a clear outlier.
It appeared that too much ERCC spike-in mix was added to this batch, which violated the assumption that the same amount of ERCC molecules was added to eac h cell.
Thus we removed this batch from all of our analyses.

Next we maintained high quality single cells that passed the following criteria:

*  Only one cell observed per well
*  At least X mapped reads
*  Less than X% unmapped reads
*  Less than X% ERCC reads
*  More than X genes with at least one read

We chose these criteria based on the distribution of these metrics in the empty wells.
Lastly, we observed that some wells classified as containing only one cell clustered with multi-cell wells when plotting the number of gene molecules versus the concentration of the samples.
We also removed these misidentified wells.
After filtering, we maintained `r nrow(anno_filter)` high quality single cells 
(NA19098: `r sum(anno_filter$individual == "NA19098")`,
 NA19101: `r sum(anno_filter$individual == "NA19101")`,
 NA19239: `r sum(anno_filter$individual == "NA19239")`).

The quality control analyses were performed using all Ensembl protein-coding genes with at least one observed read.
Using the high quality single cells, we further removed genes with low expression levels for downstream analyses.
We removed all genes with a mean log~2~ cpm less than 2.
Thus we kept
`r format(sum(grepl("ENSG", rownames(molecules_filter))), big.mark = ",")`
endogenous genes and
`r sum(grepl("ERCC", rownames(molecules_filter)))`
ERCC spike-in genes.

### Normalization

We transformed the single cell molecule counts in multiple steps (Fig. `r fig_main_normalization`).
First, we corrected for the collision probability using a method similar to that developed by @Grun2014.
Essentially we are correcting for the fact that we did not observe all the molecules originally in the cell.
The main difference between our approach and that of @Grun2014 is that we apply the correction at the level of gene counts and not individual molecule counts.
Second, we standardized the molecule counts to log~2~ counts per million (cpm).
This standardization was performed separately for the gene and ERCC molecules.
Third, we corrected for cell-to-cell technical noise using the ERCC spike-in controls.
For each single cell, we fit a linear regression between the expected ERCC molecule counts and the observed ERCC molecule counts (both were transformed to log~2~ cpm).
Next we used the slope and intercept of the least squares regression line to linearly transform the log~2~ cpm for the genes in that cell.
This is analogous to the standard curves used for qPCR measurements.
Fourth, we removed technical noise between the eight batches (three replicates each for 19101 and 19239 and two replicates for 19098).
We fit a linear mixed model with a fixed effect for individual and a random effect for the eight batches and removed the variation captured by the random effect.

For the bulk samples, we used read counts even though the reads contained UMIs.
Because these samples contained RNA molecules from millions of cells, we could not assume that the 1,024 UMIs we added could accurately estimate the true number of molecules.
We standardized them to log~2~ cpm.

### Data and code availability

The data have been deposited in NCBI's Gene Expression Omnibus [@Edgar2002] and are accessible through GEO Series accession number GSEXXXXX ([http://www.ncbi.nlm.nih.gov/geo/query/acc.cgi?acc=GSEXXXXX][geo]).
The code and processed data are available at [https://github.com/jdblischak/singleCellSeq][repo].
From the beginning of the project, we performed this research in the open.
The results of our analyses are viewable at [https://jdblischak.github.io/singleCellSeq/analysis][site].

[geo]: http://www.ncbi.nlm.nih.gov/geo/query/acc.cgi?acc=GSEXXXXX
[repo]: https://github.com/jdblischak/singleCellSeq
[site]: https://jdblischak.github.io/singleCellSeq/analysis

### Mixed effect model to correct for batch effect

### Analysis of cell-cell variability within and between individual

### Description of technologies used

1. UMIs (unique molecular identifiers) were added to each RNA transcript during the reverse transcript step, which provides the power to calculate the absolute numbers of the transcripts. 5 random nucleotides (NNNNN) were used as UMIs. Therefore, there are 1024 different UMIs.

2. ERCC spike-ins were added during the cell lysis step as the control.

3. C1 fluidigm system is used to collect single cell 

![Adding unique molecular identifiers (UMIs).](../analysis/figure/material-and-method.Rmd/method.png) 

### Preparation of single LCLs

* cell line: 19239
* sequencing data:
  1. twocell: two cells were sequenced in one rapid lane (index failed)
  2. 4 individual cells: each sequenced in 1 full flowcell lane, 2 made by in-house Tn2 and the other 2 by Epicentre
  3. 96 cells: cells collected by 1 C1 plate. Due to financial limitation, only 24 indices were used. Each pool of 24 libraries were sequenced in 1 full flowcell lane. These and the 4 individual cells were sequenced on the same flowcell.

![Report of LCL sequencing experiments.](../analysis/figure/material-and-method.Rmd/LCL.png)  
  
### Preparation of single iPSCs

* cell line: 19239, 19101, 19098
* number of replicates: 3, each collected using different passages of cells
* bulk vs single cell: each C1 plate has a bulk sample collected at the same time using the same chemicals freshly prepared each time. For the bulk, 2 libraries using 2 different indices were generated from each bulk sample (total 2*9=18). All the bulk samples were pooled as the "bulk" for sequencing. For single cell, the 96 libraries from the same C1 plates were pooled and sequenced.
* sequencing: SE 100+8. total 31 lanes are required. therefore, 4 full flowcells were used. "bulk" sample was sequenced in 1 lane of all 4 flowcells. samples were balanced as much as possible. 
* goals: ~ 3 millions good reads per cell and average 200,000 molecules per cell 
