---
bibliography: refs.bib
---

```{r chunk-options, include=FALSE}
source("chunk-options-paper.R")
```

## Materials and Methods

### Cell culture of iPSCs

Undifferentiated feeder-free iPSCs generated from LCLs from Yoruba individuals in Ibadan, Nigeria (abbreviation: YRI) were grown in E8 medium (Life Tech) [@Chen2011] on Matrigel-coated tissue culture plates with daily media feeding at 37 °C with 5% (vol/col) CO2.
For standard maintenance, cells were split every 3-4 days using cell release solution (0.5 mM EDTA and NaCl in PBS) at the confluence of roughly 80%. 
For the single cell suspension, iPSCs were individualized by Accutase Cell Detachment Solution (BD) for 5-7 minutes at 37 °C and washed twice with E8 media immediately before each experiment. 
Cell viability and cell counts were then measured by the Automated Cell Counter (Bio-Rad) to generate resuspension densities of 2.5 X 105 cells/mL in E8 medium for C1 cell capture.   

### Single cell capture and single cell library preparation

Single cell loading and capture was performed following the Fluidigm manual "Using C1 to Generate Single-Cell cDNA Libraries for mRNA Sequencing Protocol" (PN 100-7168). 
Briefly, 30 ul of C1 Suspension Reagent was added to a 70-ul aliquot of ~17,500 cells. 
Five ul of this cell mix were loaded onto 10-17 um C1 Single-Cell Auto Prep IFC microfluidic chip (Fluidigm), and the chip was then processed on a C1 instrument using the cell-loading script according to the manufacturer's instructions. 
Using the standard staining script, the iPSCs were stained with StainAlive TRA-1-60 Antibody (Stemgent, PN 09-0068). 
The capture efficiency and TRA-1-60 staining were then inspected using the EVOS FL Cell Imaging System (ThermoFisher)(supplemental Table 2). 

Immediately after imaging, reverse transcription and cDNA amplification were performed in the C1 system using the SMARTer PCR cDNA Synthesis kit (Clontech) and the Advantage 2 PCR kit (Clontech) according to the instructions in the Fluidigm user manual with minor changes to incorporate UMI labeling [@Islam2014].
Specifically, the reverse transcription primer and the 1:50,000 Ambion® ERCC Spike-In Mix1 (Life Tech) were added to the lysis buffer, and the template-switching RNA oligos which contain the UMI (5-bp random sequence) were included in the reverse transcription mix. 
When the run finished, full-length, amplified, single-cell cDNA libraries were harvested in a total of approximately 13 ul C1 Harvesting Reagent and quantified using DNA High Sensitivity LabChip (Caliper). 
The average yield of samples per C1 plate ranged from 1.26-1.88 ng per microliter (Sup. Table 1A).
A bulk sample, a 40 ul aliquot of ~10,000 cells, was collected in parallel with each C1 chip using the same reaction mixes following the C1 protocol of "Tube Controls with Purified RNA" (PN 100-7168, Appendix A).

For sequencing library preparation, tagmentation and isolation of 5' fragments were performed according to the UMI protocol [@Islam2014].
Instead of using commercial available Tn5 transposase, Tn5 protein stock was freshly purified in house using the IMPACT system (pTXB1, NEB) following the protocol previously described [@Picelli2014].
The activity of Tn5 was tested and shown to be comparable with the EZ-Tn5-Transposase (Epicentre). 
Importantly, all the libraries in this study were generated using the same batch of Tn5 protein purification. 
For each of the bulk samples, two libraries were generated using two different indices in order to get sufficient material. 
All of the 18 bulk libraries were then pooled and labelled as the "bulk" for sequencing.

### Illumina high-throughput sequencing

Single-cell RNA-seq libraries generated from 96 individual cells (namely, one C1 microfluidic prep) were pooled and then sequenced in three lanes on an Illumina Hiseq 2500 instrument using the PCR primer (C1-P1-PCR-2) as the read 1 primer and the Tn5 adapter (C1-Tn5-U) as the index read primer [@Islam2014]. 
The master mixes, which includes one mix with all the bulk samples and nine mixes corresponding to the three replicates for the three individuals, were sequenced across four flowcells using a design to minimize the introduction of technical batch effects (Fig. `r fig_supp_flowcell`). 
Single-end reads of 100 bp were generated along with 8-bp index reads corresponding to the cell-specific barcodes. 

### Read mapping

To assess read quality, we ran FastQC ([http://www.bioinformatics.babraham.ac.uk/projects/fastqc][fastqc]) and observed a decrease in base quality at the 3' end of the reads.
Thus we removed low quality bases from the 3' end using sickle with default settings [@Joshi2011] (Figure `r fig_supp_sickle`).
To handle the UMI sequences at the 5' end of each read, we used umitools [@umitools] to find all reads with a UMI of the pattern NNNNNGGG.
The UMI was cut from the 5' end of the read and added to the name of the sequence, and reads without UMIs were discarded.
We then mapped reads to human genome hg19 (only including chromosomes 1-22, X, and Y, plus the ERCC sequences) with Subread [@Liao2013], discarding non-uniquely mapped reads (option -u).
To obtain gene-level counts, we assigned reads to protein-coding genes (Ensembl GRCh37 release 82) and the ERCC spike-in genes using featureCounts [@Liao2014].
Because the UMI protocol maintains strand information, we required that reads map to a gene in the correct orientation (featureCounts flag -s 1).

In addition to read counts, we utilized the UMI information to obtain molecule counts for the single cell samples only.
We did not count molecules for the bulk samples because they violate the assumptions of the UMI protocol, i.e. they contain too many unique molecules for the 1,024 UMIs to properly tag them all.
First, we combined all reads for a given single cell using samtools [@Li2009].
Next, we converted read counts to molecule counts using UMI-tools ([https://github.com/CGATOxford/UMI-tools][UMI-tools]).
UMI-tools counts the number of UMIs at each read start position.
Furthermore, it accounts for sequencing errors in the UMIs introduced during the PCR amplication or sequencing steps using a "directional adjacency" method.
Briefly, all UMIs at a given read start position are connected in a network using an edit distance of one base pair.
However, edges between nodes (the UMIs) are only formed if the nodes have a 2x difference in reads.
The node with the highest number of reads is counted as a unique molecule, and then it and all connected nodes are removed from the network.
This is repeated until all nodes have been counted or removed.

[fastqc]: http://www.bioinformatics.babraham.ac.uk/projects/fastqc
[UMI-tools]: https://github.com/CGATOxford/UMI-tools

### Calculate the input molecule quantities of ERCC spiked-ins 

According to the information provided by Fluidigm, each of the 96 capture chamber received 13.5 nl of lysis buffer, which contain 1:50,000 Ambion® ERCC Spike-In Mix1 (Life Tech) in our setup. 
Therefore, our estimation of the total spiked-in molecule number was 16,831 per sample. 
Since the relative concentration of each ERCC genes were provided by the manufacture, we were able to calculate the molecule number of each ERCC gene added to each sample. 

### Filtering cells and genes

We performed multiple quality control analyses to detect and remove low quality cells.
In an initial analysis investigating the percentage of reads mapping to the ERCC spike-in controls, we observed that replicate 2 of individual 19098 was a clear outlier.
It appeared that too much ERCC spike-in mix was added to this batch, which violated the assumption that the same amount of ERCC molecules was added to each cell.
Thus, we removed this batch from all of our analyses.

Next, we maintained high quality single cells that passed the following criteria:

*  Only one cell observed per well
*  At least 1581061 mapped reads
*  Less than 37.46674 % unmapped reads
*  Less than 3.223869 % ERCC reads
*  More than 6875.6 genes with at least one read

We chose these criteria based on the distribution of these metrics in the empty wells (95 %).
Lastly, we observed that some wells classified as containing only one cell clustered with multi-cell wells when plotting the number of gene molecules versus the concentration of the samples (Sup Figure. 2A and B) and also the read to molecule conversion efficiency, calculated by total molecule number divided by total read number, of endogenous genes versus that of ERCC (Sup. 2C and D).
We also removed these misidentified wells using linear discriminant analysis.

After filtering, we maintained `r nrow(anno_filter)` high quality single cells 
(NA19098: `r sum(anno_filter$individual == "NA19098")`,
 NA19101: `r sum(anno_filter$individual == "NA19101")`,
 NA19239: `r sum(anno_filter$individual == "NA19239")`).

The quality control analyses were performed using all protein-coding genes (Ensembl GRCh37 release 82) with at least one observed read.
Using the high quality single cells, we further removed genes with low expression levels for downstream analyses.
We removed all genes with a mean log~2~ cpm less than 2, which we further verified that the overall structure of detected genes remain the same (Sup. Figure. 3).
We also removed genes with molecule counts larger than 1024 for the correction of collision probability.
Thus we kept
`r format(sum(grepl("ENSG", rownames(molecules_filter))), big.mark = ",")`
endogenous genes and
`r sum(grepl("ERCC", rownames(molecules_filter)))`
ERCC spike-in genes.

### Normalization

We transformed the single cell molecule counts in multiple steps (Fig. `r fig_main_normalization`).
First, we corrected for the collision probability using a method similar to that developed by Grun et al. [@Grun2014].
Essentially we are correcting for the fact that we did not observe all the molecules originally in the cell.
The main difference between our approach and that of Grun et al. [@Grun2014] is that we apply the correction at the level of gene counts and not individual molecule counts.
Second, we standardized the molecule counts to log~2~ counts per million (cpm).
This standardization was performed using only the endogenous gene molecules and not the ERCC molecules.
Third, we corrected for cell-to-cell technical noise using the ERCC spike-in controls.
For each single cell, we fit a Poisson generalized linear model (GLM) with the log~2~ expected ERCC molecule counts as the independent variable, and the observed ERCC molecule counts as the dependent variable, using the standard log link function. 
Next we used the slope and intercept of the Poisson GLM regression line to transform the log~2~ cpm for the endogenous genes in that cell.
This is analogous to the standard curves used for qPCR measurements, but taking into account that lower concentration ERCCs will have higher variance from Poisson sampling.
Fourth, we removed technical noise between the eight batches (three replicates each for 19101 and 19239 and two replicates for 19098).
We fit a linear mixed model with a fixed effect for individual and a random effect for the eight batches and removed the variation captured by the random effect (see the next section for a detailed explanation).

For the bulk samples, we used read counts even though the reads contained UMIs.
Because these samples contained RNA molecules from millions of cells, we could not assume that the 1,024 UMIs we added could accurately estimate the true number of molecules.
We standardized them to log~2~ cpm.

### Remove batch effect due to within-plate cell-to-cell gene expression variability

Our last normalization step adjusts the transformed log~2~ gene expression levels for cell-to-cell correlation within each C1 plate. The algorithm mimics a method that was initially develops for adjusting within-replicate correlation in microarray data [@Smyth2005]. We assumed that for each gene $g$, cells that belong to the same batch $j$ are correlated, for batches $j = 1, \dots, 8$. The batch effect is specific to each C1 plate and is independent of biological variation across cell lines. 

We then fit a linear mixed model for each gene $g$ that includes a fixed effect of individual cell lines and a random effect for within-batch variation attributed to cell-to-cell correlation within each C1 plate: 

$$ y_{g,ijk} = \mu_{g} + \alpha_{g,i} + b_{g,j} + \epsilon_{g,ijk}, $$

where $y_{g,ijk}$ denotes log~2~ counts-per-million (cpm) for gene $g$ in individual cell line $i$, batch $j$, and cell $k$, and $i = NA19098, NA19101, NA19239$, $j = 1, \dots, 8$, $k = 1, \dots, n_{g,j}$ with $n_{g,j}$ denotes the number of cells in batch $j$; $\mu_g$ describes gene-specific expression level across cells, $\alpha_{g,i}$ quantifies mean gene expression level for individual cell line $i$, $b_{g,j}$ models cell-to-cell correlation in batch $j$, and $\epsilon_{g,ijk}$ describes residual variation in gene expression levels. We assumed $b_{g,j}$ are independent and identically distributed with mean 0 and variance $\sigma^2_{g,b})$ for $j = 1, \dots, 8$.

Next, we applied Least-Squared (LS) method and computed the batch-corrected expression as:

$$ \widehat{y}_{g,ijk} = y_{g,ijk} - \widehat{b}_{g,j}, $$

where $\widehat{b}_{g,j}$ is the LS estimate of batch-effect specific to replicate $j$. The computations in this step are performed with the gls.series function of the limma package [@limma].


### Analysis of cell-to-cell transcriptional variation: calculating the measure of noise 

We computed a measure of transcriptional variability noise that is independent of RNA abundance levels across individuals [@Kolodziejczyk2015, @Newman2006]. First, squared coefficients of variation (CVs) for each gene were computed for each individual and across cell lines, using molecule counts after collision correction and transformations. Then we computed the distance of individual-specific CVs to the rolling median of global CVs among genes that have similar RNA abundance levels. These transformed individual CV values are referred to our noise measure. We outline the computations as follows:

1. Compute squared CVs of molecule counts in individual cell lines and across individuals. 

2. Order genes by the global average molecule counts.

3. Starting from the genes with the lowest global average gene expression level, for every sliding window of 50 genes, subtract log10 median squared CVs from log10 squared CVs of each cell line, and set 25 overlapping genes between windows. The computation was performed with the rollapply function of the R zoo package [@Zeileis2005]. 



### Identification of differential CV genes

To identify differential DM between individuals, we computed median absolute deviation (MAD) - a robust and distribution-free dissimilarity measure for gene $g$: 

$$ MAD_{g} = Median_{i= 1,2,3} \left| DM_{g,i} - Median{DM}_{g} \right|$$

. Large values of $MAD_{g}$ suggest large deviation from the median of the DM values. We then computed permutation-based p-values for each gene based on 12,192 permutation - equals to the number of genes included in the filtered and transformed data. In each permutation, the cell line labels are shuffled between cells, and the permutation-based p-value is 

$$ \frac{1}{B} \sum_{B=1}^{12,192} \mathbb{I} \left( MAD_{g} > MAD_{g,B} \right),$$

where $MAD_{g,B}$ is the gene $g$ MAD value in permutation $B$, and $\mathbb{I}$ equals 1 if $\left( MAD_{g} > MAD_{g,B} \right)$ and 0 otherwise.  


### Gene enrichment analysis

We used the Gene Ontology Consortium platform [@GO2015] to identify over-represented gene sets. The Gene Ontology Consortium platform is supported by the PANTHER Classification System [@Mi2013], which maintains up-to-date GO annotations. 


### Data and code availability

The data have been deposited in NCBI's Gene Expression Omnibus [@Edgar2002] and are accessible through GEO Series accession number GSE77288 ([http://www.ncbi.nlm.nih.gov/geo/query/acc.cgi?acc=GSE77288][geo]).
The code and processed data are available at [https://github.com/jdblischak/singleCellSeq][repo].
From the beginning of the project, we performed this research in the open.
The results of our analyses are viewable at [https://jdblischak.github.io/singleCellSeq/analysis][site].

[geo]: http://www.ncbi.nlm.nih.gov/geo/query/acc.cgi?acc=GSE77288
[repo]: https://github.com/jdblischak/singleCellSeq
[site]: https://jdblischak.github.io/singleCellSeq/analysis

