---
bibliography: refs.bib
---

```{r chunk-options, include=FALSE}
source("chunk-options-paper.R")
```

## Results
### Study design and data quality control
Single-cell transcriptomics has recently emerged as a powerful technology to explore stochastic gene expression by assaying steady state transcriptomes from numerous individual cells with minimum losses in cDNA synthesis and high reproducibility.
Specifically, randomly labeling individual transcripts with UMIs at the reverse transcription step prior to amplification of mRNA provides the power to measure absolute molecule counts per gene by eliminating the amplification bias [@Islam2014; @Grun2014; @Jaitin2014], whereas including the spike-in quantification standards of known abundance in the samples allows estimation of technical variation [@Brennecke2013; @Grun2014; @Ding2015; @Vallejos2015].
To be able to quantify mRNA molecule numbers in single cells with exceptional accuracy, we collected single cell RNA-seq (scRNA-seq) data from three Yoruba iPSC lines (19098, 19101, and 19239), with the additions of both 5-bp random sequence UMIs and ERCC spike-ins, using C1 microfluidic plates from the Fluidigm system followed by sequencing on the Illumina HiSeq system.
Importantly, for each of the Yoruba lines, we performed three independent C1 collections to more accurately capture the characteristics of technical variability across C1 replicates and then to further remove them from downstream analysis.
The three C1 replicates of each individual were collected using cells with different passage numbers, and hence they were also biological replicates. 
Moreover, to estimate how well the scRNA-seq can reassemble the RNA-seq results from population bulk samples, each individual C1 collection was accompanied by a collection of bulk sample using the same reagents.

#### Removal of low quality sample
Given the low abundance of mRNA content and the high vulnerability of single cells, scRNA-seq can be intensively affected by various technical artifacts arising from cell isolation and capture, mRNA capture efficiency, library preparation, and even sequencing procedures.
As a result, our quality control of scRNA-seq libraries was performed in multiple steps for generating reliable analysis results.
After removing the samples that were not one-cell libraries as reported by the visual inspection, we first filtered out the one-cell libraries that were similar to no-cell libraries judging from the number of total mapped reads, the percentage of unmapped reads, the percentage of ERCC spike-in reads, and the number of genes detected. 
Next, we identified several one-cell libraries that had high possibility to be multiple-cell libraries based on the number of total molecules, the concentration of cDNA amplicons, and the read-to-molecule conversion efficiency defined as the number of total molecules divided by the number of total reads.
we maintained `r nrow(anno_filter)` high quality single cells 
(NA19098: `r sum(anno_filter$individual == "NA19098")`,
 NA19101: `r sum(anno_filter$individual == "NA19101")`,
 NA19239: `r sum(anno_filter$individual == "NA19239")`).

#### Removal of genes with lower expression
Besides carrying out quality control at the library level, we next executed quality control at gene detection level to deal with the problem of genes possibly being missed during sequencing (referred to as gene “dropout”). 
Evidently, we observed that the mean of read counts and the mean of molecule counts of each gene across cells were highly correlated for both ERCC and endogenous genes (r = and r = , respectively), yet they were diverged at genes with lower expression level, indicating high detection noise of lowly expressed genes. 
We therefore applied a cutoff to filter out genes with low molecule counts and kept `r format(sum(grepl("ENSG", rownames(molecules_filter))), big.mark = ",")`
endogenous genes and
`r sum(grepl("ERCC", rownames(molecules_filter)))`
ERCC spike-in genes for the follwing analysis.
When looking at the proportion of genes detected in the high quality cells before and after the filtering, we found that the absolate numbers of the proportion were increased while keeping the relative relationship across cells, providing us with the confidence that the removed genes were truly the dropouts.

#### ERCC spike-ins 
To estimate the mRNA capture efficiency, we first calcutated the input molecule numbers of ERCC spike-in RNA transcripts and then compared the imput quantities with the observed quantities in high quality cells wihtout filtering of genes. 
While the observed levels of ERCC spike-ins strongly correlated with the known input quantities (r = 0.9932), the capture efficiency of ERCC spike-ins, defined as the fraction of total input molecules being successfully detected in each high quality cells, has the avarage of 9.119 %.

Currently, ERCC spike-ins are being used to identify low-quality single cells, to measure cell-cell technical variability, and also to indicate relative cellular mRNA content, all of which are under the assumption that each sample contains exact quatities of ERCC spike-ins.
However, when looking at total ERCC molecule versus total gene molecules, percentage of ERCC molecule versus total gene molecules, ERCC capture efficiency versus total gene molecules, and ERCC capture efficiency versus percentage of ERCC molecule, we observe a clear differences across the three individuals but not across C1 replicates in all of the above.
The fact that the C1 replicates of the same individuals were consistent in these analysis suggested that this "individual effect" cannot be arising simply from the cell-cell technical variability including PCR amplification, library prepration, and sequencing. 
Instead, some unidentified factors inherent from the individual difference to the distribution of gene expression in the single cells from these three individuals is consistently affecting the molecule counts.
For the purpose of measuring cellular mRNA content in cells, even though there was clearly a relationship between the percent ERCC molecules and the total gene molecules, the confidence in the accuracy of this estimate was inevitably dampened by the individual effect of ERCC spike-ins.
We thus decided not to explicaity estimate the technical variability and cellular content in this study. 
 
#### Batch effect of molecule counts
To avoid batch effect from the sequencing, we followed the general guidelines for bulk samples by equally distributing the nine C1 replicates of the three individuals onto four flowcells to acquire sufficient sequencing depth. 
We didn't observe any obvious counfouder that need to be corrected while carrying out quality control at the level of sequencing lane.
Not surprisingly, there was a observable batch effect of C1 collection according to the PCA results of endogenous gene expression using molecule counts.
In addition to the C1 collection itself, we noticed an unexpected batch effect arising from the difference of read-to-molecule conversion efficiency, computed from the total molecule counts divided by the total reads counts of endogenous genes in each cell, between different C1 replicates.
Intriguingly, not only the difference of conversion efficiency between C1 replicates was significant, the difference between the three individual was also siginificant, which further strengthens our conclusion that difference in certain unknown factors across these individuals are affecting the molecule counts of both ERCC spike-ins and endogenous genes in single cells.

### Subsampling
It has been shown in separate studies that as few as 10 single cells can accurately reflect gene expression from a pooled population [@Pollen2014], and 2 million reads can already reach saturation point of gene detection [@Wu2014].
However, no systematic subsampling at the levels of sequencing depth, including read counts and molecule counts, along with the consideration of cell number has been reported. 
To assess the accuracy of single-cell RNA-seq, we collected data from bulk sample (cells in bulk) together with each C1 prep from the same population of cells using the same master mixes including lysis mix, RT mix, and PCR mix. 
The results from subsampling in high quality cells indicate that sequencing depth of 2 million reads, which corresponds to ~100,000 molecule counts of protein coding genes, with 75 single cells is sufficient to capture the majority of genes expressed in the whole cell culture (~90 %). 
Using the same subsampling strategy, the pattern of the correlation between the mean expression levels of subsampled single cells and the bulk sample also coincides with the pattern of detected gene numbers in all three individuals (r > 0.93). 
Most importantly, the cell-cell variance from the same subsampling cutoffs also accurately represents all the collected single cells from one individual (r > 0.95), suggesting that a sample size of 75 high quality cells is sufficient for capturing cell-cell variability.

We also collected scRNA-seq data from Lymphoblastoid Cell Lines (LCLs) in a trial run, in which libraries from one single cell was sequenced individually on one sequencing lane. 
Since all the library materials were used in these samples, the data collected from this single cell libraries were possibly the most we could obtain using this protocol. 
The subsampling results from these libraries showed that with the sequencing depth of 50 million reads, the numbers of endogenous genes started to plateau, whereas the total molecule counts were still increasing, suggesting that we have not yet reach the satuation point of molecule counts. 

### A novel normalization for scRNA-seq data analysis 
To remove the technical variability arosed from all the factors mentioned above without explicitly estimating the technical variability of ERCC spike-in molecule counts, here, we developed a pipeline which is applicable for studying cell-cell variability aross multiple biological conditions with multiple C1 collections.  

#### Correct for collision probability
Due to the stochasticity of the UMI sampling process, not all molecules will be tagged with an unique UMI and sequenced. 
Thus, after removing genes with lower expression level in high quality cells, we first computed the molecule number of each gene by applying a mathematical correction to account for the effect of random counting statistics termed collision probability [@Fu2010]. 
This step also removed the genes with molecule counts larger than 1024, which in our case were all mitochondrial genes, becuase of the alternative transcriptional start sites. 

#### Standardization (cpm)
We then checked if the sequecning depth played a role in total molecule counts per sample and found that the total read counts highly correlated with not only the total molecule counts but also the PC1 of overall gene expression.
Not surprizingly, same results were also observed when we investigated the effects of sequencing depth on total molecule counts and PCA of ERCC spike-ins.
Therefore, we performed standardization separatly on endogenous genes and ERCC spike-ins within each sample using the unit of cpm (counts per million). 

#### Linear transformation
To eliminate the cell-to-cell technical variability coming from the differences of mRNA capture efficiency within one C1 collection, we next performed linear transforamtion on the log transformed molecule numbers using a fitted linear model generating from the observed and the expected molecule qualities of ERCC spike-ins in each sample.
In other words, we used ERCC spike-ins to make a linear standard curve for each sample.  
The PCA of gene expression after linear transformation showed that PC2 seperated not only the three individuals but also the three C1 replicates of each individual, similar to the results of filtered raw molecule data, suggesting the batch effect from the C1 replicates were yet to be removed.

shrunk linear transforamtion???

#### Mixed model for batch effect correction
Since the linear transformation with the ERCC spike-ins was not sufficient to get rid of all the unwanted technical variation, we used a mixed effect model to correct for batch effects across C1 replicates by adapting the algorithm from limma for estimating variance components due to random effects [@Smyth2005]. 
Specifically, this analysis operated under the assumption that biological replicates (or C1 replicates within an individual in this case) share similar correlation across genes. 
The results of PCA on the cleaned data showed that the samples from different C1 replicates of the same individual largely overlapped, indicating that we have successfully removed large proportion of the unwanted technical variation arcoss C1 replicates using out normalization method.
Furthermore, we were also confident to conclude that individual 19239 was the outlier of the three individuals.

### Cell-cell variability of gene expression

#### Genes with differential cell-cell variability across individuals


1. study design and data collection (Fig. `r fig_main_qc`, sup fig.1 and 2)
2. cpm (Fig. `r fig_main_cpm`, sup fig.3 and 4)
3. subsample (Fig. `r fig_main_subsample`)
4. remove batch effect (Fig. `r fig_main_normalization` and sup fig.5)
5. gene expression noise (Fig. `r fig_main_noise` and sup fig.6)
