	---
bibliography: refs.bib
---

```{r chunk-options, include=FALSE}
source("chunk-options-paper.R")
```

## Results
### Study design and quality control
To quantify mRNA molecule numbers in single cells with exceptional accuracy from different genetic backgrounds, we collected single cell RNA-seq (scRNA-seq) data from three YRI iPSC lines (19098, 19101, and 19239) with the additions of ERCC spike-in controls and 5-bp random sequence UMIs using the Fluidigm C1 microfluidic system followed by sequencing on the Illumina HiSeq system.
Importantly, for each of the YRI lines, we performed three independent C1 collections to more accurately capture the characteristics of technical variability across C1 replicates and then to further remove them from downstream analysis (Figure. 1A).
The three C1 replicates of each individual were collected on different dates using cells of different passage numbers (Sup Table. 1A), and hence they were also biological replicates. 
Moreover, each individual C1 collection was accompanied by a collection of bulk sample using the same reagents to estimate how well the scRNA-seq can recapitulate the RNA-seq results from population bulk samples.
For the sequencing, we followed the general guidelines of traditional bulk samples to avoid sequencing batch effect by equally distributing the nine C1 replicates of the three individuals onto four flowcells to acquire sufficient sequencing depth (Sup Table. 1B). 
We didn't observe any obvious technical effects due to sequencing lane or flow cell that confounded the inter-individual and inter-replicate comparisons (data not shown).

#### Removal of low quality sample
Given the low abundance of mRNA content and the high vulnerability of single cells, scRNA-seq can be intensively affected by various technical artifacts arising from cell isolation, mRNA capture efficiency, library preparation, and even sequencing procedures.
As a result, our quality control of scRNA-seq libraries was performed in multiple steps for generating reliable analysis results.
Unfortunately, the second C1 collection of individual 19098 contained extremely high ERCC contents compared to all the others (Sup Figure. 2A and B), suggesting that there might be a pippetting error during lysis mix preparation which led to excess ERCC molecule counts.
To avoid potential unidentified issues caused by the error, we decided to exclude this C1 collection from all our analyses.
Next, we removed all the samples that did not contain exactly one cell based on visual inspection on the C1 microfluidic plates.
Then we filtered these one-cell samples based on the following metrics: the number of total mapped reads, the percentage of unmapped reads, the percentage of ERCC spike-in reads, and the number of genes detected (Sup Figure. 2).
Instead of choosing arbitrary cutoffs, we defined the cutoff for each of these metrics based on the 95th percentile of the distribution for the 18 no-cell libraries (Figure. 1B-E and Sup Figure. 2C-F).
Furthermore, we identified and removed several one-cell libraries that had high possibility of being multiple-cell libraries based on the number of total molecules, the concentration of cDNA amplicons, and the read-to-molecule conversion efficiency defined as the number of total molecules divided by the number of total reads (Sup Figure. 3).
We maintained `r nrow(anno_filter)` high quality single cells 
(NA19098: `r sum(anno_filter$individual == "NA19098")`,
 NA19101: `r sum(anno_filter$individual == "NA19101")`,
 NA19239: `r sum(anno_filter$individual == "NA19239")`).

#### Removal of genes with lower expression
Besides carrying out quality control at the library level, we next executed quality control at gene detection level to deal with the problem of genes possibly being missed during sequencing (referred to as gene “dropout”) [@Brennecke2013]. 
Evidently, our results showed that the mean of read counts and the mean of molecule counts of each gene across cells were highly correlated (r = 0.9189), yet they were diverged at genes with lower expression level, indicating high detection noise of lowly expressed genes (Figure. 1F). 
We therefore applied a cutoff to filter out genes with low molecule counts and kept `r format(sum(grepl("ENSG", rownames(molecules_filter))), big.mark = ",")`
endogenous genes and
`r sum(grepl("ERCC", rownames(molecules_filter)))`
ERCC spike-in genes for the following analysis.
When looking at the proportion of genes detected in the high quality cells before and after the filtering, we found that the numbers of the proportion were increased while keeping the relative relationship across cells both within and between batches (Sup Figure. 3A and B), which provides the evidence that the removed genes were the possible dropouts.
To more rigorously analyze the effect of removing lowly expressed genes, we performed PCA on gene expression of ubiquitously expressed genes (genes detected in all cells) using the standardized molecule counts and found that the filtering had no effect on the correlation of PC1 and the proportion of genes detected, suggesting that this filtering step preserve the overall gene expression pattern in the cells (Sup Figure. 3C and D). 

#### ERCC spike-ins capture efficiency
To estimate the mRNA capture efficiency, we first calculated the input molecule numbers of ERCC spike-in RNA transcripts and then compared the input quantities with the observed quantities in high quality cells without filtering of genes. 
While the observed levels of ERCC spike-ins strongly correlated with the known input quantities (r = 0.9914), the capture efficiency of ERCC spike-ins, defined as the fraction of total input molecules being successfully detected in each high quality cells, has the average of 6.094 % (Figure. 1G).

### Subsampling
It has been shown in separate studies that as few as 10 single cells can accurately reflect gene expression from a pooled population [@Pollen2014], and 2 million reads can already reach saturation point of gene detection [@Wu2014].
However, no systematic subsampling at the levels of sequencing depth, including read counts and molecule counts, along with the consideration of cell number has been reported. 
To assess the accuracy of single-cell RNA-seq, we collected data from bulk sample (cells in bulk) together with each C1 prep from the same population of cells using the same master mixes including lysis mix, RT mix, and PCR mix (Figure. 2A). 
The results from subsampling in high quality cells indicate that sequencing depth of 2 million reads, which corresponds to ~100,000 molecule counts of endogenous genes, with 75 single cells is sufficient to capture the majority of genes expressed in the whole cell culture (~90 %, Figure. 2A). 
Using the same subsampling strategy, the pattern of the correlation between the mean expression levels of subsampled single cells and the bulk sample also coincides with the pattern of detected gene numbers in all three individuals (r > 0.93, Figure. 2B). 
Most importantly, the cell-cell variance from the same subsampling cutoffs also accurately represents all the collected single cells from one individual (r > 0.95, Figure. 2C), suggesting that a sample size of 75 high quality cells is sufficient for capturing cell-cell variability.

We also collected scRNA-seq data from Lymphoblastoid Cell Lines (LCLs) in a trial run, in which libraries from one single cell was sequenced individually on one sequencing lane (Sup Figure. 4). 
Since all the library materials were used in these samples, the data collected from this single cell libraries were possibly the highest sequencing depth that we could obtain using this protocol. 
The subsampling results from these libraries showed that, with the sequencing depth of 50 million reads, the numbers of endogenous genes started to plateau (Sup Figure. 4A and B), whereas the total molecule counts were still increasing (Sup Figure. 4D and F), suggesting that we have not yet reach the saturation point of molecule counts. 

### Technical and biologial variabiltiy 
One of the most challenging sources of unwanted technical variability in scRNA-seq experiments using the C1 system is the batch effects that take place when cells from different biological conditions are cultured, captured, and sequenced independently.
Most, if not all, published studies using the UMI-based method have been collecting only one C1 plate per biological condition. 
In other words, the potential sources of the batch effects are yet to be characterized, which motivates us to look into it from the following aspects to better distinguish technical variability from biological variability.

#### Cellular mRNA contents and cell cycle phase
When UMIs are used in scRNA-seq, cell-to-cell difference in total molecule counts could be indicative of variation in cellular mRNA contents and therefore be biological meaningful to a certain extent if the technical variability is negligible.
However, as expected, our results of plotting the sample concentration and the total number of molecules not only showed that samples with more cells have lower concentration (Sup Figure. 5A and B), but also indicated a non-negligible amount of variability across samples of one cell (Sup Figure. 5C and D).
Similar results were also found when we classified cell cycle phase of each cell based on gene expression using molecule counts [@Macosko2015] and compared the total molecule counts and the total read counts of different cell cycle phases (Sup Figure. 6).
Even though cells in G2 and G2/M phases have higher total molecule/read counts than cells in G1 and S phase in a few but not all C1 replicates, the differences are not significant due to the large variances, suggesting that the technical variability is not trivial.

#### ERCC spike-in controls
Currently, ERCC spike-ins are being used to identify low-quality single cells, to measure cell-cell technical variability, and also to indicate relative cellular mRNA content, all of which are under the assumption that each sample contains exact quantities of ERCC spike-ins.
However, when looking at total ERCC molecule versus total gene molecules, percentage of ERCC molecule versus total gene molecules, ERCC capture efficiency versus total gene molecules, and ERCC capture efficiency versus percentage of ERCC molecule, we observed a clear differences across the three individuals but not across C1 replicates in all of the above (Sup Figure. 7).
The fact that the C1 replicates of the same individuals were consistent in these analysis suggested that this "individual effect" cannot be arising simply from the cell-cell technical variability including PCR amplification, library preparation, and sequencing. 
Instead, some unidentified factors inherent from the individual difference to the distribution of gene expression in the single cells from these three individuals is consistently affecting the molecule counts.
For the purpose of measuring cellular mRNA content in cells, even though there was clearly a relationship between the percent ERCC molecules and the total gene molecules (Sup Figure. 7B), the confidence in the accuracy of this estimate was inevitably dampened by the individual effect of ERCC spike-ins (Sup Figure. 7D and F).
We thus decided not to explicitly estimate the technical variability and cellular content based on ERCC spike-ins in this study.

#### Reads to molecule conversion
While the molecule counts have been treated as the absolute numbers of mRNAs in previous studies [@Islam2014], we found that the total number of molecules per cell was dependent on the sequencing depth (Figure. 3A and C) and argued that standardization of molecule counts is required.
In addition, we noticed an unexpected batch effect arising from the difference of read-to-molecule conversion efficiency, which is computed from the total molecule counts divided by the total reads counts of endogenous genes in each cell, across C1 replicates (Figure. 2B and D).
Intriguingly, not only the difference of conversion efficiency across C1 replicates was significant, the difference between the three individuals was also significant, which further strengthens our conclusion that differences in certain unknown factors across these individuals are affecting the molecule counts of both ERCC spike-ins and endogenous genes in single cells.

### A novel normalization for scRNA-seq data analysis 
To remove the technical variability arose from all the factors mentioned above without explicitly estimating the technical variability of ERCC spike-in molecule counts, here, we developed a pipeline that is applicable for studying cell-to-cell variation of gene expression from multiple biological conditions with multiple C1 collections (Figure. 4). 

#### Correct for collision probability
Due to the stochasticity of the UMI sampling process, not all molecules will be tagged with an unique UMI and sequenced. 
Thus, after removing genes with lower expression level in high quality samples, we first computed the molecule number of each endogenous gene by applying a mathematical correction to account for the effect of random counting statistics termed collision probability [@Fu2011]. 
This step also removed the genes with molecule counts larger than 1024, which in our case were all mitochondrial genes (data not shown). 

#### Standardization (counts-per-million)
Next, we incorporated a standardization step by computing log transformed counts-per-million (cpm) of molecule counts.  
As predicted from our previous observation of the sequencing depth (Figure. 2A and C), we found that the total read counts per sample were highly correlated with not only the total molecule counts but also with the first principal component of endogenous gene expression (Figure. 4A). 
After removal of the variance caused by sequencing depth, the difference between C1 replicates, or the batch effect, became dominant (Figure. 4B).

#### Poission regression of ERCC counts
To account for the cell-to-cell technical variability coming from the differences of mRNA capture efficiency across samples within the C1 collection, we next performed Poisson GLM (generalized linear model) normalization using the observed and the expected molecule counts of ERCC spike-ins, without correction for collision probability and also without standardization, in each sample.
The PCA of gene expression after normalization showed that PC1 and PC2 separated not only the three individuals but also the three C1 replicates of each individual, suggesting that the batch effect from the C1 replicates were yet to be removed (Figure. 4C).

#### Mixed model for batch effect correction
Not surprisingly, there was a observable batch effect of C1 collection according to the PCA results of gene expression even after both standardization and normalization. 
To directly remove unwanted technical variation between C1 plates, we modeled between well correlation in gene expression within C1 plates. 
We assumed that C1 plates share similar technical variation due to gene expression similarity between cells on the same plate. 
The results of PCA on the corrected data showed that the samples from different C1 replicates of the same individual largely overlapped (Figure. 4D), indicating that we have successfully removed large proportion of the unwanted technical variation across C1 replicates. 

### Cell-to-cell variation of gene expression
Once we removed the unwanted technical variability from scRNA-seq data as described above, we next focused on analyzing cell-to-cell variation of gene expression at the transcriptional level.
We first investigated the biological gene expression variability, or gene expression noise, across single cells of each individual, and then we identified the genes that present differential expression noise across the three individuals.

#### Gene expression noise at transcriptional level
Examining endogenous transcriptional variability in iPSCs as a function of RNA abundance across C1 replicates, we found that coefficient of variation (CV) of molecule counts among endogenous genes and ERCC spike-in genes suggested similar expression variability patterns that track the expected Poisson distribution (Sup Figure. 8A), which are consistent with previous studies of mouse ESCs (embryonic stem cells) [@Islam2014; @Brennecke2013]. 
Due to the observation that transcriptional variability was correlated with mean gene abundance, we adjusted our gene variability measure (CV) by gene abundance averaged across single cells (see the Methods section, Sup Figure. 8B), a similar method used to investigate gene expression noise in mouse ESCs [@Kolodziejczyk2015]. 
Importantly, to investigate the effect of the unobserved, or the dropout, measurements, we measured the gene expression noise with or without the cells where a given gene was not detected/expressed [@Shalek2013]. 
While including all single cells passed quality filtering, we found that 120 genes were ranked among the top 1,000 noisy genes in all three indivdiuals (Figure. 5A), whereas 891 gnees were ranked among the top 1,000 highly expressed genes (Figure. 5B), indicating the presence of differential cell-to-cell variation in gene expression across individuals. 

We next focused on gene expression noise of the expressed cells - cells that were detected by one or more molecule counts in each gene, using the same noise measure as before. 
The proportion of undetected cells varied between genes; the median ranged between 21 to 30 percent in the three indivduals. Expectedly, gene abudance and variance of the expressed cells were associated with percent of undetected cells (supp). 
The variabilty measure coefficient of variation was not associated with percent of undetected cells (Spearman's correlation = .04) across the three individuals (supp), supporting that CV is not confounded with the dynamic range of gene abundance. In addition, significant individual difference in proportion of undetected genes were found in more than 10 percent of the genes (1,958 genes).
After excluding the undetected cells on a per gene basis, we also found the three individuals shared 887 highly expressed genes and 95 highly noisy genes, similar to the results described earlier while including all quality single cells. 

#### Differential gene expression noise across individuals 
Next, we identified differential noisy genes between individuals using the median of absolute deviation (MAD) to quantify the degree of dissimilarity between individual noise measures. 
While only the expressed cells were included to compute individual CVs adjusted for gene abundance, our permutation-based test indicated that 680 genes had statistically significant differences in gene expression noise between individuals, at an alpha-level of $10^{-5}$ (Sup Table. X of gene list, Sup. Figure. 9). 
Enrichment analysis suggested that these genes were enriched for genes involved in protein translation, protein disassembly, and various biosynthetic processes (Sup Table. X). 
We found two highly expressed pluripotency genes *DNMT3B* and *NR6A1* among the differential noisy genes (Figure. 5A and B, respectively), both of which were detected in all single cells (Sup. Figure. 10).
owever, majority of the pluripotency genes were not detected in all the cells and exhibited no differential noisy papttern, including key pluripotency genes *Nanog* and *ZFP42* (Figure. 5C and D, respectively), indicating that the pluripotent state was tightly regulated.
Notably, despite the association between gene abundance and proportion of undetected genes, we found no associations between the indvidual differences, and similarly for individual differences in expression noise and proportion of undetected genes (Sup. Figure. 11). 
