---
bibliography: refs.bib
---

```{r chunk-options, include=FALSE}
source("chunk-options-paper.R")
```

## Results
### Study design and quality control
We collected single cell RNA-seq (scRNA-seq) data from three YRI iPSC lines using the Fluidigm C1 microfluidic system followed by sequencing on the Illumina HiSeq system (see Methods). 
We added ERCC spike-in controls to each sample, and used 5-bp random sequence UMIs to allow for the direct quantification of mRNA molecule numbers. 
For each of the YRI lines, we performed three independent C1 collections.
This study design (Figure 1A) allows us to estimate error and variability associated with the technical processing of the samples, independently from the biological variation between individuals.  

The three C1 replicates of each individual were collected on different dates using cell cultures of different passage numbers (Sup Table 1A). 
Thus these technical replicates additionally measure the variation from growing the cells.
Each replicate of C1 collection was accompanied by a collection of matching bulk sample using the same reagents. 
This property allowed us to globally characterize the regulatory variation between replicates of the same cell culture, as well as to estimate how well scRNA-seq data can recapitulate the RNA-seq results from population bulk samples. 

In what follows, we describe data as originating from different samples when we refer to data from distinct wells of each C1 collection. 
Generally, data from each sample correspond to data from a single cell. 
In addition, we describe data as originating from different replicates when we refer to all samples from a given C1 collection, and from different individuals when we refer to data from all replicates of a given genetically distinct iPSC line.

```{r calculate-seq-depth}
library("dplyr")
total_counts <- read.table("../data/total-counts.txt", header = TRUE,
                           sep = "\t", stringsAsFactors = FALSE)
depth_stats <- total_counts %>%
  filter(stage == "raw", well != "bulk") %>%
  group_by(individual, replicate, well) %>%
  summarize(counts_per_sample = sum(counts)) %>%
  ungroup() %>%
  summarize(mean = mean(counts_per_sample), sd = sd(counts_per_sample),
            min = min(counts_per_sample), max = max(counts_per_sample))
```

We obtained an average of `r format(depth_stats$mean / 10^6, digits = 3)` +/-
`r format(depth_stats$sd / 10^6, digits = 3)` million sequencing reads per sample
(range `r format(depth_stats$min / 10^6, digits = 3)`-`r format(depth_stats$max / 10^6, digits = 3)` million reads).
We processed the sequencing reads using a standard alignment approach (see Methods). 
Given the low abundance of mRNA content and the high vulnerability of single cells and the low amount of starting material, scRNA-seq can be highly affected by various technical artifacts arising from cell isolation, mRNA capture efficiency, library preparation, and sequencing procedures. 
We thus performed multiple quality control analyses.

As a first step, we estimated the proportion of ERCC spike-in reads from each sample. 
We found that, across samples, sequencing reads from the second replicate of individual 19098 include unusually high ERCC content compared to all other samples and replicates (Sup Figure. 2A and B). 
We concluded that a pipetting error led to excess ERCC content in this replicate and we excluded the data from all samples of this replicate in subsequent analyses.

With the exception of the excluded samples, data from all other replicates seem to have similar global properties (using general metrics; Figure. 1C-E and Sup figure 2). 
We therefore proceeded by focusing on the assumption that data from each sample correspond to data from an individual cell. 
Before the processing of the individual cell samples, we performed visual inspection on the C1 microfluidic plates (see Methods). 
Based on that visual inspection, we have excluded sequencing data from 21 samples that did not contain any cell, and 54 samples that contained more than one cell (across all replicates). 
We then filtered data from the remaining samples based on the number of total mapped reads, the percentage of unmapped reads, the percentage of ERCC spike-in reads, and the number of genes detected (Figure. 1B-E).
We used data-driven inclusion cutoffs for each metric, based on the 95th percentile of the distribution for the 21 libraries that were amplified from samples that did not include a cell based on visual inspection (Sup Figure. 2C-F). 

Visual inspection of the C1 microfluidic plate is an important quality control step, but it is not infallible.
We thus identified and removed data from 15 additional samples that were classified as originating from a single cell based on visual inspection, but whose data were more consistent with a multiple-cell origin based on the number of total molecules, the concentration of cDNA amplicons, and the read-to-molecule conversion efficiency (defined as the number of total molecules divided by the number of total reads; Sup Figure 3). 
At the conclusion of these quality control analyses and exclusion steps, we retained data from 564 high quality samples, which correspond, with reasonable confidence, to 564 single cells, across eight replicates from three individuals (Table 1).

Our final quality check focused on the different properties of sequencing read and molecule count data.
We considered data from the 564 high quality samples and compared gene specific counts of sequencing read and molecules.
We found that while gene-specific read and molecule counts are exceptionally highly correlated when we considered the ERCC spike-in data (r = 0.99; Figure. 1G), these counts are somewhat less correlated when data from the endogenous genes are considered (r = 0.92). 
Moreover, the gene-specific read and molecule counts correlation is noticeably lower for genes that are expressed at lower levels. These observations concur with previous studies [@Brennecke2013] as they underscore the importance of using UMIs in single cell gene expression studies.

Beyond quality control of the samples, we also estimated the number of molecule counts required for representing the bulk samples and also for capturing the cell-to-cell variation in gene expression. 
To this end, we conducted a systematic subsampling at the levels of sequencing depth, including both read and molecule counts, along with the consideration of cell numbers, using our high quality samples (Figure. 2).
The results showed that sequencing depth of 2 million reads, which corresponds to ~100,000 molecule counts, with 75 single cell samples is sufficient to capture the majority of genes expressed (~90 %, Figure. 2A), as well as to accurately reflect mean expression levels in the whole cell culture (r > 0.93, Figure. 2B). 
More importantly, we found that the cell-cell variance from the same subsampling cutoffs can also accurately represent all the collected single cell samples (r > 0.95, Figure. 2C).

### Is there a need to standardize molecule-counts?  
Standard analysis of single cell gene expression data often relies on the implicit assumption that molecule-count data, collected by using single cell UMI-based approaches, can be thought of as an absolute estimate of the total number of RNA molecules in the cell [@Islam2014]. 
In other words, overall differences in RNA molecule-counts between cells may indicate differences in the amount of RNA content across cells, which generally correspond to differences in cell size when all cells are of the same type [@Padovan-Merhar2015].
However, this conjecture can only be correct if all RNA molecules in individual cells are being amplified (‘captured’) and sequenced or if the capturing rate and sequencing of RNA molecules is highly similar across individual cells.

Our study design allows us to carefully examine this notion. 
To do so, we considered the correlation between sample concentrations (measured at the library stage; see Methods) and total molecule-counts per sample, summed over all detected endogenous genes (between 30,408 to 113,832 molecules per cell; Sup. Figure. 4A).
We only included data from individuals 19101 and 19239 in this analysis, because sample concentration measurements from individual 19098 were not available.
Consistent with previous reports, we observed a correlation between total molecule-counts in single cells and the measured sample concentration (Sup. Figure. 4B). 
Yet, we also observed that the relationship between total single cell molecule-counts and concentration is significantly different across individuals (likelihood ratio test (LRT); p < 0.001), as well as across C1 replicates from the same individual (F-test; p < 0.01, See Method). 
This observation indicates that it is challenging to infer the sample concentration (which is a proxy for the absolute total number of mRNA molecules per cell) based on the observed number of molecules. 
While a trend exists, there is much variation, both within and across C1 replicates. 
As a consequence, it would be difficult to correctly infer the relative size of single cells based on the observed number of molecules.

We investigated this property further by inferring the cell cycle phase of single cells based on their gene expression profiles [@Macosko2015; @Whitfield2002] using molecule counts (see Methods).
We then considered the correspondence between the inferred cell cycle phase and the observed total molecule-counts in single cells. 
It is expected that cells that are inferred to be in G2 and G2/M phases (larger cells), would have higher total molecule-counts [@Vallejos2015], which is indeed our general observation (Sup. Figure 5A and 5B). 
However, variation from this pattern can clearly be seen across single cell collections from different individuals, as well as across different C1 replicates from the same individual (Sup. Figure 5). 
We conclude that classification of cell cycle phases based on observed total molecule-counts in single cells would be associated with a high error rate.

Our observations suggest that total molecule-counts in single cells are a poor proxy for the total amount of RNA in the cell. 
It therefore follows that molecule-counts should not be used as an absolute estimate of gene expression levels, but rather should be standardized, much like read counts. 
Indeed, even if one assumes that the molecule ‘capturing rate’ (the probability that a molecule is sampled) is similar across single cells, C1 replicates, and individuals, it is reasonable to assume that not all molecules in each sample are sequenced.  
Indeed, even with 50M sequencing reads (an order of magnitude more than the number of reads sequence in typical single cell experiments), while the numbers of detected endogenous genes starts to plateau (Sup Figure. 6A and B), the number of newly identified molecules is still increasing (Sup Figure. 6D and F).

### Is there a batch effect associated with UMI-based single cell data?
It is reasonable to assume that batch effects impact single cell gene expression data. 
Our previous analysis already suggests that differences in the relationship between total molecule-counts and sample concentration exist not only between individuals, but also between C1 replicates of samples from the same individual (Sup. Figure. 4).

In the context of the C1 platform, typical study designs make use of a single C1 plate (batch/replicate) per biological condition. 
In that case, it is impossible to distinguish between biological and technical effects associated with the independent capturing and sequencing of each C1 replicate. 
We designed our study with multiple technical replicates per biological condition (individual) in order to directly and explicitly estimate the batch effect associated with independent C1 preparations.

As a first step in exploring batch effects in our study, we examined the gene expression profiles across all single cells that passed our quality checks (as reported above) using raw molecule counts (without standardization). 
Using principal component analysis (PCA) for visualization purpose, we observed – as expected - that the major source of variation in data from single cells is the individual origin of the sample (Figure. 4A). 
Specifically, we found that the proportion of variance due to individual was larger (median: 20%) than variance due to C1 batch (median: 5%, Chi-squared test; *p*-value < 0.001, Sup. Figure. 4; see Methods). 
Yet, variation due to C1 batch is also substantial - data from single cell samples within a batch are more correlated than that from single cells from the same individual but different batches (Kruskal-Wallis test, Chi-squared; *p* < 0.001).

Could we account for batch effects using the ERCC spike-in controls? 
In theory, if the total ERCC molecule-counts are affected only by technical variability, the spike-ins could be used to correct for batch effects even in a study design that entirely confounds biological samples with C1 preparations. 
To examine this, we considered the relationship between total ERCC molecule-counts and total endogenous molecule-counts per sample. 
If only technical variability affects ERCC counts, we expect the magnitude of variance in the spike-ins expression profiles to be consistent across batches and individuals, in contrast to the pattern observed for endogenous molecule counts. 
In practice, however, total ERCC molecule-counts were significantly different across C1 batches (F-test; *p*-value < 0.001), yet not across individuals (LRT; *p*-value = 0.078;  Sup. Figure. 4C). 
In addition, while we observed a positive relationship between total ERCC molecule-counts and total endogenous molecule-counts per sample, this correlation pattern differed across C1 batches as well (F-test; *p*-value < 0.001; Sup. Figure. 4).
To more carefully examine the technical and biological variance of ERCC spike-in controls, we assessed the ERCC per-gene expression profile.
Perhaps even more importantly, when we considered ERCC gene expression levels, we found that the proportion of variance explained by the individual effect was significantly larger than the variance due to C1 batch (median: 23% vs. 7%, Chi-squared test; *p*-value < 0.001).
Moreover, ERCC gene expression data from samples from the same batch are more correlated than data from samples across batches (Kruskal-Wallis test, Chi-squared *p* < 0.001).
We thus conclude that ERCC spike-in controls cannot be used to effectively account for the batch effect associated with independent C1 preparations.

We explored potential reasons for the observed batch effects, and in particular, the difference in ERCC counts across batches and individuals.  
To do so, we focused on the read-to-molecule conversion rates, namely the rates at which we convert read to molecule counts based on the UMI sequences.  
We defined read-to-molecule conversion efficiency as the total molecule-counts divided by the total reads-counts in each sample, considering separately the reads/molecules that correspond to endogenous genes or ERCC spike-ins. 
In contrast to the common intuition, we observed a batch effect arising from the read-to-molecule conversion efficiency of both ERCC (F-test; *p* < 0.05) and endogenous genes (F-test; *p* < 0.001) across C1 replicates from the same individual (Figure. 3). 
We reason that the difference in read to molecule conversion efficiency across C1 preparations may contribute to the observed batch effect in this platform. 

### Measuring regulatory noise in single-cell gene expression
Our analysis indicated a considerable batch effect in the single cell gene expression data. 
We thus sought an analysis approach that will take the batch effect into account and allow us to characterize biological differences in variation of single cell gene expression levels across individuals (Figure. 4). 
As a first step, we adjusted the raw molecule counts by using a Poisson approximation to account for the random use of identical UMI sequences in molecules from highly expressed genes (this was previously termed a correction for the UMI ‘collision probability’ [@Fu2011]).
We then excluded data from genes whose inferred molecule count exceeded 1024 (the theoretical number of UMI sequences) – this step resulted in the exclusion of data from 6 genes, all mitochondrial. 

We next incorporated a standardization step by computing log transformed counts-per-million (cpm) to remove the effect of different sequencing depths, as is the common practice for the analysis of bulk RNA-seq data (Figures 4A and 4B). 
We used the framework of a Poisson GLM (generalized linear model) to normalizate the endogenous molecule counts by the observed molecule counts of ERCC spike-ins across samples (see Methods). 
While we do not expect this step to account for the batch effect (as discussed above), we reasoned that the spike-ins allow us to effectively normalize molecule counts across individuals (Figure 4C).

Finally, to account for the technical batch effect, we modeled between-sample correlations in gene expression within C1 replicates (see Methods). 
Our approach is similar in principle to a method that was initially developed for adjusting within-replicate correlations in microarray data [@Smyth2005].
Essentially, we assumed that samples within each C1 replicate share a component of technical variation, which is independent of biological variation across individuals. We fit a linear mixed model for each gene, which includes a fixed effect for individual and a random effect for batch. 
The batch effect is specific to each C1 replicate, and is independent of biological variation across individuals.
We fit a linear mixed model for each gene that includes a fixed effect of individuals and a random effect for within-batch variation attributed to cell-to-cell correlation in each C1 replicate.
We then removed the batch effect and found from the corrected data that the samples from different C1 replicates of the same individual largely overlapped (Figure 4D).

Once we removed the unwanted technical variability, we next focused on analyzing cell-to-cell variation in gene expression.
The term "gene expression noise" - the measured level of variation in gene expression among single cells in isogenic populations [@Raser2005], is generally defined as the coefficient of variation (CV) of the bio-molecule levels, either protein or mRNA [@Fehrmann2013].
In the following, we used the standardized, normalized, batch-corrected molecule count data to quantitatively measure the gene expression noise in clonal iPSCs. 
We adjusted our CV values by gene abundance averaged across single cells (see the Methods section) and used the adjusted CV as the measurements of gene expression noise.

An important consideration when analyzing scRNA-seq data is the problem of genes possibly being missed during sequencing (referred to as gene “dropout”) due do low mRNA abundance as well as low mRNA capture efficiency [@Brennecke2013].
To investigate the effects of the dropouts in our analyses, we looked at the proportion of undetected cells, or the dropout rate, of each gene and its association with gene abundance and gene expression noise.
We found that the proportion of undetected cells varied across genes, with the median ranged between 21 to 30 percent in the three individuals.
Significant individual difference in the proportion of undetected cells were found in more than 10 percent of the endogenous genes (1,958 genes).
As expected, gene abundance and variance of the expressed cells were associated with the proportion of undetected cells (Sup. Figure. 9A and 9B, respectively). 
However, importantly, the adjusted CV was not associated with proportion of undetected cells (Spearman's correlation = 0.04) across the three individuals (Sup. Figure. 9C), supporting the idea that our adjusted CV measurement is not confounded with the dynamic range of gene abundance.

We measured the expression noise and gene abundance with or without the single cell samples where a given gene was not detected/expressed (all cells in Figure. 5A and expressed cells in Figure. 5B)[@Shalek2013].
We examined the presence of differential gene expression noise across individuals by looking at the top 1,000 genes rank by expression noise and by abundance.
While including only the expressed cells - cells that were quantified by one or more molecule counts of the given gene, we found that 887 genes were ranked among the top 1,000 highly expressed genes in all three individuals (Figure. 5C), whereas only 103 genes were ranked among the top 1,000 noisy genes (Figure. 5D).
Similar results were found when all single cells were included that the three individuals shared 891 highly expressed genes and only 120 highly noisy genes (data not shown).

### Identification of genes with differential gene expression noise across individuals

Next, we identified differential noisy genes across individuals using the median of absolute deviation (MAD) to quantify the degree of dissimilarity between individual noise measures. 
While only the expressed cells were included for computing individual CVs adjusted by gene abundance, our permutation-based test indicated that 330 genes had statistically significant differences in gene expression noise between individuals, at an alpha-level of $10^{-5}$ (Sup Table. 2 of gene list, Sup. Figure. 10). 
Enrichment analysis suggested that these genes were enriched for genes involved in protein translation, protein disassembly, and various biosynthetic processes (Sup Table. 2). 
Interestingly, among the differential noisy genes that we identified, we found two highly expressed pluripotency genes *DNMT3B* and *NR6A1* among the differential noisy genes, both of which were detected in all single cells (Sup. Figure. 10), whereas the majority of the pluripotency genes were not detected in all the cells and exhibited no differential noisy pattern, including key pluripotency genes *Nanog* and *ZFP42* (Sup. Figure. 11), suggesting that the pluripotent state was tightly regulated across individuals.

Furthermore, the transcriptional noise variation pattern was characterized by a few features, such as proportion and expression variability of the expressed cells, and also the proportion of undetected cells (dropout rate, Figure. 6), despite the association between gene abundance and proportion of undetected cells (Figure. 5A and 5B). 
For instance, *TFPI* and *Nanog* in individual NA19101 exhibited a higher number of expressed cells and also a wider range of expression among the expressed cells compared to the other two individuals, yet the difference of adjusted CV of *Nanog* was not significant across individuals (*p* = 0.80).
On the other hand, *SLC25A16* exhibited a similar range of variation in the expressed cells across the three individuals, but the number of expressed cells was different, and the difference of adjusted CV was significant (*p* < 0.001). 

