---
bibliography: refs.bib
---

```{r chunk-options, include=FALSE}
source("chunk-options-paper.R")
```

## Results
### Study design and data collection
Single-cell transcriptomics has recently emerged as a powerful technology to explore stochastic gene expression by assaying steady state transcriptomes from numerous individual cells with minimum losses in cDNA synthesis and high reproducibility.

To overcome the challenge of efficiency and uniformity with which each mRNA molecule is copied and sequenced in this study, 
randomly labeling individual transcripts with UMI barcode consisting of five random nucleotides was incorparated during in vitro reverse transcription prior to the amplification of mRNA [@Islam2014; @Grun2014; @Jaitin2014].
 

 including the spike-in quantification standards of known abundance in the samples allows estimation of technical variation to achieve better quantification. 


### ERCC spike-ins and Batch effect

### Subsampling

### A novel pipeline 

### Cell-cell variability of gene expression

1. study design and data collection (Fig.1, sup fig.1 and 2)
2. cpm (Fig.2, sup fig.3 and 4)
3. subsample (fig.3)
4. remove batch effect (fig.4 and sup fig.5)
5. gene expression noise (fig.5 and sup fig.6)
