---
bibliography: refs.bib
---

```{r chunk-options, include=FALSE}
source("chunk-options-paper.R")
```

## Results
### Study design and quality control
We collected single cell RNA-seq (scRNA-seq) data from three YRI iPSC lines using the Fluidigm C1 microfluidic system followed by sequencing on the Illumina HiSeq system (see methods). 
We added ERCC spike-in controls to each sample, and used 5-bp random sequence UMIs to allow for the direct quantification of mRNA molecule numbers. 
For each of the YRI lines, we performed three independent C1 collections.
This study design (Figure 1A) allows us to estimate error and noise associated with the technical processing of the samples, independently from the biological variation between individuals.  

The three C1 replicates of each individual were collected on different dates using cell cultures of different passage numbers (Sup Table 1A). 
In that sense, the term ‘technical replicates’ is somewhat misleading, because we used biological replicates of the same cell line.
Each replicate of C1 collection was accompanied by a collection of matching bulk sample using the same reagents. 
This property allowed us to globally characterize the regulatory variation between replicates of the same cell culture, as well as to estimate how well scRNA-seq data can recapitulate the RNA-seq results from population bulk samples. 

In what follows, we describe data as originating from different samples when we refer to data from distinct wells of each C1 collection. 
Generally, data from each sample correspond to data from a single cell. 
In addition, we describe data as originating from different replicates when we refer to all samples from a given C1 collection, and from different individuals when we refer to data from all replicates of a given genetically distinct iPSC line.

```{r calculate-seq-depth}
library("dplyr")
total_counts <- read.table("../data/total-counts.txt", header = TRUE,
                           sep = "\t", stringsAsFactors = FALSE)
depth_stats <- total_counts %>%
  filter(stage == "fastq", sickle == "not-quality-trimmed",
         well != "bulk") %>%
  group_by(individual, batch, well) %>%
  summarize(counts_per_sample = sum(counts)) %>%
  ungroup() %>%
  summarize(mean = mean(counts_per_sample), sd = sd(counts_per_sample),
            min = min(counts_per_sample), max = max(counts_per_sample))
```


We obtained an average of `r format(depth_stats$mean / 10^6, digits = 3)` +/-
`r format(depth_stats$sd / 10^6, digits = 3)` million sequencing reads per sample (range `r format(depth_stats$min / 10^6, digits = 3)`-`r format(depth_stats$max / 10^6, digits = 3)` million reads).
We initially processed the sequencing reads using a standard alignment approach (see methods). 
Given the low abundance of mRNA content and the high vulnerability of single cells and the low amount of starting material, scRNA-seq can be highly affected by various technical artifacts arising from cell isolation, mRNA capture efficiency, library preparation, and sequencing procedures. 
We thus performed multiple quality control analyses.

As a first step, we estimated the proportion of ERCC spike in reads from each sample. 
We found that, across samples, sequencing reads from the second replicate of individual 19098 include unusually high ERCC content compared to all other samples and replicates (Sup Figure. 2A and B). 
We concluded that a pipetting error led to excess ERCC content in this replicate and we excluded the data from all samples of this replicate in subsequent analyses.

With the exception of the excluded samples, data from all other replicates seem to have similar global properties (using general metrics; Sup figure 2X). 
We therefore proceeded by focusing on the assumption that data from each sample correspond to data from an individual cell. 
Before the processing of the individual cell samples, we performed visual inspection on the C1 microfluidic plates (see methods). 
Based on that visual inspection, we have excluded sequencing data from 21 samples that did not contain any cell, and 54 samples that contained more than one cell (across all replicates). 
We then filtered data from the remaining samples based on the the number of total mapped reads, the percentage of unmapped reads, the percentage of ERCC spike-in reads, and the number of genes detected (Sup Figure 2).
We used arbitrary inclusion cutoffs for each metric, based on the 95th percentile of the distribution for the 21 libraries that were amplified from samples that did not include a cell based on visual inspection (Figure. 1B-E and Sup Figure. 2C-F). 

Visual inspection of the C1 microfluidic plate is an important quality control step, but it is not infallible.
We thus identified and removed data from 15 additional samples that were classified as originating from a single cell based on visual inspection, but whose data were more consistent with a multiple-cell origin based on the number of total molecules, the concentration of cDNA amplicons, and the read-to-molecule conversion efficiency (defined as the number of total molecules divided by the number of total reads; Sup Figure 3). 
At the conclusion of these quality control analyses and exclusion steps, we retained data from 564 high quality samples, which correspond, with reasonable confidence, to 564 single cells, across eight replicates from three individuals (Table 1).

Our final quality check focused on the different properties of sequencing read and molecule count data.
We considered data from the 564 high quality samples and compared gene specific counts of sequencing read and molecules.
We found that while gene-specific read and molecule counts are exceptionally highly correlated when we considered the ERCC spike-in data (r = 0.99; Figure. 1G), these counts are somewhat less correlated when data from the endogenous genes are considered (r = 0.92; Figure 1F). 
Moreover, the gene-specific read and molecule counts correlation is noticeably lower for genes that are expressed at lower levels. These observations concur with pervious studies [@Brennecke2013] as they underscore the importance of using UMIs in single cell gene expression studies.

### Is there a need to standardize molecule-counts?  
Standard analysis of single cell gene expression data often relies on the implicit assumption that molecule-count data, collected by using single cell UMI-based approaches, can be thought of as an absolute estimate of the total number of RNA molecules in the cell [@Islam2014]. 
In partial support of this notion, it has been shown that saturation of the number of genes whose expression is detected in single cells can be reached with roughly 2 million sequencing reads [@Wu2014], and that, at this depth of sequencing, mean gene expression levels in bulk samples can be recapitulated with data from as few as 10 individual cells [@Pollen2014]. 
Our study design, which provides the infomation of molecule counts, allows us to carefully examine these properties.
To estimate the number of molecule counts required for representing the bulk samples and also for capturing the cell-cell variation in gene expression, we conducted a systematic subsampling at the levels of sequencing depth, including both read and molecule counts, along with the consideration of cell numbers using our high quality samples (Figure. 2).
The results showed that sequencing depth of 2 million reads, which corresponds to ~100,000 molecule counts, with 75 single cell samples is sufficient to capture the majority of genes expressed (~90 %, Figure. 2A), as well as to accurately reflect mean expression levels in the whole cell culture (r > 0.93, Figure. 2B). 
Using the same subsampling cutoffs, we found that the cell-cell variance also accurately represents all the collected single cell samples (r > 0.95, Figure. 2C).


We also collected scRNA-seq data from Lymphoblastoid Cell Lines (LCLs) in a trial run, in which libraries from one single cell was sequenced individually on one sequencing lane (Sup Figure. 4). 
Since all the library materials were used in these samples, the data collected from these single cell libraries were possibly the highest sequencing depth that we could obtain using this protocol. 
The subsampling results from these libraries showed that, with the sequencing depth of 50 million reads, the numbers of endogenous genes started to plateau (Sup Figure. 4A and B), whereas the total molecule counts were still increasing (Sup Figure. 4D and F), suggesting that we have not yet reach the saturation point of molecule counts.

we found that the total number of molecules per sample was dependent on the sequencing depth (Figure. 3A and C) and argued that standardization of molecule counts is required. 
In addition, we noticed an unexpected batch effect arising from the difference of read-to-molecule conversion efficiency, which is computed from the total molecule counts divided by the total reads counts of endogenous genes in each cell, across C1 replicates (Figure. 2B and D). 
Intriguingly, not only the difference of conversion efficiency across replicates was significant, the difference between the three individuals was also significant, which further strengthens our conclusion that differences in certain unknown factors across these individuals are affecting the molecule counts of both ERCC spike-ins and endogenous genes in single cells.


### Is there a batch effect associated with single cell gene expression data?
It is reasonable to assume that batch effects impact single cell gene expression data. 
In the context of the C1 system, however, the typical approach to collect data from each biological condition by using a single C1 plate, makes it nearly impossible to explicitly model and estimate the effect associated with independent capturing and sequencing of each C1 batch. 
In contrast, our study design allows us to directly and explicitly estimate the batch effect associated with independent C1 preparations.

As a first step in exploring noise in the single cell gene expression data we considered the correlation between sample concentrations (measured at the library stage; see Methods) and total molecule-counts, across all detected genes. 
Part of the motivation for this analysis is the common notion that differences in total molecule-counts across cells may indicate differences in the amount of RNA between cells, which generally correspond to differences in cell size (when all cells are of the same type [@Padovan-Merhar2015]). 
Indeed, we found a general correlation between total molecule-counts in single cells and the measured sample concentration (Figure X).
Yet, we also observed that the relationship between total molecule-counts and concentration is significantly different across C1 collection replicates from the same individual (range of correlation between X and Y; TEST; P < XX), and more so across single cell collections from different individuals (range of correlation between X and Y; Test; P < XX; Figure X).

We investigated this property further by classifying the cell cycle phase of single cells based on their gene expression profiles [@Macosko2015]. 
We then considered the association between cell cycle phase and total molecule-counts. 
It is expected that cells that are in G2 and G2/M phases would have higher total molecule-counts [@Vallejos2015], which is indeed our general observation. 
However, variation in this pattern can clearly be seen across single cell collections from different individuals, as well as across different C1 collection batches from the same individual (Figure X). 
Put together, the analyses of the association of total molecule-counts provides the first indication that technical effects significantly impact the results of single cell gene expression data. 


### A novel normalization for scRNA-seq data analysis 
To remove the technical variability arose from all the factors mentioned above without explicitly estimating the technical variability of ERCC spike-in molecule counts, here, we developed a pipeline that is applicable for studying cell-to-cell variation of gene expression from multiple biological conditions with multiple C1 collections (Figure. 4). 

#### Correct for collision probability
Due to the stochasticity of the UMI sampling process, not all molecules will be tagged with an unique UMI and sequenced. 
Thus, after removing genes with lower expression level in high quality samples, we first computed the molecule number of each endogenous gene by applying a mathematical correction to account for the effect of random counting statistics termed collision probability [@Fu2011]. 
This step also removed the genes with molecule counts larger than 1024, which in our case were all mitochondrial genes (data not shown). 

#### Standardization (counts-per-million)
Next, we incorporated a standardization step by computing log transformed counts-per-million (cpm) of molecule counts.  
As predicted from our previous observation of the sequencing depth (Figure. 2A and C), we found that the total read counts per sample were highly correlated with not only the total molecule counts but also with the first principal component of endogenous gene expression (Figure. 4A). 
After removal of the variance caused by sequencing depth, the difference between C1 replicates, or the batch effect, became dominant (Figure. 4B).

#### Poission regression of ERCC counts
To account for the cell-to-cell technical variability coming from the differences of mRNA capture efficiency across samples within the C1 collection, we next performed Poisson GLM (generalized linear model) normalization using the observed and the expected molecule counts of ERCC spike-ins, without correction for collision probability and also without standardization, in each sample.
The PCA of gene expression after normalization showed that PC1 and PC2 separated not only the three individuals but also the three C1 replicates of each individual, suggesting that the batch effect from the C1 replicates were yet to be removed (Figure. 4C).

#### Mixed model for batch effect correction
Not surprisingly, there was a observable batch effect of C1 collection according to the PCA results of gene expression even after both standardization and normalization. 
To directly remove unwanted technical variation between C1 plates, we modeled between well correlation in gene expression within C1 plates. 
We assumed that C1 plates share similar technical variation due to gene expression similarity between cells on the same plate. 
The results of PCA on the corrected data showed that the samples from different C1 replicates of the same individual largely overlapped (Figure. 4D), indicating that we have successfully removed large proportion of the unwanted technical variation across C1 replicates. 

### Cell-to-cell variation of gene expression
Once we removed the unwanted technical variability from scRNA-seq data as described above, we next focused on analyzing cell-to-cell variation of gene expression at the transcriptional level.
We first investigated the biological gene expression variability, or gene expression noise, across single cells of each individual, and then we identified the genes that present differential expression noise across the three individuals.

#### Gene expression noise at transcriptional level
Examining endogenous transcriptional variability in iPSCs as a function of RNA abundance across C1 replicates, we found that coefficient of variation (CV) of molecule counts among endogenous genes and ERCC spike-in genes suggested similar expression variability patterns that track the expected Poisson distribution (Sup Figure. 8A), which are consistent with previous studies of mouse ESCs (embryonic stem cells) [@Islam2014; @Brennecke2013]. 
Due to the observation that transcriptional variability was correlated with mean gene abundance, we adjusted our gene variability measure (CV) by gene abundance averaged across single cells (see the Methods section, Sup Figure. 8B), a similar method used to investigate gene expression noise in mouse ESCs [@Kolodziejczyk2015]. 
To investigate the effect of the unobserved measurements, or the dropouts, we separately measured the CVs with or without the cells where a given gene was not detected/expressed [@Shalek2013]. 
While including only the expressed cells - cells that were quantified by one or more molecule counts of the given gene, we found that 887 genes were ranked among the top 1,000 highly expressed genes in all three individuals (Figure. 5A), whereas 103 genes were ranked among the top 1,000 noisy genes (Figure. 5B).
Similar results were found when all quality single cells were included that the three individuals shared 891 highly expressed genes and 120 highly noisy genes (data not shown), indicating the presence of differential cell-to-cell variation in gene expression across individuals.

We also looked at the proportion of undetected cells, or the dropout rate, of each gene and its association with gene abundance and gene expression noise (Figure. 5C and 5D).
We found that the proportion of undetected cells varied between genes, with the median ranged between 21 to 30 percent in the three individuals.
Significant individual difference in the proportion of undetected genes were found in more than 10 percent of the genes (1,958 genes).
As expected, gene abundance and variance of the expressed cells were associated with the proportion of undetected cells (Sup. Figure. 9A and 9B, respectively). 
However, importantly, the variability measure CV was not associated with proportion of undetected cells (Spearman's correlation = .04) across the three individuals (Sup. Figure. 9C), supporting the idea that CV is not confounded with the dynamic range of gene abundance. 

#### Differential gene expression noise across individuals 
Next, we identified differential noisy genes across individuals using the median of absolute deviation (MAD) to quantify the degree of dissimilarity between individual noise measures. 
While only the expressed cells were included for computing individual CVs adjusted by gene abundance, our permutation-based test indicated that 330 genes had statistically significant differences in gene expression noise between individuals, at an alpha-level of $10^{-5}$ (Sup Table. X of gene list, Sup. Figure. 10). 
Enrichment analysis suggested that these genes were enriched for genes involved in protein translation, protein disassembly, and various biosynthetic processes (Sup Table. X). 
We found two highly expressed pluripotency genes *DNMT3B* and *NR6A1* among the differential noisy genes, both of which were detected in all single cells (Sup. Figure. 10), whereas the majority of the pluripotency genes were not detected in all the cells and exhibited no differential noisy pattern, including key pluripotency genes *Nanog* and *ZFP42* (Sup. Figure. 11), suggesting that the pluripotent state was tightly regulated.
Furthermore, the transcriptional noise variation exhibited a few interesting patterns across individuals, such as proportion and expression variability of the expressed cells, and also the proportion of undetected cells (dropout rate, Figure. 6).
For instance, *TFPI* and *Nanog* in individual 19101 exhibited a higher number of expressed cells and also a wider range of expression among the expressed cells compared to the other two individuals, yet the difference of adjusted CV of *Nanog* was not significant across individuals (*p* = 0.80).
On the other hand, *SLC25A16* exhibited a similar range of variation in the expressed cells across the three individuals, but the number of expressed cells was different, and the difference of adjusted CV was significant. 
Most importantly, despite the association between gene abundance and proportion of undetected genes (Figure. 5D), we found no associations between the individual differences, and similarly for individual differences in expression noise and proportion of undetected genes (Figure. 6).


