---
bibliography: refs.bib
---

```{r chunk-options, include=FALSE}
source("chunk-options-paper.R")
```

## Results
### Study design and data quality control
Single-cell transcriptomics has recently emerged as a powerful technology to explore stochastic gene expression by assaying steady state transcriptomes from numerous individual cells with minimum losses in cDNA synthesis and high reproducibility.
Specifically, randomly labeling individual transcripts with UMIs at the reverse transcription step prior to amplification of mRNA provides the power to measure absolute molecule counts per gene by eliminating the amplification bias [@Islam2014; @Grun2014; @Jaitin2014], whereas including the spike-in quantification standards of known abundance in the samples allows estimation of technical variation to achieve better quantification [@Brennecke2013; @Grun2014; @Ding2015; @Vallejos2015].
In this study, we collected single cell RNA-seq (scRNA-seq) data from three Yoruba iPSC lines (19098, 19101, and 19239), with the additions of both 5-bp random sequence UMIs and ERCC spike-ins, using C1 microfluidic plates from the Fluidigm system followed by sequencing on the Illumina HiSeq system.
Importantly, for each of the Yoruba lines, we performed three independent C1 collections to more accurately capture the characteristics of technical variabilities across C1 replicates and then to further remove them from downstream analysis.
To estimate how well the scRNA-seq can reassemble the RNA-seq results from population bulk samples, each individual C1 collection was accompanied by a collection of bulk sample using the same reagents.

Given the low abundance of mRNA content and the high vulnerability of single cells compared to bulk samples, scRNA-seq can be intensively affected by various technical artifacts arising from cell isolation and capture, mRNA capture efficiency, library preparation, and even sequencing procedures.
As a result, quality control of scRNA-seq libraries was performed in multiple steps to generate reliable analysis results.
After removing the samples that were not one-cell libraries as reported by the visual inspection, we first filtered out the one-cell libraries that were similar to no-cell libraries according the number of total mapped reads, the percentage of unmapped reads, the percentage of ERCC spike-in reads, and the number of genes detected. 
Next, we identified several one-cell libraries that had high possibility to be multiple-cell libraries based on the number of total molecules, the concentration of cDNA amplicons, and the read-to-molecule conversion efficiency defined as the number of total molecules devided by the number of total reads.
Besides carrying out quality control at the library level, we also executed quality control at gene detection level to deal with the problem of genes being missed during sequencing (referred to as gene “dropout”) by filtering out genes with low molecule numbers 
Judging by the proportion of genes detected in good quality cell before and after the filtering, we are confident that the removed genes were truly the dropouts.

To ensure that 

### ERCC spike-ins and Batch effect

### Subsampling

### A novel pipeline 

### Cell-cell variability of gene expression

1. study design and data collection (Fig. `r fig_main_qc`, sup fig.1 and 2)
2. cpm (Fig. `r fig_main_cpm`, sup fig.3 and 4)
3. subsample (Fig. `r fig_main_subsample`)
4. remove batch effect (Fig. `r fig_main_normalization` and sup fig.5)
5. gene expression noise (Fig. `r fig_main_noise` and sup fig.6)
