---
title: Supplementary Information
bibliography: refs.bib
---

```{r chunk-options, include=FALSE}
source("chunk-options-paper.R")
```

## Figures

### Figure `r table_supp_collection`

![Sequencing study design.](../analysis/figure/material-and-method.Rmd/seq.png)

**Figure `r table_supp_collection`.**
The master mixes were sequenced across four flow cells.
"Bulk" is a master mix of the bulk samples from each of the nine batches (3 individuals x 3 replicates).
The other master mixes contain the libraries for the 96 single cells from one Fluidigm C1 chip.
The master mixes for the three individuals were partitioned across the four flow cells to avoid introducing bias.

## Tables

## Supplementary table `r table_supp_genotypes`

We were able to successfully determine the correct identity of each single cell sample by examining the SNPs present in their RNA sequencing reads.
Specifically, we used the method verifyBamID (https://github.com/statgen/verifyBamID) developed by Jun et al., 2012 [@Jun2012], which detects sample contamination and/or mislabeling by comparing the polymorphisms observed in the sequencing reads for a sample to the genotypes of all individuals in a study.
For our test, we included the genotypes for all 120 Yoruba individuals that are included in the International HapMap Project.
The genotypes included the HapMap SNPs with the 1000 Genomes SNPs imputed, as previously described [@McVicker2013].
We subset to include only the 528,289 SNPs that overlap Ensembl protein-coding genes.
verifyBamID used only 311,848 SNPs which passed its default thresholds (greater than 1% minor allele frequency and greater than 50% call rate).
Using the option --best to return the best matching individual, we obtained 100% accuracy identifying the single cells of all three individuals.

Cell line  | Correct | Incorrect
------------- | ------------- | -------------
NA19098  | 288 | 0
NA19101  | 288 | 0
NA19239  | 288 | 0

## References
