
```{r chunk-options, include=FALSE}
source("chunk-options-paper.R")
```

Po-Yuan Tung^1,7^,
John D. Blischak^1,2,7^,
Chiaowen Hsiao^1,7^,
David A. Knowles^3,4^,
Jonathan Burnett^1^,
Jonathan K. Pritchard^5,6^\*,
Yoav Gilad^1^\*

^1^Department of Human Genetics, University of Chicago, Chicago, Illinois, USA  
^2^Committee on Genetics, Genomics, and Systems Biology, University of Chicago, Chicago, Illinois, USA  
^3^Department of Computer Science, Stanford University, Stanford, CA, USA  
^4^Department of Radiology, Stanford University, Stanford, CA, USA  
^5^Department of Biology, Stanford University, Stanford, CA, USA  
^6^Howard Hughes Medical Institute, Stanford University, CA, USA  
^7^These authors contributed equally to this work

*Correspondence should be addressed to YG (gilad@uchicago.edu) or JKP (pritch@standford.edu).

## Abstract

Single cell RNA sequencing (scRNA-seq) provides insights into the gene regulatory processes that are unattainable in RNA-seq of bulk cell populations, e.g. identification of novel cell types and the cell-cell variability in response to external signals.
This technology also has the potential to expand our understanding of the genetic control of gene regulatory processes.
Our lab and others have identified common genetic variants (specifically single nucleotide polymorphisms, or SNPs) in humans that are associated with gene expression levels (referred to as eQTLs) and other gene regulatory features.
These studies have all focused on the mean expression level of genes in each individual, completely ignoring the variation from cell to cell, commonly referred to as the gene expression noise.
Because common SNPs often have small effect sizes, it is crucial to account for technical sources of variation introduced during the sample processing to avoid dwarfing the true biological signal, or worse creating spurious signal.
To investigate the sources of technical variation introduced during scRNA-seq, we collected three replicates of the C1 Fluidigm chip from three human induced pluripotent stem cell (iPSC) lines.
Furthermore, we added unique molecular identifiers (UMIs) to each RNA molecule to reduce the bias inherent in amplifying and sequencing a small starting amount of RNA.
As expected, we observed that the major source of variation in the gene expression data was driven by biological variation (i.e. the differences between the three individuals), but of course there was also substantial technical variation between the replicates of each individual.
Surprisingly, we also observed that the conversion of reads to molecules using the UMIs was impacted by both biological and technical variation, indicating that counting UMIs cannot be assumed to be an unbiased process.
After accounting for these technical sources of variation using a linear mixed model framework, we identified 330 genes whose expression noise was different across the three individuals.
From these results, we suggest improvements for future studies investigating interindividual differences in gene expression noise.
