
```{r chunk-options, include=FALSE}
source("chunk-options-paper.R")
```

Po-Yuan Tung^1,8^,
John D. Blischak^1,2,8^,
Chiaowen Joyce Hsiao^1,8^,
David A. Knowles^3,4^,
Jonathan Burnett^1^,
Jonathan K. Pritchard^3,5,6^,
Yoav Gilad^1,7^\*

^1^Department of Human Genetics, University of Chicago, Chicago, Illinois, USA  
^2^Committee on Genetics, Genomics, and Systems Biology, University of Chicago, Chicago, Illinois, USA  
^3^Department of Genetics, Stanford University, Stanford, CA, USA  
^4^Department of Radiology, Stanford University, Stanford, CA, USA  
^5^Department of Biology, Stanford University, Stanford, CA, USA  
^6^Howard Hughes Medical Institute, Stanford University, CA, USA  
^7^Department of Medicine, University of Chicago, Chicago, Illinois, USA  
^8^These authors contributed equally to this work

*Correspondence should be addressed to YG (gilad@uchicago.edu).

## Abstract

Single cell RNA sequencing (scRNA-seq) can be used to characterize variation in gene expression levels at high resolution.
This technology enables the identification of novel cell types and the characterization of cell-to-cell variability.
However, the sources of experimental noise in scRNA-seq are not yet well understood.
To investigate the technical variation introduced using the Fluidigm C1 system, a widely-used scRNA-seq platform, we collected three replicates from three human induced pluripotent stem cell (iPSC) lines.
We added unique molecular identifiers (UMIs) to reduce the bias inherent in amplifying and sequencing a small starting amount of RNA.
We found that the major source of variation in the gene expression data was driven by biological variation (i.e. the differences between the three individuals), but we also observed substantial technical variation between the replicates of each individual.
In contrast to standard assumptions, we observed that the conversion of reads to molecules using the UMIs was impacted by both biological and technical variation, indicating that UMI counts are not an unbiased estimator of gene expression levels.
After accounting for these technical sources of variation using a linear mixed model framework, we identified 330 genes whose variation in single cell expression levels were different across the three individuals.
Based on our observations, we suggest a framework for effective scRNA-seq studies.
