
```{r chunk-options, include=FALSE}
source("chunk-options-paper.R")
```

Po-Yuan Tung^1,8^,
John D. Blischak^1,2,8^,
Chiaowen Joyce Hsiao^1,8^,
David A. Knowles^3,4^,
Jonathan E. Burnett^1^,
Jonathan K. Pritchard^3,5,6^,
Yoav Gilad^1,7^\*

^1^Department of Human Genetics, University of Chicago, Chicago, Illinois, USA  
^2^Committee on Genetics, Genomics, and Systems Biology, University of Chicago, Chicago, Illinois, USA  
^3^Department of Genetics, Stanford University, Stanford, CA, USA  
^4^Department of Radiology, Stanford University, Stanford, CA, USA  
^5^Department of Biology, Stanford University, Stanford, CA, USA  
^6^Howard Hughes Medical Institute, Stanford University, CA, USA  
^7^Department of Medicine, University of Chicago, Chicago, Illinois, USA  
^8^These authors contributed equally to this work

*Correspondence should be addressed to YG (gilad@uchicago.edu).

## Abstract

Single-cell RNA sequencing (scRNA-seq) can be used to characterize variation in gene expression levels at high resolution.
However, the sources of experimental noise in scRNA-seq are not yet well understood.
We investigated the technical variation associated with sample processing using the single-cell Fluidigm C1 platform.
To do so, we processed three C1 replicates from three human induced pluripotent stem cell (iPSC) lines.
We added unique molecular identifiers (UMIs) to all samples, to account for amplification bias.
We found that the major source of variation in the gene expression data was driven by genotype, but we also observed substantial variation between the technical replicates.
We observed that the conversion of reads to molecules using the UMIs was impacted by both biological and technical variation, indicating that UMI counts are not an unbiased estimator of gene expression levels.
Based on our results, we suggest a framework for effective scRNA-seq studies.

## Author Summary

Single-cell RNA sequencing (scRNA-seq) can be used to study variation in gene expression levels at high resolution.
This technology enables the identification of novel cell types and the characterization of cell-to-cell variability in gene regulation.
However, the sources of experimental noise in scRNA-seq are not yet well understood.
In particular, most current single-cell studies use a study design that confounds technical batch with the biological source of the single cells, thereby making it impossible to effectively estimate the noise and error associated with single-cell sample processing.
To investigate the technical variation introduced using the Fluidigm C1 system, a widely-used scRNA-seq platform, we collected three replicates from three human induced pluripotent stem cell (iPSC) lines.
Our nested study design allows us to directly model and estimate the batch effect.
We found that the major source of variation in the gene expression data was driven by biological variation (i.e. the differences between the three individuals), but we also observed substantial technical variation between the technical replicates of each individual.
Once we had accounted for the technical variation, we estimated biological gene regulatory noise - defined as the amount of variation of gene expression levels across single cells â€“ for each individual.
We used this approach to identify hundreds of genes whose variation in single-cell expression levels are different across individuals.
Our work demonstrates the importance of a proper study design for studying gene regulatory noise in single-cell expression data.
We provide an explicit recommendation regarding an effective design for single-cell studies.
